<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>’Variance components and the response to selection</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Selection, sexual conflict and sex-specific parameters in the DGRP</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tomkeaney/DGRP_sexual_conflict">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">’Variance components and the response to selection</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-01
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DGRP_sexual_conflict/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210706code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210706)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210706code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210706)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPsexualconflicttreea4076aa5e76c8729c82465a320f59c221ba09a22targetblanka4076aaa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/tomkeaney/DGRP_sexual_conflict/tree/a4076aa5e76c8729c82465a320f59c221ba09a22" target="_blank">a4076aa</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPsexualconflicttreea4076aa5e76c8729c82465a320f59c221ba09a22targetblanka4076aaa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/tomkeaney/DGRP_sexual_conflict/tree/a4076aa5e76c8729c82465a320f59c221ba09a22" target="_blank">a4076aa</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/.DS_Store
    Ignored:    gwas_data/

Untracked files:
    Untracked:  %
    Untracked:  -Toms-laptop.Rhistory
    Untracked:  Manuscript/
    Untracked:  Reported_heritability.xlsx
    Untracked:  Rplot.pdf
    Untracked:  S_plots.pdf
    Untracked:  Selection_differential_plot.pdf
    Untracked:  Selection_differentials.pdf
    Untracked:  Selection_differentials_f.pdf
    Untracked:  Selection_differentials_m.pdf
    Untracked:  analysis/Random_plots.R
    Untracked:  code/LICENSE
    Untracked:  code/gcta64
    Untracked:  code/prettify
    Untracked:  code/toy.map
    Untracked:  code/toy.ped
    Untracked:  data/SNP_heritability.csv
    Untracked:  data/SNP_heritability_cleaned.csv
    Untracked:  data/SNP_selection.csv
    Untracked:  data/S_female.csv
    Untracked:  data/all_SNP_data.csv
    Untracked:  data/rfm_dimorphism_model.rds
    Untracked:  dimorphism_subset.pdf
    Untracked:  fitness_variation.pdf
    Untracked:  fits/
    Untracked:  lifetime_fecundity_S.pdf
    Untracked:  mating_latency_S.pdf

Unstaged changes:
    Modified:   .Rprofile
    Modified:   .gitattributes
    Modified:   .gitignore
    Modified:   DGRP_sexual_conflict.Rproj
    Modified:   README.md
    Modified:   _workflowr.yml
    Modified:   analysis/SNP_correlations.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/license.Rmd
    Modified:   analysis/process_mashr.Rmd
    Modified:   analysis/run_GWAS.Rmd
    Modified:   analysis/run_mashr.Rmd
    Modified:   code/README.md
    Modified:   code/run_mashr.R
    Modified:   data/README.md
    Modified:   data/S_male.csv
    Modified:   data/all.dgrp.phenos_unscaled.csv
    Modified:   data/rfm.complete.csv
    Modified:   output/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/GCTA_results.Rmd</code>) and HTML (<code>docs/GCTA_results.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/a4076aa5e76c8729c82465a320f59c221ba09a22/analysis/GCTA_results.Rmd" target="_blank">a4076aa</a>
</td>
<td>
ausevo
</td>
<td>
2022-06-01
</td>
<td>
adding analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/8d22c487891bef0c2bd0999def4812b8605e73c4/docs/GCTA_results.html" target="_blank">8d22c48</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/30bb3361d85525486ef3d248640432867e3e7601/analysis/GCTA_results.Rmd" target="_blank">30bb336</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-16
</td>
<td>
creating a more readable doc
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages-and-results-from-the-gcta-models" class="section level2">
<h2>Load packages and results from the <code>GCTA</code> models</h2>
<pre class="r"><code>library(tidyverse)
library(glue)
library(kableExtra)
library(DT)
library(MetBrewer)
library(patchwork)
library(rcartocolor)
library(ggbeeswarm)
library(brms)
library(tidybayes)

 SNP_heritability &lt;- read_csv(&quot;data/SNP_heritability.csv&quot;)
 SNP_selection_response_estimates &lt;- read_csv(&quot;data/SNP_selection.csv&quot;)</code></pre>
</div>
<div id="which-traits-are-responding-to-selection" class="section level2">
<h2>Which traits are responding to selection?</h2>
<pre class="r"><code>selection_female_plot &lt;-
  SNP_selection_response_estimates %&gt;% 
  filter(Fitness_sex == &quot;Female&quot;,
         `Variance_C(G)_tr12` &lt; 1 &amp; `Variance_C(G)_tr12` &gt; -1,
         Reference != &quot;Jin et al (2020) PLOS Genetics&quot; &amp;
           Reference != &quot;Everett et al (2020) Genome Research&quot; &amp;
           Reference != &quot;Dembeck et al (2015) eLife&quot;) %&gt;% 
  mutate(Trait = fct_reorder(Trait, `Variance_C(G)_tr12`)) %&gt;% 
  
  ggplot(aes(x = `Variance_C(G)_tr12`, y = Trait, xmin = `Variance_C(G)_tr12` - `SE_C(G)_tr12`, xmax = `Variance_C(G)_tr12` + `SE_C(G)_tr12`)) +
  geom_pointinterval(aes(fill = `Variance_C(G)_tr12`), shape = 21, point_size = 1.5, stroke = 1) +
  scale_fill_gradientn(colors=met.brewer(direction = -1, &quot;Hiroshige&quot;), guide = &quot;none&quot;) +
  geom_vline(xintercept = 0, linetype = 2) +
  labs(x = expression(paste(&quot;SNP covariance between fitness and trait &quot;, (R[z]))),
       title = &quot;Females&quot;) +
  theme_minimal()
  

selection_male_plot &lt;-
  SNP_selection_response_estimates %&gt;% 
  filter(Fitness_sex == &quot;Male&quot;,
         `Variance_C(G)_tr12` &lt; 1 &amp; `Variance_C(G)_tr12` &gt; -1,
         Reference != &quot;Jin et al (2020) PLOS Genetics&quot; &amp;
           Reference != &quot;Everett et al (2020) Genome Research&quot; &amp;
           Reference != &quot;Dembeck et al (2015) eLife&quot;) %&gt;% 
  mutate(Trait = fct_reorder(Trait, `Variance_C(G)_tr12`)) %&gt;% 
  
  ggplot(aes(x = `Variance_C(G)_tr12`, y = Trait, xmin = `Variance_C(G)_tr12` - `SE_C(G)_tr12`, xmax = `Variance_C(G)_tr12` + `SE_C(G)_tr12`)) +
  geom_pointinterval(aes(fill = `Variance_C(G)_tr12`), shape = 21, point_size = 1.5, stroke = 1) +
  scale_fill_gradientn(colors=met.brewer(direction = -1, &quot;Hiroshige&quot;), guide = &quot;none&quot;) +
  geom_vline(xintercept = 0, linetype = 2) +
  labs(x = expression(paste(&quot;SNP covariance between fitness and trait &quot;, (R[z]))),
       title = &quot;Males&quot;) +
  theme_minimal()

selection_female_plot + selection_male_plot</code></pre>
<p><img src="figure/GCTA_results.Rmd/unnamed-chunk-2-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>Figure X: Results from the bivariate GREML models run in GCTA. We use these models to estimate the genetic covariance between a given trait and fitness, which is the response to selection. Data are mean point estimates of covariance with standard errors. Note that 1) we have not corrected for multiple testing and 2) I don’t know what the units are here / how to interpret them as a selection response and 3) only a subset of the traits analysed are displayed in this plot</p>
</div>
<div id="is-the-response-to-selection-greater-in-males-than-it-is-in-females-across-traits" class="section level2">
<h2>Is the response to selection greater in males than it is in females across traits?</h2>
<div id="meta-analysis" class="section level3">
<h3>Meta-analysis</h3>
<ul>
<li><p>Should I only consider traits measured in both sexes?</p></li>
<li><p>We need to find an appropriate distribution to model these data</p></li>
</ul>
<pre class="r"><code>met_selection_data &lt;-
  SNP_selection_response_estimates %&gt;% 
  select(Trait, Fitness_sex, `Trait guild`, Reference, `Variance_C(G)_tr12`, `SE_C(G)_tr12`) %&gt;% 
  transmute(Trait = as.factor(Trait),
            Sex = as.factor(Fitness_sex),
            Trait_guild = as.factor(`Trait guild`),
            Reference = as.factor(Reference),
            Selection_response = abs(`Variance_C(G)_tr12`),
            SE = `SE_C(G)_tr12`) %&gt;% 
  filter(SE != 0, Selection_response &lt; 1) # note this filtering removes 8 traits


Selection_model &lt;- 
  brm(Selection_response | se(SE) ~ 0 + Sex + (1|Trait) + (1|Reference),
      family = gaussian(), # this distribution is not appropriate
      data = met_selection_data,
      prior = c(prior(normal(0, 0.2), class = b),
                prior(cauchy(0, 1), class = sd)),
      warmup = 4000, iter = 6000,
      seed = 1, cores = 4, chains = 4,
      control = list(adapt_delta = 0.9),
      file = &quot;fits/Selection.model&quot;)

print(Selection_model)</code></pre>
<pre><code> Family: gaussian 
  Links: mu = identity; sigma = identity 
Formula: Selection_response | se(SE) ~ 0 + Sex + (1 | Trait) + (1 | Reference) 
   Data: met_selection_data (Number of observations: 985) 
  Draws: 4 chains, each with iter = 6000; warmup = 4000; thin = 1;
         total post-warmup draws = 8000

Group-Level Effects: 
~Reference (Number of levels: 92) 
              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(Intercept)     0.01      0.00     0.00     0.02 1.00     4304     4049

~Trait (Number of levels: 686) 
              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(Intercept)     0.00      0.00     0.00     0.01 1.00     6361     4035

Population-Level Effects: 
          Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
SexFemale     0.08      0.01     0.07     0.09 1.00    12382     6211
SexMale       0.08      0.01     0.07     0.10 1.00    13652     5971

Family Specific Parameters: 
      Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sigma     0.00      0.00     0.00     0.00   NA       NA       NA

Draws were sampled using sampling(NUTS). For each parameter, Bulk_ESS
and Tail_ESS are effective sample size measures, and Rhat is the potential
scale reduction factor on split chains (at convergence, Rhat = 1).</code></pre>
<p>Above I mention that a Gaussian model may not be a good fit for our data. Let’s see about that:</p>
<pre class="r"><code>pp_check(Selection_model)</code></pre>
<p><img src="figure/GCTA_results.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/8d22c487891bef0c2bd0999def4812b8605e73c4/docs/figure/GCTA_results.Rmd/unnamed-chunk-4-1.png" target="_blank">8d22c48</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Yep, not great! Need to think about this.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
</div>
<div id="which-traits-exhibit-genetic-variance-explained-by-known-snps" class="section level2">
<h2>Which traits exhibit genetic variance explained by known SNPs?</h2>
<pre class="r"><code>Vg_female_plot &lt;-
  SNP_selection_response_estimates %&gt;% 
  filter(Fitness_sex == &quot;Female&quot;,
         `Variance_V(G)_trait` &lt; 1 &amp; `Variance_V(G)_trait` &gt; -1,
         Reference != &quot;Jin et al (2020) PLOS Genetics&quot; &amp;
           Reference != &quot;Everett et al (2020) Genome Research&quot; &amp;
           Reference != &quot;Dembeck et al (2015) eLife&quot;) %&gt;% 
  mutate(Trait = fct_reorder(Trait, `Variance_V(G)_trait`)) %&gt;% 
  
  ggplot(aes(x = `Variance_V(G)_trait`, y = Trait, xmin = `Variance_V(G)_trait` - `SE_V(G)_trait`, xmax = `Variance_V(G)_trait` + `SE_V(G)_trait`)) +
  geom_pointinterval(aes(fill = `Variance_V(G)_trait`), shape = 21, point_size = 1.5, stroke = 1) +
  scale_fill_gradientn(colors=met.brewer(direction = 1, &quot;OKeeffe2&quot;), guide = &quot;none&quot;) +
  geom_vline(xintercept = 0, linetype = 2) +
  labs(x = expression(paste(&quot;SNP genetic variance&quot;, (V[A]))),
       title = &quot;Females&quot;) +
  theme_minimal()
  

Vg_male_plot &lt;-
  SNP_selection_response_estimates %&gt;% 
  filter(Fitness_sex == &quot;Male&quot;,
         `Variance_V(G)_trait` &lt; 1 &amp; `Variance_V(G)_trait` &gt; -1,
         Reference != &quot;Jin et al (2020) PLOS Genetics&quot; &amp;
           Reference != &quot;Everett et al (2020) Genome Research&quot; &amp;
           Reference != &quot;Dembeck et al (2015) eLife&quot;) %&gt;% 
  mutate(Trait = fct_reorder(Trait, `Variance_V(G)_trait`)) %&gt;% 
  
  ggplot(aes(x = `Variance_V(G)_trait`, y = Trait, xmin = `Variance_V(G)_trait` - `SE_V(G)_trait`, xmax = `Variance_V(G)_trait` + `SE_V(G)_trait`)) +
  geom_pointinterval(aes(fill = `Variance_V(G)_trait`), shape = 21, point_size = 1.5, stroke = 1) +
  scale_fill_gradientn(colors=met.brewer(direction = 1, &quot;OKeeffe2&quot;), guide = &quot;none&quot;) +
  geom_vline(xintercept = 0, linetype = 2) +
  labs(x = expression(paste(&quot;SNP genetic variance&quot;, (V[A]))),
       title = &quot;Males&quot;) +
  theme_minimal()

Vg_female_plot + Vg_male_plot</code></pre>
<p><img src="figure/GCTA_results.Rmd/unnamed-chunk-5-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="does-additive-genetic-variance-differ-between-the-sexes" class="section level2">
<h2>Does additive genetic variance differ between the sexes?</h2>
<div id="exploratory-plots" class="section level3">
<h3>Exploratory plots</h3>
<p>This plot shows the distribution of genetic and environmental variance in females and males. It is simply exploratory, as we do not account for the uncertainty around the point estimate for each trait here.</p>
<pre class="r"><code># Some wrangling required

a &lt;- SNP_selection_response_estimates %&gt;% 
  filter(Fitness_sex == &quot;Male&quot;)
 

b &lt;- SNP_selection_response_estimates %&gt;% 
  filter(Fitness_sex == &quot;Female&quot;)

# there are 304 traits for which we have measures in both sexes

SNP_both_sexes &lt;- inner_join(a, b, by = &quot;Trait&quot;)

add_var_data &lt;-  
  SNP_both_sexes %&gt;% 
  pivot_longer(cols = c(`Variance_V(G)_trait.x`, `Variance_V(G)_trait.y`), names_to = &quot;Sex&quot;, values_to = &quot;Additive_genetic_variance&quot;) %&gt;% 
  mutate(Sex = if_else(str_ends(Sex, &quot;.x&quot;), &quot;Male&quot;, &quot;Female&quot;)) %&gt;% 
  select(c(Trait, Sex, `Trait guild.x`, Additive_genetic_variance)) %&gt;% 
  filter(Additive_genetic_variance &lt; 1) 

env_var_data &lt;-
  SNP_both_sexes %&gt;% 
  pivot_longer(cols = c(`Variance_V(e)_trait.x`, `Variance_V(e)_trait.y`), names_to = &quot;Sex&quot;, values_to = &quot;Env_variance&quot;) %&gt;% 
  mutate(Sex = if_else(str_ends(Sex, &quot;.x&quot;), &quot;Male&quot;, &quot;Female&quot;)) %&gt;% 
  select(c(Trait, Sex, `Trait guild.x`, Env_variance)) 
  #filter(Env_variance &gt; -1 &amp; Env_variance &lt; 1) %&gt;%</code></pre>
<pre class="r"><code># plotting

p1 &lt;-
  add_var_data %&gt;%
  ggplot(aes(x = Additive_genetic_variance)) + 
  #geom_histogram(fill = met.brewer(&quot;Hokusai2&quot;)[2]) +
  stat_halfeye(fill = met.brewer(&quot;Hokusai2&quot;)[1]) +
  #geom_vline(xintercept = Add_Median[1], linetype = 2) +
  labs(y = &quot;Number of traits&quot;,
       x = &quot;V(G) - the estimated additive genetic variance&quot;) +
  theme_minimal() +
  facet_wrap(~Sex)

p2 &lt;-
  env_var_data %&gt;% 
  ggplot(aes(x = Env_variance)) + 
  #geom_histogram(fill = met.brewer(&quot;Hokusai2&quot;)[4]) +
  stat_halfeye(fill = met.brewer(&quot;Hokusai2&quot;)[1]) +
  #geom_vline(xintercept = median(env_var_data$Env_variance), linetype = 2) +
  labs(y = &quot;Number of traits&quot;,
       x = &quot;V(e) - the estimated environmental variance&quot;) +
  theme_minimal() +
  facet_wrap(~Sex)


p1 / p2</code></pre>
<p><img src="figure/GCTA_results.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/8d22c487891bef0c2bd0999def4812b8605e73c4/docs/figure/GCTA_results.Rmd/unnamed-chunk-7-1.png" target="_blank">8d22c48</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-16
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>From our <code>GCTA</code> analysis, we find that across traits measured in both sexes, environmental variance is ~three-fold larger than additive genetic variance explained by known SNPs. We also find that V(G) is highly zero-inflated (or 0.0001 inflated…). Whether this is a true effect or a result of an inability to estimate V(G) is <strong>unknown</strong>.</p>
<p>The plots suggest that males might have slightly higher V(G) than females, but that females have higher V(e).</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="meta-analysis-1" class="section level3">
<h3>Meta-analysis</h3>
<p>To test whether V(G) is higher in males in females and that V(e) is higher in females than males, we conduct meta-analysis across all traits measured in both sexes.</p>
<p>We remove traits for which the Standard error is calculated as 0, as well as traits which have genetic variances greater than 1.</p>
<p>Note that I currently specify a Gaussian error distribution. <strong>This fits the data poorly.</strong></p>
<pre class="r"><code>met_additive_data &lt;-
  SNP_both_sexes %&gt;% 
  select(Trait, `Trait guild.x`, Reference.x, `Variance_V(G)_trait.x`, `Variance_V(G)_trait.y`, `SE_V(G)_trait.x`, `SE_V(G)_trait.y`) %&gt;% 
  pivot_longer(cols = c(`Variance_V(G)_trait.x`, `Variance_V(G)_trait.y`, ), names_to = &quot;Sex&quot;, values_to = &quot;Additive_genetic_variance&quot;) %&gt;% 
  mutate(Sex = if_else(str_ends(Sex, &quot;.x&quot;), &quot;Male&quot;, &quot;Female&quot;),
         SE = if_else(Sex == &quot;Male&quot;, `SE_V(G)_trait.x`, `SE_V(G)_trait.y`),
         Trait = as.factor(Trait),
         Trait_guild = as.factor(`Trait guild.x`),
         Reference = as.factor(Reference.x),
         Sex = as.factor(Sex)) %&gt;% 
  select(-c(`SE_V(G)_trait.x`, `SE_V(G)_trait.y`, `Trait guild.x`, Reference.x)) %&gt;% 
  filter(SE != 0, Additive_genetic_variance &lt; 1) # note this filtering removes 5 traits


VG_model &lt;- 
  brm(Additive_genetic_variance | se(SE) ~ 0 + Sex + (1|Trait) + (1|Reference),
      family = gaussian(), 
      data = met_additive_data,
      prior = c(prior(normal(0, 0.2), class = b),
                prior(cauchy(0, 1), class = sd)),
      warmup = 4000, iter = 6000,
      seed = 1, cores = 4, chains = 4,
      control = list(adapt_delta = 0.9),
      file = &quot;fits/VG.model&quot;)

# note that sigma is not estimated. This is normal for meta-analysis

print(VG_model)</code></pre>
<pre><code> Family: gaussian 
  Links: mu = identity; sigma = identity 
Formula: Additive_genetic_variance | se(SE) ~ 0 + Sex + (1 | Trait) + (1 | Reference) 
   Data: met_additive_data (Number of observations: 603) 
  Draws: 4 chains, each with iter = 6000; warmup = 4000; thin = 1;
         total post-warmup draws = 8000

Group-Level Effects: 
~Reference (Number of levels: 51) 
              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(Intercept)     0.08      0.02     0.05     0.11 1.00     2954     4948

~Trait (Number of levels: 304) 
              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(Intercept)     0.02      0.01     0.00     0.04 1.00     2389     3706

Population-Level Effects: 
          Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
SexFemale     0.20      0.02     0.17     0.23 1.00     4786     5451
SexMale       0.22      0.02     0.18     0.25 1.00     4869     5585

Family Specific Parameters: 
      Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sigma     0.00      0.00     0.00     0.00   NA       NA       NA

Draws were sampled using sampling(NUTS). For each parameter, Bulk_ESS
and Tail_ESS are effective sample size measures, and Rhat is the potential
scale reduction factor on split chains (at convergence, Rhat = 1).</code></pre>
<p>Once again, a Gaussian model may not be a good fit for our data. Let’s check:</p>
<pre class="r"><code>pp_check(VG_model)</code></pre>
<p><img src="figure/GCTA_results.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Yep, not great gain! Need to think about this.</p>
<p>Anyway, in the meantime, here’s the predictions from the model plotted as halfeyes:</p>
<pre class="r"><code>Genetic_var_sex_plot &lt;-
  as_draws_df(VG_model) %&gt;% 
  select(b_SexFemale, b_SexMale) %&gt;% 
  rename(Female = b_SexFemale, Male = b_SexMale) %&gt;% 
  pivot_longer(cols = 1:2, names_to = &quot;Sex&quot;, values_to = &quot;Genetic_variance&quot;) %&gt;% 
  ggplot(aes(x = Genetic_variance, y = Sex)) +
  stat_halfeye(point_interval = median_qi, .width = c(.5, .95),
               slab_colour = &quot;black&quot;, shape = 21, point_size = 3, 
               stroke = 1.5, point_fill = &quot;white&quot;) +
  labs(x = &quot;Mean genetic variance&quot;,
       y = &quot;Sex&quot;) +
  theme_bw()


Genetic_var_diff_plot &lt;-
  as_draws_df(VG_model) %&gt;% 
  mutate(Sex_diff = b_SexFemale - b_SexMale) %&gt;% 
  select(b_SexFemale, b_SexMale, Sex_diff) %&gt;% 
  ggplot(aes(x = Sex_diff, y = 0, fill = stat(x &gt; 0))) +
  stat_halfeye(point_interval = median_qi, .width = c(.5, .95),
               slab_colour = &quot;black&quot;, shape = 21, point_size = 3, 
               stroke = 1.5, point_fill = &quot;white&quot;) +
  geom_vline(xintercept = 0, linetype = 2) +
  labs(x = &quot;Difference in genetic variance between traits\n expressed in females and males&quot;,
       y = NULL) +
  scale_y_continuous(NULL, breaks = NULL) +
  scale_fill_manual(values = c(&quot;grey&quot;, met.brewer(&quot;Hokusai2&quot;)[2]), guide = &quot;none&quot;) +
  theme_bw()

Genetic_var_sex_plot + Genetic_var_diff_plot</code></pre>
<p><img src="figure/GCTA_results.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><span class="math inline">\(~\)</span></p>
</div>
</div>
<div id="which-traits-exhibit-environmental-variance-not-explained-by-known-snps" class="section level2">
<h2>Which traits exhibit environmental variance not explained by known SNPs?</h2>
<p>What I call environmental variance is really all residual variance that is not explained by the known set of SNPs in the DGRP. This includes all environmental and unmapped genetic explanations of variance e.g. unknown SNPs, regulatory effects, mtDNA and the genetic material of cytoplasmic endosympbionts (in some cases).</p>
<p>In theory this residual variance should be larger than the environmental variance used to find broad-sense heritability.</p>
<pre class="r"><code>Ve_female_plot &lt;-
  SNP_selection_response_estimates %&gt;% 
  filter(Fitness_sex == &quot;Female&quot;,
         `Variance_V(e)_trait` &lt; 5 &amp; `Variance_V(e)_trait` &gt; -1,
         Reference != &quot;Jin et al (2020) PLOS Genetics&quot; &amp;
           Reference != &quot;Everett et al (2020) Genome Research&quot; &amp;
           Reference != &quot;Dembeck et al (2015) eLife&quot;) %&gt;% 
  mutate(Trait = fct_reorder(Trait, `Variance_V(e)_trait`)) %&gt;% 
  
  ggplot(aes(x = `Variance_V(e)_trait`, y = Trait, xmin = `Variance_V(e)_trait` - `SE_V(e)_trait`, xmax = `Variance_V(e)_trait` + `SE_V(e)_trait`)) +
  geom_pointinterval(aes(fill = `Variance_V(e)_trait`), shape = 21, point_size = 1.5, stroke = 1) +
  scale_fill_gradientn(colors=met.brewer(direction = 1, &quot;OKeeffe2&quot;), guide = &quot;none&quot;) +
  geom_vline(xintercept = 0, linetype = 2) +
  labs(x = expression(paste(&quot;Environmental variance&quot;, (V[A]))),
       title = &quot;Females&quot;) +
  theme_minimal()
  

Ve_male_plot &lt;-
  SNP_selection_response_estimates %&gt;% 
  filter(Fitness_sex == &quot;Male&quot;,
         `Variance_V(e)_trait` &lt; 5 &amp; `Variance_V(e)_trait` &gt; -1,
         Reference != &quot;Jin et al (2020) PLOS Genetics&quot; &amp;
           Reference != &quot;Everett et al (2020) Genome Research&quot; &amp;
           Reference != &quot;Dembeck et al (2015) eLife&quot;) %&gt;% 
  mutate(Trait = fct_reorder(Trait, `Variance_V(e)_trait`)) %&gt;% 
  
  ggplot(aes(x = `Variance_V(e)_trait`, y = Trait, xmin = `Variance_V(e)_trait` - `SE_V(e)_trait`, xmax = `Variance_V(e)_trait` + `SE_V(e)_trait`)) +
  geom_pointinterval(aes(fill = `Variance_V(e)_trait`), shape = 21, point_size = 1.5, stroke = 1) +
  scale_fill_gradientn(colors=met.brewer(direction = 1, &quot;OKeeffe2&quot;), guide = &quot;none&quot;) +
  geom_vline(xintercept = 0, linetype = 2) +
  labs(x = expression(paste(&quot;Environmental variance&quot;, (V[A]))),
       title = &quot;Males&quot;) +
  theme_minimal()

Ve_female_plot + Ve_male_plot</code></pre>
<p><img src="figure/GCTA_results.Rmd/unnamed-chunk-11-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="does-enviornmental-variation-differ-between-the-sexes" class="section level2">
<h2>Does enviornmental variation differ between the sexes?</h2>
<div id="meta-analysis-2" class="section level3">
<h3>Meta-analysis</h3>
<pre class="r"><code>met_enviro_data &lt;-
  SNP_both_sexes %&gt;% 
  select(Trait, `Trait guild.x`, Reference.x, `Variance_V(e)_trait.x`, `Variance_V(e)_trait.y`, `SE_V(e)_trait.x`, `SE_V(e)_trait.y`) %&gt;% 
  pivot_longer(cols = c(`Variance_V(e)_trait.x`, `Variance_V(e)_trait.y`, ), names_to = &quot;Sex&quot;, values_to = &quot;Environmental_variance&quot;) %&gt;% 
  mutate(Sex = if_else(str_ends(Sex, &quot;.x&quot;), &quot;Male&quot;, &quot;Female&quot;),
         SE = if_else(Sex == &quot;Male&quot;, `SE_V(e)_trait.x`, `SE_V(e)_trait.y`),
         Trait = as.factor(Trait),
         Trait_guild = as.factor(`Trait guild.x`),
         Reference = as.factor(Reference.x),
         Sex = as.factor(Sex)) %&gt;% 
  select(-c(`SE_V(e)_trait.x`, `SE_V(e)_trait.y`, `Trait guild.x`, Reference.x)) %&gt;% 
  filter(SE != 0)


Ve_model &lt;- 
  brm(Environmental_variance | se(SE) ~ 0 + Sex + (1|Trait) + (1|Reference),
      family = gaussian(), 
      data = met_enviro_data,
      prior = c(prior(normal(0, 0.2), class = b),
                prior(cauchy(0, 1), class = sd)),
      warmup = 4000, iter = 6000,
      seed = 1, cores = 4, chains = 4,
      control = list(adapt_delta = 0.9),
      file = &quot;fits/Ve.model&quot;)

print(Ve_model)</code></pre>
<pre><code> Family: gaussian 
  Links: mu = identity; sigma = identity 
Formula: Environmental_variance | se(SE) ~ 0 + Sex + (1 | Trait) + (1 | Reference) 
   Data: met_enviro_data (Number of observations: 604) 
  Draws: 4 chains, each with iter = 6000; warmup = 4000; thin = 1;
         total post-warmup draws = 8000

Group-Level Effects: 
~Reference (Number of levels: 51) 
              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(Intercept)     0.14      0.03     0.09     0.21 1.00     2516     3720

~Trait (Number of levels: 304) 
              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(Intercept)     0.11      0.02     0.07     0.15 1.00     3055     4785

Population-Level Effects: 
          Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
SexFemale     0.53      0.03     0.47     0.60 1.00     3437     4769
SexMale       0.50      0.03     0.44     0.57 1.00     3403     4703

Family Specific Parameters: 
      Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sigma     0.00      0.00     0.00     0.00   NA       NA       NA

Draws were sampled using sampling(NUTS). For each parameter, Bulk_ESS
and Tail_ESS are effective sample size measures, and Rhat is the potential
scale reduction factor on split chains (at convergence, Rhat = 1).</code></pre>
<p>Check the fit</p>
<pre class="r"><code>pp_check(Ve_model)</code></pre>
<p><img src="figure/GCTA_results.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>As with all the meta-models, we need to find a distribution that precludes negative values for variances.</p>
<p>Plot the predictions:</p>
<pre class="r"><code>Env_var_sex_plot &lt;-
  as_draws_df(Ve_model) %&gt;% 
  select(b_SexFemale, b_SexMale) %&gt;% 
  rename(Female = b_SexFemale, Male = b_SexMale) %&gt;% 
  pivot_longer(cols = 1:2, names_to = &quot;Sex&quot;, values_to = &quot;Environmental_variance&quot;) %&gt;% 
  ggplot(aes(x = Environmental_variance, y = Sex)) +
  stat_halfeye(point_interval = median_qi, .width = c(.5, .95),
               slab_colour = &quot;black&quot;, shape = 21, point_size = 3, 
               stroke = 1.5, point_fill = &quot;white&quot;) +
  labs(x = &quot;Mean environmental (residual) variance&quot;,
       y = &quot;Sex&quot;) +
  theme_bw()




Env_var_diff_plot &lt;-
  as_draws_df(Ve_model) %&gt;% 
  mutate(Sex_diff = b_SexFemale - b_SexMale) %&gt;% 
  select(b_SexFemale, b_SexMale, Sex_diff) %&gt;% 
  ggplot(aes(x = Sex_diff, y = 0, fill = stat(x &gt; 0))) +
  stat_halfeye(point_interval = median_qi, .width = c(.5, .95),
               slab_colour = &quot;black&quot;, shape = 21, point_size = 3, 
               stroke = 1.5, point_fill = &quot;white&quot;) +
  geom_vline(xintercept = 0, linetype = 2) +
  labs(x = &quot;Difference in environmental variance between traits\n expressed in females and males&quot;,
       y = NULL) +
  scale_y_continuous(NULL, breaks = NULL) +
  scale_fill_manual(values = c(&quot;grey&quot;, met.brewer(&quot;Hokusai2&quot;)[2]), guide = &quot;none&quot;) +
  theme_bw()

Env_var_sex_plot + Env_var_diff_plot</code></pre>
<p><img src="figure/GCTA_results.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We detect no differences in genetic or environmental variance between the sexes.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
</div>
<div id="testing-general-predictions-of-fishers-theorem" class="section level2">
<h2>Testing general predictions of Fisher’s theorem</h2>
<p><span class="math inline">\(~\)</span></p>
<p><strong>1. Do traits correlated with fitness have lower additive genetic variance?</strong></p>
<p>Following Fisher’s theorem and assuming a constant environment, in the absence of new variation, selection is expected to erode all additive genetic variation for fitness. One implication of this expectation is that traits that are correlated with fitness should show reduced additive genetic variance relative to those that are not correlated with fitness <em>(Robertson 1955b)</em>.</p>
<p>There are previous studies that have compared fitness related traits i.e. those implicated with reproduction, with morphological traits, which are assumed to have lesser relationships with fitness.</p>
<ul>
<li><p>The problem is associating these traits with _lifetime) fitness. This shouldn’t be such a problem for us…</p></li>
<li><p>Multiple studies looking at this corollary use phenotypic correlations with fitness rather than genetic. Once again, this is no problem for us…</p></li>
<li><p>Even if there is lots of additive genetic variance for a trait and it is correlated with fitness, if fitness has a high environmental variance component the strength of selection on the trait will be weak.</p></li>
</ul>
<p><em>Houle (1992)</em> looks like an important meta-analysis type work</p>
<p>Note that heritability can also decline with an increase in environmental variance and no decrease (or an increase) in genetic variance. If true, this makes a simple regression with heritability and fitness misleading.</p>
<p>Using evolvability as his metric, <em>Houle (1992)</em> found that characters assumed to be closely related to fitness have higher evolvabilities than do trait with looser relationships to fitness. This suggests that the negative relationship between fitness and heritability is due to increased environmental variance in fitness related traits, not a decrease in additive genetic variance.</p>
<pre class="r"><code>Rz_VG_plot &lt;-
  SNP_selection_response_estimates %&gt;% 
  filter(`Variance_C(G)_tr12` &gt; -1 &amp; `Variance_V(G)_trait` &gt; -1 &amp; `Variance_C(G)_tr12` &lt; 1 &amp; `Variance_V(G)_trait` &lt; 1) %&gt;% 
  #`Trait guild` != &quot;Metabolome&quot; &amp; `Trait guild` != &quot;CHC&quot;, `Trait guild` != &quot;Microbiome&quot;) %&gt;% 
  mutate(`Variance_C(G)_tr12` = abs(`Variance_C(G)_tr12`)) %&gt;% 
  ggplot(aes(y = `Variance_C(G)_tr12`, x = `Variance_V(G)_trait`, fill = Fitness_sex)) +
  #geom_hex()
  geom_point(shape = 21, size = 3, alpha = 0.5) +
  geom_smooth(aes(colour = Fitness_sex)) +
  scale_fill_manual(values = met.brewer(&quot;Hokusai3&quot;, 2), guide = &quot;none&quot;) +
  scale_colour_manual(values = met.brewer(&quot;Hokusai3&quot;, 2), guide = &quot;none&quot;) +
  labs(y = expression(paste(&quot;Selection response &quot;, (R[z]))),
       x = expression(paste(&quot;SNP genetic variance &quot;, (V[A]))),
       fill = &quot;Sex&quot;) +
  theme_bw()

Rz_VG_plot</code></pre>
<p><img src="figure/GCTA_results.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Figure X: The SNP-covariance between fitness and trait (Robertson covariance) is positively correlated with additive genetic variance. The Robertson covariance, or the secondary theorem of natural selection, <span class="math inline">\(R_z\)</span> estimates the expected response to selection of a trait.</p>
<p>Ok so I don’t think we’re learning much here. This is very much the expected result, given additive genetic variance is the fuel for a selection response.</p>
<p><span class="math inline">\(~\)</span></p>
<p><strong>2. Do traits correlated with fitness have higher levels of residual variance?</strong></p>
<pre class="r"><code>Rz_Ve_plot &lt;-
  SNP_selection_response_estimates %&gt;% 
  filter(`Variance_C(G)_tr12` &gt; -1 &amp; `Variance_V(e)_trait` &gt; -1 &amp; `Variance_C(G)_tr12` &lt; 1 &amp; `Variance_V(e)_trait` &lt; 5) %&gt;% 
  #`Trait guild` != &quot;Metabolome&quot; &amp; `Trait guild` != &quot;CHC&quot;, `Trait guild` != &quot;Microbiome&quot;) %&gt;% 
  mutate(`Variance_C(G)_tr12` = abs(`Variance_C(G)_tr12`)) %&gt;% 
  ggplot(aes(x = `Variance_C(G)_tr12`, y = `Variance_V(e)_trait`, fill = Fitness_sex)) +
  #geom_hex()
  geom_point(shape = 21, size = 3, alpha = 0.5) +
  geom_smooth(aes(colour = Fitness_sex)) +
  scale_fill_manual(values = met.brewer(&quot;Hokusai3&quot;, 2), guide = &quot;none&quot;) +
  scale_colour_manual(values = met.brewer(&quot;Hokusai3&quot;, 2), guide = &quot;none&quot;) +
  labs(y = expression(paste(&quot;Selection response &quot;, (R[z]))),
       x = expression(paste(&quot;Environmental variance &quot;, (V[A]))),
       fill = &quot;Sex&quot;) +
  theme_bw()

Rz_Ve_plot</code></pre>
<p><img src="figure/GCTA_results.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Figure X: The SNP-covariance between fitness and trait (Robertson covariance) is negatively correlated with environmental variance underlying the trait. The Robertson covariance, or the secondary theorem of natural selection, <span class="math inline">\(R_z\)</span> estimates the expected response to selection of a trait.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="sexually-antagonistic-selection" class="section level2">
<h2>Sexually antagonistic selection</h2>
<p><span class="math inline">\(~\)</span></p>
<p>Using the Robertson covariance We have found th expected responses to selection for many traits in females and males. For those traits where we have a measure of both, we can quantify sexually antagonistic selection by finding traits that have selection operating in opposite directions, depending on which sex the trait is expressed in.</p>
<p><a href="https://doi.org/10.1111/j.1558-5646.2010.01021.x">Innocenti and Morrow (2011)</a> present an index for measuring the intensity for sexually antagonistic selection:</p>
<p><span class="math inline">\(I = \frac{\beta_M \beta_F}{\sqrt{(\beta_M^{2} + \beta_F^{2})/2}}\)</span></p>
<p>Where <span class="math inline">\(B_F\)</span> and <span class="math inline">\(B_M\)</span> are the standardised selection gradients for females and males respectively.</p>
<p>From <strong>Innocenti and Morrow</strong>:</p>
<p>This index is positive when selection is concordant in the two sexes, negative when antagonistic in the two sexes, and is zero when selection is absent in one sex (note that it will miss conflict that occurs when strong stabilizing selection is present in that sex). Finally it is proportional to the absolute intensity of selection.</p>
<p>Additionally, it has the desirable properties of being symmetrical and normally distributed for a random set of normally distributed <span class="math inline">\(B_F\)</span>, <span class="math inline">\(B_M\)</span>. |I| is also always included in the interval between the absolute values of the selection gradient in the two sexes, and it coincides with them when <span class="math inline">\(B_F\)</span> = <span class="math inline">\(B_M\)</span>. As a potential drawback, it should be noted that such quantity is not defined when <span class="math inline">\(B_F\)</span> = <span class="math inline">\(B_M\)</span> = 0, even though it makes little sense to estimate how concordant or antagonistic directional selection is when it is absent in both sexes.</p>
<p>Unfortunately, we do not have selection differentials / gradients. Instead, we have the potentially more powerful expected response to selection for many traits in both sexes.</p>
<p><span class="math inline">\(~\)</span></p>
<pre class="r"><code># create our sexual antagonism tibble by joining the female and male tibbles by traits they both share

SNP_antagonism &lt;- SNP_both_sexes %&gt;% 
  rename(Trait_covariance_male_fitness = `Variance_C(G)_tr12.x`,
         Trait_covariance_female_fitness = `Variance_C(G)_tr12.y`) %&gt;% 
  select(-c(Fitness_sex.x, Fitness_sex.y)) %&gt;% 
  # calculate the innocenti and morrow index
  mutate(Selection_index = Trait_covariance_female_fitness * Trait_covariance_male_fitness / sqrt(((Trait_covariance_female_fitness)^2 + (Trait_covariance_male_fitness)^2)/2)) %&gt;% 
  # get rid of traits with unrealistic covariances
  filter(Trait_covariance_male_fitness &gt; -5,
         Trait_covariance_female_fitness &lt; 2 &amp; Trait_covariance_female_fitness &gt; -5)

# plot the data

sexual_concordance_plot &lt;-
  ggplot(data = SNP_antagonism, aes(x = Trait_covariance_female_fitness, y = Trait_covariance_male_fitness, fill = Selection_index)) +
  #geom_errorbar(aes(ymin = Trait_covariance_male_fitness - `SE_C(G)_tr12.x`, ymax = Trait_covariance_male_fitness + `SE_C(G)_tr12.x`),
   #             width = 0.01, colour = &quot;grey11&quot;, alpha = 0.8) +
  #geom_errorbarh(aes(xmin = Trait_covariance_female_fitness - `SE_C(G)_tr12.y`, xmax = Trait_covariance_female_fitness + `SE_C(G)_tr12.y`),
   #              colour = &quot;grey11&quot;, alpha = 0.8, height = 0.01) +
  geom_point(shape = 21, alpha = 1, size = 4, show.legend = TRUE) +
  geom_smooth(method = &#39;lm&#39;, color=&#39;black&#39;) +
  geom_hline(yintercept = 0, linetype = 2) +
  geom_vline(xintercept = 0, linetype = 2) +
  #coord_cartesian(xlim = c(-0.35, 0.35), ylim = c(-0.4, 0.4)) +
  scale_fill_carto_c(palette = &quot;Geyser&quot;,
                     breaks=c(-0.25, -0.125, 0, 0.125, 0.25),
                     limits=c(-0.25,0.25)) +
  labs(x = &quot;Female response to selection&quot;,
       y = &quot;Male response to selection&quot;,
       fill = &quot;Concordance\n of selection\n&quot;) +
  theme_bw() +
  theme(panel.border= element_blank(),
        axis.line=element_line(),
        text = element_text(size=14),
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.minor.x = element_blank(),
        axis.title.x = element_text(size = 14),
        axis.title.y = element_text(size = 14))

sexual_concordance_plot</code></pre>
<p><img src="figure/GCTA_results.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Figure X:</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] tidybayes_3.0.2   brms_2.17.0       Rcpp_1.0.8.3      ggbeeswarm_0.6.0 
 [5] rcartocolor_2.0.0 patchwork_1.1.1   MetBrewer_0.2.0   DT_0.22          
 [9] kableExtra_1.3.4  glue_1.6.2        forcats_0.5.1     stringr_1.4.0    
[13] dplyr_1.0.8       purrr_0.3.4       readr_2.1.2       tidyr_1.2.0      
[17] tibble_3.1.6      ggplot2_3.3.5     tidyverse_1.3.1   workflowr_1.7.0  

loaded via a namespace (and not attached):
  [1] readxl_1.4.0         backports_1.4.1      systemfonts_1.0.4   
  [4] plyr_1.8.7           igraph_1.3.1         svUnit_1.0.6        
  [7] splines_4.1.0        crosstalk_1.2.0      TH.data_1.1-0       
 [10] rstantools_2.2.0     inline_0.3.19        digest_0.6.29       
 [13] htmltools_0.5.2      fansi_1.0.3          magrittr_2.0.3      
 [16] checkmate_2.1.0      tzdb_0.3.0           modelr_0.1.8        
 [19] RcppParallel_5.1.5   matrixStats_0.62.0   vroom_1.5.7         
 [22] xts_0.12.1           sandwich_3.0-1       svglite_2.1.0       
 [25] prettyunits_1.1.1    colorspace_2.0-3     rvest_1.0.2         
 [28] ggdist_3.1.1         haven_2.5.0          xfun_0.30           
 [31] callr_3.7.0          crayon_1.5.1         jsonlite_1.8.0      
 [34] lme4_1.1-29          survival_3.3-1       zoo_1.8-10          
 [37] gtable_0.3.0         emmeans_1.7.3        webshot_0.5.3       
 [40] distributional_0.3.0 pkgbuild_1.3.1       rstan_2.21.5        
 [43] abind_1.4-5          scales_1.2.0         mvtnorm_1.1-3       
 [46] DBI_1.1.2            miniUI_0.1.1.1       viridisLite_0.4.0   
 [49] xtable_1.8-4         diffobj_0.3.5        bit_4.0.4           
 [52] stats4_4.1.0         StanHeaders_2.21.0-7 htmlwidgets_1.5.4   
 [55] httr_1.4.2           threejs_0.3.3        arrayhelpers_1.1-0  
 [58] posterior_1.2.1      ellipsis_0.3.2       pkgconfig_2.0.3     
 [61] loo_2.5.1            farver_2.1.0         sass_0.4.1          
 [64] dbplyr_2.1.1         utf8_1.2.2           labeling_0.4.2      
 [67] tidyselect_1.1.2     rlang_1.0.2          reshape2_1.4.4      
 [70] later_1.3.0          munsell_0.5.0        cellranger_1.1.0    
 [73] tools_4.1.0          cli_3.2.0            generics_0.1.2      
 [76] broom_0.8.0          ggridges_0.5.3       evaluate_0.15       
 [79] fastmap_1.1.0        yaml_2.3.5           bit64_4.0.5         
 [82] processx_3.5.3       knitr_1.38           fs_1.5.2            
 [85] nlme_3.1-157         projpred_2.1.1       whisker_0.4         
 [88] mime_0.12            xml2_1.3.3           compiler_4.1.0      
 [91] bayesplot_1.9.0      shinythemes_1.2.0    rstudioapi_0.13     
 [94] gamm4_0.2-6          beeswarm_0.4.0       reprex_2.0.1        
 [97] bslib_0.3.1          stringi_1.7.6        highr_0.9           
[100] ps_1.7.0             Brobdingnag_1.2-7    lattice_0.20-45     
[103] Matrix_1.4-1         nloptr_2.0.0         markdown_1.1        
[106] shinyjs_2.1.0        tensorA_0.36.2       vctrs_0.4.1         
[109] pillar_1.7.0         lifecycle_1.0.1      jquerylib_0.1.4     
[112] bridgesampling_1.1-2 estimability_1.3     httpuv_1.6.5        
[115] R6_2.5.1             promises_1.2.0.1     gridExtra_2.3       
[118] vipor_0.4.5          codetools_0.2-18     boot_1.3-28         
[121] MASS_7.3-57          colourpicker_1.1.1   gtools_3.9.2        
[124] assertthat_0.2.1     rprojroot_2.0.3      withr_2.5.0         
[127] shinystan_2.6.0      multcomp_1.4-18      mgcv_1.8-40         
[130] parallel_4.1.0       hms_1.1.1            grid_4.1.0          
[133] minqa_1.2.4          coda_0.19-4          rmarkdown_2.13      
[136] git2r_0.30.1         getPass_0.2-2        shiny_1.7.1         
[139] lubridate_1.8.0      base64enc_0.1-3      dygraphs_1.1.1.6    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
