<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Estimating line means from raw data</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">The response to selection, sexual antagonism and sexual concordance in the DGRP</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Data_collation.html">Data collation</a>
</li>
<li>
  <a href="get_line_means_from_raw_data.html">Finding line means</a>
</li>
<li>
  <a href="Main_analysis.html">Analysis</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tomkeaney/DGRP_sexual_conflict">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Estimating line means from raw data</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-04-30
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DGRP_sexual_conflict/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210706code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20210706)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210706code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210706)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPsexualconflicttreecf83353ef95d45de7c1825713330f7a560919c72targetblankcf83353a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/tree/cf83353ef95d45de7c1825713330f7a560919c72" target="_blank">cf83353</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPsexualconflicttreecf83353ef95d45de7c1825713330f7a560919c72targetblankcf83353a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/tree/cf83353ef95d45de7c1825713330f7a560919c72" target="_blank">cf83353</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  %
    Untracked:  -PC18-107.Rhistory
    Untracked:  Chapter_2_Figure_S1.pdf
    Untracked:  Chapter_2_Figure_S2.pdf
    Untracked:  Figure_3.pdf
    Untracked:  Figure_4.pdf
    Untracked:  Figure_S3.pdf
    Untracked:  Figures.Rmd
    Untracked:  Manuscript/
    Untracked:  PRISMA.pptx
    Untracked:  R_transcriptome_medians.csv
    Untracked:  Reported_heritability.xlsx
    Untracked:  Supplementary_data/
    Untracked:  Useful_cuts.Rmd
    Untracked:  code/get_gene_annotations.R
    Untracked:  cuts/
    Untracked:  data/R_medians.csv
    Untracked:  data/all_dmel_genes.csv
    Untracked:  data/data_collation/
    Untracked:  data/gene_anntotations.csv
    Untracked:  data/huang_transcriptome/
    Untracked:  data/organismal_level_output/
    Untracked:  data/trait_names.rds
    Untracked:  data/transcriptome_output/
    Untracked:  figures/
    Untracked:  fits/
    Untracked:  mutation_load_analysis/

Unstaged changes:
    Modified:   _workflowr.yml
    Modified:   analysis/Transcriptome_analysis.Rmd
    Deleted:    code/export_gwas_results_for_shiny_app.R
    Deleted:    data/all.dgrp.phenos_unscaled.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/get_line_means_from_raw_data.Rmd</code>) and HTML
(<code>docs/get_line_means_from_raw_data.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/cf83353ef95d45de7c1825713330f7a560919c72/analysis/get_line_means_from_raw_data.Rmd" target="_blank">cf83353</a>
</td>
<td>
tomkeaney
</td>
<td>
2024-04-30
</td>
<td>
Prep for submission
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-in-necessary-packages" class="section level1">
<h1>Load in necessary packages</h1>
<pre class="r"><code>library(Matrix)
library(tidyverse)
library(brms) # for bayesian models
library(tidybayes) # for more bayesian things
library(broom) # convert results of functions into tables

if(!dir.exists(&quot;fits/raw_data_fits&quot;)) dir.create(&quot;fits/raw_data_fits&quot;)
my_scale &lt;- function(x) as.numeric(scale(x))</code></pre>
<div id="estimating-line-means-from-raw-data" class="section level2">
<h2>Estimating line means from raw data</h2>
<p>Some studies/authors provided the raw data that was collected when
measuring DGRP phenotypes (e.g. measurements of individual flies),
although we need line mean data to calculate the response to selection.
To get line means, we fitted Bayesian mixed effect models and used these
to estimate mean trait values for each line, using the model formula
that made the most sense to us given the study’s experimental design
(see annotations in the code below). Often, but not always, this
conformed with the original analysis. Our analyses of each paper’s data
are presented in a separate tab below.</p>
<p>Where the response follows a Gaussian distribution, we standardise
the response to have a mean of 0 and a standard deviation of 1, in order
to make prior specification more straightforward. In these cases, the
estimated line means are converted back to the response scale.</p>
<p>The Bayesian models won’t run unless you have <code>stan</code> and
the <code>brms</code> package installed. To install <code>stan</code>
follow these <a
href="https://github.com/stan-dev/rstan/wiki/RStan-Getting-Started">instructions</a>.</p>
</div>
<div id="analysis-of-raw-data" class="section level2 tabset">
<h2 class="tabset">Analysis of raw data</h2>
<div id="jumbo-lucioni-et-al-2012" class="section level3">
<h3>Jumbo-Lucioni et al 2012</h3>
<pre class="r"><code>Jumbo_mito_data &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Jumbo_Lucioni_2012.csv&quot;) %&gt;% 
  mutate(line = as.factor(Line),
         Sex = as.factor(Gender),
         Rep = as.factor(Rep),
         po_ratio = na_if(`po ratio`, &quot;.&quot;),
         po_ratio = as.numeric(po_ratio)) %&gt;% 
  rename(state_3 = `st 3 (pmol/s*mg protein)`,
         state_4 = `st 4 (pmol/s*mg protein)`) %&gt;% 
  select(line, Sex, Rep, state_3, state_4, po_ratio) %&gt;% 
  filter(!is.na(line))

# standardise 
Jumbo_mito_data &lt;-
  Jumbo_mito_data %&gt;% 
  mutate(st_state_3 = my_scale(state_3),
         st_state_4 = my_scale(state_4),
         st_po_ratio = my_scale(po_ratio)) 



# respiration state 3 model

state_3_model &lt;- brm(st_state_3 ~ 1 + Sex + (Sex|line),
                      data = Jumbo_mito_data, family = gaussian,
                      prior = c(prior(normal(0, 0.5), class = Intercept),
                                prior(normal(0, 0.5), class = b),
                                prior(exponential(1), class = sd)),
                      iter = 6000, warmup = 2000, chains = 4, cores = 4,
                      seed = 2, file = &quot;fits/raw_data_fits/state_3.model&quot;)

#pp_check(state_3_model)

new_data &lt;-
  expand_grid(line = Jumbo_mito_data$line,
              Sex = c(&quot;F&quot;, &quot;M&quot;)) %&gt;% 
  distinct(line, Sex)

     
fitted_state_3 &lt;- fitted(state_3_model,
       newdata = new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Jumbo_mito_data$state_3) + mean(Jumbo_mito_data$state_3),
         Q2.5 = Q2.5*sd(Jumbo_mito_data$state_3) + mean(Jumbo_mito_data$state_3),
         Q97.5 = Q97.5*sd(Jumbo_mito_data$state_3) + mean(Jumbo_mito_data$state_3))

state_3_estimates &lt;- tibble(new_data, fitted_state_3) %&gt;% 
  mutate(Sex = if_else(Sex == &quot;F&quot;, &quot;Female&quot;, &quot;Male&quot;))

# build a function to create tibbles holding sex specific line means

Jumbo_line_means &lt;- function(estimates, y, Trait, description){
  estimates %&gt;% 
  filter(Sex == y) %&gt;% 
  mutate(Trait = Trait,
         Reference = &quot;Jumbo-Lucioni et al (2012) BMC Genomics&quot;,
         `Trait guild` = &quot;Physiological&quot;,
         `Trait description` = description,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)
}

# create the state 3 tibbles

# females

state_3_f_line_means &lt;- Jumbo_line_means(state_3_estimates, &quot;Female&quot;, &quot;mitochondrial.state.3.respiration.f&quot;, &quot;Mean mitochondrial state 3 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 3 respiration occurs when protons re-enter the mitochondrial matrix in the presence of ADP, which results in the production of ATP. Higher values indicate higher respiration rates.&quot;)


# males

state_3_m_line_means &lt;- Jumbo_line_means(state_3_estimates, &quot;Male&quot;, &quot;mitochondrial.state.3.respiration.m&quot;, &quot;Mean mitochondrial state 3 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 3 respiration occurs when protons re-enter the mitochondrial matrix in the presence of ADP, which results in the production of ATP. Higher values indicate higher respiration rates.&quot;)


# respiration state 4 model

state_4_model &lt;- brm(st_state_4 ~ 1 + Sex + (Sex|line),
                     data = Jumbo_mito_data, family = gaussian,
                     prior = c(prior(normal(0, 0.5), class = Intercept),
                               prior(normal(0, 0.5), class = b),
                               prior(exponential(1), class = sd)),
                     iter = 10000, warmup = 4000, chains = 4, cores = 4,
                     control = list(adapt_delta = 0.95, max_treedepth = 10),
                     seed = 2, file = &quot;fits/raw_data_fits/state_4.model&quot;)

#pp_check(state_4_model)

     
fitted_state_4 &lt;- fitted(state_4_model,
       newdata = new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Jumbo_mito_data$state_4) + mean(Jumbo_mito_data$state_4),
         Q2.5 = Q2.5*sd(Jumbo_mito_data$state_4) + mean(Jumbo_mito_data$state_4),
         Q97.5 = Q97.5*sd(Jumbo_mito_data$state_4) + mean(Jumbo_mito_data$state_4))

state_4_estimates &lt;- tibble(new_data, fitted_state_4) %&gt;% 
  mutate(Sex = if_else(Sex == &quot;F&quot;, &quot;Female&quot;, &quot;Male&quot;))

# create the state 4 tibbles

# females

state_4_f_line_means &lt;- Jumbo_line_means(state_4_estimates, &quot;Female&quot;, &quot;mitochondrial.state.4.respiration.f&quot;, &quot;Mean mitochondrial state 4 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 4 respiration occurs when protons leak into the mitochondrial matrix in the absence of ADP, which is required for ATP synthase. Higher values indicate higher respiration rates.&quot;)

# males

state_4_m_line_means &lt;- Jumbo_line_means(state_4_estimates, &quot;Male&quot;, &quot;mitochondrial.state.4.respiration.m&quot;, &quot;Mean mitochondrial state 4 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 4 respiration occurs when protons leak into the mitochondrial matrix in the absence of ADP, which is required for ATP synthase. Higher values indicate higher respiration rates.&quot;)

# P:O ratio model



PO_ratio_model &lt;- brm(st_po_ratio ~ 1 + Sex + (Sex|line),
                      data = Jumbo_mito_data %&gt;% filter(po_ratio != &quot;NA&quot;), family = gaussian,
                      prior = c(prior(normal(0, 0.5), class = Intercept),
                                prior(normal(0, 0.5), class = b),
                                prior(exponential(1), class = sd)),
                      iter = 6000, warmup = 2000, chains = 4, cores = 4,
                      seed = 2, file = &quot;fits/raw_data_fits/PO_ratio.model&quot;)

#pp_check(PO_ratio_model)

Jumbo_PO_data &lt;- Jumbo_mito_data %&gt;% filter(po_ratio != &quot;NA&quot;)
     
fitted_PO_ratio &lt;- fitted(PO_ratio_model, newdata = new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Jumbo_PO_data$po_ratio) + mean(Jumbo_PO_data$po_ratio),
         Q2.5 = Q2.5*sd(Jumbo_PO_data$po_ratio) + mean(Jumbo_PO_data$po_ratio),
         Q97.5 = Q97.5*sd(Jumbo_PO_data$po_ratio) + mean(Jumbo_PO_data$po_ratio))

PO_ratio_estimates &lt;- tibble(new_data, fitted_PO_ratio) %&gt;% 
  mutate(Sex = if_else(Sex == &quot;F&quot;, &quot;Female&quot;, &quot;Male&quot;))

# create the P:O ratio tibbles

# females

PO_ratio_f_line_means &lt;- Jumbo_line_means(PO_ratio_estimates, &quot;Female&quot;, &quot;mitochondrial.PO.ratio.f&quot;, &quot;Mean phosphate/oxygen ratio (P:O ratio) - a measure of the efficiency with which mitochondria convert oxygen into ATP. The ratio was here calculated as the amount of ADP consumed per oxygen being reduced during state 3 respriation. Mitochondria were sourced from the thorax and respiration was measured in vitro. Higher values indicate more efficient ATP production.&quot;)

# males

PO_ratio_m_line_means &lt;- Jumbo_line_means(PO_ratio_estimates, &quot;Male&quot;, &quot;mitochondrial.PO.ratio.m&quot;, &quot;Mean phosphate/oxygen ratio (P:O ratio) - a measure of the efficiency with which mitochondria convert oxygen into ATP. The ratio was here calculated as the amount of ADP consumed per oxygen being reduced during state 3 respriation. Mitochondria were sourced from the thorax and respiration was measured in vitro. Higher values indicate more efficient ATP production.&quot;)

Jumbo_traits &lt;- 
  rbind(state_3_f_line_means, 
        state_3_m_line_means, 
        state_4_f_line_means, 
        state_4_m_line_means, 
        PO_ratio_f_line_means, 
        PO_ratio_m_line_means)</code></pre>
</div>
<div id="grubbs-et-al-2013" class="section level3">
<h3>Grubbs et al 2013</h3>
<pre class="r"><code># Grubbs et al 2013

Grubbs_2013_leg_props &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Grubbs_2013_leg_props.csv&quot;) %&gt;% 
  mutate(Sex = as.factor(Sex),
         Line = as.factor(Line),
         # standardise
         st_Femur = my_scale(Femur),
         st_Tibia = my_scale(Tibia),
         st_Tarsus = my_scale(Tarsus), 
         st_Total = my_scale(Total))

Grubbs_2013_T1_leg_props &lt;- Grubbs_2013_leg_props %&gt;% 
filter(T1_T2 == &quot;T1&quot;)

Grubbs_2013_T2_leg_props &lt;- Grubbs_2013_leg_props %&gt;% 
filter(T1_T2 == &quot;T2&quot;)

# mean total T1 leg length

T1_leg_length_model &lt;- brm(st_Total ~ 1 + Sex + (Sex|Line),
                      data = Grubbs_2013_T1_leg_props, family = gaussian,
                      prior = c(prior(normal(0, 5), class = Intercept),
                                prior(normal(0, 0.5), class = b),
                                prior(exponential(1), class = sd)),
                      iter = 6000, warmup = 2000, chains = 4, cores = 4,
                      seed = 2, file = &quot;fits/raw_data_fits/T1_leg.length.model&quot;, control = list(adapt_delta = 0.95, max_treedepth = 10))

new_data &lt;-
  expand_grid(Line = Grubbs_2013_leg_props$Line,
              Sex = c(&quot;F&quot;, &quot;M&quot;)) %&gt;% 
  distinct(Line, Sex)

fitted_T1_leg_total &lt;- fitted(T1_leg_length_model, newdata = new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Grubbs_2013_T1_leg_props$Total) + mean(Grubbs_2013_T1_leg_props$Total),
         Q2.5 = Q2.5*sd(Grubbs_2013_T1_leg_props$Total) + mean(Grubbs_2013_T1_leg_props$Total),
         Q97.5 = Q97.5*sd(Grubbs_2013_T1_leg_props$Total) + mean(Grubbs_2013_T1_leg_props$Total))

# build a function to create tibbles holding sex specific line means

Grubbs_line_means &lt;- function(estimates, y, Trait, guild, description){
  estimates %&gt;% 
  filter(Sex == y) %&gt;% 
  mutate(line = str_remove(Line, &quot;line_&quot;),
         Trait = Trait,
         Reference = &quot;Grubbs et al (2013) PLOS One&quot;,
         `Trait guild` = guild,
         `Trait description` = description,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)
}

T1_leg_length_estimates &lt;- tibble(new_data, fitted_T1_leg_total) %&gt;% 
  mutate(Sex = if_else(Sex == &quot;F&quot;, &quot;Female&quot;, &quot;Male&quot;))

# T1 females

T1_leg_length_f_line_means &lt;- Grubbs_line_means(T1_leg_length_estimates, &quot;Female&quot;, &quot;T1.leg.length.f&quot;, &quot;Morphological&quot;, &quot;Mean leg length for legs originating on the first thoracic segment. Higher values indicate longer legs&quot;)

# T1 males

T1_leg_length_m_line_means &lt;- Grubbs_line_means(T1_leg_length_estimates, &quot;Male&quot;, &quot;T1.leg.length.m&quot;, &quot;Morphological&quot;, &quot;Mean leg length for legs originating on the first thoracic segment. Higher values indicate longer legs&quot;)

# mean total T2 leg length

T2_leg_length_model &lt;- brm(st_Total ~ 1 + Sex + (Sex|Line),
                      data = Grubbs_2013_T2_leg_props, family = &quot;Gaussian&quot;,
                      prior = c(prior(normal(0, 5), class = Intercept),
                                prior(normal(0, 0.5), class = b),
                                prior(exponential(1), class = sd)),
                      iter = 6000, warmup = 2000, chains = 4, cores = 4,
                      seed = 2, file = &quot;fits/raw_data_fits/T2_leg.length.model&quot;, control = list(adapt_delta = 0.95, max_treedepth = 10))

fitted_T2_leg_total &lt;- fitted(T2_leg_length_model,
       newdata = new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Grubbs_2013_T2_leg_props$Total) + mean(Grubbs_2013_T2_leg_props$Total),
         Q2.5 = Q2.5*sd(Grubbs_2013_T2_leg_props$Total) + mean(Grubbs_2013_T2_leg_props$Total),
         Q97.5 = Q97.5*sd(Grubbs_2013_T2_leg_props$Total) + mean(Grubbs_2013_T2_leg_props$Total))


T2_leg_length_estimates &lt;- tibble(new_data, fitted_T2_leg_total) %&gt;% 
  mutate(Sex = if_else(Sex == &quot;F&quot;, &quot;Female&quot;, &quot;Male&quot;))

# T2 females

T2_leg_length_f_line_means &lt;- Grubbs_line_means(T2_leg_length_estimates, &quot;Female&quot;, &quot;T2.leg.length.f&quot;, &quot;Morphological&quot;, &quot;Mean leg length for legs originating on the second thoracic segment. Higher values indicate longer legs&quot;)

# T2 males

T2_leg_length_m_line_means &lt;- Grubbs_line_means(T2_leg_length_estimates, &quot;Male&quot;, &quot;T2.leg.length.m&quot;, &quot;Morphological&quot;, &quot;Mean leg length for legs originating on the second thoracic segment. Higher values indicate longer legs&quot;)

# Femur length

T1_femur_model &lt;- brm(st_Femur ~ 1 + Sex + (Sex|Line),
                      data = Grubbs_2013_T1_leg_props, family = gaussian,
                      prior = c(prior(normal(0, 5), class = Intercept),
                                prior(normal(0, 0.5), class = b),
                                prior(exponential(1), class = sd)),
                      iter = 6000, warmup = 2000, chains = 4, cores = 4,
                      seed = 2, file = &quot;fits/raw_data_fits/T1.femur.model&quot;, 
                      control = list(adapt_delta = 0.95, max_treedepth = 10))


fitted_T1_femur &lt;- fitted(T1_femur_model,
       newdata = new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Grubbs_2013_T1_leg_props$Femur) + mean(Grubbs_2013_T1_leg_props$Femur),
         Q2.5 = Q2.5*sd(Grubbs_2013_T1_leg_props$Femur) + mean(Grubbs_2013_T1_leg_props$Femur),
         Q97.5 = Q97.5*sd(Grubbs_2013_T1_leg_props$Femur) + mean(Grubbs_2013_T1_leg_props$Femur))

T1_femur_estimates &lt;- tibble(new_data, fitted_T1_femur) %&gt;% 
  mutate(Sex = if_else(Sex == &quot;F&quot;, &quot;Female&quot;, &quot;Male&quot;))

# T1 females

femur_T1_f_line_means &lt;- Grubbs_line_means(T1_femur_estimates, &quot;Female&quot;, &quot;T1.femur.f&quot;, &quot;Morphological&quot;, &quot;Mean femur length, for legs originating on the first thoracic segment. Higher values indicate longer femurs&quot;)

# T1 males

femur_T1_m_line_means &lt;- Grubbs_line_means(T1_femur_estimates, &quot;Male&quot;, &quot;T1.femur.m&quot;, &quot;Morphological&quot;, &quot;Mean femur length, for legs originating on the first thoracic segment. Higher values indicate longer femurs&quot;)

T2_femur_model &lt;- brm(st_Femur ~ 1 + Sex + (Sex|Line),
                      data = Grubbs_2013_T2_leg_props, family = gaussian,
                      prior = c(prior(normal(0, 5), class = Intercept),
                                prior(normal(0, 0.5), class = b),
                                prior(exponential(1), class = sd)),
                      iter = 6000, warmup = 2000, chains = 4, cores = 4,
                      seed = 2, file = &quot;fits/raw_data_fits/T2.femur.model&quot;, 
                      control = list(adapt_delta = 0.95, max_treedepth = 10))


fitted_T2_femur &lt;- fitted(T2_femur_model,
       newdata = new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Grubbs_2013_T2_leg_props$Femur) + mean(Grubbs_2013_T2_leg_props$Femur),
         Q2.5 = Q2.5*sd(Grubbs_2013_T2_leg_props$Femur) + mean(Grubbs_2013_T2_leg_props$Femur),
         Q97.5 = Q97.5*sd(Grubbs_2013_T2_leg_props$Femur) + mean(Grubbs_2013_T2_leg_props$Femur))

T2_femur_estimates &lt;- tibble(new_data, fitted_T2_femur) %&gt;% 
  mutate(Sex = if_else(Sex == &quot;F&quot;, &quot;Female&quot;, &quot;Male&quot;))

# T2 females

femur_T2_f_line_means &lt;- Grubbs_line_means(T2_femur_estimates, &quot;Female&quot;, &quot;T2.femur.f&quot;, &quot;Morphological&quot;, &quot;Mean femur length, for legs originating on the second thoracic segment. Higher values indicate longer femurs&quot;)

# T2 males

femur_T2_m_line_means &lt;- Grubbs_line_means(T2_femur_estimates, &quot;Male&quot;, &quot;T2.femur.m&quot;, &quot;Morphological&quot;, &quot;Mean femur length, for legs originating on the second thoracic segment. Higher values indicate longer femurs&quot;)

# Tibia length

T1_tibia_model &lt;- brm(st_Tibia ~ 1 + Sex + (Sex|Line),
                      data = Grubbs_2013_T1_leg_props, family = gaussian,
                      prior = c(prior(normal(0, 5), class = Intercept),
                                prior(normal(0, 0.5), class = b),
                                prior(exponential(1), class = sd)),
                      iter = 6000, warmup = 2000, chains = 4, cores = 4,
                      seed = 2, file = &quot;fits/raw_data_fits/T1.tibia.model&quot;, 
                      control = list(adapt_delta = 0.95, max_treedepth = 10))

fitted_T1_tibia &lt;- fitted(T1_tibia_model,
       newdata = new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Grubbs_2013_T1_leg_props$Tibia) + mean(Grubbs_2013_T1_leg_props$Tibia),
         Q2.5 = Q2.5*sd(Grubbs_2013_T1_leg_props$Tibia) + mean(Grubbs_2013_T1_leg_props$Tibia),
         Q97.5 = Q97.5*sd(Grubbs_2013_T1_leg_props$Tibia) + mean(Grubbs_2013_T1_leg_props$Tibia))

T1_tibia_estimates &lt;- tibble(new_data, fitted_T1_tibia) %&gt;% 
  mutate(Sex = if_else(Sex == &quot;F&quot;, &quot;Female&quot;, &quot;Male&quot;))

# T1 females

tibia_T1_f_line_means &lt;- Grubbs_line_means(T1_tibia_estimates, &quot;Female&quot;, &quot;T1.tibia.f&quot;, &quot;Morphological&quot;, &quot;Mean tibia length, for legs originating on the first thoracic segment. Higher values indicate longer tibias&quot;)

# T1 males

tibia_T1_m_line_means &lt;- Grubbs_line_means(T1_tibia_estimates, &quot;Male&quot;, &quot;T1.tibia.m&quot;, &quot;Morphological&quot;, &quot;Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias&quot;)

T2_tibia_model &lt;- brm(st_Tibia ~ 1 + Sex + (Sex|Line),
                      data = Grubbs_2013_T2_leg_props, family = gaussian,
                      prior = c(prior(normal(0, 5), class = Intercept),
                                prior(normal(0, 0.5), class = b),
                                prior(exponential(1), class = sd)),
                      iter = 10000, warmup = 2000, chains = 4, cores = 4,
                      seed = 2, file = &quot;fits/raw_data_fits/T2.tibia.model&quot;, 
                      control = list(adapt_delta = 0.95, max_treedepth = 10))


fitted_T2_tibia &lt;- fitted(T2_tibia_model,
       newdata = new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Grubbs_2013_T2_leg_props$Tibia) + mean(Grubbs_2013_T2_leg_props$Tibia),
         Q2.5 = Q2.5*sd(Grubbs_2013_T2_leg_props$Tibia) + mean(Grubbs_2013_T2_leg_props$Tibia),
         Q97.5 = Q97.5*sd(Grubbs_2013_T2_leg_props$Tibia) + mean(Grubbs_2013_T2_leg_props$Tibia))

T2_tibia_estimates &lt;- tibble(new_data, fitted_T2_tibia) %&gt;% 
  mutate(Sex = if_else(Sex == &quot;F&quot;, &quot;Female&quot;, &quot;Male&quot;))

# T2 females

tibia_T2_f_line_means &lt;- Grubbs_line_means(T2_tibia_estimates, &quot;Female&quot;, &quot;T2.tibia.f&quot;, &quot;Morphological&quot;, &quot;Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias&quot;)

# T2 males

tibia_T2_m_line_means &lt;- Grubbs_line_means(T2_tibia_estimates, &quot;Male&quot;, &quot;T2.tibia.m&quot;, &quot;Morphological&quot;, &quot;Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias&quot;)

# Tarsus length

T1_tarsus_model &lt;- brm(st_Tarsus ~ 1 + Sex + (Sex|Line),
                      data = Grubbs_2013_T1_leg_props, family = gaussian,
                      prior = c(prior(normal(0, 5), class = Intercept),
                                prior(normal(0, 0.5), class = b),
                                prior(exponential(1), class = sd)),
                      iter = 6000, warmup = 2000, chains = 4, cores = 4,
                      seed = 2, file = &quot;fits/raw_data_fits/T1.tarsus.model&quot;, 
                      control = list(adapt_delta = 0.95, max_treedepth = 10))

fitted_T1_tarsus &lt;- fitted(T1_tarsus_model,
       newdata = new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Grubbs_2013_T1_leg_props$Tarsus) + mean(Grubbs_2013_T1_leg_props$Tarsus),
         Q2.5 = Q2.5*sd(Grubbs_2013_T1_leg_props$Tarsus) + mean(Grubbs_2013_T1_leg_props$Tarsus),
         Q97.5 = Q97.5*sd(Grubbs_2013_T1_leg_props$Tarsus) + mean(Grubbs_2013_T1_leg_props$Tarsus))

T1_tarsus_estimates &lt;- tibble(new_data, fitted_T1_tarsus) %&gt;% 
  mutate(Sex = if_else(Sex == &quot;F&quot;, &quot;Female&quot;, &quot;Male&quot;))

# T1 females

tarsus_T1_f_line_means &lt;- Grubbs_line_means(T1_tarsus_estimates, &quot;Female&quot;, &quot;T1.tarsus.f&quot;, &quot;Morphological&quot;, &quot;Mean tarsus length, for legs originating on the first thoracic segment. Higher values indicate longer tarsus&#39;.&quot;)

# T1 males

tarsus_T1_m_line_means &lt;- Grubbs_line_means(T1_tarsus_estimates, &quot;Male&quot;, &quot;T1.tarsus.m&quot;, &quot;Morphological&quot;, &quot;Mean tarsus length, for legs originating on the first thoracic segment. Higher values indicate longer tarsus&#39;.&quot;)

T2_tarsus_model &lt;- brm(st_Tarsus ~ 1 + Sex + (Sex|Line),
                      data = Grubbs_2013_T2_leg_props, family = gaussian,
                      prior = c(prior(normal(0, 5), class = Intercept),
                                prior(normal(0, 0.5), class = b),
                                prior(exponential(1), class = sd)),
                      iter = 6000, warmup = 2000, chains = 4, cores = 4,
                      seed = 2, file = &quot;fits/raw_data_fits/T2.tarsus.model&quot;, 
                      control = list(adapt_delta = 0.95, max_treedepth = 10))

fitted_T2_tarsus &lt;- fitted(T2_tarsus_model,
       newdata = new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Grubbs_2013_T2_leg_props$Tarsus) + mean(Grubbs_2013_T2_leg_props$Tarsus),
         Q2.5 = Q2.5*sd(Grubbs_2013_T2_leg_props$Tarsus) + mean(Grubbs_2013_T2_leg_props$Tarsus),
         Q97.5 = Q97.5*sd(Grubbs_2013_T2_leg_props$Tarsus) + mean(Grubbs_2013_T2_leg_props$Tarsus))

T2_tarsus_estimates &lt;- tibble(new_data, fitted_T2_tarsus) %&gt;% 
  mutate(Sex = if_else(Sex == &quot;F&quot;, &quot;Female&quot;, &quot;Male&quot;))

# T2 females

tarsus_T2_f_line_means &lt;- Grubbs_line_means(T2_tarsus_estimates, &quot;Female&quot;, &quot;T2.tarsus.f&quot;, &quot;Morphological&quot;, &quot;Mean tarsus length, for legs originating on the second thoracic segment. Higher values indicate longer tarsus&#39;.&quot;)

# T2 males

tarsus_T2_m_line_means &lt;- Grubbs_line_means(T2_tarsus_estimates, &quot;Male&quot;, &quot;T2.tarsus.m&quot;, &quot;Morphological&quot;, &quot;Mean tarsus length, for legs originating on the second thoracic segment. Higher values indicate longer tarsus&#39;.&quot;)

Grubbs_traits &lt;- rbind(T1_leg_length_f_line_means, 
                       T1_leg_length_m_line_means, 
                       T2_leg_length_f_line_means, 
                       T2_leg_length_m_line_means,
                       femur_T1_f_line_means, 
                       femur_T1_m_line_means, 
                       femur_T2_f_line_means, 
                       femur_T2_m_line_means,
                       tibia_T1_f_line_means, 
                       tibia_T1_m_line_means, 
                       tibia_T2_f_line_means, 
                       tibia_T2_m_line_means,
                       tarsus_T1_f_line_means, 
                       tarsus_T1_m_line_means, 
                       tarsus_T2_f_line_means, 
                       tarsus_T2_m_line_means)</code></pre>
</div>
<div id="dobson-et-al-2015" class="section level3">
<h3>Dobson et al 2015</h3>
<p>From <strong>Supplementary Table 1</strong>’s caption: “In the second
approach (c), simpler models were fitted, in which microbiota, weight
and Wolbachia status were additive, with genotype nested in experimental
block as a random effect. The statistical analysis presented in Fig.1b
uses these models simplified by removal of non-significant factors.”</p>
<p>We follow this approach to best recapitulate the study’s analysis,
except that we allow the intercept for the random effect line to vary
with microbiota treatment. This is needed in our modelling approach to
estimate line-specific effects of microbiota. We also include day as a
blocking factor, which represents a block of data collection involving a
subset of lines. This controls for micro-environmental variance between
blocks of phenotyping.</p>
<pre class="r"><code># weight

Dobson_NI_microbiome.dryweight &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Dobson_2015/gwasB.filtered.dw.ab.csv&quot;) %&gt;% 
  mutate(line = as.factor(str_remove(RAL, &quot;RAL_&quot;)),
         wolb = as.factor(wolb),
         trt = as.factor(trt),
         day = as.factor(day),
         # standardise
         st_weight.5.flies = my_scale(weight.5.flies)) %&gt;% 
  select(-c(PC1, PC2, PC3, Code, RAL, ...1))

# The data is a bit tricky. Not all lines have been measured in both treatments. Below we look to see first which lines were measured in treatments a and b, then what lines were measured in both. We need this when we estimate the mean difference between treatments for each line

a &lt;- Dobson_NI_microbiome.dryweight %&gt;% filter(trt == &quot;a&quot;) %&gt;% distinct(line)
b &lt;- Dobson_NI_microbiome.dryweight %&gt;% filter(trt == &quot;b&quot;) %&gt;% distinct(line)

# inner_join selects lines that have measurements in both treatments 

weight_microbiome_effect_lines &lt;- inner_join(a, b)

# build a function to create tibbles holding specific line means

Dobson_line_means &lt;- function(estimates, Trait, guild, description){
  estimates %&gt;% 
  mutate(Trait = Trait,
         Sex = &quot;Male&quot;,
         Reference = &quot;Dobson et al (2015) Nature Communications&quot;,
          `Trait guild` = guild,
         `Trait description` = description,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)
}

weight_microbiome_model &lt;- brm(st_weight.5.flies ~ 1 + trt + wolb + day + (trt|line),
                               data = Dobson_NI_microbiome.dryweight, family = gaussian,
                               prior = c(prior(normal(0, 0.5), class = Intercept),
                                         prior(normal(0, 0.5), class = b),
                                         prior(exponential(1), class = sd)),
                               iter = 8000, warmup = 2000, chains = 4, cores = 4,
                               seed = 2, file = &quot;fits/raw_data_fits/weight.microbiome.model&quot;)

#pp_check(weight_microbiome_model)

new_data &lt;-
  tibble(line = Dobson_NI_microbiome.dryweight$line,
         trt = Dobson_NI_microbiome.dryweight$trt,
         wolb = &quot;n&quot;,
         day = &quot;a&quot;) %&gt;% 
  distinct(line, trt, wolb, day)
              

fitted_dryweight &lt;- fitted(weight_microbiome_model, newdata = new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Dobson_NI_microbiome.dryweight$weight.5.flies) + mean(Dobson_NI_microbiome.dryweight$weight.5.flies),
         Q2.5 = Q2.5*sd(Dobson_NI_microbiome.dryweight$weight.5.flies) + mean(Dobson_NI_microbiome.dryweight$weight.5.flies),
         Q97.5 = Q97.5*sd(Dobson_NI_microbiome.dryweight$weight.5.flies) + mean(Dobson_NI_microbiome.dryweight$weight.5.flies))

dry_weight_estimates_axenic &lt;- tibble(new_data, fitted_dryweight) %&gt;% 
  filter(trt == &quot;a&quot;)

weight.anexic_line_means &lt;- Dobson_line_means(dry_weight_estimates_axenic, &quot;weight.anexic.m&quot;, &quot;Microbiome&quot;, &quot;Mean combined dry weight of 5 adult flies that lacked a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate heavier flies.&quot;)

dry_weight_estimates_gnotobiotic &lt;- tibble(new_data, fitted_dryweight) %&gt;% 
  filter(trt == &quot;b&quot;)

weight.gnotobiotic_line_means &lt;- Dobson_line_means(dry_weight_estimates_gnotobiotic, &quot;weight.gnotobiotic.m&quot;, &quot;Microbiome&quot;, &quot;Mean combined dry weight of 5 adult flies that possessed a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate heavier flies.&quot;)

# Now get the difference between treatments

axenic &lt;- 
  left_join(weight_microbiome_effect_lines, dry_weight_estimates_axenic) %&gt;% 
  select(line, Estimate) %&gt;% 
  rename(axenic_estimate = Estimate)

gnoto &lt;-
  left_join(weight_microbiome_effect_lines, dry_weight_estimates_gnotobiotic) %&gt;% 
  select(line, Estimate) %&gt;% 
  rename(gnoto_estimate = Estimate)

dry_weight_estimates_diff &lt;-
  left_join(axenic, gnoto) %&gt;% 
  mutate(Estimate = gnoto_estimate - axenic_estimate)

gnoto_anexic_diff_weight_line_means &lt;- Dobson_line_means(dry_weight_estimates_diff, &quot;weight.microbiome.effect.m&quot;, &quot;Microbiome&quot;, &quot;Mean change in the combined dry weight of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota had a greater positive effect on weight. Note that the presence of micriobiota on average increased dry weight.&quot;)


# Glucose

Dobson_glucose_microbiome &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Dobson_2015/gwasB.filtered.glucose.ab.csv&quot;) %&gt;% 
  mutate(line = as.factor(str_remove(RAL, &quot;RAL_&quot;)),
         wolb = as.factor(wolb),
         trt = as.factor(trt),
         day = as.factor(day),
         # standardise
         st_glucose = my_scale(glucose)) %&gt;% 
  select(c(line, wolb, glucose, trt, day, weight.5.flies, st_glucose))

# The data is a bit tricky. Not all lines have been measured in both treatments. Below we look to see first which lines were measured in treatments a and b, then what lines were measured in both. We need this when we estimate the mean difference between treatments for each line

a &lt;- Dobson_glucose_microbiome %&gt;% filter(trt == &quot;a&quot;) %&gt;% distinct(line)
b &lt;- Dobson_glucose_microbiome %&gt;% filter(trt == &quot;b&quot;) %&gt;% distinct(line)

# inner_join selects lines that have measures for both treatments

glucose_microbiome_effect_lines &lt;- inner_join(a, b)

# Note that we include weight as a moderator variable. This means we only estimate the direct effect of line on glucose levels, instead of the total effect. This is because line likely also affects weight which is likely to have a flow-on effect on glucose levels. 

glucose_microbiome_model &lt;- brm(st_glucose ~ 1 + trt + wolb + day + weight.5.flies + (trt|line),
                               data = Dobson_glucose_microbiome, family = gaussian,
                               prior = c(prior(normal(0, 0.5), class = Intercept),
                                         prior(normal(0, 0.5), class = b),
                                         prior(exponential(1), class = sd)),
                               iter = 8000, warmup = 2000, chains = 4, cores = 4,
                               seed = 2, file = &quot;fits/raw_data_fits/glucose.microbiome.model&quot;)

#pp_check(glucose_microbiome_model)

new_data &lt;-
  tibble(line = Dobson_glucose_microbiome$line,
         trt = Dobson_glucose_microbiome$trt,
         wolb = &quot;n&quot;,
         day = &quot;a&quot;,
         weight.5.flies = 1) %&gt;% 
  distinct(line, trt, wolb, day, weight.5.flies)
              

fitted_glucose &lt;- fitted(glucose_microbiome_model,
       newdata = new_data, allow_new_levels = TRUE) %&gt;% # Why do we need to allow new levels? Might be an issue
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Dobson_glucose_microbiome$glucose) + mean(Dobson_glucose_microbiome$glucose),
         Q2.5 = Q2.5*sd(Dobson_glucose_microbiome$glucose) + mean(Dobson_glucose_microbiome$glucose),
         Q97.5 = Q97.5*sd(Dobson_glucose_microbiome$glucose) + mean(Dobson_glucose_microbiome$glucose))

glucose_estimates_axenic &lt;- tibble(new_data, fitted_glucose) %&gt;% 
  filter(trt == &quot;a&quot;)

glucose.anexic_line_means &lt;- Dobson_line_means(glucose_estimates_axenic, &quot;glucose.anexic.m&quot;, &quot;Microbiome&quot;, &quot;Mean combined glucose content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher glucose content.&quot;)

glucose_estimates_gnotobiotic &lt;- tibble(new_data, fitted_glucose) %&gt;% 
  filter(trt == &quot;b&quot;)

glucose.gnotobiotic_line_means &lt;- Dobson_line_means(glucose_estimates_gnotobiotic, &quot;glucose.gnotobiotic.m&quot;, &quot;Microbiome&quot;, &quot;Mean combined glucose content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher glucose content.&quot;)

# Now get the difference between treatments

axenic_glucose &lt;- 
  left_join(glucose_microbiome_effect_lines, glucose_estimates_axenic) %&gt;% 
  select(line, Estimate) %&gt;% 
  rename(axenic_estimate = Estimate)

gnoto_glucose &lt;-
  left_join(glucose_microbiome_effect_lines, glucose_estimates_gnotobiotic) %&gt;% 
  select(line, Estimate) %&gt;% 
  rename(gnoto_estimate = Estimate)

glucose_estimates_diff &lt;-
  left_join(axenic_glucose, gnoto_glucose) %&gt;% 
  mutate(Estimate = gnoto_estimate - axenic_estimate)

gnoto_anexic_diff_glucose_line_means &lt;- Dobson_line_means(glucose_estimates_diff, &quot;glucose.microbiome.effect.m&quot;, &quot;Microbiome&quot;, &quot;Mean change in the combined glucose content of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher glucose content. Note though that the presence of micriobiota on average decreased glucose content.&quot;)


# Glycogen

Dobson_glycogen_microbiome &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Dobson_2015/gwasB.filtered.glycogen.ab.csv&quot;) %&gt;% 
  mutate(line = as.factor(str_remove(RAL, &quot;RAL_&quot;)),
         wolb = as.factor(wolb),
         trt = as.factor(trt),
         day = as.factor(day),
         # standardise
         st_glycogen = my_scale(glycogen)) %&gt;% 
  select(c(line, wolb, glycogen, st_glycogen, trt, day, weight.5.flies))

# The data is a bit tricky. Not all lines have been measured in both treatments. Below we look to see first which lines were measured in treatments a and b, then what lines were measured in both. We need this when we estimate the mean difference between treatments for each line

a &lt;- Dobson_glycogen_microbiome %&gt;% filter(trt == &quot;a&quot;) %&gt;% distinct(line)
b &lt;- Dobson_glycogen_microbiome %&gt;% filter(trt == &quot;b&quot;) %&gt;% distinct(line)

# inner_join selects lines that have measures for both treatments

glycogen_microbiome_effect_lines &lt;- inner_join(a, b)

glycogen_microbiome_model &lt;- brm(st_glycogen ~ 1 + trt + wolb + day + weight.5.flies + (trt|line),
                               data = Dobson_glycogen_microbiome, family = gaussian,
                               prior = c(prior(normal(0, 0.5), class = Intercept),
                                         prior(normal(0, 0.5), class = b),
                                         prior(exponential(1), class = sd)),
                               iter = 8000, warmup = 2000, chains = 4, cores = 4,
                               seed = 2, file = &quot;fits/raw_data_fits/glycogen.microbiome.model&quot;)

#pp_check(glycogen_microbiome_model)

new_data &lt;-
  tibble(line = Dobson_glycogen_microbiome$line,
         trt = Dobson_glycogen_microbiome$trt,
         wolb = &quot;n&quot;,
         day = &quot;a&quot;,
         weight.5.flies = 1) %&gt;% 
  distinct(line, trt, wolb, day, weight.5.flies)
              

fitted_glycogen &lt;- fitted(glycogen_microbiome_model,
       newdata = new_data) %&gt;%
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Dobson_glycogen_microbiome$glycogen) + mean(Dobson_glycogen_microbiome$glycogen),
         Q2.5 = Q2.5*sd(Dobson_glycogen_microbiome$glycogen) + mean(Dobson_glycogen_microbiome$glycogen),
         Q97.5 = Q97.5*sd(Dobson_glycogen_microbiome$glycogen) + mean(Dobson_glycogen_microbiome$glycogen))

glycogen_estimates_axenic &lt;- tibble(new_data, fitted_glycogen) %&gt;% 
  filter(trt == &quot;a&quot;)

glycogen.anexic_line_means &lt;- Dobson_line_means(glycogen_estimates_axenic, &quot;glycogen.anexic.m&quot;, &quot;Microbiome&quot;, &quot;Mean combined glycogen content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher glycogen content.&quot;)

glycogen_estimates_gnotobiotic &lt;- tibble(new_data, fitted_glycogen) %&gt;% 
  filter(trt == &quot;b&quot;)

glycogen.gnotobiotic_line_means &lt;- Dobson_line_means(glycogen_estimates_gnotobiotic, &quot;glycogen.gnotobiotic.m&quot;, &quot;Microbiome&quot;,  &quot;Mean combined glycogen content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher glycogen content.&quot;)

# Now get the difference between treatments

axenic_glycogen &lt;- 
  left_join(glycogen_microbiome_effect_lines, glycogen_estimates_axenic) %&gt;% 
  select(line, Estimate) %&gt;% 
  rename(axenic_estimate = Estimate)

gnoto_glycogen &lt;-
  left_join(glycogen_microbiome_effect_lines, glycogen_estimates_gnotobiotic) %&gt;% 
  select(line, Estimate) %&gt;% 
  rename(gnoto_estimate = Estimate)

glycogen_estimates_diff &lt;-
  left_join(axenic_glycogen, gnoto_glycogen) %&gt;% 
  mutate(Estimate = gnoto_estimate - axenic_estimate)

gnoto_anexic_diff_glycogen_line_means &lt;- Dobson_line_means(glycogen_estimates_diff, &quot;glycogen.microbiome.effect.m&quot;, &quot;Microbiome&quot;, &quot;Mean change in the combined glycogen content of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher glycogen content. Note that the presence of micriobiota on average increased glycogen content.&quot;)

# Protein

Dobson_protein_microbiome &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Dobson_2015/gwasB.filtered.protein.ab.csv&quot;) %&gt;% 
  mutate(line = as.factor(str_remove(RAL, &quot;RAL_&quot;)),
         wolb = as.factor(wolb),
         trt = as.factor(trt),
         day = as.factor(day),
         # standardise
         st_protein = my_scale(protein)) %&gt;% 
  select(c(line, wolb, protein, st_protein, trt, day, weight.5.flies))

# The data is a bit tricky. Not all lines have been measured in both treatments. Below we look to see first which lines were measured in treatments a and b, then what lines were measured in both. We need this when we estimate the mean difference between treatments for each line

a &lt;- Dobson_protein_microbiome %&gt;% filter(trt == &quot;a&quot;) %&gt;% distinct(line)
b &lt;- Dobson_protein_microbiome %&gt;% filter(trt == &quot;b&quot;) %&gt;% distinct(line)

# inner_join selects lines that have measures for both treatments

protein_microbiome_effect_lines &lt;- inner_join(a, b)

protein_microbiome_model &lt;- brm(st_protein ~ 1 + trt + wolb + day + weight.5.flies + (trt|line),
                               data = Dobson_protein_microbiome, family = gaussian,
                               prior = c(prior(normal(0, 0.5), class = Intercept),
                                         prior(normal(0, 0.5), class = b),
                                         prior(exponential(1), class = sd)),
                               iter = 8000, warmup = 2000, chains = 4, cores = 4,
                               seed = 2, file = &quot;fits/raw_data_fits/protein.microbiome.model&quot;)

#pp_check(protein_microbiome_model)

new_data &lt;-
  tibble(line = Dobson_protein_microbiome$line,
         trt = Dobson_protein_microbiome$trt,
         wolb = &quot;n&quot;,
         day = &quot;a&quot;,
         weight.5.flies = 1) %&gt;% 
  distinct(line, trt, wolb, day, weight.5.flies)
              

fitted_protein &lt;- fitted(protein_microbiome_model,
       newdata = new_data) %&gt;%
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Dobson_protein_microbiome$protein) + mean(Dobson_protein_microbiome$protein),
         Q2.5 = Q2.5*sd(Dobson_protein_microbiome$protein) + mean(Dobson_protein_microbiome$protein),
         Q97.5 = Q97.5*sd(Dobson_protein_microbiome$protein) + mean(Dobson_protein_microbiome$protein))

protein_estimates_axenic &lt;- tibble(new_data, fitted_protein) %&gt;% 
  filter(trt == &quot;a&quot;)

protein.anexic_line_means &lt;- Dobson_line_means(protein_estimates_axenic, &quot;protein.anexic.m&quot;, &quot;Microbiome&quot;, &quot;Mean combined protein content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher protein content.&quot;)

protein_estimates_gnotobiotic &lt;- tibble(new_data, fitted_protein) %&gt;% 
  filter(trt == &quot;b&quot;)

protein.gnotobiotic_line_means &lt;- Dobson_line_means(protein_estimates_gnotobiotic, &quot;protein.gnotobiotic.m&quot;, &quot;Microbiome&quot;, &quot;Mean combined protein content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher protein content.&quot;)

# Now get the difference between treatments

axenic_protein &lt;- 
  left_join(protein_microbiome_effect_lines, protein_estimates_axenic) %&gt;% 
  select(line, Estimate) %&gt;% 
  rename(axenic_estimate = Estimate)

gnoto_protein &lt;-
  left_join(protein_microbiome_effect_lines, protein_estimates_gnotobiotic) %&gt;% 
  select(line, Estimate) %&gt;% 
  rename(gnoto_estimate = Estimate)

protein_estimates_diff &lt;-
  left_join(axenic_protein, gnoto_protein) %&gt;% 
  mutate(Estimate = gnoto_estimate - axenic_estimate)

gnoto_anexic_diff_protein_line_means &lt;- Dobson_line_means(protein_estimates_diff, &quot;protein.microbiome.effect.m&quot;, &quot;Microbiome&quot;, &quot;Mean change in the combined protein content of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher protein content. Note that the presence of micriobiota on average increased protein content.&quot;)

# TAG glycerol

Dobson_TAG_glycerol_microbiome &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Dobson_2015/gwasB.filtered.TAG_glycerol.ab.csv&quot;) %&gt;% 
  mutate(line = as.factor(str_remove(RAL, &quot;RAL_&quot;)),
         #wolb = as.factor(wolb),
         trt = as.factor(trt),
         day = as.factor(day),
         # standardise
         st_TAG_glycerol = my_scale(TAG_glycerol)) %&gt;% 
  select(c(line, TAG_glycerol, st_TAG_glycerol, trt, day, weight.5.flies))

# The data is a bit tricky. Not all lines have been measured in both treatments. Below we look to see first which lines were measured in treatments a and b, then what lines were measured in both. We need this when we estimate the mean difference between treatments for each line

a &lt;- Dobson_TAG_glycerol_microbiome %&gt;% filter(trt == &quot;a&quot;) %&gt;% distinct(line)
b &lt;- Dobson_TAG_glycerol_microbiome %&gt;% filter(trt == &quot;b&quot;) %&gt;% distinct(line)

# inner_join selects lines that have measures for both treatments

TAG_glycerol_microbiome_effect_lines &lt;- inner_join(a, b)

TAG_glycerol_microbiome_model &lt;- brm(st_TAG_glycerol ~ 1 + trt + day + weight.5.flies + (trt|line),
                               data = Dobson_TAG_glycerol_microbiome, family = gaussian,
                               prior = c(prior(normal(0, 0.5), class = Intercept),
                                         prior(normal(0, 0.5), class = b),
                                         prior(exponential(1), class = sd)),
                               iter = 8000, warmup = 2000, chains = 4, cores = 4,
                               seed = 2, file = &quot;fits/raw_data_fits/TAG_glycerol.microbiome.model&quot;)

#pp_check(TAG_glycerol_microbiome_model)

new_data &lt;-
  tibble(line = Dobson_TAG_glycerol_microbiome$line,
         trt = Dobson_TAG_glycerol_microbiome$trt,
         day = &quot;a&quot;,
         weight.5.flies = 1) %&gt;% 
  distinct(line, trt, day, weight.5.flies)
              

fitted_TAG_glycerol &lt;- fitted(TAG_glycerol_microbiome_model,
                              newdata = new_data) %&gt;%
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Dobson_TAG_glycerol_microbiome$TAG_glycerol) + mean(Dobson_TAG_glycerol_microbiome$TAG_glycerol),
         Q2.5 = Q2.5*sd(Dobson_TAG_glycerol_microbiome$TAG_glycerol) + mean(Dobson_TAG_glycerol_microbiome$TAG_glycerol),
         Q97.5 = Q97.5*sd(Dobson_TAG_glycerol_microbiome$TAG_glycerol) + mean(Dobson_TAG_glycerol_microbiome$TAG_glycerol))

TAG_glycerol_estimates_axenic &lt;- tibble(new_data, fitted_TAG_glycerol) %&gt;% 
  filter(trt == &quot;a&quot;)

TAG_glycerol.anexic_line_means &lt;- Dobson_line_means(TAG_glycerol_estimates_axenic, &quot;triglyceride.anexic.m&quot;, &quot;Microbiome&quot;, &quot;Mean combined triglyceride (TAG) content after glycerol substraction extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher TAG content.&quot;)

TAG_glycerol_estimates_gnotobiotic &lt;- tibble(new_data, fitted_TAG_glycerol) %&gt;% 
  filter(trt == &quot;b&quot;)

TAG_glycerol.gnotobiotic_line_means &lt;- Dobson_line_means(TAG_glycerol_estimates_gnotobiotic, &quot;triglyceride.gnotobiotic.m&quot;, &quot;Microbiome&quot;, &quot;Mean combined triglyceride (TAG) content after glycerol substraction extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher TAG content.&quot;)

# Now get the difference between treatments

axenic_TAG_glycerol &lt;- 
  left_join(TAG_glycerol_microbiome_effect_lines, TAG_glycerol_estimates_axenic) %&gt;% 
  select(line, Estimate) %&gt;% 
  rename(axenic_estimate = Estimate)

gnoto_TAG_glycerol &lt;-
  left_join(TAG_glycerol_microbiome_effect_lines, TAG_glycerol_estimates_gnotobiotic) %&gt;% 
  select(line, Estimate) %&gt;% 
  rename(gnoto_estimate = Estimate)

TAG_glycerol_estimates_diff &lt;-
  left_join(axenic_TAG_glycerol, gnoto_TAG_glycerol) %&gt;% 
  mutate(Estimate = gnoto_estimate - axenic_estimate)

gnoto_anexic_diff_TAG_glycerol_line_means &lt;- Dobson_line_means(TAG_glycerol_estimates_diff, &quot;TAG.glycerol.microbiome.effect.m&quot;, &quot;Microbiome&quot;, &quot;Mean change in combined triglyceride content after glycerol substraction of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher TAG content. Note though that the presence of micriobiota on average decreased TAG content.&quot;)

Dobson_traits &lt;- rbind(weight.anexic_line_means, 
                       weight.gnotobiotic_line_means, 
                       gnoto_anexic_diff_weight_line_means,
                       glucose.anexic_line_means, 
                       glucose.gnotobiotic_line_means, 
                       gnoto_anexic_diff_glucose_line_means,
                       glycogen.anexic_line_means, 
                       glycogen.gnotobiotic_line_means, 
                       gnoto_anexic_diff_glycogen_line_means,
                       protein.anexic_line_means, 
                       protein.gnotobiotic_line_means, 
                       gnoto_anexic_diff_protein_line_means,
                       TAG_glycerol.anexic_line_means, 
                       TAG_glycerol.gnotobiotic_line_means, 
                       gnoto_anexic_diff_TAG_glycerol_line_means)</code></pre>
</div>
<div id="gaertner-et-al-2015" class="section level3">
<h3>Gaertner et al 2015</h3>
<pre class="r"><code>Gaertner_2015_male_mating &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Gaertner_2015_male_mating.csv&quot;) %&gt;% 
  select(Block, Day, Assay, Chamber, Line, Subject, SUM, orient_female, approach, wing, genital_lick, attempt_cop, copulate, sum_court) %&gt;%
  rename(MMP = SUM) %&gt;% 
  mutate(Block = as.factor(Block),
         line = as.factor(Line),
         courtship_events = orient_female + approach + wing + genital_lick + attempt_cop,
         mating_latency = 31 - copulate, # we subtract from 31 to tell the model that flies did not start the assay mating 
         censor = if_else(mating_latency == 31, 1, 0))

# build a function to create tibbles holding specific line means

Gaertner_line_means &lt;- function(estimates, Trait, guild, description){
  estimates %&gt;% 
  mutate(Trait = Trait,
         Sex = &quot;Male&quot;,
         Reference = &quot;Gaertner et al (2015) G3&quot;,
          `Trait guild` = guild,
         `Trait description` = description,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)
}

# We will model the 5 courtship processes individually (note that the courtship data is very zero inflated)

# orientation towards females

orient_model &lt;- brm(orient_female ~ 1 + Block + (1|line),
                    data = Gaertner_2015_male_mating, family = zero_inflated_poisson,
                    prior = c(prior(normal(3, 0.5), class = Intercept),
                              prior(normal(0, 0.2), class = b),
                              prior(cauchy(0, 0.2), class = sd),
                              prior(beta(2, 6), class = zi)),  # the brms default is beta(1, 1)), our specified prior is more sceptical of extreme values
                    iter = 6000, warmup = 2000, chains = 4, cores = 4,
                    seed = 2, file = &quot;fits/raw_data_fits/orient.female.model&quot;,
                    control = list(adapt_delta = 0.8, max_treedepth = 12))

new_data &lt;-
  tibble(line = Gaertner_2015_male_mating$Line,
         Block = 1) %&gt;% 
  distinct(line, Block)
              

fitted_orient &lt;- fitted(orient_model,
       newdata = new_data,
       re_formula = orient_female ~ (1|line)) %&gt;% 
  as_tibble()

orient_to_female_m_estimates &lt;- tibble(new_data, fitted_orient) 

orient_to_female_m_line_means &lt;- Gaertner_line_means(orient_to_female_m_estimates, &quot;orient.towards.female.m&quot;, &quot;Behavioural&quot;, &quot;Mean number of times a DGRP male orientated towards a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of orientations, and potentially more persistent courtship.&quot;)

# approaching females

approach_model &lt;- brm(approach ~ 1 + Block + (1|line),
                      data = Gaertner_2015_male_mating, family = zero_inflated_poisson,
                      prior = c(prior(normal(3, 0.5), class = Intercept),
                                prior(normal(0, 0.2), class = b),
                                prior(cauchy(0, 0.2), class = sd),
                                prior(beta(2, 6), class = zi)),  # the brms default is beta(1, 1)), our specified prior is more sceptical of extreme values
                      iter = 6000, warmup = 2000, chains = 4, cores = 4,
                      seed = 2, file = &quot;fits/raw_data_fits/approach.female.model&quot;,
                      control = list(adapt_delta = 0.8, max_treedepth = 10))

fitted_approach &lt;- fitted(approach_model,
       newdata = new_data,
       re_formula = approach ~ (1|line)) %&gt;% 
  as_tibble()

approach_female_m_estimates &lt;- tibble(new_data, fitted_approach) 

approach_female_m_line_means &lt;- Gaertner_line_means(approach_female_m_estimates, &quot;approach.female.m&quot;, &quot;Behavioural&quot;, &quot;Mean number of times a DGRP male approached a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of approaches, and potentially more persistent courtship.&quot;)

# wing display (singing)

wing_display_model &lt;- brm(wing ~ 1 + Block + (1|line),
                          data = Gaertner_2015_male_mating, family = zero_inflated_poisson(),
                          prior = c(prior(normal(3, 0.5), class = Intercept),
                                    prior(normal(0, 0.2), class = b),
                                    prior(cauchy(0, 0.2), class = sd),
                                    prior(beta(2, 6), class = zi)),  # the brms default is beta(1, 1)), our specified prior is more sceptical of extreme values
                          iter = 6000, warmup = 2000, chains = 4, cores = 4,
                          seed = 2, file = &quot;fits/raw_data_fits/wing.display.model&quot;,
                          control = list(adapt_delta = 0.8, max_treedepth = 10))

fitted_wing_display &lt;- fitted(wing_display_model,
       newdata = new_data,
       re_formula = wing ~ (1|line)) %&gt;% 
  as_tibble()

wing_display_m_estimates &lt;- tibble(new_data, fitted_wing_display) 

wing_display_frequency_m_line_means &lt;- Gaertner_line_means(wing_display_m_estimates, &quot;wing.display.frequency.m&quot;, &quot;Behavioural&quot;, &quot;Mean number of times a DGRP male extended their wing at a 90 degree angle in rapid vibration while in the presence of a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of wing displays, and potentially more persistent courtship.&quot;)

# genital licking

genital_licking_model &lt;- brm(genital_lick ~ 1 + Block + (1|line),
                             data = Gaertner_2015_male_mating, family = zero_inflated_poisson(),
                             prior = c(prior(normal(3, 0.5), class = Intercept),
                                       prior(normal(0, 0.2), class = b),
                                       prior(cauchy(0, 0.2), class = sd),
                                       prior(beta(2, 6), class = zi)),  # the brms default is beta(1, 1)), our specified prior is more sceptical of extreme values
                             iter = 6000, warmup = 2000, chains = 4, cores = 4,
                             seed = 2, file = &quot;fits/raw_data_fits/genital.lick.model&quot;,
                             control = list(adapt_delta = 0.8, max_treedepth = 10))

fitted_genital_licking &lt;- fitted(genital_licking_model,
       newdata = new_data,
       re_formula = genital_lick ~ (1|line)) %&gt;% 
  as_tibble()

genital_licking_m_estimates &lt;- tibble(new_data, fitted_genital_licking) 

genital_licking_m_line_means &lt;- Gaertner_line_means(genital_licking_m_estimates, &quot;genital.licking.frequency.m&quot;, &quot;Behavioural&quot;, &quot;Mean number of genital licks by DGRP male while courting a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater frequency of genital licking, and potentially more persistent courtship.&quot;)

# attempted copulations

attempted_copulations_model &lt;- brm(attempt_cop ~ 1 + Block + (1|line),
                                   data = Gaertner_2015_male_mating, family = zero_inflated_poisson(),
                                   prior = c(prior(normal(3, 0.5), class = Intercept),
                                             prior(normal(0, 0.2), class = b),
                                             prior(cauchy(0, 0.2), class = sd),
                                             prior(beta(2, 6), class = zi)),  # the brms default is beta(1, 1)), our specified prior is more sceptical of extreme values
                                   iter = 6000, warmup = 2000, chains = 4, cores = 4,
                                   seed = 2, file = &quot;fits/raw_data_fits/attempt.cop.model&quot;,
                                   control = list(adapt_delta = 0.9, max_treedepth = 10))

fitted_attempted_copulations &lt;- 
  fitted(attempted_copulations_model,
         newdata = new_data) %&gt;% 
  as_tibble()

attempted_copulations_m_estimates &lt;- tibble(new_data, fitted_attempted_copulations) 

attempted_copulations_m_line_means &lt;- Gaertner_line_means(attempted_copulations_m_estimates, &quot;attempted.copulation.frequency.m&quot;, &quot;Behavioural&quot;, &quot;Mean number of attempted copulations by DGRP male while courting a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater frequency of copulation attempts, and potentially more persistent courtship.&quot;)

# Mating latency (like a no choice trial)

# The data is right censored for males that did not mate

mating_latency_model &lt;-
  brm(mating_latency | cens(censor) ~ 1 + Block + (1|line),
      data = Gaertner_2015_male_mating, 
      family = weibull(), inits = 0,
      prior = c(prior(normal(0, 1), class = Intercept), # the intercept in this model is the rate of decline in un-mated males
                prior(exponential(1), class = sd)),
      iter = 8000, warmup = 2000, chains = 4, cores = 4,
      seed = 2, file = &quot;fits/raw_data_fits/mating.latency.model&quot;,
      control = list(adapt_delta = 0.9, max_treedepth = 10))

fitted_mating_latency &lt;- fitted(mating_latency_model,
       newdata = new_data,
       re_formula = mating_latency ~ (1|line)) %&gt;% 
  as_tibble()

mating_latency_m_estimates &lt;- tibble(new_data, fitted_mating_latency) 

mating_latency_m_line_means &lt;- Gaertner_line_means(mating_latency_m_estimates, &quot;mating.latency.2015.m&quot;, &quot;Behavioural&quot;, &quot;Mean latency until a DGRP male was observed copulating with a virgin female, detected by scan sampling 30 times over a 15 minute period. Lower values indicate males that began mating faster, which could be interpreted as males that are more attractive to females.&quot;)

Gaertner_traits &lt;- rbind(orient_to_female_m_line_means, 
                         approach_female_m_line_means, 
                         wing_display_frequency_m_line_means, 
                         genital_licking_m_line_means, 
                         attempted_copulations_m_line_means, 
                         mating_latency_m_line_means)</code></pre>
</div>
<div id="najarro-et-al-2015" class="section level3">
<h3>Najarro et al 2015</h3>
<pre class="r"><code># Line 34061 doesn&#39;t exist on Bloomington or flybase. There is a 34062 though (line 49). Currently removed

Najarro_2015_caff_resistance &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Najarro_et_al_2015_caffeine_resistance.csv&quot;) %&gt;% 
  # remove unidentified line
  filter(line != 34061) %&gt;% 
  mutate(line = as.factor(line),
         # standardise
         st_CaffeineResistance = my_scale(CaffeineResistance))
 
 # mean caffeine resistance model

caffeine_resistance_model &lt;- brm(st_CaffeineResistance ~ 1 + (1|line),
                      data = Najarro_2015_caff_resistance, family = &quot;Gaussian&quot;,
                      prior = c(prior(normal(0, 0.5), class = Intercept),
                                prior(exponential(1), class = sd)),
                      iter = 6000, warmup = 2000, chains = 4, cores = 4,
                      seed = 2, file = &quot;fits/raw_data_fits/caffeine.resistance.model&quot;)

new_data &lt;-
  tibble(line = Najarro_2015_caff_resistance$line) %&gt;% 
  distinct(line)

fitted_caffeine_resistance &lt;- fitted(caffeine_resistance_model,
                                     newdata = new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Najarro_2015_caff_resistance$CaffeineResistance) + mean(Najarro_2015_caff_resistance$CaffeineResistance),
         Q2.5 = Q2.5*sd(Najarro_2015_caff_resistance$CaffeineResistance) + mean(Najarro_2015_caff_resistance$CaffeineResistance),
         Q97.5 = Q97.5*sd(Najarro_2015_caff_resistance$CaffeineResistance) + mean(Najarro_2015_caff_resistance$CaffeineResistance))

caffeine_resistance_line_means &lt;- tibble(new_data, fitted_caffeine_resistance) %&gt;% 
  mutate(Trait = &quot;caffeine.resistance.f&quot;,
         Sex = &quot;Female&quot;,
         Reference = &quot;Najarro et al (2015) PLOS Genetics&quot;,
         `Trait guild` = &quot;Drug response&quot;,
         `Trait description` = &quot;Mean time, in hours, adult females survived during continuous exposure to medium supplemented with 1% caffeine. Higher values indicate greater resistance to caffeine&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)</code></pre>
</div>
<div id="appel-et-al-2016" class="section level3">
<h3>Appel et al 2016</h3>
<p>We need to do some wrangling to fit an appropriate model for these
data. The response is a learning score that is bounded between -100 to
100. It is the percentage of flies that approached a trained target
compared with a control target. Negative values indicate that the flies
preferred the control target; positive values indicate a preference for
the trained target. This data generating process can be approximated
using the beta distribution. However, we need to re-express the data so
that it falls between 0 and 1.</p>
<p>The authors calculate the learning score using the formula <span
class="math inline">\(learning.score =\frac{(Trained.odour -
Control.odour)*100}{Trained.odour + control.odour}\)</span>. A value of
0 represents an even split of flies between both odours. We can convert
this to the proportion scale, where this even split now = 0.5.</p>
<p>We can transform our data using the following formula:</p>
<p><span class="math inline">\(normalised.learning.score =
\frac{learning.score - min(learning.score)}{max(learing.score) -
min(learning.score)}\)</span></p>
<pre class="r"><code>## Punishment learning

Appel_2016_memory_punishment &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Appel_2016_memory_learning.csv&quot;) %&gt;% 
  filter(Test == &quot;Punishment&quot;,
         memory_score != &quot;-&quot;) %&gt;% 
  mutate(line = as.factor(line),
         Group_ID = as.factor(Group_ID),
         Sex = as.factor(Sex),
         memory_score = as.numeric(memory_score),
         # ok here&#39;s our normalising step. Note that a value of -100 now = 0
         # this means that lower values = greater learning i.e. a value of 0.1 means 90% of flies avoided the punishment associated odour
         norm_mem_score = (memory_score - min(memory_score)) / (max(memory_score) - min(memory_score))
                           )

# I don&#39;t have a good read on sensible priors here, so I&#39;ll rely on brms defaults

punishment_memory_model &lt;- brm(norm_mem_score ~ 1 + Sex + (Sex|line),
                       data = Appel_2016_memory_punishment, family = zero_one_inflated_beta(),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/punishment.memory.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 10))


new_data &lt;-
  expand_grid(line = Appel_2016_memory_punishment$line,
              Sex = c(&quot;Female&quot;, &quot;Male&quot;)) %&gt;% 
  distinct(line, Sex)

fitted_punishment_memory &lt;- fitted(punishment_memory_model,
       newdata = new_data) %&gt;% 
  as_tibble()

# females

punishment_memory_f_line_means &lt;- tibble(new_data, fitted_punishment_memory) %&gt;%
  filter(Sex == &quot;Female&quot;) %&gt;% 
  mutate(Trait = &quot;punishment.memory.learning.f&quot;,
         Sex = &quot;Female&quot;,
         Reference = &quot;Appel et al (2016) Biology Letters&quot;,
         `Trait guild` = &quot;Behavioural&quot;,
         `Trait description` = &quot;Mean aversion towards an odour flies were trained to associate with the onset of an electric shock. Higher values indicate that flies showed a greater aversion toward the odour, suggesting that they associated the odor with &#39;punishment&#39;. Higher values also indicate a higher learning score&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

# males

punishment_memory_m_line_means &lt;- tibble(new_data, fitted_punishment_memory) %&gt;%
  filter(Sex == &quot;Male&quot;) %&gt;% 
  mutate(Trait = &quot;punishment.memory.learning.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Appel et al (2016) Biology Letters&quot;,
         `Trait guild` = &quot;Behavioural&quot;,
         `Trait description` = &quot;Mean aversion towards an odour flies were trained to associate with the onset of an electric shock. Higher values indicate that flies showed a greater aversion toward the odour, suggesting that they associated the odor with &#39;punishment&#39;. Higher values also indicate a greater learning score.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

## Relief learning

Appel_2016_memory_relief &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Appel_2016_memory_learning.csv&quot;) %&gt;% 
  filter(Test == &quot;Relief&quot;,
         memory_score != &quot;-&quot;) %&gt;% 
   mutate(line = as.factor(line),
         Group_ID = as.factor(Group_ID),
         Sex = as.factor(Sex),
         memory_score = as.numeric(memory_score),
          # ok here&#39;s our normalising step. Note that a value of -100 now = 0
         # this means that lower values = greater learning i.e. a value of 0.1 means 90% of flies avoided the punishment associated odour
         norm_mem_score = (memory_score - min(memory_score)) / (max(memory_score) - min(memory_score)))
         
# I don&#39;t have a good read on sensible priors here, so I&#39;ll rely on brms defaults

relief_memory_model &lt;- brm(norm_mem_score ~ 1 + Sex + (Sex|line),
                           data = Appel_2016_memory_relief, family = zero_one_inflated_beta(),
                           iter = 6000, warmup = 2000, chains = 4, cores = 4,
                           seed = 2, file = &quot;fits/raw_data_fits/relief.memory.model&quot;,
                           control = list(adapt_delta = 0.8, max_treedepth = 10))

#pp_check(relief_memory_model)

new_data &lt;-
  expand_grid(line = Appel_2016_memory_relief$line,
              Sex = c(&quot;Female&quot;, &quot;Male&quot;)) %&gt;% 
  distinct(line, Sex)

fitted_relief_memory &lt;- fitted(relief_memory_model,
       newdata = new_data) %&gt;% 
  as_tibble()

# females

relief_memory_f_line_means &lt;- tibble(new_data, fitted_relief_memory) %&gt;%
  filter(Sex == &quot;Female&quot;) %&gt;% 
  mutate(Trait = &quot;relief.memory.learning.f&quot;,
         Sex = &quot;Female&quot;,
         Reference = &quot;Appel et al (2016) Biology Letters&quot;,
         `Trait guild` = &quot;Behavioural&quot;,
         `Trait description` = &quot;Mean preference for an odor flies were trained to associate with the end of an electric shock. Higher values indicate that flies showed a stronger preference towards the odor, suggesting that they associated the odor with &#39;relief&#39;. Higher values also indicate a greater learning score.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

# males

relief_memory_m_line_means &lt;- tibble(new_data, fitted_relief_memory) %&gt;%
  filter(Sex == &quot;Male&quot;) %&gt;% 
  mutate(Trait = &quot;relief.memory.learning.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Appel et al (2016) Biology Letters&quot;,
         `Trait guild` = &quot;Behavioural&quot;,
         `Trait description` = &quot;Mean preference for an odor flies were trained to associate with the end of an electric shock. Higher values indicate that flies showed a stronger preference towards the odor, suggesting that they associated the odor with &#39;relief&#39;. Higher values also indicate a greater learning score.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

Appel_traits &lt;- rbind(punishment_memory_f_line_means,
                      punishment_memory_m_line_means,
                      relief_memory_f_line_means,
                      relief_memory_m_line_means)</code></pre>
</div>
<div id="munoz-munoz-et-al-2016" class="section level3">
<h3>Munoz-Munoz et al 2016</h3>
<pre class="r"><code>Munoz_2016_wing_size &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/munoz_munoz_2016.csv&quot;) %&gt;% 
  mutate(line = as.factor(Strain),
         Repl = (as.factor(Repl)),
         CS = as.double(CS),
         # standardise
         st_CS = my_scale(CS))

wing_size_model &lt;- brm(st_CS ~ 1 + Sex + (Sex|line),
                       data = Munoz_2016_wing_size, family = gaussian,
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(normal(0, 0.5), class = b),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/wing.size.model&quot;,
                       control = list(adapt_delta = 0.8, max_treedepth = 10))

new_data &lt;-
  expand_grid(line = Munoz_2016_wing_size$line,
              Sex = c(&quot;F&quot;, &quot;M&quot;)) %&gt;% 
  distinct(line, Sex)

fitted_wing_size &lt;- fitted(wing_size_model,
       newdata = new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Munoz_2016_wing_size$CS) + mean(Munoz_2016_wing_size$CS),
         Q2.5 = Q2.5*sd(Munoz_2016_wing_size$CS) + mean(Munoz_2016_wing_size$CS),
         Q97.5 = Q97.5*sd(Munoz_2016_wing_size$CS) + mean(Munoz_2016_wing_size$CS))

# females

wing_size_f_line_means &lt;- tibble(new_data, fitted_wing_size) %&gt;%
  mutate(Sex = if_else(Sex == &quot;F&quot;, &quot;Female&quot;, &quot;Male&quot;)) %&gt;% 
  filter(Sex == &quot;Female&quot;) %&gt;% 
  mutate(Trait = &quot;wing.size.f&quot;,
         Sex = &quot;Female&quot;,
         Reference = &quot;Muñoz-Muñoz et al (2016) Evolution&quot;,
         `Trait guild` = &quot;Morphological&quot;,
         `Trait description` = &quot;Mean centroid size - the square root of the sum of squared distances of a set of wing landmarks to the centroid of this landmark configuration. Higher values indicate larger wings.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

# males

wing_size_m_line_means &lt;- tibble(new_data, fitted_wing_size) %&gt;%
  mutate(Sex = if_else(Sex == &quot;F&quot;, &quot;Female&quot;, &quot;Male&quot;)) %&gt;% 
  filter(Sex == &quot;Male&quot;) %&gt;% 
  mutate(Trait = &quot;wing.size.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Muñoz-Muñoz et al (2016) Evolution&quot;,
         `Trait guild` = &quot;Morphological&quot;,
         `Trait description` = &quot;Mean centroid size - the square root of the sum of squared distances of a set of wing landmarks to the centroid of this landmark configuration. Higher values indicate larger wings.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

Munoz_traits &lt;- rbind(wing_size_f_line_means,
                      wing_size_m_line_means)</code></pre>
</div>
<div id="freda-2017" class="section level3">
<h3>Freda, 2017</h3>
<p>We have line mean data for larvae, but raw data for adults.
Therefore, we only need to model the adult data here.</p>
<pre class="r"><code>Freda_2017_cold_hardiness &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Freda_2017_ThermalHardiness.csv&quot;) %&gt;% 
  mutate(line = as.factor(line),
         Sex = as.factor(Sex))

# I don&#39;t have a good read on sensible priors here, so I&#39;ll rely on brms defaults

cold_hardiness_model &lt;- brm(Proportion_alive ~ 1 + Sex + (Sex|line),
                           data = Freda_2017_cold_hardiness, family = zero_one_inflated_beta(),
                           iter = 6000, warmup = 2000, chains = 4, cores = 4,
                           seed = 2, file = &quot;fits/raw_data_fits/cold_hardiness.model&quot;,
                           control = list(adapt_delta = 0.8, max_treedepth = 10))

#pp_check(cold_hardiness_model)

new_data &lt;-
  expand_grid(line = Freda_2017_cold_hardiness$line,
              Sex = c(&quot;Female&quot;, &quot;Male&quot;)) %&gt;% 
  distinct(line, Sex)

fitted_cold_hardiness &lt;- fitted(cold_hardiness_model,
       newdata = new_data) %&gt;% 
  as_tibble()

# females

cold_hardiness_f_line_means &lt;- tibble(new_data, fitted_cold_hardiness) %&gt;%
  filter(Sex == &quot;Female&quot;) %&gt;% 
  mutate(Trait = &quot;cold.hardiness.f&quot;,
         Sex = &quot;Female&quot;,
         Reference = &quot;Freda et al (2017) Integrative and Comparative Biology&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         `Trait description` = &quot;Mean proportion of surviving adults 24 hours post exposure to an hour of cold shock (-5°C). Higher values indicate flies that are more resistant to cold shock.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

# males

cold_hardiness_m_line_means &lt;- tibble(new_data, fitted_cold_hardiness) %&gt;%
  filter(Sex == &quot;Male&quot;) %&gt;% 
  mutate(Trait = &quot;cold.hardiness.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Freda et al (2017) Integrative and Comparative Biology&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         `Trait description` = &quot;Mean proportion of surviving adults 24 hours post exposure to an hour of cold shock (-5°C). Higher values indicate flies that are more resistant to cold shock.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

Freda_2017_traits &lt;- rbind(cold_hardiness_f_line_means,
                      cold_hardiness_m_line_means)</code></pre>
</div>
<div id="najarro-hackett-and-macdonald-2017" class="section level3">
<h3>Najarro, Hackett and Macdonald, 2017</h3>
<pre class="r"><code>Najarro_2017_boric_acid &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/najarro_2017.csv&quot;) %&gt;% 
  rename(lifespan = `life-span (hr) on 1.5% boric acid`) %&gt;% 
  mutate(line = as.factor(line),
         st_lifespan = my_scale(lifespan))

boric_acid_model &lt;- brm(st_lifespan ~ 1 + (1|line),
                        data = Najarro_2017_boric_acid, family = &quot;Gaussian&quot;,
                        prior = c(prior(normal(0, 0.5), class = Intercept),
                                  prior(exponential(1), class = sd)),
                        iter = 6000, warmup = 2000, chains = 4, cores = 4,
                        seed = 2, file = &quot;fits/raw_data_fits/boric.acid.model&quot;)



# get predictions from the model for each line

new_data &lt;-
  tibble(line = Najarro_2017_boric_acid$line) %&gt;% 
  distinct(line)

fitted_boric_acid &lt;- fitted(boric_acid_model,
       newdata = new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Najarro_2017_boric_acid$lifespan) + mean(Najarro_2017_boric_acid$lifespan),
         Q2.5 = Q2.5*sd(Najarro_2017_boric_acid$lifespan) + mean(Najarro_2017_boric_acid$lifespan),
         Q97.5 = Q97.5*sd(Najarro_2017_boric_acid$lifespan) + mean(Najarro_2017_boric_acid$lifespan))

boric_acid_line_means &lt;- tibble(new_data, fitted_boric_acid) %&gt;% 
  mutate(line = str_remove(line, &quot;RAL&quot;),
         Trait = &quot;boric.acid.resistance.f&quot;,
         Sex = &quot;Female&quot;,
         Reference = &quot;Najarro, Hackett and Macdonald (2017) G3&quot;,
         `Trait guild` = &quot;Insecticide response&quot;,
         `Trait description` = &quot;Mean lifespan, measured in hours, for mated females exposed to media supplemented with 1.5% boric acid (BH3O3, BP168). Boric acid is a common household pesticide. Higher values indicate a greater resistance to boric acid.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)</code></pre>
</div>
<div id="dean-et-al-2018" class="section level3">
<h3>Dean et al 2018</h3>
<p>To model copulation latency Dean <em>et al</em> log transform the
data and fit a linear model with a gaussian distribution. This model
struggles to converge using <code>brms</code>, so we treat the data as
time to event with right censoring for the flies that failed to mate
within a 2 hour window (coded as 120 in the dataset). To model this data
we fit a survival model, specifying a weibull distribution.</p>
<p>To model insemination capacity Dean <em>et al</em> specify a
quasipoisson distribution family because they treat the data as
underdispersed counts. There is no family like this in
<code>brms</code>. However, we can treat the data as binomial, with each
female treated as a YES/NO for mating success. The posterior predictive
check suggests this is a good fit to the data. Our predictions are also
a good match with those of Dean et al.</p>
<pre class="r"><code>Dean_2018_data &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Dean_2018.csv&quot;) %&gt;% 
   mutate(line = as.factor(Line),
          Rep = as.factor(Rep),
          Block = as.factor(Block),
          censor = if_else(LatCop == 120, 1, 0))

copulation_latency_model &lt;- brm(LatCop | cens(censor) ~ 1 + Block + (1|line),
                       data = Dean_2018_data, family = weibull(),
                       prior = c(prior(normal(0, 1), class = Intercept),
                                 prior(normal(0, 1), class = b),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/copulation_latency.model&quot;,
                       control = list(adapt_delta = 0.8, max_treedepth = 10))

#pp_check(copulation_latency_model)

new_data &lt;-
  tibble(line = Dean_2018_data$line,
         Block = 1) %&gt;% 
  distinct(line, Block)


fitted_cop_lat &lt;- fitted(copulation_latency_model,
       newdata = new_data) %&gt;% 
  as_tibble()

cop_lat_line_means &lt;- tibble(new_data, fitted_cop_lat) %&gt;%
  mutate(Trait = &quot;copulation.latency.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Dean et al (2020) Evolution&quot;,
         `Trait guild` = &quot;Behavioural&quot;,
         `Trait description` = &quot;Mean time (mins) from a male flies introduction to a female until the onset of mating. Lower values indicate shorter copulation latencies and potentially more attractive males&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(trait_value)

# Dean et al fit a quasipoisson model because they treat the data as underdispersed counts. There is no family like this in brms. However, I think we can treat the data as binomial with each female treated as a YES/NO for mating success. The posterior predictive check suggests this is a good fit to the data.

Dean_2018_data &lt;-
  Dean_2018_data %&gt;% 
  mutate(Cohabiting_females = 10)

insemination_capacity_model &lt;- brm(promiscuity | trials(Cohabiting_females) ~ 1 + Block + (1|line),
                                   data = Dean_2018_data, family = binomial,
                                   prior = c(prior(normal(0, 0.5), class = Intercept),
                                             prior(normal(0, 0.5), class = b),
                                             prior(exponential(1), class = sd)),
                                   iter = 6000, warmup = 2000, chains = 4, cores = 4,
                                   seed = 2, file = &quot;fits/raw_data_fits/insemination_capacity.model&quot;,
                                   control = list(adapt_delta = 0.8, max_treedepth = 10))



#pp_check(insemination_capacity_model)

new_data &lt;-
  tibble(line = Dean_2018_data$line,
         Cohabiting_females = 1,
         Block =1) %&gt;% 
  distinct(line, Cohabiting_females, Block)


fitted_insem_capac &lt;- fitted(insemination_capacity_model,
       newdata = new_data) %&gt;% 
  as_tibble()

insem_capac_line_means &lt;- tibble(new_data, fitted_insem_capac) %&gt;%
  mutate(Trait = &quot;insemination.capacity.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Dean et al (2020) Evolution&quot;,
         `Trait guild` = &quot;Life history&quot;,
         `Trait description` = &quot;Mean proportion of 10 females that a single male successfully inseminated in 48 hours. Insemination was verified by identification of viable pupal progeny. Higher values indicate males with a greater insemination capacity.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(trait_value)

Dean_traits &lt;- rbind(cop_lat_line_means,
                     insem_capac_line_means)</code></pre>
</div>
<div id="duneau-et-al-2018" class="section level3">
<h3>Duneau et al 2018</h3>
<p>We fit a slightly different parathion model to that presented in the
manuscript. This is because some lines are resistant to parathion,
leading to heavy sensoring. This results in two groups of lines: those
that are susceptible and those that are resistant. The resistant lines
have estimated mean survival times of 400+ days, which is not realistic
for D. melanogaster. Therefore, I instead model resistance to parathion
using a binomial response, which describes whether the line had
surviving flies 48 hours after exposure to parathion.</p>
<pre class="r"><code># Deltamethrin

Duneau_2018_deltamethrin_data &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Duneau_2018/Duneau_2018_deltamethrin.csv&quot;) %&gt;% 
   mutate(line = str_remove(DGRP_lines, &quot;line_&quot;),
          Experiment = as.factor(Experiment),
          line = as.factor(line)) %&gt;% 
  filter(Sample.size != &quot;NA&quot;)

deltamethrin_model &lt;-
  brm(Delta_Alive.at.48h | trials(Sample.size) ~ 1  + Experiment + (1|line),
      data = Duneau_2018_deltamethrin_data,
      prior = c(prior(normal(0, 0.5), class = Intercept),
                prior(normal(0, 0.5), class = b),
                prior(exponential(1), class = sd)),
      family = binomial(),
      iter = 10000, warmup = 4000, chains = 4, cores = 4,
      seed = 2, file = &quot;fits/raw_data_fits/deltamethrin.model&quot;,
      control = list(adapt_delta = 0.95, max_treedepth = 10))

#pp_check(deltamethrin_model)

new_data_Duneau &lt;-
  tibble(line = Duneau_2018_deltamethrin_data$line,
         Experiment = &quot;Exp 1&quot;,
         Sample.size = 1) %&gt;% 
  distinct(line, Experiment, Sample.size)


fitted_deltamethrin &lt;- fitted(deltamethrin_model,
       newdata = new_data_Duneau,
       allow_new_levels = TRUE) %&gt;% 
  as_tibble()

deltamethrin_line_means &lt;- tibble(new_data_Duneau, fitted_deltamethrin) %&gt;%
  mutate(Trait = &quot;deltamethrin.resistance.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Duneau et al (2018) G3&quot;,
         `Trait guild` = &quot;Insecticide response&quot;,
         `Trait description` = &quot;Mean survival 48 hours after exposure to deltamethrin, a pyrethroid pesticide. Higher values indicate greater resistance to deltamethrin.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)


# Parathion

# set up data for right censoring in brms - it is the opposite to what the survival frequentist package uses. We don&#39;t use this but it is useful to know if we do choose to fit a time to event model

Duneau_2018_parathion_data &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Duneau_2018/Duneau_2018_parathion.csv&quot;) %&gt;% 
   mutate(line = str_remove(DGRP_lines, &quot;line_&quot;),
          Experiment = as.factor(Experiment)) %&gt;% 
  mutate(line = as.factor(line),
         Alive.at.48hr = if_else(Censor == 1, 0, 1),
         Censor = if_else(Censor == 1, &quot;none&quot;, &quot;right&quot;)) 

# fit the new binomial model

parathion_binomial_model &lt;-
  brm(Alive.at.48hr ~ 1 + Experiment + (1|line),
      data = Duneau_2018_parathion_data,
      prior = c(prior(normal(0, 0.5), class = Intercept),
                prior(normal(0, 0.5), class = b),
                prior(exponential(1), class = sd)),
      family = bernoulli,
      cores = 4, chains = 4, iter = 6000, warmup = 2000,
      control = list(adapt_delta = 0.9, max_treedepth = 10),
      seed = 1, file = &quot;fits/raw_data_fits/parathion_binomial.model&quot;)

#pp_check(parathion_binomial_model) 

new_data &lt;-
  tibble(line = Duneau_2018_parathion_data$line,
         Experiment = &quot;Experiment_1&quot;) %&gt;% 
  distinct(line, Experiment)


fitted_parathion_binomial &lt;- fitted(parathion_binomial_model,
       newdata = new_data) %&gt;% 
  as_tibble()

parathion_binomial_line_means &lt;- tibble(new_data, fitted_parathion_binomial) %&gt;%
  mutate(Trait = &quot;parathion.resistance.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Duneau et al (2018) G3&quot;,
         `Trait guild` = &quot;Insecticide response&quot;,
         `Trait description` = &quot;Mean survival 48 hours after exposure to parathion, an organophosphate pesticide. Higher values indicate greater resistance to parathion.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

Duneau_traits &lt;- rbind(deltamethrin_line_means,
                       parathion_binomial_line_means)</code></pre>
</div>
<div id="palmer-et-al-2018" class="section level3">
<h3>Palmer et al 2018</h3>
<pre class="r"><code>Palmer_2018_KV_data &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Palmer_2018_kallithea.csv&quot;) %&gt;% 
  group_by(Injection.date) %&gt;%
  mutate(injection_date = as_factor(cur_group_id())) %&gt;% 
  ungroup() %&gt;% 
  mutate(line = as.factor(Line),
         Sex = as.factor(Sex),
         value = as.numeric(value))
  

Palmer_2018_LT50 &lt;-
  Palmer_2018_KV_data %&gt;% 
  filter(trait.column == &quot;Mortality&quot;) %&gt;% 
  mutate(# standardise
         st_value = my_scale(value))

# LT50 model

kv_LT50_model &lt;- brm(st_value ~ 1 + Sex + (Sex|line) + (1|injection_date),
                       data = Palmer_2018_LT50, family = gaussian(),
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(normal(0, 0.5), class = b),
                                 prior(exponential(1), class = sd)),
                       iter = 8000, warmup = 4000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/kv.LT50.model&quot;,
                       control = list(adapt_delta = 0.97, max_treedepth = 12))

 #pp_check(kv_LT50_model)

new_data &lt;-
  expand_grid(line = Palmer_2018_LT50$line,
              Sex = c(&quot;Female&quot;, &quot;Male&quot;)) %&gt;% 
  distinct(line, Sex)

fitted_kv_LT50 &lt;- fitted(kv_LT50_model,
       newdata = new_data,
       re_formula = value ~ (Sex|line),
       allow_new_levels = TRUE) %&gt;% # sometimes we need to include this for fitted to work. It changes nothing in the output 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Palmer_2018_LT50$value) + mean(Palmer_2018_LT50$value),
         Q2.5 = Q2.5*sd(Palmer_2018_LT50$value) + mean(Palmer_2018_LT50$value),
         Q97.5 = Q97.5*sd(Palmer_2018_LT50$value) + mean(Palmer_2018_LT50$value))

# females

kv_LT50_f_line_means &lt;- tibble(new_data, fitted_kv_LT50) %&gt;%
  filter(Sex == &quot;Female&quot;) %&gt;% 
  mutate(Trait = &quot;kallithea.resistance.LT50.f&quot;,
         Sex = &quot;Female&quot;,
         Reference = &quot;Palmer et al (2018) PLOS Pathogens&quot;,
         `Trait guild` = &quot;Pathogen response&quot;,
         `Trait description` = &quot;Median number of days adult flies survived post infection with Kallithea virus (50 nL of 105 ID50 originally injected). Measured as LT50 - the point at which 50% of 10 flies died. Higher values indicate greater resistance to Kallithea virus.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

# males

kv_LT50_m_line_means &lt;- tibble(new_data, fitted_kv_LT50) %&gt;%
  filter(Sex == &quot;Male&quot;) %&gt;% 
  mutate(Trait = &quot;kallithea.resistance.LT50.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Palmer et al (2018) PLOS Pathogens&quot;,
         `Trait guild` = &quot;Pathogen response&quot;,
         `Trait description` = &quot;Median number of days adult flies survived post infection with Kallithea virus (50 nL of 105 ID50 originally injected). Measured as LT50 - the point at which 50% of 10 flies died. Higher values indicate greater resistance to Kallithea virus.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

# titre model

Palmer_2018_KV_titre_data &lt;-
  Palmer_2018_KV_data %&gt;% 
  filter(trait.column == &quot;titre&quot;) %&gt;% 
   mutate(# standardise
         st_value = my_scale(value))

kv_titre_model &lt;- brm(st_value ~ 1 + (1|line) + (1|injection_date),
                       data = Palmer_2018_KV_titre_data, family = gaussian(),
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/kv.titre.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 12))


new_data &lt;-
  expand_grid(line = Palmer_2018_KV_titre_data$line,
              Sex = &quot;Female&quot;) %&gt;% 
  distinct(line, Sex)

fitted_kv_titre &lt;- fitted(kv_titre_model,
       newdata = new_data,
       re_formula = value ~ (1|line),
       allow_new_levels = TRUE) %&gt;% # sometimes we need to include this for fitted to work. It changes nothing in the output 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Palmer_2018_KV_titre_data$value) + mean(Palmer_2018_KV_titre_data$value),
         Q2.5 = Q2.5*sd(Palmer_2018_KV_titre_data$value) + mean(Palmer_2018_KV_titre_data$value),
         Q97.5 = Q97.5*sd(Palmer_2018_KV_titre_data$value) + mean(Palmer_2018_KV_titre_data$value))

# females

kv_titre_f_line_means &lt;- tibble(new_data, fitted_kv_titre) %&gt;%
  filter(Sex == &quot;Female&quot;) %&gt;% 
  mutate(Trait = &quot;kallithea.viral.load.f&quot;,
         Sex = &quot;Female&quot;,
         Reference = &quot;Palmer et al (2018) PLOS Pathogens&quot;,
         `Trait guild` = &quot;Pathogen response&quot;,
         `Trait description` = &quot;Mean viral titre of kallithea virus found in groups of 10 adult females, 8 days after injection of 50 nL of 105 ID50 KV.  This is a measure of viral load, with higher values indicating greater viral load.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

Palmer_traits &lt;- rbind(kv_LT50_f_line_means,
                       kv_LT50_m_line_means,
                       kv_titre_f_line_means)</code></pre>
</div>
<div id="teets-and-hahn-2018" class="section level3">
<h3>Teets and Hahn 2018</h3>
<p>Cumulative cold tolerance data was supplied by the authors in line
mean form, so we do not model it here.</p>
<pre class="r"><code>Teets_2018_cold_data &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Teets_Hahn_2018_data.csv&quot;) %&gt;% 
  mutate(line = as.factor(`Line (DGRP)`),
         Temp = as.factor(Temp))

cold_survival_model &lt;- brm(Live | trials(Total) ~ 1 + Temp + (Temp|line),
                       data = Teets_2018_cold_data, family = binomial(),
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(normal(0, 0.5), class = b),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/cold.survival.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 12))


#pp_check(cold_survival_model)

new_data &lt;-
  expand_grid(line = Teets_2018_cold_data$line,
              Temp = Teets_2018_cold_data$Temp,
              Total = 1) %&gt;% 
  distinct(line, Temp, Total)

fitted_cold_survival &lt;- fitted(cold_survival_model,
       newdata = new_data) %&gt;% 
  as_tibble()

cold_survival_line_means_data &lt;- tibble(new_data, fitted_cold_survival) %&gt;% 
  mutate(Sex = &quot;Female&quot;)

# build a function to create tibbles holding line and temp specific line means

Teets_line_means &lt;- function(estimates, value, Trait, description){
  estimates %&gt;%
  filter(Temp == value) %&gt;% 
  mutate(Trait = Trait,
         Reference = &quot;Teets and Hahn (2018) Journal of Evolutionary Biology&quot;,
         `Trait guild` =&quot;Temperature related&quot;,
         `Trait description` = description,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)
}

neg1_cold_tolerance_line_means &lt;- Teets_line_means(cold_survival_line_means_data, value = &quot;-1&quot;, &quot;cold.tolerance.-1.f&quot;, &quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -1 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;)

neg2_cold_tolerance_line_means &lt;- Teets_line_means(cold_survival_line_means_data, value = &quot;-2&quot;, &quot;cold.tolerance.-2.f&quot;, &quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -2 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;)

neg3_cold_tolerance_line_means &lt;- Teets_line_means(cold_survival_line_means_data, value = &quot;-3&quot;, &quot;cold.tolerance.-3.f&quot;, &quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -3 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;)

neg4_cold_tolerance_line_means &lt;- Teets_line_means(cold_survival_line_means_data, value = &quot;-4&quot;, &quot;cold.tolerance.-4.f&quot;, &quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -4 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;)

neg5_cold_tolerance_line_means &lt;- Teets_line_means(cold_survival_line_means_data, value = &quot;-5&quot;, &quot;cold.tolerance.-5.f&quot;, &quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -5 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;)

neg6_cold_tolerance_line_means &lt;- Teets_line_means(cold_survival_line_means_data, value = &quot;-6&quot;, &quot;cold.tolerance.-6.f&quot;, &quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -6 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;)

neg7_cold_tolerance_line_means &lt;- Teets_line_means(cold_survival_line_means_data, value = &quot;-7&quot;, &quot;cold.tolerance.-7.f&quot;, &quot;Mean proportion of surviving adults 24 hours after an hour of exposure to -7 °C. Higher values indicate flies that are more tolerant to cold exposure.&quot;)

Teets_traits &lt;- rbind(neg1_cold_tolerance_line_means,
                      neg2_cold_tolerance_line_means,
                      neg3_cold_tolerance_line_means,
                      neg4_cold_tolerance_line_means,
                      neg5_cold_tolerance_line_means,
                      neg6_cold_tolerance_line_means,
                      neg7_cold_tolerance_line_means)</code></pre>
</div>
<div id="everman-et-al-2019" class="section level3">
<h3>Everman et al 2019</h3>
<pre class="r"><code>Everman_2019_starvation_resistance &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Everman_2019_starvation_res.csv&quot;) %&gt;% 
  mutate(line = as.factor(line),
         Sex = as.factor(Sex),
         Lifespan_hrs = as.numeric(Lifespan_hrs),
         # standardise
         st_Lifespan_hrs = my_scale(Lifespan_hrs))


starvation_resistance_model &lt;- brm(st_Lifespan_hrs ~ 1 + Sex + (Sex|line),
                       data = Everman_2019_starvation_resistance, family = gaussian(),
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(normal(0, 0.5), class = b),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/starvation.res.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 10))

 #pp_check(starvation_resistance_model)


new_data &lt;-
  expand_grid(line = Everman_2019_starvation_resistance$line,
              Sex = c(&quot;Female&quot;, &quot;Male&quot;)) %&gt;% 
  distinct(line, Sex)

fitted_starvation_resistance &lt;- fitted(starvation_resistance_model,
       newdata = new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Everman_2019_starvation_resistance$Lifespan_hrs) + mean(Everman_2019_starvation_resistance$Lifespan_hrs),
         Q2.5 = Q2.5*sd(Everman_2019_starvation_resistance$Lifespan_hrs) + mean(Everman_2019_starvation_resistance$Lifespan_hrs),
         Q97.5 = Q97.5*sd(Everman_2019_starvation_resistance$Lifespan_hrs) + mean(Everman_2019_starvation_resistance$Lifespan_hrs))

# females

starvation_resistance_f_line_means &lt;- tibble(new_data, fitted_starvation_resistance) %&gt;%
  filter(Sex == &quot;Female&quot;) %&gt;% 
  mutate(Trait = &quot;starvation.resistance.2019.f&quot;,
         Sex = &quot;Female&quot;,
         Reference = &quot;Everman et al (2019) Genetics&quot;,
         `Trait guild` = &quot;Stress response&quot;,
         `Trait description` = &quot;Mean hours survived on inedible food medium, used as a measure of starvation resistance. Higher values indicate flies that exhibited greater starvation resistance.&quot;,
         trait_value = Estimate) %&gt;% 
  mutate(line = str_remove(line, &quot;RAL-&quot;)) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

# males

starvation_resistance_m_line_means &lt;- tibble(new_data, fitted_starvation_resistance) %&gt;%
  filter(Sex == &quot;Male&quot;) %&gt;% 
  mutate(Trait = &quot;starvation.resistance.2019.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Everman et al (2019) Genetics&quot;,
         `Trait guild` = &quot;Stress response&quot;,
         `Trait description` = &quot;Mean hours survived on inedible food medium, used as a measure of starvation resistance. Higher values indicate flies that exhibited greater starvation resistance.&quot;,
         trait_value = Estimate) %&gt;% 
  mutate(line = str_remove(line, &quot;RAL-&quot;)) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

Everman_traits &lt;- rbind(starvation_resistance_f_line_means,
                        starvation_resistance_m_line_means)</code></pre>
</div>
<div id="freda-2019" class="section level3">
<h3>Freda 2019</h3>
<pre class="r"><code>Freda_2019a &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Freda_2019/Cold18.csv&quot;) %&gt;% 
   mutate(line = as.factor(dgrpline))

Freda_2019b &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Freda_2019/Cold25.csv&quot;) %&gt;% 
   mutate(line = as.factor(dgrpline)) %&gt;% 
  filter(n &gt; 0)

Freda_2019c &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Freda_2019/Heat18.csv&quot;) %&gt;% 
   mutate(line = as.factor(dgrpline))

Freda_2019d &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Freda_2019/Heat25.csv&quot;) %&gt;% 
   mutate(line = as.factor(dgrpline))

# Cold hardiness at 18C for larvae model

cold_hardiness_18_larvae_model &lt;- brm(alive | trials(n) ~ 1 + (1|line),
                       data = Freda_2019a %&gt;% filter(stage == &quot;L&quot;), family = binomial(),
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/ch_18_l.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 10))

 #pp_check(cold_hardiness_18_larvae_model)

Freda_2019a_larvae_data &lt;-
  Freda_2019a %&gt;% filter(stage == &quot;L&quot;)

new_data &lt;-
  tibble(line = Freda_2019a_larvae_data$line,
         n = 1) %&gt;% 
  distinct(line, n)

fitted_ch_18_l &lt;- fitted(cold_hardiness_18_larvae_model,
       newdata = new_data) %&gt;% 
  as_tibble()

# larvae

cold_hardiness_18_l_line_means &lt;- tibble(new_data, fitted_ch_18_l) %&gt;%
  mutate(Trait = &quot;cold.hardiness.18C.larvae&quot;,
         Sex = &quot;Both&quot;,
         Reference = &quot;Freda et al (2019) Heredity&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         `Trait description` = &quot;Mean proportion of third instar larvae that survived exposure to -6.5°C for an hour. These larvae were otherwise reared at 18°C. Higher values indicate greater cold hardiness.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)


# Cold hardiness at 18C for females model

cold_hardiness_18_f_model &lt;- brm(f_female | trials(s_female) ~ 1 + (1|line),
                       data = Freda_2019a %&gt;% filter(stage == &quot;A&quot;), family = binomial(),
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/ch_18_f.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 10))

 #pp_check(cold_hardiness_18_f_model)

Freda_2019a_larvae_data &lt;-
  Freda_2019a %&gt;% filter(stage == &quot;A&quot;)

new_data &lt;-
  tibble(line = Freda_2019a_larvae_data$line,
         s_female = 1) %&gt;% 
  distinct(line, s_female)

fitted_ch_18_f &lt;- fitted(cold_hardiness_18_f_model,
       newdata = new_data,
       allow_new_levels = TRUE) %&gt;% # sometimes we need to include this for fitted to work. It changes nothing in the output
  as_tibble()

# female

cold_hardiness_18_f_line_means &lt;- tibble(new_data, fitted_ch_18_f) %&gt;%
  mutate(Trait = &quot;cold.hardiness.18C.f&quot;,
         Sex = &quot;Female&quot;,
         Reference = &quot;Freda et al (2019) Heredity&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         `Trait description` = &quot;Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater cold hardiness.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

# Cold hardiness at 18C for males model

cold_hardiness_18_m_model &lt;- brm(f_male | trials(s_male) ~ 1 + (1|line),
                       data = Freda_2019a %&gt;% filter(stage == &quot;A&quot;), family = binomial(),
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/ch_18_m.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 10))

 #pp_check(cold_hardiness_18_m_model)

Freda_2019a_larvae_data &lt;-
  Freda_2019a %&gt;% filter(stage == &quot;A&quot;)

new_data &lt;-
  tibble(line = Freda_2019a_larvae_data$line,
         s_male = 1) %&gt;% 
  distinct(line, s_male)

fitted_ch_18_m &lt;- fitted(cold_hardiness_18_m_model,
       newdata = new_data,
              allow_new_levels = TRUE) %&gt;% # sometimes we need to include this for fitted to work. It changes nothing in the output 
  as_tibble()

cold_hardiness_18_m_line_means &lt;- tibble(new_data, fitted_ch_18_m) %&gt;%
  mutate(Trait = &quot;cold.hardiness.18C.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Freda et al (2019) Heredity&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         `Trait description` = &quot;Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater cold hardiness.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

#################

# Cold hardiness at 25C for larvae model

cold_hardiness_25_larvae_model &lt;- brm(alive | trials(n) ~ 1 + (1|line),
                       data = Freda_2019b %&gt;% filter(stage == &quot;L&quot;), family = binomial(),
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/ch_25_l.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 10))

 #pp_check(cold_hardiness_25_larvae_model)

Freda_2019b_larvae_data &lt;-
  Freda_2019b %&gt;% filter(stage == &quot;L&quot;)

new_data &lt;-
  tibble(line = Freda_2019b_larvae_data$line,
         n = 1) %&gt;% 
  distinct(line, n)

fitted_ch_25_l &lt;- fitted(cold_hardiness_25_larvae_model,
       newdata = new_data) %&gt;% 
  as_tibble()

# larvae

cold_hardiness_25_l_line_means &lt;- tibble(new_data, fitted_ch_25_l) %&gt;%
  mutate(Trait = &quot;cold.hardiness.25C.larvae&quot;,
         Sex = &quot;Both&quot;,
         Reference = &quot;Freda et al (2019) Heredity&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         `Trait description` = &quot;Mean proportion of third instar larvae that survived exposure to -6.5°C for an hour. These larvae were otherwise reared at 25°C. Higher values indicate greater cold hardiness.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)


# Cold hardiness at 25C for females model

cold_hardiness_25_f_model &lt;- brm(f_female | trials(s_female) ~ 1 + (1|line),
                       data = Freda_2019b %&gt;% filter(stage == &quot;A&quot;), family = binomial(),
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/ch_25_f.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 10))

 #pp_check(cold_hardiness_25_f_model)

Freda_2019b_f_data &lt;-
  Freda_2019b %&gt;% filter(stage == &quot;A&quot;)

new_data &lt;-
  tibble(line = Freda_2019b_f_data$line,
         s_female = 1) %&gt;% 
  distinct(line, s_female)

fitted_ch_25_f &lt;- fitted(cold_hardiness_25_f_model,
       newdata = new_data) %&gt;% 
  as_tibble()

cold_hardiness_25_f_line_means &lt;- tibble(new_data, fitted_ch_25_f) %&gt;%
  mutate(Trait = &quot;cold.hardiness.25C.f&quot;,
         Sex = &quot;Female&quot;,
         Reference = &quot;Freda et al (2019) Heredity&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         `Trait description` = &quot;Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater cold hardiness.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

# Cold hardiness at 25C for males model

cold_hardiness_25_m_model &lt;- brm(f_male | trials(s_male) ~ 1 + (1|line),
                       data = Freda_2019b %&gt;% filter(stage == &quot;A&quot;), family = binomial(),
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/ch_25_m.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 10))

 #pp_check(cold_hardiness_25_m_model)

Freda_2019b_m_data &lt;-
  Freda_2019b %&gt;% filter(stage == &quot;A&quot;)

new_data &lt;-
  tibble(line = Freda_2019b_m_data$line,
         s_male = 1) %&gt;% 
  distinct(line, s_male)

fitted_ch_25_m &lt;- fitted(cold_hardiness_25_m_model,
       newdata = new_data) %&gt;% 
  as_tibble()

cold_hardiness_25_m_line_means &lt;- tibble(new_data, fitted_ch_25_m) %&gt;%
  mutate(Trait = &quot;cold.hardiness.25C.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Freda et al (2019) Heredity&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         `Trait description` = &quot;Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater cold hardiness.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

##############

# Heat hardiness at 18C for larvae model

heat_hardiness_18_larvae_model &lt;- brm(alive | trials(n) ~ 1 + (1|line),
                       data = Freda_2019c %&gt;% filter(stage == &quot;L&quot;), family = binomial(),
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/hh_18_l.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 10))

 #pp_check(heat_hardiness_18_larvae_model)

Freda_2019c_larvae_data &lt;-
  Freda_2019c %&gt;% filter(stage == &quot;L&quot;)

new_data &lt;-
  tibble(line = Freda_2019c_larvae_data$line,
         n = 1) %&gt;% 
  distinct(line, n)

fitted_hh_18_l &lt;- fitted(heat_hardiness_18_larvae_model,
       newdata = new_data) %&gt;% 
  as_tibble()

heat_hardiness_18_l_line_means &lt;- tibble(new_data, fitted_hh_18_l) %&gt;%
  mutate(Trait = &quot;heat.hardiness.18C.larvae&quot;,
         Sex = &quot;Both&quot;,
         Reference = &quot;Freda et al (2019) Heredity&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         `Trait description` = &quot;Mean proportion of third instar larvae that survived exposure to 38°C for an hour. These larvae were otherwise reared at 18°C. Higher values indicate greater heat hardiness.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)


# Heat hardiness at 18C for females model

heat_hardiness_18_f_model &lt;- brm(f_female | trials(s_female) ~ 1 + (1|line),
                       data = Freda_2019c %&gt;% filter(stage == &quot;A&quot;), family = binomial(),
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/hh_18_f.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 10))

 #pp_check(heat_hardiness_18_f_model)

Freda_2019c_f_data &lt;-
  Freda_2019c %&gt;% filter(stage == &quot;A&quot;)

new_data &lt;-
  tibble(line = Freda_2019c_f_data$line,
         s_female = 1) %&gt;% 
  distinct(line, s_female)

fitted_hh_18_f &lt;- fitted(heat_hardiness_18_f_model,
       newdata = new_data) %&gt;% 
  as_tibble()

heat_hardiness_18_f_line_means &lt;- tibble(new_data, fitted_hh_18_f) %&gt;%
  mutate(Trait = &quot;heat.hardiness.18C.f&quot;,
         Sex = &quot;Female&quot;,
         Reference = &quot;Freda et al (2019) Heredity&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         `Trait description` = &quot;Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater heat hardiness.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

# Heat hardiness at 18C for males model

heat_hardiness_18_m_model &lt;- brm(f_male | trials(s_male) ~ 1 + (1|line),
                       data = Freda_2019c %&gt;% filter(stage == &quot;A&quot;), family = binomial(),
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/hh_18_m.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 10))

 #pp_check(heat_hardiness_18_m_model)

Freda_2019c_m_data &lt;-
  Freda_2019c %&gt;% filter(stage == &quot;A&quot;)

new_data &lt;-
  tibble(line = Freda_2019c_m_data$line,
         s_male = 1) %&gt;% 
  distinct(line, s_male)

fitted_hh_18_m &lt;- fitted(heat_hardiness_18_m_model,
       newdata = new_data) %&gt;% 
  as_tibble()

heat_hardiness_18_m_line_means &lt;- tibble(new_data, fitted_hh_18_m) %&gt;%
  mutate(Trait = &quot;heat.hardiness.18C.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Freda et al (2019) Heredity&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         `Trait description` = &quot;Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater heat hardiness.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

#################

# Heat hardiness at 25C for larvae model

heat_hardiness_25_larvae_model &lt;- brm(alive | trials(n) ~ 1 + (1|line),
                       data = Freda_2019d %&gt;% filter(stage == &quot;L&quot;), family = binomial(),
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/hh_25_l.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 10))

 #pp_check(heat_hardiness_25_larvae_model)

Freda_2019d_larvae_data &lt;-
  Freda_2019d %&gt;% filter(stage == &quot;L&quot;)

new_data &lt;-
  tibble(line = Freda_2019d_larvae_data$line,
         n = 1) %&gt;% 
  distinct(line, n)

fitted_hh_25_l &lt;- fitted(heat_hardiness_25_larvae_model,
       newdata = new_data) %&gt;% 
  as_tibble()

# females

heat_hardiness_25_l_line_means &lt;- tibble(new_data, fitted_ch_25_l) %&gt;%
  mutate(Trait = &quot;heat.hardiness.25C.larvae&quot;,
         Sex = &quot;Both&quot;,
         Reference = &quot;Freda et al (2019) Heredity&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         `Trait description` = &quot;Mean proportion of third instar larvae that survived exposure to 38°C for an hour. These larvae were otherwise reared at 25°C. Higher values indicate greater heat hardiness.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)


# Heat hardiness at 25C for females model

heat_hardiness_25_f_model &lt;- brm(f_female | trials(s_female) ~ 1 + (1|line),
                       data = Freda_2019d %&gt;% filter(stage == &quot;A&quot;), family = binomial(),
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/hh_25_f.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 10))

 #pp_check(heat_hardiness_25_f_model)

Freda_2019d_f_data &lt;-
  Freda_2019d %&gt;% filter(stage == &quot;A&quot;)

new_data &lt;-
  tibble(line = Freda_2019d_f_data$line,
         s_female = 1) %&gt;% 
  distinct(line, s_female)

fitted_hh_25_f &lt;- fitted(heat_hardiness_25_f_model,
       newdata = new_data) %&gt;% 
  as_tibble()

heat_hardiness_25_f_line_means &lt;- tibble(new_data, fitted_hh_25_f) %&gt;%
  mutate(Trait = &quot;heat.hardiness.25C.f&quot;,
         Sex = &quot;Female&quot;,
         Reference = &quot;Freda et al (2019) Heredity&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         `Trait description` = &quot;Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater heat hardiness.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

# Heat hardiness at 25C for males model

heat_hardiness_25_m_model &lt;- brm(f_male | trials(s_male) ~ 1 + (1|line),
                       data = Freda_2019d %&gt;% filter(stage == &quot;A&quot;), family = binomial(),
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/hh_25_m.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 10))

 #pp_check(heat_hardiness_25_m_model)

Freda_2019d_m_data &lt;-
  Freda_2019d %&gt;% filter(stage == &quot;A&quot;)

new_data &lt;-
  tibble(line = Freda_2019d_m_data$line,
         s_male = 1) %&gt;% 
  distinct(line, s_male)

fitted_hh_25_m &lt;- fitted(heat_hardiness_25_m_model,
       newdata = new_data) %&gt;% 
  as_tibble()

heat_hardiness_25_m_line_means &lt;- tibble(new_data, fitted_hh_25_m) %&gt;%
  mutate(Trait = &quot;heat.hardiness.25C.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Freda et al (2019) Heredity&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         `Trait description` = &quot;Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater heat hardiness.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

Freda_2019_traits &lt;- rbind(cold_hardiness_18_l_line_means,
                           cold_hardiness_18_f_line_means,
                           cold_hardiness_18_m_line_means,
                           cold_hardiness_25_l_line_means,
                           cold_hardiness_25_f_line_means,
                           cold_hardiness_25_m_line_means,
                           heat_hardiness_18_l_line_means,
                           heat_hardiness_18_f_line_means,
                           heat_hardiness_18_m_line_means,
                           heat_hardiness_25_l_line_means,
                           heat_hardiness_25_f_line_means,
                           heat_hardiness_25_m_line_means)</code></pre>
</div>
<div id="pitchers-et-al-2019" class="section level3">
<h3>Pitchers et al 2019</h3>
<p>This study measures wing shape and area. They then run a multivariate
GWA. Here, we just model the mean centroid size for each line.</p>
<p>For whatever reason, this is a hard trait to model.</p>
<pre class="r"><code>Pitchers_2019_data &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Pitchers_2019_wing_multivariate.csv&quot;) %&gt;% 
  mutate(line = as.factor(Linenum),
         Lab = as.factor(Lab),
         Perp = as.factor(Perp),
         block = as.factor(block),
         Sex = as.factor(Sex))

Houle_data &lt;- Pitchers_2019_data %&gt;% 
  filter(Lab == &quot;Hou&quot;) %&gt;% 
  mutate(
    # standardise
    st_csmm = my_scale(csmm))

Dworkin_data &lt;- Pitchers_2019_data %&gt;% 
  filter(Lab == &quot;Dwo&quot;) %&gt;% 
   mutate(
    # standardise
    st_csmm = my_scale(csmm))

# Houle lab model

wing_centroid_size_model_Hou &lt;- brm(st_csmm ~ Sex + (Sex|line),
                      family = gaussian, data = Houle_data,
                      prior = c(prior(normal(0, 0.5), class = Intercept),
                                prior(normal(0, 0.5), class = b),
                                prior(exponential(1), class = sd)),
                      iter = 6000, warmup = 2000, chains = 4, cores = 4,
                      seed = 2, file = &quot;fits/raw_data_fits/Houle_cs_wing.model&quot;,
                      control = list(adapt_delta = 0.9, max_treedepth = 10))

#pp_check(wing_centroid_size_model_Hou)

new_data &lt;-
  tibble(line = Houle_data$line,
         Sex = Houle_data$Sex) %&gt;% 
  distinct(line, Sex)

fitted_wing_houle &lt;- fitted(wing_centroid_size_model_Hou,
       newdata = new_data,
       allow_new_levels = TRUE) %&gt;% # sometimes we need to include this for fitted to work. It changes nothing in the output
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Houle_data$csmm) + mean(Houle_data$csmm),
         Q2.5 = Q2.5*sd(Houle_data$csmm) + mean(Houle_data$csmm),
         Q97.5 = Q97.5*sd(Houle_data$csmm) + mean(Houle_data$csmm))

centroid_size_Houle_f_line_means &lt;- tibble(new_data, fitted_wing_houle) %&gt;%
  filter(Sex == &quot;F&quot;) %&gt;% 
  mutate(Trait = &quot;wing.centroid.size.2019_Houle.f&quot;,
         Sex = &quot;Female&quot;,
         Reference = &quot;Pitchers et al (2019) Evolution&quot;,
         `Trait guild` = &quot;Morphological&quot;,
         `Trait description` = &quot;Mean left wing centroid size for flies measured in the Houle lab. Images were taken of live flies using the &#39;Wingmachine&#39; system. Higher values indicate larger wings.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

centroid_size_Houle_m_line_means &lt;- 
  tibble(new_data, fitted_wing_houle) %&gt;%
  filter(Sex == &quot;M&quot;) %&gt;% 
  mutate(Trait = &quot;wing.centroid.size.2019_Houle.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Pitchers et al (2019) Evolution&quot;,
         `Trait guild` = &quot;Morphological&quot;,
         `Trait description` = &quot;Mean left wing centroid size for flies measured in the Houle lab. Images were taken of live flies using the &#39;Wingmachine&#39; system. Higher values indicate larger wings.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

# Dworkin lab model

wing_centroid_size_model_Dwo_f &lt;- brm(st_csmm ~ (1|line),
                      family = student, data = Dworkin_data %&gt;% filter(Sex == &quot;F&quot;),
                      prior = c(prior(normal(0, 0.5), class = Intercept),
                               #prior(normal(0, 0.5), class = b),
                                prior(exponential(1), class = sd)),
                      iter = 20000, warmup = 4000, chains = 4, cores = 4,
                      seed = 2, file = &quot;fits/raw_data_fits/Dwo_cs_wing.f.model&quot;,
                      control = list(adapt_delta = 0.9, max_treedepth = 12))

#pp_check(wing_centroid_size_model_Dwo_f)

dworkin_f &lt;- Dworkin_data %&gt;% filter(Sex == &quot;F&quot;)

new_data &lt;-
  tibble(line = dworkin_f$line,
         Sex = &quot;F&quot;) %&gt;% 
  distinct(line, Sex)

fitted_wing_dworkin_f &lt;- fitted(wing_centroid_size_model_Dwo_f,
       newdata = new_data,
       allow_new_levels = TRUE) %&gt;% # sometimes we need to include this for fitted to work. It changes nothing in the output 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Dworkin_data$csmm) + mean(Dworkin_data$csmm),
         Q2.5 = Q2.5*sd(Dworkin_data$csmm) + mean(Dworkin_data$csmm),
         Q97.5 = Q97.5*sd(Dworkin_data$csmm) + mean(Dworkin_data$csmm))

centroid_size_Dworkin_f_line_means &lt;- tibble(new_data, fitted_wing_dworkin_f) %&gt;%
  mutate(Trait = &quot;wing.centroid.size.2019_Dworkin.f&quot;,
         Sex = &quot;Female&quot;,
         Reference = &quot;Pitchers et al (2019) Evolution&quot;,
         `Trait guild` = &quot;Morphological&quot;,
         `Trait description` = &quot;Mean left wing centroid size for flies measured in the Dworkin lab. Images were taken of dissected wings using an Olympus microscope. Higher values indicate larger wings.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

wing_centroid_size_model_Dwo_m &lt;- brm(st_csmm ~ (1|line),
                      family = student, data = Dworkin_data %&gt;% filter(Sex == &quot;M&quot;),
                      prior = c(prior(normal(0, 0.5), class = Intercept),
                               #prior(normal(0, 0.5), class = b),
                                prior(exponential(1), class = sd)),
                      iter = 20000, warmup = 4000, chains = 4, cores = 4,
                      seed = 2, file = &quot;fits/raw_data_fits/Dwo_cs_wing.m.model&quot;,
                      control = list(adapt_delta = 0.9, max_treedepth = 12))


dworkin_m &lt;- Dworkin_data %&gt;% filter(Sex == &quot;M&quot;)

new_data &lt;-
  tibble(line = dworkin_m$line,
         Sex = &quot;M&quot;) %&gt;% 
  distinct(line, Sex)

fitted_wing_dworkin_m &lt;- fitted(wing_centroid_size_model_Dwo_m,
       newdata = new_data,
       allow_new_levels = TRUE) %&gt;% # sometimes we need to include this for fitted to work. It changes nothing in the output 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Dworkin_data$csmm) + mean(Dworkin_data$csmm),
         Q2.5 = Q2.5*sd(Dworkin_data$csmm) + mean(Dworkin_data$csmm),
         Q97.5 = Q97.5*sd(Dworkin_data$csmm) + mean(Dworkin_data$csmm))

centroid_size_Dworkin_m_line_means &lt;- tibble(new_data, fitted_wing_dworkin_m) %&gt;%
  mutate(Trait = &quot;wing.centroid.size.2019_Dworkin.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Pitchers et al (2019) Evolution&quot;,
         `Trait guild` = &quot;Morphological&quot;,
         `Trait description` = &quot;Mean wing centroid size for flies measured in the Dworkin lab. Images were taken of dissected wings using an Olympus microscope. Higher values indicate larger wings.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

Pitchers_traits &lt;- 
  rbind(centroid_size_Houle_f_line_means,
        centroid_size_Houle_m_line_means,
        centroid_size_Dworkin_f_line_means,
        centroid_size_Dworkin_m_line_means)</code></pre>
</div>
<div id="rohde-et-al-2019" class="section level3">
<h3>Rohde et al 2019</h3>
<pre class="r"><code>Rohde_2019_data &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Rohde_2019.csv&quot;) %&gt;% 
  mutate(line = as.factor(DGRP_id),
         arena_id = as.factor(arena_id),
         treatment = as.factor(treatment),
         day_id = as.factor(day_id),
         neighbour = as.factor(neighbour),
         plate_id = as.factor(plate_id),
         # standardise
         st_distance_cm = my_scale(distance_cm))

# distance travelled

activity_model &lt;- brm(st_distance_cm ~ treatment + (1|day_id) + (1|plate_id) + (treatment|line),
                      family = gaussian, data = Rohde_2019_data,
                      prior = c(prior(normal(0, 0.5), class = Intercept),
                                prior(normal(0, 0.5), class = b),
                                prior(exponential(1), class = sd)),
                      iter = 6000, warmup = 2000, chains = 4, cores = 4,
                      seed = 2, file = &quot;fits/raw_data_fits/activity.model&quot;,
                      control = list(adapt_delta = 0.9, max_treedepth = 10))

#pp_check(activity_model)

new_data &lt;-
  tibble(line = Rohde_2019_data$line,
         treatment = Rohde_2019_data$treatment) %&gt;% 
  distinct(line, treatment)


fitted_activity &lt;- fitted(activity_model,
       newdata = new_data,
       re_formula = distance_cm ~ (treatment|line)) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Rohde_2019_data$distance_cm) + mean(Rohde_2019_data$distance_cm),
         Q2.5 = Q2.5*sd(Rohde_2019_data$distance_cm) + mean(Rohde_2019_data$distance_cm),
         Q97.5 = Q97.5*sd(Rohde_2019_data$distance_cm) + mean(Rohde_2019_data$distance_cm))

activity_line_estimates &lt;- tibble(new_data, fitted_activity) %&gt;% 
  select(line, treatment, Estimate) %&gt;% 
  mutate(treatment = if_else(treatment == 1, &quot;MPH&quot;, &quot;SUC&quot;))

# first lets get line means for activity in the control sucrose treatment

activity_line_means &lt;- activity_line_estimates %&gt;%
  filter(treatment == &quot;SUC&quot;) %&gt;% 
  mutate(Trait = &quot;activity.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Rohde et al (2019) Genetics&quot;,
         `Trait guild` = &quot;Activity&quot;,
         `Trait description` = &quot;Mean distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height), following 24 hrs of exposure to a 5% sucrose solution diet. Higher values indicate more active flies.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

# now get line means for activity in the mph treatment

activity_mph_line_means &lt;- activity_line_estimates %&gt;%
  filter(treatment == &quot;MPH&quot;) %&gt;% 
  mutate(Trait = &quot;activity.MPH.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Rohde et al (2019) Genetics&quot;,
         `Trait guild` = &quot;Activity&quot;,
         `Trait description` = &quot;Mean distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height), following 24 hrs of exposure to a 5% sucrose solution diet, supplemented with MPH (1.5 mg ml^-1). MPH, methylphenidate, is a drug used to treat attention-deficit/hyperactivity disorder (ADHD) in humans. Higher values indicate more active flies.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

# now lets get means for the effect of mph on movement by getting the difference between treatments

SUC_activity &lt;- 
  activity_line_means %&gt;% 
  select(line, trait_value) %&gt;% 
  rename(SUC_estimate = trait_value)

mph_activity &lt;- 
  activity_mph_line_means %&gt;% 
  select(line, trait_value) %&gt;% 
  rename(mph_estimate = trait_value)

activity_estimates_diff &lt;-
  left_join(SUC_activity, mph_activity) %&gt;% 
  mutate(Estimate = mph_estimate - SUC_estimate)

mph_effect_line_means &lt;- activity_estimates_diff %&gt;%
  mutate(Trait = &quot;MPH.effect.on.activity.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Rohde et al (2019) Genetics&quot;,
         `Trait guild` = &quot;Drug response&quot;,
         `Trait description` = &quot;Mean difference in distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height) between flies fed a control 5% sucrose solution and a 5% sucrose solution supplemented with 1.5 mg ml^-1 methylphenidate (MPH). MPH is a drug used to treat the symptoms of attention-deficit/hyperactivity disorder (ADHD) in humans. Positive values indicate greater activity when flies were supplemented with MPH.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

Rohde_traits &lt;- rbind(activity_line_means,
                      activity_mph_line_means,
                      mph_effect_line_means)</code></pre>
</div>
<div id="everett-et-al-2020" class="section level3">
<h3>Everett et al 2020</h3>
<p>This is mainly data wrangling. We find the mean from two data points
for each line simply via the <code>mean()</code> function.</p>
<pre class="r"><code>Everett_2020_data &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Everett_microbiome_2020.csv&quot;) 

microbiome_line_means &lt;- 
  Everett_2020_data %&gt;% 
  pivot_longer(!`Microbial Species`, names_to = &quot;line&quot;, values_to = &quot;trait_value&quot;) %&gt;% 
  mutate(Sex = if_else(str_detect(line, &quot;F&quot;), &quot;Female&quot;, &quot;Male&quot;),
         line = str_remove(line, &quot;_F1&quot;),
         line = str_remove(line, &quot;_F2&quot;),
         line = str_remove(line, &quot;_M1&quot;),
         line = str_remove(line, &quot;_M2&quot;)) %&gt;% 
  group_by(`Microbial Species`, line, Sex) %&gt;% 
  summarise(trait_value = mean(trait_value)) %&gt;% 
  mutate( Reference = &quot;Everett et al (2020) Genome Research&quot;,
          `Trait guild` = &quot;Microbiome&quot;,
          `Trait description` = &quot;Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.&quot;,
          Trait = str_replace(`Microbial Species`, &quot; &quot;, &quot;.&quot;),
          Trait_1 = if_else(str_detect(Sex, &quot;Female&quot;), &quot;.f&quot;, &quot;.m&quot;)) %&gt;% 
  unite(Trait, Trait, Trait_1, sep = &quot;&quot;, remove = TRUE) %&gt;%
  ungroup() %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(Trait, line, Sex)</code></pre>
</div>
<div id="chapman-et-al-2020" class="section level3">
<h3>Chapman et al 2020</h3>
<p>Note that I haven’t included Wolbachia status in the model.</p>
<pre class="r"><code>Chapman_2020_data &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Chapman_2020.csv&quot;) %&gt;% 
   mutate(line = as.factor(Line),
          Treatment = as.factor(Treatment),
          Date = as.factor(Date))


E.faecalis_model &lt;- brm(Alive_at_day5 | trials(number_infected) ~ 1 + Infector + Date + (1|Vial) + (1|line),
                       data = Chapman_2020_data, family = binomial(),
                       prior = c(prior(normal(0, 0.5), class = Intercept),
                                 prior(normal(0, 0.5), class = b),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/E.faecalis.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 10))

 #pp_check(E.faecalis_model)

# set the fixed effects in the new data for which we will derive predictions to the reference levels in the model i.e. Infector = jo and Date = `5/3/17`

new_data &lt;-
  tibble(line = Chapman_2020_data$line,
         number_infected = 1,
         Infector = &quot;Jo&quot;,
         Date = &#39;5/3/17&#39;) %&gt;% 
  distinct(line, number_infected, Infector, Date) 


fitted_E.faecalis &lt;- fitted(E.faecalis_model,
       newdata = new_data,
       re_formula = Alive_at_day5 | trials(number_infected) ~ (1|line)) %&gt;% 
  as_tibble()

E.faecalis_line_means &lt;- tibble(new_data, fitted_E.faecalis) %&gt;%
  mutate(Trait = &quot;E.faecalis.resistance.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Chapman et al (2020) Genes&quot;,
         `Trait guild` = &quot;Pathogen response&quot;,
         `Trait description` = &quot;Mean proportion of adult flies that were alive 5 days after infection with the bacterial pathogen E. faecalis. Higher values indicate greater resistance to E. faecalis&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)</code></pre>
</div>
<div id="zhou-et-al-2020" class="section level3">
<h3>Zhou et al 2020</h3>
<p>Line means are provided for the non-metabolome traits in the
supplementary material. Raw data are provided for the metabolome traits,
which we model below.</p>
<p>There are 453 metabolites. We need a quick and easy way to model
these and find line means. To do this, we fit relatively simple gaussian
models with Sex as a fixed effect (where this is applicable) and line as
a random effect, with a sex-specific intercept.</p>
<p>Five traits show no genetic variation in one of the sexes; that is,
each line has the same value. We do not model these, but still need to
model the trait for the sex that shows genetic variation. We fit a very
similar model, except without the sex fixed effect.</p>
<pre class="r"><code>Zhou_data &lt;- 
  read_csv(&quot;data/data_collation/input/Raw_data_files/Zhou_2020.csv&quot;) %&gt;% 
  select(-c(`SUB PATHWAY`, `METABOLITE  ID`, PLATFORM, `CHEMICAL ID`,
            `RETENTION INDEX`, MASS, CAS, PUBCHEM, KEGG, `CLIENT IDENTIFIER`)) %&gt;% 
  pivot_longer(cols = 3:242, names_to = &quot;ID&quot;, values_to = &quot;Trait_value&quot;) %&gt;% 
  separate(ID, into = c(&quot;Line&quot;, &quot;Sex&quot;, &quot;Rep&quot;), sep = &quot;-&quot;, remove = TRUE) %&gt;% 
  mutate(line = as.factor(Line), Sex = as.factor(Sex)) %&gt;% 
  select(-Line) %&gt;% 
  # standardise
  group_by(METABOLITE, Sex) %&gt;% 
  mutate(st_trait_value = my_scale(Trait_value)) %&gt;% 
  ungroup() %&gt;%
  # some traits have the same value for every line; these can&#39;t be standardised because the SD = 0. We remove these traits from the dataset
  filter(st_trait_value != &quot;NaN&quot;)
  

# this is our model outline, based on one random metabolite. We can then update this model with new data (for another metabolite) to model every trait, which is advantageous because it prevents brms needing to recompile every time. With the model already compiled we can simply update the data, which saves a lot of time.

data &lt;- Zhou_data %&gt;% filter(METABOLITE == &quot;xanthurenate&quot;)

model_outline &lt;- brm(data = data,
             st_trait_value ~ 1 + Sex + (Sex|line),
             family = gaussian,
             prior = c(prior(normal(0, 0.5), class = Intercept),
                       prior(normal(0, 0.5), class = b),
                       prior(exponential(1), class = sd)),
             iter = 6000, warmup = 2000, chains = 4, 
             cores = 4, seed = 1, control = list(adapt_delta = 0.95, 
                                                 max_treedepth = 12))

## Now make a function to run a model for each trait and get the posterior line means for each sex 

metabolite_modeller &lt;- function(selected_trait){
  
  data &lt;- Zhou_data %&gt;% filter(METABOLITE == selected_trait)
  
  model &lt;- update(model_outline, newdata = data) # same model, but with new data each time
  
fitted_new_data &lt;-
  tibble(line = data$line,
         Sex = data$Sex,
         Trait = data$METABOLITE,
         `Trait guild` = &quot;Metabolome&quot;,
         `Trait description` = &quot;Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.&quot;,
         Reference = &quot;Zhou et al (2020) Genome Research&quot;) %&gt;% 
  distinct(line, Trait, Sex, `Trait guild`, `Trait description`, Reference)


fitted_metabolite &lt;- fitted(model,
       newdata = fitted_new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(trait_value = Estimate*sd(data$Trait_value) + mean(data$Trait_value),
         Q2.5 = Q2.5*sd(data$Trait_value) + mean(data$Trait_value),
         Q97.5 = Q97.5*sd(data$Trait_value) + mean(data$Trait_value))

metabolite_line_estimates &lt;- tibble(fitted_new_data, fitted_metabolite) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  mutate(Sex = if_else(Sex == &quot;F&quot;, &quot;Female&quot;, &quot;Male&quot;),
         Trait = if_else(Sex == &quot;Female&quot;, paste(Trait, &quot;f&quot;, sep = &quot;.&quot;), paste(Trait, &quot;m&quot;, sep = &quot;.&quot;)))
  
}

# create a list of all trait to iterate our function over

Both_sexes &lt;-
  Zhou_data %&gt;% 
  distinct(METABOLITE, Sex) %&gt;%
  group_by(METABOLITE) %&gt;% 
  summarise(n = n()) %&gt;% 
  filter(n &gt; 1)

trait_list_both_sexes &lt;- Both_sexes$METABOLITE

## Now make a function to run a model for each trait and get the posterior line means for traits only measured in a single sex

metabolite_modeller_single_sex &lt;- function(selected_trait){
  
  data &lt;- Zhou_data %&gt;% filter(METABOLITE == selected_trait)
  
  model &lt;- update(model_outline, newdata = data,
                  st_trait_value ~ 1 + (1|line),
                  family = gaussian,
                  prior = c(prior(normal(0, 0.5), class = Intercept),
                            prior(exponential(1), class = sd))) # same model, but with new data each time
  
  fitted_new_data &lt;-
    tibble(line = data$line,
           Sex = data$Sex,
           Trait = data$METABOLITE,
           `Trait guild` = &quot;Metabolome&quot;,
           `Trait description` = &quot;Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.&quot;,
           Reference = &quot;Zhou et al (2020) Genome Research&quot;) %&gt;% 
    distinct(line, Trait, Sex, `Trait guild`, `Trait description`, Reference)
  
  
  fitted_metabolite &lt;- fitted(model,
                              newdata = fitted_new_data) %&gt;% 
    as_tibble() %&gt;%
    # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
    mutate(trait_value = Estimate*sd(data$Trait_value) + mean(data$Trait_value),
           Q2.5 = Q2.5*sd(data$Trait_value) + mean(data$Trait_value),
           Q97.5 = Q97.5*sd(data$Trait_value) + mean(data$Trait_value))
  
  metabolite_line_estimates &lt;- tibble(fitted_new_data, fitted_metabolite) %&gt;% 
    select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
    mutate(Sex = if_else(Sex == &quot;F&quot;, &quot;Female&quot;, &quot;Male&quot;),
           Trait = if_else(Sex == &quot;Female&quot;, paste(Trait, &quot;f&quot;, sep = &quot;.&quot;), paste(Trait, &quot;m&quot;, sep = &quot;.&quot;)))
  
}

# create trait list to run function over

Single_sex &lt;-
  Zhou_data %&gt;% 
  distinct(METABOLITE, Sex) %&gt;%
  group_by(METABOLITE) %&gt;% 
  summarise(n = n()) %&gt;% 
  filter(n == 1)

trait_list_single_sex &lt;- Single_sex$METABOLITE

# Run the models

Run_function &lt;- FALSE # Change this to TRUE to re-run the models

if(Run_function){
  
  models_both_sexes &lt;- map_dfr(trait_list_both_sexes, metabolite_modeller) # map_dfr returns a data frame created by row-binding each output
  
  models_single_sex &lt;- map_dfr(trait_list_single_sex, metabolite_modeller_single_sex)
  
  Zhou_line_means &lt;-
    rbind(models_both_sexes,
          models_single_sex)
  
  Zhou_line_means %&gt;% 
    write_csv(&quot;data/data_collation/output/Zhou_line_means.csv&quot;)
} else {
  Zhou_line_means &lt;- read_csv(&quot;data/data_collation/output/Zhou_line_means.csv&quot;) %&gt;% 
    mutate(Trait = str_replace_all(Trait, &quot;[/]&quot;, &quot;_&quot;),
           Trait = str_remove(Trait, &quot;[*]&quot;))  # replace invalid names
}</code></pre>
</div>
<div id="zwoinska-et-al-2020" class="section level3">
<h3>Zwoinska et al 2020</h3>
<p>The authors treat temperature as a continuous variable. But they only
have measures at 3 temperatures, so we choose to consider temperature as
an ordered predictor, following Solomon Kurz’s <a
href="https://bookdown.org/content/4857/monsters-and-mixtures.html#ordered-categorical-predictors">translation</a>
of McElreath (2020) to the <code>brms</code> framework. This approach
results in an improved model fit.</p>
<pre class="r"><code>Zwoinska_male_data &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Zwoinska_2020/Zwoinska_males_2020.csv&quot;) %&gt;% 
  rename(line = DGRP) %&gt;% 
  mutate(line = as.factor(line),
         Batch = as.factor(Batch),
         Temperature_new = recode(Temperature,
                                  &quot;25&quot; = 1,
                                  &quot;27&quot; = 2,
                                  &quot;29&quot; = 3) %&gt;% 
           as.integer())


fertility_male_model_ordered &lt;-
  brm(Larvae ~ 1 + mo(Temperature_new) + (mo(Temperature_new)|line),
      data = Zwoinska_male_data,
      prior = c(prior(normal(0, 0.5), class = Intercept),
                prior(normal(0, 0.5), class = b),
                prior(dirichlet(2, 2), class = simo, coef = moTemperature_new1),
                prior(exponential(1), class = sd)),
      family = bernoulli,
      cores = 4, chains = 4, iter = 4000, warmup = 2000,
      control = list(adapt_delta = 0.9, max_treedepth = 10),
      seed = 1, file = &quot;fits/raw_data_fits/fertility_male.model_ordered&quot;)


new_data &lt;-
  tibble(line = Zwoinska_male_data$line,
         Temperature_new = Zwoinska_male_data$Temperature_new,
         Batch = 1) %&gt;% 
  distinct(line, Temperature_new, Batch)


fitted_male_fertility &lt;- fitted(fertility_male_model_ordered,
       newdata = new_data) %&gt;% 
  as_tibble()

fertility_m_line_estimates &lt;- tibble(new_data, fitted_male_fertility) %&gt;% 
  mutate(.keep = &quot;unused&quot;, 
            Temperature = recode(Temperature_new,
                                  &quot;1&quot; = 25,
                                  &quot;2&quot; = 27,
                                  &quot;3&quot; = 29)) 

# build a function to create the line means tibbles

Zwoinska_line_means &lt;- function(estimates, value, Trait, sex, description){
  estimates %&gt;%
  filter(Temperature == value) %&gt;% 
  mutate(Trait = Trait,
         Sex = sex,
         Reference = &quot;Zwoinska et al (2020) Frontiers in Genetics&quot;,
         `Trait guild` = &quot;Life history&quot;,
         `Trait description` = description,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)
}

fertility_25C_m_line_means &lt;- Zwoinska_line_means(fertility_m_line_estimates, &quot;25&quot;, &quot;fertility.25C.m&quot;, &quot;Male&quot;, &quot;Mean proportion of males that produced viable larvae. Males were reared at 25C. Higher values indicates greater fertility.&quot;)

fertility_27C_m_line_means &lt;- Zwoinska_line_means(fertility_m_line_estimates, &quot;27&quot;, &quot;fertility.27C.m&quot;, &quot;Male&quot;, &quot;Mean proportion of males that produced viable larvae. Males were reared at 27C. Higher values indicates greater fertility&quot;)

fertility_29C_m_line_means &lt;- Zwoinska_line_means(fertility_m_line_estimates, &quot;29&quot;, &quot;fertility.29C.m&quot;, &quot;Male&quot;, &quot;Mean proportion of males that produced viable larvae. Males were reared at 29C. Higher values indicates greater fertility&quot;)

# we need a plasticity measure

## The code below uses the slope estimated by the model for each line as the plasticity measure.

# get predictions from the model for each line (note fitted() isn&#39;t used, instead we grab the estimates straight from the models posterior). The line rankings produced here are very similar to the slopes estimates presented in Table S1 in Zwoinska et al (2020).

fertility_m_temp_effect_line_means &lt;- as_draws_df(fertility_male_model_ordered) %&gt;% 
  select(starts_with(&quot;r_line&quot;) &amp; contains(&quot;moTemperature&quot;)) %&gt;% 
  gather() %&gt;% 
  group_by(key) %&gt;%
  # the trait in question is plasticity, which is measured by the line specific random slope between treatments - we can get this straight from our posterior samples, without adding anything else 
  summarise(mean = mean(value) %&gt;% round(digits = 2)) %&gt;% 
  mutate(line = str_remove(key, &quot;r_line&quot;)) %&gt;% 
  mutate(line = str_remove(line, &quot;,moTemperature_new&quot;)) %&gt;% 
  mutate(line = gsub(&quot;\\[|\\]&quot;, &quot;&quot;, line),
         Trait = &quot;fertility.thermal.susceptibility.m&quot;,
         Reference = &quot;Zwoinska et al (2020) Frontiers in Genetics&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         Sex = &quot;Male&quot;,
         `Trait description` = &quot;Mean slope of the change in male fertility (the proportion of males that produced viable larvae), between 25C, 27C and 29C rearing conditions. Lower slope values indicate a greater decrease in fertility with an increase in temperature. This is a measure of thermal susceptibility.&quot;,
         trait_value = mean
  ) %&gt;%
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(trait_value)

# Slopes are hard to interpret, so we also provide this data in a different format. Below we calculate the change in fertility between 25C and 29C for each line. Greater values indicate a greater loss in fertility. Note this approach produces a very similar line ordering to the above approach

new_data &lt;-
  new_data %&gt;% 
  mutate(ID = paste0(&quot;V&quot;, 1:372))
  

fitted_male_fertility &lt;- fitted(fertility_male_model_ordered,
       newdata = new_data, summary = F) %&gt;% 
  as_tibble() %&gt;% 
  pivot_longer(cols = everything(), names_to = &quot;ID&quot;, values_to = &quot;Estimate&quot;)

fitted_male_fertility &lt;- right_join(new_data, fitted_male_fertility) %&gt;% 
  as_tibble %&gt;% 
  select(-ID)

# Different numbers of lines have been phenotyped for fertility in the 25C and 29C conditions. The code below finds the lines that are measured in one temp but not the other and removes them. We then find the difference in fertility between 25 and 29C conditions

a &lt;- 
  fitted_male_fertility %&gt;% 
  filter(Temperature_new == 1, line != &quot;38&quot; &amp; line != &quot;83&quot; &amp; line != &quot;395&quot;, line != &quot;707&quot; &amp; line != &quot;727&quot;) %&gt;%
  rename(Temp25 = Estimate)

c &lt;- 
  fitted_male_fertility %&gt;%
  filter(Temperature_new == 3, line != &quot;911&quot;) %&gt;%
  rename(Temp29 = Estimate) %&gt;%
  select(-c(Temperature_new, Batch, line))

#a_line &lt;- a %&gt;% distinct(line) 
#c_line &lt;- c %&gt;% distinct(line) # we remove the line column in the above code to make binding possible. This line of code will therefore not work, but it has already done its job.

#anti_join(a_line, c_line) # this line finds the lines present in the a_line vector but not in the b_line vector
#anti_join(c_line, a_line) # this line finds the lines present in the b_line vector but not in the a_line vector


backup_approach &lt;- cbind(a, c) %&gt;% 
  as_tibble() %&gt;% 
  mutate(fertility_diff = Temp25 - Temp29) %&gt;% 
  group_by(line) %&gt;% 
  summarise(trait_value = mean(fertility_diff)) %&gt;% 
  mutate(Trait = &quot;fertility.thermal.susceptibility.m&quot;,
         Reference = &quot;Zwoinska et al (2020) Frontiers in Genetics&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         Sex = &quot;Male&quot;,
         `Trait description` = &quot;Mean change in male fertility (the proportion of males that produced viable larvae), between 25C and 29C rearing conditions. Higher values indicate a greater decrease in fertility. This is a measure of thermal susceptibility or thermal plasticity in fertility.&quot;) %&gt;%
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(-trait_value)


# female fertility

Zwoinska_female_data &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Zwoinska_2020/Zwoinska_females_2020.csv&quot;) %&gt;% 
  rename(line = DGRP) %&gt;% 
  mutate(line = as.factor(line),
         Batch = as.factor(Batch),
         Temperature_new = recode(Temperature,
                                  &quot;25&quot; = 1,
                                  &quot;27&quot; = 2,
                                  &quot;29&quot; = 3) %&gt;% 
           as.integer())



fertility_female_model_ordered &lt;-
  brm(Larvae ~ 1 + mo(Temperature_new) + (mo(Temperature_new)|line),
      data = Zwoinska_female_data,
      prior = c(prior(normal(0, 0.5), class = Intercept),
                prior(normal(0, 0.5), class = b),
                prior(dirichlet(2, 2), class = simo, coef = moTemperature_new1),
                prior(exponential(1), class = sd)),
      family = bernoulli,
      cores = 4, chains = 4, iter = 4000, warmup = 2000,
      control = list(adapt_delta = 0.9, max_treedepth = 10),
      seed = 1, file = &quot;fits/raw_data_fits/fertility_female.model_ordered&quot;)


new_data &lt;-
  tibble(line = Zwoinska_female_data$line,
         Temperature_new = Zwoinska_female_data$Temperature_new,
         Batch = 1) %&gt;% 
  distinct(line, Temperature_new, Batch)


fitted_female_fertility &lt;- fitted(fertility_female_model_ordered,
       newdata = new_data) %&gt;% 
  as_tibble()

fertility_f_line_estimates &lt;- tibble(new_data, fitted_female_fertility) %&gt;% 
    mutate(.keep = &quot;unused&quot;, 
            Temperature = recode(Temperature_new,
                                  &quot;1&quot; = 25,
                                  &quot;2&quot; = 27,
                                  &quot;3&quot; = 29)) 

fertility_25C_f_line_means &lt;- Zwoinska_line_means(fertility_f_line_estimates, &quot;25&quot;, &quot;fertility.25C.f&quot;, &quot;Female&quot;, &quot;Mean proportion of females that produced viable larvae. Females were reared at 25C. Higher values indicates greater fertility.&quot;)

fertility_27C_f_line_means &lt;- Zwoinska_line_means(fertility_f_line_estimates, &quot;27&quot;, &quot;fertility.27C.f&quot;, &quot;Female&quot;, &quot;Mean proportion of females that produced viable larvae. Females were reared at 27C. Higher values indicates greater fertility&quot;)

fertility_29C_f_line_means &lt;- Zwoinska_line_means(fertility_f_line_estimates, &quot;29&quot;, &quot;fertility.29C.f&quot;, &quot;Female&quot;, &quot;Mean proportion of females that produced viable larvae. Females were reared at 29C. Higher values indicates greater fertility&quot;)

# get predictions from the model for each line (note fitted() isn&#39;t used, instead we grab the estimates straight from the models posterior). Once again, the line rankings produced here are very similar to the slopes estimates presented in Table S1 in Zwoinska et al (2020).

# we need a plasticity measure

## The code below uses the slope estimated by the model for each line as the plasticity measure.

# get predictions from the model for each line (note fitted() isn&#39;t used, instead we grab the estimates straight from the models posterior). The line rankings produced here are very similar to the slopes estimates presented in Table S1 in Zwoinska et al (2020).

fertility_f_temp_effect_line_means &lt;- as_draws_df(fertility_female_model_ordered) %&gt;% 
  select(starts_with(&quot;r_line&quot;) &amp; contains(&quot;moTemperature&quot;)) %&gt;% 
  gather() %&gt;% 
  group_by(key) %&gt;%
  # the trait in question is plasticity, which is measured by the line specific random slope between treatments - we can get this straight from our posterior samples, without adding anything else 
  summarise(mean = mean(value) %&gt;% round(digits = 2)) %&gt;% 
  mutate(line = str_remove(key, &quot;r_line&quot;)) %&gt;% 
  mutate(line = str_remove(line, &quot;,moTemperature_new&quot;)) %&gt;% 
  mutate(line = gsub(&quot;\\[|\\]&quot;, &quot;&quot;, line),
         Trait = &quot;fertility.thermal.susceptibility.f&quot;,
         Reference = &quot;Zwoinska et al (2020) Frontiers in Genetics&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         Sex = &quot;Female&quot;,
         `Trait description` = &quot;Mean slope of the change in female fertility (the proportion of males that produced viable larvae), between 25C, 27C and 29C rearing conditions. Lower slope values indicate a greater decrease in fertility with an increase in temperature. This is a measure of thermal susceptibility.&quot;,
         trait_value = mean
  ) %&gt;%
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(trait_value)

# Slopes are hard to interpret, so we also provide this data in a different format. Below we calculate the change in fertility between 25C and 29C for each line. Greater values indicate a greater loss in fertility. Note this approach produces a very similar line ordering to the bove approach

new_data &lt;-
  new_data %&gt;% 
  mutate(ID = paste0(&quot;V&quot;, 1:120))
  

fitted_female_fertility &lt;- fitted(fertility_female_model_ordered,
       newdata = new_data, summary = F) %&gt;% 
  as_tibble() %&gt;% 
  pivot_longer(cols = everything(), names_to = &quot;ID&quot;, values_to = &quot;Estimate&quot;)

fitted_female_fertility &lt;- right_join(new_data, fitted_female_fertility) %&gt;% 
  as_tibble %&gt;% 
  select(-ID)

# phenotypes for all 40 lines are measured so no fancy steps for binding are needed

a_female &lt;- 
  fitted_female_fertility %&gt;% 
  filter(Temperature_new == 1) %&gt;%
  rename(Temp25 = Estimate)

c_female &lt;- 
  fitted_female_fertility %&gt;%
  filter(Temperature_new == 3) %&gt;%
  rename(Temp29 = Estimate) %&gt;%
  select(-c(Temperature_new, Batch, line))


backup_approach &lt;- cbind(a_female, c_female) %&gt;% 
  as_tibble() %&gt;% 
  mutate(fertility_diff = Temp25 - Temp29) %&gt;% 
  group_by(line) %&gt;% 
  summarise(trait_value = mean(fertility_diff)) %&gt;% 
  mutate(Trait = &quot;fertility.thermal.susceptibility.f&quot;,
         Reference = &quot;Zwoinska et al (2020) Frontiers in Genetics&quot;,
         `Trait guild` = &quot;Temperature related&quot;,
         Sex = &quot;Female&quot;,
         `Trait description` = &quot;Mean change in female fertility (the proportion of males that produced viable larvae), between 25C and 29C rearing conditions. Higher values indicate a greater decrease in fertility. This is a measure of thermal susceptibility or thermal plasticity in fertility.&quot;) %&gt;%
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(-trait_value)


# combine all the data from this study into a simple dataframe

Zwoinska_traits &lt;- rbind(fertility_25C_m_line_means,
                         fertility_27C_m_line_means,
                         fertility_29C_m_line_means,
                         fertility_m_temp_effect_line_means,
                         fertility_25C_f_line_means,
                         fertility_27C_f_line_means,
                         fertility_29C_f_line_means,
                         fertility_f_temp_effect_line_means)</code></pre>
</div>
<div id="chowdhury-et-al-2021" class="section level3">
<h3>Chowdhury et al 2021</h3>
<pre class="r"><code>Chowdhury_2021_aggressive_lunges &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Chowdhury_2021_aggression.csv&quot;) %&gt;% 
  mutate(line = as.factor(line))


aggression_model &lt;- brm(agressive.lunges.per.20mins.m ~ 1 + (1|line),
                       data = Chowdhury_2021_aggressive_lunges, family = zero_inflated_poisson(),
                       prior = c(prior(normal(3, 0.5), class = Intercept),
                                 prior(exponential(1), class = sd)),
                       iter = 6000, warmup = 2000, chains = 4, cores = 4,
                       seed = 2, file = &quot;fits/raw_data_fits/aggression.model&quot;,
                       control = list(adapt_delta = 0.9, max_treedepth = 10))

 #pp_check(aggression_model)


new_data &lt;-
  tibble(line = Chowdhury_2021_aggressive_lunges$line) %&gt;% 
  distinct(line)

fitted_aggression &lt;- fitted(aggression_model,
       newdata = new_data) %&gt;% 
  as_tibble()



aggression_m_line_means &lt;- tibble(new_data, fitted_aggression) %&gt;%
  mutate(Trait = &quot;aggression.2021.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Chowdhury et al Nature Communications Biology&quot;,
         `Trait guild` = &quot;Behavioural&quot;,
         `Trait description` = &quot;Mean number of aggressive lunges made in 20 min window following removal of a divider, by pairs of DGRP males of same genotype. Trials were run after males were kept isolated for five days. Higher values indicate more aggressive males.&quot;,
         trait_value = Estimate) %&gt;% 
  mutate(line = str_remove(line, &quot;RAL-&quot;)) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)</code></pre>
</div>
<div id="watanabe-and-riddle-2021" class="section level3">
<h3>Watanabe and Riddle 2021</h3>
<pre class="r"><code>Watanabe_climbing_data &lt;- read_csv(&quot;data/data_collation/input/Raw_data_files/Watanabe_Riddle_2021.csv&quot;) %&gt;% 
  rename(line = Line) %&gt;% 
  mutate(line = str_remove(line, &quot;line_&quot;),
         line = as.factor(line),
         Vial = as.factor(Vial))

# Females

Watanabe_female_data &lt;-
  Watanabe_climbing_data %&gt;% 
  filter(Sex == &quot;F&quot;) %&gt;% 
  mutate(# standardise
    st_Climbing = my_scale(Climbing))

climbing_female_model &lt;-
  brm(st_Climbing ~ 1 +  Treatment + (1|Vial) + (Treatment|line),
      data = Watanabe_female_data,
      prior = c(prior(normal(0, 0.5), class = Intercept),
                prior(normal(0, 0.5), class = b),
                prior(exponential(1), class = sd)),
      family = gaussian,
      cores = 4, chains = 4, iter = 6000, warmup = 2000,
      control = list(adapt_delta = 0.9, max_treedepth = 10),
      seed = 1, file = &quot;fits/raw_data_fits/climbing_female.model&quot;)

new_data &lt;-
  tibble(line = Watanabe_female_data$line,
         Treatment = Watanabe_female_data$Treatment,
         Vial = 1) %&gt;% 
  distinct(line, Treatment, Vial)


fitted_female_climbing &lt;- fitted(climbing_female_model,
       newdata = new_data) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Watanabe_female_data$Climbing) + mean(Watanabe_female_data$Climbing),
         Q2.5 = Q2.5*sd(Watanabe_female_data$Climbing) + mean(Watanabe_female_data$Climbing),
         Q97.5 = Q97.5*sd(Watanabe_female_data$Climbing) + mean(Watanabe_female_data$Climbing))

climbing_f_line_estimates &lt;- tibble(new_data, fitted_female_climbing)

# build a function to create the line means tibbles

Watanabe_line_means &lt;- function(estimates, value, Trait, sex, description){
  estimates %&gt;%
  filter(Treatment == value) %&gt;% 
  mutate(Trait = Trait,
         Sex = sex,
         Reference = &quot;Watanabe and Riddle (2021) Royal Society Open Science&quot;,
         `Trait guild` =&quot;Activity&quot;,
         `Trait description` = description,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)
}

climbing_f_line_means &lt;- Watanabe_line_means(climbing_f_line_estimates, &quot;C&quot;, &quot;climbing.f&quot;, &quot;Female&quot;, &quot;Mean climbing index, following five days of standard conditions. Higher values indicate greater climbing ability.&quot;)

climbing_post_exercise_f_line_means &lt;- Watanabe_line_means(climbing_f_line_estimates, &quot;T&quot;, &quot;climbing.post.exercise.f&quot;, &quot;Female&quot;, &quot;Mean climbing index, following five days consecutive days of forced exercise (two hours a day). Higher values indicate greater climbing ability.&quot;)

# get predictions from the model for each line (note fitted() isn&#39;t used, instead we grab the estimates straight from the models posterior). Once again, the line rankings produced here are very similar to the slopes estimates presented in Watanabe and Riddle (2021).

# Now get the difference between treatments

control_climbing_f &lt;- 
  climbing_f_line_means %&gt;% 
  select(line, trait_value) %&gt;% 
  rename(control_estimate = trait_value)

exercise_climbing_f &lt;- 
  climbing_post_exercise_f_line_means %&gt;% 
  select(line, trait_value) %&gt;% 
  rename(exercise_estimate = trait_value)

climbing_estimates_diff &lt;-
  left_join(control_climbing_f, exercise_climbing_f) %&gt;% 
  mutate(Estimate = exercise_estimate - control_estimate)

climbing_f_exercise_effect_line_means &lt;- climbing_estimates_diff %&gt;%
  mutate(Trait = &quot;climbing.exercise.effect.f&quot;,
         Sex = &quot;Female&quot;,
         Reference = &quot;Watanabe and Riddle (2021) Royal Society Open Science&quot;,
         `Trait guild` = &quot;Activity&quot;,
         `Trait description` = &quot;Mean change in the climbing ability of adults, between flies that were and were not exposed to exercise five days previously. Positive values indicate an increase in climbing ability with exercise.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)

# males

Watanabe_male_data &lt;-
  Watanabe_climbing_data %&gt;% 
  filter(Sex == &quot;M&quot;) %&gt;% 
  mutate(# standardise
    st_Climbing = my_scale(Climbing))

climbing_male_model &lt;-
  brm(st_Climbing ~ 1 +  Treatment + (1|Vial) + (Treatment|line),
      data = Watanabe_male_data,
      prior = c(prior(normal(0, 0.5), class = Intercept),
                prior(normal(0, 0.5), class = b),
                prior(exponential(1), class = sd)),
      family = gaussian,
      cores = 4, chains = 4, iter = 6000, warmup = 2000,
      control = list(adapt_delta = 0.99, max_treedepth = 10),
      seed = 1, file = &quot;fits/raw_data_fits/climbing_male.model&quot;)

#pp_check(climbing_male_model)

new_data &lt;-
  tibble(line = Watanabe_male_data$line,
         Treatment = Watanabe_male_data$Treatment,
         Vial = 1) %&gt;% 
  distinct(line, Treatment, Vial)


fitted_male_climbing &lt;- fitted(climbing_male_model,
       newdata = new_data,
       allow_new_levels = TRUE) %&gt;% 
  as_tibble() %&gt;%
  # now we need to convert back to response scale (note I don&#39;t convert Est.Error)
  mutate(Estimate = Estimate*sd(Watanabe_male_data$Climbing) + mean(Watanabe_male_data$Climbing),
         Q2.5 = Q2.5*sd(Watanabe_male_data$Climbing) + mean(Watanabe_male_data$Climbing),
         Q97.5 = Q97.5*sd(Watanabe_male_data$Climbing) + mean(Watanabe_male_data$Climbing))

climbing_m_line_estimates &lt;- tibble(new_data, fitted_male_climbing)

climbing_m_line_means &lt;- Watanabe_line_means(climbing_m_line_estimates, &quot;C&quot;, &quot;climbing.m&quot;, &quot;Male&quot;, &quot;Mean climbing index, following five days of standard conditions. Higher values indicate greater climbing ability.&quot;)

climbing_post_exercise_m_line_means &lt;- Watanabe_line_means(climbing_m_line_estimates, &quot;T&quot;, &quot;climbing.post.exercise.m&quot;, &quot;Male&quot;, &quot;Mean climbing index, following five days consecutive days of forced exercise (two hours a day). Higher values indicate greater climbing ability.&quot;)

# Now get the difference between treatments

control_climbing_m &lt;- 
  climbing_m_line_means %&gt;% 
  select(line, trait_value) %&gt;% 
  rename(control_estimate = trait_value)

exercise_climbing_m &lt;- 
  climbing_post_exercise_m_line_means %&gt;% 
  select(line, trait_value) %&gt;% 
  rename(exercise_estimate = trait_value)

climbing_estimates_diff_m &lt;-
  left_join(control_climbing_m, exercise_climbing_m) %&gt;% 
  mutate(Estimate = exercise_estimate - control_estimate)

climbing_m_exercise_effect_line_means &lt;- climbing_estimates_diff_m %&gt;%
  mutate(Trait = &quot;climbing.exercise.effect.m&quot;,
         Sex = &quot;Male&quot;,
         Reference = &quot;Watanabe and Riddle (2021) Royal Society Open Science&quot;,
         `Trait guild` = &quot;Activity&quot;,
         `Trait description` = &quot;Mean change in the climbing ability of adults, between flies that were and were not exposed to exercise five days previously. Positive values indicate an increase in climbing ability with exercise.&quot;,
         trait_value = Estimate) %&gt;% 
  select(line, Trait, Sex, trait_value, `Trait guild`, `Trait description`, Reference) %&gt;% 
  arrange(line)


Watanabe_traits &lt;- rbind(climbing_f_line_means,
                         climbing_f_exercise_effect_line_means,
                         climbing_post_exercise_f_line_means,
                         climbing_m_line_means,
                         climbing_m_exercise_effect_line_means,
                         climbing_post_exercise_m_line_means)</code></pre>
</div>
</div>
<div id="combine-and-save-all-the-estimated-line-means"
class="section level2">
<h2>Combine and save all the estimated line means</h2>
<pre class="r"><code>estimated_line_means &lt;-
  rbind(Jumbo_traits,
        Grubbs_traits,
        Dobson_traits,
        Gaertner_traits,
        caffeine_resistance_line_means,
        Appel_traits,
        Munoz_traits,
        Freda_2017_traits,
        boric_acid_line_means,
        Dean_traits,
        Duneau_traits,
        Palmer_traits,
        Teets_traits,
        Everman_traits,
        Freda_2019_traits,
        Pitchers_traits,
        Rohde_traits,
        microbiome_line_means,
        E.faecalis_line_means,
        Zhou_line_means,
        Zwoinska_traits,
        aggression_m_line_means,
        Watanabe_traits)

meta_data &lt;- 
  left_join(by = &quot;Trait&quot;,
            estimated_line_means %&gt;% 
              select(Trait, Sex, `Trait guild`, `Trait description`, Reference) %&gt;% 
              distinct(),
            read_csv(&quot;data/data_collation/input/Raw_data_files/Life_stage_screen.csv&quot;)
  ) %&gt;% 
  mutate(Life_stage = case_when(Life_stage == &quot;Juvenile&quot; ~ &quot;Juvenile&quot;,
                                is.na(Life_stage) ~ &quot;Adult&quot;))

trait_data &lt;- estimated_line_means %&gt;% 
  select(-Sex, -`Trait guild`, -`Trait description`, -Reference) 

write_csv(trait_data, &quot;data/data_collation/output/dgrp_phenos_calculated_from_raw_data.csv&quot;)
write_csv(meta_data, &quot;data/data_collation/output/dgrp_phenos_calculated_from_raw_data_meta_data.csv&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22621)

Matrix products: default


locale:
[1] LC_COLLATE=English_Germany.utf8  LC_CTYPE=English_Germany.utf8   
[3] LC_MONETARY=English_Germany.utf8 LC_NUMERIC=C                    
[5] LC_TIME=English_Germany.utf8    

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] broom_1.0.5     tidybayes_3.0.6 brms_2.20.4     Rcpp_1.0.11    
 [5] lubridate_1.9.3 forcats_1.0.0   stringr_1.5.0   dplyr_1.1.3    
 [9] purrr_1.0.2     readr_2.1.4     tidyr_1.3.0     tibble_3.2.1   
[13] ggplot2_3.4.4   tidyverse_2.0.0 Matrix_1.6-1.1  workflowr_1.7.1

loaded via a namespace (and not attached):
  [1] gridExtra_2.3        inline_0.3.19        rlang_1.1.1         
  [4] magrittr_2.0.3       git2r_0.32.0         matrixStats_1.0.0   
  [7] compiler_4.3.1       getPass_0.2-2        loo_2.6.0           
 [10] callr_3.7.3          vctrs_0.6.4          reshape2_1.4.4      
 [13] arrayhelpers_1.1-0   crayon_1.5.2         pkgconfig_2.0.3     
 [16] fastmap_1.1.1        backports_1.4.1      ellipsis_0.3.2      
 [19] utf8_1.2.3           threejs_0.3.3        promises_1.2.1      
 [22] rmarkdown_2.25       markdown_1.10        tzdb_0.4.0          
 [25] ps_1.7.5             bit_4.0.5            xfun_0.40           
 [28] cachem_1.0.8         jsonlite_1.8.7       later_1.3.1         
 [31] prettyunits_1.2.0    parallel_4.3.1       R6_2.5.1            
 [34] dygraphs_1.1.1.6     StanHeaders_2.26.28  bslib_0.5.1         
 [37] stringi_1.7.12       jquerylib_0.1.4      rstan_2.32.3        
 [40] knitr_1.44           zoo_1.8-12           base64enc_0.1-3     
 [43] bayesplot_1.10.0     httpuv_1.6.11        igraph_1.5.1        
 [46] timechange_0.2.0     tidyselect_1.2.0     rstudioapi_0.15.0   
 [49] abind_1.4-5          yaml_2.3.7           codetools_0.2-19    
 [52] miniUI_0.1.1.1       curl_5.1.0           processx_3.8.2      
 [55] pkgbuild_1.4.2       lattice_0.21-8       plyr_1.8.9          
 [58] shiny_1.7.5.1        withr_2.5.1          bridgesampling_1.1-2
 [61] posterior_1.4.1      coda_0.19-4          evaluate_0.22       
 [64] RcppParallel_5.1.7   ggdist_3.3.0         xts_0.13.1          
 [67] pillar_1.9.0         tensorA_0.36.2       whisker_0.4.1       
 [70] stats4_4.3.1         checkmate_2.2.0      DT_0.30             
 [73] shinyjs_2.1.0        distributional_0.3.2 generics_0.1.3      
 [76] vroom_1.6.4          rprojroot_2.0.3      hms_1.1.3           
 [79] rstantools_2.3.1.1   munsell_0.5.0        scales_1.3.0        
 [82] gtools_3.9.4         xtable_1.8-4         glue_1.6.2          
 [85] tools_4.3.1          diffobj_0.3.5        shinystan_2.6.0     
 [88] colourpicker_1.3.0   fs_1.6.3             mvtnorm_1.2-3       
 [91] grid_4.3.1           QuickJSR_1.0.7       crosstalk_1.2.0     
 [94] colorspace_2.1-0     nlme_3.1-162         cli_3.6.1           
 [97] svUnit_1.0.6         fansi_1.0.5          Brobdingnag_1.2-9   
[100] V8_4.4.0             gtable_0.3.4         sass_0.4.7          
[103] digest_0.6.33        htmlwidgets_1.6.2    farver_2.1.1        
[106] htmltools_0.5.6.1    lifecycle_1.0.4      httr_1.4.7          
[109] mime_0.12            bit64_4.0.5          shinythemes_1.2.0   </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
