<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Thomas Keaney and Luke Holman" />


<title>The response to selection, sexual antagonism and sexual concordance in the DGRP</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.30/datatables.js"></script>
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.4/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.4/js/jquery.dataTables.min.js"></script>
<link href="site_libs/dt-ext-scroller-1.13.4/css/scroller.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-scroller-1.13.4/js/dataTables.scroller.min.js"></script>
<script src="site_libs/pdfmake-1.13.4/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.13.4/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.13.4/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.13.4/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.13.4/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">The response to selection, sexual antagonism and sexual concordance in the DGRP</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Main_analysis.html">Phenome Analysis</a>
</li>
<li>
  <a href="Transcriptome_analysis.html">Transcriptome Analysis</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tomkeaney/DGRP_sexual_conflict">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">The response to selection, sexual
antagonism and sexual concordance in the DGRP</h1>
<h4 class="author">Thomas Keaney and Luke Holman</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-10-19
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DGRP_sexual_conflict/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210706code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20210706)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210706code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210706)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPsexualconflicttree4188afb015b72c7f1dbe96e738df3e0165c40079targetblank4188afba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/tree/4188afb015b72c7f1dbe96e738df3e0165c40079" target="_blank">4188afb</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPsexualconflicttree4188afb015b72c7f1dbe96e738df3e0165c40079targetblank4188afba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/tree/4188afb015b72c7f1dbe96e738df3e0165c40079" target="_blank">4188afb</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  %
    Untracked:  -Toms-laptop-2.Rhistory
    Untracked:  -Toms-laptop-3.Rhistory
    Untracked:  -Toms-laptop.Rhistory
    Untracked:  Chapter_2_Figure_S1.pdf
    Untracked:  Chapter_2_Figure_S2.pdf
    Untracked:  Conflict_plot_females.pdf
    Untracked:  Conflict_plot_male.pdf
    Untracked:  Figures.Rmd
    Untracked:  Manuscript/
    Untracked:  PRISMA.pptx
    Untracked:  R_transcriptome_medians.csv
    Untracked:  Reported_heritability.xlsx
    Untracked:  Useful_cuts.Rmd
    Untracked:  add_later.Rmd
    Untracked:  code/get_gene_annotations.R
    Untracked:  concordance_plot_sem.pdf
    Untracked:  conflict_plot.pdf
    Untracked:  data/RFF.csv
    Untracked:  data/RFM.csv
    Untracked:  data/RMF.csv
    Untracked:  data/RMM.csv
    Untracked:  data/R_summarised_transcriptome.csv
    Untracked:  data/all.dgrp.phenos_scaled.csv
    Untracked:  data/gene_anntotations.csv
    Untracked:  data/huang_transcriptome/
    Untracked:  data/meta_data_for_all_traits.csv
    Untracked:  data/trait_names.rds
    Untracked:  data/transcriptome_chunks/
    Untracked:  data/transcriptome_output/
    Untracked:  evidence_ratios.R
    Untracked:  figures/
    Untracked:  fits/
    Untracked:  height_example.pdf
    Untracked:  mag_R_sem.pdf
    Untracked:  sem_figures.R

Unstaged changes:
    Modified:   DGRP_sexual_conflict.Rproj
    Modified:   _workflowr.yml
    Modified:   analysis/Transcriptome_analysis.Rmd
    Deleted:    code/export_gwas_results_for_shiny_app.R
    Modified:   data/all.dgrp.phenos_unscaled.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Main_analysis.Rmd</code>) and
HTML (<code>docs/Main_analysis.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/4188afb015b72c7f1dbe96e738df3e0165c40079/analysis/Main_analysis.Rmd" target="_blank">4188afb</a>
</td>
<td>
tomkeaney
</td>
<td>
2023-10-19
</td>
<td>
small wording changes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/319c84c1118d4c1c59279734e1b3432ef1d08901/docs/Main_analysis.html" target="_blank">319c84c</a>
</td>
<td>
tomkeaney
</td>
<td>
2023-10-17
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/48c7e9f2ebf56a798c432436bb9708bd9ee04dce/analysis/Main_analysis.Rmd" target="_blank">48c7e9f</a>
</td>
<td>
tomkeaney
</td>
<td>
2023-10-17
</td>
<td>
Restarting the project
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/92d2a266a617cbb1e3a1c7b47944c9652db2265b/docs/Main_analysis.html" target="_blank">92d2a26</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/5657ce75fbdf5a994d4fa9538c7ed8c5e7c2e99f/analysis/Main_analysis.Rmd" target="_blank">5657ce7</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-08
</td>
<td>
Better Table
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/414b2171adbc17c49363957d161af555932493b2/docs/Main_analysis.html" target="_blank">414b217</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/250194869ce43d151819853a6fbabd0771860260/analysis/Main_analysis.Rmd" target="_blank">2501948</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-08
</td>
<td>
Change to Tom’s analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/c33ab4e3a685c2ce3bb298be4fcb89548fdab8e1/docs/Main_analysis.html" target="_blank">c33ab4e</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/ca66d6a65302aaddc4ca2148f95faa610000bffd/analysis/Main_analysis.Rmd" target="_blank">ca66d6a</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-08
</td>
<td>
Change to Tom’s analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/daa935de0473e2ecaedca3c65f414d2858ddc598/docs/Main_analysis.html" target="_blank">daa935d</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/78ece4f2dc8517669aedfed885c6aeb3ca33f5df/analysis/Main_analysis.Rmd" target="_blank">78ece4f</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-08
</td>
<td>
Change to Tom’s analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/8f6bea5959633fc7d0bf075e8016d706adcb67c6/docs/Main_analysis.html" target="_blank">8f6bea5</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/a382ebf99e13771bc8bfbd2adc28fa0374a64716/analysis/Main_analysis.Rmd" target="_blank">a382ebf</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-03
</td>
<td>
New version
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/b682a4fd7268268f71172d62d1c5978ef2c39b5c/docs/Main_analysis.html" target="_blank">b682a4f</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-03
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/6577470b4db064ba409fa582ea45abd1aa879c59/analysis/Main_analysis.Rmd" target="_blank">6577470</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-03
</td>
<td>
New version
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/7bb248d6c288794a8dd9f70601c6a3ae068460ce/analysis/Main_analysis.Rmd" target="_blank">7bb248d</a>
</td>
<td>
ausevo
</td>
<td>
2023-01-31
</td>
<td>
rework of line means results
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/6d630f0e00db1e9167097921e8f4c5bfaa873767/docs/Main_analysis.html" target="_blank">6d630f0</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/95f4a97a95b2694250d10afcf070261b37734fb8/analysis/Main_analysis.Rmd" target="_blank">95f4a97</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-12
</td>
<td>
including ideas section
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/7bf0ee08d2c8b770d1c2d8aca0b2931897462adb/docs/Main_analysis.html" target="_blank">7bf0ee0</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/5b9883ecc77ee0ab529aa2bd15efa0caf62b5f2b/analysis/Main_analysis.Rmd" target="_blank">5b9883e</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
<td>
Add GCTA model outputs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/8e2d8be42669bfba416c2c2fd3668766ab4ba1e2/analysis/Main_analysis.Rmd" target="_blank">8e2d8be</a>
</td>
<td>
tkeaney
</td>
<td>
2021-08-17
</td>
<td>
Merge branch ‘master’ of <a
href="https://github.com/tomkeaney/DGRP_sexual_conflict"
class="uri">https://github.com/tomkeaney/DGRP_sexual_conflict</a>
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/4b095f010dea866a35dd2f7f6b021f5e7e27bd0d/analysis/Main_analysis.Rmd" target="_blank">4b095f0</a>
</td>
<td>
tkeaney
</td>
<td>
2021-08-17
</td>
<td>
edits
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/73f011cc9d650add1dd20c9f43e888fb17e4ddc5/analysis/Main_analysis.Rmd" target="_blank">73f011c</a>
</td>
<td>
lukeholman
</td>
<td>
2021-08-06
</td>
<td>
Tiny edit (digits=3)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/043902fed2d66e5b988a2f7a2dd70c734e1e95f8/docs/Main_analysis.html" target="_blank">043902f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/959123f80add4c3747918d1937e125003c982035/analysis/Main_analysis.Rmd" target="_blank">959123f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
trying to fix a bug
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/769e48d5a4afe34bb3a05faaa53829aeaadcfc05/docs/Main_analysis.html" target="_blank">769e48d</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/340c41bb74226b0fe4c241fc5d79663d2210c387/analysis/Main_analysis.Rmd" target="_blank">340c41b</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
sexual dimorphism calculation added
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/4018580c9b086ed875bd241ca2028263395bd971/docs/Main_analysis.html" target="_blank">4018580</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/65b25971953ba68583cb15a9503035ac59b1a16f/analysis/Main_analysis.Rmd" target="_blank">65b2597</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
sexual dimorphism calculation added
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/83a0c4fdf0c246cc8f853dbeecd4ab8b252df66e/docs/Main_analysis.html" target="_blank">83a0c4f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/ddea55e2cfaed0dddfe93d9dad2d1154f0e6a451/analysis/Main_analysis.Rmd" target="_blank">ddea55e</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
sexual dimorphism calculation added
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/b364d30085d882105bbb010749bcadccd8fa123a/docs/Main_analysis.html" target="_blank">b364d30</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/e404cf9f54cc05b642050a7b0bc0070a6e55f8bb/analysis/Main_analysis.Rmd" target="_blank">e404cf9</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
<td>
progression on selection calculations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/Main_analysis.html" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/6aad8a86f4107b143e2e07bb036e5064adeca7cc/analysis/Main_analysis.Rmd" target="_blank">6aad8a8</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
<td>
progression on selection calculations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/11a83906f387d1ecbb39fc5ab60f5830bdac2d8f/docs/Main_analysis.html" target="_blank">11a8390</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/42b8f12ec94d1089c153f97f0e4090edd191a279/analysis/Main_analysis.Rmd" target="_blank">42b8f12</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-07
</td>
<td>
Get the site up and running
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages-and-the-data" class="section level1">
<h1>Load packages and the data</h1>
<p><span class="math inline">\(~\)</span></p>
<p>First load the packages and build helper functions</p>
<pre class="r"><code>library(tidyverse) # for tidy coding
library(MetBrewer) # for many nice colour palettes
library(rcartocolor) # more cool colours
library(kableExtra) # for scrolling tables
library(DT) # for interactive tables
library(patchwork) # to join mulitple plots nicely
library(brms) # for bayesian models
library(tidybayes) # for more bayesian things
library(bayestestR) # for the pd metric 
library(broom) # convert results of functions into tables
library(ggtext) # for markdown features in ggplot
library(ggrepel) # for plot labels in ggplot
library(ggnewscale) # to reset scales in ggplot 

# Create a function to build HTML searchable tables

my_data_table &lt;- function(df){
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c(&quot;Scroller&quot;,  &quot;Buttons&quot;),
    options = list(
      autoWidth = TRUE,
      dom = &#39;Bfrtip&#39;,
      deferRender=TRUE,
      scrollX=TRUE, scrollY=1000,
      scrollCollapse=TRUE,
      buttons =
        list(&#39;pageLength&#39;, &#39;colvis&#39;, &#39;csv&#39;, list(
          extend = &#39;pdf&#39;,
          pageSize = &#39;A4&#39;,
          orientation = &#39;landscape&#39;,
          filename = &#39;Trait_data&#39;)),
      pageLength = 2115
    )
  )
}</code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="selecting-data-appropriate-for-analysis"
class="section level1">
<h1>Selecting data appropriate for analysis</h1>
<p>We conducted a near-exhaustive search of the literature until January
2022, to obtain line mean estimates and associated meta-data for
quantitative traits that have been measured in the DGRP. We did not
include data collected for traits that had been measured in heterozygous
combinations of multiple DGRP lines. In total, we identified 125 studies
that reported line means or raw data for 2115 phenotypic traits. To
ready the data for analysis, we grouped trait values by trait and sex
and standardised the data to have mean = 0 and sd = 1. Sex-specific
standardised line means for each trait were then combined with their
standardised fitness estimates, obtained from <a href="">Wong and Holman
(2023)</a>. We also include some helpful metadata for downstream
analysis. We then pruned the dataset to only include traits that have
been measured in single-sex cohorts, in 80 or more lines. We also
removed ‘intermediate’ traits between genotype and phenotype from three
mass datasets on the microbiome and metabolome (Everett <em>et al</em>.
2020 and Jin <em>et al</em>. 2020).</p>
<pre class="r"><code># load in the data, note that traits have already been standardised

DGRP_data &lt;- 
  left_join(
    read_csv(&quot;data/all.dgrp.phenos_scaled.csv&quot;) %&gt;% 
      mutate(line = as.factor(line)),
    read_csv(&quot;data/meta_data_for_all_traits.csv&quot;) %&gt;%
      group_by(Reference) %&gt;% 
      mutate(study_ID = as.factor(cur_group_id()),
             Pooled = if_else(Sex == &quot;Pooled&quot;, &quot;Yes&quot;, &quot;No&quot;))) %&gt;% 
  left_join(read_rds(&quot;data/trait_names.rds&quot;))
  
  
# Apply the selection criteria with the filter() function, then add a column each for female and male fitness

trait_data &lt;-
  DGRP_data %&gt;%  filter(!str_detect(Trait, &quot;fitness&quot;),
                        `# lines measured` &gt;= 80 &amp;
                          Pooled != &quot;Yes&quot; &amp; 
                          Reference != &quot;Jin et al (2020) PLOS Genetics&quot; &amp; 
                          Reference != &quot;Everett et al (2020) Genome Research&quot; &amp;
                          !str_detect(Trait, &quot;dopamine.response.to.paraquat.2021.m&quot;)) %&gt;% # data for this trait was entered into the database incorrectly, it was only measured in 10 lines so should not be included in our analysis 
  
  # join the early life fitness data from Wong and Holman
  
  left_join(
    DGRP_data %&gt;%
      filter(str_detect(Trait, &quot;fitness.early.life&quot;)) %&gt;% 
      select(line, Trait, trait_value) %&gt;% 
      pivot_wider(names_from = Trait, values_from = trait_value) %&gt;% 
      rename(female_fitness = fitness.early.life.f, male_fitness = fitness.early.life.m))


clean_meta_data &lt;-
  trait_data %&gt;% 
  select(Trait_nice, Trait, Life_stage, `Trait guild`, study_ID, Trait_nice, Reference, `Trait description`) %&gt;% 
  distinct(Trait, .keep_all = TRUE)</code></pre>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Table S1</strong>. Traits included for analysis</p>
<pre class="r"><code># how many studies did we start with?

table_data &lt;- left_join(
  DGRP_data %&gt;%
    distinct(Trait, .keep_all = TRUE) %&gt;% 
    mutate(Measured_in = case_when(Sex == &quot;Female&quot; ~ &quot;Females&quot;,
                                   Sex == &quot;Male&quot; ~ &quot;Males&quot;,
                                   Sex == &quot;Pooled&quot; ~ &quot;Mixed sex cohorts&quot;)) %&gt;% 
    select(Trait_nice, Trait, Reference, `# lines measured`, Measured_in, `Trait description`),
  
  trait_data %&gt;%
    distinct(Trait) %&gt;% 
    mutate(Included = &quot;Yes&quot;)
) %&gt;%
  mutate(Included = if_else(is.na(Included), &quot;No&quot;, Included)) %&gt;% 
  select(-Trait) %&gt;% 
  rename(Trait = Trait_nice, `Measured in` = Measured_in) %&gt;% 
  arrange(desc(Included))

# Create a function to build HTML searchable tables

my_data_table(table_data %&gt;% 
               select(Trait, Reference, `# lines measured`, `Measured in`, Included, `Trait description`))</code></pre>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-4b302075163c5fdcf2ee" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-4b302075163c5fdcf2ee">{"x":{"filter":"none","vertical":false,"extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["2 3-butanedione olfactory attraction","2 3-butanedione olfactory attraction","2 3-butanedione olfactory latency 30 percentile","2 3-butanedione olfactory latency 30 percentile","2 3-butanedione olfactory latency 90 percentile","2 3-butanedione olfactory latency 90 percentile","2 3-butanedione olfactory response intensity","2 3-butanedione olfactory response intensity","2 3-butanedione olfactory strength attraction","2 3-butanedione olfactory strength attraction","Abdomen length 18C","Abdomen length 28C","Abdomen length thermal plasticity","Acclimated acute cold exposure resistance 18C","Acclimated acute cold exposure resistance 18C","Acclimated acute cold exposure resistance 25C","Acclimated acute cold exposure resistance 25C","Acute cold exposure resistance 18C -6","Acute cold exposure resistance 18C -6","Acute cold exposure resistance 18C -8","Acute cold exposure resistance 18C -8","Acute cold exposure resistance 25C","Acute cold exposure resistance 25C","Aggression ","Anoxia tolerance adult","Anoxia tolerance adult","Basal activity","Basal activity","CHC 11 &amp; 9-Me-C23","CHC 13- &amp; 11-Me-C25","CHC 2-Me-C22","CHC 2-Me-C22","CHC 2-Me-C24","CHC 2-Me-C24","CHC 2-Me-C25","CHC 2-Me-C25","CHC 2-Me-C26","CHC 2-Me-C27","CHC 2-Me-C28 &amp; 7,11-C29:2","CHC 2-Me-C28","CHC 2-Me-C29 &amp; 8,12-C30:2","CHC 2-Me-C29","CHC 2-Me-C30","CHC 3-Me-C23","CHC 3-Me-C25 &amp; 9-C26:1","CHC 3-Me-C25","CHC 3-Me-C27 &amp; 6,10-C28:2 &amp; 9-C28:1","CHC 5-C23:1","CHC 5-C23:1","CHC 5-C24:1","CHC 5-C24:1","CHC 5-C25:1","CHC 5-C25:1","CHC 5-C27:1","CHC 5-Me-C25","CHC 5,9-C25:2 &amp; 9-C25:1","CHC 5,9-C27:2 &amp; 9-C27:1","CHC 6-C23:1","CHC 6-C23:1","CHC 6-C24:1","CHC 6-C24:1","CHC 6-C26:1 &amp; i","CHC 6,10-C25:2","CHC 6,10-C25:2","CHC 6,10-C26:2","CHC 7-C23:1","CHC 7-C23:1","CHC 7-C24:1","CHC 7-C25:1","CHC 7-C25:1","CHC 7-C26:1","CHC 7-C26:1","CHC 7-C27:1","CHC 7-C27:1","CHC 7-C29:1","CHC 7-C29:1","CHC 7,11-C23:2","CHC 7,11-C24:2","CHC 7,11-C25:2","CHC 7,11-C26:2","CHC 7,11-C27:2 &amp; 2-Me-C26","CHC 7,11-C28:2 &amp; 2-Me-C27","CHC 7,11-C31:2","CHC 8-C25:1","CHC 8,12-C26:2","CHC 8,12-C28:2","CHC 9-C23:1","CHC 9-C23:1","CHC 9-C25:1","CHC 9-C27:1","CHC 9-C29:1 &amp; 5,9-C29:2","CHC 9,13-C25:2","CHC 9,13-C27:2","CHC n-C21","CHC n-C21","CHC n-C22","CHC n-C22","CHC n-C23","CHC n-C23","CHC n-C24","CHC n-C24","CHC n-C25","CHC n-C26","CHC n-C26","CHC n-C27","CHC n-C27","CHC n-C28","CHC n-C28","CHC n-C29","CHC n-C29","CHC n-C31","CHC n-C31","CHC NI 20","CHC NI 33","CHC NI 34","CHC NI 54","CHC NI 54","CHC NI 7","CHC x-C22:1a","CHC x-C22:1b","CHC x-C23:1","CHC x,y-C24:2","CHC x,y-C24:2","Chill coma recovery","Chill coma recovery","Chill coma recovery variance","Chill coma recovery variance","Chronic cold exposure resistance","Chronic cold exposure resistance","Circadian period chi","Circadian period chi","Circadian period mesa","Circadian period mesa","Circadian rythmicity index","Circadian rythmicity index","Climbing ability","Climbing ability","Climbing ability paraquat","Climbing ability paraquat","Climbing age decline high yeast","Climbing age decline low yeast","Cold shock acclimated old","Cold shock acclimated old","Cold shock acclimated young","Cold shock acclimated young","Cold shock old","Cold shock old","Cold shock young","Cold shock young","Critical thermal max ","Critical thermal max ","Critical thermal min ","Critical thermal min ","Critical thermal min 23C","Critical thermal min developmental stability","Critical thermal min plasticity","Critical thermal min plasticity variation","Critical thermal min within environment variation plasticity","Day sleep bout length","Day sleep bout length","Day sleep bout length SD","Day sleep bout length SD","Day sleep bouts","Day sleep bouts","Day sleep bouts SD","Day sleep bouts SD","Day sleep","Day sleep","Day sleep SD","Day sleep SD","DDT resistance knockdown","DDT resistance mortality","Dessication tolerance","Dessication tolerance","Developmental acclimation cold exposure","Developmental acclimation cold exposure","Drosophila C virus resistance","Egg adult development time ","Egg adult development time ","Ethanol sensitivity first exposure","Ethanol sensitivity first exposure","Ethanol sensitivity second exposure","Ethanol sensitivity second exposure","Ethanol tolerance","Ethanol tolerance","Exercise induced activity","Exercise induced activity","Fecundity","Fertilised egg retention score","Flight performance","Flight performance","Flock house virus resistance","Food intake","Food intake","Foraging survival","Glucose ","Glucose high glucose diet","Glucose low glucose diet","Glucose medium glucose diet","Glucose yeast poor diet","Glucose yeast rich diet","Glycerol ","Glycerol high glucose diet","Glycerol low glucose diet","Glycerol medium glucose diet","Glycogen ","Glycogen high glucose diet","Glycogen low glucose diet","Glycogen medium glucose diet","Gustatory response ethanol","Gustatory response ethanol","Gut permeability","Gut permeability radiation","Hydrogen peroxide resistance","Interocular distance","Interocular distance","Isolated aggression","Kujin virus resistance","L. monocytogenes load","L. monocytogenes MTD","Lead effect development time","Lead effect development time","Lead effect locomotor activity","Lead effect locomotor activity","Lead egg adult development time ","Lead egg adult development time ","Lead locomotor activity ","Lead locomotor activity ","Lifespan 18C","Lifespan 18C","Lifespan ","Lifespan 25C ","Lifespan 25C ","Lifespan 25C ","Lifespan 25C ","Lifespan 25C high yeast","Lifespan 25C low yeast","Lifespan 28C","Lifespan 28C","Lifespan following diapause ","Lifespan high sugar diet","Lifetime fecundity 25C","Locomotor activity ","Locomotor activity ","Locomotor activity","Locomotor frequency","M. anisopliae resistance","M. anisopliae resistance","Malathion mortality 12hr","Malathion mortality 12hr","Malathion mortality 24hr","Malathion mortality 24hr","Malathion mortality 3hr","Malathion mortality 3hr","Malathion mortality 6hr","Malathion mortality 6hr","Mass yeast poor diet","Mass yeast rich diet","Micro environmental plasticity lifespan 18C","Micro environmental plasticity lifespan 18C","Micro environmental plasticity lifespan 25C","Micro environmental plasticity lifespan 25C","Micro environmental plasticity lifespan 28C","Micro environmental plasticity lifespan 28C","MSB resistance","MSB resistance","Negative geotaxis","Negative geotaxis","Negative geotaxis msb","Negative geotaxis msb","Negative geotaxis sensitivity msb","Negative geotaxis sensitivity msb","Night sleep bout length","Night sleep bout length","Night sleep bout length SD","Night sleep bout length SD","Night sleep bouts","Night sleep bouts","Night sleep bouts SD","Night sleep bouts SD","Night sleep","Night sleep","Night sleep SD","Night sleep SD","Olfactory response 1-hexanol","Olfactory response 1-hexanol","Olfactory response 2-heptanone","Olfactory response 2-heptanone","Olfactory response 2-phenyl ethyl alcohol","Olfactory response 2-phenyl ethyl alcohol","Olfactory response acetophenone","Olfactory response acetophenone","Olfactory response benzaldehyde ","Olfactory response benzaldehyde ","Olfactory response benzaldehyde ","Olfactory response benzaldehyde ","Olfactory response citral","Olfactory response citral","Olfactory response D-carvone","Olfactory response D-carvone","Olfactory response ethanol","Olfactory response ethanol","Olfactory response ethylacetate","Olfactory response ethylacetate","Olfactory response ethylbutyrate","Olfactory response ethylbutyrate","Olfactory response eugenol","Olfactory response eugenol","Olfactory response heliol","Olfactory response heliol","Olfactory response hexal","Olfactory response hexal","Olfactory response L-carvone","Olfactory response L-carvone","Olfactory response methylSalicylate","Olfactory response methylSalicylate","Ovariole asymmetry","Ovariole number","P. aeruginosa defecation","P. aeruginosa fecal pH","P. aeruginosa intestinal mitosis","P. aeruginosa resistance ","P. aeruginosa resistance ","P. aeruginosa resistance ","P. entomophila resistance","P. rettgeri load","P. rettgeri load high glucose diet","P. rettgeri load low glucose diet","P. rettgeri load medium glucose diet","P. rettgeri resistance","P. rettgeri tolerance","Paraquat resistance ","Paraquat resistance ","Paraquat resistance","Paraquat resistance","Permethrin knockdown","Permethrin knockdown","Permethrin mortality 24hr","Permethrin mortality 24hr","Phototaxis mid age","Phototaxis mid age","Phototaxis old","Phototaxis old","Phototaxis young","Phototaxis young","Posterior lobe shape PC1","Posterior lobe shape PC2","Posterior lobe size","Protein ","Protein high glucose diet","Protein low glucose diet","Protein medium glucose diet","Radiation resistance","Rapid cold hardening 18C","Rapid cold hardening 18C","Rapid cold hardening 25C","Rapid cold hardening 25C","Rapid cold hardening old","Rapid cold hardening old","Rapid cold hardening young","Rapid cold hardening young","Sigma virus sensitivity","Social aggression","Social effect aggression","Song interpulse interval","Sprinter marathoner","Sprinter marathoner","Startle response ","Startle response ","Startle response  msb","Startle response  msb","Startle response","Startle response","Startle response variance","Startle response variance","Startle sensitivity msb","Startle sensitivity msb","Starvation resistance ","Starvation resistance ","Starvation resistance ","Starvation resistance high carb diet","Starvation resistance high fat diet","Starvation resistance high protein diet","Starvation resistance old","Starvation resistance old","Starvation resistance standard diet","Starvation resistance variance","Starvation resistance variance","Starvation resistance yeast poor diet","Starvation resistance yeast rich diet","Starvation resistance young","Starvation resistance young","Tergite5","Tergite6","Thorax length 18C","Thorax length 28C","Thorax length thermal plasticity","Triglyceride ","Triglyceride high glucose diet","Triglyceride low glucose diet","Triglyceride medium glucose diet","Triglyceride poor diet","Triglyceride yeast rich diet","Tunicamycin ER stress","Tunicamycin LT50","Virgin egg retention","Waking activity","Waking activity","Waking activity SD","Waking activity SD","Week1 fecundity 25C","Week3 fecundity 25C","Week5 fecundity 25C","Week7 fecundity 25C","Weight high glucose diet","Weight low glucose diet","Weight medium glucose diet","Wet weight","Wing centroid size","Wing centroid size","T1 leg length","T1 leg length","T2 leg length","T2 leg length","T1 femur","T1 femur","T2 femur","T2 femur","T1 tibia","T1 tibia","T2 tibia","T2 tibia","T1 tarsus","T1 tarsus","T2 tarsus","T2 tarsus","Weight anexic","Weight gnotobiotic","Weight microbiome effect","Glucose anexic","Glucose gnotobiotic","Glucose microbiome effect","Protein anexic","Protein gnotobiotic","Triglyceride gnotobiotic","Orient towards female","Approach female","Wing display frequency","Genital licking frequency","Attempted copulation frequency","Mating latency ","Caffeine resistance","Cold hardiness","Cold hardiness","Boric acid resistance","Deltamethrin resistance","Parathion resistance","Kallithea resistance LT50","Kallithea resistance LT50","Kallithea viral load","Starvation resistance ","Starvation resistance ","Wing centroid size _Houle","Wing centroid size _Houle","Wing centroid size _Dworkin","Wing centroid size _Dworkin","Activity","Activity MPH","MPH effect on activity","E faecalis resistance","Fertility 25C","Fertility 27C","Fertility 29C","Fertility thermal susceptibility","1_3-Methylhistidine high yeast","1_3-Methylhistidine low yeast","1-Methyladenosine high yeast","1-Methyladenosine low yeast","1-Methylguanosine high yeast","1-Methylguanosine low yeast","13-HODE high yeast","13-HODE low yeast","2-Hydroxyglutarate high yeast","2-Hydroxyglutarate low yeast","2-Phosphoglyceric acid high yeast","2-Phosphoglyceric acid low yeast","3-Hydroxybutyric acid high yeast","3-Hydroxybutyric acid low yeast","3-Hydroxykynurenine high yeast","3-Hydroxykynurenine low yeast","4-Pyridoxic acid high yeast","4-Pyridoxic acid low yeast","5-Aminovalerate high yeast","5-Aminovalerate low yeast","A tropicalis load","Abdomen curling same sex","Acetoacetate high yeast","Acetoacetate low yeast","Acetylcarnitine high yeast","Acetylcarnitine low yeast","Acetylcholine high yeast","Acetylcholine low yeast","Activity same sex","Adenine high yeast","Adenine low yeast","Adenosine high yeast","Adenosine low yeast","ADP high yeast","ADP low yeast","Agmatine high yeast","Agmatine low yeast","Alanine high yeast","Alanine low yeast","Allantoin high yeast","Allantoin low yeast","Alpha-amanitin resistance 0.2","Alpha-amanitin resistance 2","Alpha-amanitin resistance LC50","Alpha-Ketoglutaric acid high yeast","Alpha-Ketoglutaric acid low yeast","Amino acid pref post limitation","Amino acid pref post limitation","Aminoisobutyrate high yeast","Aminoisobutyrate low yeast","AMP high yeast","AMP low yeast","Anoxia tolerance larval","Anthranilate high yeast","Anthranilate low yeast","Arachidonate high yeast","Arachidonate low yeast","Arginine high yeast","Arginine low yeast","Asparagine high yeast","Asparagine low yeast","Aspartic acid high yeast","Aspartic acid low yeast","Azinphos methyl resistance 0.25","Azinphos methyl resistance 0.5","Azinphos methyl resistance 1","Azinphos methyl resistance 2","Azinphos methyl resistance LD50","Betaine high yeast","Betaine low yeast","Biotin high yeast","Biotin low yeast","Cadaverine high yeast","Cadaverine low yeast","Carnitine high yeast","Carnitine low yeast","CGMP high yeast","CGMP low yeast","Change cocaine preferenceA","Change cocaine preferenceA","Change cocaine preferenceB","Change cocaine preferenceB","Change methamphetamine preferenceA","Change methamphetamine preferenceA","Change methamphetamine preferenceB","Change methamphetamine preferenceB","Chlorantraniliprole resistance 0.5","Chlorantraniliprole resistance 1","Chlorantraniliprole resistance 2","Chlorantraniliprole resistance 3","Chlorantraniliprole resistance 4","Chlorantraniliprole resistance 5","Chlorantraniliprole resistance LC50","Choline high yeast","Choline low yeast","Citrate synthase activity","Citrate synthase activity","Citrulline high yeast","Citrulline low yeast","Climbing ability control exercise","Climbing ability post exercise","Climbing ability pre exercise","CMP high yeast","CMP low yeast","CO2 production","Cocaine preferenceA E1","Cocaine preferenceA E1","Cocaine preferenceA E3","Cocaine preferenceA E3","Cocaine preferenceB E1","Cocaine preferenceB E1","Cocaine preferenceB E3","Cocaine preferenceB E3","Cold hardiness -5C larvae","Critical thermal max ","Critical thermal max ","Cumulative cold tolerance","Cystathionine high yeast","Cystathionine low yeast","Cystine high yeast","Cystine low yeast","Cytidine high yeast","Cytidine low yeast","Cytosine high yeast","Cytosine low yeast","D-GA3P high yeast","D-GA3P low yeast","D-Leucic acid high yeast","D-Leucic acid low yeast","Dopamine content ","Dopamine content paraquat ","Dopamine response to paraquat ","Dry weight ","Dry weight ","Duration embryogenesis","E faecalis load","Eclosion to oviposition time 16C","Eclosion to oviposition time 23C","Egg adult dev time 20C","Egg adult dev time 25C","Egg adult dev time 30C","Egg adult development time ","Egg adult development time EtOH exposure","Egg adult viability  25C","Egg adult viability ","Egg adult viability  25C","Egg adult viability ","Egg adult viability 20C","Egg adult viability 28C","Egg adult viability 30C","Egg adult viability ethanol exposure","Egg larva viability","Egg pupa viability 20C","Egg pupa viability 25C","Egg pupa viability 30C","Eggs per day 16C","Eggs per day 23C","Epinephrine high yeast","Epinephrine low yeast","Ethanol sensitivity development time","Ethanol sensitivity startle response","Ethanol sensitivity startle response","Ethanol sensitivity viability","Fipronil resistance","Fitness early life","Fitness early life","Fitness late life","Fitness late life","Following same sex","Food intake control exercise","Food intake post exercise","G1P G6P F6P F1P high yeast","G1P G6P F6P F1P low yeast","Genome size","Geranyl pyrophosphate high yeast","Geranyl pyrophosphate low yeast","Glucoronate high yeast","Glucoronate low yeast","Glucose ","Glucose ","Glucose ","Glucose ","Glucose abdomen content control exercise","Glucose abdomen content post exercise","Glucose fructose preference","Glucose fructose preference","Glucose high yeast","Glucose larvae ","Glucose low yeast","Glucose sucrose preference","Glucose sucrose preference","Glucose thorax content control exercise","Glucose thorax content post exercise","Glucose trehalose preference","Glucose trehalose preference","Glutamic acid high yeast","Glutamic acid low yeast","Glutamine high yeast","Glutamine low yeast","Glyceraldehyde high yeast","Glyceraldehyde low yeast","Glycerate high yeast","Glycerate low yeast","Glycerol-3-P high yeast","Glycerol-3-P low yeast","Glycerol ","Glycerol ","Glycerol ","Glycerol ","Glycerol abdomen content control exercise","Glycerol abdomen content post exercise","Glycerol larvae ","Glycerol thorax content control exercise","Glycerol thorax content post exercise","Glycine high yeast","Glycine low yeast","Glycine pref post limitation","Glycine pref post limitation","Glycogen ","Glycogen ","Glycogen ","Glycogen ","Glycogen larvae ","GMP high yeast","GMP low yeast","Grooming","Grooming","Guanosine high yeast","Guanosine low yeast","Heat resistance","Histamine high yeast","Histamine low yeast","Histidine high yeast","Histidine low yeast","Homogentisate high yeast","Homogentisate low yeast","Hsp70 expression","Hypoxanthine high yeast","Hypoxanthine low yeast","Imidacloprid resistance 100ppm","Imidacloprid resistance 25ppm","IMP high yeast","IMP low yeast","Inosine high yeast","Inosine low yeast","Inositol high yeast","Inositol low yeast","Iso-leucine high yeast","Iso-leucine low yeast","IsoValeric acid high yeast","IsoValeric acid low yeast","L-Kynurenine high yeast","L-Kynurenine low yeast","L bevis load","L plantarum load","Lactate ","Lactate ","Lactate high yeast","Lactate larvae ","Lactate low yeast","Lactose high yeast","Lactose low yeast","Larvae length low yeast anexic","Larvae length low yeast L plantarum","Lead effect viability","Lead egg adult viability ","Leucine high yeast","Leucine low yeast","Licking same sex","Lifespan 16C","Lifespan 23C","Lifespan td50","Linoleic acid high yeast","Linoleic acid low yeast","Linolenic acid high yeast","Linolenic acid low yeast","Lysine high yeast","Lysine low yeast","Malondialdehyde high yeast","Malondialdehyde low yeast","Margaric acid high yeast","Margaric acid low yeast","Mass ","Mass ","Mass ","Mass ","Mass ","Mass post exercise","Mass post exercise","Mating latency ","Mating latency ","MeHg tolerance","Methamphetamine preferenceA E1","Methamphetamine preferenceA E1","Methamphetamine preferenceA E3","Methamphetamine preferenceA E3","Methamphetamine preferenceB E1","Methamphetamine preferenceB E1","Methamphetamine preferenceB E3","Methamphetamine preferenceB E3","Methionine high yeast","Methionine low yeast","Methylmalonate high yeast","Methylmalonate low yeast","Min pupation time 20C","Min pupation time 25C","Min pupation time 30C","MtDNA copy number","Mushroom abnormal alpha","Mushroom abnormal beta","Mushroom fusion beta","Mushroom missing alpha","Mushroom missing beta","Mushroom short alpha","Mushroom short beta","Mushroom thin alpha","Mushroom thin beta","Myristic acid high yeast","Myristic acid low yeast","N-AcetylGlycine high yeast","N-AcetylGlycine low yeast","Net reproductive rate 16C","Net reproductive rate 23C","Orientating same sex","Ornithine high yeast","Ornithine low yeast","Orotate high yeast","Orotate low yeast","Oxaloacetate high yeast","Oxaloacetate low yeast","Oxypurinol high yeast","Oxypurinol low yeast","P-xylene susceptibility","Pantothenate high yeast","Pantothenate low yeast","PEP high yeast","PEP low yeast","Phenylalanine high yeast","Phenylalanine low yeast","Phenylalanine pref post limitation","Phenylalanine pref post limitation","Phototaxis dark","Phototaxis dark","Pipecolate high yeast","Pipecolate low yeast","Pop growth rate 16C","Pop growth rate 23C","Progeny production","Progeny sired by first male","Proline high yeast","Proline low yeast","Protein ","Protein ","Protein ","Protein ","Protein abdomen content control exercise","Protein abdomen content post exercise","Protein larvae ","Protein thorax content control exercise","Protein thorax content post exercise","Pupal case length","Pupal mass 20C","Pupal mass 25C","Pupal mass 30C","Pupation height","Putrescine high yeast","Putrescine low yeast","Pyridoxal-5-P high yeast","Pyridoxal-5-P low yeast","Pyriprole resistance","Pyruvate high yeast","Pyruvate low yeast","Rapid cold hardening 16C","Rapid cold hardening 23C","Reactive oxygen species in gut","Remating proportion","Ribose-5-P high yeast","Ribose-5-P low yeast","Sarcosine high yeast","Sarcosine low yeast","Serine high yeast","Serine low yeast","Singing same sex","Social spacing","Social spacing","Sorbitol high yeast","Sorbitol low yeast","Startle response ","Startle response ","Startle response EtOH","Startle response EtOH","Starvation resistance larvae","Succinate high yeast","Succinate low yeast","Sucrose high yeast","Sucrose low yeast","Tapped same sex","Tapping same sex","Taurine high yeast","Taurine low yeast","Thorax length","Thorax length","Thorax width","Thorax width","Threonine high yeast","Threonine low yeast","Traumatic injury mortality","Triacylglycerol ","Triacylglycerol ","Triacylglycerol larvae ","Triglyceride ","Triglyceride ","Triglyceride abdomen content control exercise","Triglyceride abdomen content post exercise","Triglyceride thorax content control exercise","Triglyceride thorax content post exercise","Trimethylamine high yeast","Trimethylamine low yeast","Tryptophan high yeast","Tryptophan low yeast","Tyrosine high yeast","Tyrosine low yeast","Unfertilised eggs post mating","Uracil high yeast","Uracil low yeast","Urate high yeast","Urate low yeast","Uridine high yeast","Uridine low yeast","Valine high yeast","Valine low yeast","Weight control exercise","Weight post exercise","Wing imaginal disk size absolute","Wing imaginal disk size absolute","Wing imaginal disk size relative","Wing imaginal disk size relative","Wolbachia copy number","Xanthosine high yeast","Xanthosine low yeast","Xanthurenate high yeast","Xanthurenate low yeast","Mitochondrial state 3 respiration","Mitochondrial state 3 respiration","Mitochondrial state 4 respiration","Mitochondrial state 4 respiration","Mitochondrial PO ratio","Mitochondrial PO ratio","Glycogen anexic","Glycogen gnotobiotic","Glycogen microbiome effect","Protein microbiome effect","Triglyceride anexic","TAG glycerol microbiome effect","Punishment memory learning","Punishment memory learning","Relief memory learning","Relief memory learning","Wing size","Wing size","Copulation latency","Insemination capacity","Cold tolerance -1","Cold tolerance -2","Cold tolerance -3","Cold tolerance -4","Cold tolerance -5","Cold tolerance -6","Cold tolerance -7","Cold hardiness 18C larvae","Cold hardiness 18C","Cold hardiness 18C","Cold hardiness 25C larvae","Cold hardiness 25C","Cold hardiness 25C","Heat hardiness 18C larvae","Heat hardiness 18C","Heat hardiness 18C","Heat hardiness 25C larvae","Heat hardiness 25C","Heat hardiness 25C","Acetobacter aceti","Acetobacter aceti","Acetobacter cibinongensis","Acetobacter cibinongensis","Acetobacter ghanensis","Acetobacter ghanensis","Acetobacter indonesiensis","Acetobacter indonesiensis","Acetobacter malorum","Acetobacter malorum","Acetobacter nitrogenifigens","Acetobacter nitrogenifigens","Acetobacter okinawensis","Acetobacter okinawensis","Acetobacter orientalis","Acetobacter orientalis","Acetobacter papayae","Acetobacter papayae","Acetobacter pasteurianus","Acetobacter pasteurianus","Acetobacter persici","Acetobacter persici","Acetobacter pomorum","Acetobacter pomorum","Acetobacter senegalensis","Acetobacter senegalensis","Acetobacter syzygii","Acetobacter syzygii","Acetobacter tropicalis","Acetobacter tropicalis","Acidiphilium angustum","Acidiphilium angustum","Acidomonas methanolica","Acidomonas methanolica","Acidovorax sp ","Acidovorax sp ","Acidovorax temperans","Acidovorax temperans","Acinetobacter gyllenbergii","Acinetobacter gyllenbergii","Acinetobacter johnsonii","Acinetobacter johnsonii","Acinetobacter sp ","Acinetobacter sp ","Agrobacterium tumefaciens","Agrobacterium tumefaciens","Asaia prunellae","Asaia prunellae","Asaia sp ","Asaia sp ","Aspergillus nidulans","Aspergillus nidulans","Aspergillus terreus,Malassezia globosa","Aspergillus terreus,Malassezia globosa","Azospirillum brasilense","Azospirillum brasilense","Azospirillum thiophilum","Azospirillum thiophilum","Borrelia coriaceae","Borrelia coriaceae","Campylobacter mucosalis","Campylobacter mucosalis","Cardinium endosymbiont","Cardinium endosymbiont","Caulobacter henricii","Caulobacter henricii","Chaetomium globosum","Chaetomium globosum","Clostridium lentocellum","Clostridium lentocellum","Corynebacterium minutissimum","Corynebacterium minutissimum","Corynebacterium pseudogenitalium","Corynebacterium pseudogenitalium","Corynebacterium sp ","Corynebacterium sp ","Corynebacterium terpenotabidum","Corynebacterium terpenotabidum","Drosophila A Virus","Drosophila A Virus","Enterococcus hirae","Enterococcus hirae","Escherichia coli","Escherichia coli","Exophiala xenobiotica","Exophiala xenobiotica","Fimbriimonas ginsengisoli","Fimbriimonas ginsengisoli","Gardnerella vaginalis","Gardnerella vaginalis","Gemmata obscuriglobus","Gemmata obscuriglobus","Gemmata sp ","Gemmata sp ","Gluconacetobacter diazotrophicus","Gluconacetobacter diazotrophicus","Gluconobacter frateurii","Gluconobacter frateurii","Gluconobacter morbifer","Gluconobacter morbifer","Gluconobacter oxydans","Gluconobacter oxydans","Gluconobacter thailandicus","Gluconobacter thailandicus","Hymenobacter sp ","Hymenobacter sp ","Hymenobacter swuensis","Hymenobacter swuensis","Ideonella sp ","Ideonella sp ","Klebsiella pneumoniae","Klebsiella pneumoniae","Kocuria polaris","Kocuria polaris","Kocuria sp ","Kocuria sp ","Komagataeibacter europaeus","Komagataeibacter europaeus","Komagataeibacter hansenii","Komagataeibacter hansenii","Komagataeibacter intermedius","Komagataeibacter intermedius","Komagataeibacter kakiaceti","Komagataeibacter kakiaceti","Komagataeibacter medellinensis","Komagataeibacter medellinensis","Komagataeibacter oboediens","Komagataeibacter oboediens","Komagataeibacter rhaeticus","Komagataeibacter rhaeticus","Komagataeibacter xylinus","Komagataeibacter xylinus","Kozakia baliensis","Kozakia baliensis","Lactobacillus brevis","Lactobacillus brevis","Lactobacillus fabifermentans","Lactobacillus fabifermentans","Lactobacillus fermentum","Lactobacillus fermentum","Lactobacillus kisonensis","Lactobacillus kisonensis","Lactobacillus plantarum","Lactobacillus plantarum","Lactobacillus rhamnosus","Lactobacillus rhamnosus","Lactococcus lactis","Lactococcus lactis","Leuconostoc carnosum","Leuconostoc carnosum","Leuconostoc pseudomesenteroides","Leuconostoc pseudomesenteroides","Luteipulveratus mongoliensis","Luteipulveratus mongoliensis","Mesorhizobium sp ","Mesorhizobium sp ","Metarhizium brunneum","Metarhizium brunneum","Methylobacterium extorquens","Methylobacterium extorquens","Methylobacterium sp ","Methylobacterium sp ","Microbacterium sp ","Microbacterium sp ","Nora virus","Nora virus","Paenibacillus borealis","Paenibacillus borealis","Paenibacillus sp ","Paenibacillus sp ","Pectobacterium carotovorum","Pectobacterium carotovorum","Plasmodium vivax","Plasmodium vivax","Podospora anserina","Podospora anserina","Propionibacterium acidipropionici","Propionibacterium acidipropionici","Propionibacterium acnes","Propionibacterium acnes","Pseudodonghicola xiamenensis","Pseudodonghicola xiamenensis","Pseudomonas fuscovaginae","Pseudomonas fuscovaginae","Pseudomonas moraviensis","Pseudomonas moraviensis","Pseudomonas putida","Pseudomonas putida","Pseudomonas sp ","Pseudomonas sp ","Pseudomonas stutzeri","Pseudomonas stutzeri","Pseudomonas syringae","Pseudomonas syringae","Psychrobacter glacincola","Psychrobacter glacincola","Psychrobacter sp ","Psychrobacter sp ","Rhizobacter sp ","Rhizobacter sp ","Saccharomyces cerevisiae","Saccharomyces cerevisiae","Sphingomonas hengshuiensis","Sphingomonas hengshuiensis","Sphingomonas parapaucimobilis","Sphingomonas parapaucimobilis","Sphingomonas phyllosphaerae","Sphingomonas phyllosphaerae","Sphingomonas sp ","Sphingomonas sp ","Sphingomonas wittichii","Sphingomonas wittichii","Sporolactobacillus terrae","Sporolactobacillus terrae","Staphylococcus aureus","Staphylococcus aureus","Staphylococcus epidermidis","Staphylococcus epidermidis","Staphylococcus gallinarum","Staphylococcus gallinarum","Streptococcus pneumoniae","Streptococcus pneumoniae","Streptococcus salivarius","Streptococcus salivarius","Streptomyces griseoflavus","Streptomyces griseoflavus","Streptomyces pristinaespiralis","Streptomyces pristinaespiralis","Streptomyces sp ","Streptomyces sp ","Sum of Non-Wolbachia Species","Sum of Non-Wolbachia Species","Terriglobus roseus","Terriglobus roseus","Tremella mesenterica","Tremella mesenterica","Wolbachia pipentis","Wolbachia pipentis","Xanthomonas sp ","Xanthomonas sp ","Xanthomonas vesicatoria","Xanthomonas vesicatoria","Yersinia mollaretii","Yersinia mollaretii","Yersinia pekkanenii","Yersinia pekkanenii","Zymomonas mobilis","Zymomonas mobilis","1-(1-enyl-palmitoyl)-2-linoleoyl-GPE (P-16:0_18:2)","1-(1-enyl-palmitoyl)-2-linoleoyl-GPE (P-16:0_18:2)","1-(1-enyl-palmitoyl)-2-oleoyl-GPC (P-16:0_18:1)","1-(1-enyl-palmitoyl)-2-oleoyl-GPC (P-16:0_18:1)","1-(1-enyl-palmitoyl)-2-oleoyl-GPE (P-16:0_18:1)","1-(1-enyl-palmitoyl)-2-oleoyl-GPE (P-16:0_18:1)","1-(1-enyl-palmitoyl)-GPE (P-16:0)","1-(1-enyl-palmitoyl)-GPE (P-16:0)","1-(1-enyl-stearoyl)-2-linoleoyl-GPE (P-18:0_18:2)","1-(1-enyl-stearoyl)-2-linoleoyl-GPE (P-18:0_18:2)","1-(1-enyl-stearoyl)-2-oleoyl-GPE (P-18:0_18:1)","1-(1-enyl-stearoyl)-2-oleoyl-GPE (P-18:0_18:1)","1-(1-enyl-stearoyl)-GPE (P-18:0)","1-(1-enyl-stearoyl)-GPE (P-18:0)","1-kestose","1-kestose","1-linolenoyl-GPC (18:3)","1-linolenoyl-GPC (18:3)","1-linoleoyl-2-linolenoyl-galactosylglycerol (18:2_18:3)","1-linoleoyl-2-linolenoyl-galactosylglycerol (18:2_18:3)","1-linoleoyl-2-linolenoyl-GPC (18:2_18:3)","1-linoleoyl-2-linolenoyl-GPC (18:2_18:3)","1-linoleoyl-GPA (18:2)","1-linoleoyl-GPA (18:2)","1-linoleoyl-GPC (18:2)","1-linoleoyl-GPC (18:2)","1-linoleoyl-GPE (18:2)","1-linoleoyl-GPE (18:2)","1-linoleoyl-GPG (18:2)","1-linoleoyl-GPG (18:2)","1-linoleoyl-GPI (18:2)","1-linoleoyl-GPI (18:2)","1-linoleoyl-GPS (18:2)","1-linoleoyl-GPS (18:2)","1-linoleoylglycerol (18:2)","1-linoleoylglycerol (18:2)","1-methyladenine","1-methyladenine","1-methylhistamine","1-methylhistamine","1-methylhistidine","1-methylhistidine","1-methylimidazoleacetate","1-methylimidazoleacetate","1-methylurate","1-methylurate","1-oleoyl-2-linoleoyl-glycerol (18:1_18:2)","1-oleoyl-2-linoleoyl-glycerol (18:1_18:2)","1-oleoyl-2-linoleoyl-GPC (18:1_18:2)","1-oleoyl-2-linoleoyl-GPC (18:1_18:2)","1-oleoyl-2-linoleoyl-GPE (18:1_18:2)","1-oleoyl-2-linoleoyl-GPE (18:1_18:2)","1-oleoyl-2-linoleoyl-GPI (18:1_18:2)","1-oleoyl-2-linoleoyl-GPI (18:1_18:2)","1-oleoyl-3-linoleoyl-glycerol (18:1_18:2)","1-oleoyl-3-linoleoyl-glycerol (18:1_18:2)","1-oleoyl-GPA (18:1)","1-oleoyl-GPA (18:1)","1-oleoyl-GPC (18:1)","1-oleoyl-GPC (18:1)","1-oleoyl-GPE (18:1)","1-oleoyl-GPE (18:1)","1-oleoyl-GPG (18:1)","1-oleoyl-GPG (18:1)","1-oleoyl-GPI (18:1)","1-oleoyl-GPI (18:1)","1-oleoyl-GPS (18:1)","1-oleoyl-GPS (18:1)","1-palmitoleoyl-2-linolenoyl-GPC (16:1_18:3)","1-palmitoleoyl-2-linolenoyl-GPC (16:1_18:3)","1-palmitoleoyl-2-linoleoyl-GPC (16:1_18:2)","1-palmitoleoyl-2-linoleoyl-GPC (16:1_18:2)","1-palmitoleoyl-2-oleoyl-glycerol (16:1_18:1)","1-palmitoleoyl-2-oleoyl-glycerol (16:1_18:1)","1-palmitoleoyl-3-oleoyl-glycerol (16:1_18:1)","1-palmitoleoyl-3-oleoyl-glycerol (16:1_18:1)","1-palmitoleoyl-GPC (16:1)","1-palmitoleoyl-GPC (16:1)","1-palmitoyl-2-linolenoyl-galactosylglycerol (16:0_18:3)","1-palmitoyl-2-linolenoyl-galactosylglycerol (16:0_18:3)","1-palmitoyl-2-linoleoyl-galactosylglycerol (16:0_18:2)","1-palmitoyl-2-linoleoyl-galactosylglycerol (16:0_18:2)","1-palmitoyl-2-linoleoyl-glycerol (16:0_18:2)","1-palmitoyl-2-linoleoyl-glycerol (16:0_18:2)","1-palmitoyl-2-linoleoyl-GPC (16:0_18:2)","1-palmitoyl-2-linoleoyl-GPC (16:0_18:2)","1-palmitoyl-2-linoleoyl-GPE (16:0_18:2)","1-palmitoyl-2-linoleoyl-GPE (16:0_18:2)","1-palmitoyl-2-linoleoyl-GPI (16:0_18:2)","1-palmitoyl-2-linoleoyl-GPI (16:0_18:2)","1-palmitoyl-2-oleoyl-GPC (16:0_18:1)","1-palmitoyl-2-oleoyl-GPC (16:0_18:1)","1-palmitoyl-2-oleoyl-GPE (16:0_18:1)","1-palmitoyl-2-oleoyl-GPE (16:0_18:1)","1-palmitoyl-2-palmitoleoyl-GPC (16:0_16:1)","1-palmitoyl-2-palmitoleoyl-GPC (16:0_16:1)","1-palmitoyl-2-stearoyl-GPC (16:0_18:0)","1-palmitoyl-2-stearoyl-GPC (16:0_18:0)","1-palmitoyl-3-linoleoyl-glycerol (16:0_18:2)","1-palmitoyl-3-linoleoyl-glycerol (16:0_18:2)","1-palmitoyl-GPA (16:0)","1-palmitoyl-GPA (16:0)","1-palmitoyl-GPC (16:0)","1-palmitoyl-GPC (16:0)","1-palmitoyl-GPE (16:0)","1-palmitoyl-GPE (16:0)","1-palmitoyl-GPG (16:0)","1-palmitoyl-GPG (16:0)","1-palmitoyl-GPI (16:0)","1-palmitoyl-GPI (16:0)","1-stearoyl-2-linoleoyl-GPC (18:0_18:2)","1-stearoyl-2-linoleoyl-GPC (18:0_18:2)","1-stearoyl-2-linoleoyl-GPE (18:0_18:2)","1-stearoyl-2-linoleoyl-GPE (18:0_18:2)","1-stearoyl-2-oleoyl-GPC (18:0_18:1)","1-stearoyl-2-oleoyl-GPC (18:0_18:1)","1-stearoyl-2-oleoyl-GPE (18:0_18:1)","1-stearoyl-2-oleoyl-GPE (18:0_18:1)","1-stearoyl-GPC (18:0)","1-stearoyl-GPC (18:0)","1-stearoyl-GPE (18:0)","1-stearoyl-GPE (18:0)","1,2-dilinoleoyl-galactosylglycerol (18:2_18:2)","1,2-dilinoleoyl-galactosylglycerol (18:2_18:2)","1,2-dilinoleoyl-GPC (18:2_18:2)","1,2-dilinoleoyl-GPC (18:2_18:2)","1,2-dilinoleoyl-GPE (18:2_18:2)","1,2-dilinoleoyl-GPE (18:2_18:2)","1,2-dioleoyl-GPC (18:1_18:1)","1,2-dioleoyl-GPC (18:1_18:1)","1,2-dioleoyl-GPE (18:1_18:1)","1,2-dioleoyl-GPE (18:1_18:1)","1,2-dipalmitoleoyl-GPC (16:1_16:1)","1,2-dipalmitoleoyl-GPC (16:1_16:1)","1,2-dipalmitoyl-GPC (16:0_16:0)","1,2-dipalmitoyl-GPC (16:0_16:0)","10-heptadecenoate (17:1n7)","10-heptadecenoate (17:1n7)","10-nonadecenoate (19:1n9)","10-nonadecenoate (19:1n9)","12,13-DiHOME","12,13-DiHOME","13-HODE + 9-HODE","13-HODE + 9-HODE","13-methylmyristate","13-methylmyristate","15-methylpalmitate","15-methylpalmitate","16-hydroxypalmitate","16-hydroxypalmitate","17-methylstearate","17-methylstearate","2-aminooctanoate","2-aminooctanoate","2-hydroxy-3-methylvalerate","2-hydroxy-3-methylvalerate","2-hydroxybutyrate_2-hydroxyisobutyrate","2-hydroxybutyrate_2-hydroxyisobutyrate","2-hydroxyglutarate","2-hydroxyglutarate","2-isopropylmalate","2-isopropylmalate","2-keto-3-deoxy-gluconate","2-keto-3-deoxy-gluconate","2-ketogluconate","2-ketogluconate","2-linoleoylglycerol (18:2)","2-linoleoylglycerol (18:2)","2-methylcitrate","2-methylcitrate","2-methylserine","2-methylserine","2-palmitoleoyl-GPC (16:1)","2-palmitoleoyl-GPC (16:1)","2-palmitoleoylglycerol (16:1)","2-palmitoleoylglycerol (16:1)","2-palmitoyl-GPC (16:0)","2-palmitoyl-GPC (16:0)","2-phosphoglycerate","2-phosphoglycerate","2-stearoyl-GPE (18:0)","2-stearoyl-GPE (18:0)","2,3-dihydroxyisovalerate","2,3-dihydroxyisovalerate","2'-deoxycytidine","2'-deoxycytidine","2'-deoxyguanosine","2'-deoxyguanosine","2'-deoxyinosine","2'-deoxyinosine","2'-deoxyuridine","2'-deoxyuridine","2'-O-methylguanosine","2'-O-methylguanosine","3-(4-hydroxyphenyl)lactate","3-(4-hydroxyphenyl)lactate","3-aminoisobutyrate","3-aminoisobutyrate","3-hydroxy-3-methylglutarate","3-hydroxy-3-methylglutarate","3-hydroxybutyrate (BHBA)","3-hydroxybutyrate (BHBA)","3-hydroxybutyrylcarnitine (2)","3-hydroxybutyrylcarnitine (2)","3-hydroxydecanoate","3-hydroxydecanoate","3-hydroxyisobutyrate","3-hydroxyisobutyrate","3-hydroxykynurenine","3-hydroxykynurenine","3-hydroxylaurate","3-hydroxylaurate","3-hydroxymyristate","3-hydroxymyristate","3-hydroxyoctanoate","3-hydroxyoctanoate","3-hydroxypalmitate","3-hydroxypalmitate","3-hydroxypropanoate","3-hydroxypropanoate","3-hydroxysebacate","3-hydroxysebacate","3-hydroxysuberate","3-hydroxysuberate","3-ketosphinganine","3-ketosphinganine","3-methyl-2-oxobutyrate","3-methyl-2-oxobutyrate","3-methyl-2-oxovalerate","3-methyl-2-oxovalerate","3-methylcytidine","3-methylcytidine","3-methylhistidine","3-methylhistidine","3-phosphoglycerate","3-phosphoglycerate","3-ureidopropionate","3-ureidopropionate","3,4-dihydroxyphenylacetate","3,4-dihydroxyphenylacetate","4-guanidinobutanoate","4-guanidinobutanoate","4-hydroxybenzoate","4-hydroxybenzoate","4-hydroxybutyrate (GHB)","4-hydroxybutyrate (GHB)","4-hydroxyhippurate","4-hydroxyhippurate","4-hydroxyphenylpyruvate","4-hydroxyphenylpyruvate","4-imidazoleacetate","4-imidazoleacetate","4-methyl-2-oxopentanoate","4-methyl-2-oxopentanoate","5-aminolevulinate","5-aminolevulinate","5-dodecenoate (12:1n7)","5-dodecenoate (12:1n7)","5-methylthioadenosine (MTA)","5-methylthioadenosine (MTA)","5-methyluridine (ribothymidine)","5-methyluridine (ribothymidine)","5-oxoproline","5-oxoproline","6-phosphogluconate","6-phosphogluconate","7-methylguanine","7-methylguanine","7-methylguanosine","7-methylguanosine","9,10-DiHOME","9,10-DiHOME","Acetylcarnitine","Acetylcarnitine","Acetylphosphate","Acetylphosphate","Aconitate [cis or trans]","Aconitate [cis or trans]","Adenine","Adenine","Adenosine","Adenosine","Adenosine 3'-monophosphate (3'-AMP)","Adenosine 3'-monophosphate (3'-AMP)","Adenosine 5'-monophosphate (AMP)","Adenosine 5'-monophosphate (AMP)","Adenosine-2',3'-cyclic monophosphate","Adenosine-2',3'-cyclic monophosphate","Adipate","Adipate","Alanine","Alanine","Alanylleucine","Alanylleucine","Allantoin","Allantoin","Alpha-hydroxyisocaproate","Alpha-hydroxyisocaproate","Alpha-hydroxyisovalerate","Alpha-hydroxyisovalerate","Alpha-ketoglutarate","Alpha-ketoglutarate","Alpha-tocopherol","Alpha-tocopherol","Alpha-tocotrienol","Alpha-tocotrienol","Arabitol_xylitol","Arabitol_xylitol","Arabonate_xylonate","Arabonate_xylonate","Arachidate (20:0)","Arachidate (20:0)","Arginine","Arginine","Argininosuccinate","Argininosuccinate","Asparagine","Asparagine","Aspartate","Aspartate","Azelate (nonanedioate)","Azelate (nonanedioate)","Benzoate","Benzoate","Beta-alanine","Beta-alanine","Beta-hydroxyisovalerate","Beta-hydroxyisovalerate","Beta-sitosterol","Beta-sitosterol","Betaine","Betaine","Biliverdin","Biliverdin","Biopterin","Biopterin","Biotin","Biotin","Campesterol","Campesterol","Caprate (10:0)","Caprate (10:0)","Carboxyethyl-GABA","Carboxyethyl-GABA","Carnitine","Carnitine","Choline","Choline","Choline phosphate","Choline phosphate","Citraconate_glutaconate","Citraconate_glutaconate","Citrate","Citrate","Citrulline","Citrulline","Cystathionine","Cystathionine","Cysteine","Cysteine","Cysteinylglycine","Cysteinylglycine","Cystine","Cystine","Cytidine","Cytidine","Cytidine 2',3'-cyclic monophosphate","Cytidine 2',3'-cyclic monophosphate","Cytidine 3'-monophosphate (3'-CMP)","Cytidine 3'-monophosphate (3'-CMP)","Cytidine 5'-diphosphocholine","Cytidine 5'-diphosphocholine","Cytidine 5'-monophosphate (5'-CMP)","Cytidine 5'-monophosphate (5'-CMP)","Cytidine-5'-diphosphoethanolamine","Cytidine-5'-diphosphoethanolamine","Cytosine","Cytosine","Dehydroascorbate","Dehydroascorbate","Deoxycarnitine","Deoxycarnitine","Dihomo-linoleate (20:2n6)","Dihomo-linoleate (20:2n6)","Dihydoxyphenylalanine (L-DOPA)","Dihydoxyphenylalanine (L-DOPA)","Dihydrobiopterin","Dihydrobiopterin","Dihydroxyacetone phosphate (DHAP)","Dihydroxyacetone phosphate (DHAP)","Dimethylarginine (SDMA + ADMA)","Dimethylarginine (SDMA + ADMA)","Docosadienoate (22:2n6)","Docosadienoate (22:2n6)","Dodecanedioate","Dodecanedioate","Eicosenoate (20:1)","Eicosenoate (20:1)","Ergosterol","Ergosterol","Ergothioneine","Ergothioneine","Erucate (22:1n9)","Erucate (22:1n9)","Erythritol","Erythritol","Erythronate","Erythronate","Ethylmalonate","Ethylmalonate","Flavin mononucleotide (FMN)","Flavin mononucleotide (FMN)","Fructose","Fructose","Fumarate","Fumarate","Galactonate","Galactonate","Galactose 1-phosphate","Galactose 1-phosphate","Gamma-aminobutyrate (GABA)","Gamma-aminobutyrate (GABA)","Gamma-glutamylalanine","Gamma-glutamylalanine","Gamma-glutamylglutamate","Gamma-glutamylglutamate","Gamma-glutamylglutamine","Gamma-glutamylglutamine","Gamma-glutamylhistidine","Gamma-glutamylhistidine","Gamma-glutamylisoleucine","Gamma-glutamylisoleucine","Gamma-glutamylleucine","Gamma-glutamylleucine","Gamma-glutamylmethionine","Gamma-glutamylmethionine","Gamma-tocopherol_beta-tocopherol","Gamma-tocopherol_beta-tocopherol","Gluconate","Gluconate","Glucosamine-6-phosphate","Glucosamine-6-phosphate","Glucose","Glucose","Glucose 1-phosphate","Glucose 1-phosphate","Glucose 6-phosphate","Glucose 6-phosphate","Glucuronate","Glucuronate","Glutamate","Glutamate","Glutamate, gamma-methyl ester","Glutamate, gamma-methyl ester","Glutamine","Glutamine","Glutaminylleucine","Glutaminylleucine","Glutarate (pentanedioate)","Glutarate (pentanedioate)","Glycerate","Glycerate","Glycerol","Glycerol","Glycerol 3-phosphate","Glycerol 3-phosphate","Glycerophosphoethanolamine","Glycerophosphoethanolamine","Glycerophosphoglycerol","Glycerophosphoglycerol","Glycerophosphoinositol","Glycerophosphoinositol","Glycerophosphorylcholine (GPC)","Glycerophosphorylcholine (GPC)","Glycine","Glycine","Glycosyl-N-palmitoyl-sphingosine","Glycosyl-N-palmitoyl-sphingosine","Glycosyl-N-stearoyl-sphingosine","Glycosyl-N-stearoyl-sphingosine","Glycylisoleucine","Glycylisoleucine","Glycylleucine","Glycylleucine","Glycylvaline","Glycylvaline","Guanine","Guanine","Guanosine","Guanosine","Guanosine 3'-monophosphate (3'-GMP)","Guanosine 3'-monophosphate (3'-GMP)","Guanosine-2',3'-cyclic monophosphate","Guanosine-2',3'-cyclic monophosphate","Gulonic acid","Gulonic acid","Heme","Heme","Hexanoylglycine","Hexanoylglycine","Hippurate","Hippurate","Histamine","Histamine","Histidine","Histidine","Homoserine lactone","Homoserine lactone","Hypoxanthine","Hypoxanthine","Inosine","Inosine","Inositol 1-phosphate (I1P)","Inositol 1-phosphate (I1P)","Isobar: fructose 1,6-diphosphate, glucose 1,6-diphosphate, myo-inositol 1,4 or 1,3-diphosphate","Isobar: fructose 1,6-diphosphate, glucose 1,6-diphosphate, myo-inositol 1,4 or 1,3-diphosphate","Isobutyrylcarnitine","Isobutyrylcarnitine","Isoleucine","Isoleucine","Isoleucylglycine","Isoleucylglycine","Isomaltose","Isomaltose","Isoxanthopterin","Isoxanthopterin","Kojibiose","Kojibiose","Kynurenate","Kynurenate","Kynurenine","Kynurenine","Lactate","Lactate","Lactosyl-N-palmitoyl-sphingosine","Lactosyl-N-palmitoyl-sphingosine","Lanthionine","Lanthionine","Laurate (12:0)","Laurate (12:0)","Leucine","Leucine","Leucylalanine","Leucylalanine","Leucylglutamine","Leucylglutamine","Leucylglycine","Leucylglycine","Linoleate (18:2n6)","Linoleate (18:2n6)","Linolenate [alpha or gamma; (18:3n3 or 6)]","Linolenate [alpha or gamma; (18:3n3 or 6)]","Linoleoylcarnitine","Linoleoylcarnitine","Lysine","Lysine","Lysylleucine","Lysylleucine","Malate","Malate","Malonate","Malonate","Maltopentaose","Maltopentaose","Maltose","Maltose","Maltotetraose","Maltotetraose","Maltotriose","Maltotriose","Mannitol_sorbitol","Mannitol_sorbitol","Mannose","Mannose","Mannose-6-phosphate","Mannose-6-phosphate","Margarate (17:0)","Margarate (17:0)","Mesaconate (methylfumarate)","Mesaconate (methylfumarate)","Methionine","Methionine","Methionine sulfone","Methionine sulfone","Methionine sulfoxide","Methionine sulfoxide","Methyl glucopyranoside (alpha + beta)","Methyl glucopyranoside (alpha + beta)","Methyl-4-hydroxybenzoate","Methyl-4-hydroxybenzoate","Methylmalonate (MMA)","Methylmalonate (MMA)","Methylphosphate","Methylphosphate","Methylsuccinate","Methylsuccinate","Mevalonate","Mevalonate","Mevalonolactone","Mevalonolactone","Myo-inositol","Myo-inositol","Myristate (14:0)","Myristate (14:0)","Myristoleate (14:1n5)","Myristoleate (14:1n5)","Myristoylcarnitine","Myristoylcarnitine","N-acetyl-1-methylhistidine","N-acetyl-1-methylhistidine","N-acetyl-beta-alanine","N-acetyl-beta-alanine","N-acetyl-glucosamine 1-phosphate","N-acetyl-glucosamine 1-phosphate","N-acetylalanine","N-acetylalanine","N-acetylarginine","N-acetylarginine","N-acetylasparagine","N-acetylasparagine","N-acetylaspartate (NAA)","N-acetylaspartate (NAA)","N-acetylcysteine","N-acetylcysteine","N-acetylglucosamine 6-phosphate","N-acetylglucosamine 6-phosphate","N-acetylglucosamine_N-acetylgalactosamine","N-acetylglucosamine_N-acetylgalactosamine","N-acetylglutamate","N-acetylglutamate","N-acetylglutamine","N-acetylglutamine","N-acetylglycine","N-acetylglycine","N-acetylhistamine","N-acetylhistamine","N-acetylhistidine","N-acetylhistidine","N-acetylleucine","N-acetylleucine","N-acetylmethionine","N-acetylmethionine","N-acetylmethionine sulfoxide","N-acetylmethionine sulfoxide","N-acetylphenylalanine","N-acetylphenylalanine","N-acetylputrescine","N-acetylputrescine","N-acetylserine","N-acetylserine","N-acetyltaurine","N-acetyltaurine","N-acetylthreonine","N-acetylthreonine","N-acetylvaline","N-acetylvaline","N-alpha-acetylornithine","N-alpha-acetylornithine","N-delta-acetylornithine","N-delta-acetylornithine","N-methylalanine","N-methylalanine","N-monomethylarginine","N-monomethylarginine","N-palmitoyl-sphinganine (d18:0_16:0)","N-palmitoyl-sphinganine (d18:0_16:0)","N-palmitoyl-sphingosine (d18:1_16:0)","N-palmitoyl-sphingosine (d18:1_16:0)","N1-methyladenosine","N1-methyladenosine","N1-methylinosine","N1-methylinosine","N2,N2-dimethylguanine","N2,N2-dimethylguanine","N2,N2-dimethylguanosine","N2,N2-dimethylguanosine","N6-acetyllysine","N6-acetyllysine","N6-carboxymethyllysine","N6-carboxymethyllysine","N6-succinyladenosine","N6-succinyladenosine","N6,N6-dimethyladenosine","N6,N6-dimethyladenosine","N6,N6,N6-trimethyllysine","N6,N6,N6-trimethyllysine","Nicotinamide","Nicotinamide","Nicotinamide ribonucleotide (NMN)","Nicotinamide ribonucleotide (NMN)","Nicotinamide riboside","Nicotinamide riboside","Nicotinate","Nicotinate","Nicotinate ribonucleoside","Nicotinate ribonucleoside","Nonadecanoate (19:0)","Nonadecanoate (19:0)","O-sulfo-L-tyrosine","O-sulfo-L-tyrosine","Oleate_vaccenate (18:1)","Oleate_vaccenate (18:1)","Oleoylcarnitine","Oleoylcarnitine","Oleoylcholine","Oleoylcholine","Ornithine","Ornithine","Orotate","Orotate","Oxalate (ethanedioate)","Oxalate (ethanedioate)","Oxaloacetate","Oxaloacetate","Palmitate (16:0)","Palmitate (16:0)","Palmitoleate (16:1n7)","Palmitoleate (16:1n7)","Palmitoleoylcarnitine","Palmitoleoylcarnitine","Palmitoloelycholine","Palmitoloelycholine","Palmitoylcarnitine","Palmitoylcarnitine","Palmitoylcholine","Palmitoylcholine","Panose","Panose","Pantethine","Pantethine","Pantothenate","Pantothenate","Pentadecanoate (15:0)","Pentadecanoate (15:0)","Phenylalanine","Phenylalanine","Phenylalanylalanine","Phenylalanylalanine","Phenylalanylglycine","Phenylalanylglycine","Phenyllactate (PLA)","Phenyllactate (PLA)","Phosphate","Phosphate","Phosphoenolpyruvate (PEP)","Phosphoenolpyruvate (PEP)","Phosphoethanolamine","Phosphoethanolamine","Phosphopantetheine","Phosphopantetheine","Phosphothreonine","Phosphothreonine","Phytosphingosine","Phytosphingosine","Pipecolate","Pipecolate","Proline","Proline","Prolylglycine","Prolylglycine","Propionylcarnitine","Propionylcarnitine","Propionylglycine","Propionylglycine","Pseudouridine","Pseudouridine","Pterin","Pterin","Putrescine","Putrescine","Pyridoxal","Pyridoxal","Pyridoxamine","Pyridoxamine","Pyridoxate","Pyridoxate","Pyroglutamine","Pyroglutamine","Pyruvate","Pyruvate","Quinate","Quinate","Ribitol","Ribitol","Riboflavin (Vitamin B2)","Riboflavin (Vitamin B2)","Ribose","Ribose","Ribose 1-phosphate","Ribose 1-phosphate","Ribose 5-phosphate","Ribose 5-phosphate","Ribulose_xylulose","Ribulose_xylulose","S-1-pyrroline-5-carboxylate","S-1-pyrroline-5-carboxylate","S-adenosylhomocysteine (SAH)","S-adenosylhomocysteine (SAH)","S-adenosylmethionine (SAM)","S-adenosylmethionine (SAM)","Saccharopine","Saccharopine","Sarcosine","Sarcosine","Sedoheptulose","Sedoheptulose","Sedoheptulose-7-phosphate","Sedoheptulose-7-phosphate","Sepiapterin","Sepiapterin","Serine","Serine","Spermidine","Spermidine","Sphinganine","Sphinganine","Stachydrine","Stachydrine","Stearate (18:0)","Stearate (18:0)","Stearidonate (18:4n3)","Stearidonate (18:4n3)","Stearoylcarnitine","Stearoylcarnitine","Succinate","Succinate","Succinimide","Succinimide","Sucrose","Sucrose","Sulfate","Sulfate","Taurine","Taurine","Tetradecanedioate","Tetradecanedioate","Thiamin (Vitamin B1)","Thiamin (Vitamin B1)","Thiamin monophosphate","Thiamin monophosphate","Thioproline","Thioproline","Threonate","Threonate","Threonine","Threonine","Threonylphenylalanine","Threonylphenylalanine","Thymidine","Thymidine","Thymine","Thymine","Trans-4-hydroxyproline","Trans-4-hydroxyproline","Tricarballylate","Tricarballylate","Trigonelline (N'-methylnicotinate)","Trigonelline (N'-methylnicotinate)","Tryptophan","Tryptophan","Tyrosine","Tyrosine","Tyrosylglycine","Tyrosylglycine","UDP-N-acetylglucosamine","UDP-N-acetylglucosamine","Undecanedioate","Undecanedioate","Uracil","Uracil","Urate","Urate","Uridine","Uridine","Uridine 3'-monophosphate (3'-UMP)","Uridine 3'-monophosphate (3'-UMP)","Uridine-2',3'-cyclic monophosphate","Uridine-2',3'-cyclic monophosphate","Valine","Valine","Valylglutamine","Valylglutamine","Valylglycine","Valylglycine","Valylleucinamide","Valylleucinamide","Valylleucine","Valylleucine","Xanthine","Xanthine","Xanthopterin","Xanthopterin","Xanthosine","Xanthosine","Xanthurenate","Xanthurenate","1-(1-enyl-palmitoyl)-2-linoleoyl-GPC (P-16:0_18:2)","1-(1-enyl-palmitoyl)-2-palmitoleoyl-GPC (P-16:0_16:1)","1-stearoyl-GPI (18:0)","Glycerol 2-phosphate","Mevalonate 5-phosphate","Fertility 25C","Fertility 27C","Fertility 29C","Fertility thermal susceptibility","Aggression ","Climbing","Climbing exercise effect","Climbing post exercise","Climbing","Climbing exercise effect","Climbing post exercise"],["Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Brown et al (2013) Genes, Brain and Behaviour","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Shorter et al (2015) PNAS","Campwell et al (2019) G3","Campwell et al (2019) G3","Watanabe et al (2020) G3","Watanabe et al (2020) G3","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Dembeck et al (2015) eLife","Mackay et al (2012) Nature","Mackay et al (2012) Nature","Morgante et al (2015) Scientific Reports","Morgante et al (2015) Scientific Reports","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Harbison et al (2018) Behavior Genetics","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Wilson et al (2020) Current Biology","Wilson et al (2020) Current Biology","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Lecheta et al (2020) Frontiers in Genetics","Lecheta et al (2020) Frontiers in Genetics","Lecheta et al (2020) Frontiers in Genetics","Lecheta et al (2020) Frontiers in Genetics","Orsted et al (2018) Evolution","Orsted et al (2018) Evolution","Orsted et al (2018) Evolution","Orsted et al (2018) Evolution","Orsted et al (2018) Evolution","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Schmidt et al (2017) Genetics","Schmidt et al (2017) Genetics","Rajpurohit et al (2018) Molecular Ecology","Rajpurohit et al (2018) Molecular Ecology","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Magwire et al (2012) PLOS Genetics","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Morozova el al. (2015) BMC Genomics","Watanabe et al (2020) G3","Watanabe et al (2020) G3","Evangelou et al (2019) G3","Horvath and Kalinka (2018) Evolution","Spierer et al (2021) PLOS Genetics","Spierer et al (2021) PLOS Genetics","Magwire et al (2012) PLOS Genetics","Garlapow et al (2015) PLOS One","Garlapow et al (2015) PLOS One","Lee et al (2017) Genome Biology and Evolution","Unckless, Rottschaefer and Lazzaro (2015) G3","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Nelson et al (2016) BMC Genomics","Nelson et al (2016) BMC Genomics","Unckless, Rottschaefer and Lazzaro (2015) G3","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) G3","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Chohan (2012) PhD thesis","Chohan (2012) PhD thesis","Sharma et al (2020) Scientific reports","Sharma et al (2020) Scientific reports","Harrison et al (2020) BMC Genomics","Vonesch et al (2016) PLOS Genetics","Vonesch et al (2016) PLOS Genetics","Rohde et al (2017) Genetics","Ahlers et al (2019) Cell Reports","Hotson and Schneider (2015) G3","Hotson and Schneider (2015) G3","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Houston (2019) Thesis","Durham et al (2014) Nature Communications","Ivanov et al (2015) The Journals of Gerontology: Series A","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Wilson et al (2020) Current Biology","Wilson et al (2020) Current Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Houston (2019) Thesis","Patel and Talbert (2021) Heliyon","Durham et al (2014) Nature Communications","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Rohde et al (2018) G3","Maesani et al (2015) PLOS Computational Biology","Wang, Lu and St. Leger (2017) PLOS Pathogens","Wang, Lu and St. Leger (2017) PLOS Pathogens","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Nelson et al (2016) BMC Genomics","Nelson et al (2016) BMC Genomics","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Huang et al (2020) PLOS Biology","Weber et al (2012) PLOS One","Weber et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Swarup et al (2013) PNAS","Swarup et al (2013) PNAS","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Chohan (2012) PhD thesis","Chohan (2012) PhD thesis","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Arya et al (2015) Chemical Senses","Lobell et al (2017) G3","Lobell et al (2017) G3","Evangelou et al (2019) G3","Evangelou et al (2019) G3","Evangelou et al (2019) G3","Wang, Lu and St. Leger (2017) PLOS Pathogens","Wang, Lu and St. Leger (2017) PLOS Pathogens","Evangelou et al (2019) G3","Bou Sleiman et al (2015) Nature Communications","Howick and Lazzaro (2017) Molecular Ecology","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Howick and Lazzaro (2017) Molecular Ecology","Howick and Lazzaro (2017) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Weber et al (2012) PLOS One","Weber et al (2012) PLOS One","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Battlay et al (2018) G3","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Takahara and Takahashi (2015) PLOS One","Takahara and Takahashi (2015) PLOS One","Takahara and Takahashi (2015) PLOS One","Unckless, Rottschaefer and Lazzaro (2015) G3","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Vaisnav et al (2014) PLOS One","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Gerken et al (2015) PNAS","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Magwire et al (2011) PLOS Genetics","Rohde et al (2017) Genetics","Rohde et al (2017) Genetics","Turner, Miller and Cochrane (2013) Molecular Biology and Evolution","Riddle (2020) Journal of Experimental Biology","Riddle (2020) Journal of Experimental Biology","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Mackay et al (2012) Nature","Mackay et al (2012) Nature","Morgante et al (2015) Scientific Reports","Morgante et al (2015) Scientific Reports","Jordan et al (2012) PLOS One","Jordan et al (2012) PLOS One","Mackay et al (2012) Nature","Mackay et al (2012) Nature","Lee et al (2017) Genome Biology and Evolution","Francis et al (2021) G3","Francis et al (2021) G3","Francis et al (2021) G3","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Francis et al (2021) G3","Morgante et al (2015) Scientific Reports","Morgante et al (2015) Scientific Reports","Nelson et al (2016) BMC Genomics","Nelson et al (2016) BMC Genomics","Everman and Morgan (2017) Evolution","Everman and Morgan (2017) Evolution","Dembeck et al (2015) PLOS Genetics","Dembeck et al (2015) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Lafuente, Duneau and Beldade (2018) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) G3","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Nelson et al (2016) BMC Genomics","Nelson et al (2016) BMC Genomics","Chow, Wolfner and Clark (2013) PNAS","Chow, Wolfner and Clark (2013) PNAS","Akhund-Zade et al (2017) Journal of Insect Science","Harbison, McCoy and Mackay (2013) BMC Genomics","Harbison, McCoy and Mackay (2013) BMC Genomics","Wu et al (2018) SLEEPJ","Wu et al (2018) SLEEPJ","Durham et al (2014) Nature Communications","Durham et al (2014) Nature Communications","Durham et al (2014) Nature Communications","Durham et al (2014) Nature Communications","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) PLOS Genetics","Unckless, Rottschaefer and Lazzaro (2015) G3","Vonesch et al (2016) PLOS Genetics","Vonesch et al (2016) PLOS Genetics","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Grubbs et al (2013) PLOS One","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Gaertner et al (2015) G3","Gaertner et al (2015) G3","Gaertner et al (2015) G3","Gaertner et al (2015) G3","Gaertner et al (2015) G3","Gaertner et al (2015) G3","Najarro et al (2015) PLOS Genetics","Freda et al (2017) Integrative and Comparative Biology","Freda et al (2017) Integrative and Comparative Biology","Najarro, Hackett and Macdonald (2017) G3","Duneau et al (2018) G3","Duneau et al (2018) G3","Palmer et al (2018) PLOS Pathogens","Palmer et al (2018) PLOS Pathogens","Palmer et al (2018) PLOS Pathogens","Everman et al (2019) Genetics","Everman et al (2019) Genetics","Pitchers et al (2019) Evolution","Pitchers et al (2019) Evolution","Pitchers et al (2019) Evolution","Pitchers et al (2019) Evolution","Rohde et al (2019) Genetics","Rohde et al (2019) Genetics","Rohde et al (2019) Genetics","Chapman et al (2020) Genes","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Early et al (2017) PLOS One","Bailey and Hoskins (2014) Evolution","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Bailey and Hoskins (2014) Evolution","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Mitchell et al (2017) PLOS One","Mitchell et al (2017) PLOS One","Mitchell et al (2017) PLOS One","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Toshima et al (2014) Journal of Insect Physiology","Toshima et al (2014) Journal of Insect Physiology","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Campwell et al (2019) G3","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Battlay et al (2016) G3","Battlay et al (2016) G3","Battlay et al (2016) G3","Battlay et al (2016) G3","Battlay et al (2016) G3","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Green et al (2019) PNAS","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jehrke et al (2018) Scientific Reports","Jehrke et al (2018) Scientific Reports","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Duun Rohde et al (2016) International Journal of Genomics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Freda et al (2017) Integrative and Comparative Biology","Rolandi et al (2018) Ecology and Evolution","Rolandi et al (2018) Ecology and Evolution","Teets and Hahn (2018) Journal of Evolutionary Biology","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Lovejoy et al (2021) Molecular Ecology","Horvath, Betancourt and Kalinka (2016) BMC Evolutionary Biology","Early et al (2017) PLOS One","Gerken, Eller and Morgan (2014) PhD thesis","Gerken, Eller and Morgan (2014) PhD thesis","Ellis et al (2014) PLOS Genetics","Ellis et al (2014) PLOS Genetics","Ellis et al (2014) PLOS Genetics","Morozova (2018) G3","Morozova (2018) G3","Ellis et al (2014) PLOS Genetics","Zhou et al (2016) Environmental Health Perspectives","Duun Rohde et al (2016) International Journal of Genomics","Morozova (2018) G3","Ellis et al (2014) PLOS Genetics","Duun Rohde et al (2016) International Journal of Genomics","Ellis et al (2014) PLOS Genetics","Morozova (2018) G3","Horvath, Betancourt and Kalinka (2016) BMC Evolutionary Biology","Ellis et al (2014) PLOS Genetics","Ellis et al (2014) PLOS Genetics","Ellis et al (2014) PLOS Genetics","Gerken, Eller and Morgan (2014) PhD thesis","Gerken, Eller and Morgan (2014) PhD thesis","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Morozova (2018) G3","Morozova (2018) G3","Morozova (2018) G3","Morozova (2018) G3","Remnant et al (2014) Insect Biochemistry and Molecular Biology","Wong and Holman (in prep)","Wong and Holman (in prep)","Wong and Holman (in prep)","Wong and Holman (in prep)","Bailey and Hoskins (2014) Evolution","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Huang et al (2014) Genome Research &amp; Ellis et al (2014) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jehrke et al (2018) Scientific Reports","Jehrke et al (2018) Scientific Reports","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Uchizono and Tamimura (2017) Chemical Senses","Uchizono and Tamimura (2017) Chemical Senses","Jin et al (2020) PLOS Genetics","Jehrke et al (2018) Scientific Reports","Jin et al (2020) PLOS Genetics","Uchizono and Tamimura (2017) Chemical Senses","Uchizono and Tamimura (2017) Chemical Senses","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Uchizono and Tamimura (2017) Chemical Senses","Uchizono and Tamimura (2017) Chemical Senses","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jehrke et al (2018) Scientific Reports","Jehrke et al (2018) Scientific Reports","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Jehrke et al (2018) Scientific Reports","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Toshima et al (2014) Journal of Insect Physiology","Toshima et al (2014) Journal of Insect Physiology","Jehrke et al (2018) Scientific Reports","Jehrke et al (2018) Scientific Reports","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jehrke et al (2018) Scientific Reports","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Hope et al (2019) Frontiers in Genetics","Hope et al (2019) Frontiers in Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Duun Rohde et al (2016) International Journal of Genomics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Duun Rohde et al (2016) International Journal of Genomics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Denecke et al (2017) Scientific Reports","Denecke et al (2017) Scientific Reports","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Early et al (2017) PLOS One","Early et al (2017) PLOS One","Jehrke et al (2018) Scientific Reports","Jehrke et al (2018) Scientific Reports","Jin et al (2020) PLOS Genetics","Jehrke et al (2018) Scientific Reports","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Ma et al (2019) iScience","Ma et al (2019) iScience","Zhou et al (2016) Environmental Health Perspectives","Zhou et al (2016) Environmental Health Perspectives","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Bailey and Hoskins (2014) Evolution","Gerken, Eller and Morgan (2014) PhD thesis","Gerken, Eller and Morgan (2014) PhD thesis","Everman and Morgan (2017) Evolution","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Arbuthnott, Levin and Promislow (2016) Journal of Evolutionary Biology","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Watanabe and Riddle (2021) Scientific Reports","Watanabe and Riddle (2021) Scientific Reports","Watanabe and Riddle (2021) Scientific Reports","Watanabe and Riddle (2021) Scientific Reports","Hope et al (2019) Frontiers in Genetics","Hope et al (2019) Frontiers in Genetics","Montgomery et al (2014) PLOS One","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Highfill et al (2019) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Ellis et al (2014) PLOS Genetics","Ellis et al (2014) PLOS Genetics","Ellis et al (2014) PLOS Genetics","Richardson (2012) PLOS Genetics","Zwarts et al (2015) Nature Communications","Zwarts et al (2015) Nature Communications","Zwarts et al (2015) Nature Communications","Zwarts et al (2015) Nature Communications","Zwarts et al (2015) Nature Communications","Zwarts et al (2015) Nature Communications","Zwarts et al (2015) Nature Communications","Zwarts et al (2015) Nature Communications","Zwarts et al (2015) Nature Communications","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Gerken, Eller and Morgan (2014) PhD thesis","Gerken, Eller and Morgan (2014) PhD thesis","Bailey and Hoskins (2014) Evolution","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Adebambo et al (2020) Frontiers in Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Toshima et al (2014) Journal of Insect Physiology","Toshima et al (2014) Journal of Insect Physiology","Carbone et al (2017) PNAS","Carbone et al (2017) PNAS","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Gerken, Eller and Morgan (2014) PhD thesis","Gerken, Eller and Morgan (2014) PhD thesis","Chow, Wolfner and Clark (2013) Genetics","Chow, Wolfner and Clark (2013) Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jehrke et al (2018) Scientific Reports","Jehrke et al (2018) Scientific Reports","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Jehrke et al (2018) Scientific Reports","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Zhang et al (2021) Current Biology","Ellis et al (2014) PLOS Genetics","Ellis et al (2014) PLOS Genetics","Ellis et al (2014) PLOS Genetics","Zhang, Reeves and Tautz (2020) Molecular Ecology","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Remnant et al (2014) Insect Biochemistry and Molecular Biology","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Gerken, Eller and Morgan (2014) PhD thesis","Gerken, Eller and Morgan (2014) PhD thesis","Early et al (2017) PLOS One","Chow, Wolfner and Clark (2013) Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Bailey and Hoskins (2014) Evolution","Hope et al (2019) Frontiers in Genetics","Hope et al (2019) Frontiers in Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Morozova (2018) G3","Morozova (2018) G3","Morozova (2018) G3","Morozova (2018) G3","Akcay (2017) Thesis","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Bailey and Hoskins (2014) Evolution","Bailey and Hoskins (2014) Evolution","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Katzenberger et al (2015) eLife","Jehrke et al (2018) Scientific Reports","Jehrke et al (2018) Scientific Reports","Jehrke et al (2018) Scientific Reports","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Horvath and Kalinka (2018) Evolution","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Mendez et al (2016) PLOS One","Mendez et al (2016) PLOS One","Okada et al (2016) Nature Communications &amp; Okada et al (2019) Molecular Systems Biology","Okada et al (2016) Nature Communications &amp; Okada et al (2019) Molecular Systems Biology","Okada et al (2016) Nature Communications &amp; Okada et al (2019) Molecular Systems Biology","Okada et al (2016) Nature Communications &amp; Okada et al (2019) Molecular Systems Biology","Richardson (2012) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jin et al (2020) PLOS Genetics","Jumbo-Lucioni et al (2012) BMC Genomics","Jumbo-Lucioni et al (2012) BMC Genomics","Jumbo-Lucioni et al (2012) BMC Genomics","Jumbo-Lucioni et al (2012) BMC Genomics","Jumbo-Lucioni et al (2012) BMC Genomics","Jumbo-Lucioni et al (2012) BMC Genomics","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Dobson et al (2015) Nature Communications","Appel et al (2016) Biology Letters","Appel et al (2016) Biology Letters","Appel et al (2016) Biology Letters","Appel et al (2016) Biology Letters","Muñoz-Muñoz et al (2016) Evolution","Muñoz-Muñoz et al (2016) Evolution","Dean et al (2020) Evolution","Dean et al (2020) Evolution","Teets and Hahn (2018) Journal of Evolutionary Biology","Teets and Hahn (2018) Journal of Evolutionary Biology","Teets and Hahn (2018) Journal of Evolutionary Biology","Teets and Hahn (2018) Journal of Evolutionary Biology","Teets and Hahn (2018) Journal of Evolutionary Biology","Teets and Hahn (2018) Journal of Evolutionary Biology","Teets and Hahn (2018) Journal of Evolutionary Biology","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Freda et al (2019) Heredity","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Everett et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zhou et al (2020) Genome Research","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Zwoinska et al (2020) Frontiers in Genetics","Chowdhury et al Nature Communications Biology","Watanabe and Riddle (2021) Royal Society Open Science","Watanabe and Riddle (2021) Royal Society Open Science","Watanabe and Riddle (2021) Royal Society Open Science","Watanabe and Riddle (2021) Royal Society Open Science","Watanabe and Riddle (2021) Royal Society Open Science","Watanabe and Riddle (2021) Royal Society Open Science"],[157,157,157,157,157,157,157,157,157,157,194,196,191,181,181,187,187,185,186,181,181,184,184,200,171,171,153,151,169,169,169,157,169,157,169,157,157,157,169,157,169,157,169,157,169,157,169,169,157,169,157,169,157,169,169,169,169,169,157,169,157,157,169,157,169,169,157,157,169,157,169,157,169,157,169,157,169,169,169,169,169,169,169,169,169,169,169,157,157,157,169,169,169,169,157,169,157,169,157,169,157,157,169,157,169,157,169,157,169,157,169,157,169,169,169,169,157,157,157,157,169,169,157,176,176,174,174,175,175,161,166,161,166,167,167,99,99,99,99,155,155,96,96,96,96,96,96,96,96,100,100,100,100,162,166,166,166,166,168,168,167,167,168,168,167,167,168,168,167,167,184,184,175,175,182,183,155,198,198,205,205,205,205,205,205,149,148,142,91,197,197,182,183,182,201,145,147,150,152,181,181,145,147,150,152,145,147,150,152,167,167,160,160,180,149,149,87,86,111,116,184,188,200,200,185,190,166,166,183,183,191,189,197,186,186,161,161,177,177,197,193,189,166,166,204,98,188,188,170,170,170,170,170,170,170,170,181,181,183,183,186,186,177,177,167,167,192,192,192,192,192,192,168,168,167,167,168,168,167,167,168,168,167,167,186,186,186,186,186,186,186,186,164,164,186,186,186,186,186,186,167,167,186,186,186,186,185,185,186,186,186,186,186,186,186,186,200,200,150,150,152,81,81,142,140,158,151,151,152,156,156,99,99,167,167,169,169,169,169,191,191,185,185,191,191,155,155,155,145,147,150,152,161,181,181,184,184,96,96,96,96,189,87,87,190,135,135,192,192,192,192,204,204,201,201,192,192,203,203,201,176,175,178,159,159,178,197,197,171,171,159,159,175,175,194,196,191,145,147,150,152,172,172,111,112,90,168,168,167,167,189,189,178,135,147,150,152,145,149,149,117,117,117,117,117,117,117,117,117,117,117,117,117,117,117,117,94,96,82,103,117,99,87,90,85,166,166,166,166,166,166,164,139,139,163,195,194,119,119,119,168,168,184,185,171,172,172,172,172,199,126,124,122,127,178,178,178,178,178,178,178,178,178,178,178,178,178,178,178,178,178,178,178,178,36,50,178,178,178,178,178,178,50,178,178,178,178,178,178,178,178,178,178,178,178,186,186,36,178,178,36,36,178,178,178,178,169,178,178,178,178,178,178,178,178,178,178,175,175,175,175,175,178,178,178,178,178,178,178,178,178,178,46,46,46,46,46,46,46,46,152,152,151,152,152,152,152,178,178,35,36,178,178,4,4,4,178,178,21,46,46,46,46,46,46,46,46,139,34,34,36,178,178,178,178,178,178,178,178,178,178,178,178,20,20,99,75,75,43,26,39,39,50,50,50,201,201,50,200,27,201,50,26,50,201,43,45,49,45,39,39,178,178,201,38,38,201,11,125,124,125,124,50,4,4,178,178,205,178,178,178,178,36,36,40,40,4,4,76,76,178,35,178,37,37,4,4,37,37,178,178,178,178,178,178,178,178,178,178,36,36,40,40,4,4,35,4,4,178,178,36,36,36,36,40,40,35,178,178,40,40,178,178,27,178,178,178,178,178,178,24,178,178,171,171,178,178,178,178,178,178,178,178,178,178,178,178,34,36,36,36,178,35,178,178,178,53,53,199,199,178,178,50,39,39,10,178,178,178,178,178,178,178,178,178,178,9,40,40,34,33,35,35,41,41,171,46,46,46,46,46,46,46,46,178,178,178,178,45,50,45,174,40,40,40,40,40,40,40,40,40,178,178,178,178,39,39,50,178,178,178,178,178,178,178,178,143,178,178,178,178,178,178,36,36,33,35,178,178,39,39,38,38,178,178,36,36,40,40,4,4,35,4,4,198,50,50,50,198,178,178,178,178,11,178,178,38,40,9,39,178,178,178,178,178,178,50,41,41,178,178,38,38,38,38,129,178,178,178,178,50,50,178,178,39,39,39,39,178,178,178,36,36,35,40,40,4,4,4,4,178,178,178,178,178,178,91,178,178,178,178,178,178,178,178,4,4,32,32,32,32,91,178,178,178,178,40,40,40,40,40,40,52,59,37,69,79,57,38,38,38,38,40,40,33,33,36,36,36,36,36,36,36,40,39,39,39,39,39,39,39,39,39,39,39,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,200,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,40,48,35,34,34,33,33,34],["Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Females","Females","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Males","Females","Males","Females","Males","Females","Females","Females","Males","Females","Males","Females","Males","Males","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Females","Males","Females","Males","Females","Males","Females","Females","Females","Females","Females","Males","Females","Males","Males","Females","Males","Females","Females","Males","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Males","Males","Males","Females","Females","Females","Females","Males","Females","Males","Females","Males","Females","Males","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Females","Females","Females","Males","Males","Males","Males","Females","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Females","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Males","Males","Males","Males","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Females","Females","Males","Females","Males","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Females","Females","Males","Males","Females","Males","Males","Males","Males","Males","Males","Females","Females","Males","Males","Males","Males","Males","Males","Males","Males","Females","Males","Females","Females","Females","Females","Males","Males","Females","Males","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Females","Females","Females","Males","Females","Females","Females","Males","Females","Females","Females","Females","Males","Males","Females","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Females","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Females","Females","Females","Females","Females","Males","Females","Females","Females","Males","Males","Males","Females","Females","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Males","Males","Males","Males","Males","Males","Males","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Males","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Males","Males","Males","Males","Females","Males","Males","Females","Males","Females","Females","Females","Males","Females","Females","Females","Females","Females","Males","Males","Males","Males","Females","Females","Males","Males","Females","Females","Males","Females","Males","Females","Females","Females","Females","Males","Males","Males","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Males","Males","Males","Males","Males","Males","Males","Males","Males","Males","Males","Males","Males","Males","Males","Females","Females","Males","Females","Males","Males","Females","Males","Females","Females","Males","Females","Males","Females","Males","Males","Males","Males","Males","Males","Males","Males","Males","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Males","Males","Females","Females","Females","Females","Females","Females","Males","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Females","Females","Females","Males","Females","Females","Females","Females","Mixed sex cohorts","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Males","Females","Males","Females","Males","Females","Males","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Females","Females","Females","Males","Females","Females","Males","Males","Males","Females","Females","Females","Females","Males","Females","Males","Females","Males","Females","Males","Mixed sex cohorts","Females","Males","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Males","Males","Males","Females","Males","Mixed sex cohorts","Males","Females","Females","Females","Females","Females","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Females","Females","Females","Females","Mixed sex cohorts","Females","Males","Mixed sex cohorts","Mixed sex cohorts","Females","Males","Females","Males","Males","Males","Males","Females","Females","Mixed sex cohorts","Females","Females","Females","Females","Females","Males","Females","Males","Males","Males","Females","Males","Females","Mixed sex cohorts","Females","Females","Males","Males","Males","Females","Males","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Males","Females","Males","Males","Males","Mixed sex cohorts","Males","Males","Females","Females","Females","Males","Females","Males","Females","Males","Mixed sex cohorts","Females","Females","Females","Males","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Mixed sex cohorts","Mixed sex cohorts","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Males","Males","Females","Males","Females","Mixed sex cohorts","Females","Females","Females","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Females","Females","Males","Females","Females","Males","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Females","Males","Females","Males","Females","Males","Females","Males","Mixed sex cohorts","Females","Males","Females","Males","Females","Males","Females","Males","Females","Females","Females","Females","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Mixed sex cohorts","Males","Males","Males","Males","Males","Males","Males","Males","Males","Females","Females","Females","Females","Mixed sex cohorts","Mixed sex cohorts","Males","Females","Females","Females","Females","Females","Females","Females","Females","Mixed sex cohorts","Females","Females","Females","Females","Females","Females","Females","Males","Females","Males","Females","Females","Mixed sex cohorts","Mixed sex cohorts","Females","Females","Females","Females","Females","Males","Females","Males","Males","Males","Mixed sex cohorts","Males","Males","Mixed sex cohorts","Females","Females","Females","Mixed sex cohorts","Females","Females","Females","Females","Mixed sex cohorts","Females","Females","Mixed sex cohorts","Mixed sex cohorts","Males","Females","Females","Females","Females","Females","Females","Females","Males","Females","Males","Females","Females","Females","Males","Females","Males","Mixed sex cohorts","Females","Females","Females","Females","Males","Males","Females","Females","Females","Males","Females","Males","Females","Females","Mixed sex cohorts","Females","Males","Mixed sex cohorts","Females","Males","Males","Males","Males","Males","Females","Females","Females","Females","Females","Females","Mixed sex cohorts","Females","Females","Females","Females","Females","Females","Females","Females","Males","Males","Females","Males","Females","Males","Mixed sex cohorts","Females","Females","Females","Females","Females","Males","Females","Males","Females","Males","Males","Males","Males","Males","Males","Males","Females","Males","Females","Males","Females","Males","Males","Males","Females","Females","Females","Females","Females","Females","Females","Mixed sex cohorts","Females","Males","Mixed sex cohorts","Females","Males","Mixed sex cohorts","Females","Males","Mixed sex cohorts","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Females","Males","Males","Males","Females","Males","Males","Females","Females","Females","Females","Males","Females","Females","Females","Males","Males","Males"],["Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","No"],["Attraction to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate greater attraction to the odorant.","Attraction to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate greater attraction to the odorant.","Median latency (seconds) to achieve 30% of the entire observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate a greater ability to detect and respond to the odorant.","Median latency (seconds) to achieve 30% of the entire observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate a greater ability to detect and respond to the odorant.","Median latency (seconds) to achieve 90% of the entire observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate faster habituation to the odorant.","Median latency (seconds) to achieve 90% of the entire observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Lower values indicate faster habituation to the odorant.","Strongest observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Higher values indicate greater attraction to the odorant.","Strongest observed response to 2,3-buntanedione, an odorant present in fermenting fruit. Higher values indicate greater attraction to the odorant.","Strength of attraction to 2,3-buntanedione, an odorant present in fermenting fruit. Higher values indicate greater attraction to the odorant.","Strength of attraction to 2,3-buntanedione, an odorant present in fermenting fruit. Higher values indicate greater attraction to the odorant.","Mean adult abdomen length, when reared at 18C. Higher values indicate larger flies.","Mean adult abdomen length, when reared at 28C. Higher values indicate larger flies.","Flies were reared at 18C and 28C, after which adult abdomen lengths were measured. Reaction norms for each DGRP line were calculated using the linear model lm (Size ~ Temperature). The absolute value of the slope is a measurement of thermal sensitivity, describing the magnitude of the plastic response to temperature. Higher values indicate greater thermal plasticity.","Mean proportion of adult flies surviving one hours of exposure to -8°C, for flies reared at 25°C. Flies were pretreated for two hours at 4°C, immediately prior to cold exposure, to acclimate them to cold exposure. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -8°C, for flies reared at 25°C. Flies were pretreated for two hours at 4°C, immediately prior to cold exposure, to acclimate them to cold exposure. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 25°C. Flies were pretreated for two hours at 4°C, immediately prior to cold exposure, to acclimate them to cold exposure. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 25°C. Flies were pretreated for two hours at 4°C, immediately prior to cold exposure, to acclimate them to cold exposure. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 18°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 18°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -8°C, for flies reared at 18°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -8°C, for flies reared at 18°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 25°C. Higher values indicate greater resistance to acute cold exposure.","Mean proportion of adult flies surviving one hours of exposure to -6°C, for flies reared at 25°C. Higher values indicate greater resistance to acute cold exposure.","Mean number of aggressive encounters between 8 males in an interaction vial over a two minute duration. Higher values indicate more aggressive males.","Mean survival for adult flies, 24 hours after exposure to 6 hours of anoxia (a vial perfused with 80% nitrogen). Higher values indicate a greater a greater tolerance of anoxia.","Mean survival for adult flies, 24 hours after exposure to 6 hours of anoxia (a vial perfused with 80% nitrogen). Higher values indicate a greater a greater tolerance of anoxia.","Basal activity levels, measured as the mean number of times a fly crossed a beam running down the midline of a vial, within a five minute interval. Higher values indicate higher basal activity levels.","Basal activity levels, measured as the mean number of times a fly crossed a beam running down the midline of a vial, within a five minute interval. Higher values indicate higher basal activity levels.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. NI indicates that the chemical composition could not be identified and the number following allows differentiation between CHCs without chemical profiles. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. How x-C22:1a differs from x-C22:1b is unclear. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. How x-C22:1a differs from x-C22:1b is unclear. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean chromatogram peak, for a particular Cuticular Hydrocarbon (CHC). Data are originally displayed as a proportion from all CHCs detected. The name of each CHC can be deciphered as follows: linear alkanes are referred to by the abbreviation n-Cx, where ‘‘x’’ is the total carbon number. Methyl-branched alkanes are referred to with a y-Me-Cx prefix, where ’y’ is the carbon onto which the methyl group is bound. For example, 9-Me-C23 is a 23 carbon chain with a methyl on the 9th carbon. For monoenes (z-Cx:1) and dienes (z,z-Cx:2) the number of double bonds is indicated after the colon and the double bond position/s are ’z’ or ’z,z\"). For example, 7-C23:1 has one double bond between the 7th and 8th carbons. Higher peaks indicate a greater amount of the CHC.","Mean time required for a fly to stand upon being returned to room temperature, after three hours of ice exposure. Higher values indicate a slower recovery from chill coma.","Mean time required for a fly to stand upon being returned to room temperature, after three hours of ice exposure. Higher values indicate a slower recovery from chill coma.","Mean variance (the log of the within line standard deviation; ln(σE)) in the time required for a fly to stand upon being returned to room temperature, after three hours of ice exposure.. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable chill coma response, and therefore greater micro-environmental plasticity. Note that the data presented in this study was originally collected by Mackay et al (2012).","Mean variance (the log of the within line standard deviation; ln(σE)) in the time required for a fly to stand upon being returned to room temperature, after three hours of ice exposure.. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable chill coma response, and therefore greater micro-environmental plasticity. Note that the data presented in this study was originally collected by Mackay et al (2012).","Mean proportion of adult flies surviving 16 hours of exposure to 0°C, after 24 hours of recovery at 25°C. Higher values indicate greater resistance to chronic cold exposure.","Mean proportion of adult flies surviving 16 hours of exposure to 0°C, after 24 hours of recovery at 25°C. Higher values indicate greater resistance to chronic cold exposure.","Mean length of the circadian cycle. Calculated using the Chi square Periodogram method. Higher values indicate a longer circadian cycle.","Mean length of the circadian cycle. Calculated using the Chi square Periodogram method. Higher values indicate a longer circadian cycle.","Mean length of the circadian cycle. Calculated using the Maximum Entropy Spectral Analysis method (MESA period). Higher values indicate a longer circadian cycle.","Mean length of the circadian cycle. Calculated using the Maximum Entropy Spectral Analysis method (MESA period). Higher values indicate a longer circadian cycle.","A measure of how consistent a flies activity patterns are, which is an indicator of strength or robustness of the circadian rhythm. Higher values indicate greater rythmicity.","A measure of how consistent a flies activity patterns are, which is an indicator of strength or robustness of the circadian rhythm. Higher values indicate greater rythmicity.","Mean centimetres climbed in three seconds following vial inversion. Flies were maintained on 1% sucrose solution prior to the climbing trial. Higher values indicate greater climbing ability.","Mean centimetres climbed in three seconds following vial inversion. Flies were maintained on 1% sucrose solution prior to the climbing trial. Higher values indicate greater climbing ability.","Mean centimetres climbed in three seconds following vial inversion, after 24 hours of exposure to paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater climbing ability.","Mean centimetres climbed in three seconds following vial inversion, after 24 hours of exposure to paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater climbing ability.","Median day at which climbing performance dropped below 50% of their day 7 climbing value. Climbing was measured as the proportion of flies in a vial that climbed above a line drawn 6cm from the bottom of the vial. Mated females were kept as adults on a diet containing 5% yeast. Higher values indicate a slower age related decline in climbing performance. Note these flies come from the same populations as used by Jin et al (2020).","Median day at which climbing performance dropped below 50% of their day 7 climbing value. Climbing was measured as the proportion of flies in a vial that climbed above a line drawn 6cm from the bottom of the vial. Mated females were kept as adults on a diet containing 0.5% yeast (dietary restricted). Higher values indicate a slower age related decline in climbing performance. Note these flies come from the same populations as used by Jin et al (2020).","Mean proportion of surviving 28-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Flies were acclimated at 4C for two hours immediately preceeding cold shock. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 28-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Flies were acclimated at 4C for two hours immediately preceeding cold shock. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 7-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Flies were acclimated at 4C for two hours immediately preceeding cold shock. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 7-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Flies were acclimated at 4C for two hours immediately preceeding cold shock. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 28-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 28-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 7-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Higher values indicate greater resistance to cold shock.","Mean proportion of surviving 7-day old adult flies, 24 hours after a one-hour cold shock treatment at -6C. Higher values indicate greater resistance to cold shock.","The critical thermal maximum, which is defined as the mean temperature in °C at which a fly loses motor function. Higher values indicate a higher CT max.","The critical thermal maximum, which is defined as the mean temperature in °C at which a fly loses motor function. Higher values indicate a higher CT max.","The critical thermal minimum, which is defined as the mean temperature in °C at which a fly loses motor function. Higher values indicate a higher CT min.","The critical thermal minimum, which is defined as the mean temperature in °C at which a fly loses motor function. Higher values indicate a higher CT min.","The critical thermal minimum (CT min), which is defined as the temperature at which a complete cessation of movement is observed as a result of the fly entering chill coma. Egg-to-adult development occurred at 23C. Higher values indicate a higher CT min.","Flies were reared at five different temperatures during development, after which adult critical thermal minimums (CT min) were measured. Developmental stability across environments is the consistency of within-environment variation in CT min across environments. The authors calculate this by taking the highest coefficient of variation (CV) in CT min at any rearing temperature, minus the lowest CV at any other given rearing temperature. Lower values indicate greater developmental stability across rearing temperatures.","Flies were reared at fiev different temperatures during development, after which adult critical thermal minimums were measured (CT min). Reaction norms for each DGRP line were calculated using the linear model lm (CTmin ~ Rearing temperature). The absolute value of the slope provides a measure of the plastic response to temperature. Higher values indicate greater thermal plasticity.","Flies were reared at fiev different temperatures during development, after which adult critical thermal minimums (CT min) were measured. Reaction norms for each DGRP line were calculated using the linear model lm (CTmin ~ Rearing temperature). The standard error of the slope provides a measure of the extent to which the same genotype can produce the same plastic response consistently. Higher values indicate greater variance in thermal plasticity.","Flies were reared at five different temperatures during development, after which adult critical thermal minimums were measured. This trait captures how consistently a genotype produces the same phenotype at a given rearing temperature. Higher values indicate greater variance in phenotype within a rearing temperature.","The mean length of a sleep bout during the 12-hour light period of the light cycle. Higher values indicate longer bouts.","The mean length of a sleep bout during the 12-hour light period of the light cycle. Higher values indicate longer bouts.","The standard deviation for the length of a sleep bout during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep bout length. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the length of a sleep bout during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep bout length. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour light period of the light cycle. Higher values indicate a higher frequency of sleep bouts.","Mean number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour light period of the light cycle. Higher values indicate a higher frequency of sleep bouts.","The standard deviation for the number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in number of sleep bouts. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in number of sleep bouts. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour light period of the light cycle. Higher values indicate a higher line average for sleep duration.","The sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour light period of the light cycle. Higher values indicate a higher line average for sleep duration.","The standard deviation for the sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep duration. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour light period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep duration. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean knockdown of adults, four hours after vials were inoculated with 0.5 ug/ml of DDT. Knockdown is defined as either flies permanently seen to be in a prone position with jolting movement of legs or wings, or a prolonged inability to right themselves from a prone position. DDT is a historically used pesticide that targets the voltage gated sodium channel, resulting in uncontrolled nerve firing. Higher values indicate greater susceptiibility to DDT.","Mean survival of adults, 24 hours after vials were inoculated with 0.5 ug/ml of DDT. This is a historically used pesticide that targets the voltage gated sodium channel, resulting in uncontrolled nerve firing. Higher values indicate greater susceptiibility to DDT.","Mean number of hours survived in a vial with very low relative humidity (~3%). Higher values indicate greater dessication tolerance.","Mean number of hours survived in a vial with very low relative humidity (~3%). Higher values indicate greater dessication tolerance.","Mean change in proportion of adult flies surviving one hours of exposure to -6°C, when rearing temperature was changed from 25°C to 18°C. Higher values indicate greater developmental acclimation capacity to cold exposure.","Mean change in proportion of adult flies surviving one hour of exposure to -6°C, when rearing temperature was changed from 25°C to 18°C. Higher values indicate greater developmental acclimation capacity to cold exposure.","Mean days survived after infection with Drosophila C Virus (DCV) - a positive sense RNA virus that naturally infects multiple Drosophila species. Higher values indicate a longer lifespan, and therefore a greater resistance to DCV.","Mean egg-to-adult development time in days, when reared at 25C on control food medium. Higher values indicate a longer development time.","Mean egg-to-adult development time in days, when reared at 25C on control food medium. Higher values indicate a longer development time.","Mean knockdown time (minutes) after a single exposure to ethanol. Higher values indicate a greater ethanol tolerance.","Mean knockdown time (minutes) after a single exposure to ethanol. Higher values indicate a greater ethanol tolerance.","Mean knockdown time (minutes) after a second exposure to ethanol, following a 2 hr recovery period. Higher values indicate a greater ethanol tolerance.","Mean knockdown time (minutes) after a second exposure to ethanol, following a 2 hr recovery period. Higher values indicate a greater ethanol tolerance.","Mean difference in knockdown time between a second and first exposure to ethanol. Higher values indicate lines with higher tolerances to ethanol.","Mean difference in knockdown time between a second and first exposure to ethanol. Higher values indicate lines with higher tolerances to ethanol.","Exercise induced activity levels, measured as the mean number of times a fly crossed a beam running down the midline of a vial, within a five minute interval. Activity was induced by vertically rotating vials at four rpm. Higher values indicate higher exercise induced activity levels.","Exercise induced activity levels, measured as the mean number of times a fly crossed a beam running down the midline of a vial, within a five minute interval. Activity was induced by vertically rotating vials at four rpm. Higher values indicate higher exercise induced activity levels.","Mean number of eggs produced by single females, per day. Higher values indicate more fecund females.","Mean fertilised egg retention score. An estimation of the progression of embryo development taken two hours after a one hour egg laying period. Higher values indicate more developed embryos which signifies a greater degree of egg retention. Note that both males and femlaes were sourced from the DGRP line, therefore variation could be explained by either the male or female genotype.","Mean height (m) at which flies catch themselves from freefall after vial inversion. Means are adjusted for wolbachia and chromosomal inversions. Higher values indicate lines that caught themselves earlier, which is interpreted as greater flight performance.","Mean height (m) at which flies catch themselves from freefall after rapid inversion. Means are adjusted for wolbachia and chromosomal inversions. Higher values indicate lines that caught themselves earlier, which is interpreted as greater flight performance.","Mean days survived after infection with Flock House Virus (FHV) - a nodavirus virus that typically infects beetles. Higher values indicate a longer lifespan, and therefore a greater resistance to FHV.","Mean quantity of 4% sucrose solution consumed per fly iover a 24 hour period. Higher values indicate greater food intake.","Mean quantity of 4% sucrose solution consumed per fly over a 24 hour period. Higher values indicate greater food intake.","Adult mean survival when flies were provided access to food over a 5 day period. Higher values indicate greater survival and assumed greater foraging ability.","Estimated mass of free glucose per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean free glucose (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of free glucose.","Mean free glucose (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of free glucose.","Mean free glucose (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of free glucose.","Mean quantity of glucose extracted from whole fly homogenate and standardised for body size. Three flies reared on food with 0.5% yeast content (dietary restricted) were homogenised. Higher values indicate higher glucose levels.","Mean quantity of glucose extracted from whole fly homogenate and standardised for body size. Three flies reared on food with 5% yeast content (ab libitum diet) were homogenised. Higher values indicate higher glucose levels.","Estimated mass of free glycerol per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean glycerol (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of glycerol.","Mean glycerol (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of glycerol.","Mean glycerol (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of glycerol.","Estimated mass of glycogen stores per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean glycogen (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of glycogen.","Mean glycogen (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of glycogen.","Mean glycogen (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of glycogen.","Flies were given a choice between sucrose solution and sucrose solution supplemented with ethanol. Higher values indicate a preference for the ethanol supplemented sucrose, while lower values indicate an aversion towards the ethanol supplemented sucrose.","Flies were given a choice between sucrose solution and sucrose solution supplemented with ethanol. Higher values indicate a preference for the ethanol supplemented sucrose, while lower values indicate an aversion towards the ethanol supplemented sucrose.","The mean proportion of 19 day-old flies with permeable guts, following no exposure to radiation. Gut permeability is an indicator of intestinal damage, and in this case was detected using the smurf assay. Higher values indicate greater intestinal damage.","The mean proportion of 19 day-old flies with permeable guts, following exposure to 100 gray units of ionising radiation when flies were 5 days old. Gut permeability is an indicator of intestinal damage, and in this case was detected using the smurf assay. Higher values indicate greater intestinal damage.","Lifespan on a diet supplemented with 2-3% hydrogen peroxide (H2O2). Data are corrected for line body mass. Higher values indicate greater resistance to hydrogen peroxide.","Mean Interocular distance. This is measured from eye edge to eye edge along the anterior edge of the posterior ocelli and palel to the base of the head. Higher values indicate a greater distance between the eyes.","Mean Interocular distance. This is measured from eye edge to eye edge along the anterior edge of the posterior ocelli and palel to the base of the head. Higher values indicate a greater distance between the eyes.","Mean number of aggressive encounters over a two-minute period between pairs of males; a DGRP male and a control Canton-SB male. The DGRP male was housed in isolation for the 3 days preceding the aggression assay. Higher values indicate more aggressive males.","Mean survival after infection with the Kunjin subtype of the West Nile Virus. Calculated as the log(hazard ratio). Higher values indicate greater susceptibility to the virus.","The median number  of colony forming units (originally log transformed) 48 hrs after initial infection with the bacterial pathogen L.monocytogenes. Higher values indicate a greater bacterial load and a more severe infection.","Median days till death after injection with 1000 colony forming units of the bacterial pathogen L. monocytogenes. Higher values indicate greater resistance to the L. monocytogenes.","Mean difference in egg-to-adult development time between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a longer development time when supplemented with lead.","Mean difference in egg-to-adult development time between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a longer development time when supplemented with lead.","Mean difference in locomotor between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a larger negative effect of lead on activity.","Mean difference in locomotor between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a larger negative effect of lead on activity.","Mean egg-to-adult development time in days, when reared at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a longer development time.","Mean egg-to-adult development time in days, when reared at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a longer development time.","Mean number of times adult flies crossed an infrared beam over a 2-day period, after developing at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate greater locomotor activity.","Mean number of times adult flies crossed an infrared beam over a 2-day period, after developing at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate greater locomotor activity.","Mean adult lifespan when reared at 18C, measured in days. Higher values indicate longer-lived flies.","Mean adult lifespan when reared at 18C, measured in days. Higher values indicate longer-lived flies.","Mean lifespan in days. Higher values indicate a longer lifespan.","Mean adult lifespan, measured as the number of days from eclosion to death, when living with a single  standard male, at 25C. Higher values indicate longer lived flies.","Adult lifespan, measured in days, for virgin females. Higher values indicate greater longevity.","Mean adult lifespan when reared at 25C, measured in days. Higher values indicate longer-lived flies.","Mean adult lifespan when reared at 25C, measured in days. Higher values indicate longer-lived flies.","Median mated female lifespan when reared at 25C and then kept as adults on a diet containing 5% yeast. Higher values indicate longer lived flies. Note these flies come from the same populations as used by Jin et al (2020).","Median mated female lifespan when reared at 25C and then kept as adults on a diet containing 0.5% yeast (dietary restricted). Higher values indicate longer lived flies. Note these flies come from the same populations as used by Jin et al (2020).","Mean adult lifespan when reared at 28C, measured in days. Higher values indicate longer-lived flies.","Mean adult lifespan when reared at 28C, measured in days. Higher values indicate longer-lived flies.","Mean lifespan in days for virgin females following diapause induction (10 ̊C for 35 days, moved to 18 ̊C for a day of acclimation, remainder of their lives at 25 ̊C). Higher values indicate a longer lifespan.","Mean lifespan of 20 females maintained on a high sugar diet. Means are adjusted for block effects. Higher values indicate longer lived flies.","Fecundity was estimated every other week of the female’s lifespan as a 2-day total of eggs laid, when living with a single standard male, at 25C. Lifetime fecundity was estimated as the sum of all egg counts across the female's lifespan. Higher values indicate greater fecundity. Note that female's were eliminated form the analysis upon death.","Mean number of times adult flies crossed an infrared beam over a 2-day period, after developing at 25C on control food medium. Higher values indicate greater locomotor activity.","Mean number of times adult flies crossed an infrared beam over a 2-day period, after developing at 25C on control food medium. Higher values indicate greater locomotor activity.","Mean total distance traveled by individual flies during a five minute interval. Higher values indicate more active lines.","The proportion of flies engaging in a ‘walking bout’ at a given time point. Higher values indicate more active lines.","Mean number of days required for 50% of 20 flies to die (LT50) when exposed to the fungal pathogen Metarhizium anisopliae (Ma549). Higher values indicate greater resistance to Ma549.","Mean number of days required for 50% of 20 flies to die (LT50) when exposed to the fungal pathogen Metarhizium anisopliae (Ma549). Higher values indicate greater resistance to Ma549.","Mean mortality after 12 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 12 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 24 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 24 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 3 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 3 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 6 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean mortality after 6 hours of exposure to melathion, an organophosphate pesticide. Higher values indicate greater susceptibility to melathion.","Mean virgin female body mass for groups of three flies, after rearing on food with 0.5% yeast content (dietary restricted). Higher values indicate heavier flies.","Mean virgin female body mass for groups of three flies, after rearing on food with 5% yeast content (ab libitum diet). Higher values indicate heavier flies.","Mean within-line (micro-environmental) variance in lifespan at 18C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 18C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 25C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 25C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 28C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean within-line (micro-environmental) variance in lifespan at 28C. Such micro-environmental variance may be due to the plasticity of lifespan in response to unmeasured environmental variation and intrinsic variability of each individual genotype. Higher values indicate greater variance in lifespan.","Mean hours survived when exposed to food medium supplemented with medione sodium bisulfite (MSB; 75 mM total). MSB facilitates oxidative stress induced mortality. Higher values indicate greater resistance to oxidative stress.","Mean hours survived when exposed to food medium supplemented with medione sodium bisulfite (MSB; 75 mM total). MSB facilitates oxidative stress induced mortality. Higher values indicate greater resistance to oxidative stress.","Mean height climbed in 5 seconds (expressed as 26 5mm bins). Higher values indicate faster climbing and greater negative geotaxis.","Mean height climbed in 5 seconds (expressed as 26 5mm bins). Higher values indicate faster climbing and greater negative geotaxis.","Mean height climbed in 5 seconds (expressed as 26 5mm bins). Prior to phenotyping flies were exposed to food medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration). MSB facilitates oxidative stress. Higher values indicate faster climbing and greater negative geotaxis.","Mean height climbed in 5 seconds (expressed as 26 5mm bins). Prior to phenotyping flies were exposed to food medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration). MSB facilitates oxidative stress. Higher values indicate faster climbing and greater negative geotaxis.","Mean difference in negative geotaxis between flies reared on medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration) and flies reared on standard medium. Calculated as the difference between control and MSB conditions, divided by the population mean difference between control and MSB conditions. MSB facilitates oxidative stress induced mortality. Higher values indicate a greater sensitivity to oxidative stress and a more compromised startle response.","Mean difference in negative geotaxis between flies reared on medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration) and flies reared on standard medium. Calculated as the difference between control and MSB conditions, divided by the population mean difference between control and MSB conditions. MSB facilitates oxidative stress induced mortality. Higher values indicate a greater sensitivity to oxidative stress and a more compromised startle response.","The mean length of a sleep bout during the 12-hour dark period of the light cycle. Higher values indicate longer bouts.","The mean length of a sleep bout during the 12-hour dark period of the light cycle. Higher values indicate longer bouts.","The standard deviation for the length of a sleep bout during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep bout length. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the length of a sleep bout during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep bout length. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour dark period of the light cycle. Higher values indicate a higher frequency of sleep bouts.","Mean number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour dark period of the light cycle. Higher values indicate a higher frequency of sleep bouts.","The standard deviation for the number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in number of sleep bouts. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the number of separate periods of inactivity lasting &gt; 5 minutes, during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in number of sleep bouts. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour dark period of the light cycle. Higher values indicate a higher line average for sleep duration.","The sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour dark period of the light cycle. Higher values indicate a higher line average for sleep duration.","The standard deviation for the sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep duration. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","The standard deviation for the sum of periods of inactivity lasting &gt; 5 minutes (the studies definition of sleep) during the 12-hour dark period of the light cycle. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more within-individual variation in sleep duration. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean number of flies that moved away from a cotton wool swab that had been dipped in 1-Hexanol solution. Higher values indicate avoidance of 1-Hexanol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 1-Hexanol solution. Higher values indicate avoidance of 1-Hexanol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 2-Heptanone solution. Higher values indicate avoidance of 2-Heptanone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 2-Heptanone solution. Higher values indicate avoidance of 2-Heptanone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 2-Phenyl ethyl alcohol solution. Higher values indicate avoidance of 2-Phenyl ethyl alcohol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in 2-Phenyl ethyl alcohol solution. Higher values indicate avoidance of 2-Phenyl ethyl alcohol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Acetophenone solution. Higher values indicate avoidance of Acetophenone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Acetophenone solution. Higher values indicate avoidance of Acetophenone, lower values indicate attraction.","Mean response of flies to a dipstick assay using 0.3% benzaldehyde. Higher values indicate aversion, lower values indicate attraction.","Mean response of flies to a dipstick assay using 0.3% benzaldehyde. Higher values indicate aversion, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Benzaldehyde solution. Higher values indicate avoidance of Benzaldehyde, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Benzaldehyde solution. Higher values indicate avoidance of Benzaldehyde, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Citral solution. Higher values indicate avoidance of Citral, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Citral solution. Higher values indicate avoidance of Citral, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in d-Carvone solution. Higher values indicate avoidance of d-Carvone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in d-Carvone solution. Higher values indicate avoidance of d-Carvone, lower values indicate attraction.","Flies were given a choice between a tube containing ethanol and a tube containing water. Higher values indicate an attraction to ethanol, while lower values indicate an aversion to ethanol.","Flies were given a choice between a tube containing ethanol and a tube containing water. Higher values indicate an attraction to ethanol, while lower values indicate an aversion to ethanol.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Ethylacetate solution. Higher values indicate avoidance of Ethylacetate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Ethylacetate solution. Higher values indicate avoidance of Ethylacetate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Ethyl butyrate solution. Higher values indicate avoidance of Ethyl butyrate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Ethyl butyrate solution. Higher values indicate avoidance of Ethyl butyrate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Eugenol solution. Higher values indicate avoidance of Eugenol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Eugenol solution. Higher values indicate avoidance of Eugenol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Heliol solution. Higher values indicate avoidance of Heliol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Heliol solution. Higher values indicate avoidance of Heliol, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Hexal solution. Higher values indicate avoidance of Hexal, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Hexal solution. Higher values indicate avoidance of Hexal, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in l-Carvone solution. Higher values indicate avoidance of l-Carvone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in l-Carvone solution. Higher values indicate avoidance of l-Carvone, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Methyl salicylate solution. Higher values indicate avoidance of Methyl salicylate, lower values indicate attraction.","Mean number of flies that moved away from a cotton wool swab that had been dipped in Methyl salicylate solution. Higher values indicate avoidance of Methyl salicylate, lower values indicate attraction.","Mean difference between the number of ovarioles in the right and left ovary. Higher values indicate greater asymmetry.","Mean number of ovarioles in the right and left ovary. Higher values indicate a greater number of ovarioles.","Mean number of fecal spots detected after 20h for 10 flies, after exposure to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate more defecation.","Mean pH of fecal matter for flies after exposure to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate a higher pH.","Mean amount of intestinal mitosis, five days after exposure to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate a greater number of mitotic itensitinal cells.","Mean number of days required for 50% of 20 flies to die (LT50) when exposed to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate greater resistance to Pa14.","Mean number of days required for 50% of 20 flies to die (LT50) when exposed to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate greater resistance to Pa14.","Mean number of days required for 50% of flies in a vial to die (LT50) when exposed to the bacterial pathogen Pseudomos aeruginosa (Pa14). Higher values indicate greater resistance to Pa14.","Mean percentage of flies dying after 3 days post infection with P.entomophila L48, an entomopathogenic bacterium. Higher values indicate greater mortality.","Mean colony forming units (CFU) produced from the homogenate of three pooled flies, 20 hours after infection with P. rettgeri, a bacterial pathogen. Data are reported on the ln scale. Higher values indicate a greater bacterial load.","Mean ln(bacterial load) per fly, 24 hours after infection with P. rettgeri, a bacterial pathogen. Prior to infection, flies were reared on a diet containing 10% glucose. Higher values indicate a greater bacterial load.","Mean ln(bacterial load) per fly, 24 hours after infection with P. rettgeri, a bacterial pathogen. Prior to infection, flies were reared on a diet containing 2.5% glucose. Higher values indicate a greater bacterial load.","Mean ln(bacterial load) per fly, 24 hours after infection with P. rettgeri, a bacterial pathogen. Prior to infection, flies were reared on a diet containing 5% glucose. Higher values indicate a greater bacterial load.","Mean proportion of flies surviving infection, two days after inoculation with the bacterial pathogen P. rettgeri. Data are reported on the logit scale. Higher values indicate greater resistance to P. rettgeri.","A general, nonlinear relationship between the proportion of flies surviving two days post infection and sustained pathogen load, after injection with the bacterial pathogen P. rettgeri, was predicted. The data presented here are the line deviations from the predicted relationship between survival and bacterial load. Positive values indicate a greater tolerance to P. rettgeri.","Mean difference in centimetres climbed in three seconds following vial inversion, after 24 hours of exposure to sucrose or sucrose supplemented with paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater resistance to paraquat and the effects of oxidative stress.","Mean difference in centimetres climbed in three seconds following vial inversion, after 24 hours of exposure to sucrose or sucrose supplemented with paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater resistance to paraquat and the effects of oxidative stress.","Mean hours survived when exposed to food medium supplemented with paraquat (20 mM/50ml concentration). Paraquat is a herbicide that facilitates oxidative stress induced mortality. Higher values indicate greater resistance to oxidative stress.","Mean hours survived when exposed to food medium supplemented with paraquat (20 mM/50ml concentration). Paraquat is a herbicide that facilitates oxidative stress induced mortality. Higher values indicate greater resistance to oxidative stress.","Mean proportion of flies 'knocked down' after 3 hours of exposure to permethrin, a pyrethroid pesticide. Knockdown is a phenotype where the fly lies paralyzed and twitching, or exhibits uncontrolled flight. Higher values indicate greater susceptibility to permethrin.","Mean proportion of flies 'knocked down' after 3 hours of exposure to permethrin, a pyrethroid pesticide. Knockdown is a phenotype where the fly lies paralyzed and twitching, or exhibits uncontrolled flight. Higher values indicate greater susceptibility to permethrin.","Mean mortality after 24 hours of exposure to permethrin, a pyrethroid pesticide. Higher values indicate greater susceptibility to permethrin.","Mean mortality after 24 hours of exposure to permethrin, a pyrethroid pesticide. Higher values indicate greater susceptibility to permethrin.","Mean phototaxis score measured in 14 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 14 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 28 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 28 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 7 day old adults. Higher values indicate a stronger phototaxic response.","Mean phototaxis score measured in 7 day old adults. Higher values indicate a stronger phototaxic response.","Mean PC1 score for posterior lobe shape, calculated from 200 Fourier coefficients. The posterior lobe is a genital structure used to grasp the female oviscape to ensure stable genital coupling. PC1 explains 78% of the variation in shape, and describes the degree of beak-like protrusion at the tip of the posterior lobes. Higher values indicate a greater degree of beak-like postrusion.","Mean PC2 score for posterior lobe shape, calculated from 200 Fourier coefficients. The posterior lobe is a genital structure used to grasp the female oviscape to ensure stable genital coupling. PC1 explains 10% of the variation in shape, and describes the width of the posterior lobes. Higher values indicate wider lobes.","Mean posterior lobe size, measured in pixels. The posterior lobe is a genital structure used to grasp the female oviscape to ensure stable genital coupling. Higher values indicate larger posterior lobes.","Estimated mass of soluble protein per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean protein (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of protein.","Mean protein (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of protein.","Mean protein (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of protein.","Mean survival 24 hours after exposure to radiation (Cs137, avg dose of 4.85 Gy/min. Total dose of 1382 Gy over the 4 hours and 45 minutes of continuous exposure). Higher values indicate greater radiation resistance.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 18C. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 18C. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C and RCH was measured when adults were 28-days old. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C and RCH was measured when adults were 28-days old. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C and RCH was measured when adults were 7-days old. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature (4 degrees C) before exposing them to a cold temperature (-6 degrees C), while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared at 25C and RCH was measured when adults were 7-days old. Higher values indicate a greater plastic response.","Mean proportion of adults infected with sigma virus (DmelSV), 13 days post infection. Sigma virus is a rhabdovirus that naturally occurs in D. melanogaster. It causes flies paralysis in flies exposed to CO2. Higher values indicate a higher (???) proportion of paralysed flies after exposure to CO2, and therefore greater sensitivity to sigma virus.","Mean number of aggressive encounters over a two-minute period between pairs of males; a DGRP male and a control Canton-SB male. The DGRP male was housed in a mixed Sex environment for the 3 days preceding the aggression assay. Higher values indicate more aggressive males.","Mean difference in aggressive behavior between male flies raised in social and socially isolated environments. Higher values indicate that males raised in a social environment were more aggressive than males experiencing isolation (the authors adjusted line means to meet normality expectations, we present udjusted means for the social and isolation line means).","Median interpulse interval (IPI) in the male courtship 'song'. Males in many Drosophila species produce a series of auditory pulses (a pulse 'song') when courting females, and the pause between pulses (IPI) varies greatly among species. Higher values indicate a longer IPI.","The percent of activity performed in the first quarter of a two hour induced exercise period. The author reasons that this is a measure of the tendency to behave like a sprinter versus a marathoner. Note: this data is recoded from that recorded by Watanabe et al (2020).","The percent of activity performed in the first quarter of a two hour induced exercise period. The author reasons that this is a measure of the tendency to behave like a sprinter versus a marathoner. Higher values indicate that a greater proportion of activity was completed in the first 30 minutes, which corresponds to a 'sprinter' phenotype. Note: this data is recoded from that recorded by Watanabe et al (2020).","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Predictions are adjusted for block effects. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Predictions are adjusted for block effects. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Prior to phenotyping flies were exposed to food medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration). MSB facilitates oxidative stress. Predictions are adjusted for block effects. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Prior to phenotyping flies were exposed to food medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration). MSB facilitates oxidative stress. Predictions are adjusted for block effects. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Higher values indicate a greater startle response.","Mean time a fly was active in the 45 seconds following a mechanical disturbance. Higher values indicate a greater startle response.","Mean variance (the log of the within line standard deviation; ln(σE)) in the time a fly was active in the 45 seconds following a mechanical disturbance. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable startle response, and therefore greater micro-environmental plasticity. Note that the data presented in this study was originally collected by Mackay et al (2012).","Mean variance (the log of the within line standard deviation; ln(σE)) in the time a fly was active in the 45 seconds following a mechanical disturbance. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable startle response, and therefore greater micro-environmental plasticity. Note that the data presented in this study was originally collected by Mackay et al (2012).","Mean difference in startle response between flies reared on medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration) and flies reared on standard medium. Calculated as the difference between control and MSB conditions, divided by the population mean difference between control and MSB conditions. MSB facilitates oxidative stress induced mortality. Higher values indicate a greater sensitivity to oxidative stress and a more compromised startle response.","Mean difference in startle response between flies reared on medium supplemented with medione sodium bisulfite (MSB; 3 mM concentration) and flies reared on standard medium. Calculated as the difference between control and MSB conditions, divided by the population mean difference between control and MSB conditions. MSB facilitates oxidative stress induced mortality. Higher values indicate a greater sensitivity to oxidative stress and a more compromised startle response.","Mean hours survived in culture vials containing non-nutritive medium. Higher values indicate greater starvation resistance.","Mean hours survived in culture vials containing non-nutritive medium. Higher values indicate greater starvation resistance.","Adult mean survival in the absence of food after two days is here used as a measure of starvation resistance. Higher values indicate greater survival and therefore higher starvation resistance.","Survival time in hours for adult flies that were first flies maintained on a high carbohydrate diet (carbs:494.15g/L, protein:52g/L, fat:2.5g/L) for 10 days, then starved in vials supplied only with water. Higher values indicate greater starvation resistance.","Survival time in hours for adult flies that were first flies maintained on a high fat diet (carbs:44.15g/L, protein:52g/L, fat:17.5g/L) for 10 days, then starved in vials supplied only with water. Higher values indicate greater starvation resistance.","Survival time in hours for adult flies that were first flies maintained on a high protein diet (carbs:44.15g/L, protein:52g/L, fat:2.5g/L) for 10 days, then starved in vials supplied only with water. Higher values indicate greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared at 25C and the stravation trial started when adults were 28-days old. Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared at 25C and the stravation trial started when adults were 28-days old. Higher values indicate flies that exhibited greater starvation resistance.","Survival time in hours for adult flies that were first flies maintained on the laboratories standard diet (carbs:119.15g/L, protein:52g/L, fat:2.5g/L) for 10 days, then starved in vials supplied only with water. Higher values indicate greater starvation resistance.","Mean variance (the log of the within line standard deviation; ln(σE)) in the hours flies survived in culture vials containing non-nutritive medium. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable starvation resistance response, and therefore greater micro-environmental plasticity. Note that the data presented in this study were originally collected by Mackay et al (2012).","Mean variance (the log of the within line standard deviation; ln(σE)) in the hours flies survived in culture vials containing non-nutritive medium. Variance is used as a measure of micro-environmental plasticity. Higher values indicate a more variable starvation resistance response, and therefore greater micro-environmental plasticity. Note that the data presented in this study were originally collected by Mackay et al (2012).","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared on food with 0.5% yeast content (dietary restricted).  Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared on food with 5% yeast content (ab libitum diet).  Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared at 25C and the stravation trial started when adults were 7-days old. Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. These flies were reared at 25C and the stravation trial started when adults were 7-days old.  Higher values indicate flies that exhibited greater starvation resistance.","Mean pigmentation scores for tergite 5. Higher values indicate higher percentages of brown/black melanin on the tergite.","Mean pigmentation scores for tergite 6. Higher values indicate higher percentages of brown/black melanin on the tergite.","Mean adult thorax length when reared at 18C. Higher values indicate larger flies.","Mean adult thorax length when reared at 28C. Higher values indicate larger flies.","Flies were reared at 18C and 28C, after which adult thorax lengths were measured. Reaction norms for each DGRP line were calculated using the linear model lm (Size ~ Temperature). The absolute value of the slope is a measurement of thermal sensitivity, describing the magnitude of the plastic response to temperature. Higher values indicate greater thermal plasticity.","Estimated mass of total trigylcerides per male fly divided by the average mass of males from that line. Higher values indicate greater content of the macronutrient.","Mean triglycerides (mg/ul) extracted from 10 flies that were reared on a diet containing 10% glucose. Higher values indicate greater levels of triglycerides.","Mean triglycerides (mg/ul) extracted from 10 flies that were reared on a diet containing 2.5% glucose. Higher values indicate greater levels of triglycerides.","Mean triglycerides (mg/ul) extracted from 10 flies that were reared on a diet containing 5% glucose. Higher values indicate greater levels of triglycerides.","Mean quantity of triglyceride extracted from whole fly homogenate and standardised for body size. Three flies reared on food with 0.5% yeast content (dietary restricted) were homogenised. Higher values indicate higher triglyceride levels.","Mean quantity of triglyceride extracted from whole fly homogenate and standardised for body size. Three flies reared on food with 5% yeast content (ab libitum diet) were homogenised. Higher values indicate higher triglyceride levels.","The hazard ratio of the death rate of male flies exposed to tunicamycin - an endoplasmic reticulum (ER) stress inducing drug – with the control group, fed drug free media. Higher values indicate greater ER stress and lower survival of flies exposed to tunicamycin.","Mean time required for 50% of male flies exposed to tunicamycin (TM) media to die. TM is an antibiotic that causes endoplasmic reticulum (ER) stress. Higher values indicate longer survival times and greater resistance to ER stress.","Mean unfertilised egg retention time, measured as the number of days before a virgin female laid her first egg. Higher values indicate a longer retention time.","Mean number of times the fly crossed the infrared beam used to measure activity, divided by the total time spent awake. Higher values indicate more active lines.","Mean number of times the fly crossed the infrared beam used to measure activity, divided by the total time spent awake. Higher values indicate more active lines.","Standard deviation for the number of times the fly crossed the infrared beam used to measure activity, divided by the total time spent awake. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more variable within-individual activity. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Standard deviation for the number of times the fly crossed the infrared beam used to measure activity, divided by the total time spent awake. Standard deviations are used to estimate the inter-individual variation in sleep. Higher values indicate more variable within-individual activity. Note that this study reanalyses the data collected in Harbison, McCoy and Mackay (2013).","Mean fecundity in the first week of a female's adult life, measured as a 2-day total of eggs laid, when living with a single standard male, at 25C. Higher values indicate greater fecundity.","Mean fecundity in the third week of a female's adult life, measured as a 2-day total of eggs laid, when living with a single standard male, at 25C. Higher values indicate greater fecundity.","Mean fecundity in the fifth week of a female's adult life, measured as a 2-day total of eggs laid, when living with a single standard male, at 25C. Higher values indicate greater fecundity. Note that female's were eliminated form the analysis upon death.","Mean fecundity in the seventh week of a female's adult life, measured as a 2-day total of eggs laid, when living with a single standard male, at 25C. Higher values indicate greater fecundity. Note that female's were eliminated form the analysis upon death.","Mean weight (mg) of live flies that were reared on a diet containing 10% glucose. Higher values indicate heavier flies.","Mean weight (mg) of live flies that were reared on a diet containing 2.5% glucose. Higher values indicate heavier flies.","Mean weight (mg) of live flies that were reared on a diet containing 5% glucose. Higher values indicate heavier flies.","Mean mass. Higher values indicate heavier flies.","Mean wing centroid size. Higher values indicate lines with larger wings, which can be used as a proxy for body size.","Mean wing centroid size. Higher values indicate lines with larger wings, which can be used as a proxy for body size.","Mean leg length for legs originating on the first thoracic segment. Higher values indicate longer legs","Mean leg length for legs originating on the first thoracic segment. Higher values indicate longer legs","Mean leg length for legs originating on the second thoracic segment. Higher values indicate longer legs","Mean leg length for legs originating on the second thoracic segment. Higher values indicate longer legs","Mean femur length, for legs originating on the first thoracic segment. Higher values indicate longer femurs","Mean femur length, for legs originating on the first thoracic segment. Higher values indicate longer femurs","Mean femur length, for legs originating on the second thoracic segment. Higher values indicate longer femurs","Mean femur length, for legs originating on the second thoracic segment. Higher values indicate longer femurs","Mean tibia length, for legs originating on the first thoracic segment. Higher values indicate longer tibias","Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias","Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias","Mean tibia length, for legs originating on the second thoracic segment. Higher values indicate longer tibias","Mean tarsus length, for legs originating on the first thoracic segment. Higher values indicate longer tarsus'.","Mean tarsus length, for legs originating on the first thoracic segment. Higher values indicate longer tarsus'.","Mean tarsus length, for legs originating on the second thoracic segment. Higher values indicate longer tarsus'.","Mean tarsus length, for legs originating on the second thoracic segment. Higher values indicate longer tarsus'.","Mean combined dry weight of 5 adult flies that lacked a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate heavier flies.","Mean combined dry weight of 5 adult flies that possessed a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate heavier flies.","Mean change in the combined dry weight of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota had a greater positive effect on weight. Note that the presence of micriobiota on average increased dry weight.","Mean combined glucose content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher glucose content.","Mean combined glucose content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher glucose content.","Mean change in the combined glucose content of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher glucose content. Note though that the presence of micriobiota on average decreased glucose content.","Mean combined protein content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher protein content.","Mean combined protein content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher protein content.","Mean combined triglyceride (TAG) content after glycerol substraction extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher TAG content.","Mean number of times a DGRP male orientated towards a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of orientations, and potentially more persistent courtship.","Mean number of times a DGRP male approached a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of approaches, and potentially more persistent courtship.","Mean number of times a DGRP male extended their wing at a 90 degree angle in rapid vibration while in the presence of a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater number of wing displays, and potentially more persistent courtship.","Mean number of genital licks by DGRP male while courting a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater frequency of genital licking, and potentially more persistent courtship.","Mean number of attempted copulations by DGRP male while courting a virgin female, detected by scan sampling 30 times over a 15 minute period. Higher values indicate a greater frequency of copulation attempts, and potentially more persistent courtship.","Mean latency until a DGRP male was observed copulating with a virgin female, detected by scan sampling 30 times over a 15 minute period. Lower values indicate males that began mating faster, which could be interpreted as males that are more attractive to females.","Mean time, in hours, adult females survived during continuous exposure to medium supplemented with 1% caffeine. Higher values indicate greater resistance to caffeine","Mean proportion of surviving adults 24 hours post exposure to an hour of cold shock (-5°C). Higher values indicate flies that are more resistant to cold shock.","Mean proportion of surviving adults 24 hours post exposure to an hour of cold shock (-5°C). Higher values indicate flies that are more resistant to cold shock.","Mean lifespan, measured in hours, for mated females exposed to media supplemented with 1.5% boric acid (BH3O3, BP168). Boric acid is a common household pesticide. Higher values indicate a greater resistance to boric acid.","Mean survival 48 hours after exposure to deltamethrin, a pyrethroid pesticide. Higher values indicate greater resistance to deltamethrin.","Mean survival 48 hours after exposure to parathion, an organophosphate pesticide. Higher values indicate greater resistance to parathion.","Median number of days adult flies survived post infection with Kallithea virus (50 nL of 105 ID50 originally injected). Measured as LT50 - the point at which 50% of 10 flies died. Higher values indicate greater resistance to Kallithea virus.","Median number of days adult flies survived post infection with Kallithea virus (50 nL of 105 ID50 originally injected). Measured as LT50 - the point at which 50% of 10 flies died. Higher values indicate greater resistance to Kallithea virus.","Mean viral titre of kallithea virus found in groups of 10 adult females, 8 days after injection of 50 nL of 105 ID50 KV.  This is a measure of viral load, with higher values indicating greater viral load.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. Higher values indicate flies that exhibited greater starvation resistance.","Mean hours survived on inedible food medium, used as a measure of starvation resistance. Higher values indicate flies that exhibited greater starvation resistance.","Mean left wing centroid size for flies measured in the Houle lab. Images were taken of live flies using the 'Wingmachine' system. Higher values indicate larger wings.","Mean left wing centroid size for flies measured in the Houle lab. Images were taken of live flies using the 'Wingmachine' system. Higher values indicate larger wings.","Mean left wing centroid size for flies measured in the Dworkin lab. Images were taken of dissected wings using an Olympus microscope. Higher values indicate larger wings.","Mean wing centroid size for flies measured in the Dworkin lab. Images were taken of dissected wings using an Olympus microscope. Higher values indicate larger wings.","Mean distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height), following 24 hrs of exposure to a 5% sucrose solution diet. Higher values indicate more active flies.","Mean distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height), following 24 hrs of exposure to a 5% sucrose solution diet, supplemented with MPH (1.5 mg ml^-1). MPH, methylphenidate, is a drug used to treat attention-deficit/hyperactivity disorder (ADHD) in humans. Higher values indicate more active flies.","Mean difference in distance moved during a 10-min trial in a circular arena (16mm diameter x 6mm height) between flies fed a control 5% sucrose solution and a 5% sucrose solution supplemented with 1.5 mg ml^-1 methylphenidate (MPH). MPH is a drug used to treat the symptoms of attention-deficit/hyperactivity disorder (ADHD) in humans. Positive values indicate greater activity when flies were supplemented with MPH.","Mean proportion of adult flies that were alive 5 days after infection with the bacterial pathogen E. faecalis. Higher values indicate greater resistance to E. faecalis","Mean proportion of males that produced viable larvae. Males were reared at 25C. Higher values indicates greater fertility.","Mean proportion of males that produced viable larvae. Males were reared at 27C. Higher values indicates greater fertility","Mean proportion of males that produced viable larvae. Males were reared at 29C. Higher values indicates greater fertility","Mean slope of the change in male fertility (the proportion of males that produced viable larvae), between 25C, 27C and 29C rearing conditions. Lower slope values indicate a greater decrease in fertility with an increase in temperature. This is a measure of thermal susceptibility.","Mean relative 1/3-Methylhistidine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 1/3-Methylhistidine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 1-Methyladenosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 1-Methyladenosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 1-Methylguanosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 1-Methylguanosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 13-HODE concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 13-HODE concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 2-Hydroxyglutarate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 2-Hydroxyglutarate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 2-Phosphoglyceric Acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 2-Phosphoglyceric Acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 3-Hydroxybutyric Acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 3-Hydroxybutyric Acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 3-Hydroxykynurenine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 3-Hydroxykynurenine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 4-Pyridoxic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 4-Pyridoxic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 5-Aminovalerate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative 5-Aminovalerate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean bacterial load, calculated from the DNA content of the bacterial strain and the DNA content of the fly. Higher values indicate a greater bacterial load. The bacteria measured here is A. tropicalis; a commensal gut bacteria commonly found in the GI tract of D.melanogaster.","Incidence of DGRP males abdomen curling in the presence of an interacting male, sourced from a wildtype laboratory strain. Higher values indicate a greater incidence of abdomen curling.","Mean relative acetoacetate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative acetoacetate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative acetylcarnitine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative acetylcarnitine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative acetylcholine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative acetylcholine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","An overall measure of activity level obtained by summing the total number of tapping, orientating, following, licking, singing and abdomen curling for a male housed in a vial with a single additional male. Higher values indicate greater activity.","Mean relative adenine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative adenine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative adenosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative adenosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative ADP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative ADP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative agmatine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative agmatine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative alanine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative alanine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative allantoin concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative allantoin concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean number of flies exposed to alpha-amanitin, at concentration 0.2 ug/g, surviving to adulthood per vial. Alpha-amanitin is the principal toxin of several deadly poisonous mushroom species, and is only found in mushrooms. Higher values indicate a greater resistance to alpha-amanitin. Note that resistance is confounded with survival in this study.","Mean number of flies exposed to alpha-amanitin, at concentration 2 ug/g, surviving to adulthood per vial. Alpha-amanitin is the principal toxin of several deadly poisonous mushroom species, and is only found in mushrooms. Higher values indicate a greater resistance to alpha-amanitin. Note that resistance is confounded with survival in this study.","Mean alpha-amanitin concentration required for 50% of larvae to die prior to reaching adulthood (LC50). Alpha-amanitin is the principal toxin of several deadly poisonous mushroom species, and is only found in mushrooms. Higher values indicate a higher LC50 and a greater resistance to alpha-amanitin. Note that resistance is confounded with survival in this study.","Mean relative Alpha-Ketoglutaric Acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative Alpha-Ketoglutaric Acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean change in amino acid diluted food medium preference after deprivation of amino acids compared with flies that had experienced no deprivation of amino acids. Positive values indicate a stronger change in preference towards the amino acid mixture after deprivation.","Mean change in amino acid diluted food medium preference after deprivation of amino acids compared with flies that had experienced no deprivation of amino acids. Positive values indicate a stronger change in preference towards the amino acid mixture after deprivation.","Mean relative aminoisobutyrate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative aminoisobutyrate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative AMP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative AMP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean survival (larvae that pupated) for late third instar larvae, 24 hours after exposure to one hour of anoxia (a vial perfused with 80% nitrogen). Higher values indicate a greater a greater tolerance of anoxia.","Mean relative anthranilate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative anthranilate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative arachidonate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative arachidonate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative arginine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative arginine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative asparagine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative asparagine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative aspartic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative aspartic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean proportion of larvae surviving to pupation after 7 days on food media containing azinphos-methyl (an organophosphate insecticide) at 0.25 ug/ml. Higher values indicate greater mortality.","Mean proportion of larvae surviving to pupation after 7 days on food media containing azinphos-methyl (an organophosphate insecticide) at 0.5 ug/ml. Higher values indicate greater mortality.","Mean proportion of larvae surviving to pupation after 7 days on food media containing azinphos-methyl (an organophosphate insecticide) at 1 ug/ml. Higher values indicate greater mortality.","Mean proportion of larvae surviving to pupation after 7 days on food media containing azinphos-methyl (an organophosphate insecticide) at 2 ug/ml. Higher values indicate greater mortality.","Mean dosage (ug/ml) of azinphos-methyl (an organophosphate insecticide) required for 50% of larvae to die. Higher values indicate greater resistance.","Mean relative betaine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative betaine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative biotin concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative biotin concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cadaverine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cadaverine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative carnitine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative carnitine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cGMP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cGMP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean change in preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, between day one and day three of exposure. Higher values indicate a more positive change in preference for cocaine supplemented solution.","Mean change in preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, between day one and day three of exposure. Higher values indicate a more positive change in preference for cocaine supplemented solution.","Mean change in preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, between day one and day three of exposure. Preference is measured as the difference between the amount of cocaine supplemented sucrose consumed and unsupplemented sucrose consumed, scaled by the total amount of both consumed during the assay. Higher values indicate a more positive change in preference for cocaine supplemented solution.","Mean change in preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, between day one and day three of exposure. Preference is measured as the difference between the amount of cocaine supplemented sucrose consumed and unsupplemented sucrose consumed, scaled by the total amount of both consumed during the assay. Higher values indicate a more positive change in preference for cocaine supplemented solution.","Mean change in preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, between day one and day three of exposure. Higher values indicate a more positive change in preference for methamphetamine supplemented solution.","Mean change in preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, between day one and day three of exposure. Higher values indicate a more positive change in preference for methamphetamine supplemented solution.","Mean change in preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, between day one and day three of exposure. Preference is measured as the difference between the amount of methamphetamine supplemented sucrose consumed and unsupplemented sucrose consumed, scaled by the total amount of both consumed during the assay. Higher values indicate a more positive change in preference for methamphetamine supplemented solution.","Mean change in preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, between day one and day three of exposure. Preference is measured as the difference between the amount of methamphetamine supplemented sucrose consumed and unsupplemented sucrose consumed, scaled by the total amount of both consumed during the assay. Higher values indicate a more positive change in preference for methamphetamine supplemented solution.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 0.5 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 1 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 2 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 3 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 4 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean survival of first instar larvae to adulthood when reared on food medium supplemented with 5 ug/ml of chlorantraniliprole. This is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean chlorantraniliprole dosage required for 50% of first instar larvae to die before eclosion. Chlorantraniliprole is an anthranilic diamide, a new class of insecticide that targets the ryanodine receptor. Higher values indicate greater resistance to chlorantraniprole.","Mean relative choline concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative choline concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean citrate synthase activity per adult fly. It is involved in the first step of the TCA cycle, and is therefore considered a proxy for mitochondrial aerobic capacity. Higher values indicate higher citrate synthase activity.","Mean citrate synthase activity per adult fly. It is involved in the first step of the TCA cycle, and is therefore considered a proxy for mitochondrial aerobic capacity. Higher values indicate higher citrate synthase activity.","Mean relative citrulline concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative citrulline concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean climbing index score - calculated using a design that takes advantage of negative geotaxis - measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate greater climbing ability.","Mean climbing index score - calculated using a design that takes advantage of negative geotaxis - following a three-day exercise regime. Higher values indicate greater climbing ability.","Mean climbing index score - calculated using a design that takes advantage of negative geotaxis - prior to a three-day exercise regime. Higher values indicate greater climbing ability.","Mean relative CMP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative CMP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean CO2 production (uL per hour, per mg dry weight). The authors use this trait as a proxy for metabolic rate. Higher values indicate a faster metabolism.","Mean preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, over unsupplemented sucrose solution. Preference was measured on the first of three days of exposure. Higher values indicate a greater preference for cocaine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, over unsupplemented sucrose solution. Preference was measured on the first of three days of exposure. Higher values indicate a greater preference for cocaine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, over unsupplemented sucrose solution. Preference was measured on the third of three days of exposure. Higher values indicate a greater preference for cocaine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, over unsupplemented sucrose solution. Preference was measured on the third of three days of exposure. Higher values indicate a greater preference for cocaine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, over unsupplemented sucrose solution. Preference was measured on the first of three days of exposure and is scaled by the total amount of solution consumed during the assay. Higher values indicate a greater preference for cocaine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, over unsupplemented sucrose solution. Preference was measured on the first of three days of exposure and is scaled by the total amount of solution consumed during the assay. Higher values indicate a greater preference for cocaine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, over unsupplemented sucrose solution. Preference was measured on the third of three days of exposure and is scaled by the total amount of solution consumed during the assay. Higher values indicate a greater preference for cocaine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.2 mg/ml cocaine, over unsupplemented sucrose solution. Preference was measured on the third of three days of exposure and is scaled by the total amount of solution consumed during the assay. Higher values indicate a greater preference for cocaine supplemented solution.","Mean proportion of successfully eclosed individuals following an hour of cold shock (-5°C) at 5 days post oviposition (third instar). Higher values indicate larvae that are more resistant to cold shock.","The critical thermal maximum, which is defined as the mean temperature in °C at which a fly ceases activity. Higher values indicate a higher CT max.","The critical thermal maximum, which is defined as the mean temperature in °C at which a fly ceases activity. Higher values indicate a higher CT max.","Mean cumulative cold tolerance (CCT) score. This metric represents the shape of cold-survival curves calculated between -1°C and -7°C. Higher values indicate greater area under the curve, and therefore flies that are more resistant to cold shock.","Mean relative cystathionine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cystathionine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cystine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cystine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cytidine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cytidine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cytosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative cytosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative D-GA3P concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative D-GA3P concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative D-Leucic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative D-Leucic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean Dopamine content (pg/ng tissue) extracted from the heads of two flies. Higher values indicate a greater dopamine content. Note that the 20 lines were chosen for extreme paraquat resistance or susceptibility values.","Mean Dopamine content (pg/ng tissue) extracted from the heads of two flies, after flies had been exposed to sucrose supplemented with paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate a greater dopamine content. Note that the 20 lines were chosen for extreme paraquat resistance or susceptibility values.","Mean difference in Dopamine content (pg/ng tissue) extracted from the heads of two flies, after 24 hours of exposure to sucrose or sucrose supplemented with paraquat (200 uM in 1% sucrose solution). Paraquat is a herbicide that causes oxidative stress in D. melanogaster. Higher values indicate greater resistance to paraquat and the effects of oxidative stress on Dopamine content.","Dry weight of 24-hour old adult virgins. Higher values indicate heavier flies. Note that this data has not been published, but was collected and provided by G. Tollin, N. Elfiky, E. Davis, N. Morano, and H. Karp of the A. Fiumera lab.","Dry weight of 24-hour old adult virgins. Higher values indicate heavier flies. Note that this data has not been published, but was collected and provided by G. Tollin, N. Elfiky, E. Davis, N. Morano, and H. Karp of the A. Fiumera lab.","Mean duration of embryogenesis (DOE) - from fertilisation of an egg to hatching as a larva. Higher values indicate a longer period of embryogenesis.","Mean bacterial load, calculated from the DNA content of the bacterial strain and the DNA content of the fly. Higher values indicate a greater bacterial load. The bacteria measured here is E.faecalis; a commensal gut bacteria commonly found in the GI tract of D.melanogaster.","Estimated mean time from eclosion to laying an egg (days), when reared under fluctuating temperature controlled conditions, with a mean of 16.8C ± 6. Females cohabited a food vial with two males from the same DGRP line as the female. Calculated as T= Σ(lxmx)/R0, where lx = the proportion of females surviving each day, mx = the number of eggs laid each day and R0 is the net reproductive rate (or replacment rate). Higher values indicate a longer generation time. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Estimated mean time from eclosion to laying an egg (days), when reared under fluctuating temperature controlled conditions, with a mean of 16.8C ± 6. Females cohabited a food vial with two males from the same DGRP line as the female. Calculated as T= Σ(lxmx)/R0, where lx = the proportion of females surviving each day, mx = the number of eggs laid each day and R0 is the net reproductive rate (or replacment rate). Higher values indicate a longer generation time. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Egg-to-adult development time when reared at 20C, recorded to the nearest 8 hours. Higher values indicate a longer development time.","Egg-to-adult development time when reared at 25C, recorded to the nearest 8 hours. Higher values indicate a longer development time.","Egg-to-adult development time when reared at 30C, recorded to the nearest 8 hours. Higher values indicate a longer development time.","Mean development time in days from egg to adult eclosion, on a standard diet. Higher values indicate a longer development time.","Mean development time in days from egg to adult eclosion, on a diet supplemented with 10% EtOH (ethanol). Higher values indicate a longer development time.","Proportion of 75 eggs in a vial that successfully survived to adulthood at 25C. Higher values indicate greater survival.","Mean proportion of flies surviving to adulthood, when reared at 25C on control food medium. Higher values indicate greater survival.","Mean proportion of eggs that survived to adulthood, when reared at 25C. Higher values indicate greater viability.","Percentage of flies surviving to eclosion, after development on standard food. Higher values indicate greater viability.","Proportion of 75 eggs in a vial that successfully survived to adulthood at 20C. Higher values indicate greater survival.","Mean proportion of eggs that survived to adulthood, when reared at 28C. Higher values indicate greater viability.","Proportion of 75 eggs in a vial that successfully survived to adulthood at 30C. Higher values indicate greater survival.","Percentage of flies surviving to eclosion, after development on food supplemented with 10% EtOH (ethanol). Higher values indicate greater viability.","Mean proportion of eggs that hatched. Higher values indicate a greater proportion of viable eggs.","Proportion of 75 eggs in a vial that successfully survived to pupation at 20C. Higher values indicate greater survival.","Proportion of 75 eggs in a vial that successfully survived to pupation at 25C. Higher values indicate greater survival.","Proportion of 75 eggs in a vial that successfully survived to pupation at 30C. Higher values indicate greater survival.","Average number of eggs laid per day across a female’s lifetime, when cohabiting a food vial with two males from the same DGRP line as the female. Measured under fluctuating temperature controlled conditions, with a mean of 16.8C ± 6. Higher values indicate greater fecundity. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Average number of eggs laid per day across a female’s lifetime, when cohabiting a food vial with two males from the same DGRP line as the female. Measured under fluctuating temperature controlled conditions, with a mean of 23.8C ± 6. Higher values indicate greater fecundity. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Mean relative epinephrine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative epinephrine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","The mean difference in egg-to-adult development time in days, between those reared on ethanol-supplemented (10% EtOH) and standard food medium. Higher values indicate a longer development time for ethanol-supplemented flies.","Mean difference in the amount of time adult flies remained mobile across a 60 second period following mechanical disturbance, between those grown on 10% EtOH (ethanol) supplemented food and those that developed on standard medium. Lower values indicate a greater negative effect on startle response.","Mean difference in the amount of time adult flies remained mobile across a 60 second period following mechanical disturbance, between those grown on 10% EtOH (ethanol) supplemented food and those that developed on standard medium. Lower values indicate a greater negative effect on startle response.","The mean difference in the percentage of flies surviving to eclosion, between those reared on ethanol-supplemented (10% EtOH) and standard food medium. Lower values indicate a greater negative effect of ethanol on egg to adult viability.","Larval to adult survival on food media supplemented with 0.5PPM fipronil – an insecticide from the phenylpyrazole family, released for commercial use in 1993. Higher values indicate greater resistance.","Mean number of L1 larvae produced by females aged 5-6 days over a 24h period. For each replicate measurement, 5 DGRP females were housed with standardised mates and opposite-sex competitors (15 males and 10 females, giving an overall population size of 30 and 1:1 sex ratio). Fitness was estimated on the latent scale from a generalised linear mixed model, and is scaled to have a mean of zero and unit variance.","Mean number of L1 larvae sired by males in competition for matings/fertilisations with same-aged competitor males over 3 days, beginning when all males were aged 2-3 days. For each replicate measurement, 5 DGRP males were housed with standardised mates and opposite-sex competitors (15 females and 10 males, giving an overall population size of 30 and 1:1 sex ratio). Fitness was estimated on the latent scale from a generalised linear mixed model, and is scaled to have a mean of zero and unit variance.","Mean number of L1 larvae produced by females aged 17-18 days over a 24h period. For each replicate measurement, 5 DGRP females were housed with standardised mates and opposite-sex competitors (15 males and 10 females, giving an overall population size of 30 and 1:1 sex ratio). Fitness was estimated on the latent scale from a generalised linear mixed model, and is scaled to have a mean of zero and unit variance.","Mean number of L1 larvae sired by males in competition for matings/fertilisations with same-aged competitor males over 3 days, beginning when all males were aged 14-15 days. For each replicate measurement, 5 DGRP males were housed with standardised mates and opposite-sex competitors (15 females and 10 males, giving an overall population size of 30 and 1:1 sex ratio). Fitness was estimated on the latent scale from a generalised linear mixed model, and is scaled to have a mean of zero and unit variance.","Incidence of DGRP males following an interacting male, sourced from a wildtype laboratory strain. Higher values indicate a greater incidence of following.","Mean quantity of 10% sucrose, 5% yeast nutrient solution consumed by five virgin flies over an 8 hour period, measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate greater food consumption.","Mean quantity of 10% sucrose, 5% yeast nutrient solution consumed by five virgin flies over an 8 hour period, following a three-day exercise regime. Higher values indicate greater food consumption.","Mean relative G1P, G6P, F6P and F1P concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolites. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative G1P, G6P, F6P and F1P concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolites. Note these flies come from the same populations as used by Wilson et al (2020).","Mean size of the genome in mb, estimated using flow cytometry. Higher values indicate a larger genome.","Mean relative geranyl pyrophosphate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative geranyl pyrophosphate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glucoronate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glucoronate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean quantity of glucose per adult fly. Higher values indicate a higher glucose content.","Mean quantity of glucose per adult fly. Higher values indicate a higher glucose content.","Mean free glucose (mg/ul) extracted from 10 virgin flies. Higher values indicate greater levels of free glucose.","Mean free glucose (mg/ul) extracted from 10 virgin flies. Higher values indicate greater levels of free glucose.","Mean abdomen glucose content (log-transformed data is originally reported), measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate flies with a greater glucose content.","Mean abdomen glucose content (log-transformed data is originally reported), following a three-day exercise regime. Higher values indicate flies with a greater glucose content.","Mean preference for glucose when given a choice between glucose and fructose. Higher values indicate a greater preference for glucose consumption.","Mean preference for glucose when given a choice between glucose and fructose. Higher values indicate a greater preference for glucose consumption.","Mean relative glucose concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean quantity of glucose per third instar larvae. Higher values indicate a higher glucose content.","Mean relative glucose concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean preference for glucose when given a choice between glucose and sucrose. Higher values indicate a greater preference for glucose consumption.","Mean preference for glucose when given a choice between glucose and sucrose. Higher values indicate a greater preference for glucose consumption.","Mean thorax glucose content (log-transformed data is originally reported), measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate flies with a greater glucose content.","Mean thorax glucose content (log-transformed data is originally reported), following a three-day exercise regime. Higher values indicate flies with a greater glucose content.","Mean preference for glucose when given a choice between glucose and trehalose. Higher values indicate a greater preference for glucose consumption.","Mean preference for glucose when given a choice between glucose and trehalose. Higher values indicate a greater preference for glucose consumption.","Mean relative glutamic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glutamic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glutamine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glutamine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glyceraldehyde concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glyceraldehyde concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glycerate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glycerate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glycerol-3-P concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glycerol-3-P concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean quantity of glycerol per adult fly. Higher values indicate a higher glycerol content.","Mean quantity of glycerol per adult fly. Higher values indicate a higher glycerol content.","Mean free glycerol (mg/ul) extracted from 10 virgin flies. Higher values indicate greater levels of free glycerol.","Mean free glycerol (mg/ul) extracted from 10 virgin flies. Higher values indicate greater levels of free glycerol.","Mean abdomen glycerol content (log-transformed data is originally reported), measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate flies with a greater glucose content.","Mean abdomen glycerol content (log-transformed data is originally reported), following a three-day exercise regime. Higher values indicate flies with a greater glucose content.","Mean quantity of glycerol per third instar larvae. Higher values indicate a higher glycerol content.","Mean thorax glycerol content (log-transformed data is originally reported), measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate flies with a greater glucose content.","Mean thorax glycerol content (log-transformed data is originally reported), following a three-day exercise regime. Higher values indicate flies with a greater glucose content.","Mean relative glycine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative glycine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean change in glycine diluted food medium preference after deprivation of amino acids compared with flies that had experienced no deprivation of amino acids. Positive values indicate a stronger change in preference towards the glycine supplemented medium after deprivation.","Mean change in glycine diluted food medium preference after deprivation of amino acids compared with flies that had experienced no deprivation of amino acids. Positive values indicate a stronger change in preference towards the glycine supplemented medium after deprivation.","Mean quantity of glycogen per adult fly. Higher values indicate a higher glycogen content.","Mean quantity of glycogen per adult fly. Higher values indicate a higher glycogen content.","Mean glycogen (mg/ul) extracted from 10 virgin flies. Higher values indicate greater levels of glycogen.","Mean glycogen (mg/ul) extracted from 10 virgin flies. Higher values indicate greater levels of glycogen.","Mean quantity of glycogen per third instar larvae. Higher values indicate a higher glycogen content.","Mean relative GMP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative GMP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean seconds spent grooming in a five minute interval. Higher values indicate more frequent grooming.","Mean seconds spent grooming in a five minute interval. Higher values indicate more frequent grooming.","Mean relative guanosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative guanosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean time elapsed within a 37C waterbath before the fly was comatose. Higher values indicate greater resistance to heat stress.","Mean relative histamine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative histamine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative histidine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative histidine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative homogentisate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative homogentisate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean expression of the heat shock protein Hsp70, after exposure to 35C for one hour and an additiol hour of recovery at 25C. Higher values indicate greater expression of Hsp70.","Mean relative hypoxanthine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative hypoxanthine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean wiggle index for third instar larvae 1 hour into exposure to 100ppm of imidacloprid, a neonicotinoid pesticide. The wiggle index measures the relative movement ratio for larvae before and after they were exposed to imidacloprid. Higher values indicate a smaller decrease in motility and therefore greater resistance to imidacloprid.","Mean wiggle index for third instar larvae 1 hour into exposure to 25ppm of imidacloprid, a neonicotinoid pesticide. The wiggle index measures the relative movement ratio for larvae before and after they were exposed to imidacloprid. Higher values indicate a smaller decrease in motility and therefore greater resistance to imidacloprid.","Mean relative IMP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative IMP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative inosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative inosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative inositol concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative inositol concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative iso-leucine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative iso-leucine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative isoValeric acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative isoValeric acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative L-Kynurenine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative L-Kynurenine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean bacterial load, calculated from the DNA content of the bacterial strain and the DNA content of the fly. Higher values indicate a greater bacterial load. The bacteria measured here is L.bevis; a commensal gut bacteria commonly found in the GI tract of D.melanogaster.","Mean bacterial load, calculated from the DNA content of the bacterial strain and the DNA content of the fly. Higher values indicate a greater bacterial load. The bacteria measured here is L.plantarum; a commensal gut bacteria commonly found in the GI tract of D.melanogaster.","Mean quantity of lactate per adult fly. Higher values indicate a higher lactate content.","Mean quantity of lactate per adult fly. Higher values indicate a higher lactate content.","Mean relative lactate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean quantity of lactate per third instar larvae. Higher values indicate a higher lactate content.","Mean relative lactate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative lactose concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative lactose concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Larval size in mm 7 days post-embryonic development in protein poor conditions (lacking yeast). Larvae were purged of a microbiome and left to develop. Larger values indicate longer larvae.","Larval size in mm 7 days post-embryonic development in protein poor conditions (lacking yeast). Larvae were purged of a microbiome and subsequently associated with the commensal gut bacteria Lactobacillus plantarum. Larger values indicate longer larvae.","Mean difference in egg-to-adult viability between flies reared on control food medium and those reared on food medium supplemented with 0.5 mM lead acetate. Higher values indicate a smaller negative effect of lead on viability.","Mean proportion of flies surviving to adulthood, when reared at 25C on food medium supplemented with 0.5 mM lead acetate. Higher values indicate greater survival.","Mean relative leucine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative leucine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Incidence of DGRP males licking an interacting male, sourced from a wildtype laboratory strain. Higher values indicate a greater incidence of licking.","Mean days survived, when cohabiting a food vial with two males from the same DGRP line as the female. Measured under fluctuating temperature controlled conditions, with a mean of 16.8C ± 6. Higher values indicate greater longevity. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Mean days survived, when cohabiting a food vial with two males from the same DGRP line as the female. Measured under fluctuating temperature controlled conditions, with a mean of 23.8C ± 6. Higher values indicate greater longevity. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Mean number of days at which 50% of 20 adult flies had died in a vial filled with standard medium. Higher values indicate longer lived flies.","Mean relative linoleic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative linoleic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative linolenic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative linolenic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative lysine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative lysine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative malondialdehyde concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative malondialdehyde concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative margaric acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative margaric acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean mass of 4–6 females weighed together in milligrams. Higher values indicate heavier females.","Mean mass (mg) of 10 virgin flies. Higher values indicate heavier flies.","Mean mass (mg) of 10 virgin flies. Higher values indicate heavier flies.","Mean mass (mg). Higher values indicate heavier flies.","Mean mass (mg). Higher values indicate heavier flies.","Mean mass (mg) following forced exercise via rotational stimulation for 2h a day for 5 days. Higher values indicate heavier flies.","Mean mass (mg) following forced exercise via rotational stimulation for 2h a day for 5 days. Higher values indicate heavier flies.","Mean time from the start of cohabitation until a male from the Canton S line successfully initiated copulation with a DGRP female (unclear whether these flies were virgin). Higher values indicate a longer mating latency.","Mean time from the start of cohabitation until a male from a DGRP line successfully initiated copulation with a Canton S female (unclear whether these flies were virgin). Higher values indicate a longer mating latency.","Methylmercury (MeHg) is a persistent environmental toxin present in seafood that can compromise the developing nervous system in humans. To assess overall developmental tolerance on MeHg media, the authors calculate an eclosion index – higher values indicate greater relative egg-to-adult viability when reared on MeHg supplemented media.","Mean preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, over unsupplemented sucrose solution. Preference was measured on the first of three days of exposure. Higher values indicate a greater preference for methamphetamine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, over unsupplemented sucrose solution. Preference was measured on the first of three days of exposure. Higher values indicate a greater preference for methamphetamine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, over unsupplemented sucrose solution. Preference was measured on the third of three days of exposure. Higher values indicate a greater preference for methamphetamine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, over unsupplemented sucrose solution. Preference was measured on the third of three days of exposure. Higher values indicate a greater preference for methamphetamine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, over unsupplemented sucrose solution. Preference was measured on the first of three days of exposure and is scaled by the total amount of solution consumed during the assay. Higher values indicate a greater preference for methamphetamine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, over unsupplemented sucrose solution. Preference was measured on the first of three days of exposure and is scaled by the total amount of solution consumed during the assay. Higher values indicate a greater preference for methamphetamine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, over unsupplemented sucrose solution. Preference was measured on the third of three days of exposure and is scaled by the total amount of solution consumed during the assay. Higher values indicate a greater preference for methamphetamine supplemented solution.","Mean preferance for sucrose solution supplemented with 0.5 mg/ml methamphetamine, over unsupplemented sucrose solution. Preference was measured on the third of three days of exposure and is scaled by the total amount of solution consumed during the assay. Higher values indicate a greater preference for methamphetamine supplemented solution.","Mean relative methionine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative methionine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative methylmalonate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative methylmalonate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","The time elapsed from when eggs were placed into the vial until observation of the first pupal case at an 8 hour tempo resolution, at 20C. Higher values indicate a longer development time.","The time elapsed from when eggs were placed into the vial until observation of the first pupal case at an 8 hour tempo resolution, at 25C. Higher values indicate a longer development time.","The time elapsed from when eggs were placed into the vial until observation of the first pupal case at an 8 hour tempo resolution, at 30C. Higher values indicate a longer development time.","Mean copy number of mitochondrial genomes carried by adults, relative to a standard nuclear reference genome of equal size. Measured using whole genome shotgun sequencing, which is not tissue specific. Higher values indicate more mitochondrial genomes, which the authors suggest is a proxy for mitochondrial density.","% of male brains dissected that had abnormally formed alpha lobes in their mushroom bodies. Higher values indicate a greater % of abnormally formed lobes.","% of male brains dissected that had abnormally formed beta lobes in their mushroom bodies. Higher values indicate a greater % of abnormally formed lobes.","% of male brains dissected that had fused beta lobes in their mushroom bodies. Higher values indicate a greater % of fused lobes.","% of male brains dissected that had missing alpha lobes in their mushroom bodies. Higher values indicate a greater % of missing lobes.","% of male brains dissected that had missing beta lobes in their mushroom bodies. Higher values indicate a greater % of missing lobes.","% of male brains dissected that had short alpha lobes in their mushroom bodies. Higher values indicate a greater % of short lobes.","% of male brains dissected that had short beta lobes in their mushroom bodies. Higher values indicate a greater % of short lobes.","% of male brains dissected that had thin alpha lobes in their mushroom bodies. Higher values indicate a greater % of thin lobes.","% of male brains dissected that had thin beta lobes in their mushroom bodies. Higher values indicate a greater % of thin lobes.","Mean relative myristic acid concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative myristic acid concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative N-AcetylGlycine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative N-AcetylGlycine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean number of eggs laid over a female's entire lifetime, when cohabiting a food vial with two males from the same DGRP line as the female. Calculated as R0= Σ(lxmx), where lx = the proportion of females surviving each day and mx = the number of eggs laid each day. Higher values indicate a higher net reproductive rate. This trait was measured under fluctuating temperature controlled conditions, with a mean of 16.8C ± 6. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Mean number of eggs laid over a female's entire lifetime, when cohabiting a food vial with two males from the same DGRP line as the female. Calculated as R0= Σ(lxmx), where lx = the proportion of females surviving each day and mx = the number of eggs laid each day. Higher values indicate a higher net reproductive rate. This trait was measured under fluctuating temperature controlled conditions, with a mean of 23.8C ± 6. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Incidence of DGRP males orientating towards an interacting male, sourced from a wildtype laboratory strain. Higher values indicate a greater incidence of orientating.","Mean relative ornithine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative ornithine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative orotate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative orotate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative oxaloacetate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative oxaloacetate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative oxypurinol concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative oxypurinol concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean difference in larval viability between control and p-xylene-treated (concentration 1.068mM) first instar larvae after 14 days of development. P-xylene is an aromatic hydrocarbon associated with gasoline and diesel-powered engine emissions. Higher values indicate a greater negative effect of P-xylene on larval viability.","Mean relative pantothenate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative pantothenate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative PEP concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative PEP concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative phenylalanine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative phenylalanine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean change in phenylalanine diluted food medium preference after deprivation of amino acids compared with flies that had experienced no deprivation of amino acids. Positive values indicate a stronger change in preference towards the phenylalanine supplemented medium after deprivation.","Mean change in phenylalanine diluted food medium preference after deprivation of amino acids compared with flies that had experienced no deprivation of amino acids. Positive values indicate a stronger change in preference towards the phenylalanine supplemented medium after deprivation.","Mean phototaxis score measured in 7 day old adults, without a light source. This measure was taken to approximate the effect of fly activity on the phototaxis response. Higher values indicate more active flies.","Mean phototaxis score measured in 7 day old adults, without a light source. This measure was taken to approximate the effect of fly activity on the phototaxis response. Higher values indicate more active flies.","Mean relative pipecolate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative pipecolate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Lambda. Population growth rate or intrinsic rate of increase. Defined as e^r, where r = ln(net reproductive rate) / generation time. Higher values indicate faster population growth rates. This trait was measured under fluctuating temperature controlled conditions, with a mean of 16.8C ± 6. Populations consisted of one female and two males from the same DGRP line. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Lambda. Population growth rate or intrinsic rate of increase. Defined as e^r, where r = ln(net reproductive rate) / generation time. Higher values indicate faster population growth rates. This trait was measured under fluctuating temperature controlled conditions, with a mean of 23.8C ± 6. Populations consisted of one female and two males from the same DGRP line. Note that this trait should be treated as a characteristic of a population, rather than that of an individual genotype.","Mean number of progeny produced in 5 vials over 10 days by twice-mated females.","P1 is the proportion of progeny sired by the first male to mate with a twice-mated female. In this study, a DGRP female mated with two standard males. Hence, in this study, the variation in P1 is attributed to the female genotype. Higher values indicate a greater proportion of offspring sired by the male that mated first.","Mean relative proline concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative proline concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean quantity of protein per adult fly. Higher values indicate a higher protein content.","Mean quantity of protein per adult fly. Higher values indicate a higher protein content.","Mean protein (mg/ul) extracted from 10 virgin flies. Higher values indicate greater levels of protein.","Mean protein (mg/ul) extracted from 10 virgin flies. Higher values indicate greater levels of protein.","Mean abdomen protein content (log-transformed data is originally reported), measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate flies with a greater glucose content.","Mean abdomen protein content (log-transformed data is originally reported), following a three-day exercise regime. Higher values indicate flies with a greater glucose content.","Mean quantity of protein per third instar larvae. Higher values indicate a higher protein content.","Mean thorax protein content (log-transformed data is originally reported), measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate flies with a greater glucose content.","Mean thorax protein content (log-transformed data is originally reported), following a three-day exercise regime. Higher values indicate flies with a greater glucose content.","Mean pupal case length in mm. Measurements were taken with image analysis software from at least 6 vials with a minimum of 15 pupa per vial, and were converted from pixels to mm based on the ratio to the diameter measurement of 1 euro cent coin (16.25 mm diameter)","Average female pupal weight, reared at 20C. Higher vales indicate heavier pupa.","Average female pupal weight, reared at 25C. Higher vales indicate heavier pupa.","Average female pupal weight, reared at 30C. Higher vales indicate heavier pupa.","Mean height at which larvae pupated in a 95mm Drosophila vial, after accounting for food height. Higher values indicate higher pupation sites.","Mean relative putrescine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative putrescine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative pyridoxal-5-P concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative pyridoxal-5-P concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Larval to adult survival on food media supplemented with 4PPM Pyripole – an insecticide from the phenylpyrazole family, released for commercial use in 2007. Higher values indicate greater resistance.","Mean relative pyruvate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative pyruvate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature before exposing them to a cold temperature, while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared under fluctuating controlled temperatures, with a mean of 16.8C ± 6, prior to phenotyping. Higher values indicate a greater plastic response.","Rapid cold hardening; a type of phenotypic plasticity that allows ectotherms to quickly enhance cold tolerance in response to brief chilling (lasting minutes to hours). This was measured by exposing one group of flies to a mild pre-treatment temperature before exposing them to a cold temperature, while another group of flies was immediately exposed to the cold temperature. RCH = prop survival after hardening – prop survival without hardening. These flies were reared under fluctuating controlled temperatures, with a mean of 23.8C ± 6, prior to phenotyping. Higher values indicate a greater plastic response.","Mean Reactive oxygen species (ROS) levels in the guts of flies. Higher values indicate greater levels of ROS.","Mean proportion of females that remated with a second male (genotype bwD), ~ 1 day after first mating with a male (genotype cn bw).","Mean relative ribose-5-P concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative ribose-5-P concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative sarcosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative sarcosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative serine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative serine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Incidence of courtship ‘song’ produced by a DGRP male when interacting with another male, sourced from a wildtype laboratory strain. Higher values indicate a greater incidence of singing.","Mean nearest neighbour distances for a group of 25 flies, after one hour of acclimation in the environment. Higher values indicate greater spacing between individuals.","Mean nearest neighbour distances for a group of 25 flies, after one hour of acclimation in the environment. Higher values indicate greater spacing between individuals.","Mean relative sorbitol concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative sorbitol concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean time in seconds that an adult fly, grown on standard food, remained mobile across a 60 second period following mechanical disturbance. Higher values indicate a greater startle response.","Mean time in seconds that an adult fly, grown on standard food, remained mobile across a 60 second period following mechanical disturbance. Higher values indicate a greater startle response.","Mean time in seconds that an adult fly, grown on 10% EtOH (ethanol) supplemented food, remained mobile across a 60 second period following mechanical disturbance. Higher values indicate a greater startle response.","Mean time in seconds that an adult fly, grown on 10% EtOH (ethanol) supplemented food, remained mobile across a 60 second period following mechanical disturbance. Higher values indicate a greater startle response.","Mean hours survived for larvae subjected to nutrient poor starvation medium (20% sucrose in PBS solution). Higher values indicate greater starvation resistance.","Mean relative succinate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative succinate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative sucrose concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative sucrose concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Incidence of a male being tapped by an interacting male, sourced from a wildtype laboratory strain. Higher values indicate a greater incidence of being tapped.","Male tapping; in this case when a male brings the tarsus of a foreleg into contact with another male. Higher values indicate a greater incidence of tapping.","Mean relative taurine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative taurine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean thorax length (mm). The authors use this as a proxy for body size. Higher values indicate a longer thorax.","Mean thorax length (mm). The authors use this as a proxy for body size. Higher values indicate a longer thorax.","Mean thorax width (mm). The authors use this as a proxy for body size. Higher values indicate a wider thorax.","Mean thorax width (mm). The authors use this as a proxy for body size. Higher values indicate a wider thorax.","Mean relative threonine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative threonine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","The mean percentage of flies that died within 24 hours, after suffering traumatic injury. Higher values indicate greater mortality.","Mean quantity of triacylglycerol per adult fly. Higher values indicate a higher triacylglycerol content.","Mean quantity of triacylglycerol per adult fly. Higher values indicate a higher triacylglycerol content.","Mean quantity of triacylglycerol per third instar larvae. Higher values indicate a higher triacylglycerol content.","Mean triglyceride (mg/ul) extracted from 10 virgin flies. Higher values indicate greater levels of triglyceride.","Mean triglyceride (mg/ul) extracted from 10 virgin flies. Higher values indicate greater levels of triglyceride.","Mean abdomen triglyceride content (log-transformed data is originally reported), measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate flies with a greater glucose content.","Mean abdomen triglyceride content (log-transformed data is originally reported), following a three-day exercise regime. Higher values indicate flies with a greater glucose content.","Mean thorax triglyceride content (log-transformed data is originally reported), measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate flies with a greater glucose content.","Mean thorax triglyceride content (log-transformed data is originally reported), following a three-day exercise regime. Higher values indicate flies with a greater glucose content.","Mean relative trimethylamine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative trimethylamine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative tryptophan concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative tryptophan concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative tyrosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative tyrosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Proportion of unfertilised eggs produced by 3-4 flies that had been allowed to mate for at least two days beforehand. Note that flies were kept in population cages that contained both sexes from the same line during the mating period. This means that genetic variance in this trait variation could be explained by either the male or female genotype. Higher values indicate a greater proportion of unfertilised eggs.","Mean relative uracil concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative uracil concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative urate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative urate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative uridine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative uridine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative valine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative valine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean wet weight of flies, measured in flies that exhibited normal movement (control) during the exercise period. Higher values indicate heavier flies.","Mean wet weight of flies, following a three day exercise regime. Higher values indicate heavier flies.","Wing imaginal disk size in third instar larvae – measured as absolute centroid size. Larger values indicate larger wing disks.","Wing imaginal disk size in third instar larvae – measured as absolute centroid size. Larger values indicate larger wing disks.","Wing imaginal disk size in third instar larvae – measured as centroid size after controlling for body size. Larger values indicate larger wing disks relative to body size.","Wing imaginal disk size in third instar larvae – measured as centroid size after controlling for body size. Larger values indicate larger wing disks relative to body size.","Mean copy number of Wolbachia genomes carried by adults, relative to a standard nuclear reference genome of equal size. Measured using whole genome shotgun sequencing, which is not tissue specific. Higher values indicate more Wolbachia genomes, which the authors suggest is a proxy for Wolbachia density. Note that only lines found to harbour Wolbachia are inlcuded in this dataset.","Mean relative xanthosine concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative xanthosine concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative xanthurenate concentration extracted from 3 flies that lived as adults on a diet containing 5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean relative xanthurenate concentration extracted from 3 flies that lived as adults on a diet containing 0.5% yeast. Higher values indicate greater levels of the metabolite. Note these flies come from the same populations as used by Wilson et al (2020).","Mean mitochondrial state 3 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 3 respiration occurs when protons re-enter the mitochondrial matrix in the presence of ADP, which results in the production of ATP. Higher values indicate higher respiration rates.","Mean mitochondrial state 3 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 3 respiration occurs when protons re-enter the mitochondrial matrix in the presence of ADP, which results in the production of ATP. Higher values indicate higher respiration rates.","Mean mitochondrial state 4 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 4 respiration occurs when protons leak into the mitochondrial matrix in the absence of ADP, which is required for ATP synthase. Higher values indicate higher respiration rates.","Mean mitochondrial state 4 respiration rate, measured as oxygen consumption (pmol/s* mg of protein). Mitochondria were sourced from the thorax and respiration was measured in vitro. State 4 respiration occurs when protons leak into the mitochondrial matrix in the absence of ADP, which is required for ATP synthase. Higher values indicate higher respiration rates.","Mean phosphate/oxygen ratio (P:O ratio) - a measure of the efficiency with which mitochondria convert oxygen into ATP. The ratio was here calculated as the amount of ADP consumed per oxygen being reduced during state 3 respriation. Mitochondria were sourced from the thorax and respiration was measured in vitro. Higher values indicate more efficient ATP production.","Mean phosphate/oxygen ratio (P:O ratio) - a measure of the efficiency with which mitochondria convert oxygen into ATP. The ratio was here calculated as the amount of ADP consumed per oxygen being reduced during state 3 respriation. Mitochondria were sourced from the thorax and respiration was measured in vitro. Higher values indicate more efficient ATP production.","Mean combined glycogen content extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher glycogen content.","Mean combined glycogen content extracted from 5 adult flies that possessed a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate higher glycogen content.","Mean change in the combined glycogen content of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher glycogen content. Note that the presence of micriobiota on average increased glycogen content.","Mean change in the combined protein content of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher protein content. Note that the presence of micriobiota on average increased protein content.","Mean combined triglyceride (TAG) content after glycerol substraction extracted from 5 adult flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared on sterile food. Higher values indicate higher TAG content.","Mean change in combined triglyceride content after glycerol substraction of 5 adult flies that possessed a microbiome compared with flies that lacked a microbiome, after controlling for dry weight. All micro-organisms were purged from egg-surfaces via dechorionation and larvae were reared either on sterile medium or on medium supplemented with Acetobacter pomorum, Acetobacter tropicalis, Lactobacillus brevis, Lactobacillus fructivorans and Lactobacillus plantarum. All are common commensal bacteria that make up a large proportion of the D. melanogaster microbiome. Higher values indicate that the addition of microbiota resulted in a higher TAG content. Note though that the presence of micriobiota on average decreased TAG content.","Mean aversion towards an odour flies were trained to associate with the onset of an electric shock. Higher values indicate that flies showed a greater aversion toward the odour, suggesting that they associated the odor with 'punishment'. Higher values also indicate a higher learning score","Mean aversion towards an odour flies were trained to associate with the onset of an electric shock. Higher values indicate that flies showed a greater aversion toward the odour, suggesting that they associated the odor with 'punishment'. Higher values also indicate a greater learning score.","Mean preference for an odor flies were trained to associate with the end of an electric shock. Higher values indicate that flies showed a stronger preference towards the odor, suggesting that they associated the odor with 'relief'. Higher values also indicate a greater learning score.","Mean preference for an odor flies were trained to associate with the end of an electric shock. Higher values indicate that flies showed a stronger preference towards the odor, suggesting that they associated the odor with 'relief'. Higher values also indicate a greater learning score.","Mean centroid size - the square root of the sum of squared distances of a set of wing landmarks to the centroid of this landmark configuration. Higher values indicate larger wings.","Mean centroid size - the square root of the sum of squared distances of a set of wing landmarks to the centroid of this landmark configuration. Higher values indicate larger wings.","Mean time (mins) from a male flies introduction to a female until the onset of mating. Lower values indicate shorter copulation latencies and potentially more attractive males","Mean proportion of 10 females that a single male successfully inseminated in 48 hours. Insemination was verified by identification of viable pupal progeny. Higher values indicate males with a greater insemination capacity.","Mean proportion of surviving adults 24 hours after an hour of exposure to -1 °C. Higher values indicate flies that are more tolerant to cold exposure.","Mean proportion of surviving adults 24 hours after an hour of exposure to -2 °C. Higher values indicate flies that are more tolerant to cold exposure.","Mean proportion of surviving adults 24 hours after an hour of exposure to -3 °C. Higher values indicate flies that are more tolerant to cold exposure.","Mean proportion of surviving adults 24 hours after an hour of exposure to -4 °C. Higher values indicate flies that are more tolerant to cold exposure.","Mean proportion of surviving adults 24 hours after an hour of exposure to -5 °C. Higher values indicate flies that are more tolerant to cold exposure.","Mean proportion of surviving adults 24 hours after an hour of exposure to -6 °C. Higher values indicate flies that are more tolerant to cold exposure.","Mean proportion of surviving adults 24 hours after an hour of exposure to -7 °C. Higher values indicate flies that are more tolerant to cold exposure.","Mean proportion of third instar larvae that survived exposure to -6.5°C for an hour. These larvae were otherwise reared at 18°C. Higher values indicate greater cold hardiness.","Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater cold hardiness.","Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater cold hardiness.","Mean proportion of third instar larvae that survived exposure to -6.5°C for an hour. These larvae were otherwise reared at 25°C. Higher values indicate greater cold hardiness.","Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater cold hardiness.","Mean proportion of adult flies that survived exposure to -6.5°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater cold hardiness.","Mean proportion of third instar larvae that survived exposure to 38°C for an hour. These larvae were otherwise reared at 18°C. Higher values indicate greater heat hardiness.","Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater heat hardiness.","Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 18°C. Higher values indicate greater heat hardiness.","Mean proportion of third instar larvae that survived exposure to 38°C for an hour. These larvae were otherwise reared at 25°C. Higher values indicate greater heat hardiness.","Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater heat hardiness.","Mean proportion of adult flies that survived exposure to 38°C for an hour. These flies were originally reared at 25°C. Higher values indicate greater heat hardiness.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Mean reads per million (originally Log2 normalised) values for reads from each DGRP RNA-seq sample uniquely aligning to each microbial species. Higher values indicate a greater load of the microbial species.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Metabolites identified by mass spectrometry with scaled measurement for each sample (100 flies, replicated three times). Data were scaled to account for run-day blocks. Higher values indicate greater quantities.","Mean proportion of females that produced viable larvae. Females were reared at 25C. Higher values indicates greater fertility.","Mean proportion of females that produced viable larvae. Females were reared at 27C. Higher values indicates greater fertility","Mean proportion of females that produced viable larvae. Females were reared at 29C. Higher values indicates greater fertility","Mean slope of the change in female fertility (the proportion of males that produced viable larvae), between 25C, 27C and 29C rearing conditions. Lower slope values indicate a greater decrease in fertility with an increase in temperature. This is a measure of thermal susceptibility.","Mean number of aggressive lunges made in 20 min window following removal of a divider, by pairs of DGRP males of same genotype. Trials were run after males were kept isolated for five days. Higher values indicate more aggressive males.","Mean climbing index, following five days of standard conditions. Higher values indicate greater climbing ability.","Mean change in the climbing ability of adults, between flies that were and were not exposed to exercise five days previously. Positive values indicate an increase in climbing ability with exercise.","Mean climbing index, following five days consecutive days of forced exercise (two hours a day). Higher values indicate greater climbing ability.","Mean climbing index, following five days of standard conditions. Higher values indicate greater climbing ability.","Mean change in the climbing ability of adults, between flies that were and were not exposed to exercise five days previously. Positive values indicate an increase in climbing ability with exercise.","Mean climbing index, following five days consecutive days of forced exercise (two hours a day). Higher values indicate greater climbing ability."]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Trait<\/th>\n      <th>Reference<\/th>\n      <th># lines measured<\/th>\n      <th>Measured in<\/th>\n      <th>Included<\/th>\n      <th>Trait description<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"autoWidth":true,"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":1000,"scrollCollapse":true,"buttons":["pageLength","colvis","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Trait_data"}],"pageLength":2115,"columnDefs":[{"className":"dt-right","targets":2}],"order":[],"orderClasses":false,"lengthMenu":[10,25,50,100,2115]},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<p><span class="math inline">\(~\)</span></p>
<p>Find the number of traits and studies included in our analysis.</p>
<pre class="r"><code># how many studies and traits do we have after filtering?

num_unique_traits &lt;- table_data %&gt;% filter(Included == &quot;Yes&quot;) %&gt;% nrow()
# in females
num_unique_traits_f &lt;- table_data %&gt;% filter(Included == &quot;Yes&quot; &amp; `Measured in` == &quot;Females&quot;) %&gt;% nrow()
# in males
num_unique_traits_m &lt;- table_data %&gt;% filter(Included == &quot;Yes&quot; &amp; `Measured in` == &quot;Males&quot;) %&gt;% nrow()
# how many studies are they measured across in total?
num_unique_studies &lt;- table_data %&gt;% filter(Included == &quot;Yes&quot;) %&gt;%  distinct(Reference) %&gt;% nrow()
# in females
num_unique_studies_f &lt;- table_data %&gt;% filter(Included == &quot;Yes&quot; &amp; `Measured in` == &quot;Females&quot;) %&gt;% distinct(Reference) %&gt;% nrow()
# in males
num_unique_studies_m &lt;- table_data %&gt;% filter(Included == &quot;Yes&quot; &amp; `Measured in` == &quot;Males&quot;) %&gt;% distinct(Reference) %&gt;% nrow()</code></pre>
<p>After this selection process, 474 remain, that were measured across
76 studies. There are 232 measured in females and 242 in males across 56
and 54 respectively.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="calculating-r-the-response-to-selection"
class="section level1">
<h1>Calculating <em>R</em>: the response to selection</h1>
<p><span class="math inline">\(~\)</span></p>
<p>We assume that line means approximate the breeding value for a trait
for that given line, and that the variance across lines equals the
genetic variance in the DGRP population for that trait.</p>
<p>We can then estimate the response to selection (<span
class="math inline">\(R\)</span>) for a trait using <strong>Robertson’s
Secondary Theorem of Natural Selection</strong> (also known as the
<strong>Robertson covariance</strong>; Robertson, 1968), which states
that <span class="math inline">\(R\)</span> is equivalent to the
covariance between a traits breeding values (<span
class="math inline">\(A_z\)</span>) and the breeding values for relative
fitness (<span class="math inline">\(A_w\)</span>):</p>
<p><span class="math display">\[R = \sigma(A_w, Az)\]</span> The
secondary theorem typically does not partition <span
class="math inline">\(R\)</span> between the sexes. However, because we
have <span class="math inline">\(A_w\)</span> values in both sexes and
often <span class="math inline">\(A_z\)</span> values for each sex, we
calculate four versions of <span class="math inline">\(R\)</span>:</p>
<p>For traits expressed in females:</p>
<p><span class="math display">\[R_{FF} = \sigma(A_w^F,
A_z^F)\]</span></p>
<p>and</p>
<p><span class="math display">\[R_{MF} = \sigma(A_w^M, A_z^F)\]</span>
For traits expressed in males:</p>
<p><span class="math display">\[R_{MM} = \sigma(A_w^M,
A_z^M)\]</span></p>
<p>and</p>
<p><span class="math display">\[R_{FM} = \sigma(A_w^F,
A_z^M)\]</span></p>
<p>where the first subscripted letter indicates which sex breeding
values for fitness were measured in.</p>
<p><span class="math inline">\(R_{MF}\)</span> and <span
class="math inline">\(R_{FM}\)</span> indicate the expected response
when a trait is measured in one sex and fitness is measured in the
other. We calculate these versions of <span
class="math inline">\(R\)</span> because a trait can respond to
selection acting on the other sex even if it is not expressed, because
the phenotype is a multivariate amalgamation of traits and genetic
correlations between traits are likely abundant. For example, alleles
encoding a female-specific CHC may also effect male desiccation
resistance, which could be under selection in males.</p>
<p><span class="math inline">\(~\)</span></p>
<div id="build-models-to-calculate-r_ff-r_mf" class="section level2">
<h2>Build models to calculate <span
class="math inline">\(R_{FF}\)</span> &amp; <span
class="math inline">\(R_{MF}\)</span></h2>
<p>To estimate the covariance between <span
class="math inline">\(A_w\)</span> and <span
class="math inline">\(A_z\)</span>, we fitted bivariate Bayesian linear
models using the <code>brms</code> package (Bürkner, 2017) for
<code>R version 4.2.2</code>. For each combination of trait and sex, we
used line means for the focal trait and the fitness of the focal sex as
the two response variables and fitted an intercept-only Gaussian model.
Each model returned a posterior distribution of the residual correlation
between trait and fitness, which for data expressed in standard units is
equivalent to the covariance.</p>
<p>Build functions to run the models</p>
<pre class="r"><code># RFF estimates

female_traits &lt;- trait_data %&gt;% filter(Sex == &quot;Female&quot;)

trait_list_female &lt;- unique(female_traits$Trait) # an input to the map_dfr() function that we&#39;ll need in a few chunks time

# code the model structure we will use for all traits using one example - `flight.performance.f`. We can then use the update() function to run this model many times, once for each trait measured in females. update() makes this process many times faster, because the model can immediately start sampling, without the need to recompile. 

RFF_model &lt;-
  brm(data = female_traits %&gt;% filter(Trait == &quot;flight.performance.f&quot;),
      family = gaussian,
      bf(mvbind(female_fitness, trait_value) ~ 1) + set_rescor(TRUE),
      prior = c(prior(normal(0, 0.1), class = Intercept, resp = femalefitness),
                prior(normal(0, 0.1), class = Intercept, resp = traitvalue),
                prior(normal(1, 0.1), class = sigma, resp = femalefitness),
                prior(normal(1, 0.1), class = sigma, resp = traitvalue),
                prior(lkj(2), class = rescor)),
      chains = 4, cores = 4, iter = 6000, warmup = 2000,
      seed = 1, file = &quot;fits/RFF_test_model&quot;)   

# make a function to update the model and the posterior sample output with the &#39;selected trait&#39;

RFF_calculator &lt;- function(selected_trait){
  
  data &lt;- female_traits %&gt;% filter(Trait == selected_trait)
  
  model &lt;- update(
    RFF_model, newdata = data,
      chains = 4, cores = 4, iter = 6000, warmup = 2000,
      seed = 1)
  
  posterior &lt;- 
    as_draws_df(model) %&gt;% 
    rename(Response_to_selection_female = rescor__femalefitness__traitvalue) %&gt;% 
    mutate(Trait = selected_trait) %&gt;% 
    select(Trait, Response_to_selection_female) %&gt;% 
    as_tibble()
  
  posterior
}

# RMF estimates

RMF_model &lt;-
  brm(data = female_traits %&gt;% filter(Trait == &quot;flight.performance.f&quot;),
      family = gaussian,
      bf(mvbind(male_fitness, trait_value) ~ 1) + set_rescor(TRUE),
      prior = c(prior(normal(0, 0.1), class = Intercept, resp = malefitness),
                prior(normal(0, 0.1), class = Intercept, resp = traitvalue),
                prior(normal(1, 0.1), class = sigma, resp = malefitness),
                prior(normal(1, 0.1), class = sigma, resp = traitvalue),
                prior(lkj(2), class = rescor)),
      chains = 4, cores = 4, iter = 6000, warmup = 2000,
      seed = 1, file = &quot;fits/RMF_test_model&quot;)

# make a function to update the model and the posterior sample output with your desired trait

RMF_calculator &lt;- function(selected_trait){
  
  data &lt;- female_traits %&gt;% filter(Trait == selected_trait)
  
  model &lt;- update(
    RMF_model, newdata = data,
    chains = 4, cores = 4, iter = 6000, warmup = 2000,
    seed = 1)
  
  posterior &lt;- 
    as_draws_df(model) %&gt;% 
    rename(Response_to_selection_male = rescor__malefitness__traitvalue) %&gt;% 
    mutate(Trait = selected_trait) %&gt;% 
    select(Trait, Response_to_selection_male) %&gt;% 
    as_tibble()
  
  posterior
}</code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="build-models-to-calculate-r_fm-r_mf" class="section level2">
<h2>Build models to calculate <span
class="math inline">\(R_{FM}\)</span> &amp; <span
class="math inline">\(R_{MF}\)</span></h2>
<p><span class="math inline">\(~\)</span></p>
<pre class="r"><code># RMM estimates

male_traits &lt;- trait_data %&gt;% filter(Sex == &quot;Male&quot;)

trait_list_male &lt;- unique(male_traits$Trait)

RMM_model &lt;-
  brm(data = male_traits %&gt;% filter(Trait == &quot;flight.performance.m&quot;),
      family = gaussian,
      bf(mvbind(male_fitness, trait_value) ~ 1) + set_rescor(TRUE),
      prior = c(prior(normal(0, 0.1), class = Intercept, resp = malefitness),
                prior(normal(0, 0.1), class = Intercept, resp = traitvalue),
                prior(normal(1, 0.1), class = sigma, resp = malefitness),
                prior(normal(1, 0.1), class = sigma, resp = traitvalue),
                prior(lkj(2), class = rescor)),
      chains = 4, cores = 4, iter = 6000, warmup = 2000,
      seed = 1, file = &quot;fits/RMM_test_model&quot;)   

# make a function to update the model and the posterior sample output with your desired trait

RMM_calculator &lt;- function(selected_trait){
  
  data &lt;- male_traits %&gt;% filter(Trait == selected_trait)
  
  model &lt;- update(
    RMM_model, newdata = data,
    chains = 4, cores = 4, iter = 6000, warmup = 2000,
    seed = 1)
  
  posterior &lt;- 
    as_draws_df(model) %&gt;% 
    rename(Response_to_selection_male = rescor__malefitness__traitvalue) %&gt;%
    mutate(Trait = selected_trait) %&gt;% 
    select(Trait, Response_to_selection_male) %&gt;% 
    as_tibble()
  
  posterior
}

# RFM estimates

RFM_model &lt;-
  brm(data = male_traits %&gt;% filter(Trait == &quot;flight.performance.m&quot;),
      family = gaussian,
      bf(mvbind(female_fitness, trait_value) ~ 1) + set_rescor(TRUE),
      prior = c(prior(normal(0, 0.1), class = Intercept, resp = femalefitness),
                prior(normal(0, 0.1), class = Intercept, resp = traitvalue),
                prior(normal(1, 0.1), class = sigma, resp = femalefitness),
                prior(normal(1, 0.1), class = sigma, resp = traitvalue),
                prior(lkj(2), class = rescor)),
      chains = 4, cores = 4, iter = 6000, warmup = 2000,
      seed = 1, file = &quot;fits/RFM_test_model&quot;)   

# make a function to update the model and the posterior sample output with your desired trait

RFM_calculator &lt;- function(selected_trait){
  
  data &lt;- male_traits %&gt;% filter(Trait == selected_trait)
  
  model &lt;- update(
    RFM_model, newdata = data,
      chains = 4, cores = 4, iter = 6000, warmup = 2000,
      seed = 1)
  
  posterior &lt;- 
    as_draws_df(model) %&gt;% 
    rename(Response_to_selection_female = rescor__femalefitness__traitvalue) %&gt;% 
    mutate(Trait = selected_trait) %&gt;% 
    select(Trait, Response_to_selection_female) %&gt;% 
    as_tibble()
  
  posterior
}</code></pre>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Correlation ~ Covariance</strong></p>
<p>The <code>brms</code> model computes the posterior distribution of
the residual correlation between standardised trait and fitness breeding
values. Standardising traits to have standard deviation of 1 makes
correlation equal to covariance. For our own sanity, we plot the
residual correlation against covariance and show that, outside a small
amount of noise at the extremes, correlation ~ covariance. We calculated
point estimates of these variables for each trait measured in females
using the <code>cor</code> and <code>cov</code> functions.</p>
<pre class="r"><code>cor_cov_calculator &lt;- function(selected_trait){
  
  data &lt;- trait_data %&gt;% filter(Trait == selected_trait,
                                             !is.na(female_fitness),
                                             !is.na(trait_value))
  
  output &lt;- 
    bind_cols(
      cor(data$female_fitness, data$trait_value) %&gt;% 
        as_tibble() %&gt;%
        rename(cor = value),
      
      
      cov(data$female_fitness, data$trait_value) %&gt;% 
        as_tibble() %&gt;%
        rename(cov = value)) %&gt;% 
    mutate(Trait = selected_trait)
  
  output
}

cor_cov &lt;- map_dfr(trait_list_female, cor_cov_calculator)

cor_cov %&gt;% 
  ggplot(aes(cor, cov)) +
  geom_point(size = 2) +
  labs(x = &quot;Trait correlation with fitness&quot;, y = &quot;Trait covariance with fitness&quot;) +
  theme_bw() + 
  theme(text = element_text(size = 14))</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/319c84c1118d4c1c59279734e1b3432ef1d08901/docs/figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">319c84c</a>
</td>
<td>
tomkeaney
</td>
<td>
2023-10-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/daa935de0473e2ecaedca3c65f414d2858ddc598/docs/figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">daa935d</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/b682a4fd7268268f71172d62d1c5978ef2c39b5c/docs/figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">b682a4f</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/7bf0ee08d2c8b770d1c2d8aca0b2931897462adb/docs/figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">7bf0ee0</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/769e48d5a4afe34bb3a05faaa53829aeaadcfc05/docs/figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">769e48d</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/83a0c4fdf0c246cc8f853dbeecd4ab8b252df66e/docs/figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">83a0c4f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/11a83906f387d1ecbb39fc5ab60f5830bdac2d8f/docs/figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">11a8390</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="run-the-models-for-all-the-traits" class="section level2">
<h2>Run the models for all the traits</h2>
<p>Run the models using <code>RFF_calculator</code>,
<code>RMF_calculator</code>, <code>RMM_calculator</code> and
<code>RFM_calculator</code></p>
<pre class="r"><code># run the RFF function

Run_function &lt;- FALSE # Change this to TRUE to run the models

if(Run_function){
  RFF &lt;- map_dfr(trait_list_female, RFF_calculator) # map_dfr returns a data frame created by row-binding each output
  write_csv(RFF, file = &quot;data/RFF.csv&quot;)
} else RFF &lt;- read_csv(&quot;data/RFF.csv&quot;)

# run the RMF function

if(Run_function){
  RMF &lt;- map_dfr(trait_list_female, RMF_calculator) 
  write_csv(RMF, file = &quot;data/RMF.csv&quot;)
} else RMF &lt;- read_csv(&quot;data/RMF.csv&quot;)

# run the RMM function

if(Run_function){
  RMM &lt;- map_dfr(trait_list_male, RMM_calculator) 
  write_csv(RMM, file = &quot;data/RMM.csv&quot;)
} else RMM &lt;- read_csv(&quot;data/RMM.csv&quot;)

# run the RFM function

if(Run_function){
  RFM &lt;- map_dfr(trait_list_male, RFM_calculator) 
  write_csv(RFM, file = &quot;data/RFM.csv&quot;)
} else RFM &lt;- read_csv(&quot;data/RFM.csv&quot;)</code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
</div>
<div id="interpreting-r" class="section level1">
<h1>Interpreting R</h1>
<div id="calculate-various-metrics" class="section level2">
<h2>Calculate various metrics</h2>
<p><span class="math inline">\(~\)</span></p>
<p>Combine the data frames and estimate the overall expected response to
selection as</p>
<p><span class="math display">\[R_{overall}^F = \frac{R_{FF} +
R_{MF}}{2}\]</span> and</p>
<p><span class="math display">\[R_{overall}^M = \frac{R_{MM} +
R_{FM}}{2}\]</span></p>
<p>In the same code chunk, we also calculate the difference between the
sexes for the response to selection in each trait as</p>
<p><span class="math display">\[\Delta R^F = R_{FF} - R_{MF}\]</span>
and</p>
<p><span class="math display">\[\Delta R^M = R_{FM} -
R_{MM}\]</span></p>
<p>to test for potential intralocus sexual conflict. Note that this
measure does not reveal whether a trait is truly sexually antagonistic,
as a difference is possible when a selection response is sexually
concordant but with sex differences in strength. A difference can also
result from the absence of a response in one sex and a strong response
in the other.</p>
<pre class="r"><code>R_female_traits &lt;-
  bind_cols(RFF, RMF) %&gt;% 
  filter(Trait...1 == Trait...3) %&gt;% 
  rename(Trait = Trait...1) %&gt;% 
  mutate(Trait_Sex = &quot;Female&quot;,
         R_overall = (Response_to_selection_female + Response_to_selection_male)/2,
         R_diff = Response_to_selection_female - Response_to_selection_male) %&gt;% 
    select(Trait, Trait_Sex, everything(), -(Trait...3)) 

R_male_traits &lt;-
  bind_cols(RFM, RMM) %&gt;% 
  rename(Trait = Trait...1) %&gt;%  
  mutate(Trait_Sex = &quot;Male&quot;,
         R_overall = (Response_to_selection_female + Response_to_selection_male)/2,
         R_diff = Response_to_selection_female - Response_to_selection_male) %&gt;% 
  select(Trait, Trait_Sex, everything(), -(Trait...3)) %&gt;% 
  filter(Trait != &quot;dopamine.response.to.paraquat.2021.m&quot;)

R_all_traits &lt;- bind_rows(R_female_traits, R_male_traits)

R_long_form &lt;-
  R_all_traits %&gt;% 
  select(1:4) %&gt;% 
    pivot_longer(cols = 3:4, names_to = &quot;Fitness_Sex&quot;, values_to = &quot;R&quot;) %&gt;% 
    mutate(Fitness_Sex = case_when(Fitness_Sex == &quot;Response_to_selection_female&quot; ~ &quot;Female&quot;,
                                   Fitness_Sex == &quot;Response_to_selection_male&quot; ~ &quot;Male&quot;))</code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="visualise-the-responses-to-selection-for-each-trait"
class="section level2">
<h2>Visualise the responses to selection for each trait</h2>
<p><span class="math inline">\(~\)</span></p>
<pre class="r"><code>p1 &lt;-
  R_female_traits %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_R = median(Response_to_selection_female)) %&gt;%
  ggplot(aes(Response_to_selection_female, fct_reorder(Trait, avg_R))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 1, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Purp&quot;) +
  coord_cartesian(xlim = c(-0.5, 0.5)) +
  geom_vline(linetype = 2, xintercept = 0, linewidth = 1) +
  labs(x = &quot;Response to selection in females&quot;,
       y = &quot;Trait expressed in females&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        text = element_text(size=14),
        axis.text.y = element_text(size = 8))

p2 &lt;-
  R_female_traits %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_R = median(Response_to_selection_male)) %&gt;%
  ggplot(aes(Response_to_selection_male, fct_reorder(Trait, avg_R))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 1, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  coord_cartesian(xlim = c(-0.5, 0.5)) +
  geom_vline(linetype = 2, xintercept = 0, linewidth = 1) +
  labs(x = &quot;Response to selection in males&quot;,
       y = &quot;Trait expressed in females&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        text = element_text(size=14),
        axis.text.y = element_text(size = 8)) 


p1 + p2 +
  plot_annotation(tag_levels = &#39;a&#39;)</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-9-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/319c84c1118d4c1c59279734e1b3432ef1d08901/docs/figure/Main_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">319c84c</a>
</td>
<td>
tomkeaney
</td>
<td>
2023-10-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/daa935de0473e2ecaedca3c65f414d2858ddc598/docs/figure/Main_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">daa935d</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/b682a4fd7268268f71172d62d1c5978ef2c39b5c/docs/figure/Main_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">b682a4f</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/7bf0ee08d2c8b770d1c2d8aca0b2931897462adb/docs/figure/Main_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">7bf0ee0</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/043902fed2d66e5b988a2f7a2dd70c734e1e95f8/docs/figure/Main_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">043902f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/769e48d5a4afe34bb3a05faaa53829aeaadcfc05/docs/figure/Main_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">769e48d</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/83a0c4fdf0c246cc8f853dbeecd4ab8b252df66e/docs/figure/Main_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">83a0c4f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure S1</strong>. The estimated response to selection in
<strong>a</strong> females and <strong>b</strong> males for all traits
measured in females. Innermost bands approximate the median, while outer
bands show the 66 and 95% credible intervals.</p>
<p><span class="math inline">\(~\)</span></p>
<pre class="r"><code>p3 &lt;-
  R_male_traits %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_R = median(Response_to_selection_male)) %&gt;%
  ggplot(aes(Response_to_selection_male, fct_reorder(Trait, avg_R))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 1, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  coord_cartesian(xlim = c(-0.5, 0.5)) +
  geom_vline(linetype = 2, xintercept = 0, linewidth = 1) +
  labs(x = &quot;Response to selection in males&quot;,
       y = &quot;Trait expressed in males&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        text = element_text(size=14),
        axis.text.y = element_text(size = 8))

p4 &lt;- 
  R_male_traits %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_R = median(Response_to_selection_female)) %&gt;%
  ggplot(aes(Response_to_selection_female, fct_reorder(Trait, avg_R))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 1, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Purp&quot;) +
  coord_cartesian(xlim = c(-0.5, 0.5)) +
  geom_vline(linetype = 2, xintercept = 0, linewidth = 1) +
  labs(x = &quot;Response to selection in females&quot;,
       y = &quot;Trait expressed in males&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        text = element_text(size=14),
        axis.text.y = element_text(size = 8)) 




p3 + p4 +
  plot_annotation(tag_levels = &#39;a&#39;)</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-10-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/319c84c1118d4c1c59279734e1b3432ef1d08901/docs/figure/Main_analysis.Rmd/unnamed-chunk-10-1.png" target="_blank">319c84c</a>
</td>
<td>
tomkeaney
</td>
<td>
2023-10-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/daa935de0473e2ecaedca3c65f414d2858ddc598/docs/figure/Main_analysis.Rmd/unnamed-chunk-10-1.png" target="_blank">daa935d</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/b682a4fd7268268f71172d62d1c5978ef2c39b5c/docs/figure/Main_analysis.Rmd/unnamed-chunk-10-1.png" target="_blank">b682a4f</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-03
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/7bf0ee08d2c8b770d1c2d8aca0b2931897462adb/docs/figure/Main_analysis.Rmd/unnamed-chunk-10-1.png" target="_blank">7bf0ee0</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure S2</strong>. The response to selection in
<strong>a</strong> males and <strong>b</strong> females for all traits
measured in males. Innermost bands approximate the median, while outer
bands show the 66 and 95% credible intervals.</p>
<p><span class="math inline">\(~\)</span></p>
<p>Choose the notable traits measured in females in terms of absolute
<span class="math inline">\(R_{FF}\)</span> and <span
class="math inline">\(R_{MF}\)</span> and present them in a Table</p>
<p><strong>Table 1</strong>. The traits measured in females with the
strongest responses to selection on females (left column) and males
(right column) in the DGRP. The sign of <span
class="math inline">\(R\)</span> indicates the direction of the
response, with positive expected responses bolded. Estimated error is
the standard deviation of the posterior distribution. The notable column
indicates that the 95% CI does not overlap zero.</p>
<pre class="r"><code>R_summary_female &lt;-
bind_cols(
  R_female_traits %&gt;% 
    select(Trait, Response_to_selection_female) %&gt;% 
    group_by(Trait) %&gt;% 
    summarise_draws(&quot;median&quot;, &quot;sd&quot;, ~quantile(.x, probs = c(0.025, 0.975), na.rm = TRUE)) %&gt;%
    ungroup() %&gt;% 
    left_join(trait_data %&gt;% distinct(Trait, Trait_nice)) %&gt;% # add the nice names
    rename(`R in females` = median, `Estimated error` = sd,
           `2.5% CI` = `2.5%`, `97.5% CI` = `97.5%`, `Trait in females` = Trait_nice)  %&gt;% 
    mutate(R_abs = abs(`R in females`)) %&gt;% 
    arrange(-R_abs) %&gt;% 
    mutate(across(3:6, ~ round(.x, digits = 3)),
           Notable = case_when(`2.5% CI` &lt; 0 &amp; `97.5% CI` &lt; 0 ~ &quot;*&quot;,
                               `2.5% CI` &gt; 0 &amp; `97.5% CI` &gt; 0 ~ &quot;*&quot;,
                               TRUE ~ &quot; &quot;)) %&gt;% 
    filter(Notable == &quot;*&quot;) %&gt;% 
    mutate(`Trait value` = if_else(`R in females` &lt; 0, &quot;Decreasing&quot;, &quot;Increasing&quot;)) %&gt;% 
    select(`Trait in females`, `Trait value`, `R in females`, `Estimated error`, 
           `2.5% CI`, `97.5% CI`, Notable) %&gt;% 
    #the next lines add empty rows and remove the NA values in them. We do this so we bind the two dataframes 
    add_row() %&gt;%
    add_row() %&gt;%
    add_row() %&gt;%
    mutate_all(~replace(., is.na(.), &quot;&quot;)),
  
  R_female_traits %&gt;% 
    select(Trait, Response_to_selection_male) %&gt;% 
    group_by(Trait) %&gt;% 
    summarise_draws(&quot;median&quot;, &quot;sd&quot;, ~quantile(.x, probs = c(0.025, 0.975), na.rm = TRUE)) %&gt;%
    ungroup() %&gt;% 
    left_join(trait_data %&gt;% distinct(Trait, Trait_nice)) %&gt;% # add the nice names
    rename(`R in males` = median, `Estimated error ` = sd,
           `2.5% CI ` = `2.5%`, `97.5% CI ` = `97.5%`, `Trait in females ` = Trait_nice)  %&gt;% 
    mutate(R_abs = abs(`R in males`)) %&gt;% 
    arrange(-R_abs) %&gt;% 
    mutate(across(3:6, ~ round(.x, digits = 3)),
           `Notable ` = case_when(`2.5% CI ` &lt; 0 &amp; `97.5% CI ` &lt; 0 ~ &quot;*&quot;,
                                 `2.5% CI ` &gt; 0 &amp; `97.5% CI ` &gt; 0 ~ &quot;*&quot;,
                                 TRUE ~ &quot; &quot;)) %&gt;% 
    filter(`Notable ` == &quot;*&quot;) %&gt;% 
    mutate(`Trait value ` = if_else(`R in males` &lt; 0, &quot;Decreasing&quot;, &quot;Increasing&quot;)) %&gt;%
    select(`Trait in females `, `Trait value `, `R in males`, `Estimated error `, `2.5% CI `, `97.5% CI `, `Notable `)
) 
  
  kbl(R_summary_female) %&gt;%
  kable_minimal(font_size = 12) %&gt;% 
  column_spec(1,
              bold = if_else(R_summary_female$`R in females` &gt; 0, TRUE, FALSE),
              width = &quot;22em&quot;) %&gt;% 
  column_spec(7,
              bold = if_else(R_summary_female$`R in males` &gt; 0, TRUE, FALSE),
              width = &quot;24em&quot;) </code></pre>
<table class=" lightable-minimal" style='font-size: 12px; font-family: "Trebuchet MS", verdana, sans-serif; margin-left: auto; margin-right: auto;'>
<thead>
<tr>
<th style="text-align:left;">
Trait in females
</th>
<th style="text-align:left;">
Trait value
</th>
<th style="text-align:left;">
R in females
</th>
<th style="text-align:left;">
Estimated error
</th>
<th style="text-align:left;">
2.5% CI
</th>
<th style="text-align:left;">
97.5% CI
</th>
<th style="text-align:left;">
Notable
</th>
<th style="text-align:left;">
Trait in females
</th>
<th style="text-align:left;">
Trait value
</th>
<th style="text-align:right;">
R in males
</th>
<th style="text-align:right;">
Estimated error
</th>
<th style="text-align:right;">
2.5% CI
</th>
<th style="text-align:right;">
97.5% CI
</th>
<th style="text-align:left;">
Notable
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;width: 22em; font-weight: bold;">
Lifetime fecundity 25C
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:left;">
0.28
</td>
<td style="text-align:left;">
0.088
</td>
<td style="text-align:left;">
0.097
</td>
<td style="text-align:left;">
0.438
</td>
<td style="text-align:left;width: 24em; font-weight: bold;">
<ul>
<li></td>
<td style="text-align:left;">
Locomotor frequency
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.324
</td>
<td style="text-align:right;">
0.113
</td>
<td style="text-align:right;">
0.075
</td>
<td style="text-align:right;">
0.517
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
Day sleep bout length
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:left;">
-0.273
</td>
<td style="text-align:left;">
0.091
</td>
<td style="text-align:left;">
-0.435
</td>
<td style="text-align:left;">
-0.084
</td>
<td style="text-align:left;width: 24em; font-weight: bold;">
<ul>
<li></td>
<td style="text-align:left;">
Critical thermal max
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.289
</td>
<td style="text-align:right;">
0.109
</td>
<td style="text-align:right;">
0.061
</td>
<td style="text-align:right;">
0.482
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; font-weight: bold;">
Week1 fecundity 25C
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:left;">
0.267
</td>
<td style="text-align:left;">
0.087
</td>
<td style="text-align:left;">
0.087
</td>
<td style="text-align:left;">
0.427
</td>
<td style="text-align:left;width: 24em; ">
<ul>
<li></td>
<td style="text-align:left;">
CHC 6,10-C26:2
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:right;">
-0.269
</td>
<td style="text-align:right;">
0.085
</td>
<td style="text-align:right;">
-0.422
</td>
<td style="text-align:right;">
-0.092
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
Cold shock young
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:left;">
-0.246
</td>
<td style="text-align:left;">
0.114
</td>
<td style="text-align:left;">
-0.453
</td>
<td style="text-align:left;">
-0.005
</td>
<td style="text-align:left;width: 24em; font-weight: bold;">
<ul>
<li></td>
<td style="text-align:left;">
Paraquat resistance
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.251
</td>
<td style="text-align:right;">
0.112
</td>
<td style="text-align:right;">
0.013
</td>
<td style="text-align:right;">
0.451
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; font-weight: bold;">
Circadian period mesa
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:left;">
0.241
</td>
<td style="text-align:left;">
0.099
</td>
<td style="text-align:left;">
0.037
</td>
<td style="text-align:left;">
0.42
</td>
<td style="text-align:left;width: 24em; font-weight: bold;">
<ul>
<li></td>
<td style="text-align:left;">
Climbing ability paraquat
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.241
</td>
<td style="text-align:right;">
0.112
</td>
<td style="text-align:right;">
0.008
</td>
<td style="text-align:right;">
0.442
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; font-weight: bold;">
M. anisopliae resistance
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:left;">
0.24
</td>
<td style="text-align:left;">
0.086
</td>
<td style="text-align:left;">
0.064
</td>
<td style="text-align:left;">
0.399
</td>
<td style="text-align:left;width: 24em; font-weight: bold;">
<ul>
<li></td>
<td style="text-align:left;">
CHC n-C26
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.236
</td>
<td style="text-align:right;">
0.086
</td>
<td style="text-align:right;">
0.060
</td>
<td style="text-align:right;">
0.394
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
Sprinter marathoner
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:left;">
-0.233
</td>
<td style="text-align:left;">
0.1
</td>
<td style="text-align:left;">
-0.417
</td>
<td style="text-align:left;">
-0.025
</td>
<td style="text-align:left;width: 24em; font-weight: bold;">
<ul>
<li></td>
<td style="text-align:left;">
M. anisopliae resistance
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.226
</td>
<td style="text-align:right;">
0.084
</td>
<td style="text-align:right;">
0.054
</td>
<td style="text-align:right;">
0.381
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
CHC 8-C25:1
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:left;">
-0.229
</td>
<td style="text-align:left;">
0.091
</td>
<td style="text-align:left;">
-0.392
</td>
<td style="text-align:left;">
-0.041
</td>
<td style="text-align:left;width: 24em; ">
<ul>
<li></td>
<td style="text-align:left;">
CHC 5,9-C25:2 &amp; 9-C25:1
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:right;">
-0.213
</td>
<td style="text-align:right;">
0.088
</td>
<td style="text-align:right;">
-0.377
</td>
<td style="text-align:right;">
-0.035
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; font-weight: bold;">
Week3 fecundity 25C
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:left;">
0.226
</td>
<td style="text-align:left;">
0.088
</td>
<td style="text-align:left;">
0.048
</td>
<td style="text-align:left;">
0.391
</td>
<td style="text-align:left;width: 24em; ">
<ul>
<li></td>
<td style="text-align:left;">
CHC 8,12-C26:2
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:right;">
-0.209
</td>
<td style="text-align:right;">
0.091
</td>
<td style="text-align:right;">
-0.379
</td>
<td style="text-align:right;">
-0.020
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
Waking activity
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:left;">
-0.226
</td>
<td style="text-align:left;">
0.094
</td>
<td style="text-align:left;">
-0.398
</td>
<td style="text-align:left;">
-0.03
</td>
<td style="text-align:left;width: 24em; font-weight: bold;">
<ul>
<li></td>
<td style="text-align:left;">
Phototaxis mid age
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.207
</td>
<td style="text-align:right;">
0.086
</td>
<td style="text-align:right;">
0.034
</td>
<td style="text-align:right;">
0.365
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; font-weight: bold;">
2 3-butanedione olfactory response intensity
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:left;">
0.211
</td>
<td style="text-align:left;">
0.096
</td>
<td style="text-align:left;">
0.013
</td>
<td style="text-align:left;">
0.39
</td>
<td style="text-align:left;width: 24em; ">
<ul>
<li></td>
<td style="text-align:left;">
CHC 9,13-C25:2
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:right;">
-0.203
</td>
<td style="text-align:right;">
0.092
</td>
<td style="text-align:right;">
-0.374
</td>
<td style="text-align:right;">
-0.017
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
Day sleep bout length SD
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:left;">
-0.194
</td>
<td style="text-align:left;">
0.092
</td>
<td style="text-align:left;">
-0.365
</td>
<td style="text-align:left;">
-0.005
</td>
<td style="text-align:left;width: 24em; font-weight: bold;">
<ul>
<li></td>
<td style="text-align:left;">
Startle response
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.194
</td>
<td style="text-align:right;">
0.086
</td>
<td style="text-align:right;">
0.021
</td>
<td style="text-align:right;">
0.354
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; font-weight: bold;">
CHC 7-C27:1
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:left;">
0.192
</td>
<td style="text-align:left;">
0.089
</td>
<td style="text-align:left;">
0.01
</td>
<td style="text-align:left;">
0.358
</td>
<td style="text-align:left;width: 24em; font-weight: bold;">
<ul>
<li></td>
<td style="text-align:left;">
Micro environmental plasticity lifespan 25C
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.191
</td>
<td style="text-align:right;">
0.085
</td>
<td style="text-align:right;">
0.019
</td>
<td style="text-align:right;">
0.351
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
Thorax length thermal plasticity
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:left;">
-0.189
</td>
<td style="text-align:left;">
0.092
</td>
<td style="text-align:left;">
-0.358
</td>
<td style="text-align:left;">
-0.001
</td>
<td style="text-align:left;width: 24em; font-weight: bold;">
<ul>
<li></td>
<td style="text-align:left;">
Ovariole number
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.191
</td>
<td style="text-align:right;">
0.086
</td>
<td style="text-align:right;">
0.017
</td>
<td style="text-align:right;">
0.350
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; font-weight: bold;">
CHC NI 33
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:left;">
0.188
</td>
<td style="text-align:left;">
0.092
</td>
<td style="text-align:left;">
0.002
</td>
<td style="text-align:left;">
0.36
</td>
<td style="text-align:left;width: 24em; ">
<ul>
<li></td>
<td style="text-align:left;">
CHC 6-C23:1
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:right;">
-0.189
</td>
<td style="text-align:right;">
0.090
</td>
<td style="text-align:right;">
-0.355
</td>
<td style="text-align:right;">
-0.006
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
Starvation resistance variance
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:left;">
-0.18
</td>
<td style="text-align:left;">
0.086
</td>
<td style="text-align:left;">
-0.339
</td>
<td style="text-align:left;">
-0.003
</td>
<td style="text-align:left;width: 24em; font-weight: bold;">
<ul>
<li></td>
<td style="text-align:left;">
CHC n-C29
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.185
</td>
<td style="text-align:right;">
0.090
</td>
<td style="text-align:right;">
0.001
</td>
<td style="text-align:right;">
0.350
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 24em; font-weight: bold;">
</td>
<td style="text-align:left;">
Negative geotaxis msb
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.184
</td>
<td style="text-align:right;">
0.087
</td>
<td style="text-align:right;">
0.011
</td>
<td style="text-align:right;">
0.348
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 24em; font-weight: bold;">
</td>
<td style="text-align:left;">
Day sleep bout length SD
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.184
</td>
<td style="text-align:right;">
0.090
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
0.349
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 24em; font-weight: bold;">
</td>
<td style="text-align:left;">
CHC n-C28
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.182
</td>
<td style="text-align:right;">
0.088
</td>
<td style="text-align:right;">
0.004
</td>
<td style="text-align:right;">
0.349
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
</tbody>
</table></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul>
<pre class="r"><code>#make_table_no_linespaces(R_summary_female_f)</code></pre>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Table 2</strong>. The traits in measured in males with the
strongest responses to selection on females (left column) and males
(right column) in the DGRP. The sign of <span
class="math inline">\(R\)</span> indicates the direction of the
response, with positive expected responses bolded. Estimated error is
the standard deviation of the posterior distribution. The notable column
indicates that the 95% CI does not overlap zero.</p>
<pre class="r"><code>R_summary_male &lt;-
 bind_cols(
    R_male_traits %&gt;% 
      select(Trait, Response_to_selection_female) %&gt;% 
      group_by(Trait) %&gt;% 
      summarise_draws(&quot;median&quot;, &quot;sd&quot;, ~quantile(.x, probs = c(0.025, 0.975), na.rm = TRUE)) %&gt;%
      ungroup() %&gt;% 
      left_join(trait_data %&gt;% distinct(Trait, Trait_nice)) %&gt;% # add the nice names
      rename(`R in females` = median, `Estimated error` = sd,
             `2.5% CI` = `2.5%`, `97.5% CI` = `97.5%`, `Trait in males` = Trait_nice)  %&gt;% 
      mutate(R_abs = abs(`R in females`)) %&gt;% 
      arrange(-R_abs) %&gt;% 
      mutate(across(3:6, ~ round(.x, digits = 3)),
             Notable = case_when(`2.5% CI` &lt; 0 &amp; `97.5% CI` &lt; 0 ~ &quot;*&quot;,
                                 `2.5% CI` &gt; 0 &amp; `97.5% CI` &gt; 0 ~ &quot;*&quot;,
                                 TRUE ~ &quot; &quot;)) %&gt;% 
      filter(Notable == &quot;*&quot;) %&gt;% 
   mutate(`Trait value` = if_else(`R in females` &lt; 0, &quot;Decreasing&quot;, &quot;Increasing&quot;)) %&gt;%
      select(`Trait in males`, `Trait value`, `R in females`, `Estimated error`, `2.5% CI`, `97.5% CI`, Notable) %&gt;% 
      # the next lines add empty rows and remove the NA values in them. We do this so we bind the two dataframes 
      add_row() %&gt;% 
      add_row() %&gt;%
      add_row() %&gt;%
      add_row() %&gt;%
      add_row() %&gt;%
      add_row() %&gt;%
      add_row() %&gt;%
      add_row() %&gt;%
      add_row() %&gt;%
      add_row() %&gt;%
      add_row() %&gt;%
      add_row() %&gt;%
      mutate_all(~replace(., is.na(.), &quot;&quot;)),
   
  R_male_traits %&gt;% 
  select(Trait, Response_to_selection_male) %&gt;% 
  group_by(Trait) %&gt;% 
  summarise_draws(&quot;median&quot;, &quot;sd&quot;, ~quantile(.x, probs = c(0.025, 0.975), na.rm = TRUE)) %&gt;%
  ungroup() %&gt;% 
  left_join(trait_data %&gt;% distinct(Trait, Trait_nice)) %&gt;% # add the nice names
  rename(`R in males` = median, `Estimated error ` = sd,
         `2.5% CI ` = `2.5%`, `97.5% CI ` = `97.5%`, `Trait in males ` = Trait_nice)  %&gt;% 
  mutate(R_abs = abs(`R in males`)) %&gt;% 
  arrange(-R_abs) %&gt;% 
  mutate(across(3:6, ~ round(.x, digits = 3)),
         `Notable ` = case_when(`2.5% CI ` &lt; 0 &amp; `97.5% CI ` &lt; 0 ~ &quot;*&quot;,
                               `2.5% CI ` &gt; 0 &amp; `97.5% CI ` &gt; 0 ~ &quot;*&quot;,
                               TRUE ~ &quot; &quot;)) %&gt;% 
  filter(`Notable ` == &quot;*&quot;) %&gt;% 
     mutate(`Trait value ` = if_else(`R in males` &lt; 0, &quot;Decreasing&quot;, &quot;Increasing&quot;)) %&gt;%
  select(`Trait in males `, `Trait value `, `R in males`, `Estimated error `, `2.5% CI `, `97.5% CI `, `Notable `)
  )


kbl(R_summary_male) %&gt;%
  kable_minimal(font_size = 12) %&gt;% 
  column_spec(1,
              bold = if_else(R_summary_male$`R in females` &gt; 0, TRUE, FALSE),
              width = &quot;22em&quot;) %&gt;% 
  column_spec(7,
              bold = if_else(R_summary_male$`R in males` &gt; 0, TRUE, FALSE),
              width = &quot;28em&quot;)  </code></pre>
<table class=" lightable-minimal" style='font-size: 12px; font-family: "Trebuchet MS", verdana, sans-serif; margin-left: auto; margin-right: auto;'>
<thead>
<tr>
<th style="text-align:left;">
Trait in males
</th>
<th style="text-align:left;">
Trait value
</th>
<th style="text-align:left;">
R in females
</th>
<th style="text-align:left;">
Estimated error
</th>
<th style="text-align:left;">
2.5% CI
</th>
<th style="text-align:left;">
97.5% CI
</th>
<th style="text-align:left;">
Notable
</th>
<th style="text-align:left;">
Trait in males
</th>
<th style="text-align:left;">
Trait value
</th>
<th style="text-align:right;">
R in males
</th>
<th style="text-align:right;">
Estimated error
</th>
<th style="text-align:right;">
2.5% CI
</th>
<th style="text-align:right;">
97.5% CI
</th>
<th style="text-align:left;">
Notable
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;width: 22em; ">
Protein gnotobiotic
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:left;">
-0.284
</td>
<td style="text-align:left;">
0.126
</td>
<td style="text-align:left;">
-0.502
</td>
<td style="text-align:left;">
-0.014
</td>
<td style="text-align:left;width: 28em; ">
<ul>
<li></td>
<td style="text-align:left;">
Mating latency
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:right;">
-0.317
</td>
<td style="text-align:right;">
0.084
</td>
<td style="text-align:right;">
-0.470
</td>
<td style="text-align:right;">
-0.144
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
Cold shock young
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:left;">
-0.254
</td>
<td style="text-align:left;">
0.111
</td>
<td style="text-align:left;">
-0.455
</td>
<td style="text-align:left;">
-0.022
</td>
<td style="text-align:left;width: 28em; ">
<ul>
<li></td>
<td style="text-align:left;">
Triglyceride gnotobiotic
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:right;">
-0.288
</td>
<td style="text-align:right;">
0.118
</td>
<td style="text-align:right;">
-0.496
</td>
<td style="text-align:right;">
-0.033
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; font-weight: bold;">
M. anisopliae resistance
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:left;">
0.238
</td>
<td style="text-align:left;">
0.088
</td>
<td style="text-align:left;">
0.059
</td>
<td style="text-align:left;">
0.4
</td>
<td style="text-align:left;width: 28em; ">
<ul>
<li></td>
<td style="text-align:left;">
P. rettgeri load high glucose diet
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:right;">
-0.272
</td>
<td style="text-align:right;">
0.090
</td>
<td style="text-align:right;">
-0.438
</td>
<td style="text-align:right;">
-0.085
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; font-weight: bold;">
Circadian period mesa
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:left;">
0.215
</td>
<td style="text-align:left;">
0.095
</td>
<td style="text-align:left;">
0.02
</td>
<td style="text-align:left;">
0.393
</td>
<td style="text-align:left;width: 28em; font-weight: bold;">
<ul>
<li></td>
<td style="text-align:left;">
Rapid cold hardening old
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.262
</td>
<td style="text-align:right;">
0.108
</td>
<td style="text-align:right;">
0.036
</td>
<td style="text-align:right;">
0.453
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; font-weight: bold;">
Flight performance
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:left;">
0.191
</td>
<td style="text-align:left;">
0.085
</td>
<td style="text-align:left;">
0.019
</td>
<td style="text-align:left;">
0.35
</td>
<td style="text-align:left;width: 28em; font-weight: bold;">
<ul>
<li></td>
<td style="text-align:left;">
Climbing ability paraquat
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.261
</td>
<td style="text-align:right;">
0.113
</td>
<td style="text-align:right;">
0.023
</td>
<td style="text-align:right;">
0.462
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; font-weight: bold;">
Starvation resistance high protein diet
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:left;">
0.179
</td>
<td style="text-align:left;">
0.086
</td>
<td style="text-align:left;">
0.003
</td>
<td style="text-align:left;">
0.342
</td>
<td style="text-align:left;width: 28em; font-weight: bold;">
<ul>
<li></td>
<td style="text-align:left;">
Lifespan 28C
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.258
</td>
<td style="text-align:right;">
0.085
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
0.415
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 28em; ">
</td>
<td style="text-align:left;">
Cold shock young
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:right;">
-0.245
</td>
<td style="text-align:right;">
0.107
</td>
<td style="text-align:right;">
-0.438
</td>
<td style="text-align:right;">
-0.022
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 28em; font-weight: bold;">
</td>
<td style="text-align:left;">
M. anisopliae resistance
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.228
</td>
<td style="text-align:right;">
0.087
</td>
<td style="text-align:right;">
0.047
</td>
<td style="text-align:right;">
0.388
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 28em; ">
</td>
<td style="text-align:left;">
Olfactory response ethylacetate
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:right;">
-0.209
</td>
<td style="text-align:right;">
0.085
</td>
<td style="text-align:right;">
-0.367
</td>
<td style="text-align:right;">
-0.033
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 28em; font-weight: bold;">
</td>
<td style="text-align:left;">
Glucose low glucose diet
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.208
</td>
<td style="text-align:right;">
0.096
</td>
<td style="text-align:right;">
0.009
</td>
<td style="text-align:right;">
0.384
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 28em; font-weight: bold;">
</td>
<td style="text-align:left;">
Day sleep bout length SD
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.208
</td>
<td style="text-align:right;">
0.091
</td>
<td style="text-align:right;">
0.019
</td>
<td style="text-align:right;">
0.376
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 28em; font-weight: bold;">
</td>
<td style="text-align:left;">
Micro environmental plasticity lifespan 25C
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.202
</td>
<td style="text-align:right;">
0.084
</td>
<td style="text-align:right;">
0.030
</td>
<td style="text-align:right;">
0.357
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 28em; ">
</td>
<td style="text-align:left;">
CHC 2-Me-C22
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:right;">
-0.198
</td>
<td style="text-align:right;">
0.094
</td>
<td style="text-align:right;">
-0.373
</td>
<td style="text-align:right;">
-0.008
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 28em; font-weight: bold;">
</td>
<td style="text-align:left;">
Day sleep bout length
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.197
</td>
<td style="text-align:right;">
0.092
</td>
<td style="text-align:right;">
0.010
</td>
<td style="text-align:right;">
0.366
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 28em; ">
</td>
<td style="text-align:left;">
Critical thermal min within environment variation plasticity
</td>
<td style="text-align:left;">
Decreasing
</td>
<td style="text-align:right;">
-0.194
</td>
<td style="text-align:right;">
0.090
</td>
<td style="text-align:right;">
-0.362
</td>
<td style="text-align:right;">
-0.012
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 28em; font-weight: bold;">
</td>
<td style="text-align:left;">
Activity MPH
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.193
</td>
<td style="text-align:right;">
0.088
</td>
<td style="text-align:right;">
0.012
</td>
<td style="text-align:right;">
0.356
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 28em; font-weight: bold;">
</td>
<td style="text-align:left;">
Locomotor activity
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.181
</td>
<td style="text-align:right;">
0.086
</td>
<td style="text-align:right;">
0.008
</td>
<td style="text-align:right;">
0.341
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 22em; ">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 28em; font-weight: bold;">
</td>
<td style="text-align:left;">
Acclimated acute cold exposure resistance 25C
</td>
<td style="text-align:left;">
Increasing
</td>
<td style="text-align:right;">
0.181
</td>
<td style="text-align:right;">
0.087
</td>
<td style="text-align:right;">
0.005
</td>
<td style="text-align:right;">
0.345
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
</tbody>
</table></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul>
<pre class="r"><code>#make_table_no_linespaces(R_summary_male_f)</code></pre>
<p><strong>Table S2</strong>. Traits with the strongest overall
responses to selection in the DGRP (total traits measured = 232 in
females and 242 in males). The sign of <span
class="math inline">\(R\)</span> indicates the direction of the
response, with positive expected responses bolded. Estimated error is
the standard deviation of the posterior distribution. The notable column
indicates whether the 95% CI overlaps with zero.</p>
<pre class="r"><code>R_total_summary &lt;-
  bind_rows(
    left_join(
      R_female_traits %&gt;% 
        select(Trait, R_overall) %&gt;% 
        group_by(Trait) %&gt;% 
        summarise_draws(&quot;median&quot;, &quot;sd&quot;, ~quantile(.x, probs = c(0.025, 0.975), na.rm = TRUE)) %&gt;%
        ungroup() %&gt;% 
        left_join(trait_data %&gt;% distinct(Trait, Trait_nice)) %&gt;% # add the nice names
        rename(`R overall` = median, `Estimated error` = sd,
               `2.5% CI` = `2.5%`, `97.5% CI` = `97.5%`)  %&gt;% 
        mutate(R_abs = abs(`R overall`)) %&gt;% 
        arrange(-R_abs) %&gt;% 
        mutate(across(3:6, ~ round(.x, digits = 3)),
               Notable = case_when(`2.5% CI` &lt; 0 &amp; `97.5% CI` &lt; 0 ~ &quot;*&quot;,
                                   `2.5% CI` &gt; 0 &amp; `97.5% CI` &gt; 0 ~ &quot;*&quot;,
                                   TRUE ~ &quot; &quot;)) %&gt;% 
        filter(Notable == &quot;*&quot;),
      
      R_female_traits %&gt;% 
        select(Trait, Trait_Sex, Response_to_selection_female, Response_to_selection_male) %&gt;% 
        group_by(Trait, Trait_Sex) %&gt;% 
        summarise(`R in females` = round(median(Response_to_selection_female), 3),
                  `R in males` = round(median(Response_to_selection_male), 3)) %&gt;% 
        ungroup()
    ) %&gt;% 
      select(Trait_nice, Trait_Sex, `R in females`, `R in males`, `R overall`, `Estimated error`, `2.5% CI`, `97.5% CI`, Notable) %&gt;% 
      rename(Trait = Trait_nice, `Trait sex` = Trait_Sex),
    
     left_join(
      R_male_traits %&gt;% 
        select(Trait, R_overall) %&gt;% 
        group_by(Trait) %&gt;% 
        summarise_draws(&quot;median&quot;, &quot;sd&quot;, ~quantile(.x, probs = c(0.025, 0.975), na.rm = TRUE)) %&gt;%
        ungroup() %&gt;% 
        left_join(trait_data %&gt;% distinct(Trait, Trait_nice)) %&gt;% # add the nice names
        rename(`R overall` = median, `Estimated error` = sd,
               `2.5% CI` = `2.5%`, `97.5% CI` = `97.5%`)  %&gt;% 
        mutate(R_abs = abs(`R overall`)) %&gt;% 
        arrange(-R_abs) %&gt;% 
        mutate(across(3:6, ~ round(.x, digits = 3)),
               Notable = case_when(`2.5% CI` &lt; 0 &amp; `97.5% CI` &lt; 0 ~ &quot;*&quot;,
                                   `2.5% CI` &gt; 0 &amp; `97.5% CI` &gt; 0 ~ &quot;*&quot;,
                                   TRUE ~ &quot; &quot;)) %&gt;% 
        filter(Notable == &quot;*&quot;),
      
      R_male_traits %&gt;% 
        select(Trait, Trait_Sex, Response_to_selection_female, Response_to_selection_male) %&gt;% 
        group_by(Trait, Trait_Sex) %&gt;% 
        summarise(`R in females` = round(median(Response_to_selection_female), 3),
                  `R in males` = round(median(Response_to_selection_male), 3)) %&gt;% 
        ungroup()
    ) %&gt;% 
      select(Trait_nice, Trait_Sex, `R in females`, `R in males`, `R overall`, `Estimated error`, `2.5% CI`, `97.5% CI`, Notable) %&gt;% 
      rename(Trait = Trait_nice,
             `Trait sex` = Trait_Sex)
  ) %&gt;% 
  select(Trait, `R overall`, everything())
    

kbl(R_total_summary %&gt;% select(-`Trait sex`)) %&gt;%
  kable_minimal(full_width = F, font_size = 12) %&gt;% 
  column_spec(1,
              bold = if_else(R_total_summary$`R overall` &gt; 0, TRUE, FALSE),
              width = &quot;21em&quot;)%&gt;% 
  pack_rows(&quot;Traits measured in females&quot;, 1, 31, hline_after = TRUE, italic = TRUE) %&gt;% 
  pack_rows(&quot;Traits measured in males&quot;, 32, 52, hline_after = TRUE, italic = TRUE)</code></pre>
<table class=" lightable-minimal" style='font-size: 12px; font-family: "Trebuchet MS", verdana, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;'>
<thead>
<tr>
<th style="text-align:left;">
Trait
</th>
<th style="text-align:right;">
R overall
</th>
<th style="text-align:right;">
R in females
</th>
<th style="text-align:right;">
R in males
</th>
<th style="text-align:right;">
Estimated error
</th>
<th style="text-align:right;">
2.5% CI
</th>
<th style="text-align:right;">
97.5% CI
</th>
<th style="text-align:left;">
Notable
</th>
</tr>
</thead>
<tbody>
<tr grouplength="31">
<td colspan="8" style="border-bottom: 0;">
<em><strong>Traits measured in females</strong></em>
</td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Locomotor frequency
</td>
<td style="text-align:right;">
0.254
</td>
<td style="text-align:right;">
0.192
</td>
<td style="text-align:right;">
0.324
</td>
<td style="text-align:right;">
0.088
</td>
<td style="text-align:right;">
0.071
</td>
<td style="text-align:right;">
0.413
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Cold shock young
</td>
<td style="text-align:right;">
-0.233
</td>
<td style="text-align:right;">
-0.246
</td>
<td style="text-align:right;">
-0.224
</td>
<td style="text-align:right;">
0.080
</td>
<td style="text-align:right;">
-0.382
</td>
<td style="text-align:right;">
-0.067
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
M. anisopliae resistance
</td>
<td style="text-align:right;">
0.232
</td>
<td style="text-align:right;">
0.240
</td>
<td style="text-align:right;">
0.226
</td>
<td style="text-align:right;">
0.060
</td>
<td style="text-align:right;">
0.109
</td>
<td style="text-align:right;">
0.348
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
P. aeruginosa resistance
</td>
<td style="text-align:right;">
0.214
</td>
<td style="text-align:right;">
0.219
</td>
<td style="text-align:right;">
0.214
</td>
<td style="text-align:right;">
0.092
</td>
<td style="text-align:right;">
0.027
</td>
<td style="text-align:right;">
0.384
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Lifetime fecundity 25C
</td>
<td style="text-align:right;">
0.197
</td>
<td style="text-align:right;">
0.280
</td>
<td style="text-align:right;">
0.116
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
0.068
</td>
<td style="text-align:right;">
0.316
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
CHC 9,13-C25:2
</td>
<td style="text-align:right;">
-0.182
</td>
<td style="text-align:right;">
-0.163
</td>
<td style="text-align:right;">
-0.203
</td>
<td style="text-align:right;">
0.066
</td>
<td style="text-align:right;">
-0.309
</td>
<td style="text-align:right;">
-0.048
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Week1 fecundity 25C
</td>
<td style="text-align:right;">
0.182
</td>
<td style="text-align:right;">
0.267
</td>
<td style="text-align:right;">
0.097
</td>
<td style="text-align:right;">
0.062
</td>
<td style="text-align:right;">
0.058
</td>
<td style="text-align:right;">
0.301
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
CHC 6-C23:1
</td>
<td style="text-align:right;">
-0.177
</td>
<td style="text-align:right;">
-0.168
</td>
<td style="text-align:right;">
-0.189
</td>
<td style="text-align:right;">
0.065
</td>
<td style="text-align:right;">
-0.300
</td>
<td style="text-align:right;">
-0.048
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Cold shock old
</td>
<td style="text-align:right;">
-0.176
</td>
<td style="text-align:right;">
-0.128
</td>
<td style="text-align:right;">
-0.226
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
-0.331
</td>
<td style="text-align:right;">
-0.007
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Waking activity SD
</td>
<td style="text-align:right;">
-0.175
</td>
<td style="text-align:right;">
-0.188
</td>
<td style="text-align:right;">
-0.166
</td>
<td style="text-align:right;">
0.072
</td>
<td style="text-align:right;">
-0.313
</td>
<td style="text-align:right;">
-0.033
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Negative geotaxis msb
</td>
<td style="text-align:right;">
0.169
</td>
<td style="text-align:right;">
0.156
</td>
<td style="text-align:right;">
0.184
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
0.043
</td>
<td style="text-align:right;">
0.288
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Phototaxis mid age
</td>
<td style="text-align:right;">
0.168
</td>
<td style="text-align:right;">
0.131
</td>
<td style="text-align:right;">
0.207
</td>
<td style="text-align:right;">
0.062
</td>
<td style="text-align:right;">
0.044
</td>
<td style="text-align:right;">
0.286
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Circadian period mesa
</td>
<td style="text-align:right;">
0.161
</td>
<td style="text-align:right;">
0.241
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
0.071
</td>
<td style="text-align:right;">
0.016
</td>
<td style="text-align:right;">
0.297
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Week3 fecundity 25C
</td>
<td style="text-align:right;">
0.157
</td>
<td style="text-align:right;">
0.226
</td>
<td style="text-align:right;">
0.090
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
0.033
</td>
<td style="text-align:right;">
0.277
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
CHC 8-C25:1
</td>
<td style="text-align:right;">
-0.156
</td>
<td style="text-align:right;">
-0.229
</td>
<td style="text-align:right;">
-0.085
</td>
<td style="text-align:right;">
0.065
</td>
<td style="text-align:right;">
-0.280
</td>
<td style="text-align:right;">
-0.024
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Fecundity
</td>
<td style="text-align:right;">
0.151
</td>
<td style="text-align:right;">
0.190
</td>
<td style="text-align:right;">
0.114
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
0.017
</td>
<td style="text-align:right;">
0.280
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
CHC 2-Me-C24
</td>
<td style="text-align:right;">
-0.148
</td>
<td style="text-align:right;">
-0.114
</td>
<td style="text-align:right;">
-0.185
</td>
<td style="text-align:right;">
0.068
</td>
<td style="text-align:right;">
-0.279
</td>
<td style="text-align:right;">
-0.011
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
CHC n-C21
</td>
<td style="text-align:right;">
-0.148
</td>
<td style="text-align:right;">
-0.165
</td>
<td style="text-align:right;">
-0.130
</td>
<td style="text-align:right;">
0.066
</td>
<td style="text-align:right;">
-0.272
</td>
<td style="text-align:right;">
-0.015
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
CHC x-C23:1
</td>
<td style="text-align:right;">
-0.146
</td>
<td style="text-align:right;">
-0.136
</td>
<td style="text-align:right;">
-0.159
</td>
<td style="text-align:right;">
0.065
</td>
<td style="text-align:right;">
-0.273
</td>
<td style="text-align:right;">
-0.015
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
CHC n-C26
</td>
<td style="text-align:right;">
0.145
</td>
<td style="text-align:right;">
0.058
</td>
<td style="text-align:right;">
0.236
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
0.021
</td>
<td style="text-align:right;">
0.267
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
CHC NI 33
</td>
<td style="text-align:right;">
0.143
</td>
<td style="text-align:right;">
0.188
</td>
<td style="text-align:right;">
0.099
</td>
<td style="text-align:right;">
0.065
</td>
<td style="text-align:right;">
0.012
</td>
<td style="text-align:right;">
0.265
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Startle response msb
</td>
<td style="text-align:right;">
0.142
</td>
<td style="text-align:right;">
0.137
</td>
<td style="text-align:right;">
0.148
</td>
<td style="text-align:right;">
0.062
</td>
<td style="text-align:right;">
0.016
</td>
<td style="text-align:right;">
0.261
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
CHC n-C23
</td>
<td style="text-align:right;">
-0.141
</td>
<td style="text-align:right;">
-0.153
</td>
<td style="text-align:right;">
-0.129
</td>
<td style="text-align:right;">
0.065
</td>
<td style="text-align:right;">
-0.263
</td>
<td style="text-align:right;">
-0.009
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Abdomen length 28C
</td>
<td style="text-align:right;">
0.141
</td>
<td style="text-align:right;">
0.126
</td>
<td style="text-align:right;">
0.156
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
0.014
</td>
<td style="text-align:right;">
0.261
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Startle response
</td>
<td style="text-align:right;">
0.140
</td>
<td style="text-align:right;">
0.087
</td>
<td style="text-align:right;">
0.194
</td>
<td style="text-align:right;">
0.062
</td>
<td style="text-align:right;">
0.015
</td>
<td style="text-align:right;">
0.260
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
CHC NI 34
</td>
<td style="text-align:right;">
0.139
</td>
<td style="text-align:right;">
0.130
</td>
<td style="text-align:right;">
0.152
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
0.015
</td>
<td style="text-align:right;">
0.259
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Waking activity
</td>
<td style="text-align:right;">
-0.138
</td>
<td style="text-align:right;">
-0.226
</td>
<td style="text-align:right;">
-0.052
</td>
<td style="text-align:right;">
0.067
</td>
<td style="text-align:right;">
-0.265
</td>
<td style="text-align:right;">
-0.003
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
CHC 7-C27:1
</td>
<td style="text-align:right;">
0.134
</td>
<td style="text-align:right;">
0.192
</td>
<td style="text-align:right;">
0.078
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
0.008
</td>
<td style="text-align:right;">
0.255
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Dessication tolerance
</td>
<td style="text-align:right;">
0.134
</td>
<td style="text-align:right;">
0.166
</td>
<td style="text-align:right;">
0.104
</td>
<td style="text-align:right;">
0.064
</td>
<td style="text-align:right;">
0.009
</td>
<td style="text-align:right;">
0.257
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Boric acid resistance
</td>
<td style="text-align:right;">
0.130
</td>
<td style="text-align:right;">
0.137
</td>
<td style="text-align:right;">
0.121
</td>
<td style="text-align:right;">
0.064
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
0.252
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
CHC 6,10-C26:2
</td>
<td style="text-align:right;">
-0.128
</td>
<td style="text-align:right;">
0.013
</td>
<td style="text-align:right;">
-0.269
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
-0.248
</td>
<td style="text-align:right;">
-0.003
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr grouplength="21">
<td colspan="8" style="border-bottom: 0;">
<em><strong>Traits measured in males</strong></em>
</td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Cold shock young
</td>
<td style="text-align:right;">
-0.247
</td>
<td style="text-align:right;">
-0.254
</td>
<td style="text-align:right;">
-0.245
</td>
<td style="text-align:right;">
0.078
</td>
<td style="text-align:right;">
-0.393
</td>
<td style="text-align:right;">
-0.089
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
P. aeruginosa resistance
</td>
<td style="text-align:right;">
0.233
</td>
<td style="text-align:right;">
0.231
</td>
<td style="text-align:right;">
0.240
</td>
<td style="text-align:right;">
0.087
</td>
<td style="text-align:right;">
0.051
</td>
<td style="text-align:right;">
0.395
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
M. anisopliae resistance
</td>
<td style="text-align:right;">
0.232
</td>
<td style="text-align:right;">
0.238
</td>
<td style="text-align:right;">
0.228
</td>
<td style="text-align:right;">
0.062
</td>
<td style="text-align:right;">
0.106
</td>
<td style="text-align:right;">
0.347
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Triglyceride gnotobiotic
</td>
<td style="text-align:right;">
-0.229
</td>
<td style="text-align:right;">
-0.174
</td>
<td style="text-align:right;">
-0.288
</td>
<td style="text-align:right;">
0.091
</td>
<td style="text-align:right;">
-0.397
</td>
<td style="text-align:right;">
-0.037
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
P. rettgeri load high glucose diet
</td>
<td style="text-align:right;">
-0.216
</td>
<td style="text-align:right;">
-0.164
</td>
<td style="text-align:right;">
-0.272
</td>
<td style="text-align:right;">
0.069
</td>
<td style="text-align:right;">
-0.345
</td>
<td style="text-align:right;">
-0.077
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Protein gnotobiotic
</td>
<td style="text-align:right;">
-0.188
</td>
<td style="text-align:right;">
-0.284
</td>
<td style="text-align:right;">
-0.095
</td>
<td style="text-align:right;">
0.086
</td>
<td style="text-align:right;">
-0.349
</td>
<td style="text-align:right;">
-0.011
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Olfactory response ethylacetate
</td>
<td style="text-align:right;">
-0.175
</td>
<td style="text-align:right;">
-0.143
</td>
<td style="text-align:right;">
-0.209
</td>
<td style="text-align:right;">
0.061
</td>
<td style="text-align:right;">
-0.291
</td>
<td style="text-align:right;">
-0.051
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Flight performance
</td>
<td style="text-align:right;">
0.169
</td>
<td style="text-align:right;">
0.191
</td>
<td style="text-align:right;">
0.150
</td>
<td style="text-align:right;">
0.061
</td>
<td style="text-align:right;">
0.048
</td>
<td style="text-align:right;">
0.285
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Mating latency
</td>
<td style="text-align:right;">
-0.165
</td>
<td style="text-align:right;">
-0.014
</td>
<td style="text-align:right;">
-0.317
</td>
<td style="text-align:right;">
0.065
</td>
<td style="text-align:right;">
-0.290
</td>
<td style="text-align:right;">
-0.036
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Day sleep SD
</td>
<td style="text-align:right;">
-0.162
</td>
<td style="text-align:right;">
-0.159
</td>
<td style="text-align:right;">
-0.167
</td>
<td style="text-align:right;">
0.064
</td>
<td style="text-align:right;">
-0.284
</td>
<td style="text-align:right;">
-0.035
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Glucose low glucose diet
</td>
<td style="text-align:right;">
0.162
</td>
<td style="text-align:right;">
0.117
</td>
<td style="text-align:right;">
0.208
</td>
<td style="text-align:right;">
0.071
</td>
<td style="text-align:right;">
0.016
</td>
<td style="text-align:right;">
0.297
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Circadian period mesa
</td>
<td style="text-align:right;">
0.160
</td>
<td style="text-align:right;">
0.215
</td>
<td style="text-align:right;">
0.109
</td>
<td style="text-align:right;">
0.068
</td>
<td style="text-align:right;">
0.025
</td>
<td style="text-align:right;">
0.289
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Parathion resistance
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
0.174
</td>
<td style="text-align:right;">
0.132
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
0.026
</td>
<td style="text-align:right;">
0.272
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Circadian period chi
</td>
<td style="text-align:right;">
0.153
</td>
<td style="text-align:right;">
0.194
</td>
<td style="text-align:right;">
0.113
</td>
<td style="text-align:right;">
0.072
</td>
<td style="text-align:right;">
0.006
</td>
<td style="text-align:right;">
0.290
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
CHC 2-Me-C22
</td>
<td style="text-align:right;">
-0.151
</td>
<td style="text-align:right;">
-0.107
</td>
<td style="text-align:right;">
-0.198
</td>
<td style="text-align:right;">
0.068
</td>
<td style="text-align:right;">
-0.280
</td>
<td style="text-align:right;">
-0.016
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Activity
</td>
<td style="text-align:right;">
0.141
</td>
<td style="text-align:right;">
0.107
</td>
<td style="text-align:right;">
0.175
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
0.013
</td>
<td style="text-align:right;">
0.261
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Activity MPH
</td>
<td style="text-align:right;">
0.139
</td>
<td style="text-align:right;">
0.085
</td>
<td style="text-align:right;">
0.193
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
0.010
</td>
<td style="text-align:right;">
0.262
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Night sleep SD
</td>
<td style="text-align:right;">
-0.139
</td>
<td style="text-align:right;">
-0.116
</td>
<td style="text-align:right;">
-0.162
</td>
<td style="text-align:right;">
0.066
</td>
<td style="text-align:right;">
-0.265
</td>
<td style="text-align:right;">
-0.005
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Startle response msb
</td>
<td style="text-align:right;">
0.130
</td>
<td style="text-align:right;">
0.086
</td>
<td style="text-align:right;">
0.174
</td>
<td style="text-align:right;">
0.063
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
0.249
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Negative geotaxis msb
</td>
<td style="text-align:right;">
0.129
</td>
<td style="text-align:right;">
0.099
</td>
<td style="text-align:right;">
0.159
</td>
<td style="text-align:right;">
0.064
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
0.253
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Starvation resistance high protein diet
</td>
<td style="text-align:right;">
0.126
</td>
<td style="text-align:right;">
0.179
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
0.062
</td>
<td style="text-align:right;">
0.002
</td>
<td style="text-align:right;">
0.244
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
</tbody>
</table></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Table 3</strong>. The traits with the largest <span
class="math inline">\(\Delta R\)</span> between the sexes in the DGRP.
Traits where <span class="math inline">\(R_F &gt; R_M\)</span> are
indicated in bold. Estimated error is the standard deviation of the
posterior distribution. The notable column indicates whether the 95% CI
overlaps with zero.</p>
<pre class="r"><code>R_diff_summary &lt;-
  bind_rows(
    left_join(
      R_female_traits %&gt;% 
        select(Trait, R_diff) %&gt;% 
        group_by(Trait) %&gt;% 
        summarise_draws(&quot;median&quot;, &quot;sd&quot;, ~quantile(.x, probs = c(0.025, 0.975), na.rm = TRUE)) %&gt;%
        ungroup() %&gt;% 
        left_join(trait_data %&gt;% distinct(Trait, Trait_nice)) %&gt;% # add the nice names
        rename(`Female R - Male R` = median, `Estimated error` = sd,
               `2.5% CI` = `2.5%`, `97.5% CI` = `97.5%`)  %&gt;% 
        mutate(R_abs = abs(`Female R - Male R`)) %&gt;% 
        arrange(-R_abs) %&gt;% 
        mutate(across(3:6, ~ round(.x, digits = 3)),
               Notable = case_when(`2.5% CI` &lt; 0 &amp; `97.5% CI` &lt; 0 ~ &quot;*&quot;,
                                   `2.5% CI` &gt; 0 &amp; `97.5% CI` &gt; 0 ~ &quot;*&quot;,
                                   TRUE ~ &quot; &quot;)) %&gt;% 
        filter(Notable == &quot;*&quot;),
      
      R_female_traits %&gt;% 
        select(Trait, Response_to_selection_female, Response_to_selection_male) %&gt;% 
        group_by(Trait) %&gt;% 
        summarise(`R in females` = round(median(Response_to_selection_female), 3),
                  `R in males` = round(median(Response_to_selection_male), 3)) %&gt;% 
        ungroup()
    ) %&gt;% 
      select(Trait_nice, `R in females`, `R in males`, `Female R - Male R`, `Estimated error`, `2.5% CI`, `97.5% CI`, Notable) %&gt;% 
      rename(Trait = Trait_nice),
    
    left_join(
    R_male_traits %&gt;% 
      select(Trait, R_diff) %&gt;% 
      group_by(Trait) %&gt;% 
      summarise_draws(&quot;median&quot;, &quot;sd&quot;, ~quantile(.x, probs = c(0.025, 0.975), na.rm = TRUE)) %&gt;%
      ungroup() %&gt;% 
      left_join(trait_data %&gt;% distinct(Trait, Trait_nice)) %&gt;% # add the nice names
      rename(`Female R - Male R` = median, `Estimated error` = sd,
             `2.5% CI` = `2.5%`, `97.5% CI` = `97.5%`)  %&gt;% 
      mutate(R_abs = abs(`Female R - Male R`)) %&gt;% 
      arrange(-R_abs) %&gt;% 
      mutate(across(3:6, ~ round(.x, digits = 3)),
             Notable = case_when(`2.5% CI` &lt; 0 &amp; `97.5% CI` &lt; 0 ~ &quot;*&quot;,
                                 `2.5% CI` &gt; 0 &amp; `97.5% CI` &gt; 0 ~ &quot;*&quot;,
                                 TRUE ~ &quot; &quot;)) %&gt;% 
      filter(Notable == &quot;*&quot;),
    
    R_male_traits %&gt;% 
      select(Trait, Response_to_selection_female, Response_to_selection_male) %&gt;% 
      group_by(Trait) %&gt;% 
      summarise(`R in females` = round(median(Response_to_selection_female), 3),
                `R in males` = round(median(Response_to_selection_male), 3)) %&gt;% 
      ungroup()
  ) %&gt;% 
  select(Trait_nice, `R in females`, `R in males`, `Female R - Male R`, `Estimated error`, `2.5% CI`, `97.5% CI`, Notable) %&gt;% 
  rename(Trait = Trait_nice) %&gt;% 
  filter(!is.na(Trait))
  ) %&gt;% 
  select(Trait, `Female R - Male R`, everything())


kbl(R_diff_summary) %&gt;%
  kable_minimal(full_width = F, font_size = 12) %&gt;% 
  column_spec(1,
              bold = if_else(R_diff_summary$`Female R - Male R` &gt; 0, TRUE, FALSE),
              width = &quot;21em&quot;) %&gt;% 
  pack_rows(&quot;Traits measured in females&quot;, 1, 8, hline_after = TRUE, italic = TRUE) %&gt;% 
  pack_rows(&quot;Traits measured in males&quot;, 9, 18, hline_after = TRUE, italic = TRUE)</code></pre>
<table class=" lightable-minimal" style='font-size: 12px; font-family: "Trebuchet MS", verdana, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;'>
<thead>
<tr>
<th style="text-align:left;">
Trait
</th>
<th style="text-align:right;">
Female R - Male R
</th>
<th style="text-align:right;">
R in females
</th>
<th style="text-align:right;">
R in males
</th>
<th style="text-align:right;">
Estimated error
</th>
<th style="text-align:right;">
2.5% CI
</th>
<th style="text-align:right;">
97.5% CI
</th>
<th style="text-align:left;">
Notable
</th>
</tr>
</thead>
<tbody>
<tr grouplength="8">
<td colspan="8" style="border-bottom: 0;">
<em><strong>Traits measured in females</strong></em>
</td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Day sleep bout length
</td>
<td style="text-align:right;">
-0.442
</td>
<td style="text-align:right;">
-0.273
</td>
<td style="text-align:right;">
0.173
</td>
<td style="text-align:right;">
0.129
</td>
<td style="text-align:right;">
-0.683
</td>
<td style="text-align:right;">
-0.178
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Sprinter marathoner
</td>
<td style="text-align:right;">
-0.422
</td>
<td style="text-align:right;">
-0.233
</td>
<td style="text-align:right;">
0.191
</td>
<td style="text-align:right;">
0.142
</td>
<td style="text-align:right;">
-0.687
</td>
<td style="text-align:right;">
-0.133
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Critical thermal max
</td>
<td style="text-align:right;">
-0.415
</td>
<td style="text-align:right;">
-0.129
</td>
<td style="text-align:right;">
0.289
</td>
<td style="text-align:right;">
0.160
</td>
<td style="text-align:right;">
-0.711
</td>
<td style="text-align:right;">
-0.084
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Day sleep bout length SD
</td>
<td style="text-align:right;">
-0.376
</td>
<td style="text-align:right;">
-0.194
</td>
<td style="text-align:right;">
0.184
</td>
<td style="text-align:right;">
0.129
</td>
<td style="text-align:right;">
-0.619
</td>
<td style="text-align:right;">
-0.119
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
2 3-butanedione olfactory response intensity
</td>
<td style="text-align:right;">
0.301
</td>
<td style="text-align:right;">
0.211
</td>
<td style="text-align:right;">
-0.092
</td>
<td style="text-align:right;">
0.136
</td>
<td style="text-align:right;">
0.023
</td>
<td style="text-align:right;">
0.561
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
2 3-butanedione olfactory latency 30 percentile
</td>
<td style="text-align:right;">
0.288
</td>
<td style="text-align:right;">
0.160
</td>
<td style="text-align:right;">
-0.130
</td>
<td style="text-align:right;">
0.136
</td>
<td style="text-align:right;">
0.015
</td>
<td style="text-align:right;">
0.549
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
CHC 6,10-C26:2
</td>
<td style="text-align:right;">
0.280
</td>
<td style="text-align:right;">
0.013
</td>
<td style="text-align:right;">
-0.269
</td>
<td style="text-align:right;">
0.125
</td>
<td style="text-align:right;">
0.028
</td>
<td style="text-align:right;">
0.518
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
DDT resistance mortality
</td>
<td style="text-align:right;">
-0.262
</td>
<td style="text-align:right;">
-0.097
</td>
<td style="text-align:right;">
0.167
</td>
<td style="text-align:right;">
0.128
</td>
<td style="text-align:right;">
-0.509
</td>
<td style="text-align:right;">
-0.004
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr grouplength="10">
<td colspan="8" style="border-bottom: 0;">
<em><strong>Traits measured in males</strong></em>
</td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Rapid cold hardening old
</td>
<td style="text-align:right;">
-0.435
</td>
<td style="text-align:right;">
-0.177
</td>
<td style="text-align:right;">
0.262
</td>
<td style="text-align:right;">
0.159
</td>
<td style="text-align:right;">
-0.730
</td>
<td style="text-align:right;">
-0.109
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Day sleep bout length SD
</td>
<td style="text-align:right;">
-0.383
</td>
<td style="text-align:right;">
-0.177
</td>
<td style="text-align:right;">
0.208
</td>
<td style="text-align:right;">
0.132
</td>
<td style="text-align:right;">
-0.634
</td>
<td style="text-align:right;">
-0.114
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Day sleep bout length
</td>
<td style="text-align:right;">
-0.365
</td>
<td style="text-align:right;">
-0.169
</td>
<td style="text-align:right;">
0.197
</td>
<td style="text-align:right;">
0.133
</td>
<td style="text-align:right;">
-0.615
</td>
<td style="text-align:right;">
-0.096
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Cold shock acclimated old
</td>
<td style="text-align:right;">
-0.358
</td>
<td style="text-align:right;">
-0.189
</td>
<td style="text-align:right;">
0.170
</td>
<td style="text-align:right;">
0.155
</td>
<td style="text-align:right;">
-0.648
</td>
<td style="text-align:right;">
-0.042
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Paraquat resistance
</td>
<td style="text-align:right;">
-0.352
</td>
<td style="text-align:right;">
-0.215
</td>
<td style="text-align:right;">
0.140
</td>
<td style="text-align:right;">
0.165
</td>
<td style="text-align:right;">
-0.662
</td>
<td style="text-align:right;">
-0.014
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Sprinter marathoner
</td>
<td style="text-align:right;">
-0.331
</td>
<td style="text-align:right;">
-0.138
</td>
<td style="text-align:right;">
0.195
</td>
<td style="text-align:right;">
0.145
</td>
<td style="text-align:right;">
-0.608
</td>
<td style="text-align:right;">
-0.039
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Critical thermal max
</td>
<td style="text-align:right;">
-0.324
</td>
<td style="text-align:right;">
-0.166
</td>
<td style="text-align:right;">
0.160
</td>
<td style="text-align:right;">
0.157
</td>
<td style="text-align:right;">
-0.620
</td>
<td style="text-align:right;">
-0.009
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Mating latency
</td>
<td style="text-align:right;">
0.299
</td>
<td style="text-align:right;">
-0.014
</td>
<td style="text-align:right;">
-0.317
</td>
<td style="text-align:right;">
0.127
</td>
<td style="text-align:right;">
0.050
</td>
<td style="text-align:right;">
0.547
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; padding-left: 2em;" indentlevel="1">
Lifespan 28C
</td>
<td style="text-align:right;">
-0.291
</td>
<td style="text-align:right;">
-0.033
</td>
<td style="text-align:right;">
0.258
</td>
<td style="text-align:right;">
0.124
</td>
<td style="text-align:right;">
-0.530
</td>
<td style="text-align:right;">
-0.045
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 21em; font-weight: bold;padding-left: 2em;" indentlevel="1">
Critical thermal min within environment variation plasticity
</td>
<td style="text-align:right;">
0.282
</td>
<td style="text-align:right;">
0.089
</td>
<td style="text-align:right;">
-0.194
</td>
<td style="text-align:right;">
0.128
</td>
<td style="text-align:right;">
0.027
</td>
<td style="text-align:right;">
0.532
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
</tbody>
</table></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="build-figure-3" class="section level2">
<h2>Build Figure 3</h2>
<p>Yep, this is out of order but it makes more sense in the text to
order it as such.</p>
<pre class="r"><code>female_R_diff_plot &lt;-
  R_female_traits %&gt;%
  group_by(Trait) %&gt;%
  mutate(diff_mean = mean(R_diff)) %&gt;%
  ungroup() %&gt;%
  filter(diff_mean &gt; 0.26 | diff_mean &lt; -0.26) %&gt;%
  left_join(clean_meta_data) %&gt;%
  filter(!is.na(Trait_nice)) %&gt;%
  #filter(!str_detect(Trait_nice, &quot;environment&quot;)) %&gt;%
  #filter(!str_detect(Trait_nice, &quot;SD&quot;)) %&gt;%
  select(Response_to_selection_female, Response_to_selection_male, Trait_nice) %&gt;%
  rename(Female = Response_to_selection_female, Male = Response_to_selection_male) %&gt;%
  pivot_longer(cols = 1:2, names_to = &quot;Fitness_Sex&quot;, values_to = &quot;R&quot;) %&gt;%
  mutate(Trait_nice = case_when(Trait_nice == &quot;DDT resistance mortality&quot; ~
                                  &quot;DDT susceptibility&quot;,
                                Trait_nice == &quot;CHC 6,10-C26:2&quot; ~
                                  &quot;CHC 6,10-C26:2 peak&quot;,
                                .default = Trait_nice)) %&gt;% 


  ggplot(aes(x = R, y = Trait_nice, fill = Fitness_Sex)) +
  stat_slab(alpha = 0.9, shape = 21) +#,
  labs(x = expression(&quot;The expected response to selection, &quot;* italic(R)),
       y = NULL,
       fill = &quot;Sex under selection&quot;,
       subtitle = &quot;a. Measured in females&quot;) +
  geom_vline(xintercept = 0, linetype = 2, linewidth = 1.2) +
  scale_fill_manual(values = c(carto_pal(7, &quot;Purp&quot;)[5], carto_pal(7, &quot;Peach&quot;)[5])) +
  coord_cartesian(xlim = c(-0.6, 0.6)) +
  theme_minimal() +
  theme(panel.background = element_rect(fill=&#39;transparent&#39;),
        #panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        plot.background = element_rect(fill=&#39;transparent&#39;, color=NA),
        legend.position = &quot;none&quot;,
        text = element_text(size=12),
        axis.text.y = element_text(size = 9),
        plot.subtitle=element_text(hjust=0.5))

male_R_diff_plot &lt;-
  R_male_traits %&gt;%
  group_by(Trait) %&gt;%
  mutate(diff_mean = mean(R_diff)) %&gt;%
  ungroup() %&gt;%
  filter(diff_mean &gt; 0.28 | diff_mean &lt; -0.28) %&gt;%
  left_join(clean_meta_data) %&gt;%
  filter(!is.na(Trait_nice)) %&gt;%
  #filter(!str_detect(Trait_nice, &quot;environment&quot;)) %&gt;%
  #filter(!str_detect(Trait_nice, &quot;SD&quot;)) %&gt;%
  select(Response_to_selection_female, Response_to_selection_male, Trait_nice) %&gt;%
  rename(Female = Response_to_selection_female, Male = Response_to_selection_male) %&gt;%
  pivot_longer(cols = 1:2, names_to = &quot;Fitness_Sex&quot;, values_to = &quot;R&quot;) %&gt;%


  ggplot(aes(x = R, y = Trait_nice, fill = Fitness_Sex)) +
  stat_slab(alpha = 0.9, shape = 21) +#,
  labs(x = expression(&quot;The expected response to selection, &quot;* italic(R)),
       y = NULL,
       fill = &quot;Sex under selection&quot;,
       subtitle = &quot;b. Measured in males&quot;) +
  geom_vline(xintercept = 0, linetype = 2, linewidth = 1.2) +
  scale_fill_manual(values = c(carto_pal(7, &quot;Purp&quot;)[5], carto_pal(7, &quot;Peach&quot;)[5])) +
  coord_cartesian(xlim = c(-0.6, 0.6)) +
  theme_minimal() +
  theme(panel.background = element_rect(fill=&#39;transparent&#39;),
        #panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        plot.background = element_rect(fill=&#39;transparent&#39;, color=NA),
        legend.position = &quot;bottom&quot;,
        text = element_text(size=12),
        axis.text.y = element_text(size = 9),
        plot.subtitle=element_text(hjust=0.5))

fig3 &lt;- female_R_diff_plot / male_R_diff_plot

fig3</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/319c84c1118d4c1c59279734e1b3432ef1d08901/docs/figure/Main_analysis.Rmd/unnamed-chunk-15-1.png" target="_blank">319c84c</a>
</td>
<td>
tomkeaney
</td>
<td>
2023-10-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure 3</strong>. Candidate traits predicted to have strong
sex-specific responses to selection. The traits shown are those where
the 95% credible interval of the difference between <span
class="math inline">\(R_f\)</span> and <span
class="math inline">\(R_m\)</span> does not overlap zero. Traits are
likely to have sex-specific genetic architectures, therefore we consider
traits measured in females as distinct from traits measured in males,
and <em>vice versa</em>, even if they are putatively the same e.g. mean
lifespan. Hence, mating latency is measured in males (panel b), but
<span class="math inline">\(R\)</span> can be estimated for both sexes,
because we simply need the genetic covariance between trait and fitness.
Detailed definitions of each trait are given in Table S1.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
</div>
<div id="estimate-across-trait-grand-means-for-r-in-each-sex"
class="section level1">
<h1>Estimate across-trait grand means for <span
class="math inline">\(|R|\)</span> in each sex</h1>
<pre class="r"><code># First, fit a simple model to estimate the overall absolute response, |$R_{overall}$| across traits

R_overall_medians &lt;-
  R_all_traits %&gt;% 
  select(Trait, Trait_Sex, R_overall) %&gt;% 
  group_by(Trait, Trait_Sex) %&gt;% 
  summarise_draws(&quot;median&quot;, &quot;sd&quot;, ~quantile(.x, probs = c(0.025, 0.975), na.rm = TRUE)) %&gt;% 
  ungroup() %&gt;% 
  select(-variable) %&gt;% 
  left_join(clean_meta_data) %&gt;% 
  mutate(absolute_R_overall = abs(median),
         Trait_Sex = as.factor(Trait_Sex),
         Trait = as.factor(Trait)) 


R_intercept_model &lt;- 
  brm(absolute_R_overall | weights(1/sd) ~ 1 + Trait_Sex + (1|study_ID),
      family = brmsfamily(family = &quot;Gamma&quot;), # gamma is appropriate for the half-normal distribution created by taking the absolute
      data = R_overall_medians,
      prior = c(prior(normal(-2.2, 1), class = Intercept),
                prior(exponential(1), class = sd),
                prior(exponential(1), class = shape),
                prior(normal(0, 0.25), class = b)),
      warmup = 4000, iter = 12000,
      seed = 1, cores = 4, chains = 4,
      control = list(adapt_delta = 0.9, max_treedepth = 10),
      file = &quot;fits/intercept_R_model&quot;)

print(R_intercept_model)</code></pre>
<pre class="r"><code># Find summary measures to report in results section
new_data &lt;- tibble(Trait_Sex = c(&quot;Female&quot;, &quot;Male&quot;))

  fitted(R_intercept_model, newdata = new_data, re_formula = NA, summary = F) %&gt;% 
  as.data.frame() %&gt;% 
  rename(Female_trait_R = V1, Male_trait_R = V2) %&gt;% 
  as_tibble() %&gt;% 
  mutate(percent_diff = ((Female_trait_R - Male_trait_R) / Male_trait_R)*100) %&gt;% 
  pivot_longer(cols = everything(), names_to = &quot;Parameter&quot;, values_to = &quot;R_mean&quot;) %&gt;% 
  group_by(Parameter) %&gt;% 
  median_qi(R_mean)</code></pre>
<p><span class="math inline">\(~\)</span></p>
<p>Fit the model to test whether <span
class="math inline">\(|R|\)</span> depends on the sex fitness and trait
values were measured in:</p>
<pre class="r"><code>R_medians &lt;-
  R_long_form %&gt;%
  group_by(Trait, Fitness_Sex, Trait_Sex) %&gt;% 
  summarise_draws(&quot;median&quot;, &quot;sd&quot;, ~quantile(.x, probs = c(0.025, 0.975), na.rm = TRUE)) %&gt;% 
  ungroup() %&gt;% 
  select(-variable) %&gt;% 
  left_join(clean_meta_data) %&gt;% 
  mutate(absolute_R = abs(median),
         Fitness_Sex = as.factor(Fitness_Sex),
         Trait_Sex = as.factor(Trait_Sex),
         Trait = as.factor(Trait)) 

median_R_model &lt;- 
  brm(absolute_R | weights(1/sd) ~ 1 + Fitness_Sex * Trait_Sex + (1|study_ID) + (1|Trait),
      family = brmsfamily(family = &quot;Gamma&quot;), # gamma is appropriate for the half-normal distribution created by taking the absolute
      data = R_medians,
      prior = c(prior(normal(-2.2, 1), class = Intercept),
                prior(exponential(1), class = sd),
                prior(exponential(1), class = shape),
                prior(normal(0, 0.25), class = b)),
      warmup = 4000, iter = 12000,
      seed = 1, cores = 4, chains = 4,
      control = list(adapt_delta = 0.9, max_treedepth = 10),
      file = &quot;fits/median_R_model&quot;)

print(median_R_model)</code></pre>
<pre><code> Family: gamma 
  Links: mu = log; shape = identity 
Formula: absolute_R | weights(1/sd) ~ 1 + Fitness_Sex * Trait_Sex + (1 | study_ID) + (1 | Trait) 
   Data: R_medians (Number of observations: 950) 
  Draws: 4 chains, each with iter = 12000; warmup = 4000; thin = 1;
         total post-warmup draws = 32000

Group-Level Effects: 
~study_ID (Number of levels: 76) 
              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(Intercept)     0.32      0.05     0.23     0.43 1.00     6142    10131

~Trait (Number of levels: 475) 
              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(Intercept)     0.59      0.02     0.54     0.63 1.00     8421    12837

Population-Level Effects: 
                              Estimate Est.Error l-95% CI u-95% CI Rhat
Intercept                        -2.70      0.06    -2.82    -2.58 1.00
Fitness_SexMale                   0.18      0.02     0.14     0.22 1.00
Trait_SexMale                    -0.15      0.06    -0.27    -0.03 1.00
Fitness_SexMale:Trait_SexMale     0.11      0.03     0.05     0.17 1.00
                              Bulk_ESS Tail_ESS
Intercept                        16404    18313
Fitness_SexMale                  36862    23998
Trait_SexMale                    11981    15341
Fitness_SexMale:Trait_SexMale    34545    23661

Family Specific Parameters: 
      Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
shape     2.42      0.03     2.36     2.49 1.00    39872    21791

Draws were sampled using sampling(NUTS). For each parameter, Bulk_ESS
and Tail_ESS are effective sample size measures, and Rhat is the potential
scale reduction factor on split chains (at convergence, Rhat = 1).</code></pre>
<p><span class="math inline">\(~\)</span></p>
<div id="build-panels-a-and-b-for-figure-2" class="section level2">
<h2>Build Panels a and b for Figure 2</h2>
<p>Get model predictions and plot</p>
<pre class="r"><code>new_data &lt;- expand_grid(Fitness_Sex = c(&quot;Female&quot;, &quot;Male&quot;),
                        Trait_Sex = c(&quot;Female&quot;, &quot;Male&quot;))

R_fitted &lt;-
  fitted(median_R_model, newdata = new_data, re_formula = NA, summary = F) %&gt;% 
  as.data.frame() %&gt;% 
  rename(FemaleFitness_FemaleTrait = V1, FemaleFitness_MaleTrait = V2, 
         MaleFitness_FemaleTrait = V3, MaleFitness_MaleTrait = V4) %&gt;% 
  as_tibble() %&gt;% 
  mutate(percent_diff_female = ((MaleFitness_FemaleTrait - FemaleFitness_FemaleTrait) / FemaleFitness_FemaleTrait)*100,
         percent_diff_male = ((MaleFitness_MaleTrait - FemaleFitness_MaleTrait)/ FemaleFitness_MaleTrait)*100) %&gt;% 
  pivot_longer(cols = everything(), names_to = &quot;Parameter&quot;, values_to = &quot;R_mean&quot;) %&gt;% 
  mutate(Fitness_Sex = case_when(str_detect(Parameter, &quot;FemaleFitness&quot;) ~ &quot;Female&quot;,
                                 str_detect(Parameter, &quot;MaleFitness&quot;) ~ &quot;Male&quot;),
         Trait_Sex = case_when(str_detect(Parameter, &quot;FemaleTrait&quot;) ~ &quot;Female&quot;,
                               str_detect(Parameter, &quot;MaleTrait&quot;) ~ &quot;Male&quot;,
                               str_detect(Parameter, &quot;percent_diff_female&quot;) ~ &quot;Female&quot;,
                               str_detect(Parameter, &quot;percent_diff_male&quot;) ~ &quot;Male&quot;))


R_p1 &lt;-
  R_fitted %&gt;% 
  filter(!str_detect(Parameter, &quot;percent&quot;)) %&gt;% 
  ggplot(aes(x = R_mean, y = Trait_Sex, fill = Fitness_Sex)) + #fill = Fitness_Sex)) +
  stat_slab(alpha = 0.9, shape = 21) +#,
  labs(x = expression(&quot;Across-trait estimated magnitude for |&quot;* italic(R) * &quot;|&quot;),
       y = &quot;Sex traits were measured in&quot;,
       fill = &quot;Sex under selection&quot;) +
  scale_fill_manual(values = c(carto_pal(7, &quot;Purp&quot;)[5], carto_pal(7, &quot;Peach&quot;)[5])) +
  theme_minimal() + 
  theme(panel.background = element_rect(fill=&#39;transparent&#39;),
        #panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        plot.background = element_rect(fill=&#39;transparent&#39;, color=NA),
        legend.position = &quot;bottom&quot;,
        text = element_text(size=13))

R_p2 &lt;-
  R_fitted %&gt;% 
  filter(str_detect(Parameter, &quot;percent&quot;)) %&gt;%
  ggplot(aes(x = R_mean, y = Trait_Sex, fill = Trait_Sex)) +
  stat_halfeye(.width = c(0.66, 0.95), alpha = 0.9,
               point_interval = &quot;median_qi&quot;, point_fill = &quot;white&quot;,
               shape = 21, point_size = 4, stroke = 1.5) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals + # width indicates the uncertainty intervals: here we have 66% and 95% intervals+
  scale_fill_manual(values = c(carto_pal(7, &quot;Peach&quot;)[5], carto_pal(7, &quot;Peach&quot;)[5], carto_pal(7, &quot;Purp&quot;)[5])) +
  geom_vline(xintercept = 0, linetype = 2, linewidth = 1.2) +
  coord_cartesian(xlim = c(-5, 50)) +
  xlab(expression(&quot;% increase in |&quot;*italic(R)* &quot;| through males&quot;)) +
  ylab(&quot;Sex traits were measured in&quot;) +
  theme_minimal() + 
  theme(panel.background = element_rect(fill=&#39;transparent&#39;),
        #panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        plot.background = element_rect(fill=&#39;transparent&#39;, color=NA),
        legend.position = &quot;none&quot;,
        text = element_text(size=13))

R_p1 + R_p2 +
  plot_annotation(tag_levels = &#39;a&#39;)</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-19-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/319c84c1118d4c1c59279734e1b3432ef1d08901/docs/figure/Main_analysis.Rmd/unnamed-chunk-19-1.png" target="_blank">319c84c</a>
</td>
<td>
tomkeaney
</td>
<td>
2023-10-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/daa935de0473e2ecaedca3c65f414d2858ddc598/docs/figure/Main_analysis.Rmd/unnamed-chunk-19-1.png" target="_blank">daa935d</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure 2</strong>. <strong>a</strong> The posterior
distribution for across-trait |<span class="math inline">\(R\)</span>|,
the mean expected response to selection (absolute value) in females and
males and split by the sex the trait was measured in. For each trait
<span class="math inline">\(R = \sigma(A_w, A_z)\)</span>, where Aw is
the breeding value for fitness and <span
class="math inline">\(A_z\)</span> is the trait breeding value. |<span
class="math inline">\(R\)</span>| is expressed in trait standard
deviations. <strong>b</strong> The percentage increase in |<span
class="math inline">\(R\)</span>| through males compared to through
females. Points indicate the estimated median with associated 66 and 95%
credible intervals.</p>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Calculating stats for the results section</strong></p>
<pre class="r"><code>R_fitted %&gt;% 
  group_by(Parameter) %&gt;% 
  median_qi(R_mean)</code></pre>
<pre><code># A tibble: 6 × 7
  Parameter                  R_mean  .lower  .upper .width .point .interval
  &lt;chr&gt;                       &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;    
1 FemaleFitness_FemaleTrait  0.0673  0.0598  0.0761   0.95 median qi       
2 FemaleFitness_MaleTrait    0.0579  0.0513  0.0653   0.95 median qi       
3 MaleFitness_FemaleTrait    0.0808  0.0716  0.0915   0.95 median qi       
4 MaleFitness_MaleTrait      0.0776  0.0689  0.0876   0.95 median qi       
5 percent_diff_female       20.1    15.3    25.1      0.95 median qi       
6 percent_diff_male         34.1    28.7    39.7      0.95 median qi       </code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
</div>
<div
id="identifying-traits-responding-to-sexually-antagonistic-selection"
class="section level1">
<h1>Identifying traits responding to sexually antagonistic
selection</h1>
<div id="innocenti-and-morrows-i-index" class="section level2">
<h2>Innocenti and Morrow’s I index</h2>
<p><span class="math inline">\(~\)</span></p>
<p>We have found <span class="math inline">\(R\)</span> for traits in
females and males. For those traits where we have a measure of both, we
can estimate the concordance of the response to selection between the
sexes using the <span class="math inline">\(I\)</span> metric presented
in <a href="https://doi.org/10.1111/j.1558-5646.2010.01021.x">Innocenti
and Morrow (2011)</a>.</p>
<p><span class="math display">\[I = \frac{\beta&#39;_M
\beta&#39;_F}{\sqrt{\frac{(\beta&#39;_M)^2 +
(\beta&#39;_F)^2}{2}}}\]</span></p>
<p>Where <span class="math inline">\(\beta&#39;_F\)</span> and <span
class="math inline">\(\beta&#39;_M\)</span> = are phenotypic selection
gradients. Taking traits expressed in females as an example, we
substitute <span class="math inline">\(R_FF\)</span> and <span
class="math inline">\(R_MF\)</span> for <span
class="math inline">\(\beta&#39;_F\)</span> and <span
class="math inline">\(\beta&#39;_M\)</span>. This makes our test for
sexually antagonistic traits more conservative, as non-zero <span
class="math inline">\(R\)</span> requires both directional selection on
the trait and the presence of additive genetic variance for the
trait.</p>
<p>This index is positive when selection is concordant in the two sexes,
negative when antagonistic in the two sexes, and is zero when selection
is absent in one sex (note that it will miss conflict that occurs when
strong stabilizing selection is present in that sex). Finally it is
proportional to the absolute intensity of selection (track the diagonals
in the visualisation below to get a sense of this). Note that because we
use <span class="math inline">\(R\)</span> instead of <span
class="math inline">\(\beta&#39;\)</span>, <span
class="math inline">\(I\)</span> is bounded between -1 to 1 in our
analysis.</p>
<p>Additionally, it is symmetrical and normally distributed for a random
set of normally distributed <span
class="math inline">\(\beta&#39;_M\)</span>, <span
class="math inline">\(\beta&#39;_F\)</span>. <span
class="math inline">\(|I|\)</span> is also always included in the
interval between the absolute values of the selection gradient in the
two sexes, and it coincides with them when <span
class="math inline">\(\beta&#39;_M = \beta&#39;_F\)</span>. As a
potential drawback, it should be noted that such quantity is not defined
when <span class="math inline">\(\beta&#39;_M = \beta&#39;_F =
0\)</span>.</p>
<p>First let’s look at the <span class="math inline">\(I\)</span>
parameter space. We simulate combinations of <span
class="math inline">\(R_F\)</span> and <span
class="math inline">\(R_M\)</span> and calculate <span
class="math inline">\(I\)</span>.</p>
<pre class="r"><code># Let&#39;s test that the I index behaves as it should

sim_I &lt;-
  expand_grid(R_F = seq(from = -0.5, to = 0.5, by = 0.01),
         R_M = seq(from = -0.5, to = 0.5, by = 0.01)) %&gt;% 
  mutate(I = R_F * R_M / sqrt(((R_F)^2 + (R_M)^2)/2))

I_plot &lt;-
  sim_I %&gt;% 
  ggplot(aes(x = R_F, y = R_M, fill = I)) +
  geom_raster() +
  stat_contour(aes(z = I), colour = &quot;black&quot;, binwidth = 0.075) +
  stat_contour(aes(z = I), colour = &quot;black&quot;, breaks = 8, linetype = 2) +
  scale_fill_gradientn(colours = met.brewer(name = &quot;Paquin&quot;, 7),
                       labels = c(&quot;Antagonistic&quot;, -0.25, 0, 0.25, &quot;Concordant&quot;)) +
  scale_x_continuous(expand = c(0, 0)) + 
  scale_y_continuous(expand = c(0, 0)) +
  labs(x = &quot;Response to selection in females&quot;, y = &quot;Response to selection in males&quot;, fill = &quot;I index&quot;) +
  theme_tidybayes() +
  theme(text = element_text(size = 17),
        legend.position = &quot;right&quot;,
        legend.direction = &quot;vertical&quot;)

I_plot</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/319c84c1118d4c1c59279734e1b3432ef1d08901/docs/figure/Main_analysis.Rmd/unnamed-chunk-21-1.png" target="_blank">319c84c</a>
</td>
<td>
tomkeaney
</td>
<td>
2023-10-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/daa935de0473e2ecaedca3c65f414d2858ddc598/docs/figure/Main_analysis.Rmd/unnamed-chunk-21-1.png" target="_blank">daa935d</a>
</td>
<td>
ausevo
</td>
<td>
2023-03-08
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Calculate <span class="math inline">\(I\)</span> for each draw of
each trait</p>
<p><strong>Table S4</strong>. The traits with the most concordant and
antagonistic <em>I</em> index values in the DGRP. Estimated error is the
standard deviation of the posterior distribution. The notable column
indicates whether the 95% CI overlaps with zero.</p>
<pre class="r"><code>I_index &lt;-
  R_all_traits %&gt;% 
  mutate(I = Response_to_selection_female * Response_to_selection_male / 
           sqrt(((Response_to_selection_female)^2 + (Response_to_selection_male)^2)/2))
# make the tables

bind_cols(
  bind_rows(
    I_index %&gt;% 
      filter(Trait_Sex == &quot;Female&quot;) %&gt;% 
      select(Trait, I) %&gt;% 
      group_by(Trait) %&gt;% 
      summarise_draws(&quot;median&quot;, &quot;sd&quot;, ~quantile(.x, probs = c(0.025, 0.975), na.rm = TRUE)) %&gt;%
      ungroup() %&gt;% 
      rename(I = median, `Estimated error` = sd, `2.5% CI` = `2.5%`, `97.5% CI` = `97.5%`)  %&gt;%
      arrange(-I) %&gt;% 
      slice_head(n = 20) %&gt;% 
      select(-variable) %&gt;% 
      mutate(across(I:`97.5% CI`, ~ round(.x, digits = 3)),
             `Notable` = case_when(`2.5% CI` &lt; 0 &amp; `97.5% CI` &lt; 0 ~ &quot;*&quot;,
                                   `2.5% CI` &gt; 0 &amp; `97.5% CI` &gt; 0 ~ &quot;*&quot;,
                                   TRUE ~ &quot; &quot;)) %&gt;% #, 
      left_join(trait_data %&gt;% distinct(Trait, Trait_nice)) %&gt;% # add the nice names 
      rename(`Trait in females` = Trait_nice) %&gt;% 
      select(`Trait in females`, everything(), -Trait),
    
    
    I_index %&gt;% 
      filter(Trait_Sex == &quot;Female&quot;) %&gt;% 
      select(Trait, I) %&gt;% 
      group_by(Trait) %&gt;% 
      summarise_draws(&quot;median&quot;, &quot;sd&quot;, ~quantile(.x, probs = c(0.025, 0.975), na.rm = TRUE)) %&gt;%
      ungroup() %&gt;% 
      rename(I = median, `Estimated error` = sd, `2.5% CI` = `2.5%`, `97.5% CI` = `97.5%`)  %&gt;%
      arrange(I) %&gt;% 
      slice_head(n = 20) %&gt;% 
      select(-variable) %&gt;% 
      mutate(across(I:`97.5% CI`, ~ round(.x, digits = 3)),
             `Notable` = case_when(`2.5% CI` &lt; 0 &amp; `97.5% CI` &lt; 0 ~ &quot;*&quot;,
                                   `2.5% CI` &gt; 0 &amp; `97.5% CI` &gt; 0 ~ &quot;*&quot;,
                                   TRUE ~ &quot; &quot;)) %&gt;% 
      left_join(trait_data %&gt;% distinct(Trait, Trait_nice)) %&gt;% # add the nice names 
      rename(`Trait in females` = Trait_nice) %&gt;% 
      select(`Trait in females`, everything(), -Trait) 
  ),
  
  bind_rows(
    I_index %&gt;% 
      filter(Trait_Sex == &quot;Male&quot;) %&gt;% 
      select(Trait, I) %&gt;% 
      group_by(Trait) %&gt;% 
      summarise_draws(&quot;median&quot;, &quot;sd&quot;, ~quantile(.x, probs = c(0.025, 0.975), na.rm = TRUE)) %&gt;%
      ungroup() %&gt;% 
      rename(`I ` = median, `Estimated error ` = sd, `2.5% CI ` = `2.5%`, `97.5% CI ` = `97.5%`)  %&gt;%
      arrange(-`I `) %&gt;% 
      slice_head(n = 20) %&gt;% 
      select(-variable) %&gt;% 
      mutate(across(`I `:`97.5% CI `, ~ round(.x, digits = 3)),
             `Notable ` = case_when(`2.5% CI ` &lt; 0 &amp; `97.5% CI ` &lt; 0 ~ &quot;*&quot;,
                                    `2.5% CI ` &gt; 0 &amp; `97.5% CI ` &gt; 0 ~ &quot;*&quot;,
                                    TRUE ~ &quot; &quot;)) %&gt;% 
      left_join(trait_data %&gt;% distinct(Trait, Trait_nice)) %&gt;% # add the nice names 
      rename(`Trait in males` = Trait_nice) %&gt;% 
      select(`Trait in males`, everything(), -Trait), 
    
    I_index %&gt;% 
      filter(Trait_Sex == &quot;Male&quot;) %&gt;% 
      select(Trait, I) %&gt;%  
      group_by(Trait) %&gt;% 
      summarise_draws(&quot;median&quot;, &quot;sd&quot;, ~quantile(.x, probs = c(0.025, 0.975), na.rm = TRUE)) %&gt;%
      ungroup() %&gt;% 
      select(-variable) %&gt;% 
      rename(`I ` = median, `Estimated error ` = sd, `2.5% CI ` = `2.5%`, `97.5% CI ` = `97.5%`)  %&gt;%
      arrange(`I `) %&gt;% 
      slice_head(n = 20) %&gt;% 
      mutate(across(`I `:`97.5% CI `, ~ round(.x, digits = 3)),
             `Notable ` = case_when(`2.5% CI ` &lt; 0 &amp; `97.5% CI ` &lt; 0 ~ &quot;*&quot;,
                                    `2.5% CI ` &gt; 0 &amp; `97.5% CI ` &gt; 0 ~ &quot;*&quot;,
                                    TRUE ~ &quot; &quot;)) %&gt;% 
      left_join(trait_data %&gt;% distinct(Trait, Trait_nice)) %&gt;% # add the nice names 
      rename(`Trait in males` = Trait_nice) %&gt;% 
      select(`Trait in males`, everything(), -Trait) #%&gt;% 
      #filter(!is.na(`Trait in males`))
  )
) %&gt;% 
  
  
  
  kbl() %&gt;%
  kable_minimal(full_width = F, font_size = 12) %&gt;% 
  column_spec(1, width = &quot;27em&quot;) %&gt;% 
  pack_rows(&quot;Sexually concordant R&quot;, 1, 20, italic = TRUE) %&gt;% 
  pack_rows(&quot;Sexually antagonistic R&quot;, 21, 40, italic = TRUE) %&gt;% 
  column_spec(7, width = &quot;27em&quot;)</code></pre>
<table class=" lightable-minimal" style='font-size: 12px; font-family: "Trebuchet MS", verdana, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;'>
<thead>
<tr>
<th style="text-align:left;">
Trait in females
</th>
<th style="text-align:right;">
I
</th>
<th style="text-align:right;">
Estimated error
</th>
<th style="text-align:right;">
2.5% CI
</th>
<th style="text-align:right;">
97.5% CI
</th>
<th style="text-align:left;">
Notable
</th>
<th style="text-align:left;">
Trait in males
</th>
<th style="text-align:right;">
I
</th>
<th style="text-align:right;">
Estimated error
</th>
<th style="text-align:right;">
2.5% CI
</th>
<th style="text-align:right;">
97.5% CI
</th>
<th style="text-align:left;">
Notable
</th>
</tr>
</thead>
<tbody>
<tr grouplength="20">
<td colspan="12" style="border-bottom: 0;">
<em><strong>Sexually concordant R</strong></em>
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
M. anisopliae resistance
</td>
<td style="text-align:right;">
0.214
</td>
<td style="text-align:right;">
0.074
</td>
<td style="text-align:right;">
0.046
</td>
<td style="text-align:right;">
0.339
</td>
<td style="text-align:left;">
<ul>
<li></td>
<td style="text-align:left;width: 27em; ">
Cold shock young
</td>
<td style="text-align:right;">
0.222
</td>
<td style="text-align:right;">
0.099
</td>
<td style="text-align:right;">
-0.014
</td>
<td style="text-align:right;">
0.382
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Locomotor frequency
</td>
<td style="text-align:right;">
0.209
</td>
<td style="text-align:right;">
0.132
</td>
<td style="text-align:right;">
-0.118
</td>
<td style="text-align:right;">
0.401
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
M. anisopliae resistance
</td>
<td style="text-align:right;">
0.212
</td>
<td style="text-align:right;">
0.076
</td>
<td style="text-align:right;">
0.036
</td>
<td style="text-align:right;">
0.338
</td>
<td style="text-align:left;">
<ul>
<li></td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Cold shock young
</td>
<td style="text-align:right;">
0.203
</td>
<td style="text-align:right;">
0.104
</td>
<td style="text-align:right;">
-0.050
</td>
<td style="text-align:right;">
0.369
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
P. aeruginosa resistance
</td>
<td style="text-align:right;">
0.197
</td>
<td style="text-align:right;">
0.116
</td>
<td style="text-align:right;">
-0.081
</td>
<td style="text-align:right;">
0.383
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
P. aeruginosa resistance
</td>
<td style="text-align:right;">
0.174
</td>
<td style="text-align:right;">
0.122
</td>
<td style="text-align:right;">
-0.106
</td>
<td style="text-align:right;">
0.370
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
P. rettgeri load high glucose diet
</td>
<td style="text-align:right;">
0.182
</td>
<td style="text-align:right;">
0.100
</td>
<td style="text-align:right;">
-0.065
</td>
<td style="text-align:right;">
0.334
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
CHC 9,13-C25:2
</td>
<td style="text-align:right;">
0.154
</td>
<td style="text-align:right;">
0.088
</td>
<td style="text-align:right;">
-0.056
</td>
<td style="text-align:right;">
0.298
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Triglyceride gnotobiotic
</td>
<td style="text-align:right;">
0.181
</td>
<td style="text-align:right;">
0.135
</td>
<td style="text-align:right;">
-0.144
</td>
<td style="text-align:right;">
0.382
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
CHC 6-C23:1
</td>
<td style="text-align:right;">
0.152
</td>
<td style="text-align:right;">
0.085
</td>
<td style="text-align:right;">
-0.047
</td>
<td style="text-align:right;">
0.290
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Olfactory response ethylacetate
</td>
<td style="text-align:right;">
0.146
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
-0.052
</td>
<td style="text-align:right;">
0.279
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Waking activity SD
</td>
<td style="text-align:right;">
0.144
</td>
<td style="text-align:right;">
0.095
</td>
<td style="text-align:right;">
-0.070
</td>
<td style="text-align:right;">
0.301
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Flight performance
</td>
<td style="text-align:right;">
0.143
</td>
<td style="text-align:right;">
0.081
</td>
<td style="text-align:right;">
-0.045
</td>
<td style="text-align:right;">
0.275
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Negative geotaxis msb
</td>
<td style="text-align:right;">
0.143
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
-0.046
</td>
<td style="text-align:right;">
0.278
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Day sleep SD
</td>
<td style="text-align:right;">
0.137
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
-0.054
</td>
<td style="text-align:right;">
0.274
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Lifetime fecundity 25C
</td>
<td style="text-align:right;">
0.143
</td>
<td style="text-align:right;">
0.098
</td>
<td style="text-align:right;">
-0.089
</td>
<td style="text-align:right;">
0.293
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Parathion resistance
</td>
<td style="text-align:right;">
0.124
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
-0.066
</td>
<td style="text-align:right;">
0.261
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Phototaxis mid age
</td>
<td style="text-align:right;">
0.136
</td>
<td style="text-align:right;">
0.086
</td>
<td style="text-align:right;">
-0.070
</td>
<td style="text-align:right;">
0.275
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Glucose low glucose diet
</td>
<td style="text-align:right;">
0.120
</td>
<td style="text-align:right;">
0.103
</td>
<td style="text-align:right;">
-0.114
</td>
<td style="text-align:right;">
0.282
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Cold shock old
</td>
<td style="text-align:right;">
0.129
</td>
<td style="text-align:right;">
0.118
</td>
<td style="text-align:right;">
-0.138
</td>
<td style="text-align:right;">
0.315
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Circadian period mesa
</td>
<td style="text-align:right;">
0.118
</td>
<td style="text-align:right;">
0.097
</td>
<td style="text-align:right;">
-0.108
</td>
<td style="text-align:right;">
0.272
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Week1 fecundity 25C
</td>
<td style="text-align:right;">
0.121
</td>
<td style="text-align:right;">
0.099
</td>
<td style="text-align:right;">
-0.107
</td>
<td style="text-align:right;">
0.279
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Circadian period chi
</td>
<td style="text-align:right;">
0.112
</td>
<td style="text-align:right;">
0.100
</td>
<td style="text-align:right;">
-0.111
</td>
<td style="text-align:right;">
0.274
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
CHC x-C23:1
</td>
<td style="text-align:right;">
0.117
</td>
<td style="text-align:right;">
0.087
</td>
<td style="text-align:right;">
-0.079
</td>
<td style="text-align:right;">
0.262
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Protein gnotobiotic
</td>
<td style="text-align:right;">
0.110
</td>
<td style="text-align:right;">
0.129
</td>
<td style="text-align:right;">
-0.172
</td>
<td style="text-align:right;">
0.320
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
CHC n-C21
</td>
<td style="text-align:right;">
0.116
</td>
<td style="text-align:right;">
0.087
</td>
<td style="text-align:right;">
-0.078
</td>
<td style="text-align:right;">
0.261
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
CHC 2-Me-C22
</td>
<td style="text-align:right;">
0.110
</td>
<td style="text-align:right;">
0.097
</td>
<td style="text-align:right;">
-0.109
</td>
<td style="text-align:right;">
0.265
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Startle response msb
</td>
<td style="text-align:right;">
0.113
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
-0.072
</td>
<td style="text-align:right;">
0.251
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Activity
</td>
<td style="text-align:right;">
0.106
</td>
<td style="text-align:right;">
0.087
</td>
<td style="text-align:right;">
-0.092
</td>
<td style="text-align:right;">
0.247
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Fecundity
</td>
<td style="text-align:right;">
0.113
</td>
<td style="text-align:right;">
0.093
</td>
<td style="text-align:right;">
-0.097
</td>
<td style="text-align:right;">
0.265
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Night sleep SD
</td>
<td style="text-align:right;">
0.106
</td>
<td style="text-align:right;">
0.089
</td>
<td style="text-align:right;">
-0.091
</td>
<td style="text-align:right;">
0.252
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Abdomen length 28C
</td>
<td style="text-align:right;">
0.112
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
-0.075
</td>
<td style="text-align:right;">
0.250
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Climbing ability
</td>
<td style="text-align:right;">
0.097
</td>
<td style="text-align:right;">
0.118
</td>
<td style="text-align:right;">
-0.155
</td>
<td style="text-align:right;">
0.294
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
CHC 2-Me-C24
</td>
<td style="text-align:right;">
0.111
</td>
<td style="text-align:right;">
0.093
</td>
<td style="text-align:right;">
-0.101
</td>
<td style="text-align:right;">
0.265
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Negative geotaxis msb
</td>
<td style="text-align:right;">
0.094
</td>
<td style="text-align:right;">
0.087
</td>
<td style="text-align:right;">
-0.100
</td>
<td style="text-align:right;">
0.240
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
CHC NI 34
</td>
<td style="text-align:right;">
0.110
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
-0.078
</td>
<td style="text-align:right;">
0.246
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Activity MPH
</td>
<td style="text-align:right;">
0.093
</td>
<td style="text-align:right;">
0.092
</td>
<td style="text-align:right;">
-0.110
</td>
<td style="text-align:right;">
0.243
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
CHC n-C23
</td>
<td style="text-align:right;">
0.110
</td>
<td style="text-align:right;">
0.086
</td>
<td style="text-align:right;">
-0.084
</td>
<td style="text-align:right;">
0.252
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
CHC NI 54
</td>
<td style="text-align:right;">
0.091
</td>
<td style="text-align:right;">
0.089
</td>
<td style="text-align:right;">
-0.103
</td>
<td style="text-align:right;">
0.239
</td>
<td style="text-align:left;">
</td>
</tr>
<tr grouplength="20">
<td colspan="12" style="border-bottom: 0;">
<em><strong>Sexually antagonistic R</strong></em>
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Day sleep bout length
</td>
<td style="text-align:right;">
-0.189
</td>
<td style="text-align:right;">
0.088
</td>
<td style="text-align:right;">
-0.328
</td>
<td style="text-align:right;">
0.026
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Rapid cold hardening old
</td>
<td style="text-align:right;">
-0.180
</td>
<td style="text-align:right;">
0.111
</td>
<td style="text-align:right;">
-0.354
</td>
<td style="text-align:right;">
0.086
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Sprinter marathoner
</td>
<td style="text-align:right;">
-0.183
</td>
<td style="text-align:right;">
0.092
</td>
<td style="text-align:right;">
-0.333
</td>
<td style="text-align:right;">
0.035
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Day sleep bout length SD
</td>
<td style="text-align:right;">
-0.166
</td>
<td style="text-align:right;">
0.087
</td>
<td style="text-align:right;">
-0.307
</td>
<td style="text-align:right;">
0.039
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Day sleep bout length SD
</td>
<td style="text-align:right;">
-0.163
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
-0.299
</td>
<td style="text-align:right;">
0.033
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Day sleep bout length
</td>
<td style="text-align:right;">
-0.156
</td>
<td style="text-align:right;">
0.088
</td>
<td style="text-align:right;">
-0.298
</td>
<td style="text-align:right;">
0.055
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Critical thermal max
</td>
<td style="text-align:right;">
-0.149
</td>
<td style="text-align:right;">
0.123
</td>
<td style="text-align:right;">
-0.336
</td>
<td style="text-align:right;">
0.135
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Cold shock acclimated old
</td>
<td style="text-align:right;">
-0.144
</td>
<td style="text-align:right;">
0.101
</td>
<td style="text-align:right;">
-0.311
</td>
<td style="text-align:right;">
0.086
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
2 3-butanedione olfactory latency 30 percentile
</td>
<td style="text-align:right;">
-0.112
</td>
<td style="text-align:right;">
0.090
</td>
<td style="text-align:right;">
-0.261
</td>
<td style="text-align:right;">
0.086
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Paraquat resistance
</td>
<td style="text-align:right;">
-0.133
</td>
<td style="text-align:right;">
0.113
</td>
<td style="text-align:right;">
-0.315
</td>
<td style="text-align:right;">
0.124
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
2 3-butanedione olfactory response intensity
</td>
<td style="text-align:right;">
-0.101
</td>
<td style="text-align:right;">
0.099
</td>
<td style="text-align:right;">
-0.262
</td>
<td style="text-align:right;">
0.118
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Sprinter marathoner
</td>
<td style="text-align:right;">
-0.131
</td>
<td style="text-align:right;">
0.099
</td>
<td style="text-align:right;">
-0.290
</td>
<td style="text-align:right;">
0.100
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
DDT resistance mortality
</td>
<td style="text-align:right;">
-0.094
</td>
<td style="text-align:right;">
0.089
</td>
<td style="text-align:right;">
-0.240
</td>
<td style="text-align:right;">
0.105
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Critical thermal max
</td>
<td style="text-align:right;">
-0.125
</td>
<td style="text-align:right;">
0.103
</td>
<td style="text-align:right;">
-0.297
</td>
<td style="text-align:right;">
0.104
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
CHC 8,12-C28:2
</td>
<td style="text-align:right;">
-0.086
</td>
<td style="text-align:right;">
0.087
</td>
<td style="text-align:right;">
-0.234
</td>
<td style="text-align:right;">
0.099
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Critical thermal min within environment variation plasticity
</td>
<td style="text-align:right;">
-0.097
</td>
<td style="text-align:right;">
0.093
</td>
<td style="text-align:right;">
-0.250
</td>
<td style="text-align:right;">
0.113
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Lifespan 18C
</td>
<td style="text-align:right;">
-0.082
</td>
<td style="text-align:right;">
0.081
</td>
<td style="text-align:right;">
-0.219
</td>
<td style="text-align:right;">
0.092
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Day sleep
</td>
<td style="text-align:right;">
-0.094
</td>
<td style="text-align:right;">
0.096
</td>
<td style="text-align:right;">
-0.250
</td>
<td style="text-align:right;">
0.118
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
CHC 7-C26:1
</td>
<td style="text-align:right;">
-0.071
</td>
<td style="text-align:right;">
0.089
</td>
<td style="text-align:right;">
-0.225
</td>
<td style="text-align:right;">
0.116
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Lead effect locomotor activity
</td>
<td style="text-align:right;">
-0.083
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
-0.225
</td>
<td style="text-align:right;">
0.098
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
2 3-butanedione olfactory strength attraction
</td>
<td style="text-align:right;">
-0.068
</td>
<td style="text-align:right;">
0.088
</td>
<td style="text-align:right;">
-0.222
</td>
<td style="text-align:right;">
0.112
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Triglyceride low glucose diet
</td>
<td style="text-align:right;">
-0.077
</td>
<td style="text-align:right;">
0.090
</td>
<td style="text-align:right;">
-0.235
</td>
<td style="text-align:right;">
0.110
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Day sleep bouts
</td>
<td style="text-align:right;">
-0.062
</td>
<td style="text-align:right;">
0.088
</td>
<td style="text-align:right;">
-0.217
</td>
<td style="text-align:right;">
0.120
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Day sleep bouts
</td>
<td style="text-align:right;">
-0.069
</td>
<td style="text-align:right;">
0.100
</td>
<td style="text-align:right;">
-0.235
</td>
<td style="text-align:right;">
0.144
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
CHC 3-Me-C25 &amp; 9-C26:1
</td>
<td style="text-align:right;">
-0.060
</td>
<td style="text-align:right;">
0.086
</td>
<td style="text-align:right;">
-0.212
</td>
<td style="text-align:right;">
0.114
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
CHC 5-C23:1
</td>
<td style="text-align:right;">
-0.066
</td>
<td style="text-align:right;">
0.087
</td>
<td style="text-align:right;">
-0.217
</td>
<td style="text-align:right;">
0.114
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Week5 fecundity 25C
</td>
<td style="text-align:right;">
-0.054
</td>
<td style="text-align:right;">
0.088
</td>
<td style="text-align:right;">
-0.212
</td>
<td style="text-align:right;">
0.124
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Wing centroid size _Houle
</td>
<td style="text-align:right;">
-0.066
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
-0.208
</td>
<td style="text-align:right;">
0.107
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
CHC NI 20
</td>
<td style="text-align:right;">
-0.054
</td>
<td style="text-align:right;">
0.083
</td>
<td style="text-align:right;">
-0.205
</td>
<td style="text-align:right;">
0.112
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Lifespan 18C
</td>
<td style="text-align:right;">
-0.066
</td>
<td style="text-align:right;">
0.080
</td>
<td style="text-align:right;">
-0.206
</td>
<td style="text-align:right;">
0.099
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
CHC 5-C24:1
</td>
<td style="text-align:right;">
-0.051
</td>
<td style="text-align:right;">
0.099
</td>
<td style="text-align:right;">
-0.217
</td>
<td style="text-align:right;">
0.150
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Olfactory response benzaldehyde
</td>
<td style="text-align:right;">
-0.065
</td>
<td style="text-align:right;">
0.086
</td>
<td style="text-align:right;">
-0.220
</td>
<td style="text-align:right;">
0.109
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
2 3-butanedione olfactory latency 90 percentile
</td>
<td style="text-align:right;">
-0.051
</td>
<td style="text-align:right;">
0.087
</td>
<td style="text-align:right;">
-0.207
</td>
<td style="text-align:right;">
0.124
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Critical thermal min developmental stability
</td>
<td style="text-align:right;">
-0.056
</td>
<td style="text-align:right;">
0.082
</td>
<td style="text-align:right;">
-0.202
</td>
<td style="text-align:right;">
0.110
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Lifespan 25C
</td>
<td style="text-align:right;">
-0.048
</td>
<td style="text-align:right;">
0.085
</td>
<td style="text-align:right;">
-0.197
</td>
<td style="text-align:right;">
0.125
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
L. monocytogenes load
</td>
<td style="text-align:right;">
-0.056
</td>
<td style="text-align:right;">
0.100
</td>
<td style="text-align:right;">
-0.241
</td>
<td style="text-align:right;">
0.140
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
Night sleep bout length
</td>
<td style="text-align:right;">
-0.048
</td>
<td style="text-align:right;">
0.085
</td>
<td style="text-align:right;">
-0.203
</td>
<td style="text-align:right;">
0.122
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Critical thermal min
</td>
<td style="text-align:right;">
-0.051
</td>
<td style="text-align:right;">
0.115
</td>
<td style="text-align:right;">
-0.255
</td>
<td style="text-align:right;">
0.181
</td>
<td style="text-align:left;">
</td>
</tr>
<tr>
<td style="text-align:left;width: 27em; padding-left: 2em;" indentlevel="1">
CHC 5,9-C25:2 &amp; 9-C25:1
</td>
<td style="text-align:right;">
-0.048
</td>
<td style="text-align:right;">
0.104
</td>
<td style="text-align:right;">
-0.223
</td>
<td style="text-align:right;">
0.166
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;width: 27em; ">
Attempted copulation frequency
</td>
<td style="text-align:right;">
-0.050
</td>
<td style="text-align:right;">
0.094
</td>
<td style="text-align:right;">
-0.213
</td>
<td style="text-align:right;">
0.143
</td>
<td style="text-align:left;">
</td>
</tr>
</tbody>
</table></li>
</ul></li>
</ul>
<p>Build panel a for Figure 4.</p>
<pre class="r"><code>I_summary &lt;- 
  I_index %&gt;% 
  group_by(Trait, Trait_Sex) %&gt;% 
  summarise(R_F = median(Response_to_selection_female),
            R_M = median(Response_to_selection_male),
            I = median(I)) %&gt;%
  ungroup() %&gt;% 
  mutate(I_abs = abs(I),
         Trait_Sex = case_when(Trait_Sex == &quot;Female&quot; ~ &quot;Traits measured in females&quot;,
                               Trait_Sex == &quot;Male&quot; ~ &quot;Traits measured in males&quot;)) %&gt;% 
  left_join(trait_data %&gt;% distinct(Trait, Trait_nice))

I_plot_labelled &lt;-
  I_plot +
  new_scale_fill() +
  geom_point(data = I_summary, #%&gt;% filter(Trait_Sex == &quot;Female&quot;),
             aes(x = R_F, y = R_M, fill = Trait_Sex),
             size = 4, alpha = 0.75, shape = 21) +
  scale_fill_manual(values = c(&quot;Female&quot; = &quot;white&quot;, &quot;Male&quot; = &quot;lightblue&quot;)) +
  labs(fill = &quot;Sex measured in&quot;) +
  new_scale_fill() +
  geom_label_repel(data = I_summary %&gt;% filter(I_abs &gt; 0.15),  
                   aes(label = Trait_nice, fill = if_else(I &gt; 0, met.brewer(name = &quot;Paquin&quot;, 7)[3], met.brewer(name = &quot;Paquin&quot;, 7)[5])), size = 3, alpha = 0.9,
                   min.segment.length = 0, seed = 1, box.padding = 0.5, point.padding = 0.5) +
  scale_fill_manual(values = c(met.brewer(name = &quot;Paquin&quot;, 7)[3], met.brewer(name = &quot;Paquin&quot;, 7)[5]), guide = &quot;none&quot;)

   
fig4_panel_a &lt;-
  I_summary %&gt;%
  ggplot(aes(x = R_F, y = R_M)) +
  geom_point(aes(fill = I),
             size = 4, alpha = 0.75, shape = 21) +
  geom_hline(yintercept = 0, linetype = 2) +
  geom_vline(xintercept = 0, linetype = 2) +
  scale_fill_gradientn(colours = met.brewer(name = &quot;Paquin&quot;, 7),
                       limits = c(-0.3, 0.3)) +
  guides(fill = guide_colourbar(barwidth = 8, barheight = 0.5)) +
  labs(x = &quot;Reponse to selection on females&quot;,
       y = &quot;Reponse to selection on males&quot;,
       fill = &quot;_I_ index&quot;) +
  coord_cartesian(xlim = c(-0.4, 0.4), ylim = c(-0.4, 0.4)) +
  facet_wrap(~Trait_Sex) +
  theme_bw() +
  theme(legend.position = &quot;bottom&quot;,
        strip.background = element_rect(fill = &quot;aliceblue&quot;),
        legend.title = element_markdown(),
        text = element_text(size = 11))</code></pre>
<p>Fit the across-trait <span class="math inline">\(I\)</span> model</p>
<pre class="r"><code>I_medians &lt;-
  I_index %&gt;% 
  select(Trait, Trait_Sex, I) %&gt;% 
  group_by(Trait, Trait_Sex) %&gt;% 
  summarise_draws(&quot;median&quot;, &quot;sd&quot;, ~quantile(.x, probs = c(0.025, 0.975), na.rm = TRUE)) %&gt;%
  ungroup() %&gt;% 
  rename(I = median) %&gt;% 
  select(-variable) %&gt;% 
  left_join(trait_data %&gt;% 
              distinct(Trait, .keep_all = TRUE) %&gt;% 
              select(Trait, Trait_nice, Life_stage, `Trait guild`, study_ID))


 # fit the model

I_model &lt;- 
  brm(data = I_medians, 
      I | weights(sd) ~ 1 + Trait_Sex + (1|study_ID) + (1|Trait),
      family = gaussian,
      prior = c(prior(normal(0, 0.25), class = Intercept),
                prior(exponential(1), class = sd),
                prior(exponential(5), class = sigma),
                prior(normal(0, 0.25), class = b)), 
      warmup = 4000, iter = 8000,
      seed = 1, cores = 4, chains = 4,
      control = list(adapt_delta = 0.9, max_treedepth = 10),
      file = &quot;fits/I_model&quot;)

print(I_model)</code></pre>
<pre><code> Family: gaussian 
  Links: mu = identity; sigma = identity 
Formula: I | weights(sd) ~ 1 + Trait_Sex + (1 | study_ID) + (1 | Trait) 
   Data: I_medians (Number of observations: 474) 
  Draws: 4 chains, each with iter = 8000; warmup = 4000; thin = 1;
         total post-warmup draws = 16000

Group-Level Effects: 
~study_ID (Number of levels: 76) 
              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(Intercept)     0.01      0.01     0.00     0.03 1.00    11461     7576

~Trait (Number of levels: 474) 
              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(Intercept)     0.01      0.01     0.00     0.02 1.00    11508     6276

Population-Level Effects: 
              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
Intercept         0.02      0.02    -0.01     0.05 1.00    18616    11551
Trait_SexMale    -0.01      0.02    -0.05     0.03 1.00    20614    11646

Family Specific Parameters: 
      Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sigma     0.07      0.01     0.05     0.08 1.00    18289    10839

Draws were sampled using sampling(NUTS). For each parameter, Bulk_ESS
and Tail_ESS are effective sample size measures, and Rhat is the potential
scale reduction factor on split chains (at convergence, Rhat = 1).</code></pre>
<p>Get model predictions</p>
<pre class="r"><code>new_data &lt;- tibble(Trait_Sex = c(&quot;Female&quot;, &quot;Male&quot;))

I_fitted &lt;-
  fitted(I_model, newdata = new_data, re_formula = NA, summary = F) %&gt;% 
  as.data.frame() %&gt;% 
  rename(Females = V1, Males = V2) %&gt;% 
  as_tibble() %&gt;% 
  pivot_longer(cols = everything(), names_to = &quot;Trait_Sex&quot;, values_to = &quot;I&quot;)


I_posterior &lt;- 
  as_draws_df(I_model) %&gt;% 
  mutate(Females = b_Intercept,
         Males = b_Intercept + b_Trait_SexMale) %&gt;% 
    select(Females, Males, sigma) 

I_posterior_long &lt;-
bind_cols(
I_posterior,
    tibble(predictive_I_female = rnorm(16000, mean = I_posterior$Females, sd = I_posterior$sigma),
           predictive_I_male = rnorm(16000, mean = I_posterior$Males, sd = I_posterior$sigma))
  ) %&gt;% 
  select(-sigma) %&gt;% 
  pivot_longer(cols = everything(), names_to = &quot;parameter&quot;, values_to = &quot;I&quot;)</code></pre>
<p>Build Figure 4 panel b</p>
<pre class="r"><code>fig4_panel_b &lt;-
  I_fitted %&gt;% 
  ggplot(aes(x = I, y = Trait_Sex)) +
  stat_halfeye(aes(fill = after_stat(x &gt; 0)),
               .width = c(0.66, 0.95), alpha = 0.9, 
               point_interval = &quot;median_qi&quot;, point_fill = &quot;white&quot;,
               shape = 21, point_size = 4, stroke = 1.5) + 
  scale_fill_manual(values = c(met.brewer(name = &quot;Paquin&quot;, 7)[2], 
                    met.brewer(name = &quot;Tsimshian&quot;, 7)[3])) +
  geom_vline(xintercept = 0, linetype = 2, linewidth = 0.75) +
  ylab(&quot;Sex traits were measured in&quot;) +
  xlab(expression(italic(I)* &quot; across-trait mean (posterior)&quot;)) +
  theme_minimal() + 
  theme(panel.background = element_rect(fill=&#39;transparent&#39;),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        plot.background = element_rect(fill=&#39;transparent&#39;, color=NA),
        legend.position = &quot;none&quot;,
        text = element_text(size=11))</code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="using-evidence-ratios-to-detect-sexually-antagonistic-traits"
class="section level2">
<h2>Using evidence ratios to detect sexually antagonistic traits</h2>
<p>We calculate evidence ratios a for sexually concordant responses to
selection by:</p>
<ol style="list-style-type: decimal">
<li><p>Using the <code>p_direction</code> function from the
<code>bayestestR</code> package to find the proportion of the posterior
distribution that is of the median’s sign for each trait in each sex. We
use this as the local false sign rate (LFSR).</p></li>
<li><p>Using the <code>p_direction</code> results we found the
probability that a trait has positive <span
class="math inline">\(R\)</span>, <span
class="math inline">\(P(pos)\)</span> or negative <span
class="math inline">\(R\)</span>, <span class="math inline">\(P(neg) = 1
- P(pos)\)</span>.</p></li>
<li><p>Finding <span class="math inline">\(P(concord) = P(pos)_f *
P(pos)_m + P(neg)_f * P(neg)_m\)</span></p></li>
<li><p>Finding <span class="math inline">\(P(antag) = P(pos)_f *
P(neg)_m + P(neg)_f * P(pos)_m\)</span></p></li>
<li><p>Taking the ratio of <span
class="math inline">\(P(concord)\)</span> and <span
class="math inline">\(P(antag)\)</span></p></li>
</ol>
<pre class="r"><code>pd_function &lt;-function(Trait_name){
  R_all_traits %&gt;% filter(Trait == Trait_name) %&gt;%
    select(Trait, Response_to_selection_female, Response_to_selection_male) %&gt;%
    p_direction(Response_to_selection_female, method = &quot;direct&quot;, null = 0) %&gt;%
    as_tibble() %&gt;%
    pivot_wider(names_from = Parameter, values_from = pd) %&gt;%
    mutate(Trait = Trait_name) %&gt;%
    rename(pd_female = Response_to_selection_female,
           pd_male = Response_to_selection_male)
}

trait_list &lt;- unique(R_all_traits$Trait)

pd_data &lt;- map_dfr(trait_list, pd_function)

Trait_medians &lt;-
  R_all_traits %&gt;%
  group_by(Trait) %&gt;%
  summarise(median_female = median(Response_to_selection_female),
            median_male = median(Response_to_selection_male)) %&gt;%
  ungroup()

evidence_ratios &lt;-
  left_join(pd_data, Trait_medians) %&gt;%
  mutate(prob_pos_female = if_else(median_female &gt; 0, pd_female, 1 - pd_female),
         prob_pos_male = if_else(median_male &gt; 0, pd_male, 1 - pd_male)) %&gt;%
  select(Trait, prob_pos_female, prob_pos_male) %&gt;%

    # Calculate the probabilities that beta_i and beta_j have the same/opposite signs
    mutate(p_sex_concord = prob_pos_female  * prob_pos_male +
             (1 - prob_pos_female) * (1 - prob_pos_male),
           p_sex_antag = prob_pos_female * (1 - prob_pos_male) +
             (1 - prob_pos_female) * prob_pos_male) %&gt;%
    # Find the ratios of these two probabilities (i.e. the &quot;evidence ratios&quot;)
    mutate(evidence_ratio_concord = p_sex_concord / p_sex_antag) %&gt;%
  left_join(R_all_traits %&gt;% distinct(Trait, Trait_Sex) %&gt;%
              mutate(Trait_Sex = if_else(Trait_Sex == &quot;Female&quot;, &quot;Traits measured in females&quot;,
                                         &quot;Traits measured in males&quot;))) %&gt;% 
  left_join(clean_meta_data)</code></pre>
<p>Build panel c for figure 4 and update panel a</p>
<pre class="r"><code>fig4_panel_c &lt;-
  evidence_ratios %&gt;%
    ggplot(aes(log2(evidence_ratio_concord))) +
    geom_histogram(data=subset(evidence_ratios, evidence_ratio_concord &lt; 1),
                   bins = 50, fill = met.brewer(name = &quot;Tam&quot;, 8)[4]) +
    geom_histogram(data=subset(evidence_ratios, evidence_ratio_concord &gt;= 1),
                   bins = 50, fill = met.brewer(name = &quot;Tsimshian&quot;, 7)[3]) +
    coord_cartesian(xlim = c(-8, 8), ylim = c(0, 20)) +
  geom_vline(xintercept = log2(19), linetype = 2) +
  geom_vline(xintercept = log2(1/19), linetype = 2) +
    scale_x_continuous(breaks = c(-8,  -6, -4, -2, 0, 2, 4, 6,  8),
                       labels = c(paste(&quot;1/&quot;,2 ^ c(8, 6, 4, 2), sep = &quot;&quot;), 2 ^ c(0, 2,4,6,8))) +
    scale_y_continuous(expand = expansion(mult = c(0.01, 0.01))) +

    facet_wrap(~ Trait_Sex) +
    xlab(&quot;Evidence ratio (log~2~ scale)&quot;) + ylab(&quot;Number of traits&quot;) +
    theme_bw() +
    theme(#panel.border = element_rect(size = 0.8),
          text = element_text(size = 11),
          axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
          strip.background = element_rect(fill = &quot;aliceblue&quot;),
          axis.title.x = element_markdown())

plotting_data_er &lt;-
  evidence_ratios %&gt;% select(Trait, Trait_Sex, evidence_ratio_concord) %&gt;% 
  left_join(I_summary, by = c(&quot;Trait&quot;, &quot;Trait_Sex&quot;)) %&gt;% 
  mutate(er_abs = abs(log2(evidence_ratio_concord)))


# add labels to fig3 panel a, using the &#39;significant&#39; er values

fig4_panel_a &lt;-
  fig4_panel_a +
  geom_label_repel(data = plotting_data_er %&gt;% filter(er_abs &gt; 4.24),  
                   aes(label = Trait_nice), 
                   fill = met.brewer(name = &quot;Paquin&quot;, 7)[5],
                   size = 2, alpha = 0.9,
                   min.segment.length = 0, seed = 1,
                   box.padding = 0.5, point.padding = 0.5,
                   max.overlaps = 30) </code></pre>
<p>Fit the across-trait evidence ratio model</p>
<pre class="r"><code> # fit the model

er_model &lt;- 
  brm(data = evidence_ratios, 
      log2(evidence_ratio_concord) ~ 1 + Trait_Sex + (1|study_ID),
      family = gaussian,
      prior = c(prior(normal(0, 0.5), class = Intercept),
                prior(exponential(1), class = sd),
                prior(exponential(5), class = sigma),
                prior(normal(0, 0.25), class = b)), 
      warmup = 4000, iter = 8000,
      seed = 1, cores = 4, chains = 4,
      control = list(adapt_delta = 0.9, max_treedepth = 10),
      file = &quot;fits/er_model&quot;)

print(er_model)</code></pre>
<pre><code> Family: gaussian 
  Links: mu = identity; sigma = identity 
Formula: log2(evidence_ratio_concord) ~ 1 + Trait_Sex + (1 | study_ID) 
   Data: evidence_ratios (Number of observations: 474) 
  Draws: 4 chains, each with iter = 8000; warmup = 4000; thin = 1;
         total post-warmup draws = 16000

Group-Level Effects: 
~study_ID (Number of levels: 76) 
              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sd(Intercept)     0.78      0.12     0.56     1.02 1.00     5736    10138

Population-Level Effects: 
                              Estimate Est.Error l-95% CI u-95% CI Rhat
Intercept                         0.33      0.13     0.07     0.57 1.00
Trait_SexTraitmeasuredinmales    -0.10      0.11    -0.31     0.11 1.00
                              Bulk_ESS Tail_ESS
Intercept                         8082    10687
Trait_SexTraitmeasuredinmales    29138    12916

Family Specific Parameters: 
      Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sigma     1.20      0.04     1.12     1.29 1.00    19306    11972

Draws were sampled using sampling(NUTS). For each parameter, Bulk_ESS
and Tail_ESS are effective sample size measures, and Rhat is the potential
scale reduction factor on split chains (at convergence, Rhat = 1).</code></pre>
<p>Get model predictions and plot</p>
<pre class="r"><code>new_data &lt;- tibble(Trait_Sex = c(&quot;Trait measured in females&quot;, &quot;Trait measured in males&quot;))

er_fitted &lt;-
  fitted(er_model, newdata = new_data, re_formula = NA, summary = F) %&gt;% 
  as.data.frame() %&gt;% 
  rename(Females = V1, Males = V2) %&gt;% 
  as_tibble() %&gt;% 
  pivot_longer(cols = everything(), names_to = &quot;Trait_Sex&quot;, values_to = &quot;evidence_ratio_mean&quot;) %&gt;% 
  mutate(evidence_ratio_mean = exp(evidence_ratio_mean))</code></pre>
<pre class="r"><code>fig4_panel_d &lt;-
  er_fitted %&gt;% 
  ggplot(aes(x = evidence_ratio_mean, y = Trait_Sex)) +
  stat_halfeye(aes(fill = after_stat(x &gt; 1)),
               .width = c(0.66, 0.95), alpha = 0.9, 
               point_interval = &quot;median_qi&quot;, point_fill = &quot;white&quot;,
               shape = 21, point_size = 4, stroke = 1.5) + 
  scale_fill_manual(values = c(met.brewer(name = &quot;Paquin&quot;, 7)[2], 
                    met.brewer(name = &quot;Tsimshian&quot;, 7)[3])) +
  geom_vline(xintercept = 1, linetype = 2, linewidth = 0.75) +
  ylab(&quot;Sex traits were measured in&quot;) +
  xlab(&quot;Evidence ratio across-trait\nmean (posterior)&quot;) +
  #scale_x_continuous(limits = c(-1, 1), breaks = c(-1,  -0.2876821, 0, 0.4054651, 0.6931472),
    #                   labels = c(&quot;1/2&quot;, &quot;3/4&quot;, 1, &quot;1.5&quot;, &quot;2&quot;)) +
  theme_minimal() + 
  theme(panel.background = element_rect(fill=&#39;transparent&#39;),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        plot.background = element_rect(fill=&#39;transparent&#39;, color=NA),
        legend.position = &quot;none&quot;,
        text = element_text(size=11))</code></pre>
</div>
<div id="figure-4" class="section level2">
<h2>Figure 4</h2>
<pre class="r"><code>(fig4_panel_a | plot_spacer() | fig4_panel_b) / 
  (fig4_panel_c | plot_spacer() | fig4_panel_d) &amp;
  plot_layout(widths = c(3, -0.5, 1)) &amp;  
  plot_annotation(tag_levels = &#39;a&#39;)</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-32-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-32-1">
Past versions of unnamed-chunk-32-1.png
</button>
</p>
<div id="fig-unnamed-chunk-32-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/319c84c1118d4c1c59279734e1b3432ef1d08901/docs/figure/Main_analysis.Rmd/unnamed-chunk-32-1.png" target="_blank">319c84c</a>
</td>
<td>
tomkeaney
</td>
<td>
2023-10-17
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure 4</strong>. Concordance and conflict in the
<em>Drosophila melanogaster</em> phenome. <strong>a</strong> shows
traits coloured by Innocenti and Morrow’s I index. Extreme positive
values indicate a trait is predicted to have a sexually concordant
response to selection, while negative values indicate a trait is
expected to have a sexually antagonistic response to selection. Traits
with evidence ratios &gt; 19 are labelled. <strong>b</strong> the mean
<em>I</em> value estimated across all traits, split by the sex traits
were measured in. <strong>c</strong> evidence ratios for sexually
concordant responses to selection. Dashed lines indicate an evidence
ratio of 19, which is equivalent to a frequentist <em>p</em>-value =
0.05. <strong>d</strong> the mean evidence ratio estimated across all
traits, split by the sex traits were measured in. Points indicate the
mean point estimate, with associated 67 and 95% credible intervals.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22621)

Matrix products: default


locale:
[1] LC_COLLATE=English_Australia.utf8  LC_CTYPE=English_Australia.utf8   
[3] LC_MONETARY=English_Australia.utf8 LC_NUMERIC=C                      
[5] LC_TIME=English_Australia.utf8    

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggnewscale_0.4.9  ggrepel_0.9.4     ggtext_0.1.2      broom_1.0.5      
 [5] bayestestR_0.13.1 tidybayes_3.0.6   brms_2.20.4       Rcpp_1.0.11      
 [9] patchwork_1.1.3   DT_0.30           kableExtra_1.3.4  rcartocolor_2.1.1
[13] MetBrewer_0.2.0   lubridate_1.9.3   forcats_1.0.0     stringr_1.5.0    
[17] dplyr_1.1.3       purrr_1.0.2       readr_2.1.4       tidyr_1.3.0      
[21] tibble_3.2.1      ggplot2_3.4.4     tidyverse_2.0.0   workflowr_1.7.1  

loaded via a namespace (and not attached):
  [1] tensorA_0.36.2       rstudioapi_0.15.0    jsonlite_1.8.7      
  [4] magrittr_2.0.3       farver_2.1.1         rmarkdown_2.25      
  [7] fs_1.6.3             vctrs_0.6.4          base64enc_0.1-3     
 [10] webshot_0.5.5        htmltools_0.5.6.1    distributional_0.3.2
 [13] curl_5.1.0           sass_0.4.7           StanHeaders_2.26.28 
 [16] bslib_0.5.1          htmlwidgets_1.6.2    plyr_1.8.9          
 [19] zoo_1.8-12           cachem_1.0.8         commonmark_1.9.0    
 [22] whisker_0.4.1        igraph_1.5.1         mime_0.12           
 [25] lifecycle_1.0.3      pkgconfig_2.0.3      colourpicker_1.3.0  
 [28] Matrix_1.6-1.1       R6_2.5.1             fastmap_1.1.1       
 [31] shiny_1.7.5.1        digest_0.6.33        colorspace_2.1-0    
 [34] ps_1.7.5             rprojroot_2.0.3      crosstalk_1.2.0     
 [37] labeling_0.4.3       fansi_1.0.5          timechange_0.2.0    
 [40] httr_1.4.7           abind_1.4-5          compiler_4.3.1      
 [43] bit64_4.0.5          withr_2.5.1          backports_1.4.1     
 [46] inline_0.3.19        shinystan_2.6.0      highr_0.10          
 [49] QuickJSR_1.0.7       pkgbuild_1.4.2       gtools_3.9.4        
 [52] loo_2.6.0            tools_4.3.1          httpuv_1.6.11       
 [55] threejs_0.3.3        glue_1.6.2           callr_3.7.3         
 [58] nlme_3.1-162         promises_1.2.1       gridtext_0.1.5      
 [61] grid_4.3.1           checkmate_2.2.0      getPass_0.2-2       
 [64] reshape2_1.4.4       generics_0.1.3       isoband_0.2.7       
 [67] gtable_0.3.4         tzdb_0.4.0           hms_1.1.3           
 [70] xml2_1.3.5           utf8_1.2.3           pillar_1.9.0        
 [73] ggdist_3.3.0         markdown_1.10        vroom_1.6.4         
 [76] posterior_1.4.1      later_1.3.1          lattice_0.21-8      
 [79] bit_4.0.5            tidyselect_1.2.0     miniUI_0.1.1.1      
 [82] knitr_1.44           git2r_0.32.0         arrayhelpers_1.1-0  
 [85] gridExtra_2.3        V8_4.4.0             svglite_2.1.2       
 [88] stats4_4.3.1         xfun_0.40            bridgesampling_1.1-2
 [91] matrixStats_1.0.0    rstan_2.32.3         stringi_1.7.12      
 [94] yaml_2.3.7           evaluate_0.22        codetools_0.2-19    
 [97] cli_3.6.1            RcppParallel_5.1.7   shinythemes_1.2.0   
[100] xtable_1.8-4         systemfonts_1.0.5    munsell_0.5.0       
[103] processx_3.8.2       jquerylib_0.1.4      coda_0.19-4         
[106] svUnit_1.0.6         parallel_4.3.1       rstantools_2.3.1.1  
[109] ellipsis_0.3.2       prettyunits_1.2.0    dygraphs_1.1.1.6    
[112] bayesplot_1.10.0     Brobdingnag_1.2-9    viridisLite_0.4.2   
[115] mvtnorm_1.2-3        scales_1.2.1         xts_0.13.1          
[118] insight_0.19.6       crayon_1.5.2         rlang_1.1.1         
[121] rvest_1.0.3          shinyjs_2.1.0       </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
