<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Thomas Keaney and Luke Holman" />


<title>Intralocus sexual conflict in the DGRP</title>

<script src="site_libs/header-attrs-2.7/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Intralocus sexual conflict in the DGRP</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Main_analysis.html">Main analysis</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tomkeaney/DGRP_sexual_conflict">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Intralocus sexual conflict in the DGRP</h1>
<h4 class="author">Thomas Keaney and Luke Holman</h4>
<h4 class="date">25/06/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DGRP_sexual_conflict/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210706code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210706)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210706code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210706)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPsexualconflicttree959123f80add4c3747918d1937e125003c982035targetblank959123fa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/tomkeaney/DGRP_sexual_conflict/tree/959123f80add4c3747918d1937e125003c982035" target="_blank">959123f</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPsexualconflicttree959123f80add4c3747918d1937e125003c982035targetblank959123fa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/tomkeaney/DGRP_sexual_conflict/tree/959123f80add4c3747918d1937e125003c982035" target="_blank">959123f</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  Selection_differential_plot.pdf
    Untracked:  data/S_female.csv
    Untracked:  fits/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Main_analysis.Rmd</code>) and HTML (<code>docs/Main_analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/959123f80add4c3747918d1937e125003c982035/analysis/Main_analysis.Rmd" target="_blank">959123f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
trying to fix a bug
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/769e48d5a4afe34bb3a05faaa53829aeaadcfc05/docs/Main_analysis.html" target="_blank">769e48d</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/340c41bb74226b0fe4c241fc5d79663d2210c387/analysis/Main_analysis.Rmd" target="_blank">340c41b</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
sexual dimorphism calculation added
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/4018580c9b086ed875bd241ca2028263395bd971/docs/Main_analysis.html" target="_blank">4018580</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/65b25971953ba68583cb15a9503035ac59b1a16f/analysis/Main_analysis.Rmd" target="_blank">65b2597</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
sexual dimorphism calculation added
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/83a0c4fdf0c246cc8f853dbeecd4ab8b252df66e/docs/Main_analysis.html" target="_blank">83a0c4f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/ddea55e2cfaed0dddfe93d9dad2d1154f0e6a451/analysis/Main_analysis.Rmd" target="_blank">ddea55e</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
sexual dimorphism calculation added
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/b364d30085d882105bbb010749bcadccd8fa123a/docs/Main_analysis.html" target="_blank">b364d30</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/e404cf9f54cc05b642050a7b0bc0070a6e55f8bb/analysis/Main_analysis.Rmd" target="_blank">e404cf9</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
<td>
progression on selection calculations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/Main_analysis.html" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/6aad8a86f4107b143e2e07bb036e5064adeca7cc/analysis/Main_analysis.Rmd" target="_blank">6aad8a8</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
<td>
progression on selection calculations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/11a83906f387d1ecbb39fc5ab60f5830bdac2d8f/docs/Main_analysis.html" target="_blank">11a8390</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/42b8f12ec94d1089c153f97f0e4090edd191a279/analysis/Main_analysis.Rmd" target="_blank">42b8f12</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-07
</td>
<td>
Get the site up and running
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages-and-the-data" class="section level2">
<h2>Load packages and the data</h2>
<p>First load the packages</p>
<pre class="r"><code>library(tidyverse)
library(googlesheets4) # for reading google sheets
library(pheatmap) # for heatmaps
library(RColorBrewer) # for custom colours
library(rcartocolor) # cool colours
library(wesanderson) # cool colours
library(kableExtra) # for scrolling tables
library(pander) # more tables
library(patchwork) # for cool plots
library(brms) # for bayesian models
library(tidybayes) # for more bayesian things
library(shiny) # for interactive app
library(DT) # for interactive tables
library(broom) # convert results of functions into tables
library(ggExtra) # for marginal plots
library(ggtips) # for hovering tooltips in R shiny
library(ggtext) # for markdown features in ggplot
library(job) # to run things in the background</code></pre>
<p>Now load the data. Correlation analysis is not robust when sample sizes are small, so we only include traits that have been measured in 80+ lines.</p>
<pre class="r"><code>DGRP_data &lt;- read_csv(&quot;data/all.dgrp.phenos_unscaled.csv&quot;)

# fix the typo (temporary - keep until we reload the data from google sheets)

DGRP_data$Trait &lt;-recode(DGRP_data$Trait, gustatory.reponse.ethanol.m = &quot;gustatory.response.ethanol.m&quot;)

# Now trim the dataset so that it only includes traits that have been measured in 80+ lines - we are left with 905 traits from an original 1119

DGRP_subset &lt;- DGRP_data %&gt;% 
  group_by(Trait) %&gt;% 
  filter(length(line) &gt; 80) %&gt;% 
  ungroup()

# Add a study ID variable 

DGRP_subset &lt;- 
  DGRP_subset %&gt;% 
  group_by(Reference) %&gt;% 
  mutate(study_ID = cur_group_id()) %&gt;% 
  arrange(study_ID) %&gt;% 
  ungroup()

# Assign each trait to a category based on whether it was measured in 1) both sexes separately, 2) just females, 3) just males or 4) both sexes together. We can use a shortcut and use the study IDs as each study tends to be consistent with their measurement approach (with 4 exceptions).

# This is dangerous coding. Using the ID number rather than the name can easily lead to an error. Consider changing.

DGRP_subset &lt;-
  DGRP_subset %&gt;% 
  mutate(rmf_applicable = if_else(study_ID == &quot;3&quot; | study_ID == &quot;5&quot; | study_ID == &quot;7&quot; | study_ID == &quot;8&quot; | study_ID == &quot;9&quot; | study_ID == &quot;11&quot; | study_ID == &quot;13&quot; | study_ID == &quot;19&quot; | study_ID == &quot;20&quot; | study_ID == &quot;21&quot; | study_ID == &quot;23&quot; | study_ID == &quot;24&quot; | study_ID == &quot;26&quot; | study_ID == &quot;27&quot; | study_ID == &quot;28&quot; | study_ID == &quot;34&quot; | study_ID == &quot;37&quot; | study_ID == &quot;40&quot; | study_ID == &quot;43&quot; | study_ID == &quot;48&quot; | study_ID == &quot;49&quot; | study_ID == &quot;54&quot; | study_ID == &quot;55&quot; | study_ID == &quot;57&quot; | study_ID == &quot;64&quot; | study_ID == &quot;65&quot; | study_ID == &quot;71&quot; | study_ID == &quot;72&quot; | study_ID == &quot;73&quot; | study_ID == &quot;75&quot;| study_ID == &quot;76&quot;, &quot;YES&quot;, &quot;NO&quot;))

# Above, we identified the studies that measured traits in the sexes separately. However, some studies measure multiple traits, some of which they only measure in one sex or both (e.g. larval traits are often measured with the sexes pooled). Below we remove the specific traits that are not measured in both sexes separately from our already trimmed data  

DGRP_rmf &lt;- 
  DGRP_subset %&gt;% 
  filter(rmf_applicable == &quot;YES&quot;) %&gt;% 
  filter(Trait != &quot;anoxia.tolerance.larval&quot;,
         Trait != &quot;CHC.11 &amp; 9-Me-C23.f&quot;, 
         Trait != &quot;CHC.13- &amp; 11-Me-C25.f&quot;,
         Trait != &quot;CHC.2-Me-C26.m&quot;, 
         Trait !=&quot;CHC.2-Me-C27.m&quot;, 
         Trait !=&quot;CHC.2-Me-C28 &amp; 7,11-C29:2.f&quot;,
         Trait != &quot;CHC.2-Me-C28.m&quot;, 
         Trait != &quot;CHC.2-Me-C29 &amp; 8,12-C30:2.f&quot;,
         Trait != &quot;CHC.2-Me-C29.m&quot;, 
         Trait != &quot;CHC.2-Me-C30.f&quot;,
         Trait != &quot;CHC.3-Me-C23.m&quot;, 
         Trait != &quot;CHC.3-Me-C25 &amp; 9-C26:1.f&quot;,
         Trait != &quot;CHC.3-Me-C25.m&quot;,
         Trait != &quot;CHC.3-Me-C27 &amp; 6,10-C28:2 &amp; 9-C28:1.f&quot;,
         Trait != &quot;CHC.5,9-C25:2 &amp; 9-C25:1.f&quot;, 
         Trait != &quot;CHC.5,9-C27:2 &amp; 9-C27:1.f&quot;,
         Trait != &quot;CHC.5-C27:1.f&quot;, 
         Trait != &quot;CHC.5-Me-C25.f&quot;,
         Trait != &quot;CHC.6,10-C26:2.f&quot;, 
         Trait != &quot;CHC.6-C26:1 &amp; i.m&quot;,
         Trait != &quot;CHC.7,11-C23:2.f&quot;, 
         Trait != &quot;CHC.7,11-C24:2.f&quot;, 
         Trait != &quot;CHC.7,11-C25:2.f&quot;,
         Trait != &quot;CHC.7,11-C26:2.f&quot;, 
         Trait != &quot;CHC.7,11-C27:2 &amp; 2-Me-C26.f&quot;,
         Trait != &quot;CHC.7,11-C28:2 &amp; 2-Me-C27.f&quot;, 
         Trait != &quot;CHC.7,11-C31:2.f&quot;,
         Trait != &quot;CHC.7-C24:1.m&quot;,
         Trait != &quot;CHC.8,12-C26:2.f&quot;,
         Trait != &quot;CHC.8,12-C28:2.f&quot;,
         Trait != &quot;CHC.8-C25:1.f&quot;,
         Trait != &quot;CHC.9,13-C25:2.f&quot;, 
         Trait != &quot;CHC.9,13-C27:2.f&quot;,
         Trait != &quot;CHC.9-C25:1.m&quot;, 
         Trait != &quot;CHC.9-C27:1.m&quot;,
         Trait != &quot;CHC.9-C29:1 &amp; 5,9-C29:2.f&quot;, 
         Trait != &quot;CHC.n-C25.m&quot;,
         Trait != &quot;CHC.NI.20.f&quot;, 
         Trait != &quot;CHC.NI.33.f&quot;, 
         Trait != &quot;CHC.NI.34.f&quot;,
         Trait != &quot;CHC.NI.7.m&quot;, 
         Trait != &quot;CHC.x-C22:1a.m&quot;,
         Trait != &quot;CHC.x-C22:1b.m&quot;, 
         Trait != &quot;CHC.x-C23:1.f&quot;,
         Trait != &quot;kallithea.viral.load.f&quot;)

# Make a female trait dataset

DGRP_rf_unscaled &lt;-
  DGRP_rmf %&gt;% filter(Sex == &quot;Female&quot;) %&gt;% 
  rename(Female_trait_value = trait_value) %&gt;% 
  select(-c(Sex, `Trait description`, study_ID, rmf_applicable)) %&gt;% 
  mutate(Trait = gsub(&#39;.{2}$&#39;, &#39;&#39;, Trait)) # remove the final two characters from the name of each trait: .f

DGRP_rf &lt;-
  DGRP_rf_unscaled %&gt;% 
  group_by(Trait) %&gt;% # group the data by trait so that scaling works on a per trait basis
  mutate(Female_trait_value = as.numeric(scale(Female_trait_value))) %&gt;% # scale the data for each trait 
  ungroup() 
 
# Make a male trait dataset          

DGRP_rm_unscaled &lt;-
  DGRP_rmf %&gt;% filter(Sex == &quot;Male&quot;) %&gt;% 
  rename(Male_trait_value = trait_value) %&gt;% 
  select(-c(Sex, `Trait description`, study_ID, rmf_applicable)) %&gt;%
  mutate(Trait = gsub(&#39;.{2}$&#39;, &#39;&#39;, Trait)) # remove the final two characters from the name of each trait: .m

DGRP_rm &lt;-
  DGRP_rm_unscaled %&gt;% 
  group_by(Trait) %&gt;% # group the data by trait so that scaling works on a per trait basis
  mutate(Male_trait_value = as.numeric(scale(Male_trait_value))) %&gt;% # scale the data for each trait
  ungroup() 

# Join the two together      

rfm_unscaled_data &lt;- full_join(DGRP_rf_unscaled, DGRP_rm_unscaled)

rfm_data &lt;- full_join(DGRP_rf, DGRP_rm)

# Now return to our larger tibble: DGRP_subset. Change the format of the data and scale it.

# This next section changes the format of the data. Each row represents a single line, while each column represents a trait and contains the lines trait mean. Scaling is also applied to each trait value, so that each trait has a mean of 0 and an SD of 1. This makes the variation found for each trait broadly comparable with all other traits.

DGRP_subset_scaled &lt;- 
  split(DGRP_subset, DGRP_subset$Trait) %&gt;% # split divides the data into trait groups
  lapply(function(x){
    trait_name &lt;- x$Trait[1]
    x &lt;- x[,c(1,4)]
    names(x) &lt;- c(&quot;line&quot;, &quot;value&quot;)
    x &lt;- x %&gt;% 
      group_by(line) %&gt;% 
      summarise(mean = mean(value))
    names(x) &lt;- c(&quot;line&quot;, trait_name)
    x}) %&gt;% 
  purrr::reduce(full_join, by = &quot;line&quot;, match = &quot;first&quot;) %&gt;% 
  mutate_at(vars(-(&quot;line&quot;)), ~ as.numeric(scale(.x))) # this is the scaling step</code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="calculate-the-intersex-genetic-correlation-for-fitness-raw-fm" class="section level2">
<h2>Calculate the intersex genetic correlation for fitness r<sup>A</sup><sub><em>w fm</em></sub></h2>
<pre class="r"><code># Note that because we are using standarised data, the intercept will always = 0 and the sigma = 1 (we have no predictors so sigma just predicts the population SD). We make the intercept 0 because we have standardised the mean for every trait to = 0. We fix sigma to 0, which makes our estimate more precise. We make sigma = 0 rather than 1 because brms by default actually models the log of sigma. log(1) = 1, so lets set sigma to 0.

rfm_fitness_model &lt;-
  brm(data = rfm_data %&gt;% filter(Trait == &quot;fitness.early&quot;),
      family = gaussian,
      bf(mvbind(Female_trait_value, Male_trait_value) ~ 0, sigma ~ 0) + set_rescor(TRUE),
      prior(lkj(2), class = rescor),
      chains = 4, cores = 4, 
      seed = 1, file = &quot;fits/rfm_fitness.model&quot;)   

posterior_summary(rfm_fitness_model) %&gt;% 
  kable()</code></pre>
<table>
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
Estimate
</th>
<th style="text-align:right;">
Est.Error
</th>
<th style="text-align:right;">
Q2.5
</th>
<th style="text-align:right;">
Q97.5
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
rescor__Femaletraitvalue__Maletraitvalue
</td>
<td style="text-align:right;">
0.2148652
</td>
<td style="text-align:right;">
0.0821809
</td>
<td style="text-align:right;">
0.046072
</td>
<td style="text-align:right;">
0.3620269
</td>
</tr>
<tr>
<td style="text-align:left;">
lp__
</td>
<td style="text-align:right;">
-348.9107198
</td>
<td style="text-align:right;">
0.6811722
</td>
<td style="text-align:right;">
-350.918982
</td>
<td style="text-align:right;">
-348.4131047
</td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="plot-raw-fm" class="section level2">
<h2>Plot r<sup>A</sup><sub><em>w fm</em></sub></h2>
<pre class="r"><code>DGRP_no_line &lt;- 
  DGRP_subset_scaled %&gt;% 
  select(-line)

pos_colour &lt;- wes_palette(&quot;Zissou1&quot;)[4]
neg_colour &lt;- wes_palette(&quot;Zissou1&quot;)[2]

fitness_regression_plot &lt;- function(a, b, c, d) {
  ggplot(data = a, aes(x = b, y = c)) +
    #geom_hex(bins = 20) +
    #scale_fill_carto_c(palette = &quot;Temps&quot;) +
    geom_point(shape = 21, alpha = 0.75, size = 3, fill = d, show.legend = FALSE) +
    geom_smooth(method = &#39;lm&#39;, color=&#39;black&#39;) +
    geom_hline(yintercept = 0, linetype = 2) +
    geom_vline(xintercept = 0, linetype = 2) +
    #coord_cartesian(xlim = c(-3, 3), ylim = c(-3, 3)) +
    theme_bw() +
    theme(panel.border= element_blank(),
          axis.line=element_line(),
          text = element_text(size=14),
          panel.grid.major.x = element_blank(),
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank(),
          panel.grid.minor.x = element_blank(),
          axis.title.x = element_text(size = 12),
          axis.title.y = element_text(size = 12))
}


rfm_fitness_plot &lt;-
  fitness_regression_plot(DGRP_no_line, 
                          DGRP_no_line$fitness.early.f,
                          DGRP_no_line$fitness.early.m, pos_colour) +
  labs(x = &quot;Female fitness&quot;, y = &quot;Male fitness&quot;) +
  annotate(
    &quot;text&quot;, label = &quot;residual correlation = 0.21 (95% CIs: 0.05 to 0.36)&quot;,
    x = -2.4, y = 2.55, size = 4, colour = &quot;black&quot;
  )

ggMarginal(rfm_fitness_plot,
      type = &#39;histogram&#39;,
      margins = &#39;both&#39;,
      size = 5,
      colour = wes_palette(&quot;Zissou1&quot;)[1],
      fill = wes_palette(&quot;Zissou1&quot;)[2]
    )</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/b364d30085d882105bbb010749bcadccd8fa123a/docs/figure/Main_analysis.Rmd/unnamed-chunk-3-1.png" target="_blank">b364d30</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/figure/Main_analysis.Rmd/unnamed-chunk-3-1.png" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/11a83906f387d1ecbb39fc5ab60f5830bdac2d8f/docs/figure/Main_analysis.Rmd/unnamed-chunk-3-1.png" target="_blank">11a8390</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure 1</strong>: The intersex genetic correlation for fitness r<sup>A</sup><sub><em>w fm</em></sub> across the DGRP. The orange points each represent a DGRP line, and where it falls on the female-male fitness continuum. Data points are standardised to have a mean of 0 and a standard deviation of 1. This means that lines that have negative values have lower than average fitness, while lines that have positive values have above average fitness. The dotted lines separate the plot into four quadrants: lines that fall within the top left quadrant have above average male fitness, but below average female fitness, lines in the top right have above average fitness for both sexes, lines in the bottom left have below average fitness in both sexes, and lines in the bottom right have below average fitness in males, but above average fitness in females.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="calculating-raz-fm-for-all-traits" class="section level2">
<h2>Calculating r<sup>A</sup><sub><em>z fm</em></sub> for all traits</h2>
<p><span class="math inline">\(~\)</span></p>
<p>The traits we consider in this analysis fall into four categories. 1) They have been measured in the two sexes separately, 2) they have only been measured females, 3) they have only been measured in males and 4) they have been measured across the two sexes. For those traits that fall into category 1, we can calculate r<sup>A</sup><sub><em>fm</em></sub>.</p>
<p>We use bayesian models to find the residual correlation between traits expressed in males and females. Each line represents a different genotype, and we have measures for both sexes per line. This correlation therefore corresponds to the additive intersex genetic correlation for each trait.</p>
<p>A problem here is that my current approach assumes that each trait is normally distributed, which is clearly false. This is the same problem that researchers face with the spearman correlation coefficient, and the main reason we have used the pearson method in earlier analyses. To solve this problem, I can categorise each trait into a ‘distribution family’ then run the models in subsets. Using this bayesian approach is preferred because it gives us measures of uncertainty which you don’t get from the pearson frequentist approach. <strong>Currently not done</strong></p>
<pre class="r"><code># find all the traits in our dataset

check &lt;- unique(rfm_data$Trait)

# this is our model outline - we can update this with new filtered data to calculate rfm for every trait. We do this because we don&#39;t want brms to recompile every time. With the model already compiled we can simply update, which saves a load of time.

rfm_model &lt;-
  brm(data = rfm_data %&gt;% filter(Trait == &quot;gustatory.response.ethanol&quot;),
      family = gaussian,
      bf(mvbind(Female_trait_value, Male_trait_value) ~ 0, sigma ~ 0) + set_rescor(TRUE),
      prior(lkj(2), class = rescor),
      chains = 4, cores = 4, 
      seed = 1)   </code></pre>
<pre><code>Running /Library/Frameworks/R.framework/Resources/bin/R CMD SHLIB foo.c
clang -mmacosx-version-min=10.13 -I&quot;/Library/Frameworks/R.framework/Resources/include&quot; -DNDEBUG   -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/Rcpp/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/unsupported&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include&quot; -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/StanHeaders/include/src/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/StanHeaders/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppParallel/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/rstan/include&quot; -DEIGEN_NO_DEBUG  -DBOOST_DISABLE_ASSERTS  -DBOOST_PENDING_INTEGER_LOG2_HPP  -DSTAN_THREADS  -DBOOST_NO_AUTO_PTR  -include &#39;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp&#39;  -D_REENTRANT -DRCPP_PARALLEL_USE_TBB=1   -I/usr/local/include   -fPIC  -Wall -g -O2  -c foo.c -o foo.o
In file included from &lt;built-in&gt;:1:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/Dense:1:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/Core:88:
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/src/Core/util/Macros.h:628:1: error: unknown type name &#39;namespace&#39;
namespace Eigen {
^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/src/Core/util/Macros.h:628:16: error: expected &#39;;&#39; after top level declarator
namespace Eigen {
               ^
               ;
In file included from &lt;built-in&gt;:1:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/Dense:1:
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/Core:96:10: fatal error: &#39;complex&#39; file not found
#include &lt;complex&gt;
         ^~~~~~~~~
3 errors generated.
make: *** [foo.o] Error 1</code></pre>
<pre class="r"><code># sample from the posterior - this will create a tibble with two columns - Trait and rfm

post &lt;- posterior_samples(rfm_model) %&gt;% 
  rename(rfm = rescor__Femaletraitvalue__Maletraitvalue) %&gt;% 
  mutate(Trait = &quot;olfactory.response.1-hexanol&quot;) %&gt;% 
  select(Trait, rfm) %&gt;% 
  as_tibble()

# make a function to update the model and the posterior sample output with your desired trait

rmf_calculator &lt;- function(selected_trait){
  
  data &lt;- rfm_data %&gt;% filter(Trait == selected_trait)
  
  model &lt;- update(
    rfm_model, newdata = data,
    chains = 4, cores = 4, 
    seed = 1)
  
  post &lt;- 
    posterior_samples(model) %&gt;% 
    rename(rfm = rescor__Femaletraitvalue__Maletraitvalue) %&gt;% 
    mutate(Trait = selected_trait) %&gt;% 
    select(Trait, rfm) %&gt;% 
    as_tibble() #%&gt;% 
    rbind(post) # we add this to the bottom of the previous post object: it gains progressively more rows as traits are added
  
  post
}

# create a list of all trait to iterate our function over

trait_list &lt;- unique(rfm_data$Trait)

# run the function

if(!file.exists(&quot;data/rfm.complete.csv&quot;)){
  rfm_complete &lt;- map_dfr(trait_list, rmf_calculator)
  write_csv(rfm_complete, file = &quot;data/rfm.complete.csv&quot;)
} else rfm_complete &lt;- read_csv(&quot;data/rfm.complete.csv&quot;)



# now plot rfm for all traits

forest_plot &lt;-
  rfm_complete %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_rfm = median(rfm)) %&gt;%
  ggplot(aes(rfm, fct_reorder(Trait, avg_rfm))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  #stat_pointinterval(aes(), .width = c(0.66, 0.95)) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-0.5, 1)) +
  ylab(&quot;Trait&quot;) +
  xlab(&quot;Estimated _r_^A z fm&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

forest_plot</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-4-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/769e48d5a4afe34bb3a05faaa53829aeaadcfc05/docs/figure/Main_analysis.Rmd/unnamed-chunk-4-1.png" target="_blank">769e48d</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/figure/Main_analysis.Rmd/unnamed-chunk-4-1.png" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure X</strong>: The intersex additive genetic correlation for traits that have been measured in both sexes in &gt; 80 DGRP lines. A value of 0 indicates that there is no intersex genetic correlation, while a value of 1 means that additive genetic effects in one sex affect the other one identically. Data are means (depicted as 5% uncertainty intervals) with 66% and 95% uncertainty intervals.</p>
<p><span class="math inline">\(~\)</span></p>
<p>This figure is cluttered, mainly because there are so many microbiome traits. Below I attempt to visualise the data without these microbiome traits. This code is also useful because we can tag each trait by its guild.</p>
<p><strong>It has problems</strong></p>
<pre class="r"><code># I don&#39;t really get what left_join is doing here. Why do we have 186 million rows! I just want the trait guild and reference columns added to rfm_complete, grouping by trait. I would like to retain all of the rows in rfm_complete (which has many more rows than rfm_data).

# the distinct() hack I use here is not great. It removes values draws from the posterior that are the same by chance, instead of because I have accidentally duplicated them with my full_join. This removes ~50,000 of the ~ 1 million rows that should be retained. 

rfm_complete_guilds &lt;- 
  full_join(rfm_complete, rfm_data %&gt;% 
              select(-line)) #%&gt;% 
  distinct()

# now plot rfm in guild groups

forest_plot_2 &lt;-
  rfm_complete_guilds %&gt;%
  filter(`Trait guild` != &quot;Microbiome&quot;) %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_rfm = median(rfm)) %&gt;%
  ggplot(aes(rfm, fct_reorder(Trait, avg_rfm))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  #stat_pointinterval(aes(), .width = c(0.66, 0.95)) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-0.5, 1)) +
  ylab(&quot;Trait&quot;) +
  xlab(&quot;Estimated _r_^A z fm&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())</code></pre>
</div>
<div id="calculating-sexual-dimorphism-for-all-traits" class="section level2">
<h2>Calculating sexual dimorphism for all traits</h2>
<p><span class="math inline">\(~\)</span></p>
<p>We use the <span class="math inline">\(\beta\)</span> estimate for the effect of Sex:Male on the mean for each trait to estimate sexual dimorphism.</p>
<pre class="r"><code>d1 &lt;- 
  rfm_unscaled_data %&gt;% 
  select(-Male_trait_value) %&gt;% 
  mutate(Sex = &quot;Female&quot;) %&gt;% 
  rename(Trait_value = Female_trait_value)

d2 &lt;-
  rfm_unscaled_data %&gt;% 
  select(-Female_trait_value) %&gt;% 
  mutate(Sex = &quot;Male&quot;) %&gt;% 
  rename(Trait_value = Male_trait_value)

dimorphism_data &lt;- rbind(d1, d2) %&gt;% 
  group_by(Trait) %&gt;% # group the data by trait so that scaling works on a per trait basis
  mutate(Trait_value = as.numeric(scale(Trait_value))) %&gt;% # scale the data for each trait
  ungroup() 

# now we model dimorphism score to find an average value for each trait 

#job::job({
dimorphism_model_2 &lt;-
  brm(data = dimorphism_data,
      family = gaussian,
      Trait_value ~ 1 + (Sex|Trait),
      prior = c(prior(normal(0, 5), class = Intercept),
                prior(cauchy(0, 2), class = sd)),
      chains = 4, cores = 4, warmup = 3000, iter = 8000, 
      seed = 1, file = &quot;fits/dimorphism_model_2&quot;) 
#})


# Now find the difference between males and females for each trait. This is calculated in the model as the trait specific effect of being male. We can find posterior estimates for these parameters using posterior samples(). Each trait has a column of estimated values named r_Trait[**insert specified trait**,SexMale]. We wrangle the data so that we get two columns: one with the trait name and another with an estimated value sampled from the posterior distribution for that parameter.

# Interpreting the dimorphism values: positive values indicate that males have higher trait values than females, a value of 0 indicates that there is no sexual dimorphism, and negative values indicate that females have higher trait values

dimorphism_post &lt;- posterior_samples(dimorphism_model_2) %&gt;% 
  as_tibble() %&gt;% 
  select(starts_with(&quot;r_Trait&quot;) &amp;
           contains(&quot;SexMale&quot;)) %&gt;% 
  pivot_longer(cols = everything(), names_to = &quot;Trait&quot;, values_to = &quot;dimorphism_value&quot;) %&gt;% 
  mutate(Trait = str_remove(Trait, &quot;r_Trait&quot;),
         Trait =  str_remove(Trait, &quot;,SexMale&quot;),
         Trait = gsub(&quot;\\[|\\]&quot;, &quot;&quot;, Trait)) %&gt;% 
  arrange(Trait)

# now plot dimorphism for all traits

forest_plot_dimorphism &lt;-
  dimorphism_post %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_dimorph = median(dimorphism_value)) %&gt;%
  ggplot(aes(dimorphism_value, fct_reorder(Trait, avg_dimorph))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  #stat_pointinterval(aes(), .width = c(0.66, 0.95)) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-2, 2)) +
  ylab(&quot;Trait&quot;) +
  xlab(&quot;Estimated dimorphism&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

forest_plot_dimorphism</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/769e48d5a4afe34bb3a05faaa53829aeaadcfc05/docs/figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">769e48d</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/83a0c4fdf0c246cc8f853dbeecd4ab8b252df66e/docs/figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">83a0c4f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/11a83906f387d1ecbb39fc5ab60f5830bdac2d8f/docs/figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">11a8390</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure XX</strong>: Sexual dimorphism scores for traits that have been phenotyped in females and males. Positive values indicate that males have higher trait values than females, a value of 0 indicates that there is no sexual dimorphism, and negative values indicate that females have higher trait values than males. Bars represent means (displayed as 5% CI intervals), 66% and 95% intervals.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="calculating-s-the-selection-differential" class="section level2">
<h2>Calculating <em>S</em>: the selection differential</h2>
<p><span class="math inline">\(~\)</span></p>
<p>We can calculate the selection differential associated with each phenotype using the <strong>Robertson Price identity</strong>, which states that the selection differential is equivalent to the covariance between phenotype and relative fitness</p>
<p><span class="math inline">\(S = \sigma(w_i, z_i)\)</span></p>
<p>or</p>
<p><span class="math inline">\(S = cov(relative.fitness, standarised.phenotype)\)</span></p>
<p>To do this, I currently use the pearson correlation coefficient. I would like to use bayesian analysis instead, so that we get uncertainty estimates on the selection differentials.</p>
<p>We can also calculate the broad sense heritability associated with each phenotype… if we could calculate the narrow sense heritability we could use the <strong>Breeder’s equation</strong> to calculate the response to selection for each phenotype (seems unlikely).</p>
<p><strong>Table 1:</strong> Selection coefficients (<span class="math inline">\(S\)</span>) for each phenotype, when expressed in each sex. The antagonism index column shows how different selection on the trait is in the two sexes. Positive values indicate that the trait has a more positive selection differential in females than it does in males, a value of 0 indicates that the selection differential is the same in each sex, and negative values indicate a more positive selection differential in males compared to females.</p>
<pre class="r"><code># Calculate female selection differentials - identify all traits that don&#39;t end in .m

Female_selection_data &lt;- 
  DGRP_subset_scaled %&gt;% 
  select(!ends_with(&quot;.m&quot;))

# Calculate the correlation between female fitness and all these traits - maybe we go univariate here...

# this is our model outline - we can update this with new filtered data to calculate rfm for every trait. We do this because we don&#39;t want brms to recompile every time. With the model already compiled we can simply update, which saves a load of time.

data &lt;- Female_selection_data %&gt;% 
  select(&quot;fitness.early.f&quot;, &quot;Acetobacter.aceti.f&quot;) %&gt;% 
  rename(V1 = 1, V2 = 2)

s_female_model&lt;-
  brm(data = data,
      family = gaussian,
      V1 ~ 0 + V2,
      prior = c(prior(normal(0, 1), class = b),
                prior(normal(0, 1), class = sigma)),
      chains = 4, cores = 4, 
      seed = 1)</code></pre>
<pre><code>Running /Library/Frameworks/R.framework/Resources/bin/R CMD SHLIB foo.c
clang -mmacosx-version-min=10.13 -I&quot;/Library/Frameworks/R.framework/Resources/include&quot; -DNDEBUG   -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/Rcpp/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/unsupported&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/BH/include&quot; -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/StanHeaders/include/src/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/StanHeaders/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppParallel/include/&quot;  -I&quot;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/rstan/include&quot; -DEIGEN_NO_DEBUG  -DBOOST_DISABLE_ASSERTS  -DBOOST_PENDING_INTEGER_LOG2_HPP  -DSTAN_THREADS  -DBOOST_NO_AUTO_PTR  -include &#39;/Library/Frameworks/R.framework/Versions/4.0/Resources/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp&#39;  -D_REENTRANT -DRCPP_PARALLEL_USE_TBB=1   -I/usr/local/include   -fPIC  -Wall -g -O2  -c foo.c -o foo.o
In file included from &lt;built-in&gt;:1:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/Dense:1:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/Core:88:
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/src/Core/util/Macros.h:628:1: error: unknown type name &#39;namespace&#39;
namespace Eigen {
^
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/src/Core/util/Macros.h:628:16: error: expected &#39;;&#39; after top level declarator
namespace Eigen {
               ^
               ;
In file included from &lt;built-in&gt;:1:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/StanHeaders/include/stan/math/prim/mat/fun/Eigen.hpp:13:
In file included from /Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/Dense:1:
/Library/Frameworks/R.framework/Versions/4.0/Resources/library/RcppEigen/include/Eigen/Core:96:10: fatal error: &#39;complex&#39; file not found
#include &lt;complex&gt;
         ^~~~~~~~~
3 errors generated.
make: *** [foo.o] Error 1</code></pre>
<pre class="r"><code># make a function to update the model and the posterior sample output with your desired trait

s_female_calculator &lt;- function(selected_trait){
  
  data &lt;- Female_selection_data %&gt;% 
    select(&quot;fitness.early.f&quot;, selected_trait) %&gt;% 
    rename(V1 = 1, V2 = 2)
  
  model &lt;- update(
    s_female_model,
    newdata = data,
    chains = 4, cores = 4, 
    seed = 1)
  
  post &lt;- 
    posterior_samples(model) %&gt;% 
    rename(S_female = 1) %&gt;% 
    mutate(Trait = selected_trait) %&gt;% 
    select(Trait, S_female) %&gt;% 
    as_tibble() #%&gt;% 
  rbind(post) # we add this to the bottom of the previous post object: it gains progressively more rows as traits are added
  
  post
}

# create a list of all trait to iterate our function over

trait_list_female &lt;- colnames(Female_selection_data %&gt;% select(-fitness.early.f))

# run the function

if(!file.exists(&quot;data/S_female.csv&quot;)){
  S_female &lt;- map_dfr(trait_list_female, s_female_calculator)
  write_csv(S_female, file = &quot;data/S_female.csv&quot;)
} else S_female &lt;- read_csv(&quot;data/S_female.csv&quot;)

S_female &lt;-
  S_female %&gt;% 
  filter(Trait != &quot;line&quot;)

# now plot S_female for all traits

S_female_forest_plot &lt;-
  S_female %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_S = median(S_female)) %&gt;%
  ggplot(aes(S_female, fct_reorder(Trait, avg_S))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  #stat_pointinterval(aes(), .width = c(0.66, 0.95)) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-0.5, 1)) +
  ylab(&quot;Trait&quot;) +
  xlab(&quot;Estimated female selection differential&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())


# Calculate male selection differentials - identify all traits that don&#39;t end in .f

Male_selection_data &lt;- 
  DGRP_subset_scaled %&gt;% 
  select(!ends_with(&quot;.f&quot;))

# Repeat process for males

# make a function to update the model and the posterior sample output with your desired trait

s_male_calculator &lt;- function(selected_trait){
  
  male_data &lt;- Male_selection_data %&gt;% 
    select(&quot;fitness.early.m&quot;, selected_trait) %&gt;% 
    rename(V1 = 1, V2 = 2)
  
  model &lt;- update(
    s_female_model,
    newdata = male_data,
    chains = 4, cores = 4, 
    seed = 1)
  
  post &lt;- 
    posterior_samples(model) %&gt;% 
    rename(S_male = 1) %&gt;% 
    mutate(Trait = selected_trait) %&gt;% 
    select(Trait, S_male) %&gt;% 
    as_tibble() #%&gt;% 
  rbind(post) # we add this to the bottom of the previous post object: it gains progressively more rows as traits are added
  
  post
}

# create a list of all trait to iterate our function over

trait_list_male &lt;- colnames(Male_selection_data %&gt;% select(-fitness.early.m))

# run the function

if(!file.exists(&quot;data/S_male.csv&quot;)){
  S_male &lt;- map_dfr(trait_list_male, s_male_calculator)
  write_csv(S_male, file = &quot;data/S_male.csv&quot;)
} else S_male &lt;- read_csv(&quot;data/S_male.csv&quot;)

S_male &lt;-
  S_male %&gt;% 
  filter(Trait != &quot;line&quot;)

# now plot S_female for all traits

S_male_forest_plot &lt;-
  S_male %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_S = median(S_male)) %&gt;%
  ggplot(aes(S_male, fct_reorder(Trait, avg_S))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  #stat_pointinterval(aes(), .width = c(0.66, 0.95)) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-0.5, 1)) +
  ylab(NULL) +
  xlab(&quot;Estimated male selection differential&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

S_female_forest_plot + S_male_forest_plot</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-7-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/769e48d5a4afe34bb3a05faaa53829aeaadcfc05/docs/figure/Main_analysis.Rmd/unnamed-chunk-7-1.png" target="_blank">769e48d</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/figure/Main_analysis.Rmd/unnamed-chunk-7-1.png" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/11a83906f387d1ecbb39fc5ab60f5830bdac2d8f/docs/figure/Main_analysis.Rmd/unnamed-chunk-7-1.png" target="_blank">11a8390</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure XX</strong>: selection differentials for a) females and b) males</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="identifying-traits-under-sexually-antagonistic-selection" class="section level2">
<h2>Identifying traits under sexually antagonistic selection</h2>
<p><span class="math inline">\(~\)</span></p>
<p>We have found selection differentials acting on traits in females and males. For those traits where we have a measure of both, we can quantify sexually antagonistic selection by finding traits that have selection operating in opposite directions, depending on which sex the trait is expressed in.</p>
<p><a href="https://doi.org/10.1111/j.1558-5646.2010.01021.x">Innocenti and Morrow (2011)</a> present an index for measuring the intensity for sexually antagonistic selection:</p>
<p>Where <span class="math inline">\(B&#39;_F\)</span> = <span class="math inline">\(S_F\)</span> and <span class="math inline">\(B&#39;_M\)</span> = <span class="math inline">\(S_M\)</span> that we have calculated above.</p>
<p>From <strong>Innocenti and Morrow</strong>:</p>
<p>This index is positive when selection is concordant in the two sexes, negative when antagonistic in the two sexes, and is zero when selection is absent in one sex (note that it will miss conflict that occurs when strong stabilizing selection is present in that sex). Finally it is proportional to the absolute intensity of selection.</p>
<p>Additionally, it has the desirable properties of being symmetrical and normally distributed for a random set of normally distributed β′M,β′F. |I| is also always included in the interval between the absolute values of the selection gradient in the two sexes, and it coincides with them when β′M = β′F. As a potential drawback, it should be noted that such quantity is not defined when β′M = β′F = 0, even though it makes little sense to estimate how concordant or antagonstic directional selection is when it is absent in both sexes.</p>
<pre class="r"><code>S_female_a &lt;-
  S_female %&gt;% 
  filter(str_ends(Trait, &quot;.f&quot;)) %&gt;% 
  mutate(Trait = gsub(&#39;.{2}$&#39;, &#39;&#39;, Trait))

S_female_b &lt;-
  S_female %&gt;% 
  filter(!str_ends(Trait, &quot;.f&quot;)) 

S_female &lt;- rbind(S_female_a, S_female_b) %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(draw = 1:n())

S_male_a &lt;-
  S_male %&gt;% 
  filter(str_ends(Trait, &quot;.m&quot;)) %&gt;% 
  mutate(Trait = gsub(&#39;.{2}$&#39;, &#39;&#39;, Trait))

S_male_b &lt;-
  S_male %&gt;% 
  filter(!str_ends(Trait, &quot;.m&quot;))

S_male &lt;- rbind(S_male_a, S_male_b) %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(draw = 1:n())

# Create the combined dataset

S_complete &lt;- full_join(S_female, S_male)

# Calculate Morrow and Innocenti&#39;s concordance of selection index

S_complete &lt;-
  S_complete %&gt;% 
  select(-draw) %&gt;% 
  mutate(SI = S_female * S_male / sqrt(((S_female)^2 + (S_male)^2)/2))

Selection_index_forest_plot &lt;-
  S_complete %&gt;% filter(!is.na(SI)) %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_SI = median(SI)) %&gt;%
  ggplot(aes(SI, fct_reorder(Trait, avg_SI))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  #stat_pointinterval(aes(), .width = c(0.66, 0.95)) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-0.5, 1)) +
  ylab(NULL) +
  xlab(&quot;Concordance of selection index&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

Selection_index_forest_plot</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-8-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/83a0c4fdf0c246cc8f853dbeecd4ab8b252df66e/docs/figure/Main_analysis.Rmd/unnamed-chunk-8-1.png" target="_blank">83a0c4f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/figure/Main_analysis.Rmd/unnamed-chunk-8-1.png" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Here&#39;s a table showing the correlations

S_complete %&gt;% 
  group_by(Trait) %&gt;% 
  summarise()

my_data_table &lt;- function(df){
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c(&quot;Scroller&quot;,  &quot;Buttons&quot;),
    options = list(
      dom = &#39;Bfrtip&#39;,
      deferRender=TRUE,
      scrollX=TRUE, scrollY=800,
      scrollCollapse=TRUE,
      buttons =
        list(&#39;pageLength&#39;, &#39;colvis&#39;, &#39;csv&#39;, list(
          extend = &#39;pdf&#39;,
          pageSize = &#39;A4&#39;,
          orientation = &#39;landscape&#39;,
          filename = &#39;selection_data&#39;)),
      pageLength = 905
    )
  )
}

my_data_table(trait_selection_gradients)</code></pre>
<p><span class="math inline">\(~\)</span></p>
<pre class="r"><code># There are some issues here. It does not make sense to calculate the covariance between e.g. female fitness and a phenotype expressed in males, when we also have a measure of the phenotype in females.

# this is not perfect

ggplot(data = trait_selection_gradients, aes(x = `S female`, y = `S male`, fill = `antagonism index`)) +
  #geom_hex(bins = 20) +
  geom_jitter(shape = 21, alpha = 0.8, size = 4, show.legend = FALSE, width = 0.003, height = 0.003) +
  #geom_point(shape = 21, alpha = 0.9, size = 5, show.legend = FALSE) +
  geom_smooth(method = &#39;lm&#39;, color=&#39;black&#39;) +
  geom_hline(yintercept = 0, linetype = 2) +
  geom_vline(xintercept = 0, linetype = 2) +
  #coord_cartesian(xlim = c(-3, 3), ylim = c(-3, 3)) +
  scale_fill_carto_c(palette = &quot;Geyser&quot;,
                     breaks=c(-1, -0.5, 0, 0.5, 1),
                     limits=c(-1,1)) +
  theme_bw() +
  theme(panel.border= element_blank(),
        axis.line=element_line(),
        text = element_text(size=14),
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.minor.x = element_blank(),
        axis.title.x = element_text(size = 12),
        axis.title.y = element_text(size = 12))</code></pre>
<p><span class="math inline">\(~\)</span></p>
<p>While this figure comes from relatively rough data, we can see that trait values that are postively selected in females tend also to be positively selected in males. However, there are numerous traits that appear to be under sexually antagonstic selection: that is, trait values that make fit males make low fitness females and vice versa.</p>
</div>
<div id="female-fitness" class="section level2">
<h2>Female fitness</h2>
<pre class="r"><code># I&#39;m trying to find a way to automatically find the strongest correlations with each fitness component 

# Now let&#39;s plot the trait correlations that are most positively correlated  

early.female.1 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$glutamic.acid.low.yeast.f,
                          DGRP_no_line$fitness.early.f, pos_colour) +
  labs(x = &quot;Level of glutamic\nacid on low yeast\ndiet&quot;, y = &quot;Female fitness&quot;)

early.female.2 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$lifetime.fecundity.25C.f   ,
                          DGRP_no_line$fitness.early.f, pos_colour) +
  labs(x = &quot;Lifetime fecundity&quot;, y = NULL)

early.female.3 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$week1.fecundity.25C.f  ,
                          DGRP_no_line$fitness.early.f, pos_colour) +
  labs(x = &quot;Week 1 fecundity\n at 25C&quot;, y = NULL)

early.female.4 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$M.anisopliae.resistance.f  ,
                          DGRP_no_line$fitness.early.f, pos_colour) +
  labs(x = &quot;Resistance to\nM. anisopliae&quot;, y = NULL)

early.female.5 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$acetylcarnitine.low.yeast.f    ,
                          DGRP_no_line$fitness.early.f, pos_colour) +
  labs(x = &quot;Level of\nacetylcarnitine on low\nyeast diet&quot;, y = NULL)

early.female.6 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$circadian.period.mesa.f        ,
                          DGRP_no_line$fitness.early.f, pos_colour) +
  labs(x = &quot;Circadian period length&quot;, y = &quot;Female fitness&quot;)

early.female.7 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$ornithine.low.yeast.f  ,
                          DGRP_no_line$fitness.early.f, pos_colour) +
  labs(x = &quot;Level of ornithine\non low yeast diet&quot;, y = NULL)

early.female.8 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$P.aeruginosa.resistance.2017.f ,
                          DGRP_no_line$fitness.early.f, pos_colour) +
  labs(x = &quot;Resistance to\nP. aeruginosa&quot;, y = NULL)

early.female.9 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$serine.high.yeast.f    ,
                          DGRP_no_line$fitness.early.f, pos_colour) +
  labs(x = &quot;Level of serine\non low yeast diet&quot;, y = NULL)

early.female.10 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$week3.fecundity.25C.f  ,
                          DGRP_no_line$fitness.early.f, pos_colour) +
  labs(x = &quot;Week 3 fecundity\n at 25C&quot;, y = NULL)

# Now lets do the most negative correlations

early.female.11 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$epinephrine.high.yeast.f   ,
                          DGRP_no_line$fitness.early.f, neg_colour) +
  labs(x = &quot;Level of epinephrine\non high yeast diet&quot;, y = &quot;Female fitness&quot;)

early.female.12 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$cGMP.low.yeast.f   ,
                          DGRP_no_line$fitness.early.f, neg_colour) +
  labs(x = &quot;Level of cGMP\non low yeast diet&quot;, y = NULL)

early.female.13&lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$day.sleep.bout.length.f     ,
                          DGRP_no_line$fitness.early.f, neg_colour) +
  labs(x = &quot;Day sleep bout\n length&quot;, y = NULL)

early.female.14 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$Sphingomonas.wittichii.f    ,
                          DGRP_no_line$fitness.early.f, neg_colour) +
  coord_cartesian(xlim = c(-3, 5), ylim = c(-3, 3)) +
  labs(x = &quot;S. wittichii load\n(microbiome)&quot;, y = NULL)

early.female.15 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$`alpha-amanitin.resistance.0.2`    ,
                          DGRP_no_line$fitness.early.f, neg_colour) +
  labs(x = &quot;Resistance to mushroom toxin\n (alpha-amanitin 0.2 ug/g)&quot;, y = NULL)

early.female.16 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$sprinter.marathoner.f  ,
                          DGRP_no_line$fitness.early.f, neg_colour) +
  labs(x = &quot;&#39;Sprinter&#39; phenotype&quot;, y = &quot;Female fitness&quot;)

early.female.17 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$`CHC.8-C25:1.f`    ,
                          DGRP_no_line$fitness.early.f, neg_colour) +
  labs(x = &quot;Level of the CHC\n 8-C25:1&quot;, y = NULL)

early.female.18 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$cold.shock.young.f ,
                          DGRP_no_line$fitness.early.f, neg_colour) +
  labs(x = &quot;Resistance to cold\n shock&quot;, y = NULL)

early.female.19 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$proline.low.yeast.f    ,
                          DGRP_no_line$fitness.early.f, neg_colour) +
  labs(x = &quot;Level of proline\non low yeast diet&quot;, y = NULL)

early.female.20 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$`3-Hydroxykynurenine.high.yeast.f` ,
                          DGRP_no_line$fitness.early.f, neg_colour) +
  labs(x = &quot;Level of\n3-Hydroxykynurenine on\n low yeast diet&quot;, y = NULL)

(early.female.1 | early.female.2 | early.female.3 | early.female.4 | early.female.5) /
  (early.female.6 | early.female.7 | early.female.8 | early.female.9 | early.female.10) /
  (early.female.11 | early.female.12 | early.female.13| early.female.14| early.female.15) /
  (early.female.16 | early.female.17 | early.female.18 | early.female.19 | early.female.20)</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-12-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/83a0c4fdf0c246cc8f853dbeecd4ab8b252df66e/docs/figure/Main_analysis.Rmd/unnamed-chunk-12-1.png" target="_blank">83a0c4f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/figure/Main_analysis.Rmd/unnamed-chunk-12-1.png" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure 2</strong>: The 10 phenotypes with the strongest positive (orange points) and negative (blue points) selection coefficients in females. All traits are standardised to have a mean = 0 and SD = 1.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="male-fitness" class="section level2">
<h2>Male fitness</h2>
<pre class="r"><code># I&#39;m trying to find a way to automatically find the strongest correlations with each fitness component 

young.male.1 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$locomotor.frequency.f,
                          DGRP_no_line$fitness.early.m, pos_colour) +
  labs(x = &quot;Locomotor activity&quot;, y = &quot;Male fitness&quot;)

young.male.2 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$P.aeruginosa.resistance.2017.m,
                          DGRP_no_line$fitness.early.m, pos_colour) +
  labs(x = &quot;Resistance to bacterial infection\n (P. aeruginosa)&quot;, y = NULL)

young.male.3 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$lifespan.28C.m,
                          DGRP_no_line$fitness.early.m, pos_colour) +
  labs(x = &quot;Lifespan at 28C&quot;, y = NULL)

young.male.4 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$RCH.old.m,
                          DGRP_no_line$fitness.early.m, pos_colour) +
  labs(x = &quot;Rapid cold hardening\n in old flies&quot;, y = NULL)

young.male.5 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$Enterococcus.hirae.m   ,
                          DGRP_no_line$fitness.early.m, pos_colour) +
  labs(x = &quot;E. hirae load\n(microbiome)&quot;, y = NULL)

young.male.6 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$M.anisopliae.resistance.m  ,
                          DGRP_no_line$fitness.early.m, pos_colour) +
  labs(x = &quot;Resistance to fungal infection\n (M. anisopliae)&quot;,
       y = &quot;Male fitness&quot;)

young.male.7 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$acetylcholine.low.yeast.f  ,
                          DGRP_no_line$fitness.early.m, pos_colour) +
  labs(x = &quot;Level of\nacetylcholine on\n low yeast diet&quot;, y = NULL)

young.male.8 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$glucose.low.glucose.diet.m ,
                          DGRP_no_line$fitness.early.m, pos_colour) +
  labs(x = &quot;Level of glucose\non low glucose diet&quot;, y = NULL)

young.male.9 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$Streptomyces.sp..m ,
                          DGRP_no_line$fitness.early.m, pos_colour) +
  labs(x = &quot;Streptomyces sp. load\n(microbiome)&quot;, y = NULL)

young.male.10 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$day.sleep.bout.length.SD.m ,
                          DGRP_no_line$fitness.early.m, pos_colour) +
  labs(x = &quot;Variation in day\nsleep bout length&quot;, y = NULL)

# Now lets do the most negative correlations

young.male.11 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$`CHC.6,10-C26:2.f` ,
                          DGRP_no_line$fitness.early.m, neg_colour) +
  labs(x = &quot;Level of the CHC\n 6, 10-C26:2\n(only produced by females)&quot;, y = NULL)

young.male.12 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$P.rettgeri.load.high.glucose.diet.m     ,
                          DGRP_no_line$fitness.early.m, neg_colour) +
  labs(x = &quot;Bacterial load \n(P. rettgeri) 24hrs post\ninfection&quot;, y = NULL)

young.male.13 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$`3-Hydroxybutyric.acid.low.yeast.f`     ,
                          DGRP_no_line$fitness.early.m, neg_colour) +
  labs(x = &quot;Level of\3-Hydroxybutyric acid on\n low yeast diet&quot;, y = NULL)

young.male.14 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$cadaverine.high.yeast.f    ,
                          DGRP_no_line$fitness.early.m, neg_colour) +
  labs(x = &quot;Level of\ncadaverine on\n high yeast diet&quot;, y = NULL)

young.male.15 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$cold.shock.young.m ,
                          DGRP_no_line$fitness.early.m, neg_colour) +
  labs(x = &quot;Resistance to cold\nshock in young flies&quot;, y = &quot;Male fitness&quot;)

young.male.16 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$`CHC.5,9-C25:2 &amp; 9-C25:1.f`,
                          DGRP_no_line$fitness.early.m, neg_colour) +
  labs(x = &quot;Level of the CHCs\n5,9-C25:2 &amp; 9-C25:1\n(only produced by females)&quot;, y = NULL)

young.male.17 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$pantothenate.low.yeast.f,
                          DGRP_no_line$fitness.early.m, neg_colour) +
  labs(x = &quot;Level of\npantothenate on\n low yeast diet&quot;, y = NULL)

young.male.18 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$aminoisobutyrate.high.yeast.f  ,
                          DGRP_no_line$fitness.early.m, neg_colour) +
  labs(x = &quot;Level of\naminoisobutyrate on\n high yeast diet&quot;, y = NULL)

young.male.19 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$`CHC.8,12-C26:2.f` ,
                          DGRP_no_line$fitness.early.m, neg_colour) +
  labs(x = &quot;Level of the CHC\n8,12-C26:2\n(only produced by females)&quot;, y = NULL)

young.male.20 &lt;-
  fitness_regression_plot(DGRP_no_line, DGRP_no_line$myristic.acid.low.yeast.f      ,
                          DGRP_no_line$fitness.early.m, neg_colour) +
  labs(x = &quot;Level of myristic acid\n on low yeast diet&quot;, y = &quot;Male fitness&quot;)

(young.male.1 | young.male.2 | young.male.3 | young.male.4 | young.male.5) /
  (young.male.6 | young.male.7 | young.male.8 | young.male.9 | young.male.10) /
  (young.male.11 | young.male.12 | young.male.13 | young.male.14 | young.male.15) /
  (young.male.16 | young.male.17 | young.male.18 | young.male.19 | young.male.20)</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-13-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/83a0c4fdf0c246cc8f853dbeecd4ab8b252df66e/docs/figure/Main_analysis.Rmd/unnamed-chunk-13-1.png" target="_blank">83a0c4f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/figure/Main_analysis.Rmd/unnamed-chunk-13-1.png" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure 3</strong>: The 10 phenotypes with the strongest positive (orange points) and negative (blue points) selection coefficients in males. All phenotypes are standardised to have a mean = 0 and SD = 1.</p>
<p><span class="math inline">\(~\)</span></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] job_0.3.0           ggtext_0.1.1        ggtips_0.3.4       
 [4] ggExtra_0.9         broom_0.7.6         DT_0.18            
 [7] shiny_1.6.0         tidybayes_2.3.1     brms_2.15.0        
[10] Rcpp_1.0.6          patchwork_1.1.1     pander_0.6.3       
[13] kableExtra_1.3.4    wesanderson_0.3.6   rcartocolor_2.0.0  
[16] RColorBrewer_1.1-2  pheatmap_1.0.12     googlesheets4_0.3.0
[19] forcats_0.5.1       stringr_1.4.0       dplyr_1.0.5        
[22] purrr_0.3.4         readr_1.4.0         tidyr_1.1.3        
[25] tibble_3.1.1        ggplot2_3.3.3       tidyverse_1.3.1    
[28] workflowr_1.6.2    

loaded via a namespace (and not attached):
  [1] readxl_1.3.1         backports_1.2.1      systemfonts_1.0.1   
  [4] plyr_1.8.6           igraph_1.2.6         svUnit_1.0.6        
  [7] splines_4.0.5        crosstalk_1.1.1      rstantools_2.1.1    
 [10] inline_0.3.17        digest_0.6.27        htmltools_0.5.1.1   
 [13] rsconnect_0.8.17     fansi_0.4.2          magrittr_2.0.1      
 [16] modelr_0.1.8         RcppParallel_5.1.2   matrixStats_0.58.0  
 [19] xts_0.12.1           svglite_2.0.0        prettyunits_1.1.1   
 [22] colorspace_2.0-0     rvest_1.0.0          ggdist_2.4.0        
 [25] haven_2.4.0          xfun_0.22            callr_3.7.0         
 [28] crayon_1.4.1         jsonlite_1.7.2       lme4_1.1-26         
 [31] zoo_1.8-9            glue_1.4.2           gtable_0.3.0        
 [34] gargle_1.1.0         emmeans_1.5.5-1      webshot_0.5.2       
 [37] V8_3.4.0             distributional_0.2.2 pkgbuild_1.2.0      
 [40] rstan_2.21.2         abind_1.4-5          scales_1.1.1        
 [43] mvtnorm_1.1-1        DBI_1.1.1            miniUI_0.1.1.1      
 [46] gridtext_0.1.4       viridisLite_0.4.0    xtable_1.8-4        
 [49] StanHeaders_2.21.0-7 stats4_4.0.5         htmlwidgets_1.5.3   
 [52] httr_1.4.2           threejs_0.3.3        arrayhelpers_1.1-0  
 [55] ellipsis_0.3.1       farver_2.1.0         pkgconfig_2.0.3     
 [58] loo_2.4.1            sass_0.3.1           dbplyr_2.1.1        
 [61] utf8_1.2.1           labeling_0.4.2       tidyselect_1.1.0    
 [64] rlang_0.4.10         reshape2_1.4.4       later_1.1.0.1       
 [67] munsell_0.5.0        cellranger_1.1.0     tools_4.0.5         
 [70] cli_2.4.0            generics_0.1.0       ggridges_0.5.3      
 [73] evaluate_0.14        fastmap_1.1.0        yaml_2.2.1          
 [76] processx_3.5.1       knitr_1.32           fs_1.5.0            
 [79] nlme_3.1-152         whisker_0.4          mime_0.10           
 [82] projpred_2.0.2       xml2_1.3.2           compiler_4.0.5      
 [85] bayesplot_1.8.0      shinythemes_1.2.0    rstudioapi_0.13     
 [88] gamm4_0.2-6          curl_4.3             reprex_2.0.0        
 [91] statmod_1.4.35       bslib_0.2.4          stringi_1.5.3       
 [94] highr_0.9            ps_1.6.0             Brobdingnag_1.2-6   
 [97] lattice_0.20-41      Matrix_1.3-2         nloptr_1.2.2.2      
[100] markdown_1.1         shinyjs_2.0.0        vctrs_0.3.7         
[103] pillar_1.6.0         lifecycle_1.0.0      jquerylib_0.1.3     
[106] bridgesampling_1.1-2 estimability_1.3     httpuv_1.5.5        
[109] R6_2.5.0             promises_1.2.0.1     gridExtra_2.3       
[112] codetools_0.2-18     boot_1.3-27          colourpicker_1.1.0  
[115] MASS_7.3-53.1        gtools_3.8.2         assertthat_0.2.1    
[118] rprojroot_2.0.2      withr_2.4.2          shinystan_2.5.0     
[121] mgcv_1.8-35          parallel_4.0.5       hms_1.0.0           
[124] grid_4.0.5           coda_0.19-4          minqa_1.2.4         
[127] rmarkdown_2.7        googledrive_1.0.1    git2r_0.28.0        
[130] lubridate_1.7.10     base64enc_0.1-3      dygraphs_1.1.1.6    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
