<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Thomas Keaney and Luke Holman" />


<title>Intralocus sexual conflict in the DGRP</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.22/datatables.js"></script>
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.11.3/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.11.3/js/jquery.dataTables.min.js"></script>
<link href="site_libs/dt-ext-scroller-1.11.3/css/scroller.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-scroller-1.11.3/js/dataTables.scroller.min.js"></script>
<script src="site_libs/pdfmake-1.11.3/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.11.3/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.11.3/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.11.3/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.11.3/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.2.0/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Selection, sexual conflict and sex-specific parameters in the DGRP</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tomkeaney/DGRP_sexual_conflict">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Intralocus sexual conflict in the DGRP</h1>
<h4 class="author">Thomas Keaney and Luke Holman</h4>
<h4 class="date">25/06/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-05-12
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DGRP_sexual_conflict/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210706code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210706)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210706code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210706)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPsexualconflicttree95f4a97a95b2694250d10afcf070261b37734fb8targetblank95f4a97a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/tomkeaney/DGRP_sexual_conflict/tree/95f4a97a95b2694250d10afcf070261b37734fb8" target="_blank">95f4a97</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPsexualconflicttree95f4a97a95b2694250d10afcf070261b37734fb8targetblank95f4a97a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/tomkeaney/DGRP_sexual_conflict/tree/95f4a97a95b2694250d10afcf070261b37734fb8" target="_blank">95f4a97</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/.DS_Store
    Ignored:    gwas_data/

Untracked files:
    Untracked:  -Toms-laptop.Rhistory
    Untracked:  Manuscript/
    Untracked:  Reported_heritability.xlsx
    Untracked:  Rplot.pdf
    Untracked:  S_plots.pdf
    Untracked:  Selection_differential_plot.pdf
    Untracked:  Selection_differentials.pdf
    Untracked:  Selection_differentials_f.pdf
    Untracked:  Selection_differentials_m.pdf
    Untracked:  analysis/Random_plots.R
    Untracked:  code/LICENSE
    Untracked:  code/gcta64
    Untracked:  code/prettify
    Untracked:  code/toy.map
    Untracked:  code/toy.ped
    Untracked:  data/SNP_heritability.csv
    Untracked:  data/SNP_heritability_cleaned.csv
    Untracked:  data/SNP_selection.csv
    Untracked:  data/S_female.csv
    Untracked:  data/rfm_dimorphism_model.rds
    Untracked:  dimorphism_subset.pdf
    Untracked:  fitness_variation.pdf
    Untracked:  fits/
    Untracked:  lifetime_fecundity_S.pdf
    Untracked:  mating_latency_S.pdf

Unstaged changes:
    Modified:   .Rprofile
    Modified:   .gitattributes
    Modified:   .gitignore
    Modified:   DGRP_sexual_conflict.Rproj
    Modified:   README.md
    Modified:   _workflowr.yml
    Modified:   analysis/SNP_correlations.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/license.Rmd
    Modified:   analysis/process_mashr.Rmd
    Modified:   analysis/run_GWAS.Rmd
    Modified:   analysis/run_mashr.Rmd
    Modified:   code/README.md
    Modified:   code/run_mashr.R
    Modified:   data/README.md
    Modified:   data/S_male.csv
    Modified:   data/all.dgrp.phenos_unscaled.csv
    Modified:   data/rfm.complete.csv
    Modified:   output/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Main_analysis.Rmd</code>) and HTML (<code>docs/Main_analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/95f4a97a95b2694250d10afcf070261b37734fb8/analysis/Main_analysis.Rmd" target="_blank">95f4a97</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-12
</td>
<td>
including ideas section
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/7bf0ee08d2c8b770d1c2d8aca0b2931897462adb/docs/Main_analysis.html" target="_blank">7bf0ee0</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/5b9883ecc77ee0ab529aa2bd15efa0caf62b5f2b/analysis/Main_analysis.Rmd" target="_blank">5b9883e</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
<td>
Add GCTA model outputs
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/8e2d8be42669bfba416c2c2fd3668766ab4ba1e2/analysis/Main_analysis.Rmd" target="_blank">8e2d8be</a>
</td>
<td>
tkeaney
</td>
<td>
2021-08-17
</td>
<td>
Merge branch ‘master’ of <a href="https://github.com/tomkeaney/DGRP_sexual_conflict" class="uri">https://github.com/tomkeaney/DGRP_sexual_conflict</a>
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/4b095f010dea866a35dd2f7f6b021f5e7e27bd0d/analysis/Main_analysis.Rmd" target="_blank">4b095f0</a>
</td>
<td>
tkeaney
</td>
<td>
2021-08-17
</td>
<td>
edits
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/73f011cc9d650add1dd20c9f43e888fb17e4ddc5/analysis/Main_analysis.Rmd" target="_blank">73f011c</a>
</td>
<td>
lukeholman
</td>
<td>
2021-08-06
</td>
<td>
Tiny edit (digits=3)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/043902fed2d66e5b988a2f7a2dd70c734e1e95f8/docs/Main_analysis.html" target="_blank">043902f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/959123f80add4c3747918d1937e125003c982035/analysis/Main_analysis.Rmd" target="_blank">959123f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
trying to fix a bug
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/769e48d5a4afe34bb3a05faaa53829aeaadcfc05/docs/Main_analysis.html" target="_blank">769e48d</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/340c41bb74226b0fe4c241fc5d79663d2210c387/analysis/Main_analysis.Rmd" target="_blank">340c41b</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
sexual dimorphism calculation added
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/4018580c9b086ed875bd241ca2028263395bd971/docs/Main_analysis.html" target="_blank">4018580</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/65b25971953ba68583cb15a9503035ac59b1a16f/analysis/Main_analysis.Rmd" target="_blank">65b2597</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
sexual dimorphism calculation added
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/83a0c4fdf0c246cc8f853dbeecd4ab8b252df66e/docs/Main_analysis.html" target="_blank">83a0c4f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/ddea55e2cfaed0dddfe93d9dad2d1154f0e6a451/analysis/Main_analysis.Rmd" target="_blank">ddea55e</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
<td>
sexual dimorphism calculation added
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/b364d30085d882105bbb010749bcadccd8fa123a/docs/Main_analysis.html" target="_blank">b364d30</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/e404cf9f54cc05b642050a7b0bc0070a6e55f8bb/analysis/Main_analysis.Rmd" target="_blank">e404cf9</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
<td>
progression on selection calculations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/Main_analysis.html" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/6aad8a86f4107b143e2e07bb036e5064adeca7cc/analysis/Main_analysis.Rmd" target="_blank">6aad8a8</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
<td>
progression on selection calculations
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/11a83906f387d1ecbb39fc5ab60f5830bdac2d8f/docs/Main_analysis.html" target="_blank">11a8390</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-07
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/42b8f12ec94d1089c153f97f0e4090edd191a279/analysis/Main_analysis.Rmd" target="_blank">42b8f12</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-07
</td>
<td>
Get the site up and running
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages-and-the-data" class="section level2">
<h2>Load packages and the data</h2>
<p>First load the packages</p>
<pre class="r"><code>library(tidyverse) # for tidy coding
library(googlesheets4) # for reading google sheets
library(pheatmap) # for heatmaps
library(RColorBrewer) # for custom colours
library(rcartocolor) # cool colours
library(wesanderson) # cool colours
library(kableExtra) # for scrolling tables
library(pander) # more tables
library(patchwork) # for cool plots
library(brms) # for bayesian models
library(tidybayes) # for more bayesian things
library(shiny) # for interactive app
library(DT) # for interactive tables
library(broom) # convert results of functions into tables
library(ggExtra) # for marginal plots
library(ggtips) # for hovering tooltips in R shiny
library(ggtext) # for markdown features in ggplot
library(job) # to run things in the background

# for plots later

pos_colour &lt;- wes_palette(&quot;Zissou1&quot;)[4]
neg_colour &lt;- wes_palette(&quot;Zissou1&quot;)[2]

fitness_regression_plot &lt;- function(a, b, c) {
  a %&gt;% filter(Trait == b) %&gt;% 
    ggplot(aes(trait_value, female_fitness)) +
    geom_point(shape = 21, alpha = 0.75, size = 3, fill = c, show.legend = FALSE) +
    geom_smooth(method = &#39;lm&#39;, color=&#39;black&#39;) +
    geom_hline(yintercept = 0, linetype = 2) +
    geom_vline(xintercept = 0, linetype = 2) +
    #coord_cartesian(xlim = c(-3, 3), ylim = c(-3, 3)) +
    theme_bw() +
    theme(panel.border= element_blank(),
          axis.line=element_line(),
          text = element_text(size=14),
          panel.grid.major.x = element_blank(),
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank(),
          panel.grid.minor.x = element_blank(),
          axis.title.x = element_text(size = 12),
          axis.title.y = element_text(size = 12))
}</code></pre>
<p>Now load the data. Correlation analysis is not robust when sample sizes are small, so we only include traits that have been measured in 80+ lines.</p>
<pre class="r"><code>DGRP_data &lt;- read_csv(&quot;data/all.dgrp.phenos_unscaled.csv&quot;) %&gt;% 
  filter(Reference != &quot;Vonesch et al (2016) PLOS Genetics&quot;)

# Now trim the dataset so that it only includes traits that have been measured in 80+ lines - we are left with 955 traits from an original 1157

DGRP_subset &lt;- DGRP_data %&gt;% 
  group_by(Trait) %&gt;% 
  filter(length(line) &gt; 80) %&gt;% 
  ungroup()

# Add a study ID variable 

DGRP_subset &lt;- 
  DGRP_subset %&gt;% 
  group_by(Reference) %&gt;% 
  mutate(study_ID = cur_group_id()) %&gt;% 
  arrange(study_ID) %&gt;% 
  ungroup()

# Assign each trait to a category based on whether it was measured in 1) both sexes separately, 2) just females, 3) just males or 4) both sexes together. We can use a shortcut and use the study names as each study tends to be consistent with their measurement approach (with 4 exceptions).

DGRP_subset &lt;-
  DGRP_subset %&gt;% 
  mutate(rmf_applicable = if_else(Reference == &quot;Arya et al (2015) Chemical Senses&quot; | Reference == &quot;Battlay et al (2018) G3&quot; | Reference == &quot;Brown et al (2013) Genes, Brain and Behaviour&quot; | Reference == &quot;Campwell et al (2019) G3&quot; | Reference == &quot;Carbone et al (2017) PNAS&quot; | Reference == &quot;Chohan (2012) PhD thesis&quot; | Reference == &quot;Dembeck et al (2015) eLife&quot; | Reference == &quot;Everett et al (2020) Genome Research&quot; | Reference == &quot;Everman and Morgan (2017) Evolution&quot; | Reference == &quot;Everman et al (2019) Genetics&quot; | Reference == &quot;Garlapow et al (2015) PLOS One&quot; | Reference == &quot;Gerken et al (2015) PNAS&quot; | Reference == &quot;Grubbs et al (2013) PLOS One&quot; | Reference == &quot;Harbison et al (2018) Behavior Genetics&quot; | Reference == &quot;Harbison, McCoy and Mackay (2013) BMC Genomics&quot; | Reference == &quot;Huang et al (2020) PLOS Biology&quot; | Reference == &quot;Jordan et al (2012) PLOS One&quot; | Reference == &quot;Lecheta et al (2020) Frontiers in Genetics&quot; | Reference == &quot;Lovejoy et al (2021) Molecular Ecology&quot; | Reference == &quot;Mackay et al (2012) Nature&quot; | Reference == &quot;Morgante et al (2015) Scientific Reports&quot; | Reference == &quot;Morozova el al. (2015) BMC Genomics&quot; | Reference == &quot;Palmer et al (2018) PLOS Pathogens&quot; | Reference == &quot;Rajpurohit et al (2018) Molecular Ecology&quot; | Reference == &quot;Riddle (2020) Journal of Experimental Biology&quot; | Reference == &quot;Spierer et al (2021) PLOS Genetics&quot; | Reference == &quot;Swarup et al (2013) PNAS&quot; | Reference == &quot;Wang, Lu and St. Leger (2017) PLOS Pathogens&quot; | Reference == &quot;Watanabe et al (2020) G3&quot; | Reference == &quot;Weber et al (2012) PLOS One&quot; | Reference == &quot;Wong and Holman (in prep)&quot;| Reference == &quot;Wu et al (2018) SLEEPJ&quot; | Reference == &quot;Zhou et al (2016) Environmental Health Perspectives&quot;, &quot;YES&quot;, &quot;NO&quot;))

# Above, we identified the studies that measured traits in the sexes separately. However, some studies measure multiple traits, some of which they only measure in one sex or both (e.g. larval traits are often measured with the sexes pooled). Below we remove the specific traits that are not measured in both sexes separately from our already trimmed data  

DGRP_rmf &lt;- 
  DGRP_subset %&gt;% 
  filter(rmf_applicable == &quot;YES&quot;) %&gt;% 
  filter(Trait != &quot;anoxia.tolerance.larval&quot;,
         Trait != &quot;CHC.11 &amp; 9-Me-C23.f&quot;, 
         Trait != &quot;CHC.13- &amp; 11-Me-C25.f&quot;,
         Trait != &quot;CHC.2-Me-C26.m&quot;, 
         Trait !=&quot;CHC.2-Me-C27.m&quot;, 
         Trait !=&quot;CHC.2-Me-C28 &amp; 7,11-C29:2.f&quot;,
         Trait != &quot;CHC.2-Me-C28.m&quot;, 
         Trait != &quot;CHC.2-Me-C29 &amp; 8,12-C30:2.f&quot;,
         Trait != &quot;CHC.2-Me-C29.m&quot;, 
         Trait != &quot;CHC.2-Me-C30.f&quot;,
         Trait != &quot;CHC.3-Me-C23.m&quot;, 
         Trait != &quot;CHC.3-Me-C25 &amp; 9-C26:1.f&quot;,
         Trait != &quot;CHC.3-Me-C25.m&quot;,
         Trait != &quot;CHC.3-Me-C27 &amp; 6,10-C28:2 &amp; 9-C28:1.f&quot;,
         Trait != &quot;CHC.5,9-C25:2 &amp; 9-C25:1.f&quot;, 
         Trait != &quot;CHC.5,9-C27:2 &amp; 9-C27:1.f&quot;,
         Trait != &quot;CHC.5-C27:1.f&quot;, 
         Trait != &quot;CHC.5-Me-C25.f&quot;,
         Trait != &quot;CHC.6,10-C26:2.f&quot;, 
         Trait != &quot;CHC.6-C26:1 &amp; i.m&quot;,
         Trait != &quot;CHC.7,11-C23:2.f&quot;, 
         Trait != &quot;CHC.7,11-C24:2.f&quot;, 
         Trait != &quot;CHC.7,11-C25:2.f&quot;,
         Trait != &quot;CHC.7,11-C26:2.f&quot;, 
         Trait != &quot;CHC.7,11-C27:2 &amp; 2-Me-C26.f&quot;,
         Trait != &quot;CHC.7,11-C28:2 &amp; 2-Me-C27.f&quot;, 
         Trait != &quot;CHC.7,11-C31:2.f&quot;,
         Trait != &quot;CHC.7-C24:1.m&quot;,
         Trait != &quot;CHC.8,12-C26:2.f&quot;,
         Trait != &quot;CHC.8,12-C28:2.f&quot;,
         Trait != &quot;CHC.8-C25:1.f&quot;,
         Trait != &quot;CHC.9,13-C25:2.f&quot;, 
         Trait != &quot;CHC.9,13-C27:2.f&quot;,
         Trait != &quot;CHC.9-C25:1.m&quot;, 
         Trait != &quot;CHC.9-C27:1.m&quot;,
         Trait != &quot;CHC.9-C29:1 &amp; 5,9-C29:2.f&quot;, 
         Trait != &quot;CHC.n-C25.m&quot;,
         Trait != &quot;CHC.NI.20.f&quot;, 
         Trait != &quot;CHC.NI.33.f&quot;, 
         Trait != &quot;CHC.NI.34.f&quot;,
         Trait != &quot;CHC.NI.7.m&quot;, 
         Trait != &quot;CHC.x-C22:1a.m&quot;,
         Trait != &quot;CHC.x-C22:1b.m&quot;, 
         Trait != &quot;CHC.x-C23:1.f&quot;,
         Trait != &quot;kallithea.viral.load.f&quot;,
         Trait != &quot;dopamine.content.2021.m&quot;, 
         Trait != &quot;dopamine.content.paraquat.2021.m&quot;,
         Trait != &quot;dopamine.response.to.paraquat.2021.m&quot;)

# Make a female trait dataset

DGRP_rf_unscaled &lt;-
  DGRP_rmf %&gt;% filter(Sex == &quot;Female&quot;) %&gt;% 
  rename(Female_trait_value = trait_value) %&gt;% 
  select(-c(Sex, `Trait description`, study_ID, rmf_applicable)) %&gt;% 
  mutate(Trait = gsub(&#39;.{2}$&#39;, &#39;&#39;, Trait)) # remove the final two characters from the name of each trait: .f

DGRP_rf &lt;-
  DGRP_rf_unscaled %&gt;% 
  group_by(Trait) %&gt;% # group the data by trait so that scaling works on a per trait basis
  mutate(Female_trait_value = as.numeric(scale(Female_trait_value))) %&gt;% # scale the data for each trait 
  ungroup() 
 
# Make a male trait dataset          

DGRP_rm_unscaled &lt;-
  DGRP_rmf %&gt;% filter(Sex == &quot;Male&quot;) %&gt;% 
  rename(Male_trait_value = trait_value) %&gt;% 
  select(-c(Sex, `Trait description`, study_ID, rmf_applicable)) %&gt;%
  mutate(Trait = gsub(&#39;.{2}$&#39;, &#39;&#39;, Trait)) # remove the final two characters from the name of each trait: .m

DGRP_rm &lt;-
  DGRP_rm_unscaled %&gt;% 
  group_by(Trait) %&gt;% # group the data by trait so that scaling works on a per trait basis
  mutate(Male_trait_value = as.numeric(scale(Male_trait_value))) %&gt;% # scale the data for each trait
  ungroup() 

# Join the two together      

rfm_unscaled_data &lt;- full_join(DGRP_rf_unscaled, DGRP_rm_unscaled)

rfm_data &lt;- full_join(DGRP_rf, DGRP_rm)

# Now return to our larger tibble: DGRP_subset. Do some wrangling and scale it. Scaling is also applied to each trait value, so that each trait has a mean of 0 and an SD of 1. This makes the variation found for each trait broadly comparable with all other traits.

s_data &lt;-
  DGRP_subset %&gt;%  filter(Trait != &quot;fitness.early.f&quot; &amp;
                            Trait != &quot;fitness.early.m&quot;)
s_data_1 &lt;-
  DGRP_subset %&gt;%  filter(Trait == &quot;fitness.early.f&quot;) %&gt;% 
  rename(female_fitness = trait_value) %&gt;% 
  select(line, female_fitness)


s_data_2 &lt;-
  DGRP_subset %&gt;%  filter(Trait == &quot;fitness.early.m&quot;) %&gt;% 
  rename(male_fitness = trait_value) %&gt;% 
  select(line, male_fitness)

s_data &lt;-
  left_join(s_data, s_data_1)

s_data &lt;-
  left_join(s_data, s_data_2) %&gt;% 
  group_by(Trait) %&gt;% # group the data by trait so that scaling works on a per trait basis
  mutate(trait_value = as.numeric(scale(trait_value))) # scale the data for each trait</code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="sexual-dimorphism-and-the-intersex-genetic-correlation" class="section level1">
<h1>Sexual dimorphism and the intersex genetic correlation</h1>
<p><span class="math inline">\(~\)</span></p>
<div id="calculating-raz-fm-for-all-traits" class="section level2">
<h2>Calculating r<sup>A</sup><sub><em>z fm</em></sub> for all traits</h2>
<p><span class="math inline">\(~\)</span></p>
<p>The traits we consider in this analysis fall into four categories. 1) They have been measured in the two sexes separately, 2) they have only been measured females, 3) they have only been measured in males and 4) they have been measured across the two sexes. For those traits that fall into category 1, we can calculate r<sup>A</sup><sub><em>fm</em></sub>.</p>
<p>We use bayesian models to find the residual correlation between traits expressed in males and females. Each line represents a different genotype, and we have measures for both sexes per line. This correlation therefore corresponds to the additive intersex genetic correlation for each trait.</p>
<p>A problem here is that my current approach assumes that each trait is normally distributed, which is clearly false. This is the same problem that researchers face with the spearman correlation coefficient, and the main reason we have used the pearson method in earlier analyses. To solve this problem, I can categorise each trait into a ‘distribution family’ then run the models in subsets. Using this bayesian approach is preferred because it gives us measures of uncertainty which you don’t get from the pearson frequentist approach. <strong>Currently not done</strong></p>
<pre class="r"><code># find all the traits in our dataset

check &lt;- unique(rfm_data$Trait)

# this is our model outline - we can update this with new filtered data to calculate rfm for every trait. We do this because we don&#39;t want brms to recompile every time. With the model already compiled we can simply update, which saves a load of time.

rfm_model &lt;-
  brm(data = rfm_data %&gt;% filter(Trait == &quot;flight.performance&quot;),
      family = gaussian,
      bf(mvbind(Female_trait_value, Male_trait_value) ~ 0, sigma ~ 0) + set_rescor(TRUE),
      prior(lkj(2), class = rescor),
      chains = 4, cores = 4, 
      seed = 1)   

# sample from the posterior - this will create a tibble with two columns - Trait and rfm

post &lt;- posterior_samples(rfm_model) %&gt;% 
  rename(rfm = rescor__Femaletraitvalue__Maletraitvalue) %&gt;% 
  mutate(Trait = &quot;flight.performance&quot;) %&gt;% 
  select(Trait, rfm) %&gt;% 
  as_tibble()

# make a function to update the model and the posterior sample output with your desired trait

rfm_calculator &lt;- function(selected_trait){
  
  data &lt;- rfm_data %&gt;% filter(Trait == selected_trait)
  
  model &lt;- update(
    rfm_model, newdata = data,
    chains = 4, cores = 4, 
    seed = 1)
  
  post &lt;- 
    posterior_samples(model) %&gt;% 
    rename(rfm = rescor__Femaletraitvalue__Maletraitvalue) %&gt;% 
    mutate(Trait = selected_trait,
           `Trait guild` = unique(data$`Trait guild`)) %&gt;% 
    select(Trait, rfm, `Trait guild`) %&gt;% 
    as_tibble()
  
  post
}

# create a list of all trait to iterate our function over

trait_list &lt;- unique(rfm_data$Trait)

#trait_list_test_subset &lt;- rfm_data %&gt;% filter(`Trait guild` == &quot;Insecticide response&quot;)
#trait_list_test_subset &lt;- unique(trait_list_test_subset$Trait)

#rfm_test &lt;- map_dfr(trait_list_test_subset, rfm_calculator)

# run the function

if(!file.exists(&quot;data/rfm.complete.csv&quot;)){
  rfm_complete &lt;- map_dfr(trait_list, rfm_calculator) # map_dfr returns a data frame created by row-binding each output
  write_csv(rfm_complete, file = &quot;data/rfm.complete.csv&quot;)
} else rfm_complete &lt;- read_csv(&quot;data/rfm.complete.csv&quot;)



# now plot rfm for all traits

rfm_forest_plot &lt;-
  rfm_complete %&gt;%
  filter(`Trait guild` != &quot;CHC&quot;) %&gt;% 
  filter(`Trait guild` != &quot;Microbiome&quot;) %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_rfm = median(rfm)) %&gt;%
  ggplot(aes(rfm, fct_reorder(Trait, avg_rfm))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  #stat_pointinterval(aes(), .width = c(0.66, 0.95)) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-0.1, 1)) +
  ylab(&quot;Trait&quot;) +
  xlab(&quot;Estimated intersex genetic correlation (_r^fm_ )&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

rfm_microbiome_forest_plot &lt;-
  rfm_complete %&gt;%
  filter(`Trait guild` == &quot;Microbiome&quot;) %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_rfm = median(rfm)) %&gt;%
  ggplot(aes(rfm, fct_reorder(Trait, avg_rfm))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  #stat_pointinterval(aes(), .width = c(0.66, 0.95)) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-0.2, 1)) +
  ylab(&quot;Microbiome trait&quot;) +
  xlab(&quot;Estimated intersex genetic correlation (_r^fm_ )&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

rfm_CHC_forest_plot &lt;-
  rfm_complete %&gt;%
  filter(`Trait guild` == &quot;CHC&quot;) %&gt;% 
  mutate(Trait = str_remove(Trait, &quot;CHC.&quot;)) %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_rfm = median(rfm)) %&gt;%
  ggplot(aes(rfm, fct_reorder(Trait, avg_rfm))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  #stat_pointinterval(aes(), .width = c(0.66, 0.95)) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-0.5, 1)) +
  ylab(&quot;CHC&quot;) +
  xlab(&quot;Estimated intersex genetic correlation (_r^fm_ )&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())


rfm_forest_plot + rfm_microbiome_forest_plot + rfm_CHC_forest_plot</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-3-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/7bf0ee08d2c8b770d1c2d8aca0b2931897462adb/docs/figure/Main_analysis.Rmd/unnamed-chunk-3-1.png" target="_blank">7bf0ee0</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/b364d30085d882105bbb010749bcadccd8fa123a/docs/figure/Main_analysis.Rmd/unnamed-chunk-3-1.png" target="_blank">b364d30</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/figure/Main_analysis.Rmd/unnamed-chunk-3-1.png" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/11a83906f387d1ecbb39fc5ab60f5830bdac2d8f/docs/figure/Main_analysis.Rmd/unnamed-chunk-3-1.png" target="_blank">11a8390</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure 1</strong>: The intersex additive genetic correlation for traits that have been measured in both sexes in &gt; 80 DGRP lines. A value of 0 indicates that there is no intersex genetic correlation, while a value of 1 means that additive genetic effects in one sex affect the other one identically. Data are means (depicted as 5% uncertainty intervals) with 66% and 95% uncertainty intervals.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="calculating-sexual-dimorphism-for-all-traits" class="section level2">
<h2>Calculating sexual dimorphism for all traits</h2>
<p><span class="math inline">\(~\)</span></p>
<p>Sexual dimorphism is a distinct concept from the intersex genetic correlation. The intersex genetic correlation is the correlation between the effects of alleles on females and males. When this is high, the sexes are affected by alleles in a similar way. However, a high <span class="math inline">\(r_{fm}\)</span> does not necessarily main that there is not sexual dimorphism, which is the difference in the trait mean between females and males. If processes other than genetic variation affect phenotype, then sexual dimorphism can be decoupled from <span class="math inline">\(r_{fm}\)</span>. For example, suppose an allele affects a trait expressed by both sexes in the same direction, but that it is expressed to different extents in the two sexes. This will result in a positive genetic correlation, but a sexually dimorphic trait.</p>
<p>Recall that we calculated <span class="math inline">\(r_{fm}\)</span> as the residual correlation between trait means for each line that had been scaled within their sex groupings. To calculate sexual dimorphism, we instead scale trait means across both sexes. This means that a male that has a high trait value relative to other males can still have a low trait value if females have higher trait values on average.</p>
<p>We fit the model <code>Trait_value ~ 1 + (Sex|Trait)</code> to estimate the effect of sex on trait means.Specifically, use the <span class="math inline">\(\beta\)</span> estimate for the effect of <code>Sex:Male</code> on the mean for each trait to estimate sexual dimorphism.</p>
<pre class="r"><code>d1 &lt;- 
  rfm_unscaled_data %&gt;% 
  select(-Male_trait_value) %&gt;% 
  mutate(Sex = &quot;Female&quot;) %&gt;% 
  rename(Trait_value = Female_trait_value)

d2 &lt;-
  rfm_unscaled_data %&gt;% 
  select(-Female_trait_value) %&gt;% 
  mutate(Sex = &quot;Male&quot;) %&gt;% 
  rename(Trait_value = Male_trait_value)

dimorphism_data &lt;- rbind(d1, d2) %&gt;% 
  group_by(Trait) %&gt;% # group the data by trait so that scaling works on a per trait basis
  mutate(Trait_value = as.numeric(scale(Trait_value))) %&gt;% # scale the data for each trait
  ungroup() 

# now we model dimorphism score to find an average value for each trait 

#job::job({
dimorphism_model &lt;-
  brm(data = dimorphism_data,
      family = gaussian,
      Trait_value ~ 1 + (Sex|Trait),
      prior = c(prior(normal(0, 5), class = Intercept),
                prior(cauchy(0, 2), class = sd)),
      chains = 4, cores = 4, warmup = 3000, iter = 8000, 
      seed = 1, file = &quot;fits/dimorphism_model&quot;) 
#})


# Now find the difference between males and females for each trait. This is calculated in the model as the trait specific effect of being male. We can find posterior estimates for these parameters using posterior samples(). Each trait has a column of estimated values named r_Trait[**insert specified trait**,SexMale]. We wrangle the data so that we get two columns: one with the trait name and another with an estimated value sampled from the posterior distribution for that parameter.

# Interpreting the dimorphism values: positive values indicate that males have higher trait values than females, a value of 0 indicates that there is no sexual dimorphism, and negative values indicate that females have higher trait values

dimorphism_post &lt;- posterior_samples(dimorphism_model) %&gt;% 
  as_tibble() %&gt;% 
  select(starts_with(&quot;r_Trait&quot;) &amp;
           contains(&quot;SexMale&quot;)) %&gt;% 
  pivot_longer(cols = everything(), names_to = &quot;Trait&quot;, values_to = &quot;dimorphism_value&quot;) %&gt;% 
  mutate(Trait = str_remove(Trait, &quot;r_Trait&quot;),
         Trait =  str_remove(Trait, &quot;,SexMale&quot;),
         Trait = gsub(&quot;\\[|\\]&quot;, &quot;&quot;, Trait)) %&gt;% 
  arrange(Trait)

# Now add the guild data to our tibble

d3 &lt;- dimorphism_data %&gt;% select(Trait, `Trait guild`) %&gt;% 
  distinct(Trait, `Trait guild`)

dimorphism_post &lt;-
  left_join(dimorphism_post, d3)

# plot dimorphism for all traits

forest_plot_dimorphism &lt;-
  dimorphism_post %&gt;% 
  filter(`Trait guild` != &quot;CHC&quot;) %&gt;% 
  filter(`Trait guild` != &quot;Microbiome&quot;) %&gt;%
  group_by(Trait) %&gt;% 
  mutate(avg_dimorph = median(dimorphism_value)) %&gt;%
  ggplot(aes(dimorphism_value, fct_reorder(Trait, avg_dimorph))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-2, 2)) +
  ylab(&quot;Trait&quot;) +
  xlab(&quot;Estimated sexual dimorphism&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

forest_plot_microbiome_dimorphism &lt;-
  dimorphism_post %&gt;% 
  filter(`Trait guild` == &quot;Microbiome&quot;) %&gt;%
  group_by(Trait) %&gt;% 
  mutate(avg_dimorph = median(dimorphism_value)) %&gt;%
  ggplot(aes(dimorphism_value, fct_reorder(Trait, avg_dimorph))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  #stat_pointinterval(aes(), .width = c(0.66, 0.95)) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-2, 2)) +
  ylab(&quot;Microbiome trait&quot;) +
  xlab(&quot;Estimated sexual dimorphism&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

forest_plot_CHC_dimorphism &lt;-
  dimorphism_post %&gt;% 
  filter(`Trait guild` == &quot;CHC&quot;) %&gt;% 
  mutate(Trait = str_remove(Trait, &quot;CHC.&quot;)) %&gt;%
  group_by(Trait) %&gt;% 
  mutate(avg_dimorph = median(dimorphism_value)) %&gt;%
  ggplot(aes(dimorphism_value, fct_reorder(Trait, avg_dimorph))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  #stat_pointinterval(aes(), .width = c(0.66, 0.95)) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-2, 2)) +
  ylab(&quot;CHC&quot;) +
  xlab(&quot;Estimated sexual dimorphism&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

forest_plot_dimorphism + forest_plot_microbiome_dimorphism + forest_plot_CHC_dimorphism</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-4-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/7bf0ee08d2c8b770d1c2d8aca0b2931897462adb/docs/figure/Main_analysis.Rmd/unnamed-chunk-4-1.png" target="_blank">7bf0ee0</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/769e48d5a4afe34bb3a05faaa53829aeaadcfc05/docs/figure/Main_analysis.Rmd/unnamed-chunk-4-1.png" target="_blank">769e48d</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/figure/Main_analysis.Rmd/unnamed-chunk-4-1.png" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure 2</strong>: Sexual dimorphism scores for traits that have been phenotyped in females and males. Positive values indicate that males have higher trait values than females, a value of 0 indicates that there is no sexual dimorphism, and negative values indicate that females have higher trait values than males. Bars represent means (displayed as 5% CI intervals), 66% and 95% intervals.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
</div>
<div id="what-effect-does-the-intersex-genetic-correlation-have-on-sexual-dimorphism" class="section level1">
<h1>What effect does the intersex genetic correlation have on sexual dimorphism?</h1>
<p><span class="math inline">\(~\)</span></p>
<p>To answer this question, we take the absolute value of our dimorphism measure. While this means we are unable to identify which sex has the higher trait value, it allows us to test whether there is a negative relationship between <span class="math inline">\(r_{fm}\)</span> and sexual dimorphism, as predicted if <span class="math inline">\(r_{fm}\)</span> holds the sexes back from attaining their phenotypic optima.</p>
<pre class="r"><code>abs_dimorphism_summary &lt;-
  dimorphism_post %&gt;% 
  group_by(Trait) %&gt;% 
  mean_qi(dimorphism_value) %&gt;% 
  rename(dimorphism_lower = .lower,
         dimorphism_upper = .upper) %&gt;% 
  select(Trait, dimorphism_value, dimorphism_lower, dimorphism_upper) %&gt;% 
  mutate(dimorphism_direction = if_else(dimorphism_value &gt; 0, &quot;Male&quot;, &quot;Female&quot;),
         dimorphism_value = abs(dimorphism_value),
         dimorphism_lower = abs(dimorphism_lower),
         dimorphism_upper = abs(dimorphism_upper))

rfm_summary &lt;-
  rfm_complete %&gt;% 
  group_by(Trait) %&gt;% 
  mean_qi(rfm) %&gt;% 
  rename(rfm_lower = .lower,
         rfm_upper = .upper) %&gt;% 
  select(Trait, rfm, rfm_lower, rfm_upper)

rfm_dimorphism &lt;-
  left_join(rfm_summary, abs_dimorphism_summary)

rfm_dimorphism &lt;- 
  full_join(rfm_dimorphism, d3)

rfm_dimorphism_plot &lt;-
    rfm_dimorphism %&gt;% 
    ggplot(aes(x = rfm, y = dimorphism_value)) +
    #geom_hex(bins = 20) +
    #scale_fill_carto_c(palette = &quot;Temps&quot;) +
    geom_point(aes(fill = pos_colour), shape = 21, alpha = 0.75, size = 4, show.legend = FALSE) +
    geom_smooth(method = &#39;lm&#39;, color=&#39;black&#39;) +
    labs(x = &quot;Intersex genetic correlation&quot;, y = &quot;Sexual dimorphism (standard deviations)&quot;) +
    theme_bw() +
    theme(panel.border= element_blank(),
          axis.line=element_line(),
          text = element_text(size=14),
          panel.grid.major.x = element_blank(),
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank(),
          panel.grid.minor.x = element_blank(),
          axis.title.x = element_text(size = 12),
          axis.title.y = element_text(size = 12))


rfm_dimorphism_plot_guilds &lt;-
    rfm_dimorphism %&gt;% 
    filter(`Trait guild` != &quot;Pathogen response&quot; &amp;
             `Trait guild` != &quot;Reproduction&quot; &amp;
             `Trait guild` != &quot;Stress response&quot; &amp;
             `Trait guild` != &quot;Drug response&quot; &amp;
             `Trait guild` != &quot;Insecticide response&quot; &amp;
             `Trait guild` != &quot;Morphological&quot; &amp;
             `Trait guild` != &quot;Oxidative stress resistance&quot;) %&gt;% 
    ggplot(aes(x = rfm, y = dimorphism_value)) +
    #geom_hex(bins = 20) +
    #scale_fill_carto_c(palette = &quot;Temps&quot;) +
    geom_point(aes(fill = pos_colour), shape = 21, alpha = 0.75, size = 4, show.legend = FALSE) +
    geom_smooth(method = &#39;lm&#39;, color=&#39;black&#39;) +
    labs(x = &quot;Intersex genetic correlation&quot;, y = &quot;Sexual dimorphism (standard deviations)&quot;) +
    theme_bw() +
    facet_wrap(~`Trait guild`) +
    theme(panel.border= element_blank(),
          axis.line=element_line(),
          text = element_text(size=14),
          panel.grid.major.x = element_blank(),
          panel.grid.major.y = element_blank(),
          panel.grid.minor.y = element_blank(),
          panel.grid.minor.x = element_blank(),
          axis.title.x = element_text(size = 12),
          axis.title.y = element_text(size = 12))

rfm_dimorphism_plot /
  rfm_dimorphism_plot_guilds</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/7bf0ee08d2c8b770d1c2d8aca0b2931897462adb/docs/figure/Main_analysis.Rmd/unnamed-chunk-5-1.png" target="_blank">7bf0ee0</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/figure/Main_analysis.Rmd/unnamed-chunk-5-1.png" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/11a83906f387d1ecbb39fc5ab60f5830bdac2d8f/docs/figure/Main_analysis.Rmd/unnamed-chunk-5-1.png" target="_blank">11a8390</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># the distribution family is not ideal. Ask Luke.

rfm_dimorphism_model &lt;-
  brm(dimorphism_value ~ 1 + rfm, 
      data = rfm_dimorphism, family = skew_normal(),
      prior = c(prior(normal(0, 5), class = Intercept),
                prior(normal(0, 2), class = b)),
      warmup = 2000, iter = 6000, chains = 4, cores = 4, seed = 1,
      file = &quot;data/rfm_dimorphism_model&quot;)

#pp_check(rfm_dimorphism_model)</code></pre>
<p><strong>Figure 3</strong>: The intersex genetic correlation for a trait does not affect sexual dimorphism in the trait. Panel a shows the entire dataset, while panels b-g show traits broken down into broad guilds. Note that guilds with less than 10 traits are not shown.</p>
<p><span class="math inline">\(~\)</span></p>
<p>This analysis is not perfect! However, there does not appear to be any substantial effect of the intersex genetic correlation on sexual dimorphism This suggests that sharing a genome may not be the constraint on adaptation that is often assumed (at least in the DGRP).</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="selection-differentials-and-the-opportunity-for-selection" class="section level1">
<h1>Selection differentials and the opportunity for selection</h1>
<p><span class="math inline">\(~\)</span></p>
<p>There is variance in fitness for both sexes</p>
<pre class="r"><code># Calculate female selection differentials - identify all traits that don&#39;t end in .m

Female_selection_data &lt;- 
  s_data %&gt;%
  filter(!str_ends(Trait, &quot;.m&quot;))


Female_fitness &lt;-
Female_selection_data %&gt;% filter(Trait == &quot;fitness.late.f&quot;) %&gt;% 
  ggplot(aes(x = female_fitness)) +
  geom_bar(fill = &quot;#f17854&quot;) +
  scale_x_binned(breaks = c(-5, -4, -3, -2, -1, 0, 1, 2)) +
  theme_minimal() +
  labs(x = &quot;Female fitness (standardised)&quot;,
       y = &quot;No. lines&quot;)

# Calculate male selection differentials - identify all traits that don&#39;t end in .f

Male_selection_data &lt;- 
  s_data %&gt;%
  filter(!str_ends(Trait, &quot;.f&quot;))


Male_fitness &lt;-
Male_selection_data %&gt;% filter(Trait == &quot;fitness.late.m&quot;) %&gt;% 
  ggplot(aes(x = male_fitness)) +
  geom_bar(fill = &quot;#f17854&quot;) +
  scale_x_binned(breaks = c(-5, -4, -3, -2, -1, 0, 1, 2, 3)) +
  theme_minimal() +
  labs(x = &quot;Male fitness (standardised)&quot;,
       y = NULL)

Female_fitness + Male_fitness</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/7bf0ee08d2c8b770d1c2d8aca0b2931897462adb/docs/figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">7bf0ee0</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/769e48d5a4afe34bb3a05faaa53829aeaadcfc05/docs/figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">769e48d</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/83a0c4fdf0c246cc8f853dbeecd4ab8b252df66e/docs/figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">83a0c4f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/11a83906f387d1ecbb39fc5ab60f5830bdac2d8f/docs/figure/Main_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">11a8390</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure XX</strong>: variation in fitness for a) females and b) males.</p>
<p><span class="math inline">\(~\)</span></p>
<div id="calculating-i-the-opportunity-for-selection" class="section level2">
<h2>Calculating <em>I</em>: the opportunity for selection</h2>
<p><span class="math inline">\(~\)</span></p>
<p>We run into the <strong>problem</strong> that fitness has been standardised, meaning that variance is uniform and known i.e. SD = 1. I think we need raw values?</p>
<p><em>I</em> is the variance in relative fitness within a population, and represents the upper bound for a change in any trait due to selection within a generation. The usefulness of <em>I</em> for working out how selection changes trait values depends upon the correlation of the trait in question with relative fitness. The stronger this correlation, the more insightful <em>I</em> becomes. This correlation is the Price-Robertson identity, <span class="math inline">\(S = \sigma(z, w)\)</span>, which we shall consider later.</p>
<p>Here we calculate <em>I</em> to set a bound for the maximum intensity of selection that is possible in our population, within the environmental conditions that our fitness measurements occurred. <em>I</em> is estimated by <span class="math display">\[\hat{I} =\frac{n}{n-1}(\overline{w}^{2} - 1)\]</span> where n is the number of individuals that we have measured fitness for (in our case lines), and <span class="math inline">\(\overline{w}^{2}\)</span> is the mean of the squared relative fitness in the population. To calculate relative fitness we need to know the mean proportion of offspring produced across lines (our measure of fitness), then divide each lines individual fitness measure by this number. This will not work for scaled data. <strong>Ask Luke for unscaled means</strong></p>
<p><strong>No work on this yet</strong></p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="calculating-s-the-selection-differential" class="section level2">
<h2>Calculating <em>S</em>: the selection differential</h2>
<p><span class="math inline">\(~\)</span></p>
<p>The selection differential is the change in trait mean that occurs within a generation. Positive values indicate positive directional selection, while negative values indicate negative directional selection. We can calculate the selection differential associated with each phenotype using the <strong>Robertson Price identity</strong>, which states that the selection differential is equivalent to the covariance between phenotype and relative fitness</p>
<p><span class="math inline">\(S = \sigma(w_i, z_i)\)</span></p>
<p>or</p>
<p><span class="math inline">\(S = cov(relative.fitness, standarised.phenotype)\)</span></p>
<p>To do this, I use bayesian analysis, so that we get uncertainty estimates on the selection differentials. We have standardised each trait to have mean = 0 and SD = 1. Therefore, if <span class="math inline">\(S\)</span> = 0.25, this can be interpreted as a change of 0.25 standard deviations in the mean within a generation. However, this does not represent a change in the mean across generations because <span class="math inline">\(S\)</span> provides no information about the inheritance of the trait.</p>
<p>We can also calculate the broad sense heritability associated with each phenotype… if we could calculate the narrow sense heritability we could use the <strong>Breeder’s equation</strong>, <span class="math inline">\(\Delta Z = h^2S\)</span>, to calculate the response to selection for each phenotype (I don’t think we can do this).</p>
<p><span class="math inline">\(~\)</span></p>
<pre class="r"><code>female_selection_model &lt;-
  brm(data = Female_selection_data %&gt;% filter(Trait == &quot;flight.performance.f&quot;),
      family = gaussian,
      bf(mvbind(female_fitness, trait_value) ~ 0, sigma ~ 0) + set_rescor(TRUE),
      prior(lkj(2), class = rescor),
      chains = 4, cores = 4, 
      seed = 1)   

# sample from the posterior - this will create a tibble with two columns - Trait and rfm

post &lt;- posterior_samples(female_selection_model) %&gt;% 
  rename(S_female = rescor__femalefitness__traitvalue) %&gt;% 
  mutate(Trait = &quot;flight.performance&quot;) %&gt;% 
  select(Trait, S_female) %&gt;% 
  as_tibble()

# make a function to update the model and the posterior sample output with your desired trait

S_female_calculator &lt;- function(selected_trait){
  
  data &lt;- Female_selection_data %&gt;% filter(Trait == selected_trait)
  
  model &lt;- update(
    female_selection_model, newdata = data,
    chains = 4, cores = 4, 
    seed = 1)
  
  post &lt;- 
    posterior_samples(model) %&gt;% 
    rename(S_female = rescor__femalefitness__traitvalue) %&gt;% 
    mutate(Trait = selected_trait,
           `Trait guild` = unique(data$`Trait guild`)) %&gt;% 
    select(Trait, S_female, `Trait guild`) %&gt;% 
    as_tibble()
  
  post
}

trait_list_female &lt;- unique(Female_selection_data$Trait)

#trait_list_test_subset &lt;- Female_selection_data %&gt;% filter(`Trait guild` == &quot;Stress response&quot;)
#trait_list_test_subset &lt;- unique(trait_list_test_subset$Trait)

#S_f_test &lt;- map_dfr(trait_list_test_subset, S_female_calculator)

# run the function

if(!file.exists(&quot;data/S_female.csv&quot;)){
  S_female &lt;- map_dfr(trait_list_female, S_female_calculator) # map_dfr returns a data frame created by row-binding each output
  write_csv(S_female, file = &quot;data/S_female.csv&quot;)
} else S_female &lt;- read_csv(&quot;data/S_female.csv&quot;)

# now plot S_female for all traits

S_female_forest_plot &lt;-
  S_female %&gt;% 
  filter(`Trait guild` != &quot;Microbiome&quot; &amp;
           `Trait guild` != &quot;CHC&quot; &amp;
           `Trait guild` != &quot;Metabolome&quot;) %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_S = median(S_female)) %&gt;%
  ggplot(aes(S_female, fct_reorder(Trait, avg_S))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-0.5, 1)) +
  ylab(&quot;Trait&quot;) +
  xlab(&quot;Estimated female selection differential&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

# microbiome traits only

S_female_microbiome_plot &lt;-
  S_female %&gt;% 
  filter(`Trait guild` == &quot;Microbiome&quot;) %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_S = median(S_female)) %&gt;%
  ggplot(aes(S_female, fct_reorder(Trait, avg_S))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-0.5, 1)) +
  ylab(&quot;Microbiome trait&quot;) +
  xlab(&quot;Estimated female selection differential&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

# CHCs

S_female_CHC_plot &lt;-
  S_female %&gt;% 
  filter(`Trait guild` == &quot;CHC&quot;) %&gt;% 
  mutate(Trait = str_remove(Trait, &quot;CHC.&quot;)) %&gt;%
  group_by(Trait) %&gt;% 
  mutate(avg_S = median(S_female)) %&gt;%
  ggplot(aes(S_female, fct_reorder(Trait, avg_S))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-0.5, 1)) +
  ylab(&quot;CHC abundance&quot;) +
  xlab(&quot;Estimated female selection differential&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

# Metabolome 

S_female_metabolome_plot &lt;-
  S_female %&gt;% 
  filter(`Trait guild` == &quot;Metabolome&quot;) %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_S = median(S_female)) %&gt;%
  ggplot(aes(S_female, fct_reorder(Trait, avg_S))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-0.5, 1)) +
  ylab(&quot;Metabolome (abundance)&quot;) +
  xlab(&quot;Estimated female selection differential&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

(S_female_forest_plot | S_female_microbiome_plot) /
(S_female_CHC_plot | S_female_metabolome_plot)</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-7-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/7bf0ee08d2c8b770d1c2d8aca0b2931897462adb/docs/figure/Main_analysis.Rmd/unnamed-chunk-7-1.png" target="_blank">7bf0ee0</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/769e48d5a4afe34bb3a05faaa53829aeaadcfc05/docs/figure/Main_analysis.Rmd/unnamed-chunk-7-1.png" target="_blank">769e48d</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/figure/Main_analysis.Rmd/unnamed-chunk-7-1.png" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/11a83906f387d1ecbb39fc5ab60f5830bdac2d8f/docs/figure/Main_analysis.Rmd/unnamed-chunk-7-1.png" target="_blank">11a8390</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure XX</strong>: selection differentials for traits expressed in females</p>
<pre class="r"><code>male_selection_model &lt;-
  brm(data = Male_selection_data %&gt;% filter(Trait == &quot;flight.performance.m&quot;),
      family = gaussian,
      bf(mvbind(male_fitness, trait_value) ~ 0, sigma ~ 0) + set_rescor(TRUE),
      prior(lkj(2), class = rescor),
      chains = 4, cores = 4, 
      seed = 1)   

# sample from the posterior - this will create a tibble with two columns - Trait and rfm

post &lt;- posterior_samples(male_selection_model) %&gt;% 
  rename(S_male = rescor__malefitness__traitvalue) %&gt;% 
  mutate(Trait = &quot;flight.performance&quot;) %&gt;% 
  select(Trait, S_male) %&gt;% 
  as_tibble()

# make a function to update the model and the posterior sample output with your desired trait

S_male_calculator &lt;- function(selected_trait){
  
  data &lt;- Male_selection_data %&gt;% filter(Trait == selected_trait)
  
  model &lt;- update(
    male_selection_model, newdata = data,
    chains = 4, cores = 4, 
    seed = 1)
  
  post &lt;- 
    posterior_samples(model) %&gt;% 
    rename(S_male = rescor__malefitness__traitvalue) %&gt;% 
    mutate(Trait = selected_trait,
           `Trait guild` = unique(data$`Trait guild`)) %&gt;% 
    select(Trait, S_male, `Trait guild`) %&gt;% 
    as_tibble()
  
  post
}

trait_list_male &lt;- unique(Male_selection_data$Trait)

# run the function

if(!file.exists(&quot;data/S_male.csv&quot;)){
  S_male &lt;- map_dfr(trait_list_male, S_male_calculator) # map_dfr returns a data frame created by row-binding each output
  write_csv(S_male, file = &quot;data/S_male.csv&quot;)
} else S_male &lt;- read_csv(&quot;data/S_male.csv&quot;)

# now plot S_male for all traits

S_male_forest_plot &lt;-
  S_male %&gt;% 
  filter(`Trait guild` != &quot;Microbiome&quot; &amp;
           `Trait guild` != &quot;CHC&quot; &amp;
           `Trait guild` != &quot;Metabolome&quot; ) %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_S = median(S_male)) %&gt;%
  ggplot(aes(S_male, fct_reorder(Trait, avg_S))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-0.5, 1)) +
  ylab(&quot;Trait&quot;) +
  xlab(&quot;Estimated male selection differential&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

# microbiome traits only

S_male_microbiome_plot &lt;-
  S_male %&gt;% 
  filter(`Trait guild` == &quot;Microbiome&quot;) %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_S = median(S_male)) %&gt;%
  ggplot(aes(S_male, fct_reorder(Trait, avg_S))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-0.5, 1)) +
  ylab(&quot;Microbiome trait&quot;) +
  xlab(&quot;Estimated male selection differential&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

# CHCs

S_male_CHC_plot &lt;-
  S_male %&gt;% 
  filter(`Trait guild` == &quot;CHC&quot;) %&gt;% 
  mutate(Trait = str_remove(Trait, &quot;CHC.&quot;)) %&gt;%
  group_by(Trait) %&gt;% 
  mutate(avg_S = median(S_male)) %&gt;%
  ggplot(aes(S_male, fct_reorder(Trait, avg_S))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-0.5, 1)) +
  ylab(&quot;CHC abundance&quot;) +
  xlab(&quot;Estimated male selection differential&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

# Metabolome 

S_male_metabolome_plot &lt;-
  S_male %&gt;% 
  filter(`Trait guild` == &quot;Metabolome&quot;) %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_S = median(S_male)) %&gt;%
  ggplot(aes(S_male, fct_reorder(Trait, avg_S))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-0.5, 1)) +
  ylab(&quot;Metabolome (abundance)&quot;) +
  xlab(&quot;Estimated male selection differential&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

(S_male_forest_plot | S_male_microbiome_plot) /
(S_male_CHC_plot | S_male_metabolome_plot)</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/7bf0ee08d2c8b770d1c2d8aca0b2931897462adb/docs/figure/Main_analysis.Rmd/unnamed-chunk-8-1.png" target="_blank">7bf0ee0</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/043902fed2d66e5b988a2f7a2dd70c734e1e95f8/docs/figure/Main_analysis.Rmd/unnamed-chunk-8-1.png" target="_blank">043902f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/83a0c4fdf0c246cc8f853dbeecd4ab8b252df66e/docs/figure/Main_analysis.Rmd/unnamed-chunk-8-1.png" target="_blank">83a0c4f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/figure/Main_analysis.Rmd/unnamed-chunk-8-1.png" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="top-10-bottom-10-traits-under-selection-in-females" class="section level2">
<h2>Top 10 / bottom 10 traits under selection in females</h2>
<pre class="r"><code># Now let&#39;s plot the trait correlations that are most positively correlated  

female.1 &lt;-
  fitness_regression_plot(Female_selection_data, &quot;glutamic.acid.low.yeast.f&quot;, pos_colour) +
  labs(x = &quot;Level of glutamic\nacid on low yeast\ndiet&quot;, y = &quot;Female fitness&quot;) +
  annotate(&quot;text&quot;, label = &quot;S = 0.27 (0.1-0.42)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;)

female.2 &lt;-
  fitness_regression_plot(Female_selection_data, &quot;lifetime.fecundity.25C.f&quot;, pos_colour) +
  labs(x = &quot;Lifetime fecundity at 25C&quot;, y = NULL) +
  annotate(&quot;text&quot;, label = &quot;S = 0.28 (0.1-0.44)&quot;, x = 2, y = -2, size = 4, colour = &quot;black&quot;)

female.3 &lt;-
fitness_regression_plot(Female_selection_data, &quot;week1.fecundity.25C.f&quot;, pos_colour) +
  labs(x = &quot;Week 1 fecundity\n at 25C&quot;, y = NULL) +
  annotate(&quot;text&quot;, label = &quot;S = 0.27 (0.09-0.43)&quot;, x = 1.75, y = -2, size = 4, colour = &quot;black&quot;)

female.4 &lt;-
fitness_regression_plot(Female_selection_data, &quot;M.anisopliae.resistance.f&quot;, pos_colour) +
  labs(x = &quot;Resistance to\nM. anisopliae&quot;, y = NULL) +
  annotate(&quot;text&quot;, label = &quot;S = 0.24 (0.07-0.39)&quot;, x = 2, y = -4, size = 4, colour = &quot;black&quot;)

female.5 &lt;-
  fitness_regression_plot(Female_selection_data, &quot;serine.high.yeast.f&quot;, pos_colour) +   
  labs(x = &quot;Level of serine\n on high yeast diet&quot;, y = NULL) +
  annotate(&quot;text&quot;, label = &quot;S = 0.25 (0.06-0.41)&quot;, x = 1.75, y = -3, size = 4, colour = &quot;black&quot;)

female.6 &lt;-
  fitness_regression_plot(Female_selection_data, &quot;circadian.period.mesa.f&quot;, pos_colour) +   
  labs(x = &quot;Circadian period length&quot;, y = &quot;Female fitness&quot;) +
  coord_cartesian(xlim = c(-3, 3)) +
  annotate(&quot;text&quot;, label = &quot;S = 0.28 (0.05-0.47)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;)

female.7 &lt;-
  fitness_regression_plot(Female_selection_data, &quot;ornithine.low.yeast.f&quot;, pos_colour) + 
  labs(x = &quot;Level of ornithine\non low yeast diet&quot;, y = NULL) +
  annotate(&quot;text&quot;, label = &quot;S = 0.23 (0.05-0.38)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;)

female.8 &lt;-
  fitness_regression_plot(Female_selection_data, &quot;acetylcarnitine.low.yeast.f&quot;, pos_colour) +   
  labs(x = &quot;Level of acetylcarnitine\non low yeast diet&quot;, y = NULL) +
  annotate(&quot;text&quot;, label = &quot;S = 0.22 (0.05-0.37)&quot;, x = -2.5, y = -3, size = 4, colour = &quot;black&quot;)

female.9 &lt;-
  fitness_regression_plot(Female_selection_data, &quot;week3.fecundity.25C.f&quot;, pos_colour) + 
  labs(x = &quot;Week 3 fecundity\n at 25C&quot;, y = NULL) +
  annotate(&quot;text&quot;, label = &quot;S = 0.23 (0.05-0.39)&quot;, x = 1.8, y = -2.5, size = 4, colour = &quot;black&quot;)

female.10 &lt;-
  fitness_regression_plot(Female_selection_data, &quot;glycerate.high.yeast.f&quot;, pos_colour) +    
  labs(x = &quot;Level of glycerate\n on high yeast diet&quot;, y = NULL) +
  annotate(&quot;text&quot;, label = &quot;S = 0.22 (0.04-0.39)&quot;, x = 1.8, y = -3, size = 4, colour = &quot;black&quot;)

# Now lets do the most negative correlations

female.11 &lt;-
  fitness_regression_plot(Female_selection_data, &quot;epinephrine.high.yeast.f&quot;, neg_colour) +
  labs(x = &quot;Level of epinephrine\non high yeast diet&quot;, y = &quot;Female fitness&quot;) +
  annotate(&quot;text&quot;, label = &quot;S = -0.32 (-0.47 to -0.13)&quot;, x = 1.8, y = -3, size = 4, colour = &quot;black&quot;)

female.12 &lt;-
  fitness_regression_plot(Female_selection_data, &quot;day.sleep.bout.length.f&quot;, neg_colour) +
  labs(x = &quot;Day sleep bout\n length&quot;, y = NULL) +
  annotate(&quot;text&quot;, label = &quot;S = -0.29 (-0.45 to -0.09)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;)

female.13&lt;-
  fitness_regression_plot(Female_selection_data, &quot;cGMP.low.yeast.f&quot;, neg_colour) +
  labs(x = &quot;Level of cGMP\non low yeast diet&quot;, y = NULL) +
  annotate(&quot;text&quot;, label = &quot;S = -0.26 (-0.41 to -0.09)&quot;, x = -1.25, y = -3, size = 4, colour = &quot;black&quot;)

female.14 &lt;-
  fitness_regression_plot(Female_selection_data, &quot;Sphingomonas.wittichii.f&quot;, neg_colour) +
  labs(x = &quot;S. wittichii load\n(microbiome)&quot;, y = NULL) +
  annotate(&quot;text&quot;, label = &quot;S = -0.24 (-0.38 to -0.08)&quot;, x = -2, y = -3, size = 4, colour = &quot;black&quot;)

female.15 &lt;-
  fitness_regression_plot(Female_selection_data, &quot;alpha-amanitin.resistance.0.2&quot;, neg_colour) +
  labs(x = &quot;Resistance to mushroom toxin\n (alpha-amanitin 0.2 ug/g)&quot;, y = NULL) +
  annotate(&quot;text&quot;, label = &quot;S = -0.26 (-0.42 to -0.08)&quot;, x = 1.1, y = -3, size = 4, colour = &quot;black&quot;)

female.16 &lt;-
  fitness_regression_plot(Female_selection_data, &quot;proline.low.yeast.f&quot;, neg_colour) +
  labs(x = &quot;Level of proline\non low yeast diet&quot;, y = &quot;Female fitness&quot;) +
  annotate(&quot;text&quot;, label = &quot;S = -0.24 (-0.39 to -0.06)&quot;, x = -1.4, y = -3, size = 4, colour = &quot;black&quot;)

female.17 &lt;-
  fitness_regression_plot(Female_selection_data, &quot;CHC.8-C25:1.f&quot;, neg_colour) +
  labs(x = &quot;Level of the CHC\n 8-C25:1&quot;, y = NULL) +
  annotate(&quot;text&quot;, label = &quot;S = -0.23 (-0.38 to -0.04)&quot;, x = 2.3, y = -3, size = 4, colour = &quot;black&quot;)

female.18 &lt;-
  fitness_regression_plot(Female_selection_data, &quot;waking.activity.f&quot;, neg_colour) +
  labs(x = &quot;Activity while awake&quot;, y = NULL) +
  annotate(&quot;text&quot;, label = &quot;S = -0.25 (-0.42 to -0.04)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;)

female.19 &lt;-
  fitness_regression_plot(Female_selection_data, &quot;3-Hydroxykynurenine.high.yeast.f&quot;, neg_colour) +
  labs(x = &quot;Level of\n3-Hydroxykynurenine on\n low yeast diet&quot;, y = NULL) +
  annotate(&quot;text&quot;, label = &quot;S = -0.22 (-0.39 to -0.04)&quot;, x = -1.5, y = -3, size = 4, colour = &quot;black&quot;)

female.20 &lt;-
  fitness_regression_plot(Female_selection_data, &quot;sprinter.marathoner.f&quot;, neg_colour) +
  labs(x = &quot;&#39;Sprinter&#39; phenotype&quot;, y = &quot;Female fitness&quot;) +
  annotate(&quot;text&quot;, label = &quot;S = -0.23 (-0.4 to -0.03)&quot;, x = -1.5, y = -3, size = 4, colour = &quot;black&quot;)

(female.1 | female.2 | female.3 | female.4 | female.5) /
  (female.6 | female.7 | female.8 | female.9 | female.10) /
  (female.11 | female.12 | female.13| female.14| female.15) /
  (female.16 | female.17 | female.18 | female.19 | female.20)</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-9-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/7bf0ee08d2c8b770d1c2d8aca0b2931897462adb/docs/figure/Main_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">7bf0ee0</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/043902fed2d66e5b988a2f7a2dd70c734e1e95f8/docs/figure/Main_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">043902f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/769e48d5a4afe34bb3a05faaa53829aeaadcfc05/docs/figure/Main_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">769e48d</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/83a0c4fdf0c246cc8f853dbeecd4ab8b252df66e/docs/figure/Main_analysis.Rmd/unnamed-chunk-9-1.png" target="_blank">83a0c4f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure 4</strong>: The 10 phenotypes with the strongest positive (orange points) and negative (blue points) selection differentials in females. All traits are standardised to have a mean = 0 and SD = 1.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="top-10-bottom-10-traits-under-selection-in-males" class="section level2">
<h2>Top 10 / bottom 10 traits under selection in males</h2>
<pre class="r"><code>male.1 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;lifespan.28C.m&quot;, pos_colour) +  
  annotate(&quot;text&quot;, label = &quot;S = 0.26 (0.09-0.40)&quot;, x = -1.5, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Lifespan at 28C&quot;, y = NULL)

male.2 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;Enterococcus.hirae.m&quot;, pos_colour) +    
  annotate(&quot;text&quot;, label = &quot;S = 0.23 (0.06-0.38)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;E. hirae load\n(microbiome)&quot;, y = NULL)

male.3 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;M.anisopliae.resistance.m&quot;, pos_colour) +   
  annotate(&quot;text&quot;, label = &quot;S = 0.23 (0.05-0.39)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Resistance to fungal infection\n (M. anisopliae)&quot;,
       y = &quot;Male fitness&quot;)

male.4 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;Streptomyces.sp..m&quot;, pos_colour) +  
  annotate(&quot;text&quot;, label = &quot;S = 0.22 (0.04-0.38)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Streptomyces sp. load\n(microbiome)&quot;, y = NULL)

male.5 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;RCH.old.m&quot;, pos_colour) +   
  annotate(&quot;text&quot;, label = &quot;S = 0.26 (0.04-0.45)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Rapid cold hardening\n in old flies&quot;, y = NULL)

male.6 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;micro.environmental.plasticity.lifespan.25C.m&quot;, pos_colour) +   
  annotate(&quot;text&quot;, label = &quot;S = 0.21 (0.03-0.38)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Within-line variance\nin lifespan&quot;, y = NULL)


male.7 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;Acinetobacter.johnsonii.m&quot;, pos_colour) +   
  annotate(&quot;text&quot;, label = &quot;S = 0.20 (0.01-0.37)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;A. johnsonii load\n(microbiome)&quot;, y = &quot;Male fitness&quot;)

male.8 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;day.sleep.bout.length.SD.m&quot;, pos_colour) +  
  annotate(&quot;text&quot;, label = &quot;S = 0.18 (0.01-0.35)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Variation in day\nsleep bout length&quot;, y = NULL)

male.9 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;climbing.ability.paraquat.m&quot;, pos_colour) + 
  annotate(&quot;text&quot;, label = &quot;S = 0.18 (0.01-0.34)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Climbing ability with\n high oxidative stress&quot;, y = NULL)

male.10 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;day.sleep.bout.length.m&quot;, pos_colour) + 
  annotate(&quot;text&quot;, label = &quot;S = 0.16 (0.01-0.31)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Day sleep bout length&quot;, y = NULL)

# Now lets do the most negative correlations

male.11 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;mating.latency.2015.m&quot;, neg_colour) +   
  annotate(&quot;text&quot;, label = &quot;S = -0.27 (-0.42 to -0.1)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Mating latency&quot;, y = NULL)

male.12 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;P.rettgeri.load.high.glucose.diet.m&quot;, neg_colour) + 
  annotate(&quot;text&quot;, label = &quot;S = -0.26 (-0.42 to -0.08)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Bacterial load \n(P. rettgeri) 24hrs post\ninfection&quot;, y = NULL)

male.13 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;olfactory.response.ethylacetate.m&quot;, neg_colour) +   
  annotate(&quot;text&quot;, label = &quot;S = -0.21 (-0.37 to -0.04)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Olfactory response to\nethylacetate&quot;, y = NULL)

male.14 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;triglyceride.gnotobiotic.m&quot;, neg_colour) +  
  annotate(&quot;text&quot;, label = &quot;S = -0.27 (-0.47 to -0.03)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Triglyceride levels\nwith a microbiome&quot;, y = NULL)

male.15 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;cold.shock.young.m&quot;, neg_colour) +  
  annotate(&quot;text&quot;, label = &quot;S = 0.24 (-0.43 to -0.03)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Resistance to cold\nshock in young flies&quot;, y = &quot;Male fitness&quot;)

male.16 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;CT.min.within.environment.variation.plasticity.m&quot;, neg_colour) +    
  annotate(&quot;text&quot;, label = &quot;S = -0.19 (-0.35 to -0.02)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Variation in critical\n thermal minimum&quot;, y = &quot;Male fitness&quot;)

male.17 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;CHC.2-Me-C22.m&quot;, neg_colour) +  
  annotate(&quot;text&quot;, label = &quot;S = -0.2 (-0.36 to -0.01)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Level of the CHC\n2-Me-C22&quot;, y = NULL)

male.18 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;imidacloprid.resistance.100ppm&quot;, neg_colour) +  
  annotate(&quot;text&quot;, label = &quot;S = -0.18 (-0.35 to -0.01)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Resistance to imidacloprid&quot;, y = NULL)

male.19 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;day.sleep.bouts.m&quot;, neg_colour) +   
  annotate(&quot;text&quot;, label = &quot;S = -0.19 (-0.36 to -0.00)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Number of day sleep bouts&quot;, y = NULL)

male.20 &lt;-
  fitness_regression_plot(Male_selection_data, &quot;day.sleep.SD.m&quot;, neg_colour) +  
  annotate(&quot;text&quot;, label = &quot;S = -0.16 (-0.31 to 0.01)&quot;, x = 2, y = -3, size = 4, colour = &quot;black&quot;) +
  labs(x = &quot;Variation in amount of\nsleep during the day&quot;, y = NULL)

(male.1 | male.2 | male.3 | male.4 | male.5) /
  (male.6 | male.7 | male.8 | male.9 | male.10) /
  (male.11 | male.12 | male.13 | male.14 | male.15) /
  (male.16 | male.17 | male.18 | male.19 | male.20)</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-10-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/7bf0ee08d2c8b770d1c2d8aca0b2931897462adb/docs/figure/Main_analysis.Rmd/unnamed-chunk-10-1.png" target="_blank">7bf0ee0</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure 5</strong>: The 10 phenotypes with the strongest positive (orange points) and negative (blue points) selection coefficients in males. All phenotypes are standardised to have a mean = 0 and SD = 1.</p>
</div>
<div id="identifying-traits-under-sexually-antagonistic-selection" class="section level2">
<h2>Identifying traits under sexually antagonistic selection</h2>
<p><span class="math inline">\(~\)</span></p>
<p>We have found selection differentials acting on traits in females and males. For those traits where we have a measure of both, we can quantify sexually antagonistic selection by finding traits that have selection operating in opposite directions, depending on which sex the trait is expressed in.</p>
<p><a href="https://doi.org/10.1111/j.1558-5646.2010.01021.x">Innocenti and Morrow (2011)</a> present an index for measuring the intensity for sexually antagonistic selection:</p>
<p>Where <span class="math inline">\(B&#39;_F\)</span> = <span class="math inline">\(S_F\)</span> and <span class="math inline">\(B&#39;_M\)</span> = <span class="math inline">\(S_M\)</span> that we have calculated above.</p>
<p>From <strong>Innocenti and Morrow</strong>:</p>
<p>This index is positive when selection is concordant in the two sexes, negative when antagonistic in the two sexes, and is zero when selection is absent in one sex (note that it will miss conflict that occurs when strong stabilizing selection is present in that sex). Finally it is proportional to the absolute intensity of selection.</p>
<p>Additionally, it has the desirable properties of being symmetrical and normally distributed for a random set of normally distributed β′M,β′F. |I| is also always included in the interval between the absolute values of the selection gradient in the two sexes, and it coincides with them when β′M = β′F. As a potential drawback, it should be noted that such quantity is not defined when β′M = β′F = 0, even though it makes little sense to estimate how concordant or antagonstic directional selection is when it is absent in both sexes.</p>
<pre class="r"><code>S_female_a &lt;-
  S_female %&gt;% 
  filter(str_ends(Trait, &quot;.f&quot;)) %&gt;% 
  mutate(Trait = gsub(&#39;.{2}$&#39;, &#39;&#39;, Trait))

S_female_b &lt;-
  S_female %&gt;% 
  filter(!str_ends(Trait, &quot;.f&quot;)) 

S_female &lt;- rbind(S_female_a, S_female_b) %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(draw = 1:n())

S_male_a &lt;-
  S_male %&gt;% 
  filter(str_ends(Trait, &quot;.m&quot;)) %&gt;% 
  mutate(Trait = gsub(&#39;.{2}$&#39;, &#39;&#39;, Trait))

S_male_b &lt;-
  S_male %&gt;% 
  filter(!str_ends(Trait, &quot;.m&quot;))

S_male &lt;- rbind(S_male_a, S_male_b) %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(draw = 1:n())

# Create the combined dataset

S_complete &lt;- full_join(S_female, S_male)

# Calculate Morrow and Innocenti&#39;s concordance of selection index

S_complete &lt;-
  S_complete %&gt;% 
  mutate(Selection_index = S_female * S_male / sqrt(((S_female)^2 + (S_male)^2)/2))

Selection_index_forest_plot &lt;-
  S_complete %&gt;% 
  filter(!is.na(Selection_index)) %&gt;% 
  group_by(Trait) %&gt;% 
  mutate(avg_Selection_index = median(Selection_index)) %&gt;%
  ggplot(aes(Selection_index, fct_reorder(Trait, avg_Selection_index))) +
  stat_interval(.width = c(0.05, 0.66, 0.95), 
                height = 3, show.legend = F) +
  rcartocolor::scale_color_carto_d(palette = &quot;Peach&quot;) +
  #stat_pointinterval(aes(), .width = c(0.66, 0.95)) +
  geom_vline(linetype = 2, xintercept = 0) +
  coord_cartesian(xlim = c(-1, 1)) +
  ylab(NULL) +
  xlab(&quot;Concordance of selection index&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank(),
        axis.title.x = ggtext::element_markdown())

Selection_index_forest_plot</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-11-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/7bf0ee08d2c8b770d1c2d8aca0b2931897462adb/docs/figure/Main_analysis.Rmd/unnamed-chunk-11-1.png" target="_blank">7bf0ee0</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure 6</strong>: Sexual concordance of selection for traits measured across the DGRP.</p>
<pre class="r"><code># Here&#39;s a table showing the correlations
S_complete_summary &lt;-
  S_complete %&gt;%
  group_by(Trait) %&gt;% 
  mean_qi(S_female, S_male, Selection_index) %&gt;% 
  select(-c(.width, .point, .interval)) %&gt;% 
  mutate(S_female = round(S_female, 3),
         S_female.lower = round(S_female.lower, 3),
         S_female.upper = round(S_female.upper, 3),
         S_male = round(S_male, 3),
         S_male.lower = round(S_male.lower, 3),
         S_male.upper = round(S_male.upper, 3),
         Selection_index = round(Selection_index, 3),
         Selection_index.lower = round(Selection_index.lower, 3),
         Selection_index.upper = round(Selection_index.upper, 3))

my_data_table &lt;- function(df){
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c(&quot;Scroller&quot;,  &quot;Buttons&quot;),
    options = list(
      dom = &#39;Bfrtip&#39;,
      deferRender=TRUE,
      scrollX=TRUE, scrollY=800,
      scrollCollapse=TRUE,
      buttons =
        list(&#39;pageLength&#39;, &#39;colvis&#39;, &#39;csv&#39;, list(
          extend = &#39;pdf&#39;,
          pageSize = &#39;A4&#39;,
          orientation = &#39;landscape&#39;,
          filename = &#39;selection_data&#39;)),
      pageLength = 673
    )
  )
}

my_data_table(S_complete_summary)</code></pre>
<div id="htmlwidget-2a1ffe038dddd448f159" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-2a1ffe038dddd448f159">{"x":{"filter":"none","vertical":false,"extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["1-Methyladenosine.high.yeast","1-Methyladenosine.low.yeast","1-Methylguanosine.high.yeast","1-Methylguanosine.low.yeast","1/3-Methylhistidine.high.yeast","1/3-Methylhistidine.low.yeast","13-HODE.high.yeast","13-HODE.low.yeast","2-Hydroxyglutarate.high.yeast","2-Hydroxyglutarate.low.yeast","2-Phosphoglyceric.acid.high.yeast","2-Phosphoglyceric.acid.low.yeast","2.3-butanedione.olfactory.attraction","2.3-butanedione.olfactory.latency.30.percentile","2.3-butanedione.olfactory.latency.90.percentile","2.3-butanedione.olfactory.response.intensity","2.3-butanedione.olfactory.strength.attraction","3-Hydroxybutyric.acid.high.yeast","3-Hydroxybutyric.acid.low.yeast","3-Hydroxykynurenine.high.yeast","3-Hydroxykynurenine.low.yeast","4-Pyridoxic.acid.high.yeast","4-Pyridoxic.acid.low.yeast","5-Aminovalerate.high.yeast","5-Aminovalerate.low.yeast","abdomen.length.18C","abdomen.length.28C","abdomen.length.thermal.plasticity","acclimated.acute.cold.exposure.resistance.18C","acclimated.acute.cold.exposure.resistance.25C","acetoacetate.high.yeast","acetoacetate.low.yeast","Acetobacter.aceti","Acetobacter.cibinongensis","Acetobacter.ghanensis","Acetobacter.indonesiensis","Acetobacter.malorum","Acetobacter.nitrogenifigens","Acetobacter.okinawensis","Acetobacter.orientalis","Acetobacter.papayae","Acetobacter.pasteurianus","Acetobacter.persici","Acetobacter.pomorum","Acetobacter.senegalensis","Acetobacter.syzygii","Acetobacter.tropicalis","acetylcarnitine.high.yeast","acetylcarnitine.low.yeast","acetylcholine.high.yeast","acetylcholine.low.yeast","Acidiphilium.angustum","Acidomonas.methanolica","Acidovorax.sp.","Acidovorax.temperans","Acinetobacter.gyllenbergii","Acinetobacter.johnsonii","Acinetobacter.sp.","activity","acute.cold.exposure.resistance.18C.-6","acute.cold.exposure.resistance.18C.-8","acute.cold.exposure.resistance.25C","adenine.high.yeast","adenine.low.yeast","adenosine.high.yeast","adenosine.low.yeast","ADP.high.yeast","ADP.low.yeast","aggression.2015","agmatine.high.yeast","agmatine.low.yeast","Agrobacterium.tumefaciens","alanine.high.yeast","alanine.low.yeast","allantoin.high.yeast","allantoin.low.yeast","alpha-amanitin.resistance.0.2","alpha-amanitin.resistance.2","alpha-Ketoglutaric.acid.high.yeast","alpha-Ketoglutaric.acid.low.yeast","aminoisobutyrate.high.yeast","aminoisobutyrate.low.yeast","AMP.high.yeast","AMP.low.yeast","anoxia.tolerance.adult","anoxia.tolerance.larval","anthranilate.high.yeast","anthranilate.low.yeast","approach.female","arachidonate.high.yeast","arachidonate.low.yeast","arginine.high.yeast","arginine.low.yeast","Asaia.prunellae","Asaia.sp.","asparagine.high.yeast","asparagine.low.yeast","aspartic.acid.high.yeast","aspartic.acid.low.yeast","Aspergillus.nidulans","Aspergillus.terreus,Malassezia globosa","attempted.copulation.frequency","azinphos.methyl.resistance.0.25","azinphos.methyl.resistance.0.5","azinphos.methyl.resistance.1","azinphos.methyl.resistance.2","azinphos.methyl.resistance.LD50","Azospirillum.brasilense","Azospirillum.thiophilum","basal.activity","betaine.high.yeast","betaine.low.yeast","biotin.high.yeast","biotin.low.yeast","boric.acid.resistance","Borrelia.coriaceae","cadaverine.high.yeast","cadaverine.low.yeast","caffeine.resistance","Campylobacter.mucosalis","Cardinium.endosymbiont","carnitine.high.yeast","carnitine.low.yeast","Caulobacter.henricii","cGMP.high.yeast","cGMP.low.yeast","Chaetomium.globosum","CHC.11 &amp; 9-Me-C23","CHC.13- &amp; 11-Me-C25","CHC.2-Me-C22","CHC.2-Me-C24","CHC.2-Me-C25","CHC.2-Me-C26","CHC.2-Me-C27","CHC.2-Me-C28","CHC.2-Me-C28 &amp; 7,11-C29:2","CHC.2-Me-C29","CHC.2-Me-C29 &amp; 8,12-C30:2","CHC.2-Me-C30","CHC.3-Me-C23","CHC.3-Me-C25","CHC.3-Me-C25 &amp; 9-C26:1","CHC.3-Me-C27 &amp; 6,10-C28:2 &amp; 9-C28:1","CHC.5-C23:1","CHC.5-C24:1","CHC.5-C25:1","CHC.5-C27:1","CHC.5-Me-C25","CHC.5,9-C25:2 &amp; 9-C25:1","CHC.5,9-C27:2 &amp; 9-C27:1","CHC.6-C23:1","CHC.6-C24:1","CHC.6-C26:1 &amp; i","CHC.6,10-C25:2","CHC.6,10-C26:2","CHC.7-C23:1","CHC.7-C24:1","CHC.7-C25:1","CHC.7-C26:1","CHC.7-C27:1","CHC.7-C29:1","CHC.7,11-C23:2","CHC.7,11-C24:2","CHC.7,11-C25:2","CHC.7,11-C26:2","CHC.7,11-C27:2 &amp; 2-Me-C26","CHC.7,11-C28:2 &amp; 2-Me-C27","CHC.7,11-C31:2","CHC.8-C25:1","CHC.8,12-C26:2","CHC.8,12-C28:2","CHC.9-C23:1","CHC.9-C25:1","CHC.9-C27:1","CHC.9-C29:1 &amp; 5,9-C29:2","CHC.9,13-C25:2","CHC.9,13-C27:2","CHC.n-C21","CHC.n-C22","CHC.n-C23","CHC.n-C24","CHC.n-C25","CHC.n-C26","CHC.n-C27","CHC.n-C28","CHC.n-C29","CHC.n-C31","CHC.NI.20","CHC.NI.33","CHC.NI.34","CHC.NI.54","CHC.NI.7","CHC.x-C22:1a","CHC.x-C22:1b","CHC.x-C23:1","CHC.x,y-C24:2","chill.coma.recovery","chill.coma.recovery.variance","chlorantraniliprole.resistance.0.5","chlorantraniliprole.resistance.1","chlorantraniliprole.resistance.2","chlorantraniliprole.resistance.3","chlorantraniliprole.resistance.4","chlorantraniliprole.resistance.5","chlorantraniliprole.resistance.LC50","choline.high.yeast","choline.low.yeast","chronic.cold.exposure.resistance","circadian.period.chi","circadian.period.mesa","circadian.rythmicity.index","citrulline.high.yeast","citrulline.low.yeast","climbing.ability","climbing.ability.paraquat","climbing.age.decline.high.yeast","climbing.age.decline.low.yeast","Clostridium.lentocellum","CMP.high.yeast","CMP.low.yeast","cold.hardiness.-5C.larvae","cold.shock.acclimated.old","cold.shock.acclimated.young","cold.shock.old","cold.shock.young","Corynebacterium.minutissimum","Corynebacterium.pseudogenitalium","Corynebacterium.sp.","Corynebacterium.terpenotabidum","CT.max.2020","CT.min.2020","CT.min.23C","CT.min.developmental.stability","CT.min.plasticity","CT.min.plasticity.variation","CT.min.within.environment.variation.plasticity","cystathionine.high.yeast","cystathionine.low.yeast","cystine.high.yeast","cystine.low.yeast","cytidine.high.yeast","cytidine.low.yeast","cytosine.high.yeast","cytosine.low.yeast","D-GA3P.high.yeast","D-GA3P.low.yeast","D-Leucic.acid.high.yeast","D-Leucic.acid.low.yeast","day.sleep","day.sleep.bout.length","day.sleep.bout.length.SD","day.sleep.bouts","day.sleep.bouts.SD","day.sleep.SD","DCV.resistance","DDT.resistance.knockdown","DDT.resistance.mortality","deltamethrin.resistance","dessication.tolerance","developmental.acclimation.cold.exposure","dopamine.response.to.paraquat.2021","Drosophila.A Virus","E.faecalis.resistance","egg.adult.development.time.2016","egg.adult.viability.2016","Enterococcus.hirae","epinephrine.high.yeast","epinephrine.low.yeast","Escherichia.coli","ethanol.sensitivity.first.exposure","ethanol.sensitivity.second.exposure","ethanol.tolerance","exercise.induced.activity","Exophiala.xenobiotica","fecundity","fertilised.egg.retention.score","fertility.25C","fertility.27C","fertility.29C","fertility.thermal.plasticity","FHV.resistance","Fimbriimonas.ginsengisoli","fitness.late","flight.performance","food.intake","foraging.survival","G1P.G6P.F6P.F1P.high.yeast","G1P.G6P.F6P.F1P.low.yeast","Gardnerella.vaginalis","Gemmata.obscuriglobus","Gemmata.sp.","genital.licking.frequency","genome.size","geranyl.pyrophosphate.high.yeast","geranyl.pyrophosphate.low.yeast","Gluconacetobacter.diazotrophicus","Gluconobacter.frateurii","Gluconobacter.morbifer","Gluconobacter.oxydans","Gluconobacter.thailandicus","glucoronate.high.yeast","glucoronate.low.yeast","glucose.2015","glucose.anexic","glucose.gnotobiotic","glucose.high.glucose.diet","glucose.high.yeast","glucose.low.glucose.diet","glucose.low.yeast","glucose.medium.glucose.diet","glucose.microbiome.effect","glucose.yeast.poor.diet","glucose.yeast.rich.diet","glutamic.acid.high.yeast","glutamic.acid.low.yeast","glutamine.high.yeast","glutamine.low.yeast","glyceraldehyde.high.yeast","glyceraldehyde.low.yeast","glycerate.high.yeast","glycerate.low.yeast","glycerol-3-P.high.yeast","glycerol-3-P.low.yeast","glycerol.2015","glycerol.high.glucose.diet","glycerol.low.glucose.diet","glycerol.medium.glucose.diet","glycine.high.yeast","glycine.low.yeast","glycogen.2015","glycogen.high.glucose.diet","glycogen.low.glucose.diet","glycogen.medium.glucose.diet","GMP.high.yeast","GMP.low.yeast","guanosine.high.yeast","guanosine.low.yeast","gustatory.response.ethanol","histamine.high.yeast","histamine.low.yeast","histidine.high.yeast","histidine.low.yeast","homogentisate.high.yeast","homogentisate.low.yeast","hydrogen.peroxide.resistance","Hymenobacter.sp.","Hymenobacter.swuensis","hypoxanthine.high.yeast","hypoxanthine.low.yeast","Ideonella.sp.","imidacloprid.resistance.100p","imidacloprid.resistance.25p","IMP.high.yeast","IMP.low.yeast","inosine.high.yeast","inosine.low.yeast","inositol.high.yeast.fInositol","inositol.low.yeast","IPI","iso-leucine.high.yeast","iso-leucine.low.yeast","isolated.aggression","isoValeric.acid.high.yeast","isoValeric.acid.low.yeast","kallithea.resistance.LT50","kallithea.viral.load","Klebsiella.pneumoniae","Kocuria.polaris","Kocuria.sp.","Komagataeibacter.europaeus","Komagataeibacter.hansenii","Komagataeibacter.intermedius","Komagataeibacter.kakiaceti","Komagataeibacter.medellinensis","Komagataeibacter.oboediens","Komagataeibacter.rhaeticus","Komagataeibacter.xylinus","Kozakia.baliensis","kujin.virus.resistance","L-Kynurenine.high.yeast","L-Kynurenine.low.yeast","L.monocytogenes.load","L.monocytogenes.MTD","lactate.high.yeast","lactate.low.yeast","Lactobacillus.brevis","Lactobacillus.fabifermentans","Lactobacillus.fermentum","Lactobacillus.kisonensis","Lactobacillus.plantarum","Lactobacillus.rhamnosus","Lactococcus.lactis","lactose.high.yeast","lactose.low.yeast","larvae.learning","lead.effect.development.time","lead.effect.locomotor.activity","lead.effect.viability","lead.egg.adult.development.time.2016","lead.egg.adult.viability.2016","lead.locomotor.activity.2016","leucine.high.yeast","leucine.low.yeast","Leuconostoc.carnosum","Leuconostoc.pseudomesenteroides","lifespan.18C","lifespan.25C.2014","lifespan.25C.2015","lifespan.25C.2020","lifespan.25C.high.yeast","lifespan.25C.low.yeast","lifespan.28C","lifespan.high.sugar.diet","lifetime.fecundity.25C","linoleic.acid.high.yeast","linoleic.acid.low.yeast","linolenic.acid.high.yeast","linolenic.acid.low.yeast","locomotor.activity","locomotor.activity.2016","locomotor.frequency","Luteipulveratus.mongoliensis","lysine.high.yeast","lysine.low.yeast","M.anisopliae.resistance","malathion.mortality.12hr","malathion.mortality.24hr","malathion.mortality.3hr","malathion.mortality.6hr","malondialdehyde.high.yeast","malondialdehyde.low.yeast","margaric.acid.high.yeast","margaric.acid.low.yeast","mass.yeast.poor.diet","mass.yeast.rich.diet","mating.latency.2015","MeHg.tolerance","Mesorhizobium.sp.","Metarhizium.brunneum","methionine.high.yeast","methionine.low.yeast","methylmalonate.high.yeast","methylmalonate.low.yeast","Methylobacterium.extorquens","Methylobacterium.sp.","micro.environmental.plasticity.lifespan.18C","micro.environmental.plasticity.lifespan.25C","micro.environmental.plasticity.lifespan.28C","Microbacterium.sp.","MPH.effect.on.activity","msb.resistance","mtDNA.copy.number","myristic.acid.high.yeast","myristic.acid.low.yeast","N-AcetylGlycine.high.yeast","N-AcetylGlycine.low.yeast","night.sleep","night.sleep.bout.length","night.sleep.bout.length.SD","night.sleep.bouts","night.sleep.bouts.SD","night.sleep.SD","Nora.virus","olfactory.response.1-hexanol","olfactory.response.2-heptanone","olfactory.response.2-phenyl ethyl alcohol","olfactory.response.acetophenone","olfactory.response.benzaldehyde.2013","olfactory.response.benzaldehyde.2015","olfactory.response.citral","olfactory.response.D-carvone","olfactory.response.ethanol","olfactory.response.ethylacetate","olfactory.response.ethylbutyrate","olfactory.response.eugenol","olfactory.response.heliol","olfactory.response.hexal","olfactory.response.L-carvone","olfactory.response.methylSalicylate","orient.towards.female","ornithine.high.yeast","ornithine.low.yeast","orotate.high.yeast","orotate.low.yeast","ovariole.asymmetry","ovariole.number","oxaloacetate.high.yeast","oxaloacetate.low.yeast","oxypurinol.high.yeast","oxypurinol.low.yeast","P.aeruginosa.defecation","P.aeruginosa.fecal.pH","P.aeruginosa.intestinal.mitosis","P.aeruginosa.resistance.2017","P.aeruginosa.resistance.2019","P.entomophila.resistance","P.rettgeri.load","P.rettgeri.load.high.glucose.diet","P.rettgeri.load.low.glucose.diet","P.rettgeri.load.medium.glucose.diet","P.rettgeri.resistance","P.rettgeri.tolerance","Paenibacillus.borealis","Paenibacillus.sp.","pantothenate.high.yeast","pantothenate.low.yeast","paraquat.resistance","paraquat.resistance.2021","parathion.resistance","Pectobacterium.carotovorum","PEP.high.yeast","PEP.low.yeast","permethrin.knockdown","permethrin.mortality.24hr","phenylalanine.high.yeast","phenylalanine.low.yeast","phototaxis.mid.age","phototaxis.old","phototaxis.young","pipecolate.high.yeast","pipecolate.low.yeast","Plasmodium.vivax","Podospora.anserina","posterior.lobe.shape.PC1","posterior.lobe.shape.PC2","posterior.lobe.size","proline.high.yeast","proline.low.yeast","Propionibacterium.acidipropionici","Propionibacterium.acnes","protein.2015","protein.anexic","protein.gnotobiotic","protein.high.glucose.diet","protein.low.glucose.diet","protein.medium.glucose.diet","Pseudodonghicola.xiamenensis","Pseudomonas.fuscovaginae","Pseudomonas.moraviensis","Pseudomonas.putida","Pseudomonas.sp.","Pseudomonas.stutzeri","Pseudomonas.syringae","Psychrobacter.glacincola","Psychrobacter.sp.","pupal.case.length","pupation.height","putrescine.high.yeast","putrescine.low.yeast","pyridoxal-5-P.high.yeast","pyridoxal-5-P.low.yeast","pyruvate.high.yeast","pyruvate.low.yeast","radiation.resistance","RCH.18C","RCH.25C","RCH.old","RCH.young","Rhizobacter.sp.","ribose-5-P.high.yeast","ribose-5-P.low.yeast","Saccharomyces.cerevisiae","sarcosine.high.yeast","sarcosine.low.yeast","serine.high.yeast","serine.low.yeast","sigma.virus.sensitivity","social.aggression","social.effect.aggression","sorbitol.high.yeast","sorbitol.low.yeast","Sphingomonas.hengshuiensis","Sphingomonas.parapaucimobilis","Sphingomonas.phyllosphaerae","Sphingomonas.sp.","Sphingomonas.wittichii","Sporolactobacillus.terrae","sprinter.marathoner","Staphylococcus.aureus","Staphylococcus.epidermidis","Staphylococcus.gallinarum","startle.response","startle.response.variance","startle.sensitivity.msb","starvation.resistance.2012","starvation.resistance.2017","starvation.resistance.2019","starvation.resistance.old","starvation.resistance.variance","starvation.resistance.yeast.poor.diet","starvation.resistance.yeast.rich.diet","starvation.resistance.young","Streptococcus.pneumoniae","Streptococcus.salivarius","Streptomyces.griseoflavus","Streptomyces.pristinaespiralis","Streptomyces.sp.","succinate.high.yeast","succinate.low.yeast","sucrose.high.yeast","sucrose.low.yeast","Sum.of Non-Wolbachia Species","T1.femur","T1.leg.length","T1.tarsus","T1.tibia","T2.femur","T2.leg.length","T2.tarsus","T2.tibia","taurine.high.yeast","taurine.low.yeast","tergite5","tergite6","Terriglobus.roseus","thorax.length.18C","thorax.length.28C","thorax.length.thermal.plasticity","threonine.high.yeast","threonine.low.yeast","traumatic.injury.mortality","Tremella.mesenterica","triglyceride.2015","triglyceride.gnotobiotic","triglyceride.high.glucose.diet","triglyceride.low.glucose.diet","triglyceride.medium.glucose.diet","triglyceride.poor.diet","triglyceride.yeast.rich.diet","trimethylamine.high.yeast","trimethylamine.low.yeast","tryptophan.high.yeast","tryptophan.low.yeast","tunicamycin.ER.stress","tunicamycin.LT50","tyrosine.high.yeast","tyrosine.low.yeast","unfertilised.eggs.post.mating","uracil.high.yeast","uracil.low.yeast","urate.high.yeast","urate.low.yeast","uridine.high.yeast","uridine.low.yeast","valine.high.yeast","valine.low.yeast","virgin.egg.retention","waking.activity","waking.activity.SD","week1.fecundity.25C","week3.fecundity.25C","week5.fecundity.25C","week7.fecundity.25C","weight.anexic","weight.gnotobiotic","weight.high.glucose.diet","weight.low.glucose.diet","weight.medium.glucose.diet","weight.microbiome.effect","wet.weight","wing.centroid.size.2019_Houle","wing.display.frequency","wolbachia.copy.number","Wolbachia.pipentis","Xanthomonas.sp.","Xanthomonas.vesicatoria","xanthosine.high.yeast","xanthosine.low.yeast","xanthurenate.high.yeast","xanthurenate.low.yeast","Yersinia.mollaretii","Yersinia.pekkanenii","Zymomonas.mobilis"],[0.022,0.011,0.015,-0.069,0.172,0.066,-0.017,0.068,0.116,0.176,0.056,0.049,-0.017,0.16,0.115,0.222,0.084,0.034,0.104,-0.222,-0.156,-0.067,-0.129,0.127,-0.035,0.048,0.129,-0.042,0.014,0.063,-0.109,-0.022,-0.006,-0.021,-0.007,-0.051,-0.054,-0.009,0.007,-0.013,-0.1,0.022,-0.046,0.022,-0.035,0,-0.046,0.11,0.218,0.097,0.068,0.005,-0.053,0.024,-0.015,0.044,0.084,-0.046,null,0.069,-0.036,0.034,0.097,-0.109,0.016,-0.023,0.029,-0.126,null,0.07,-0.007,0.032,-0.061,-0.127,-0.045,0.023,-0.26,0.007,0.102,0.052,0.011,-0.101,-0.041,-0.05,-0.076,-0.05,-0.085,-0.073,null,0.192,-0.082,-0.015,-0.061,0.001,0.001,0.218,0.036,0.122,0.146,0.077,0.158,null,-0.022,-0.045,-0.055,0.051,0.017,-0.007,0.14,0.138,-0.031,0.041,0.123,-0.11,0.132,0.032,-0.089,-0.069,-0.001,0.004,0.098,-0.041,-0.092,0.081,-0.032,-0.257,0.009,-0.015,0.065,-0.112,-0.121,-0.007,null,null,null,-0.035,null,-0.029,0.046,null,null,0.13,0.055,0.084,0.043,0.133,0.119,0.06,0.038,0.132,-0.172,0.005,null,-0.046,0.01,0.044,null,0.087,-0.09,0.181,0.01,-0.084,-0.117,-0.003,-0.032,-0.099,-0.023,-0.053,-0.225,-0.044,-0.113,0.033,null,null,0.072,-0.178,-0.136,-0.165,-0.117,-0.15,-0.103,null,0.056,0.037,0.046,0.002,-0.046,0.098,0.188,0.118,0.121,null,null,null,-0.141,-0.023,-0.103,-0.08,-0.093,-0.103,-0.062,-0.142,-0.197,-0.131,-0.067,-0.131,-0.077,0.069,0.099,0.275,0.145,-0.053,-0.188,0.069,0.074,-0.006,0.014,-0.005,0.049,-0.053,0.04,-0.085,-0.134,-0.145,-0.263,-0.01,-0.026,0.099,0.005,-0.135,-0.039,null,null,null,null,null,-0.124,0.057,-0.114,0.119,0.173,0.136,0.19,0.081,0.048,0.206,-0.067,0.046,-0.149,-0.29,-0.195,0.084,-0.009,-0.11,null,-0.035,-0.104,null,0.175,0.039,null,-0.054,null,-0.001,0.026,0.011,-0.318,-0.158,0.023,-0.064,-0.074,-0.031,0.133,0.07,0.198,0.01,null,null,null,null,null,-0.021,0.754,0.09,0.135,null,0.152,0.118,0.098,0.049,-0.063,null,-0.111,-0.145,-0.096,-0.022,-0.031,-0.049,-0.054,-0.037,-0.119,0.147,null,null,null,null,-0.138,null,-0.047,null,null,-0.01,-0.013,0.152,0.27,0.103,0.103,-0.043,0.066,0.221,0.064,0.1,0.042,null,null,null,null,-0.009,0.016,null,null,null,null,-0.005,-0.019,0.077,-0.016,0.055,-0.06,0.048,-0.103,-0.054,-0.129,-0.015,0.141,0.055,-0.074,-0.104,0.027,-0.048,null,null,0.033,-0.06,-0.015,-0.013,0.084,0.195,null,0.164,-0.046,null,-0.127,-0.01,-0.015,0.048,0.121,0.07,-0.052,-0.047,-0.017,-0.033,0.023,0.013,-0.016,-0.006,-0.009,-0.001,-0.159,0.149,0.106,null,null,-0.18,-0.134,0.045,-0.013,0.052,0.019,0.01,0.014,-0.038,-0.198,-0.015,0.084,0.017,-0.125,null,0.006,0.002,0.096,0.171,-0.08,-0.108,0.044,-0.109,0.046,-0.138,-0.053,0.033,-0.078,-0.019,0.045,0.282,-0.181,-0.035,-0.058,-0.035,null,0.058,0.233,0.082,0.005,-0.007,0.238,-0.076,-0.03,-0.108,-0.064,-0.102,-0.045,-0.066,0.039,-0.078,-0.077,null,-0.003,0.085,0.008,0.156,-0.006,-0.005,0.215,0.161,0.051,0.013,-0.016,-0.019,0.062,null,-0.023,0.059,-0.038,-0.149,-0.085,0.046,-0.031,-0.072,0.036,0.04,0.162,0.03,0.067,0.027,-0.001,-0.073,-0.032,-0.173,0.029,-0.083,0.04,-0.045,-0.055,0.007,0.045,0.069,-0.033,-0.021,-0.05,null,0.016,0.225,-0.036,-0.155,-0.048,0.032,0.201,0.023,-0.151,0.002,0.025,-0.078,-0.018,0.223,0.064,-0.015,-0.006,null,null,null,0.076,0.107,0.091,0.031,-0.029,0.023,-0.016,0.022,null,-0.099,-0.007,0.078,-0.066,-0.083,0.179,-0.037,0.136,0.119,-0.017,0.075,0.018,0.104,0.075,null,null,null,0.111,-0.238,0.039,0.063,null,null,null,null,null,null,-0.111,-0.057,0.015,0.1,0.107,0.119,0.068,-0.004,-0.17,-0.199,0.118,-0.166,0.037,-0.047,-0.099,-0.121,-0.166,null,0.04,0.027,-0.014,0.098,-0.031,0.015,0.026,-0.032,0.069,-0.149,0.246,0.128,0.022,null,null,0.161,0.084,0.015,-0.056,-0.015,-0.092,-0.239,0.122,-0.227,-0.094,-0.107,-0.105,0.024,-0.044,-0.044,-0.029,null,-0.093,-0.022,-0.181,0.019,-0.067,-0.015,0.099,0.022,0.045,0.002,0.13,0.014,-0.031,-0.07,-0.141,-0.002,0.147,0.107,0.106,0.03,0.015,0.015,0.026,-0.01,-0.162,-0.069,0.142,0.076,-0.138,-0.145,0.031,-0.218,0.004,-0.089,-0.097,-0.072,null,null,null,null,null,-0.046,-0.002,0.03,-0.096,0.057,-0,null,null,0.162,0.149,-0.171,0.094,-0.075,-0.142,-0.014,0.114,-0.023,0.123,-0.042,-0.116,-0.248,-0.247,0.27,0.226,0.133,-0.022,null,null,null,null,null,null,null,0.059,null,0.072,-0.037,-0.136,0.067,-0.001,0.057,-0.098,-0.03,0.064,-0.123,0.003],[-0.162,-0.164,-0.19,-0.24,-0.007,-0.118,-0.19,-0.112,-0.07,-0.005,-0.125,-0.135,-0.221,-0.029,-0.073,0.029,-0.094,-0.146,-0.076,-0.385,-0.329,-0.234,-0.301,-0.067,-0.211,-0.145,-0.05,-0.221,-0.182,-0.126,-0.284,-0.215,-0.169,-0.173,-0.161,-0.219,-0.226,-0.17,-0.14,-0.172,-0.257,-0.136,-0.223,-0.134,-0.199,-0.151,-0.214,-0.072,0.049,-0.085,-0.111,-0.147,-0.209,-0.14,-0.166,-0.115,-0.083,-0.223,null,-0.098,-0.234,-0.149,-0.094,-0.282,-0.186,-0.21,-0.157,-0.306,null,-0.107,-0.195,-0.125,-0.241,-0.298,-0.249,-0.173,-0.415,-0.166,-0.08,-0.133,-0.163,-0.268,-0.223,-0.234,-0.239,-0.237,-0.258,-0.259,null,0.011,-0.253,-0.188,-0.232,-0.153,-0.151,0.035,-0.156,-0.066,-0.047,-0.094,-0.025,null,-0.19,-0.216,-0.223,-0.113,-0.141,-0.161,-0.019,-0.045,-0.211,-0.138,-0.065,-0.282,-0.038,-0.135,-0.266,-0.242,-0.183,-0.16,-0.075,-0.224,-0.264,-0.083,-0.215,-0.408,-0.158,-0.204,-0.126,-0.293,-0.316,-0.201,null,null,null,-0.221,null,-0.214,-0.127,null,null,-0.062,-0.14,-0.077,-0.135,-0.029,-0.036,-0.129,-0.145,-0.058,-0.345,-0.202,null,-0.247,-0.158,-0.128,null,-0.083,-0.276,0.012,-0.191,-0.299,-0.322,-0.257,-0.209,-0.273,-0.21,-0.248,-0.383,-0.239,-0.299,-0.194,null,null,-0.113,-0.357,-0.311,-0.336,-0.307,-0.326,-0.276,null,-0.119,-0.158,-0.126,-0.185,-0.226,-0.103,0.001,-0.047,-0.075,null,null,null,-0.317,-0.228,-0.287,-0.268,-0.3,-0.307,-0.258,-0.338,-0.381,-0.307,-0.261,-0.296,-0.244,-0.125,-0.213,0.053,-0.05,-0.242,-0.38,-0.175,-0.169,-0.22,-0.204,-0.162,-0.133,-0.228,-0.157,-0.32,-0.368,-0.388,-0.476,-0.174,-0.185,-0.08,-0.185,-0.359,-0.278,null,null,null,null,null,-0.298,-0.125,-0.288,-0.07,-0.005,-0.064,0.017,-0.121,-0.149,0.025,-0.249,-0.141,-0.344,-0.454,-0.364,-0.118,-0.198,-0.292,null,-0.228,-0.281,null,-0.014,-0.159,null,-0.223,null,-0.172,-0.144,-0.15,-0.469,-0.321,-0.153,-0.245,-0.245,-0.204,-0.054,-0.102,-0.005,-0.221,null,null,null,null,null,-0.186,0.687,-0.075,-0.062,null,-0.033,-0.063,-0.082,-0.123,-0.224,null,-0.288,-0.304,-0.276,-0.18,-0.188,-0.209,-0.209,-0.188,-0.306,-0.037,null,null,null,null,-0.307,null,-0.225,null,null,-0.186,-0.193,-0.035,0.103,-0.071,-0.071,-0.22,-0.099,0.037,-0.132,-0.091,-0.147,null,null,null,null,-0.205,-0.166,null,null,null,null,-0.2,-0.209,-0.109,-0.193,-0.15,-0.225,-0.12,-0.274,-0.231,-0.311,-0.197,-0.037,-0.142,-0.219,-0.279,-0.147,-0.215,null,null,-0.158,-0.242,-0.199,-0.195,-0.112,0.025,null,-0.033,-0.225,null,-0.299,-0.202,-0.219,-0.155,-0.047,-0.099,-0.22,-0.216,-0.176,-0.196,-0.126,-0.133,-0.172,-0.17,-0.162,-0.152,-0.416,-0.036,-0.075,null,null,-0.346,-0.295,-0.119,-0.176,-0.115,-0.141,-0.158,-0.151,-0.199,-0.358,-0.186,-0.141,-0.147,-0.28,null,-0.155,-0.167,-0.109,-0.022,-0.255,-0.27,-0.11,-0.278,-0.143,-0.304,-0.229,-0.158,-0.269,-0.199,-0.125,0.102,-0.354,-0.225,-0.215,-0.201,null,-0.141,-0.106,-0.096,-0.178,-0.185,0.072,-0.247,-0.204,-0.271,-0.233,-0.269,-0.226,-0.249,-0.15,-0.248,-0.248,null,-0.177,-0.08,-0.162,-0.033,-0.173,-0.175,0.023,-0.021,-0.128,-0.173,-0.195,-0.188,-0.109,null,-0.216,-0.119,-0.219,-0.316,-0.259,-0.143,-0.227,-0.264,-0.163,-0.173,-0.016,-0.164,-0.1,-0.174,-0.183,-0.248,-0.209,-0.356,-0.154,-0.253,-0.138,-0.23,-0.236,-0.167,-0.139,-0.118,-0.207,-0.198,-0.23,null,-0.156,0.051,-0.219,-0.328,-0.224,-0.137,0.02,-0.166,-0.309,-0.177,-0.157,-0.274,-0.242,-0.039,-0.136,-0.212,-0.2,null,null,null,-0.136,-0.106,-0.072,-0.136,-0.215,-0.158,-0.214,-0.226,null,-0.254,-0.196,-0.101,-0.235,-0.258,-0.014,-0.214,-0.045,-0.064,-0.196,-0.121,-0.174,-0.071,-0.091,null,null,null,-0.064,-0.388,-0.115,-0.11,null,null,null,null,null,null,-0.268,-0.223,-0.151,-0.059,-0.062,-0.046,-0.097,-0.17,-0.316,-0.358,-0.035,-0.332,-0.172,-0.221,-0.267,-0.291,-0.328,null,-0.155,-0.15,-0.259,-0.155,-0.21,-0.18,-0.169,-0.204,-0.132,-0.33,0.063,-0.063,-0.162,null,null,-0.021,-0.101,-0.141,-0.211,-0.183,-0.257,-0.379,-0.043,-0.398,-0.246,-0.267,-0.258,-0.144,-0.199,-0.218,-0.19,null,-0.277,-0.219,-0.336,-0.163,-0.254,-0.192,-0.07,-0.145,-0.127,-0.166,-0.056,-0.174,-0.229,-0.245,-0.323,-0.154,-0.086,-0.128,-0.147,-0.204,-0.212,-0.211,-0.202,-0.229,-0.335,-0.247,-0.044,-0.105,-0.29,-0.306,-0.144,-0.394,-0.173,-0.264,-0.295,-0.229,null,null,null,null,null,-0.24,-0.177,-0.161,-0.261,-0.12,-0.187,null,null,-0.015,-0.028,-0.436,-0.088,-0.253,-0.323,-0.203,-0.075,-0.207,-0.067,-0.226,-0.364,-0.422,-0.462,0.093,0.046,-0.051,-0.241,null,null,null,null,null,null,null,-0.119,null,-0.159,-0.192,-0.281,-0.117,-0.183,-0.13,-0.269,-0.206,-0.101,-0.285,-0.164],[0.2,0.185,0.214,0.109,0.338,0.249,0.152,0.247,0.296,0.339,0.236,0.234,0.186,0.338,0.295,0.394,0.256,0.22,0.275,-0.04,0.032,0.111,0.05,0.312,0.142,0.237,0.298,0.137,0.201,0.248,0.076,0.176,0.156,0.136,0.148,0.12,0.131,0.153,0.153,0.142,0.061,0.175,0.133,0.182,0.126,0.15,0.122,0.285,0.372,0.274,0.24,0.157,0.106,0.185,0.137,0.197,0.246,0.132,null,0.236,0.169,0.217,0.276,0.063,0.209,0.165,0.204,0.073,null,0.248,0.183,0.195,0.12,0.057,0.169,0.214,-0.079,0.187,0.275,0.231,0.181,0.073,0.139,0.141,0.093,0.139,0.096,0.123,null,0.356,0.095,0.171,0.114,0.157,0.149,0.382,0.23,0.302,0.322,0.242,0.323,null,0.146,0.127,0.115,0.212,0.173,0.147,0.293,0.31,0.151,0.217,0.299,0.074,0.297,0.193,0.092,0.114,0.187,0.172,0.265,0.151,0.087,0.242,0.15,-0.091,0.177,0.165,0.251,0.079,0.09,0.187,null,null,null,0.159,null,0.162,0.221,null,null,0.311,0.244,0.242,0.224,0.289,0.275,0.243,0.216,0.309,0.011,0.213,null,0.16,0.19,0.212,null,0.243,0.111,0.338,0.213,0.134,0.107,0.248,0.147,0.08,0.171,0.152,-0.043,0.157,0.088,0.255,null,null,0.251,0.026,0.053,0.015,0.087,0.033,0.079,null,0.228,0.222,0.211,0.187,0.142,0.281,0.356,0.277,0.306,null,null,null,0.055,0.198,0.092,0.112,0.128,0.119,0.145,0.067,0.009,0.067,0.13,0.043,0.096,0.259,0.376,0.465,0.33,0.146,0.033,0.304,0.309,0.205,0.231,0.163,0.235,0.136,0.23,0.161,0.115,0.132,0.004,0.155,0.138,0.271,0.198,0.106,0.21,null,null,null,null,null,0.061,0.231,0.071,0.303,0.344,0.328,0.354,0.268,0.239,0.374,0.122,0.226,0.066,-0.091,-0.013,0.27,0.181,0.081,null,0.167,0.085,null,0.346,0.226,null,0.124,null,0.164,0.197,0.174,-0.132,0.021,0.191,0.124,0.103,0.141,0.303,0.242,0.379,0.249,null,null,null,null,null,0.151,0.808,0.245,0.32,null,0.318,0.293,0.272,0.216,0.105,null,0.066,0.032,0.093,0.141,0.122,0.11,0.105,0.115,0.071,0.323,null,null,null,null,0.048,null,0.138,null,null,0.172,0.163,0.322,0.419,0.266,0.266,0.141,0.229,0.387,0.244,0.283,0.228,null,null,null,null,0.18,0.208,null,null,null,null,0.191,0.172,0.259,0.163,0.242,0.109,0.214,0.075,0.128,0.062,0.171,0.309,0.243,0.085,0.079,0.2,0.121,null,null,0.216,0.13,0.167,0.174,0.265,0.353,null,0.343,0.135,null,0.058,0.176,0.194,0.249,0.283,0.228,0.123,0.123,0.139,0.132,0.171,0.159,0.137,0.158,0.147,0.151,0.137,0.319,0.276,null,null,0.001,0.041,0.204,0.148,0.215,0.18,0.18,0.18,0.133,-0.026,0.156,0.296,0.178,0.038,null,0.17,0.179,0.288,0.345,0.094,0.062,0.198,0.071,0.231,0.036,0.125,0.226,0.129,0.159,0.218,0.44,0.01,0.161,0.102,0.136,null,0.249,0.49,0.244,0.194,0.177,0.391,0.104,0.144,0.064,0.109,0.074,0.139,0.116,0.217,0.098,0.109,null,0.162,0.252,0.174,0.33,0.164,0.167,0.375,0.335,0.225,0.204,0.163,0.159,0.23,null,0.171,0.235,0.151,0.03,0.102,0.236,0.167,0.126,0.241,0.248,0.326,0.221,0.225,0.226,0.185,0.11,0.148,0.029,0.207,0.094,0.218,0.145,0.13,0.187,0.223,0.242,0.146,0.16,0.128,null,0.192,0.384,0.15,0.031,0.132,0.203,0.366,0.212,0.014,0.185,0.208,0.124,0.206,0.455,0.262,0.184,0.196,null,null,null,0.283,0.306,0.248,0.197,0.162,0.202,0.171,0.274,null,0.061,0.177,0.25,0.111,0.103,0.349,0.15,0.306,0.29,0.157,0.266,0.206,0.272,0.237,null,null,null,0.278,-0.058,0.193,0.23,null,null,null,null,null,null,0.057,0.115,0.178,0.256,0.273,0.27,0.228,0.159,-0.016,-0.018,0.269,0.014,0.242,0.133,0.086,0.058,0.004,null,0.234,0.209,0.231,0.328,0.153,0.209,0.218,0.141,0.253,0.046,0.411,0.305,0.214,null,null,0.324,0.253,0.174,0.1,0.146,0.077,-0.084,0.283,-0.032,0.068,0.065,0.056,0.191,0.118,0.144,0.133,null,0.102,0.174,-0.009,0.197,0.131,0.168,0.269,0.184,0.215,0.179,0.301,0.201,0.173,0.11,0.057,0.15,0.358,0.323,0.337,0.252,0.247,0.236,0.254,0.217,0.034,0.115,0.322,0.255,0.025,0.03,0.206,-0.016,0.182,0.093,0.116,0.08,null,null,null,null,null,0.162,0.175,0.222,0.082,0.232,0.18,null,null,0.33,0.317,0.144,0.268,0.113,0.055,0.18,0.286,0.163,0.31,0.143,0.154,-0.04,0.031,0.428,0.388,0.306,0.2,null,null,null,null,null,null,null,0.233,null,0.294,0.127,0.019,0.241,0.18,0.238,0.086,0.154,0.224,0.043,0.168],[null,null,null,null,null,null,null,null,null,null,null,null,0.07,-0.093,-0.087,0.023,-0.011,null,null,null,null,null,null,null,null,null,null,null,-0.126,0.177,null,null,-0.041,-0.021,-0.022,-0.067,-0.056,-0.044,-0.017,-0.002,-0.076,0.015,-0.018,0.012,-0.073,-0.04,-0.088,null,null,null,null,-0.039,-0.016,-0.035,0.043,-0.153,0.19,-0.007,0.161,-0.077,0.02,0.111,null,null,null,null,null,null,0.036,null,null,0.092,null,null,null,null,0.06,0.025,null,null,null,null,null,null,0.069,0.023,null,null,0.054,null,null,null,null,-0.068,-0.055,null,null,null,null,0.047,0.028,0.018,-0.043,-0.099,-0.104,0.084,-0.043,-0.024,-0.029,-0.017,null,null,null,null,null,-0.096,null,null,null,0.099,-0.122,null,null,0.026,null,null,0.096,null,null,-0.196,-0.078,-0.118,-0.091,-0.037,-0.005,null,0.126,null,null,0.003,-0.015,null,null,-0.124,0.028,-0.087,null,null,null,null,0.102,0.073,-0.034,0.093,null,-0.052,-0.018,0.076,0.045,-0.057,-0.035,null,null,null,null,null,null,null,null,null,null,0.006,0.12,-0.095,null,null,null,-0.076,-0.055,-0.013,0.077,0.076,0.103,0.136,0.05,0.075,-0.047,null,null,null,0.104,0.043,0.012,-0.097,null,-0.105,-0.124,-0.125,-0.033,0.052,0.045,0.008,0.014,0.017,-0.017,null,null,0.124,0.129,0.113,0.083,null,null,0.202,0.264,null,null,-0.039,null,null,0.02,0.16,-0.089,-0.119,-0.244,-0.061,0.16,0.11,0.067,0.152,0.186,0.022,-0.105,-0.041,-0.168,-0.191,null,null,null,null,null,null,null,null,null,null,null,null,0.18,0.198,0.208,-0.191,0.069,-0.158,-0.104,null,null,0.055,0.111,-0.124,0.142,-0.102,-0.019,0.141,0.035,0.229,null,null,-0.118,-0.089,-0.077,-0.034,0.011,0.159,null,null,0.138,0.138,0.054,-0.004,0.003,0.062,0.86,0.15,0.055,-0.038,null,null,0.013,0.059,0.097,-0.138,-0.052,null,null,-0.108,-0.018,-0.01,-0.045,-0.011,null,null,0.081,-0.049,-0.103,0.069,null,0.204,null,0.134,0.028,null,null,null,null,null,null,null,null,null,null,null,null,0.033,0.049,0.062,0.082,null,null,0.11,0.142,0.144,0.163,null,null,null,null,0.038,null,null,null,null,null,null,null,0.089,0.146,null,null,0.076,-0.183,-0.164,null,null,null,null,-0.14,null,-0.059,null,null,0.105,null,null,-0.006,null,0.171,0.097,-0.123,-0.042,0,-0.149,0.021,-0.045,-0.02,-0.05,-0.041,-0.06,null,null,null,0.092,-0.132,null,null,0.164,0.053,0.053,0.114,0.051,0.01,-0.022,null,null,-0.004,0.04,-0.091,-0.086,0.089,-0.059,0.027,null,null,-0.001,0.156,0.085,null,null,0.15,null,null,0.257,null,null,null,null,null,null,0.182,0.017,null,0.056,null,null,0.232,-0.043,-0.003,0.007,0.021,null,null,null,null,null,null,-0.269,0.059,0.036,0.024,null,null,null,null,-0.072,-0.149,0.049,0.19,0.041,0.003,-0.019,-0.013,0.033,null,null,null,null,0.136,0.161,0.165,-0.166,-0.146,-0.155,-0.05,0.122,0.143,-0.016,-0.111,0.097,0.028,0.01,-0.159,0.042,-0.212,0.088,-0.043,-0.052,-0.001,-0.054,-0.064,-0.126,null,null,null,null,null,null,null,null,null,null,null,null,null,0.217,null,null,null,-0.257,-0.092,-0.165,null,null,0.128,0.049,null,null,0.016,0.142,0.131,0.013,null,null,-0.028,-0.061,null,null,-0.035,-0.026,0.104,null,null,0.065,0.049,0.022,-0.035,-0.026,null,null,0.027,-0.016,-0.103,-0.079,-0.08,-0.071,-0.017,-0.04,0.075,0.169,0.112,0.116,0.063,0.052,-0.016,0.063,-0.077,-0.071,-0.02,null,null,null,null,null,null,0.042,-0.082,0.027,0.262,0.158,0.094,null,null,-0.005,null,null,null,null,null,0.019,-0.055,null,null,-0.095,0.038,0.104,0.067,0.014,0.06,0.186,0.108,-0.027,0.039,0.062,-0.105,-0.033,-0.05,-0.026,-0.008,-0.001,-0.039,null,null,0.041,0.012,0.037,0.033,0.039,0.223,null,null,null,null,-0.076,0.038,0.039,-0.026,0.097,-0.007,-0.028,-0.096,0.054,null,null,null,null,0.08,null,null,null,null,null,0.024,-0.003,0.015,-0.269,0.116,-0.099,0.011,null,null,null,null,null,null,0.055,-0.056,null,null,-0.262,null,null,null,null,null,null,null,null,null,0.044,0.003,null,null,null,null,-0.079,-0.056,0.074,0.153,0.138,0.057,0.101,null,-0.136,-0.087,0.078,0.067,0.046,null,null,null,null,0.02,-0.021,-0.081],[null,null,null,null,null,null,null,null,null,null,null,null,-0.111,-0.261,-0.264,-0.159,-0.195,null,null,null,null,null,null,null,null,null,null,null,-0.28,0.007,null,null,-0.205,-0.17,-0.173,-0.236,-0.244,-0.203,-0.169,-0.149,-0.235,-0.139,-0.2,-0.146,-0.239,-0.187,-0.254,null,null,null,null,-0.193,-0.159,-0.204,-0.123,-0.323,0.031,-0.177,-0.007,-0.234,-0.148,-0.061,null,null,null,null,null,null,-0.115,null,null,-0.084,null,null,null,null,-0.112,-0.144,null,null,null,null,null,null,-0.101,-0.166,null,null,-0.154,null,null,null,null,-0.219,-0.203,null,null,null,null,-0.117,-0.13,-0.165,-0.212,-0.27,-0.274,-0.081,-0.199,-0.185,-0.18,-0.201,null,null,null,null,null,-0.258,null,null,null,-0.089,-0.301,null,null,-0.147,null,null,-0.066,null,null,-0.36,-0.243,-0.309,-0.277,-0.211,-0.186,null,-0.054,null,null,-0.172,-0.205,null,null,-0.274,-0.142,-0.249,null,null,null,null,-0.082,-0.121,-0.226,-0.08,null,-0.23,-0.208,-0.109,-0.125,-0.216,-0.216,null,null,null,null,null,null,null,null,null,null,-0.162,-0.062,-0.259,null,null,null,-0.245,-0.229,-0.179,-0.103,-0.106,-0.07,-0.036,-0.118,-0.093,-0.199,null,null,null,-0.077,-0.127,-0.16,-0.265,null,-0.267,-0.297,-0.305,-0.214,-0.128,-0.129,-0.175,-0.165,-0.147,-0.189,null,null,-0.067,-0.115,-0.068,-0.105,null,null,-0.045,0.026,null,null,-0.215,null,null,-0.17,-0.052,-0.302,-0.339,-0.429,-0.236,0.007,-0.052,-0.123,-0.056,-0.034,-0.157,-0.265,-0.224,-0.351,-0.349,null,null,null,null,null,null,null,null,null,null,null,null,-0.005,0.013,0.026,-0.363,-0.115,-0.313,-0.294,null,null,-0.104,-0.055,-0.288,-0.09,-0.265,-0.184,-0.036,-0.137,0.063,null,null,-0.295,-0.247,-0.227,-0.197,-0.167,-0.011,null,null,-0.132,-0.149,-0.169,-0.216,-0.144,-0.101,0.822,-0.021,-0.124,-0.206,null,null,-0.158,-0.116,-0.066,-0.3,-0.235,null,null,-0.261,-0.164,-0.165,-0.202,-0.157,null,null,-0.112,-0.231,-0.28,-0.132,null,0.005,null,-0.058,-0.178,null,null,null,null,null,null,null,null,null,null,null,null,-0.154,-0.152,-0.124,-0.115,null,null,-0.086,-0.052,-0.048,-0.031,null,null,null,null,-0.147,null,null,null,null,null,null,null,-0.076,-0.026,null,null,-0.096,-0.346,-0.324,null,null,null,null,-0.3,null,-0.245,null,null,-0.152,null,null,-0.213,null,-0.023,-0.076,-0.285,-0.199,-0.169,-0.309,-0.13,-0.196,-0.175,-0.205,-0.195,-0.207,null,null,null,-0.13,-0.32,null,null,-0.005,-0.118,-0.11,-0.078,-0.119,-0.166,-0.199,null,null,-0.215,-0.12,-0.245,-0.238,-0.071,-0.228,-0.158,null,null,-0.166,-0.015,-0.079,null,null,-0.016,null,null,0.087,null,null,null,null,null,null,0.007,-0.161,null,-0.119,null,null,0.053,-0.214,-0.167,-0.161,-0.158,null,null,null,null,null,null,-0.415,-0.108,-0.13,-0.141,null,null,null,null,-0.239,-0.314,-0.124,0.034,-0.119,-0.156,-0.18,-0.187,-0.13,null,null,null,null,-0.039,-0.036,-0.019,-0.334,-0.308,-0.313,-0.205,-0.051,-0.023,-0.185,-0.276,-0.086,-0.148,-0.157,-0.314,-0.13,-0.368,-0.076,-0.203,-0.226,-0.164,-0.22,-0.234,-0.29,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.01,null,null,null,-0.416,-0.268,-0.333,null,null,-0.045,-0.125,null,null,-0.162,-0.09,-0.038,-0.161,null,null,-0.194,-0.227,null,null,-0.207,-0.197,-0.062,null,null,-0.096,-0.118,-0.15,-0.231,-0.194,null,null,-0.136,-0.181,-0.282,-0.296,-0.289,-0.246,-0.192,-0.215,-0.102,-0.01,-0.074,-0.067,-0.107,-0.13,-0.189,-0.106,-0.226,-0.248,-0.178,null,null,null,null,null,null,-0.136,-0.246,-0.142,0.037,-0.045,-0.096,null,null,-0.175,null,null,null,null,null,-0.228,-0.316,null,null,-0.251,-0.13,-0.047,-0.11,-0.146,-0.104,-0.005,-0.061,-0.205,-0.121,-0.114,-0.263,-0.204,-0.217,-0.187,-0.18,-0.194,-0.212,null,null,-0.145,-0.163,-0.117,-0.142,-0.139,0.042,null,null,null,null,-0.222,-0.166,-0.169,-0.229,-0.111,-0.21,-0.22,-0.295,-0.155,null,null,null,null,-0.085,null,null,null,null,null,-0.163,-0.164,-0.176,-0.471,-0.058,-0.271,-0.163,null,null,null,null,null,null,-0.141,-0.256,null,null,-0.495,null,null,null,null,null,null,null,null,null,-0.151,-0.25,null,null,null,null,-0.315,-0.272,-0.104,-0.038,-0.04,-0.184,-0.093,null,-0.318,-0.315,-0.092,-0.091,-0.115,null,null,null,null,-0.156,-0.192,-0.24],[null,null,null,null,null,null,null,null,null,null,null,null,0.245,0.085,0.098,0.199,0.169,null,null,null,null,null,null,null,null,null,null,null,0.041,0.336,null,null,0.121,0.135,0.135,0.106,0.134,0.118,0.136,0.154,0.09,0.175,0.171,0.173,0.097,0.111,0.086,null,null,null,null,0.118,0.134,0.144,0.211,0.026,0.344,0.16,0.322,0.082,0.188,0.274,null,null,null,null,null,null,0.187,null,null,0.267,null,null,null,null,0.235,0.189,null,null,null,null,null,null,0.242,0.201,null,null,0.25,null,null,null,null,0.092,0.1,null,null,null,null,0.209,0.188,0.195,0.128,0.075,0.072,0.244,0.121,0.135,0.121,0.172,null,null,null,null,null,0.071,null,null,null,0.275,0.067,null,null,0.195,null,null,0.253,null,null,-0.012,0.096,0.083,0.108,0.143,0.176,null,0.297,null,null,0.185,0.176,null,null,0.04,0.196,0.079,null,null,null,null,0.269,0.267,0.158,0.262,null,0.124,0.174,0.248,0.211,0.111,0.148,null,null,null,null,null,null,null,null,null,null,0.174,0.29,0.076,null,null,null,0.093,0.121,0.156,0.248,0.249,0.266,0.3,0.216,0.236,0.107,null,null,null,0.273,0.215,0.182,0.076,null,0.066,0.062,0.066,0.154,0.23,0.215,0.192,0.196,0.18,0.164,null,null,0.302,0.345,0.291,0.264,null,null,0.416,0.466,null,null,0.141,null,null,0.207,0.36,0.127,0.116,-0.03,0.124,0.305,0.265,0.251,0.346,0.39,0.196,0.064,0.144,0.029,-0.015,null,null,null,null,null,null,null,null,null,null,null,null,0.357,0.373,0.375,-0.001,0.242,0.005,0.098,null,null,0.211,0.271,0.045,0.358,0.07,0.14,0.305,0.202,0.38,null,null,0.071,0.08,0.078,0.132,0.192,0.322,null,null,0.386,0.392,0.263,0.213,0.147,0.223,0.891,0.311,0.228,0.132,null,null,0.189,0.226,0.254,0.04,0.131,null,null,0.054,0.133,0.147,0.114,0.137,null,null,0.264,0.142,0.081,0.261,null,0.377,null,0.317,0.238,null,null,null,null,null,null,null,null,null,null,null,null,0.212,0.239,0.242,0.264,null,null,0.297,0.328,0.32,0.337,null,null,null,null,0.218,null,null,null,null,null,null,null,0.247,0.31,null,null,0.242,-0.005,0.011,null,null,null,null,0.031,null,0.138,null,null,0.344,null,null,0.201,null,0.345,0.261,0.044,0.119,0.16,0.024,0.17,0.113,0.137,0.109,0.112,0.097,null,null,null,0.303,0.074,null,null,0.322,0.218,0.216,0.292,0.218,0.185,0.161,null,null,0.203,0.198,0.075,0.077,0.243,0.111,0.205,null,null,0.166,0.316,0.241,null,null,0.308,null,null,0.403,null,null,null,null,null,null,0.346,0.195,null,0.224,null,null,0.389,0.132,0.162,0.175,0.198,null,null,null,null,null,null,-0.102,0.215,0.199,0.188,null,null,null,null,0.096,0.027,0.219,0.341,0.195,0.159,0.142,0.161,0.198,null,null,null,null,0.301,0.337,0.339,0.023,0.023,0.006,0.109,0.289,0.302,0.154,0.066,0.272,0.212,0.176,0.013,0.214,-0.038,0.243,0.12,0.14,0.165,0.109,0.118,0.047,null,null,null,null,null,null,null,null,null,null,null,null,null,0.424,null,null,null,-0.082,0.09,0.014,null,null,0.286,0.22,null,null,0.189,0.358,0.291,0.185,null,null,0.141,0.109,null,null,0.135,0.141,0.266,null,null,0.223,0.212,0.19,0.168,0.154,null,null,0.194,0.148,0.088,0.152,0.136,0.115,0.157,0.132,0.239,0.333,0.286,0.286,0.229,0.226,0.154,0.227,0.075,0.11,0.13,null,null,null,null,null,null,0.217,0.09,0.19,0.453,0.349,0.266,null,null,0.16,null,null,null,null,null,0.277,0.223,null,null,0.067,0.2,0.247,0.242,0.172,0.22,0.363,0.27,0.152,0.195,0.228,0.06,0.147,0.123,0.138,0.161,0.192,0.14,null,null,0.228,0.188,0.19,0.2,0.217,0.383,null,null,null,null,0.079,0.231,0.241,0.186,0.292,0.199,0.174,0.109,0.254,null,null,null,null,0.238,null,null,null,null,null,0.206,0.159,0.21,-0.034,0.286,0.078,0.184,null,null,null,null,null,null,0.25,0.157,null,null,0.037,null,null,null,null,null,null,null,null,null,0.233,0.253,null,null,null,null,0.168,0.165,0.254,0.327,0.31,0.291,0.28,null,0.059,0.153,0.242,0.228,0.211,null,null,null,null,0.194,0.15,0.079],[null,null,null,null,null,null,null,null,null,null,null,null,-0.009,-0.08,-0.059,0.025,-0.007,null,null,null,null,null,null,null,null,null,null,null,-0.01,0.061,null,null,0.005,0.003,0.003,0.024,0.022,0.004,0.001,0.001,0.053,0.002,0.005,0.002,0.017,-0.001,0.03,null,null,null,null,-0.001,0.009,-0.003,-0.005,-0.042,0.087,0.003,null,-0.038,-0.006,0.026,null,null,null,null,null,null,null,null,null,0.023,null,null,null,null,-0.07,0.004,null,null,null,null,null,null,-0.036,-0.007,null,null,null,null,null,null,null,0.001,0.003,null,null,null,null,0.026,0.026,null,0.009,0.03,0.038,0.03,-0.005,0.004,-0.026,-0.014,null,null,null,null,null,-0.019,null,null,null,0.001,-0.069,null,null,0.014,null,null,0.006,null,null,0.109,0.054,0.006,null,null,null,null,null,null,null,null,null,null,null,-0.067,0.009,-0.07,null,null,null,null,-0.092,0,null,-0.024,null,-0.014,null,0.043,-0.026,-0.059,-0.001,null,null,null,null,null,null,null,null,null,null,0.004,null,null,null,null,null,0.072,0.04,0.013,-0.05,null,0.038,0.03,0.017,0.003,0.017,null,null,null,0.073,null,null,null,null,0.013,0.074,0.058,0.021,-0.028,-0.018,-0.005,-0.013,-0.013,0.007,null,null,0.049,0.058,0.125,0.069,null,null,0.06,0.077,null,null,0.005,null,null,0.005,-0.064,0.057,0.077,0.217,0.003,-0.025,0.071,0.004,-0.095,-0.033,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.124,-0.208,-0.173,-0.08,-0.004,0.095,null,null,null,null,0.102,-0.028,null,0.037,null,-0,0.009,0.013,null,null,-0.015,0.04,0.04,0.01,0.009,0.064,null,null,null,null,null,null,null,-0.009,0.802,0.08,0.046,null,null,null,0.008,0.02,-0.039,null,0.035,null,null,0.02,0.006,0.005,0.018,0.003,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.014,null,null,null,null,null,null,null,0.031,-0.065,null,null,-0.027,null,null,null,null,null,null,-0.067,null,null,null,null,null,null,null,0.002,null,0.108,0.046,0.041,0.015,0.001,0.03,0.006,-0.005,0,0.003,0.003,0,null,null,null,null,null,null,null,0.045,-0.003,0.022,0.014,0.01,0.002,0.011,null,null,-0,0.004,0.071,null,0.004,-0.002,0.016,null,null,0.002,0.041,-0.059,null,null,-0.048,null,null,-0.023,null,null,null,null,null,null,null,0.007,null,0.032,null,null,0.213,0.022,0,-0.005,-0.007,null,null,null,null,null,null,null,-0.001,0.022,0.002,null,null,null,null,-0.067,-0.043,0.005,-0.017,-0.006,0.002,null,0.003,0.014,null,null,null,null,-0.024,-0.062,0.032,-0.036,-0.12,-0.026,-0.024,0.021,-0.001,0.009,0.025,-0.088,0.004,-0.005,-0.036,-0.01,0.059,0.004,-0.013,-0.023,-0,0.012,0.022,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.176,null,null,null,null,null,null,null,null,0.072,0.013,null,null,0,0.015,null,-0.008,null,null,0.012,0.035,null,null,-0.029,-0.018,-0.013,null,null,0.045,0.024,null,null,null,null,null,0.009,-0.006,null,null,null,null,null,null,-0.053,-0.054,0.012,0.07,0.046,0.041,-0.006,-0.002,0.077,0.073,-0.013,null,null,null,null,null,null,null,-0.021,0.006,-0.014,0.075,-0.015,null,null,0.003,null,null,null,null,null,null,null,null,null,-0.01,-0.018,-0.008,-0.039,-0.016,0.047,-0.173,-0.065,0.022,-0.027,0.009,0.031,0.011,0.015,null,0.005,0.001,0.04,null,null,-0.004,0.009,0.007,0.011,0.001,0.132,null,null,null,null,0.004,0.029,0.023,-0.008,0.018,0.002,-0.005,-0.01,0,null,null,null,null,-0.067,null,null,null,null,null,-0.012,0.003,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.148,null,null,null,null,null,null,null,null,null,-0.048,0,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.032,-0.021,-0.059,0.019,null,null,null,null,0.01,0.017,-0.001],[null,null,null,null,null,null,null,null,null,null,null,null,-0.165,-0.234,-0.221,-0.178,-0.153,null,null,null,null,null,null,null,null,null,null,null,-0.174,-0.137,null,null,-0.119,-0.113,-0.113,-0.115,-0.127,-0.121,-0.106,-0.111,-0.094,-0.113,-0.132,-0.113,-0.116,-0.111,-0.113,null,null,null,null,-0.117,-0.114,-0.137,-0.131,-0.196,-0.099,-0.126,null,-0.17,-0.147,-0.137,null,null,null,null,null,null,null,null,null,-0.117,null,null,null,null,-0.247,-0.124,null,null,null,null,null,null,-0.18,-0.154,null,null,null,null,null,null,null,-0.124,-0.113,null,null,null,null,-0.115,-0.141,null,-0.12,-0.126,-0.117,-0.108,-0.13,-0.111,-0.177,-0.185,null,null,null,null,null,-0.167,null,null,null,-0.15,-0.216,null,null,-0.128,null,null,-0.134,null,null,-0.096,-0.113,-0.165,null,null,null,null,null,null,null,null,null,null,null,-0.206,-0.122,-0.213,null,null,null,null,-0.245,-0.163,null,-0.18,null,-0.153,null,-0.106,-0.181,-0.211,-0.146,null,null,null,null,null,null,null,null,null,null,-0.142,null,null,null,null,null,-0.106,-0.122,-0.158,-0.204,null,-0.112,-0.155,-0.112,-0.144,-0.113,null,null,null,-0.103,null,null,null,null,-0.165,-0.103,-0.119,-0.142,-0.19,-0.176,-0.181,-0.201,-0.172,-0.144,null,null,-0.127,-0.186,-0.092,-0.106,null,null,-0.189,-0.192,null,null,-0.126,null,null,-0.136,-0.263,-0.141,-0.152,-0.029,-0.133,-0.181,-0.092,-0.143,-0.28,-0.25,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.29,-0.357,-0.317,-0.25,-0.154,-0.091,null,null,null,null,-0.073,-0.196,null,-0.118,null,-0.164,-0.12,-0.173,null,null,-0.172,-0.112,-0.099,-0.119,-0.155,-0.117,null,null,null,null,null,null,null,-0.144,0.756,-0.087,-0.127,null,null,null,-0.153,-0.113,-0.177,null,-0.127,null,null,-0.131,-0.11,-0.114,-0.105,-0.112,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.134,null,null,null,null,null,null,null,-0.127,-0.203,null,null,-0.168,null,null,null,null,null,null,-0.226,null,null,null,null,null,null,null,-0.145,null,-0.068,-0.101,-0.123,-0.115,-0.117,-0.135,-0.104,-0.129,-0.115,-0.123,-0.114,-0.124,null,null,null,null,null,null,null,-0.125,-0.134,-0.108,-0.133,-0.128,-0.119,-0.124,null,null,-0.172,-0.114,-0.089,null,-0.14,-0.143,-0.141,null,null,-0.154,-0.12,-0.199,null,null,-0.21,null,null,-0.222,null,null,null,null,null,null,null,-0.136,null,-0.115,null,null,0.055,-0.12,-0.124,-0.156,-0.154,null,null,null,null,null,null,null,-0.138,-0.116,-0.122,null,null,null,null,-0.223,-0.208,-0.131,-0.199,-0.135,-0.13,null,-0.133,-0.118,null,null,null,null,-0.194,-0.235,-0.159,-0.226,-0.261,-0.2,-0.158,-0.155,-0.173,-0.135,-0.132,-0.254,-0.127,-0.147,-0.202,-0.149,-0.149,-0.141,-0.152,-0.171,-0.128,-0.126,-0.124,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.08,null,null,null,null,null,null,null,null,-0.088,-0.117,null,null,-0.137,-0.204,null,-0.157,null,null,-0.126,-0.115,null,null,-0.191,-0.176,-0.168,null,null,-0.108,-0.113,null,null,null,null,null,-0.113,-0.146,null,null,null,null,null,null,-0.199,-0.215,-0.141,-0.088,-0.108,-0.128,-0.146,-0.133,-0.092,-0.131,-0.159,null,null,null,null,null,null,null,-0.178,-0.127,-0.262,-0.151,-0.169,null,null,-0.123,null,null,null,null,null,null,null,null,null,-0.146,-0.152,-0.153,-0.19,-0.192,-0.108,-0.324,-0.201,-0.139,-0.17,-0.123,-0.113,-0.13,-0.12,null,-0.144,-0.146,-0.151,null,null,-0.144,-0.144,-0.116,-0.119,-0.129,-0.072,null,null,null,null,-0.125,-0.167,-0.161,-0.201,-0.168,-0.154,-0.165,-0.191,-0.166,null,null,null,null,-0.209,null,null,null,null,null,-0.182,-0.132,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.182,null,null,null,null,null,null,null,null,null,-0.245,-0.26,null,null,null,null,null,null,null,null,null,null,null,null,null,-0.229,-0.16,-0.201,-0.122,null,null,null,null,-0.13,-0.147,-0.146],[null,null,null,null,null,null,null,null,null,null,null,null,0.145,0.1,0.107,0.214,0.146,null,null,null,null,null,null,null,null,null,null,null,0.167,0.239,null,null,0.135,0.12,0.117,0.168,0.175,0.129,0.111,0.111,0.193,0.115,0.151,0.12,0.155,0.116,0.177,null,null,null,null,0.117,0.133,0.122,0.117,0.124,0.236,0.138,null,0.104,0.129,0.182,null,null,null,null,null,null,null,null,null,0.17,null,null,null,null,0.142,0.136,null,null,null,null,null,null,0.105,0.139,null,null,null,null,null,null,null,0.128,0.125,null,null,null,null,0.168,0.182,null,0.148,0.183,0.189,0.169,0.117,0.123,0.13,0.163,null,null,null,null,null,0.125,null,null,null,0.149,0.103,null,null,0.165,null,null,0.151,null,null,0.27,0.212,0.179,null,null,null,null,null,null,null,null,null,null,null,0.088,0.144,0.095,null,null,null,null,0.1,0.161,null,0.142,null,0.118,null,0.194,0.123,0.124,0.15,null,null,null,null,null,null,null,null,null,null,0.145,null,null,null,null,null,0.227,0.198,0.181,0.109,null,0.19,0.2,0.154,0.151,0.154,null,null,null,0.233,null,null,null,null,0.182,0.238,0.22,0.182,0.139,0.131,0.17,0.185,0.148,0.155,null,null,0.213,0.285,0.3,0.233,null,null,0.287,0.308,null,null,0.138,null,null,0.156,0.164,0.26,0.286,0.392,0.137,0.144,0.221,0.158,0.119,0.21,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.088,-0.011,0.007,0.136,0.144,0.245,null,null,null,null,0.251,0.149,null,0.185,null,0.167,0.143,0.198,null,null,0.145,0.187,0.183,0.14,0.179,0.221,null,null,null,null,null,null,null,0.125,0.837,0.223,0.205,null,null,null,0.165,0.166,0.11,null,0.193,null,null,0.165,0.124,0.128,0.151,0.117,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.164,null,null,null,null,null,null,null,0.183,0.102,null,null,0.115,null,null,null,null,null,null,0.115,null,null,null,null,null,null,null,0.153,null,0.257,0.189,0.191,0.15,0.121,0.19,0.123,0.111,0.116,0.127,0.123,0.125,null,null,null,null,null,null,null,0.201,0.131,0.159,0.166,0.146,0.126,0.146,null,null,0.173,0.132,0.208,null,0.147,0.135,0.178,null,null,0.157,0.191,0.096,null,null,0.128,null,null,0.185,null,null,null,null,null,null,null,0.152,null,0.179,null,null,0.34,0.168,0.122,0.152,0.131,null,null,null,null,null,null,null,0.132,0.162,0.126,null,null,null,null,0.108,0.135,0.15,0.177,0.12,0.139,null,0.138,0.155,null,null,null,null,0.163,0.134,0.22,0.175,0.057,0.166,0.11,0.185,0.178,0.148,0.182,0.104,0.145,0.142,0.143,0.131,0.233,0.154,0.121,0.127,0.127,0.15,0.167,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.365,null,null,null,null,null,null,null,null,0.214,0.151,null,null,0.133,0.23,null,0.138,null,null,0.151,0.184,null,null,0.14,0.144,0.142,null,null,0.191,0.165,null,null,null,null,null,0.138,0.131,null,null,null,null,null,null,0.105,0.128,0.162,0.214,0.195,0.193,0.132,0.13,0.216,0.238,0.132,null,null,null,null,null,null,null,0.136,0.138,0.249,0.268,0.144,null,null,0.129,null,null,null,null,null,null,null,null,null,0.131,0.112,0.141,0.112,0.171,0.19,0.025,0.094,0.174,0.12,0.152,0.174,0.15,0.154,null,0.157,0.142,0.209,null,null,0.134,0.159,0.127,0.145,0.133,0.29,null,null,null,null,0.136,0.222,0.213,0.183,0.209,0.156,0.149,0.175,0.167,null,null,null,null,0.098,null,null,null,null,null,0.152,0.139,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.388,null,null,null,null,null,null,null,null,null,0.177,0.265,null,null,null,null,null,null,null,null,null,null,null,null,null,0.156,0.117,0.098,0.164,null,null,null,null,0.151,0.174,0.14]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Trait<\/th>\n      <th>S_female<\/th>\n      <th>S_female.lower<\/th>\n      <th>S_female.upper<\/th>\n      <th>S_male<\/th>\n      <th>S_male.lower<\/th>\n      <th>S_male.upper<\/th>\n      <th>Selection_index<\/th>\n      <th>Selection_index.lower<\/th>\n      <th>Selection_index.upper<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":800,"scrollCollapse":true,"buttons":["pageLength","colvis","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"selection_data"}],"pageLength":673,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,25,50,100,673]}},"evals":[],"jsHooks":[]}</script>
<p><strong>Table 2:</strong> Selection differentials (<span class="math inline">\(S\)</span>) for each phenotype, when expressed in each sex. The SI column shows the concordance of selection between the sexes. Positive values indicate that the trait is under concordant selection, a value of 0 indicates that selection is absent in one sex (note that it will miss conflict that occurs when strong stabilizing selection is present in that sex), and negative values indicate sexually antagonistic selection.</p>
<p><span class="math inline">\(~\)</span></p>
<pre class="r"><code>ggplot(data = S_complete_summary %&gt;% filter(Trait != &quot;fitness.late&quot;), aes(x = S_female, y = S_male, fill = Selection_index)) +
  #geom_hex(bins = 20) +
  geom_jitter(shape = 21, alpha = 0.9, size = 4, show.legend = TRUE, width = 0.003, height = 0.003) +
  #geom_point(shape = 21, alpha = 0.9, size = 5, show.legend = FALSE) +
  #geom_smooth(method = &#39;lm&#39;, color=&#39;black&#39;) +
  geom_hline(yintercept = 0, linetype = 2) +
  geom_vline(xintercept = 0, linetype = 2) +
  #coord_cartesian(xlim = c(-3, 3), ylim = c(-3, 3)) +
  scale_fill_carto_c(palette = &quot;Geyser&quot;,
                     breaks=c(-0.3, -0.15, 0, 0.15, 0.3),
                     limits=c(-0.3,0.3)) +
  labs(x = &quot;Female selection differential&quot;,
       y = &quot;Male selection differential&quot;,
       fill = &quot;Concordance\n of selection\n&quot;) +
  theme_bw() +
  theme(panel.border= element_blank(),
        axis.line=element_line(),
        text = element_text(size=14),
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.minor.x = element_blank(),
        axis.title.x = element_text(size = 14),
        axis.title.y = element_text(size = 14))</code></pre>
<p><img src="figure/Main_analysis.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/7bf0ee08d2c8b770d1c2d8aca0b2931897462adb/docs/figure/Main_analysis.Rmd/unnamed-chunk-13-1.png" target="_blank">7bf0ee0</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/043902fed2d66e5b988a2f7a2dd70c734e1e95f8/docs/figure/Main_analysis.Rmd/unnamed-chunk-13-1.png" target="_blank">043902f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/83a0c4fdf0c246cc8f853dbeecd4ab8b252df66e/docs/figure/Main_analysis.Rmd/unnamed-chunk-13-1.png" target="_blank">83a0c4f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-28
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/32f107fb9019fd1c55e06e4826fa11fed5dfc221/docs/figure/Main_analysis.Rmd/unnamed-chunk-13-1.png" target="_blank">32f107f</a>
</td>
<td>
tkeaney
</td>
<td>
2021-07-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure 5</strong>: an alternative way to show selection and the concordance of selection for traits in the DGRP. Each point represents a trait.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] job_0.3.0           ggtext_0.1.1        ggtips_0.4.1       
 [4] ggExtra_0.10.0      broom_0.8.0         DT_0.22            
 [7] shiny_1.7.1         tidybayes_3.0.2     brms_2.17.0        
[10] Rcpp_1.0.8.3        patchwork_1.1.1     pander_0.6.5       
[13] kableExtra_1.3.4    wesanderson_0.3.6   rcartocolor_2.0.0  
[16] RColorBrewer_1.1-3  pheatmap_1.0.12     googlesheets4_1.0.0
[19] forcats_0.5.1       stringr_1.4.0       dplyr_1.0.8        
[22] purrr_0.3.4         readr_2.1.2         tidyr_1.2.0        
[25] tibble_3.1.6        ggplot2_3.3.5       tidyverse_1.3.1    
[28] workflowr_1.7.0    

loaded via a namespace (and not attached):
  [1] utf8_1.2.2           tidyselect_1.1.2     lme4_1.1-29         
  [4] htmlwidgets_1.5.4    grid_4.1.0           munsell_0.5.0       
  [7] codetools_0.2-18     miniUI_0.1.1.1       withr_2.5.0         
 [10] Brobdingnag_1.2-7    colorspace_2.0-3     highr_0.9           
 [13] knitr_1.38           rstudioapi_0.13      stats4_4.1.0        
 [16] bayesplot_1.9.0      labeling_0.4.2       emmeans_1.7.3       
 [19] git2r_0.30.1         rstan_2.21.5         bit64_4.0.5         
 [22] farver_2.1.0         bridgesampling_1.1-2 rprojroot_2.0.3     
 [25] coda_0.19-4          vctrs_0.4.1          generics_0.1.2      
 [28] TH.data_1.1-0        xfun_0.30            R6_2.5.1            
 [31] markdown_1.1         gamm4_0.2-6          projpred_2.1.1      
 [34] assertthat_0.2.1     promises_1.2.0.1     scales_1.2.0        
 [37] vroom_1.5.7          multcomp_1.4-18      gtable_0.3.0        
 [40] processx_3.5.3       sandwich_3.0-1       rlang_1.0.2         
 [43] systemfonts_1.0.4    splines_4.1.0        gargle_1.2.0        
 [46] checkmate_2.1.0      inline_0.3.19        yaml_2.3.5          
 [49] reshape2_1.4.4       abind_1.4-5          modelr_0.1.8        
 [52] threejs_0.3.3        crosstalk_1.2.0      backports_1.4.1     
 [55] httpuv_1.6.5         gridtext_0.1.4       tensorA_0.36.2      
 [58] tools_4.1.0          ellipsis_0.3.2       jquerylib_0.1.4     
 [61] posterior_1.2.1      ggridges_0.5.3       plyr_1.8.7          
 [64] base64enc_0.1-3      ps_1.7.0             prettyunits_1.1.1   
 [67] zoo_1.8-10           haven_2.5.0          fs_1.5.2            
 [70] magrittr_2.0.3       ggdist_3.1.1         colourpicker_1.1.1  
 [73] reprex_2.0.1         googledrive_2.0.0    mvtnorm_1.1-3       
 [76] whisker_0.4          matrixStats_0.62.0   hms_1.1.1           
 [79] shinyjs_2.1.0        mime_0.12            evaluate_0.15       
 [82] arrayhelpers_1.1-0   xtable_1.8-4         shinystan_2.6.0     
 [85] readxl_1.4.0         gridExtra_2.3        rstantools_2.2.0    
 [88] compiler_4.1.0       crayon_1.5.1         minqa_1.2.4         
 [91] StanHeaders_2.21.0-7 htmltools_0.5.2      mgcv_1.8-40         
 [94] later_1.3.0          tzdb_0.3.0           RcppParallel_5.1.5  
 [97] lubridate_1.8.0      DBI_1.1.2            dbplyr_2.1.1        
[100] MASS_7.3-57          boot_1.3-28          Matrix_1.4-1        
[103] cli_3.2.0            parallel_4.1.0       igraph_1.3.1        
[106] pkgconfig_2.0.3      getPass_0.2-2        xml2_1.3.3          
[109] svUnit_1.0.6         dygraphs_1.1.1.6     svglite_2.1.0       
[112] bslib_0.3.1          webshot_0.5.3        estimability_1.3    
[115] rvest_1.0.2          distributional_0.3.0 callr_3.7.0         
[118] digest_0.6.29        rmarkdown_2.13       cellranger_1.1.0    
[121] gtools_3.9.2         nloptr_2.0.0         lifecycle_1.0.1     
[124] nlme_3.1-157         jsonlite_1.8.0       viridisLite_0.4.0   
[127] fansi_1.0.3          pillar_1.7.0         lattice_0.20-45     
[130] loo_2.5.1            fastmap_1.1.0        httr_1.4.2          
[133] pkgbuild_1.3.1       survival_3.3-1       glue_1.6.2          
[136] xts_0.12.1           shinythemes_1.2.0    bit_4.0.4           
[139] stringi_1.7.6        sass_0.4.1          </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
