<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Running GWAS on the DGRP phenotypes</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Intralocus sexual conflict in the DGRP</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Main_analysis.html">Main analysis</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tomkeaney/DGRP_sexual_conflict">
    <span class="fab fa fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Running GWAS on the DGRP phenotypes</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DGRP_sexual_conflict/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210706code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210706)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210706code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210706)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPsexualconflicttree416fa1cf281f0795fc968d0c4c905e68354f3630targetblank416fa1ca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/tomkeaney/DGRP_sexual_conflict/tree/416fa1cf281f0795fc968d0c4c905e68354f3630" target="_blank">416fa1c</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPsexualconflicttree416fa1cf281f0795fc968d0c4c905e68354f3630targetblank416fa1ca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/tomkeaney/DGRP_sexual_conflict/tree/416fa1cf281f0795fc968d0c4c905e68354f3630" target="_blank">416fa1c</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/SNP_correlations.Rmd
    Untracked:  analysis/process_mashr.Rmd
    Untracked:  analysis/run_mashr.Rmd
    Untracked:  code/gcta64
    Untracked:  code/launch_slurm.bash
    Untracked:  code/plink
    Untracked:  code/run_mashr.R
    Untracked:  gwas_data/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/run_GWAS.Rmd</code>) and HTML (<code>docs/run_GWAS.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/416fa1cf281f0795fc968d0c4c905e68354f3630/analysis/run_GWAS.Rmd" target="_blank">416fa1c</a>
</td>
<td>
lukeholman
</td>
<td>
2021-08-05
</td>
<td>
wflow_publish(“analysis/run_GWAS.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(bigsnpr) # to install:   devtools::install_github(&quot;privefl/bigsnpr&quot;)
library(glue)
plink &lt;- file.path(getwd(), &quot;code/plink&quot;) # I use plink v1.9
# gcta &lt;- file.path(getwd(), &quot;code/gcta64&quot;)
options(readr.show_col_types = FALSE)


# helper function to pass commands to the terminal
# Note that we set `intern = TRUE`, and pass the result of `system()` to `cat()`,
# ensuring that the Terminal output will be printed in this knitr report.
run_command &lt;- function(shell_command, wd = getwd(), path = &quot;&quot;){
  cat(system(glue(&quot;cd &quot;, wd, path, &quot;\n&quot;,shell_command), intern = TRUE), sep = &#39;\n&#39;)
}</code></pre>
<div id="load-trait-data-to-be-used-in-gwas" class="section level2">
<h2>Load trait data to be used in GWAS</h2>
<pre class="r"><code># Tom to check: some line/trait combinations have multiple values. Look at this:
# read_csv(&quot;data/all.dgrp.phenos_unscaled.csv&quot;) %&gt;% group_by(line, Trait) %&gt;% summarise(n = n()) %&gt;% arrange(-n)
# read_csv(&quot;data/all.dgrp.phenos_unscaled.csv&quot;) %&gt;% filter(Trait == &quot;starvation.resistance.yeast.poor.diet.f&quot; &amp; line == 306) %&gt;% as.data.frame()

traits_for_gwas &lt;- read_csv(&quot;data/all.dgrp.phenos_unscaled.csv&quot;) %&gt;% 
  # filter(!(`Trait guild` %in% allowed_guilds) | is.na(`Trait guild`)) %&gt;% 
  group_by(Trait) %&gt;% 
  summarise(lines_measured = length(unique(line))) %&gt;% 
  ungroup() %&gt;% 
  filter(lines_measured &gt; 99) %&gt;% pull(Trait) # find traits with 100+ replicates

traits_for_gwas &lt;- traits_for_gwas[!(grepl(&quot;fitness.late&quot;, traits_for_gwas))]

# replace any slashes in the trait name, e.g. &quot;1/3-Methylhistidine.high.yeast.f&quot;, as this is not ok to use as a file name
traits_for_gwas &lt;- str_replace_all(traits_for_gwas, &quot;[/]&quot;, &quot;_&quot;) 

line_mean_phenotypes &lt;- read_csv(&quot;data/all.dgrp.phenos_unscaled.csv&quot;) %&gt;%
  filter(Trait %in% traits_for_gwas) %&gt;% 
  select(line, Trait, trait_value) %&gt;% 
  distinct(line, Trait, .keep_all = TRUE) %&gt;% # Remove this line once Tom resolves the duplicates
  spread(Trait, trait_value) %&gt;% 
  mutate_at(vars(-line), ~ as.numeric(scale(.x))) %&gt;%  # scale the traits
  mutate(line = paste(&quot;line&quot;, line, sep = &quot;&quot;))</code></pre>
</div>
<div id="perform-snp-quality-control-and-imputation" class="section level2">
<h2>Perform SNP quality control and imputation</h2>
<p>We cleaned up the DGRP’s .bed/.bim/.fam files (available from the <a href="http://dgrp2.gnets.ncsu.edu/">Mackay lab website</a>) as follows:</p>
<ol style="list-style-type: decimal">
<li>Remove any SNPs for which genotypes are missing for &gt;10% of the DGRP lines. We then use the software <a href="https://faculty.washington.edu/browning/beagle/beagle.html">Beagle</a> to impute the remaining missing genotypes.</li>
<li>Remove SNPs with a minor allele frequency of less than 5%</li>
</ol>
<p>Note that in the PLINK-formatted genotype files, lines fixed for the major allele are coded as 2, and lines fixed for the minor allele as 0. This means that in the association tests we calculate, negative effect sizes mean that the minor allele is associated with lower fitness, while positive effect sizes means that the minor allele is associated with higher fitness.</p>
<pre class="r"><code>perform_SNP_QC_and_imputation &lt;- function(phenotypes){
  
  beagle &lt;- bigsnpr::download_beagle()
  
  # Use Plink to clean and subset the DGRP&#39;s SNP data as follows:
  # Only keep SNPs for which at least 90% of DGRP lines were successfully genotyped (--geno 0.1)
  # Only keep SNPs with a minor allele frequency of 0.05 or higher (--maf 0.05)
  # Finally, write the processed BIM/BED/FAM files to the data/derived directory
  run_command(glue(&quot;{plink} --bfile dgrp2&quot;,
                   &quot; --geno 0.1 --maf 0.05 --allow-no-sex&quot;, 
                   &quot; --make-bed --out ../derived/dgrp2_QC_all_lines&quot;), path = &quot;/gwas_data/input/&quot;)
  
  # Use the shell command &#39;sed&#39; to remove underscores from the DGRP line names in the .fam file (e.g. &#39;line_120&#39; becomes &#39;line120&#39;)
  # Otherwise, these underscores cause trouble when we need to convert from PLINK to vcf format (vcf format uses underscore as a separator)
  for(i in 1:2) run_command(&quot;sed -i &#39;&#39; &#39;s/_//&#39; dgrp2_QC_all_lines.fam&quot;, path = &quot;/gwas_data/derived/&quot;)
  
  # Now impute the missing genotypes using Beagle
  # This part uses the data for the full DGRP panel of &gt;200 lines, to infer missing genotypes as accurately as possible. 
  # This step uses a lot of memory (I set to 28MB max, and it used 26.5GB), but maybe it can also run on a less powerful computer?
  # The bigsnpr package provides a helpful wrapper for Beagle called snp_beagleImpute(): it translates to a VCF file and back again using PLINK
  snp_beagleImpute(beagle, plink, 
                   bedfile.in = &quot;gwas_data/derived/dgrp2_QC_all_lines.bed&quot;, 
                   bedfile.out = &quot;gwas_data/derived/dgrp2_QC_all_lines_imputed.bed&quot;,
                   ncores = 7, 
                   memory.max = 20)
  
  # assign a sex of &#39;female&#39; to all the DGRP lines (Beagle removes the sex, and it seems PLINK needs individuals to have a sex)
  run_command(&quot;sed -i &#39;&#39; &#39;s/    0   0   0/  0   0   2/&#39; dgrp2_QC_all_lines_imputed.fam&quot;, path = &quot;/gwas_data/derived/&quot;)
  
  # Re-write the .bed file, to make sure the MAF and genotyping thresholds are correctly assigned post-Beagle
  run_command(glue(&quot;{plink} --bfile dgrp2_QC_all_lines_imputed&quot;,
                   &quot; --geno 0.1 --maf 0.05&quot;, 
                   &quot; --make-bed --out dgrp2_QC_all_lines_imputed_correct&quot;), path = &quot;/gwas_data/derived/&quot;)

  # Use PLINK to get the allele IDs and calculate the MAFs across the whole DGRP, for all SNPs that survived QC
  # The file created is called data/derived/plink.frq
  run_command(&quot;{plink} --bfile dgrp2_QC_all_lines_imputed_correct --freq&quot;, path = &quot;/gwas_data/derived&quot;)

  # Clean up:
  unlink(c(&quot;gwas_data/derived/plink.log&quot;,
           &quot;gwas_data/derived/dgrp2_QC_all_lines_imputed.bed&quot;,
           &quot;gwas_data/derived/dgrp2_QC_all_lines_imputed.bim&quot;,
           &quot;gwas_data/derived/dgrp2_QC_all_lines_imputed.fam&quot;,
           &quot;gwas_data/derived/dgrp2_QC_all_lines_imputed.log&quot;,
           &quot;gwas_data/derived/dgrp2_QC_all_lines_imputed_correct.log&quot;))
}

if(!file.exists(&quot;gwas_data/derived/dgrp2_QC_all_lines_imputed_correct.bed&quot;)) perform_SNP_QC_and_imputation(phenotypes = predicted_line_means)

# creates the following 3 files: dgrp2_QC_all_lines_imputed_correct.bed/bim/fam

# These are the 205 DGRP lines with genotype data. Check for incorrect line names in the phenotype data
genotyped_lines &lt;- read.table(&quot;gwas_data/derived/dgrp2_QC_all_lines_imputed_correct.fam&quot;)[,1]
line_mean_phenotypes &lt;- line_mean_phenotypes %&gt;% filter(line %in% genotyped_lines)</code></pre>
</div>
<div id="create-a-reduced-list-of-ld-pruned-snps-with-plink" class="section level2">
<h2>Create a reduced list of LD-pruned SNPs with PLINK</h2>
<p>To keep the computation time and memory usage manageable, we did not analyse every SNP that passed quality control (i.e. 1,646,652 SNPs), but rather a subset of them that were approximately in linkage disequilibrium. We identified this LD-pruned set of SNPs using the PLINK arguments <code>--indep-pairwise 100 10 0.2</code>, i.e. pruning within 100kB sliding windows, sliding 10 variants along with each step, and allowing a maximum pairwise <span class="math inline">\(r^2\)</span> threshold of 0.2 between loci. With these parameters, 1420071 SNPs were removed, leaving 2.2658110^{5} for downstream analysis.</p>
<pre class="r"><code># indep-pairwise arguments are: 
# 100kB window size, 
# variant count to shift the window by 10 variants at the end of each step, 
# pairwise r^2 threshold of 0.2
run_command(glue(&quot;{plink} --bfile dgrp2_QC_all_lines_imputed_correct&quot;,
                 &quot; --indep-pairwise 100 10 0.2&quot;), path = &quot;/gwas_data/derived/&quot;)


run_command(glue(&quot;{plink} --bfile dgrp2_QC_all_lines_imputed_correct&quot;,
                 &quot; --extract plink.prune.in&quot;,
                 &quot; --make-bed --out dgrp2_QC_all_lines_LDpruned&quot;), path = &quot;/gwas_data/derived/&quot;)
unlink(&quot;gwas_data/derived/plink.prune.in&quot;)
unlink(&quot;gwas_data/derived/plink.prune.out&quot;)</code></pre>
</div>
<div id="run-all-the-gwas" class="section level2">
<h2>Run all the GWAS</h2>
<p>Note: because plink defines the minor allele as the alt allele (so, lines fixed for the minor allele are scored as genotype: 2, and those with the major allele as genotype: 0), a <em>positive</em> effect size in these association tests means the <em>minor</em> allele is associated with a <em>higher</em> value of the trait in question.</p>
<pre class="r"><code>gwas_one_trait &lt;- function(focal_phenotype){
  
  # First make &#39;focal_data&#39;, a 2-column data frame with the line and the focal phenotype value
  focal_data &lt;- line_mean_phenotypes %&gt;% 
    select(line, !! focal_phenotype) 
  names(focal_data)[2] &lt;- &quot;focal_pheno&quot;
  
  # Prune it to just the lines that have some phenotype data
  focal_data &lt;- focal_data %&gt;% 
    filter(!is.na(line), !is.na(focal_pheno)) 
  
  # Make a list of the lines in our sample and save as a text file for passing to PLINK
  lines_to_keep &lt;- gsub(&quot;_&quot;, &quot;&quot;, focal_data$line) %&gt;% cbind(.,.)
  write.table(lines_to_keep, row.names = FALSE, col.names = FALSE, file = &quot;gwas_data/derived/lines_to_keep.txt&quot;, quote = FALSE)
  
  # Now cull the PLINK files to just the lines that we measured, and re-apply the 
  # MAF cut-off of 0.05 for the new smaller sample of DGRP lines
  run_command(glue(&quot;{plink} --bfile dgrp2_QC_all_lines_LDpruned&quot;,
                   &quot; --keep-allele-order&quot;, 
                   &quot; --keep lines_to_keep.txt --geno 0.1 --maf 0.05&quot;, 
                   &quot; --make-bed --out dgrp2_QC_focal_lines&quot;), path = &quot;/gwas_data/derived/&quot;)
  
  # Define a function to add our phenotype data to a .fam file, which is needed for GWAS analysis and to make sure PLINK includes these samples
  # The &#39;phenotypes&#39; data frame needs to have a column called &#39;line&#39;
  add_phenotypes_to_fam &lt;- function(filepath, focal_data){
    read_delim(filepath, col_names = FALSE, delim = &quot; &quot;) %&gt;% 
      select(X1, X2, X3, X4, X5) %&gt;% # Get all the non-phenotype columns
      left_join(focal_data, 
                by = c(&quot;X1&quot; = &quot;line&quot;)) %&gt;%
      write.table(file = &quot;gwas_data/derived/dgrp2_QC_focal_lines_NEW.fam&quot;, 
                  col.names = FALSE, row.names = FALSE, 
                  quote = FALSE, sep = &quot; &quot;)
    unlink(&quot;gwas_data/derived/dgrp2_QC_focal_lines.fam&quot;)
    file.rename(&quot;gwas_data/derived/dgrp2_QC_focal_lines_NEW.fam&quot;, &quot;gwas_data/derived/dgrp2_QC_focal_lines.fam&quot;)
  }
  
  add_phenotypes_to_fam(&quot;gwas_data/derived/dgrp2_QC_focal_lines.fam&quot;, focal_data)
 
  # # Write a file with the line and phenotype data called phenotype.txt, for gcta64
  # pheno_data &lt;- focal_data %&gt;% 
  #   mutate(line_copy = line) %&gt;% 
  #   select(line, line_copy, focal_pheno) %&gt;% as.matrix() 
  # pheno_data %&gt;% 
  #   write.table(row.names = FALSE, col.names = FALSE, 
  #               file = &quot;gwas_data/derived/phenotype.txt&quot;, quote = FALSE)

  # Run mixed-model GWAS (in practice, the relatedness is low so it just a runs linear model)
  print(focal_phenotype)
  run_command(&quot;{plink} --bfile dgrp2_QC_focal_lines  --assoc --maf 0.05 --out gwas_results/new&quot;, 
              path = &quot;/gwas_data/derived&quot;)

  # Rename and compress the GWAS summary stats file:
  read.table(&quot;gwas_data/derived/gwas_results/new.qassoc&quot;, header = TRUE) %&gt;% 
    select(SNP, BETA, SE, P) %&gt;% 
    write_tsv(glue(&quot;gwas_data/derived/gwas_results/{focal_phenotype}.tsv.gz&quot;))
  unlink(&quot;gwas_data/derived/gwas_results/new.qassoc&quot;)
  
  # Rename the plink log file
  file.rename(&quot;gwas_data/derived/gwas_results/new.log&quot;, glue(&quot;gwas_data/derived/gwas_results/{focal_phenotype}_log.txt&quot;))
  
}

files &lt;- list.files(&quot;gwas_data/derived/gwas_results&quot;, pattern = &quot;tsv.gz&quot;, full.names = TRUE)
already_run_traits &lt;- str_split(files, &quot;/&quot;) %&gt;% map_chr(~ .x[4]) %&gt;% str_remove_all(&quot;.tsv.gz&quot;)

traits_to_run_for_gwas &lt;- traits_for_gwas[!(traits_for_gwas %in% already_run_traits)]

lapply(traits_to_run_for_gwas, gwas_one_trait)

unlink(list.files(&quot;gwas_data/derived&quot;, pattern = &quot;focal_lines&quot;, full.names = T))
unlink(c(&quot;gwas_data/derived/lines_to_keep.txt&quot;))</code></pre>
<!-- ```{r} -->
<!-- # First make 'focal_data', a 2-column data frame with the line and the focal phenotype value -->
<!-- focal_data <- line_mean_phenotypes %>%  -->
<!--   select(line, lifespan.25C.2020.m, fitness.early.f)  -->
<!-- names(focal_data)[2:3] <- c("focal_pheno1", "focal_pheno2") -->
<!-- # Write a file with the line and phenotype data called phenotype.txt, for gcta64 -->
<!-- pheno_data <- focal_data %>% -->
<!--   mutate(line_copy = line) %>% -->
<!--   select(line, line_copy, focal_pheno1) %>%  -->
<!--   as.matrix() -->
<!-- pheno_data[is.na(pheno_data)] <- -9 -->
<!-- pheno_data %>% -->
<!--   write.table(row.names = FALSE, col.names = FALSE, -->
<!--               file = "gwas_data/derived/phenotype.txt", quote = FALSE) -->
<!-- run_command(glue("{gcta} --bfile dgrp2_QC_all_lines_imputed_correct", -->
<!--                  " --make-grm",  -->
<!--                  " --out gcta_GRM"), path = "/gwas_data/derived/") -->
<!-- run_command(glue("{gcta} --grm gcta_GRM --make-bK-sparse 0.05 --out sparse_GRM"), path = "/gwas_data/derived/") -->
<!-- # fastGWA mixed model (based on the sparse GRM generated above) -->
<!-- run_command(glue("{gcta} --bfile dgrp2_QC_all_lines_imputed_correct --grm-sparse sparse_GRM --fastGWA-mlm --pheno phenotype.txt 1 --threads 10 --out geno_assoc"), path = "/gwas_data/derived/") -->
<!-- run_command(glue("{gcta} --bfile dgrp2_QC_all_lines_imputed_correct", -->
<!--                  " --pheno phenotype.txt --grm gcta_GRM",  -->
<!--                  " --reml-bivar 1 2 --reml-bivar-lrt-rg 0 --out gcta_test"), path = "/gwas_data/derived/") -->
<!-- run_command(glue("{gcta} --bfile dgrp2_QC_all_lines_imputed_correct --pheno phenotype.txt --fastGWA-mlm --grm-sparse sparse_GRM --out gcta_test"), path = "/gwas_data/derived/") -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
