<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="lukeholman" />

<meta name="date" content="2021-07-28" />

<title>run_mashr</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Intralocus sexual conflict in the DGRP</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Main_analysis.html">Main analysis</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tomkeaney/DGRP_sexual_conflict">
    <span class="fab fa fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">run_mashr</h1>
<h4 class="author">lukeholman</h4>
<h4 class="date">2021-07-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DGRP_sexual_conflict/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210706code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210706)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210706code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210706)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPsexualconflicttree7d22f2db972e2a34f5a03541d72c450f40e75138targetblank7d22f2da"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/tomkeaney/DGRP_sexual_conflict/tree/7d22f2db972e2a34f5a03541d72c450f40e75138" target="_blank">7d22f2d</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPsexualconflicttree7d22f2db972e2a34f5a03541d72c450f40e75138targetblank7d22f2da" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/tomkeaney/DGRP_sexual_conflict/tree/7d22f2db972e2a34f5a03541d72c450f40e75138" target="_blank">7d22f2d</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/SNP_correlations.Rmd
    Untracked:  analysis/process_mashr.Rmd
    Untracked:  code/gcta64
    Untracked:  code/launch_slurm.bash
    Untracked:  code/plink
    Untracked:  code/run_mashr.R
    Untracked:  gwas_data/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/run_mashr.Rmd</code>) and HTML (<code>docs/run_mashr.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/7d22f2db972e2a34f5a03541d72c450f40e75138/analysis/run_mashr.Rmd" target="_blank">7d22f2d</a>
</td>
<td>
lukeholman
</td>
<td>
2021-08-05
</td>
<td>
wflow_publish(“analysis/run_mashr.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(tidyverse)
library(ashr) 
library(mashr) 
library(glue)
library(rslurm)

files &lt;- list.files(&quot;gwas_data/derived/gwas_results&quot;, pattern = &quot;tsv.gz&quot;, full.names = TRUE)
traits &lt;- str_split(files, &quot;/&quot;) %&gt;% map_chr(~ .x[4]) %&gt;% str_remove_all(&quot;.tsv.gz&quot;)
nonfitness_traits &lt;- traits[!(traits %in% c(&quot;fitness.early.f&quot;, &quot;fitness.early.m&quot;))]</code></pre>
<div id="code-for-running-mashr" class="section level2">
<h2>Code for running mashr</h2>
<p>The following code takes one phenotypic trait (other than early male/female fitness as input) as its argument, and runs <code>mashr</code> on 3 sets of GWAS-derived effect sizes for each SNP: 1 from the GWAS of female early life fitness, one from the GWAS of male early life fitness, and the 1 from the GWAS of the focal trait. We run <code>mashr</code> in the data-driven mode (see the vignette), and save the ouput of <code>mash()</code> as well as the list of SNPs that were analysed (i.e. the set of SNPs for which an effect size was calculated for all three traits). This code is designed to be run on a computer cluster using the SLURM workload manager.</p>
<pre class="r"><code>run_mashr &lt;- function(nonfitness_trait, overwrite = FALSE){
  
  # project folder on Spartan
  setwd(&quot;/data/projects/punim0243/DGRP_sexual_conflict&quot;)
  
  print(nonfitness_trait)
  
  # Set up the output file name. If it already exists and overwrite = FALSE, just quit.
  # nonfitness_trait_neat &lt;- str_replace_all(nonfitness_trait, &quot;[.]&quot;, &quot;_&quot;)
  output_file &lt;- glue(&quot;gwas_data/derived/mashr_results/{nonfitness_trait}.rds&quot;)
  if(!overwrite &amp; file.exists(output_file)) return(NULL)
  
  # Input file names (GWAS stats for 2 fitness traits and a another phenotype)
  files &lt;- c(
    &quot;gwas_data/derived/gwas_results/fitness.early.f.tsv.gz&quot;,
    &quot;gwas_data/derived/gwas_results/fitness.early.m.tsv.gz&quot;,
    paste(&quot;gwas_data/derived/gwas_results/&quot;, nonfitness_trait, &quot;.tsv.gz&quot;, sep = &quot;&quot;))
  traits &lt;- c(&quot;fitness.early.f&quot;, &quot;fitness.early.m&quot;, &quot;nonfitness_trait&quot;)
  
  # Load the GWAS data for these 3 things
  all_dat &lt;- lapply(1:length(files), function(i){
    trait &lt;- traits[i]
    dat &lt;- read_tsv(files[i]) %&gt;% select(SNP, BETA, SE)
    names(dat)[2] &lt;- paste(&quot;beta&quot;, trait, sep = &quot;_&quot;)
    names(dat)[3] &lt;- paste(&quot;se&quot;, trait, sep = &quot;_&quot;)
    dat
  })
  names(all_dat) &lt;- traits

  ##### 1. Data set up
  # Define functions to get the focal data, and set up as betas and SE for mashr
  make_mashr_data_one_phenotype &lt;- function(nonfitness_trait){
    left_join(all_dat[[1]], all_dat[[2]], by = &quot;SNP&quot;) %&gt;% # 1=female fitness, 2=male fitness, 3=phenotype
      left_join(all_dat[[3]], by = &quot;SNP&quot;) %&gt;% 
      select(SNP, starts_with(&quot;beta&quot;), starts_with(&quot;SE&quot;))
  }
  
  mashr_setup &lt;- function(beta_and_se){
    betas &lt;- beta_and_se %&gt;% select(starts_with(&quot;beta&quot;)) %&gt;% as.matrix()
    SEs &lt;- beta_and_se %&gt;% select(starts_with(&quot;SE&quot;)) %&gt;% as.matrix()
    rownames(betas) &lt;- beta_and_se$SNP
    rownames(SEs) &lt;- beta_and_se$SNP
    mash_set_data(betas, SEs)
  }
  
  beta_and_se &lt;- make_mashr_data_one_phenotype(nonfitness_trait)
  mash_data &lt;- mashr_setup(beta_and_se)
  
  ##### 2. Obtain data-driven covariance matrices for mashr (see mashr vignette)
  # This first finds strong signals in the complete SNP effect size data, and use
  # to estimate the underlying covariance matrices for the SNPs&#39; effect sizes
  m.1by1 &lt;- mash_1by1(mash_data) 
  strong &lt;- get_significant_results(m.1by1, thresh = 0.2)   
  U.pca &lt;- cov_pca(mash_data, npc = 3, subset = strong)
  U &lt;- cov_ed(mash_data, U.pca, subset = strong)
  
  ##### 3. Run mashr and save the results
  # Now that the data and cov matrices are set up, we can run mashr (takes a bit over 1 hour)
  mash_ED &lt;- mash(data = mash_data, Ulist = U)
  list(SNPs = beta_and_se %&gt;% pull(SNP), mash_ED) %&gt;% saveRDS(output_file)
}

# Run all the mashr jobs on Spartan using SLURM (using the rslurm package) 
slurm_parameters_df &lt;- data.frame(nonfitness_trait = nonfitness_traits)
sopt1 &lt;- list(time = &#39;6:00:00&#39;) # 6 hour time limit, usually takes &lt;2 hours

sjob &lt;- slurm_apply(f = run_mashr, params = slurm_parameters_df, 
                    jobname = &#39;DGRP_mashr&#39;, slurm_options = sopt1,
                    nodes = nrow(slurm_parameters_df), cpus_per_node = 1, submit = TRUE)</code></pre>
<!-- ```{r} -->
<!-- run_mashr <- function(beta_and_se){ -->
<!--   mashr_setup <- function(beta_and_se){ -->
<!--     betas <- beta_and_se %>% select(starts_with("beta")) %>% as.matrix() -->
<!--     SEs <- beta_and_se %>% select(starts_with("SE")) %>% as.matrix() -->
<!--     rownames(betas) <- beta_and_se$SNP -->
<!--     rownames(SEs) <- beta_and_se$SNP -->
<!--     mash_set_data(betas, SEs) -->
<!--   } -->
<!--   mash_data <- mashr_setup(beta_and_se) -->
<!--   # define the covariance matrices ourselves (a long list of a priori interesting matrices are checked) -->
<!--   # 2-trait matrices -->
<!--   make_2trait_matrix <- function(missing_one, name) { -->
<!--     U <- cov_canonical(mash_data) -->
<!--     U <- U[-(1:4)] -->
<!--     U <- lapply(U, function(x){ -->
<!--       x[,missing_one] <- 0 -->
<!--       x[missing_one,] <- 0 -->
<!--       x -->
<!--     }) -->
<!--     names(U) <- paste(names(U), name, sep = "_") -->
<!--     U -->
<!--   } -->
<!--   make_negative <- function(twotrait_matrix, i, j){ -->
<!--     xx <- lapply(twotrait_matrix, function(x) { -->
<!--                    x[x<1] <- x[x<1]*-1 -->
<!--                    x}) -->
<!--     xx[[1]][i,j] <- -1 -->
<!--     xx[[1]][j,i] <- -1 -->
<!--     xx -->
<!--   } -->
<!--   # doesn't affect female fitness, but affects the trait and male fitness -->
<!--   pos_nofemale <- make_2trait_matrix(1, "nofemale")  -->
<!--   names(pos_nofemale) <- c("pos_selected_males1", "pos_selected_males0.25", "pos_selected_males0.5", "pos_selected_males0.75") -->
<!--   neg_nofemale <- make_negative(pos_nofemale, 2, 3) -->
<!--   names(neg_nofemale) <- c("neg_selected_males1", "neg_selected_males0.25", "neg_selected_males0.5", "neg_selected_males0.75") -->
<!--   # doesn't affect male fitness, but affects the trait and female fitness -->
<!--   pos_nomale <- make_2trait_matrix(2, "nomale")  -->
<!--   names(pos_nomale) <- c("pos_selected_females1", "pos_selected_females0.25", "pos_selected_females0.5", "pos_selected_females0.75") -->
<!--   neg_nomale <- make_negative(pos_nomale, 1, 3) -->
<!--   names(neg_nomale) <- c("neg_selected_females1", "neg_selected_females0.25", "neg_selected_females0.5", "neg_selected_females0.75") -->
<!--   # Doesn't affect the trait, but affects fitness in both sexes -->
<!--   pos_notrait <- make_2trait_matrix(3, "notrait")  -->
<!--   names(pos_notrait) <- c("sex_concord_1_notrait", "sex_concord_0.25_notrait", "sex_concord_0.5_notrait", "sex_concord_0.75_notrait") -->
<!--   neg_notrait <- make_negative(pos_notrait, 1, 2) -->
<!--   names(neg_notrait) <- c("sex_antag_1_notrait", "sex_antag_0.25_notrait", "sex_antag_0.5_notrait", "sex_antag_0.75_notrait") -->
<!--   # Add the 3-trait stuff.... -->
<!--   # Get the mashr default canonical covariance matrices: this includes the ones  -->
<!--   # called "null", "uniform", and "same sign" in the list that precedes this code chunk -->
<!--   U <- cov_canonical(mash_data) -->
<!--   # And now our custom covariance matrices:  -->
<!--   U <- c(U, pos_nofemale, neg_nofemale, pos_nomale, neg_nomale, pos_notrait, neg_notrait) -->
<!--   return(U) -->
<!--   return(mash(data = mash_data, Ulist = U)) # Run mashr -->
<!-- } -->
<!-- run_mashr(beta_and_se) -->
<!-- mash_output <- run_mashr(beta_and_se) -->
<!-- # if(!file.exists("data/derived/mashr_results_canonical.rds")){ -->
<!-- #   run_mashr(data_for_mashr, mashr_mode = "ED", ED_p_cutoff = 0.2) %>% -->
<!-- #     write_rds(path = "data/derived/mashr_results_ED.rds") -->
<!-- #    -->
<!-- #   run_mashr(data_for_mashr, mashr_mode = "canonical") %>% -->
<!-- #     write_rds(path = "data/derived/mashr_results_canonical.rds") -->
<!-- # } else { -->
<!-- #   mashr_results_ED <- read_rds("data/derived/mashr_results_ED.rds") -->
<!-- #   mashr_results_canonical <- read_rds("data/derived/mashr_results_canonical.rds") -->
<!-- # } -->
<!-- ``` -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
