<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>‘GCTA analysis’</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Selection, sexual conflict and sex-specific parameters in the DGRP</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tomkeaney/DGRP_sexual_conflict">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">‘GCTA analysis’</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-01
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DGRP_sexual_conflict/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210706code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210706)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210706code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210706)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPsexualconflicttreea4076aa5e76c8729c82465a320f59c221ba09a22targetblanka4076aaa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/tomkeaney/DGRP_sexual_conflict/tree/a4076aa5e76c8729c82465a320f59c221ba09a22" target="_blank">a4076aa</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneyDGRPsexualconflicttreea4076aa5e76c8729c82465a320f59c221ba09a22targetblanka4076aaa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/tomkeaney/DGRP_sexual_conflict/tree/a4076aa5e76c8729c82465a320f59c221ba09a22" target="_blank">a4076aa</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/.DS_Store
    Ignored:    gwas_data/

Untracked files:
    Untracked:  %
    Untracked:  -Toms-laptop.Rhistory
    Untracked:  Manuscript/
    Untracked:  Reported_heritability.xlsx
    Untracked:  Rplot.pdf
    Untracked:  S_plots.pdf
    Untracked:  Selection_differential_plot.pdf
    Untracked:  Selection_differentials.pdf
    Untracked:  Selection_differentials_f.pdf
    Untracked:  Selection_differentials_m.pdf
    Untracked:  analysis/Random_plots.R
    Untracked:  code/LICENSE
    Untracked:  code/gcta64
    Untracked:  code/prettify
    Untracked:  code/toy.map
    Untracked:  code/toy.ped
    Untracked:  data/SNP_heritability.csv
    Untracked:  data/SNP_heritability_cleaned.csv
    Untracked:  data/SNP_selection.csv
    Untracked:  data/S_female.csv
    Untracked:  data/all_SNP_data.csv
    Untracked:  data/rfm_dimorphism_model.rds
    Untracked:  dimorphism_subset.pdf
    Untracked:  fitness_variation.pdf
    Untracked:  fits/
    Untracked:  lifetime_fecundity_S.pdf
    Untracked:  mating_latency_S.pdf

Unstaged changes:
    Modified:   .Rprofile
    Modified:   .gitattributes
    Modified:   .gitignore
    Modified:   DGRP_sexual_conflict.Rproj
    Modified:   README.md
    Modified:   _workflowr.yml
    Modified:   analysis/SNP_correlations.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/license.Rmd
    Modified:   analysis/process_mashr.Rmd
    Modified:   analysis/run_GWAS.Rmd
    Modified:   analysis/run_mashr.Rmd
    Modified:   code/README.md
    Modified:   code/run_mashr.R
    Modified:   data/README.md
    Modified:   data/S_male.csv
    Modified:   data/all.dgrp.phenos_unscaled.csv
    Modified:   data/rfm.complete.csv
    Modified:   output/README.md

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/SNP_heritabilities.Rmd</code>) and HTML (<code>docs/SNP_heritabilities.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/a4076aa5e76c8729c82465a320f59c221ba09a22/analysis/SNP_heritabilities.Rmd" target="_blank">a4076aa</a>
</td>
<td>
ausevo
</td>
<td>
2022-06-01
</td>
<td>
adding analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/55867df25da20bdccbcbd4649acba6149d7eb6d0/docs/SNP_heritabilities.html" target="_blank">55867df</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-16
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/1c86981014c956e141ab746a22e6838bebc494e7/analysis/SNP_heritabilities.Rmd" target="_blank">1c86981</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-16
</td>
<td>
creating a more readable doc
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/5f1d9b90f38cba46e4770b5817116722799d531f/docs/SNP_heritabilities.html" target="_blank">5f1d9b9</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-11
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/e82fc6550fc866889833c780f9b299060b6754c1/analysis/SNP_heritabilities.Rmd" target="_blank">e82fc65</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-11
</td>
<td>
adding exploratory plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/DGRP_sexual_conflict/7bf0ee08d2c8b770d1c2d8aca0b2931897462adb/docs/SNP_heritabilities.html" target="_blank">7bf0ee0</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/DGRP_sexual_conflict/blob/5b9883ecc77ee0ab529aa2bd15efa0caf62b5f2b/analysis/SNP_heritabilities.Rmd" target="_blank">5b9883e</a>
</td>
<td>
ausevo
</td>
<td>
2022-05-10
</td>
<td>
Add GCTA model outputs
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages-and-prepare-for-gcta-analysis" class="section level2">
<h2>Load packages and prepare for <code>GCTA</code> analysis</h2>
<p>Here we load the necessary packages to build this document and code in ‘tidy’ style. We also set the file path so that R can access the <code>gcta64</code> software. Finally, we build a helper function to pass <code>GCTA</code> commands to the terminal. This function allows us to code entirely within R-studio, which is our strong preference.</p>
<pre class="r"><code>library(tidyverse)
library(glue)
library(kableExtra)
library(DT)
library(MetBrewer)
library(patchwork)
library(rcartocolor)

gcta &lt;- file.path(getwd(), &quot;code/gcta64&quot;)
options(readr.show_col_types = FALSE)


# helper function to pass commands to the terminal
# Note that we set `intern = TRUE`, and pass the result of `system()` to `cat()`,
# ensuring that the Terminal output will be printed in this knitr report.
run_command &lt;- function(shell_command, wd = getwd(), path = &quot;&quot;){
  cat(system(glue(&quot;cd &quot;, wd, path, &quot;\n&quot;,shell_command), intern = TRUE), sep = &#39;\n&#39;)
}


### useful code that I didn&#39;t end up needing

  #remove the suffix that denotes the sex the trait was measured in and replace the level &#39;both&#39; with &#39;pooled&#39;
 # mutate(Sex = recode(Sex, Both = &quot;Pooled&quot;),
  #       Trait = case_when(
   #        str_ends(Trait, &quot;.f&quot;) | str_ends(Trait, &quot;.m&quot;)  ~ gsub(&#39;.{2}$&#39;, &#39;&#39;, Trait)),
    #     `Trait 2` = case_when(
     #      str_ends(`Trait 2`, &quot;.f&quot;) | str_ends(`Trait 2`, &quot;.m&quot;)  ~ gsub(&#39;.{2}$&#39;, &#39;&#39;, `Trait 2`))) %&gt;% </code></pre>
</div>
<div id="load-phenotype-data" class="section level2">
<h2>Load phenotype data</h2>
<pre class="r"><code>traits &lt;- read_csv(&quot;data/all.dgrp.phenos_unscaled.csv&quot;) 

traits_for_gwas &lt;- 
  traits %&gt;% 
  # filter(!(`Trait guild` %in% allowed_guilds) | is.na(`Trait guild`)) %&gt;% 
  group_by(Trait) %&gt;% 
  summarise(lines_measured = length(unique(line))) %&gt;% 
  ungroup() %&gt;% 
  filter(lines_measured &gt; 79) %&gt;%
  filter(Trait != &quot;paraquat.resistance.2021.f&quot; &amp; Trait != &quot;starvation.resistance.high.carb.diet.m&quot;) %&gt;% # these traits call a gcta error, we remove for now
  pull(Trait)  # find traits with 80+ replicates
 

traits_for_gwas &lt;- traits_for_gwas[!(grepl(&quot;fitness.late&quot;, traits_for_gwas))]

# replace any slashes in the trait name, e.g. &quot;1/3-Methylhistidine.high.yeast.f&quot;, as this is not ok to use as a file name
traits_for_gwas &lt;- str_replace_all(traits_for_gwas, &quot;[/]&quot;, &quot;_&quot;) 

line_mean_phenotypes &lt;- read_csv(&quot;data/all.dgrp.phenos_unscaled.csv&quot;) %&gt;%
  mutate(Trait = str_replace_all(Trait, &quot;[/]&quot;, &quot;_&quot;)) %&gt;%  # change to match `traits_for_gwas` list
  filter(Trait %in% traits_for_gwas) %&gt;%
   filter(Trait != &quot;paraquat.resistance.2021.f&quot; &amp; Trait != &quot;starvation.resistance.high.carb.diet.m&quot;) %&gt;% # these traits produce a gcta error, we remove for now
  select(line, Trait, trait_value) %&gt;% 
  #distinct(line, Trait, .keep_all = TRUE) %&gt;% # Remove this line once Tom resolves the duplicates
  spread(Trait, trait_value) %&gt;% 
  mutate_at(vars(-line), ~ as.numeric(scale(.x))) %&gt;%  # scale the traits
  mutate(line = paste(&quot;line&quot;, line, sep = &quot;&quot;)) # add &quot;line&quot; in front of the line number to fit with plink/gcta formatting</code></pre>
</div>
<div id="use-gcta-to-estimate-the-genetic-variances-and-heritabilities-for-each-trait" class="section level2">
<h2>Use <code>GCTA</code> to estimate the genetic variances and heritabilities for each trait</h2>
<p>Here we estimate ‘SNP-heritability’ for each phenotype using a genomic relatedness matrix Restricted Maximum Likelihood (GREML) approach. GREML uses a set of SNPs <span class="math inline">\(S\)</span> that are known across a population of individuals (or lines, in our case) to estimate the relatedness between random individuals (lines) and compares that with phenotypic variation across that set of individuals. This allows us to estimate the variation in phenotype that is explained by the known set of SNPs in our population. We can then divide this variance by the total variation found for the phenotype, giving the SNP-heritability. Note that SNP-heritability should be smaller than other estimates of heritability such as broad- or narrow-sense heritability, as it is limited to additive effects from a subset of the genome, while the latter two estimates encompass a broader range of genetic effects.</p>
<p>We use the <code>GCTA</code> software to run this analysis.</p>
<p>A few helpful exerts from the literature about GREML are provided below:</p>
<blockquote>
<p>“Data from a GWAS that are collected to detect statistical associations between SNPs and complex traits are usually analyzed by testing each SNP individually for an association with the trait. To account for the large number of significance tests carried out, a very stringent P value is used. This reduces the occurrence of false positives, but it may cause many real associations to be missed, especially if individual SNPs have a small effect on the trait.”</p>
</blockquote>
<footer>
— Yang et al 2010
</footer>
<blockquote>
<p>“In contrast to single-SNP association analysis, the basic concept behind the <code>GCTA</code> method is to fit the effects of all the SNPs as random effects using a mixed linear model (MLM).”</p>
</blockquote>
<footer>
— Yang et al 2011
</footer>
<div id="make-a-genomic-relatedness-matrix-for-all-the-lines" class="section level3">
<h3>Make a genomic relatedness matrix for all the lines</h3>
<p>This uses the <code>GCTA</code> command <code>--make-grm</code> to make a genomic relatedness matrix (GRM) for the DGRP SNP data. This matrix is needed to run the GREML models below.</p>
<pre class="r"><code>run_command(glue(&quot;{gcta} --bfile dgrp2_QC_all_lines&quot;,
                 &quot; --make-grm&quot;,
                 &quot; --out gcta_GRM&quot;), path = &quot;/gwas_data/derived/&quot;)</code></pre>
</div>
<div id="run-univariate-greml-models-to-find-heritabilites" class="section level3">
<h3>Run univariate GREML models to find heritabilites</h3>
<p>The below uses the <code>GCTA</code> command –reml to estimate the genetic variance (V(G)), environmental variance (V(e)), their sum (Vp), and the so-called “SNP heritability” (V(G)/Vp).</p>
<pre class="r"><code>do_univariate_greml &lt;- function(trait1){

  # First make &#39;focal_data&#39;, a 2-column data frame with the line and the focal phenotype value
  focal_data &lt;- line_mean_phenotypes %&gt;%
    select(line, !! trait1)
  names(focal_data)[2] &lt;- c(&quot;focal_phenotype&quot;)
  
  # Write a file with the line and phenotype data called phenotype.txt, for gcta64
  pheno_data &lt;- focal_data %&gt;%
    # filter(!(line %in% lines_to_prune)) %&gt;% 
    mutate(line_copy = line) %&gt;%
    select(line, line_copy, focal_phenotype) %&gt;%
    as.matrix()
  
  pheno_data %&gt;%
    write.table(row.names = FALSE, col.names = FALSE,
                file = &quot;gwas_data/derived/phenotype.txt&quot;, quote = FALSE)
  
  console_output &lt;- suppressWarnings(capture.output(
    run_command(glue(&quot;{gcta}  --reml --reml-maxit 10000 --grm gcta_GRM  --pheno phenotype.txt  --out delete_me&quot;), 
                path = &quot;/gwas_data/derived/&quot;)))
  
  errors &lt;- console_output[str_detect(console_output, &quot;[Ee]rror&quot;)]
  if(length(errors) == 0) errors &lt;- NA
  if(length(errors) &gt; 1) errors &lt;- paste(errors, sep = &quot;; &quot;)

  rbind(read.delim(&quot;gwas_data/derived/delete_me.hsq&quot;, sep = &quot;\t&quot;) , 
        data.frame(Source = &quot;Errors&quot;, Variance = errors, SE = NA)) %&gt;% 
    mutate(Trait = trait1) %&gt;% select(Trait, everything())
}

# Now run the model for every trait and create a large tibble using map_dfr

if(!file.exists(&quot;gwas_data/variance_components.csv&quot;)){
  variance_components &lt;-
    map_dfr(traits_for_gwas, do_univariate_greml) %&gt;% 
    filter(Source %in% c(&quot;V(G)&quot;, &quot;V(e)&quot;, &quot;Vp&quot;, &quot;V(G)/Vp&quot;)) %&gt;% 
    mutate(Variance = as.numeric(Variance)) %&gt;% 
    rename(Parameter = Source) %&gt;% 
    as_tibble()
  write_csv(variance_components, file = &quot;gwas_data/variance_components.csv&quot;)
} else variance_components &lt;- read_csv(&quot;gwas_data/variance_components.csv&quot;)</code></pre>
<p>Currently two phenotypes call errors when we run the univariate GREML model: <code>paraquat.resistance.2021.f</code> and <code>starvation.resistance.high.carb.diet.m</code>. The error is <em>Error: the information matrix is not invertible.</em></p>
<p>For now we simply don’t run models for these traits, so there are no variance estimates for these traits.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="wrangle-data-and-save" class="section level3">
<h3>Wrangle data and save</h3>
<pre class="r"><code>trait_info &lt;- read_csv(&quot;data/all.dgrp.phenos_unscaled.csv&quot;) %&gt;%
  mutate(Trait = str_replace_all(Trait, &quot;[/]&quot;, &quot;_&quot;),
         Sex = fct_recode(Sex, Pooled = &quot;Both&quot;)) %&gt;% 
  select(-trait_value) %&gt;% 
  distinct(Trait, Sex, `Trait guild`, `Trait description`, Reference)

variance_components &lt;- left_join(variance_components, trait_info)

# wrangle so that each variance component is its own column

variance_components_wide &lt;- 
  variance_components %&gt;% 
  pivot_wider(names_from = Parameter,
              values_from = c(Variance, SE)) %&gt;% 
  mutate(across(where(is.numeric), round, 2)) %&gt;% 
  rename(`V(G)` = `Variance_V(G)`,
         `V(e)` = `Variance_V(e)`,
         `Vp` = `Variance_Vp`,
         `V(G)/Vp` = `Variance_V(G)/Vp`)
  
# Below we remove the suffix that denotes the sex the trait was measured in and replace the level &#39;both&#39; with &#39;pooled&#39;

SNP_heritability &lt;-
  variance_components_wide %&gt;% 
  mutate(Sex = recode(Sex, Both = &quot;Pooled&quot;),
         Trait = case_when(
           str_ends(Trait, &quot;.f&quot;) | str_ends(Trait, &quot;.m&quot;)  ~ gsub(&#39;.{2}$&#39;, &#39;&#39;, Trait))) %&gt;% 
  select(Trait, Sex, everything()) %&gt;%  # reorder the columns for the table %&gt;% 
  arrange(Trait)


write_csv(SNP_heritability, file = &quot;data/SNP_heritability_cleaned.csv&quot;)</code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
</div>
<div id="use-gcta-to-estimate-genetic-covariance-between-traits-and-fitness" class="section level2">
<h2>Use <code>GCTA</code> to estimate genetic covariance between traits and fitness</h2>
<p>Here we use the <code>gcta64</code> command <code>--reml-bivar</code> to estimate the genetic covariance <code>(C(G)_tr12)</code>, i.e. the co-variance explained by the known SNPs in the DGRP, and the genetic correlation (<code>rG</code>), in addition to the univariate parameters for trait 1 and trait 2 mentioned above. Note that for traits where the genetic variance is estimated to be low, the model cannot reliably estimate genetic covariance or correlations.</p>
<p>We use <code>(C(G)_tr12)</code> to calculate the <strong>Robertson covariance</strong>, <span class="math inline">\(R_z = \sigma(A_w, A_z) = \sigma_A(w,z)\)</span>, which is sometimes referred to as the secondary theorem of natural selection. This is the additive covariance between a trait and fitness, or equivalently, the expected response to selection of a trait.</p>
<p>We specify the following options in <code>GCTA</code>:</p>
<p>–reml-bivar-lrt-rg 0: To test for the hypothesis of fixing the genetic correlation at a particular value, e.g. fixing genetic correlation at -1, 0 and 1. By default bivariate GCTA-GREML does not perform a log likelihood test unless this option is specified.</p>
<p>Note that this means the p-values returned in the bivariate GREML output correspond to the genetic correlation <span class="math inline">\(r_G\)</span>. We focus our analysis on<code>(C(G)_tr12)</code>; therefore we do not report these p-values.</p>
<div id="female-fitness" class="section level3">
<h3>Female fitness</h3>
<pre class="r"><code># wrangle data so that it is specific to the sex the trait was measured for i.e. female, male or pooled sexes

female_line_means &lt;- line_mean_phenotypes %&gt;%
  select(line, ends_with(&quot;.f&quot;))

female_trait_list &lt;- line_mean_phenotypes %&gt;% 
  select(ends_with(&quot;.f&quot;)) %&gt;% 
  select(-fitness.early.f) %&gt;% 
  colnames()

# build the function to run the bivariate GREML  

do_bivar_greml_f &lt;- function(trait1, trait2){

  # First make &#39;focal_data&#39;, a 2-column data frame with the line and the focal phenotype value
  focal_data &lt;- female_line_means %&gt;%
    select(line, !! trait1, !! trait2)
  names(focal_data)[2:3] &lt;- c(&quot;focal_pheno1&quot;, &quot;focal_pheno2&quot;)
  
  # Write a file with the line and phenotype data called phenotype.txt, for gcta64
  pheno_data &lt;- focal_data %&gt;%
    # filter(!(line %in% lines_to_prune)) %&gt;% 
    mutate(line_copy = line) %&gt;%
    select(line, line_copy, focal_pheno1, focal_pheno2) %&gt;%
    as.matrix()
  
  pheno_data %&gt;%
    write.table(row.names = FALSE, col.names = FALSE,
                file = &quot;gwas_data/derived/phenotype_pair.txt&quot;, quote = FALSE)
  
  # Note that the warnings come when gcta could not fit the model properly. I save these errors especially, so no need to print them.
  console_output &lt;- suppressWarnings(capture.output(
    run_command(glue(&quot;{gcta}  --reml-bivar --reml-bivar-lrt-rg 0 --reml-maxit 10000 --grm gcta_GRM  --pheno phenotype_pair.txt  --out delete_me&quot;), 
                path = &quot;/gwas_data/derived/&quot;)))
  
  errors &lt;- console_output[str_detect(console_output, &quot;[Ee]rror&quot;)]
  if(length(errors) == 0) errors &lt;- NA
  if(length(errors) &gt; 1) errors &lt;- paste(errors, sep = &quot;; &quot;)
  traits &lt;- sort(c(trait1, trait2))
  
  rbind(read.delim(&quot;gwas_data/derived/delete_me.hsq&quot;, sep = &quot;\t&quot;) , 
        data.frame(Source = &quot;Errors&quot;, Variance = errors, SE = NA)) %&gt;% 
    mutate(`Trait 1` = traits[1], `Trait 2` = traits[2]) %&gt;% 
    select(`Trait 1`, `Trait 2`, everything())
}

# run for all traits - this works, except I need to rewrite the function for each level of sex (done below)

if(!file.exists(&quot;data/SNP_selection.csv&quot;)){
  female_results &lt;- map2_dfr(&quot;fitness.early.f&quot;, female_trait_list, do_bivar_greml_f)
  # unlink deletes files
  unlink(c(&quot;data/derived/delete_me.log&quot;, &quot;data/derived/phenotype_pair.txt&quot;, &quot;data/derived/delete_me.hsq&quot;))
  unlink(list.files(&quot;data/derived/&quot;, pattern = &quot;gcta_GRM&quot;, full.names = TRUE))
  # clean the data up
  female_results_concise &lt;-
    female_results %&gt;% 
    as_tibble() %&gt;% 
    filter(Source %in% c(&quot;V(G)_tr1&quot;, &quot;V(e)_tr1&quot;, &quot;Vp_tr1&quot;, &quot;V(G)/Vp_tr1&quot;,
                         &quot;V(G)_tr2&quot;, &quot;V(e)_tr2&quot;, &quot;Vp_tr2&quot;, &quot;V(G)/Vp_tr2&quot;,
                         &quot;C(G)_tr12&quot;, &quot;rG&quot;)) %&gt;% 
    mutate(Variance = as.numeric(Variance),
           Fitness_sex = &quot;Female&quot;) %&gt;% 
    rename(Parameter = Source)
} else &quot;Already calculated&quot;</code></pre>
</div>
<div id="male-fitness" class="section level3">
<h3>Male fitness</h3>
<pre class="r"><code># wrangle data so that it is specific to the sex the trait was measured for i.e. female, male or pooled sexes

male_line_means &lt;- line_mean_phenotypes %&gt;%
  select(line, ends_with(&quot;.m&quot;))

male_trait_list &lt;- line_mean_phenotypes %&gt;% 
  select(ends_with(&quot;.m&quot;)) %&gt;% 
  select(-fitness.early.m) %&gt;% 
  colnames()

# build the function to run the bivariate GREML  

do_bivar_greml_m &lt;- function(trait1, trait2){

  # First make &#39;focal_data&#39;, a 2-column data frame with the line and the focal phenotype value
  focal_data &lt;- male_line_means %&gt;%
    select(line, !! trait1, !! trait2)
  names(focal_data)[2:3] &lt;- c(&quot;focal_pheno1&quot;, &quot;focal_pheno2&quot;)
  
  # Write a file with the line and phenotype data called phenotype.txt, for gcta64
  pheno_data &lt;- focal_data %&gt;%
    # filter(!(line %in% lines_to_prune)) %&gt;% 
    mutate(line_copy = line) %&gt;%
    select(line, line_copy, focal_pheno1, focal_pheno2) %&gt;%
    as.matrix()
  
  pheno_data %&gt;%
    write.table(row.names = FALSE, col.names = FALSE,
                file = &quot;gwas_data/derived/phenotype_pair.txt&quot;, quote = FALSE)
  
  # Note that the warnings come when gcta could not fit the model properly. I save these errors especially, so no need to print them.
  console_output &lt;- suppressWarnings(capture.output(
    run_command(glue(&quot;{gcta}  --reml-bivar --reml-bivar-lrt-rg 0 --reml-maxit 10000 --grm gcta_GRM  --pheno phenotype_pair.txt  --out delete_me&quot;), 
                path = &quot;/gwas_data/derived/&quot;)))
  
  errors &lt;- console_output[str_detect(console_output, &quot;[Ee]rror&quot;)]
  if(length(errors) == 0) errors &lt;- NA
  if(length(errors) &gt; 1) errors &lt;- paste(errors, sep = &quot;; &quot;)
  traits &lt;- sort(c(trait1, trait2))
  
  rbind(read.delim(&quot;gwas_data/derived/delete_me.hsq&quot;, sep = &quot;\t&quot;) , 
        data.frame(Source = &quot;Errors&quot;, Variance = errors, SE = NA)) %&gt;% 
    mutate(`Trait 1` = traits[1], `Trait 2` = traits[2]) %&gt;% 
    select(`Trait 1`, `Trait 2`, everything())
}

# run for all traits 

if(!file.exists(&quot;data/SNP_selection.csv&quot;)){
  male_results &lt;- map2_dfr(&quot;fitness.early.m&quot;, male_trait_list, do_bivar_greml_m)
  # unlink deletes files
  unlink(c(&quot;data/derived/delete_me.log&quot;, &quot;data/derived/phenotype_pair.txt&quot;, &quot;data/derived/delete_me.hsq&quot;))
  unlink(list.files(&quot;data/derived/&quot;, pattern = &quot;gcta_GRM&quot;, full.names = TRUE))
  # clean the data up
  male_results_concise &lt;-
    male_results %&gt;% 
    as_tibble() %&gt;% 
    filter(Source %in% c(&quot;V(G)_tr1&quot;, &quot;V(e)_tr1&quot;, &quot;Vp_tr1&quot;, &quot;V(G)/Vp_tr1&quot;,
                         &quot;V(G)_tr2&quot;, &quot;V(e)_tr2&quot;, &quot;Vp_tr2&quot;, &quot;V(G)/Vp_tr2&quot;,
                         &quot;C(G)_tr12&quot;, &quot;rG&quot;)) %&gt;% 
    mutate(Variance = as.numeric(Variance),
           Fitness_sex = &quot;Male&quot;) %&gt;% 
    rename(Parameter = Source)
} else &quot;Already calculated&quot;</code></pre>
</div>
<div id="data-where-the-sexes-were-pooled" class="section level3">
<h3>Data where the sexes were pooled</h3>
<pre class="r"><code># wrangle data so that it is specific to the sex the trait was measured for i.e. female, male or pooled sexes

pooled_line_means &lt;- line_mean_phenotypes %&gt;%
  select(line, fitness.early.f, fitness.early.m, !ends_with(c(&quot;.f&quot;, &quot;.m&quot;)))

pooled_trait_list &lt;- line_mean_phenotypes %&gt;% 
  select(!ends_with(c(&quot;.f&quot;, &quot;.m&quot;))) %&gt;%  
  select(-line) %&gt;% 
  colnames()

# build the function to run the bivariate GREML  

do_bivar_greml_p &lt;- function(trait1, trait2){

  # First make &#39;focal_data&#39;, a 2-column data frame with the line and the focal phenotype value
  focal_data &lt;- pooled_line_means %&gt;%
    select(line, !! trait1, !! trait2)
  names(focal_data)[2:3] &lt;- c(&quot;focal_pheno1&quot;, &quot;focal_pheno2&quot;)
  
  # Write a file with the line and phenotype data called phenotype.txt, for gcta64
  pheno_data &lt;- focal_data %&gt;%
    # filter(!(line %in% lines_to_prune)) %&gt;% 
    mutate(line_copy = line) %&gt;%
    select(line, line_copy, focal_pheno1, focal_pheno2) %&gt;%
    as.matrix()
  
  pheno_data %&gt;%
    write.table(row.names = FALSE, col.names = FALSE,
                file = &quot;gwas_data/derived/phenotype_pair.txt&quot;, quote = FALSE)
  
  # Note that the warnings come when gcta could not fit the model properly. I save these errors especially, so no need to print them.
  console_output &lt;- suppressWarnings(capture.output(
    run_command(glue(&quot;{gcta}  --reml-bivar --reml-bivar-lrt-rg 0 --reml-maxit 10000 --grm gcta_GRM  --pheno phenotype_pair.txt  --out delete_me&quot;), 
                path = &quot;/gwas_data/derived/&quot;)))
  
  errors &lt;- console_output[str_detect(console_output, &quot;[Ee]rror&quot;)]
  if(length(errors) == 0) errors &lt;- NA
  if(length(errors) &gt; 1) errors &lt;- paste(errors, sep = &quot;; &quot;)
  traits &lt;- sort(c(trait1, trait2))
  
  rbind(read.delim(&quot;gwas_data/derived/delete_me.hsq&quot;, sep = &quot;\t&quot;) , 
        data.frame(Source = &quot;Errors&quot;, Variance = errors, SE = NA)) %&gt;% 
    mutate(`Trait 1` = traits[1], `Trait 2` = traits[2]) %&gt;% 
    select(`Trait 1`, `Trait 2`, everything())
}

# run for all traits

if(!file.exists(&quot;data/SNP_selection.csv&quot;)){
  pooled_female &lt;- map2_dfr(&quot;fitness.early.f&quot;, pooled_trait_list, do_bivar_greml_p)
  # unlink deletes files
  unlink(c(&quot;data/derived/delete_me.log&quot;, &quot;data/derived/phenotype_pair.txt&quot;, &quot;data/derived/delete_me.hsq&quot;))
  unlink(list.files(&quot;data/derived/&quot;, pattern = &quot;gcta_GRM&quot;, full.names = TRUE))
  pooled_male &lt;- map2_dfr(&quot;fitness.early.m&quot;, pooled_trait_list, do_bivar_greml_p)
  # unlink deletes files
  unlink(c(&quot;data/derived/delete_me.log&quot;, &quot;data/derived/phenotype_pair.txt&quot;, &quot;data/derived/delete_me.hsq&quot;))
  unlink(list.files(&quot;data/derived/&quot;, pattern = &quot;gcta_GRM&quot;, full.names = TRUE))
  # clean the data up
  pooled_results &lt;- rbind(pooled_female, pooled_male)
  
  pooled_results_concise &lt;-
    pooled_results %&gt;% 
    as_tibble() %&gt;% 
    filter(Source %in% c(&quot;V(G)_tr1&quot;, &quot;V(e)_tr1&quot;, &quot;Vp_tr1&quot;, &quot;V(G)/Vp_tr1&quot;,
                         &quot;V(G)_tr2&quot;, &quot;V(e)_tr2&quot;, &quot;Vp_tr2&quot;, &quot;V(G)/Vp_tr2&quot;,
                         &quot;C(G)_tr12&quot;, &quot;rG&quot;)) %&gt;% 
    mutate(Variance = as.numeric(Variance)) %&gt;% 
    rename(Parameter = Source)
  
    a_pooled &lt;- 
    pooled_results_concise %&gt;% 
    filter(`Trait 1` == &quot;fitness.early.f&quot; | `Trait 1` == &quot;fitness.early.m&quot;) %&gt;% 
    #rename(Trait = `Trait 1`) %&gt;% 
    pivot_wider(names_from = Parameter, values_from = c(Variance, SE)) %&gt;% 
    rename(# rename to more specific names. No name switching needed for this subset
      `Variance_V(G)_trait` = `Variance_V(G)_tr2`,
      `Variance_V(G)_fitness` = `Variance_V(G)_tr1`,
      `Variance_V(e)_trait` = `Variance_V(e)_tr2`,
      `Variance_V(e)_fitness` = `Variance_V(e)_tr1`,
      Variance_Vp_trait = Variance_Vp_tr2,
      Variance_Vp_fitness = Variance_Vp_tr1,
      `Variance_V(G)/Vp_trait` = `Variance_V(G)/Vp_tr2`,
      `Variance_V(G)/Vp_fitness` = `Variance_V(G)/Vp_tr1`,
      `SE_V(G)_trait` = `SE_V(G)_tr2`,
      `SE_V(G)_fitness` = `SE_V(G)_tr1`,
      `SE_V(e)_trait` = `SE_V(e)_tr2`,
      `SE_V(e)_fitness` = `SE_V(e)_tr1`,
      SE_Vp_trait = SE_Vp_tr2,
      SE_Vp_fitness = SE_Vp_tr1,
      `SE_V(G)/Vp_trait` = `SE_V(G)/Vp_tr2`,
      `SE_V(G)/Vp_fitness` = `SE_V(G)/Vp_tr1`)
  
  b_pooled &lt;-
    pooled_results_concise %&gt;% 
    filter(`Trait 2` == &quot;fitness.early.f&quot; | `Trait 2` == &quot;fitness.early.m&quot;) %&gt;% 
    rename(Trait = `Trait 1`,
           `Trait 1` = `Trait 2`) %&gt;% 
    rename(`Trait 2` = Trait) %&gt;% 
    pivot_wider(names_from = Parameter, values_from = c(Variance, SE)) %&gt;% 
    rename(# switch the names so that they match the order of trait1 and trait2
      `Variance_V(G)_trait` = `Variance_V(G)_tr1`,
      `Variance_V(G)_fitness` = `Variance_V(G)_tr2`,
      `Variance_V(e)_trait` = `Variance_V(e)_tr1`,
      `Variance_V(e)_fitness` = `Variance_V(e)_tr2`,
      Variance_Vp_trait = Variance_Vp_tr1,
      Variance_Vp_fitness = Variance_Vp_tr2,
      `Variance_V(G)/Vp_trait` = `Variance_V(G)/Vp_tr1`,
      `Variance_V(G)/Vp_fitness` = `Variance_V(G)/Vp_tr2`,
      `SE_V(G)_trait` = `SE_V(G)_tr1`,
      `SE_V(G)_fitness` = `SE_V(G)_tr2`,
      `SE_V(e)_trait` = `SE_V(e)_tr1`,
      `SE_V(e)_fitness` = `SE_V(e)_tr2`,
      SE_Vp_trait = SE_Vp_tr1,
      SE_Vp_fitness = SE_Vp_tr2,
      `SE_V(G)/Vp_trait` = `SE_V(G)/Vp_tr1`,
      `SE_V(G)/Vp_fitness` = `SE_V(G)/Vp_tr2`)
  
  
  
  pooled_results_concise &lt;-
    rbind(a_pooled, b_pooled) %&gt;% 
    rename(Fitness_component = `Trait 1`,
           Trait = `Trait 2`) %&gt;% 
    mutate(Fitness_sex = if_else(Fitness_component == &quot;fitness.early.f&quot;, &quot;Female&quot;, &quot;Male&quot;)) %&gt;% 
    select(Fitness_component, Trait, Fitness_sex, everything()) %&gt;% arrange(Trait)
  
  # join the  meta-data
  
   pooled_results_concise &lt;-
     left_join(pooled_results_concise, trait_info)
  
} else &quot;Already calculated&quot;</code></pre>
</div>
<div id="combine-results-and-save" class="section level3">
<h3>Combine results and save</h3>
<p>Note that the <code>SNP_heritability</code> tibble and the <code>SNP_selection</code> tibble contain duplicate columns. This is because the bivariate GREML calculates everything that the univariate GREML does. I’ll check if they are identical and delete redundant columns if so. I get the feeling that they’ll be slightly different because the bivariate GREML only uses the lines that have been measured for both traits; generally a smaller subset of those a trait was measured across.</p>
<pre class="r"><code>if(!file.exists(&quot;data/SNP_selection.csv&quot;)){
  female_male_data &lt;-
    rbind(female_results_concise, male_results_concise) %&gt;% # note that the pooled data is binded later
    # create a common name so that we can join the meta data
    rename(Trait = `Trait 1`)
  
  # The code below is written to make sure fitness is in the same trait column throughout the tibble. We need to do this because Trait 1 is determined alphabetically, so it&#39;s sometimes fitness and sometimes is not.
  
  # wrangle so that each variance component is its own column
  
  a &lt;- 
    female_male_data %&gt;% 
    filter(Trait == &quot;fitness.early.f&quot; | Trait == &quot;fitness.early.m&quot;) %&gt;% 
    rename(`Trait 1` = Trait) %&gt;% 
    pivot_wider(names_from = Parameter, values_from = c(Variance, SE)) %&gt;% 
    rename(# rename to more specific names. No name switching needed for this subset
      `Variance_V(G)_trait` = `Variance_V(G)_tr2`,
      `Variance_V(G)_fitness` = `Variance_V(G)_tr1`,
      `Variance_V(e)_trait` = `Variance_V(e)_tr2`,
      `Variance_V(e)_fitness` = `Variance_V(e)_tr1`,
      Variance_Vp_trait = Variance_Vp_tr2,
      Variance_Vp_fitness = Variance_Vp_tr1,
      `Variance_V(G)/Vp_trait` = `Variance_V(G)/Vp_tr2`,
      `Variance_V(G)/Vp_fitness` = `Variance_V(G)/Vp_tr1`,
      `SE_V(G)_trait` = `SE_V(G)_tr2`,
      `SE_V(G)_fitness` = `SE_V(G)_tr1`,
      `SE_V(e)_trait` = `SE_V(e)_tr2`,
      `SE_V(e)_fitness` = `SE_V(e)_tr1`,
      SE_Vp_trait = SE_Vp_tr2,
      SE_Vp_fitness = SE_Vp_tr1,
      `SE_V(G)/Vp_trait` = `SE_V(G)/Vp_tr2`,
      `SE_V(G)/Vp_fitness` = `SE_V(G)/Vp_tr1`)
  
  b &lt;-
    female_male_data %&gt;% 
    filter(`Trait 2` == &quot;fitness.early.f&quot; | `Trait 2` == &quot;fitness.early.m&quot;) %&gt;% 
    rename(`Trait 1` = `Trait 2`,
           `Trait 2` = Trait) %&gt;% 
    pivot_wider(names_from = Parameter, values_from = c(Variance, SE)) %&gt;% 
    rename(# switch the names so that they match the order of trait1 and trait2
      `Variance_V(G)_trait` = `Variance_V(G)_tr1`,
      `Variance_V(G)_fitness` = `Variance_V(G)_tr2`,
      `Variance_V(e)_trait` = `Variance_V(e)_tr1`,
      `Variance_V(e)_fitness` = `Variance_V(e)_tr2`,
      Variance_Vp_trait = Variance_Vp_tr1,
      Variance_Vp_fitness = Variance_Vp_tr2,
      `Variance_V(G)/Vp_trait` = `Variance_V(G)/Vp_tr1`,
      `Variance_V(G)/Vp_fitness` = `Variance_V(G)/Vp_tr2`,
      `SE_V(G)_trait` = `SE_V(G)_tr1`,
      `SE_V(G)_fitness` = `SE_V(G)_tr2`,
      `SE_V(e)_trait` = `SE_V(e)_tr1`,
      `SE_V(e)_fitness` = `SE_V(e)_tr2`,
      SE_Vp_trait = SE_Vp_tr1,
      SE_Vp_fitness = SE_Vp_tr2,
      `SE_V(G)/Vp_trait` = `SE_V(G)/Vp_tr1`,
      `SE_V(G)/Vp_fitness` = `SE_V(G)/Vp_tr2`)
  
  female_male_wide &lt;-
    rbind(a, b) %&gt;% 
    rename(Fitness_component = `Trait 1`,
           Trait = `Trait 2`)
  
  # join meta data
  
  female_male_wide &lt;-
  left_join(female_male_wide, trait_info) %&gt;% 
    # this next line removes the .f and .m suffix&#39;s for each trait, as this info is held in the `Fitness_sex` column
    mutate(Trait = case_when(str_ends(Trait, &quot;.f&quot;) | str_ends(Trait, &quot;.m&quot;)  ~ gsub(&#39;.{2}$&#39;, &#39;&#39;, Trait)))
  
  # combine the pooled dataset with the female and male data
  
  SNP_selection_response_estimates &lt;- rbind(pooled_results_concise, female_male_wide) %&gt;% 
    select(Trait, Fitness_component, Fitness_sex, Sex, `Trait guild`, `Trait description`, Reference, everything()) %&gt;% 
    rename(Trait_sex = Sex)
  
      # Trait_sex shows which sex the trait was measured in, while Fitness_sex shows the sex that `(C(G)_tr12)` and `rG` were calculated for
    # These are generally the same, with the exception of traits that were measured with the sexes pooled (commonly larval traits), coded as pooled in the trait_sex column
    # For these pooled traits we ran the bivariate GREML twice - once for female fitness and again for male fitness
  
  write_csv(SNP_selection_response_estimates, file = &quot;data/SNP_selection.csv&quot;)
  
} else SNP_selection_response_estimates &lt;- read_csv(&quot;data/SNP_selection.csv&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
