---
title: "'Variance components and the response to selection"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE, cache = FALSE)
```

## Load packages and results from the `GCTA` models

```{r}

library(tidyverse)
library(glue)
library(kableExtra)
library(DT)
library(MetBrewer)
library(patchwork)
library(rcartocolor)
library(ggbeeswarm)
library(brms)
library(tidybayes)

 SNP_heritability <- read_csv("data/SNP_heritability.csv")
 SNP_selection_response_estimates <- read_csv("data/SNP_selection.csv")
```


## Which traits are responding to selection?

```{r, fig.height= 25, fig.width= 12}

selection_female_plot <-
  SNP_selection_response_estimates %>% 
  filter(Fitness_sex == "Female",
         `Variance_C(G)_tr12` < 1 & `Variance_C(G)_tr12` > -1,
         Reference != "Jin et al (2020) PLOS Genetics" &
           Reference != "Everett et al (2020) Genome Research" &
           Reference != "Dembeck et al (2015) eLife") %>% 
  mutate(Trait = fct_reorder(Trait, `Variance_C(G)_tr12`)) %>% 
  
  ggplot(aes(x = `Variance_C(G)_tr12`, y = Trait, xmin = `Variance_C(G)_tr12` - `SE_C(G)_tr12`, xmax = `Variance_C(G)_tr12` + `SE_C(G)_tr12`)) +
  geom_pointinterval(aes(fill = `Variance_C(G)_tr12`), shape = 21, point_size = 1.5, stroke = 1) +
  scale_fill_gradientn(colors=met.brewer(direction = -1, "Hiroshige"), guide = "none") +
  geom_vline(xintercept = 0, linetype = 2) +
  labs(x = expression(paste("SNP covariance between fitness and trait ", (R[z]))),
       title = "Females") +
  theme_minimal()
  

selection_male_plot <-
  SNP_selection_response_estimates %>% 
  filter(Fitness_sex == "Male",
         `Variance_C(G)_tr12` < 1 & `Variance_C(G)_tr12` > -1,
         Reference != "Jin et al (2020) PLOS Genetics" &
           Reference != "Everett et al (2020) Genome Research" &
           Reference != "Dembeck et al (2015) eLife") %>% 
  mutate(Trait = fct_reorder(Trait, `Variance_C(G)_tr12`)) %>% 
  
  ggplot(aes(x = `Variance_C(G)_tr12`, y = Trait, xmin = `Variance_C(G)_tr12` - `SE_C(G)_tr12`, xmax = `Variance_C(G)_tr12` + `SE_C(G)_tr12`)) +
  geom_pointinterval(aes(fill = `Variance_C(G)_tr12`), shape = 21, point_size = 1.5, stroke = 1) +
  scale_fill_gradientn(colors=met.brewer(direction = -1, "Hiroshige"), guide = "none") +
  geom_vline(xintercept = 0, linetype = 2) +
  labs(x = expression(paste("SNP covariance between fitness and trait ", (R[z]))),
       title = "Males") +
  theme_minimal()

selection_female_plot + selection_male_plot

```

Figure X: Results from the bivariate GREML models run in GCTA. We use these models to estimate the genetic covariance between a given trait and fitness, which is the response to selection. Data are mean point estimates of covariance with standard errors. Note that 1) we have not corrected for multiple testing and 2) I don't know what the units are here / how to interpret them as a selection response and 3) only a subset of the traits analysed are displayed in this plot

## Is the response to selection greater in males than it is in females across traits?

### Meta-analysis

- Should I only consider traits measured in both sexes?

- We need to find an appropriate distribution to model these data

```{r}
met_selection_data <-
  SNP_selection_response_estimates %>% 
  select(Trait, Fitness_sex, `Trait guild`, Reference, `Variance_C(G)_tr12`, `SE_C(G)_tr12`) %>% 
  transmute(Trait = as.factor(Trait),
            Sex = as.factor(Fitness_sex),
            Trait_guild = as.factor(`Trait guild`),
            Reference = as.factor(Reference),
            Selection_response = abs(`Variance_C(G)_tr12`),
            SE = `SE_C(G)_tr12`) %>% 
  filter(SE != 0, Selection_response < 1) # note this filtering removes 8 traits


Selection_model <- 
  brm(Selection_response | se(SE) ~ 0 + Sex + (1|Trait) + (1|Reference),
      family = gaussian(), # this distribution is not appropriate
      data = met_selection_data,
      prior = c(prior(normal(0, 0.2), class = b),
                prior(cauchy(0, 1), class = sd)),
      warmup = 4000, iter = 6000,
      seed = 1, cores = 4, chains = 4,
      control = list(adapt_delta = 0.9),
      file = "fits/Selection.model")

print(Selection_model)
```

Above I mention that a Gaussian model may not be a good fit for our data. Letâ€™s see about that:

```{r}
pp_check(Selection_model)
```

Yep, not great! Need to think about this.

$~$

## Which traits exhibit genetic variance explained by known SNPs?

```{r, fig.height= 25, fig.width=12}

Vg_female_plot <-
  SNP_selection_response_estimates %>% 
  filter(Fitness_sex == "Female",
         `Variance_V(G)_trait` < 1 & `Variance_V(G)_trait` > -1,
         Reference != "Jin et al (2020) PLOS Genetics" &
           Reference != "Everett et al (2020) Genome Research" &
           Reference != "Dembeck et al (2015) eLife") %>% 
  mutate(Trait = fct_reorder(Trait, `Variance_V(G)_trait`)) %>% 
  
  ggplot(aes(x = `Variance_V(G)_trait`, y = Trait, xmin = `Variance_V(G)_trait` - `SE_V(G)_trait`, xmax = `Variance_V(G)_trait` + `SE_V(G)_trait`)) +
  geom_pointinterval(aes(fill = `Variance_V(G)_trait`), shape = 21, point_size = 1.5, stroke = 1) +
  scale_fill_gradientn(colors=met.brewer(direction = 1, "OKeeffe2"), guide = "none") +
  geom_vline(xintercept = 0, linetype = 2) +
  labs(x = expression(paste("SNP genetic variance", (V[A]))),
       title = "Females") +
  theme_minimal()
  

Vg_male_plot <-
  SNP_selection_response_estimates %>% 
  filter(Fitness_sex == "Male",
         `Variance_V(G)_trait` < 1 & `Variance_V(G)_trait` > -1,
         Reference != "Jin et al (2020) PLOS Genetics" &
           Reference != "Everett et al (2020) Genome Research" &
           Reference != "Dembeck et al (2015) eLife") %>% 
  mutate(Trait = fct_reorder(Trait, `Variance_V(G)_trait`)) %>% 
  
  ggplot(aes(x = `Variance_V(G)_trait`, y = Trait, xmin = `Variance_V(G)_trait` - `SE_V(G)_trait`, xmax = `Variance_V(G)_trait` + `SE_V(G)_trait`)) +
  geom_pointinterval(aes(fill = `Variance_V(G)_trait`), shape = 21, point_size = 1.5, stroke = 1) +
  scale_fill_gradientn(colors=met.brewer(direction = 1, "OKeeffe2"), guide = "none") +
  geom_vline(xintercept = 0, linetype = 2) +
  labs(x = expression(paste("SNP genetic variance", (V[A]))),
       title = "Males") +
  theme_minimal()

Vg_female_plot + Vg_male_plot

```

## Does additive genetic variance differ between the sexes?

### Exploratory plots

This plot shows the distribution of genetic and environmental variance in females and males. It is simply exploratory, as we do not account for the uncertainty around the point estimate for each trait here.

```{r}

# Some wrangling required

a <- SNP_selection_response_estimates %>% 
  filter(Fitness_sex == "Male")
 

b <- SNP_selection_response_estimates %>% 
  filter(Fitness_sex == "Female")

# there are 304 traits for which we have measures in both sexes

SNP_both_sexes <- inner_join(a, b, by = "Trait")

add_var_data <-  
  SNP_both_sexes %>% 
  pivot_longer(cols = c(`Variance_V(G)_trait.x`, `Variance_V(G)_trait.y`), names_to = "Sex", values_to = "Additive_genetic_variance") %>% 
  mutate(Sex = if_else(str_ends(Sex, ".x"), "Male", "Female")) %>% 
  select(c(Trait, Sex, `Trait guild.x`, Additive_genetic_variance)) %>% 
  filter(Additive_genetic_variance < 1) 

env_var_data <-
  SNP_both_sexes %>% 
  pivot_longer(cols = c(`Variance_V(e)_trait.x`, `Variance_V(e)_trait.y`), names_to = "Sex", values_to = "Env_variance") %>% 
  mutate(Sex = if_else(str_ends(Sex, ".x"), "Male", "Female")) %>% 
  select(c(Trait, Sex, `Trait guild.x`, Env_variance)) 
  #filter(Env_variance > -1 & Env_variance < 1) %>%


```


```{r}

# plotting

p1 <-
  add_var_data %>%
  ggplot(aes(x = Additive_genetic_variance)) + 
  #geom_histogram(fill = met.brewer("Hokusai2")[2]) +
  stat_halfeye(fill = met.brewer("Hokusai2")[1]) +
  #geom_vline(xintercept = Add_Median[1], linetype = 2) +
  labs(y = "Number of traits",
       x = "V(G) - the estimated additive genetic variance") +
  theme_minimal() +
  facet_wrap(~Sex)

p2 <-
  env_var_data %>% 
  ggplot(aes(x = Env_variance)) + 
  #geom_histogram(fill = met.brewer("Hokusai2")[4]) +
  stat_halfeye(fill = met.brewer("Hokusai2")[1]) +
  #geom_vline(xintercept = median(env_var_data$Env_variance), linetype = 2) +
  labs(y = "Number of traits",
       x = "V(e) - the estimated environmental variance") +
  theme_minimal() +
  facet_wrap(~Sex)


p1 / p2

```

From our `GCTA` analysis, we find that across traits measured in both sexes, environmental variance is ~three-fold larger than additive genetic variance explained by known SNPs. We also find that V(G) is highly zero-inflated (or 0.0001 inflated...). Whether this is a true effect or a result of an inability to estimate V(G) is **unknown**.

The plots suggest that males might have slightly higher V(G) than females, but that females have higher V(e).

$~$

### Meta-analysis

To test whether V(G) is higher in males in females and that V(e) is higher in females than males, we conduct meta-analysis across all traits measured in both sexes. 

We remove traits for which the Standard error is calculated as 0, as well as traits which have genetic variances greater than 1.

Note that I currently specify a Gaussian error distribution. **This fits the data poorly.**

```{r}
met_additive_data <-
  SNP_both_sexes %>% 
  select(Trait, `Trait guild.x`, Reference.x, `Variance_V(G)_trait.x`, `Variance_V(G)_trait.y`, `SE_V(G)_trait.x`, `SE_V(G)_trait.y`) %>% 
  pivot_longer(cols = c(`Variance_V(G)_trait.x`, `Variance_V(G)_trait.y`, ), names_to = "Sex", values_to = "Additive_genetic_variance") %>% 
  mutate(Sex = if_else(str_ends(Sex, ".x"), "Male", "Female"),
         SE = if_else(Sex == "Male", `SE_V(G)_trait.x`, `SE_V(G)_trait.y`),
         Trait = as.factor(Trait),
         Trait_guild = as.factor(`Trait guild.x`),
         Reference = as.factor(Reference.x),
         Sex = as.factor(Sex)) %>% 
  select(-c(`SE_V(G)_trait.x`, `SE_V(G)_trait.y`, `Trait guild.x`, Reference.x)) %>% 
  filter(SE != 0, Additive_genetic_variance < 1) # note this filtering removes 5 traits


VG_model <- 
  brm(Additive_genetic_variance | se(SE) ~ 0 + Sex + (1|Trait) + (1|Reference),
      family = gaussian(), 
      data = met_additive_data,
      prior = c(prior(normal(0, 0.2), class = b),
                prior(cauchy(0, 1), class = sd)),
      warmup = 4000, iter = 6000,
      seed = 1, cores = 4, chains = 4,
      control = list(adapt_delta = 0.9),
      file = "fits/VG.model")

# note that sigma is not estimated. This is normal for meta-analysis

print(VG_model)

```


Once again, a Gaussian model may not be a good fit for our data. Let's check:

```{r}
pp_check(VG_model)
```

Yep, not great gain! Need to think about this.

Anyway, in the meantime, here's the predictions from the model plotted as halfeyes:

```{r}

Genetic_var_sex_plot <-
  as_draws_df(VG_model) %>% 
  select(b_SexFemale, b_SexMale) %>% 
  rename(Female = b_SexFemale, Male = b_SexMale) %>% 
  pivot_longer(cols = 1:2, names_to = "Sex", values_to = "Genetic_variance") %>% 
  ggplot(aes(x = Genetic_variance, y = Sex)) +
  stat_halfeye(point_interval = median_qi, .width = c(.5, .95),
               slab_colour = "black", shape = 21, point_size = 3, 
               stroke = 1.5, point_fill = "white") +
  labs(x = "Mean genetic variance",
       y = "Sex") +
  theme_bw()


Genetic_var_diff_plot <-
  as_draws_df(VG_model) %>% 
  mutate(Sex_diff = b_SexFemale - b_SexMale) %>% 
  select(b_SexFemale, b_SexMale, Sex_diff) %>% 
  ggplot(aes(x = Sex_diff, y = 0, fill = stat(x > 0))) +
  stat_halfeye(point_interval = median_qi, .width = c(.5, .95),
               slab_colour = "black", shape = 21, point_size = 3, 
               stroke = 1.5, point_fill = "white") +
  geom_vline(xintercept = 0, linetype = 2) +
  labs(x = "Difference in genetic variance between traits\n expressed in females and males",
       y = NULL) +
  scale_y_continuous(NULL, breaks = NULL) +
  scale_fill_manual(values = c("grey", met.brewer("Hokusai2")[2]), guide = "none") +
  theme_bw()

Genetic_var_sex_plot + Genetic_var_diff_plot
```

$~$

## Which traits exhibit environmental variance not explained by known SNPs?

What I call environmental variance is really all residual variance that is not explained by the known set of SNPs in the DGRP. This includes all environmental and unmapped genetic explanations of variance e.g. unknown SNPs, regulatory effects, mtDNA and the genetic material of cytoplasmic endosympbionts (in some cases).

In theory this residual variance should be larger than the environmental variance used to find broad-sense heritability. 

```{r, fig.height= 25, fig.width=12}

Ve_female_plot <-
  SNP_selection_response_estimates %>% 
  filter(Fitness_sex == "Female",
         `Variance_V(e)_trait` < 5 & `Variance_V(e)_trait` > -1,
         Reference != "Jin et al (2020) PLOS Genetics" &
           Reference != "Everett et al (2020) Genome Research" &
           Reference != "Dembeck et al (2015) eLife") %>% 
  mutate(Trait = fct_reorder(Trait, `Variance_V(e)_trait`)) %>% 
  
  ggplot(aes(x = `Variance_V(e)_trait`, y = Trait, xmin = `Variance_V(e)_trait` - `SE_V(e)_trait`, xmax = `Variance_V(e)_trait` + `SE_V(e)_trait`)) +
  geom_pointinterval(aes(fill = `Variance_V(e)_trait`), shape = 21, point_size = 1.5, stroke = 1) +
  scale_fill_gradientn(colors=met.brewer(direction = 1, "OKeeffe2"), guide = "none") +
  geom_vline(xintercept = 0, linetype = 2) +
  labs(x = expression(paste("Environmental variance", (V[A]))),
       title = "Females") +
  theme_minimal()
  

Ve_male_plot <-
  SNP_selection_response_estimates %>% 
  filter(Fitness_sex == "Male",
         `Variance_V(e)_trait` < 5 & `Variance_V(e)_trait` > -1,
         Reference != "Jin et al (2020) PLOS Genetics" &
           Reference != "Everett et al (2020) Genome Research" &
           Reference != "Dembeck et al (2015) eLife") %>% 
  mutate(Trait = fct_reorder(Trait, `Variance_V(e)_trait`)) %>% 
  
  ggplot(aes(x = `Variance_V(e)_trait`, y = Trait, xmin = `Variance_V(e)_trait` - `SE_V(e)_trait`, xmax = `Variance_V(e)_trait` + `SE_V(e)_trait`)) +
  geom_pointinterval(aes(fill = `Variance_V(e)_trait`), shape = 21, point_size = 1.5, stroke = 1) +
  scale_fill_gradientn(colors=met.brewer(direction = 1, "OKeeffe2"), guide = "none") +
  geom_vline(xintercept = 0, linetype = 2) +
  labs(x = expression(paste("Environmental variance", (V[A]))),
       title = "Males") +
  theme_minimal()

Ve_female_plot + Ve_male_plot

```

## Does enviornmental variation differ between the sexes?

### Meta-analysis

```{r}
met_enviro_data <-
  SNP_both_sexes %>% 
  select(Trait, `Trait guild.x`, Reference.x, `Variance_V(e)_trait.x`, `Variance_V(e)_trait.y`, `SE_V(e)_trait.x`, `SE_V(e)_trait.y`) %>% 
  pivot_longer(cols = c(`Variance_V(e)_trait.x`, `Variance_V(e)_trait.y`, ), names_to = "Sex", values_to = "Environmental_variance") %>% 
  mutate(Sex = if_else(str_ends(Sex, ".x"), "Male", "Female"),
         SE = if_else(Sex == "Male", `SE_V(e)_trait.x`, `SE_V(e)_trait.y`),
         Trait = as.factor(Trait),
         Trait_guild = as.factor(`Trait guild.x`),
         Reference = as.factor(Reference.x),
         Sex = as.factor(Sex)) %>% 
  select(-c(`SE_V(e)_trait.x`, `SE_V(e)_trait.y`, `Trait guild.x`, Reference.x)) %>% 
  filter(SE != 0)


Ve_model <- 
  brm(Environmental_variance | se(SE) ~ 0 + Sex + (1|Trait) + (1|Reference),
      family = gaussian(), 
      data = met_enviro_data,
      prior = c(prior(normal(0, 0.2), class = b),
                prior(cauchy(0, 1), class = sd)),
      warmup = 4000, iter = 6000,
      seed = 1, cores = 4, chains = 4,
      control = list(adapt_delta = 0.9),
      file = "fits/Ve.model")

print(Ve_model)
```

Check the fit

```{r}
pp_check(Ve_model)
```

As with all the meta-models, we need to find a distribution that precludes negative values for variances.

Plot the predictions:

```{r}
Env_var_sex_plot <-
  as_draws_df(Ve_model) %>% 
  select(b_SexFemale, b_SexMale) %>% 
  rename(Female = b_SexFemale, Male = b_SexMale) %>% 
  pivot_longer(cols = 1:2, names_to = "Sex", values_to = "Environmental_variance") %>% 
  ggplot(aes(x = Environmental_variance, y = Sex)) +
  stat_halfeye(point_interval = median_qi, .width = c(.5, .95),
               slab_colour = "black", shape = 21, point_size = 3, 
               stroke = 1.5, point_fill = "white") +
  labs(x = "Mean environmental (residual) variance",
       y = "Sex") +
  theme_bw()




Env_var_diff_plot <-
  as_draws_df(Ve_model) %>% 
  mutate(Sex_diff = b_SexFemale - b_SexMale) %>% 
  select(b_SexFemale, b_SexMale, Sex_diff) %>% 
  ggplot(aes(x = Sex_diff, y = 0, fill = stat(x > 0))) +
  stat_halfeye(point_interval = median_qi, .width = c(.5, .95),
               slab_colour = "black", shape = 21, point_size = 3, 
               stroke = 1.5, point_fill = "white") +
  geom_vline(xintercept = 0, linetype = 2) +
  labs(x = "Difference in environmental variance between traits\n expressed in females and males",
       y = NULL) +
  scale_y_continuous(NULL, breaks = NULL) +
  scale_fill_manual(values = c("grey", met.brewer("Hokusai2")[2]), guide = "none") +
  theme_bw()

Env_var_sex_plot + Env_var_diff_plot
  
```

We detect no differences in genetic or environmental variance between the sexes.

$~$

## Testing general predictions of Fisher's theorem 

$~$

**1. Do traits correlated with fitness have lower additive genetic variance?**

Following Fisher's theorem and assuming a constant environment, in the absence of new variation, selection is expected to erode all additive genetic variation for fitness. One implication of this expectation is that traits that are correlated with fitness should show reduced additive genetic variance relative to those that are not correlated with fitness _(Robertson 1955b)_. 

There are previous studies that have compared fitness related traits i.e. those implicated with reproduction, with morphological traits, which are assumed to have lesser relationships with fitness.

- The problem is associating these traits with _lifetime) fitness. This shouldn't be such a problem for us...

- Multiple studies looking at this corollary use phenotypic correlations with fitness rather than genetic. Once again, this is no problem for us...

- Even if there is lots of additive genetic variance for a trait and it is correlated with fitness, if fitness has a high environmental variance component the strength of selection on the trait will be weak.

_Houle (1992)_ looks like an important meta-analysis type work

Note that heritability can also decline with an increase in environmental variance and no decrease (or an increase) in genetic variance. If true, this makes a simple regression with heritability and fitness misleading.

Using evolvability as his metric, _Houle (1992)_ found that characters assumed to be closely related to fitness have higher evolvabilities than do trait with looser relationships to fitness. This suggests that the negative relationship between fitness and heritability is due to increased environmental variance in fitness related traits, not a decrease in additive genetic variance. 

```{r, include=FALSE}

# relationship between Rg and Cov

  SNP_selection_response_estimates %>% 
  filter(`Variance_C(G)_tr12` > -1 & Variance_rG > -1 & `Variance_C(G)_tr12` < 1 & Variance_rG < 1) %>% 
         #`Trait guild` != "Metabolome" & `Trait guild` != "CHC", `Trait guild` != "Microbiome") %>% 
  ggplot(aes(x = `Variance_C(G)_tr12`, y = Variance_rG, fill = Fitness_sex)) +
  geom_point(shape = 21, size = 3, alpha = 0.5) +
  scale_fill_manual(values = met.brewer("VanGogh2", 2)) +
  geom_abline(intercept = 0, slope = 1, linetype = 2) +
  labs(x = "SNP covariance between fitness and trait",
       y = "Genetic correlation between fitness\nand trait (Rg)",
       fill = "Sex") +
  theme_bw() 

```



```{r}

Rz_VG_plot <-
  SNP_selection_response_estimates %>% 
  filter(`Variance_C(G)_tr12` > -1 & `Variance_V(G)_trait` > -1 & `Variance_C(G)_tr12` < 1 & `Variance_V(G)_trait` < 1) %>% 
  #`Trait guild` != "Metabolome" & `Trait guild` != "CHC", `Trait guild` != "Microbiome") %>% 
  mutate(`Variance_C(G)_tr12` = abs(`Variance_C(G)_tr12`)) %>% 
  ggplot(aes(y = `Variance_C(G)_tr12`, x = `Variance_V(G)_trait`, fill = Fitness_sex)) +
  #geom_hex()
  geom_point(shape = 21, size = 3, alpha = 0.5) +
  geom_smooth(aes(colour = Fitness_sex)) +
  scale_fill_manual(values = met.brewer("Hokusai3", 2), guide = "none") +
  scale_colour_manual(values = met.brewer("Hokusai3", 2), guide = "none") +
  labs(y = expression(paste("Selection response ", (R[z]))),
       x = expression(paste("SNP genetic variance ", (V[A]))),
       fill = "Sex") +
  theme_bw()

Rz_VG_plot
```

Figure X: The SNP-covariance between fitness and trait (Robertson covariance) is positively correlated with additive genetic variance. The Robertson covariance, or the secondary theorem of natural selection, $R_z$ estimates the expected response to selection of a trait.

Ok so I don't think we're learning much here. This is very much the expected result, given additive genetic variance is the fuel for a selection response. 

$~$

**2. Do traits correlated with fitness have higher levels of residual variance?**

```{r}
Rz_Ve_plot <-
  SNP_selection_response_estimates %>% 
  filter(`Variance_C(G)_tr12` > -1 & `Variance_V(e)_trait` > -1 & `Variance_C(G)_tr12` < 1 & `Variance_V(e)_trait` < 5) %>% 
  #`Trait guild` != "Metabolome" & `Trait guild` != "CHC", `Trait guild` != "Microbiome") %>% 
  mutate(`Variance_C(G)_tr12` = abs(`Variance_C(G)_tr12`)) %>% 
  ggplot(aes(x = `Variance_C(G)_tr12`, y = `Variance_V(e)_trait`, fill = Fitness_sex)) +
  #geom_hex()
  geom_point(shape = 21, size = 3, alpha = 0.5) +
  geom_smooth(aes(colour = Fitness_sex)) +
  scale_fill_manual(values = met.brewer("Hokusai3", 2), guide = "none") +
  scale_colour_manual(values = met.brewer("Hokusai3", 2), guide = "none") +
  labs(y = expression(paste("Selection response ", (R[z]))),
       x = expression(paste("Environmental variance ", (V[A]))),
       fill = "Sex") +
  theme_bw()

Rz_Ve_plot

```

Figure X: The SNP-covariance between fitness and trait (Robertson covariance) is negatively correlated with environmental variance underlying the trait. The Robertson covariance, or the secondary theorem of natural selection, $R_z$ estimates the expected response to selection of a trait.

$~$

## Sexually antagonistic selection

$~$

Using the Robertson covariance We have found th expected responses to selection for many traits in females and males. For those traits where we have a measure of both, we can quantify sexually antagonistic selection by finding traits that have selection operating in opposite directions, depending on which sex the trait is expressed in. 

[Innocenti and Morrow (2011)]( https://doi.org/10.1111/j.1558-5646.2010.01021.x) present an index for measuring the intensity for sexually antagonistic selection:

$I = \frac{\beta_M \beta_F}{\sqrt{(\beta_M^{2} + \beta_F^{2})/2}}$


Where $B_F$ and $B_M$ are the standardised selection gradients for females and males respectively.

From **Innocenti and Morrow**:

This index is positive when selection is concordant in the two sexes, negative when antagonistic in the two sexes, and is zero when selection is absent in one sex (note that it will miss conflict that occurs when strong stabilizing selection is present in that sex). Finally it is proportional to the absolute intensity of selection.

Additionally, it has the desirable properties of being symmetrical and normally distributed for a random set of normally distributed $B_F$, $B_M$. |I| is also always included in the interval between the absolute values of the selection gradient in the two sexes, and it coincides with them when $B_F$ = $B_M$. As a potential drawback, it should be noted that such quantity is not defined when $B_F$ = $B_M$ = 0, even though it makes little sense to estimate how concordant or antagonistic directional selection is when it is absent in both sexes.

Unfortunately, we do not have selection differentials / gradients. Instead, we have the potentially more powerful expected response to selection for many traits in both sexes.  


$~$

```{r}


# create our sexual antagonism tibble by joining the female and male tibbles by traits they both share

SNP_antagonism <- SNP_both_sexes %>% 
  rename(Trait_covariance_male_fitness = `Variance_C(G)_tr12.x`,
         Trait_covariance_female_fitness = `Variance_C(G)_tr12.y`) %>% 
  select(-c(Fitness_sex.x, Fitness_sex.y)) %>% 
  # calculate the innocenti and morrow index
  mutate(Selection_index = Trait_covariance_female_fitness * Trait_covariance_male_fitness / sqrt(((Trait_covariance_female_fitness)^2 + (Trait_covariance_male_fitness)^2)/2)) %>% 
  # get rid of traits with unrealistic covariances
  filter(Trait_covariance_male_fitness > -5,
         Trait_covariance_female_fitness < 2 & Trait_covariance_female_fitness > -5)

# plot the data

sexual_concordance_plot <-
  ggplot(data = SNP_antagonism, aes(x = Trait_covariance_female_fitness, y = Trait_covariance_male_fitness, fill = Selection_index)) +
  #geom_errorbar(aes(ymin = Trait_covariance_male_fitness - `SE_C(G)_tr12.x`, ymax = Trait_covariance_male_fitness + `SE_C(G)_tr12.x`),
   #             width = 0.01, colour = "grey11", alpha = 0.8) +
  #geom_errorbarh(aes(xmin = Trait_covariance_female_fitness - `SE_C(G)_tr12.y`, xmax = Trait_covariance_female_fitness + `SE_C(G)_tr12.y`),
   #              colour = "grey11", alpha = 0.8, height = 0.01) +
  geom_point(shape = 21, alpha = 1, size = 4, show.legend = TRUE) +
  geom_smooth(method = 'lm', color='black') +
  geom_hline(yintercept = 0, linetype = 2) +
  geom_vline(xintercept = 0, linetype = 2) +
  #coord_cartesian(xlim = c(-0.35, 0.35), ylim = c(-0.4, 0.4)) +
  scale_fill_carto_c(palette = "Geyser",
                     breaks=c(-0.25, -0.125, 0, 0.125, 0.25),
                     limits=c(-0.25,0.25)) +
  labs(x = "Female response to selection",
       y = "Male response to selection",
       fill = "Concordance\n of selection\n") +
  theme_bw() +
  theme(panel.border= element_blank(),
        axis.line=element_line(),
        text = element_text(size=14),
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.minor.x = element_blank(),
        axis.title.x = element_text(size = 14),
        axis.title.y = element_text(size = 14))

sexual_concordance_plot


```

Figure X: 
