---
title: "Running GWAS on the DGRP phenotypes"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

```{r, setup, include=FALSE}
knitr::opts_chunk$set(
  message = FALSE, results='hide'
)
```


```{r}
library(tidyverse)
library(bigsnpr) # to install:   devtools::install_github("privefl/bigsnpr")
library(glue)
plink <- file.path(getwd(), "code/plink") # I use plink v1.9
# gcta <- file.path(getwd(), "code/gcta64")
options(readr.show_col_types = FALSE)


# helper function to pass commands to the terminal
# Note that we set `intern = TRUE`, and pass the result of `system()` to `cat()`,
# ensuring that the Terminal output will be printed in this knitr report.
run_command <- function(shell_command, wd = getwd(), path = ""){
  cat(system(glue("cd ", wd, path, "\n",shell_command), intern = TRUE), sep = '\n')
}
```


## Load trait data to be used in GWAS

```{r}
# Tom to check: some line/trait combinations have multiple values. Look at this:
# read_csv("data/all.dgrp.phenos_unscaled.csv") %>% group_by(line, Trait) %>% summarise(n = n()) %>% arrange(-n)
# read_csv("data/all.dgrp.phenos_unscaled.csv") %>% filter(Trait == "starvation.resistance.yeast.poor.diet.f" & line == 306) %>% as.data.frame()

traits_for_gwas <- read_csv("data/all.dgrp.phenos_unscaled.csv") %>% 
  # filter(!(`Trait guild` %in% allowed_guilds) | is.na(`Trait guild`)) %>% 
  group_by(Trait) %>% 
  summarise(lines_measured = length(unique(line))) %>% 
  ungroup() %>% 
  filter(lines_measured > 99) %>% pull(Trait) # find traits with 100+ replicates

traits_for_gwas <- traits_for_gwas[!(grepl("fitness.late", traits_for_gwas))]

# replace any slashes in the trait name, e.g. "1/3-Methylhistidine.high.yeast.f", as this is not ok to use as a file name
traits_for_gwas <- str_replace_all(traits_for_gwas, "[/]", "_") 

line_mean_phenotypes <- read_csv("data/all.dgrp.phenos_unscaled.csv") %>%
  filter(Trait %in% traits_for_gwas) %>% 
  select(line, Trait, trait_value) %>% 
  distinct(line, Trait, .keep_all = TRUE) %>% # Remove this line once Tom resolves the duplicates
  spread(Trait, trait_value) %>% 
  mutate_at(vars(-line), ~ as.numeric(scale(.x))) %>%  # scale the traits
  mutate(line = paste("line", line, sep = ""))
```



## Perform SNP quality control and imputation

We cleaned up the DGRP's .bed/.bim/.fam files (available from the [Mackay lab website](http://dgrp2.gnets.ncsu.edu/)) as follows:

1. Remove any SNPs for which genotypes are missing for >10% of the DGRP lines. We then use the software [Beagle](https://faculty.washington.edu/browning/beagle/beagle.html) to impute the remaining missing genotypes.
2. Remove SNPs with a minor allele frequency of less than 5%

Note that in the PLINK-formatted genotype files, lines fixed for the major allele are coded as 2, and lines fixed for the minor allele as 0. This means that in the association tests we calculate, negative effect sizes mean that the minor allele is associated with lower fitness, while positive effect sizes means that the minor allele is associated with higher fitness.

```{r QC_and_imputation, results='hide'}
perform_SNP_QC_and_imputation <- function(phenotypes){
  
  beagle <- bigsnpr::download_beagle()
  
  # Use Plink to clean and subset the DGRP's SNP data as follows:
  # Only keep SNPs for which at least 90% of DGRP lines were successfully genotyped (--geno 0.1)
  # Only keep SNPs with a minor allele frequency of 0.05 or higher (--maf 0.05)
  # Finally, write the processed BIM/BED/FAM files to the data/derived directory
  run_command(glue("{plink} --bfile dgrp2",
                   " --geno 0.1 --maf 0.05 --allow-no-sex", 
                   " --make-bed --out ../derived/dgrp2_QC_all_lines"), path = "/gwas_data/input/")
  
  # Use the shell command 'sed' to remove underscores from the DGRP line names in the .fam file (e.g. 'line_120' becomes 'line120')
  # Otherwise, these underscores cause trouble when we need to convert from PLINK to vcf format (vcf format uses underscore as a separator)
  for(i in 1:2) run_command("sed -i '' 's/_//' dgrp2_QC_all_lines.fam", path = "/gwas_data/derived/")
  
  # Now impute the missing genotypes using Beagle
  # This part uses the data for the full DGRP panel of >200 lines, to infer missing genotypes as accurately as possible. 
  # This step uses a lot of memory (I set to 28MB max, and it used 26.5GB), but maybe it can also run on a less powerful computer?
  # The bigsnpr package provides a helpful wrapper for Beagle called snp_beagleImpute(): it translates to a VCF file and back again using PLINK
  snp_beagleImpute(beagle, plink, 
                   bedfile.in = "gwas_data/derived/dgrp2_QC_all_lines.bed", 
                   bedfile.out = "gwas_data/derived/dgrp2_QC_all_lines_imputed.bed",
                   ncores = 7, 
                   memory.max = 20)
  
  # assign a sex of 'female' to all the DGRP lines (Beagle removes the sex, and it seems PLINK needs individuals to have a sex)
  run_command("sed -i '' 's/	0	0	0/	0	0	2/' dgrp2_QC_all_lines_imputed.fam", path = "/gwas_data/derived/")
  
  # Re-write the .bed file, to make sure the MAF and genotyping thresholds are correctly assigned post-Beagle
  run_command(glue("{plink} --bfile dgrp2_QC_all_lines_imputed",
                   " --geno 0.1 --maf 0.05", 
                   " --make-bed --out dgrp2_QC_all_lines_imputed_correct"), path = "/gwas_data/derived/")

  # Use PLINK to get the allele IDs and calculate the MAFs across the whole DGRP, for all SNPs that survived QC
  # The file created is called data/derived/plink.frq
  run_command("{plink} --bfile dgrp2_QC_all_lines_imputed_correct --freq", path = "/gwas_data/derived")

  # Clean up:
  unlink(c("gwas_data/derived/plink.log",
           "gwas_data/derived/dgrp2_QC_all_lines_imputed.bed",
           "gwas_data/derived/dgrp2_QC_all_lines_imputed.bim",
           "gwas_data/derived/dgrp2_QC_all_lines_imputed.fam",
           "gwas_data/derived/dgrp2_QC_all_lines_imputed.log",
           "gwas_data/derived/dgrp2_QC_all_lines_imputed_correct.log"))
}

if(!file.exists("gwas_data/derived/dgrp2_QC_all_lines_imputed_correct.bed")) perform_SNP_QC_and_imputation(phenotypes = predicted_line_means)

# creates the following 3 files: dgrp2_QC_all_lines_imputed_correct.bed/bim/fam

# These are the 205 DGRP lines with genotype data. Check for incorrect line names in the phenotype data
genotyped_lines <- read.table("gwas_data/derived/dgrp2_QC_all_lines_imputed_correct.fam")[,1]
line_mean_phenotypes <- line_mean_phenotypes %>% filter(line %in% genotyped_lines)
```


## Create a reduced list of LD-pruned SNPs with PLINK

To keep the computation time and memory usage manageable, we did not analyse every SNP that passed quality control (i.e. 1,646,652 SNPs), but rather a subset of them that were approximately in linkage disequilibrium. We identified this LD-pruned set of SNPs using the PLINK arguments `--indep-pairwise 100 10 0.2`, i.e. pruning within 100kB sliding windows, sliding 10 variants along with each step, and allowing a maximum pairwise $r^2$ threshold of 0.2 between loci. With these parameters, 1420071 SNPs were removed, leaving 226581 for downstream analysis.

```{r}
# indep-pairwise arguments are: 
# 100kB window size, 
# variant count to shift the window by 10 variants at the end of each step, 
# pairwise r^2 threshold of 0.2
run_command(glue("{plink} --bfile dgrp2_QC_all_lines_imputed_correct",
                 " --indep-pairwise 100 10 0.2"), path = "/gwas_data/derived/")


run_command(glue("{plink} --bfile dgrp2_QC_all_lines_imputed_correct",
                 " --extract plink.prune.in",
                 " --make-bed --out dgrp2_QC_all_lines_LDpruned"), path = "/gwas_data/derived/")
unlink("gwas_data/derived/plink.prune.in")
unlink("gwas_data/derived/plink.prune.out")
```



## Run all the GWAS
Note: because PLINK defines the minor allele as the alt allele (so, lines fixed for the minor allele are scored as genotype: 2, and those with the major allele as genotype: 0), a _positive_ effect size in these association tests means the _minor_ allele is associated with a _higher_ value of the trait in question.

```{r}
gwas_one_trait <- function(focal_phenotype){
  
  # First make 'focal_data', a 2-column data frame with the line and the focal phenotype value
  focal_data <- line_mean_phenotypes %>% 
    select(line, !! focal_phenotype) 
  names(focal_data)[2] <- "focal_pheno"
  
  # Prune it to just the lines that have some phenotype data
  focal_data <- focal_data %>% 
    filter(!is.na(line), !is.na(focal_pheno)) 
  
  # Make a list of the lines in our sample and save as a text file for passing to PLINK
  lines_to_keep <- gsub("_", "", focal_data$line) %>% cbind(.,.)
  write.table(lines_to_keep, row.names = FALSE, col.names = FALSE, file = "gwas_data/derived/lines_to_keep.txt", quote = FALSE)
  
  # Now cull the PLINK files to just the lines that we measured, and re-apply the 
  # MAF cut-off of 0.05 for the new smaller sample of DGRP lines
  run_command(glue("{plink} --bfile dgrp2_QC_all_lines_LDpruned",
                   " --keep-allele-order", 
                   " --keep lines_to_keep.txt --geno 0.1 --maf 0.05", 
                   " --make-bed --out dgrp2_QC_focal_lines"), path = "/gwas_data/derived/")
  
  # Define a function to add our phenotype data to a .fam file, which is needed for GWAS analysis and to make sure PLINK includes these samples
  # The 'phenotypes' data frame needs to have a column called 'line'
  add_phenotypes_to_fam <- function(filepath, focal_data){
    read_delim(filepath, col_names = FALSE, delim = " ") %>% 
      select(X1, X2, X3, X4, X5) %>% # Get all the non-phenotype columns
      left_join(focal_data, 
                by = c("X1" = "line")) %>%
      write.table(file = "gwas_data/derived/dgrp2_QC_focal_lines_NEW.fam", 
                  col.names = FALSE, row.names = FALSE, 
                  quote = FALSE, sep = " ")
    unlink("gwas_data/derived/dgrp2_QC_focal_lines.fam")
    file.rename("gwas_data/derived/dgrp2_QC_focal_lines_NEW.fam", "gwas_data/derived/dgrp2_QC_focal_lines.fam")
  }
  
  add_phenotypes_to_fam("gwas_data/derived/dgrp2_QC_focal_lines.fam", focal_data)
 
  # # Write a file with the line and phenotype data called phenotype.txt, for gcta64
  # pheno_data <- focal_data %>% 
  #   mutate(line_copy = line) %>% 
  #   select(line, line_copy, focal_pheno) %>% as.matrix() 
  # pheno_data %>% 
  #   write.table(row.names = FALSE, col.names = FALSE, 
  #               file = "gwas_data/derived/phenotype.txt", quote = FALSE)

  # Run mixed-model GWAS (in practice, the relatedness is low so it just a runs a linear model)
  print(focal_phenotype)
  run_command("{plink} --bfile dgrp2_QC_focal_lines  --assoc --maf 0.05 --out gwas_results/new", 
              path = "/gwas_data/derived")

  # Rename and compress the GWAS summary stats file:
  read.table("gwas_data/derived/gwas_results/new.qassoc", header = TRUE) %>% 
    select(SNP, BETA, SE, P) %>% 
    write_tsv(glue("gwas_data/derived/gwas_results/{focal_phenotype}.tsv.gz"))
  unlink("gwas_data/derived/gwas_results/new.qassoc")
  
  # Rename the plink log file
  file.rename("gwas_data/derived/gwas_results/new.log", glue("gwas_data/derived/gwas_results/{focal_phenotype}_log.txt"))
  
}

files <- list.files("gwas_data/derived/gwas_results", pattern = "tsv.gz", full.names = TRUE)
already_run_traits <- str_split(files, "/") %>% map_chr(~ .x[4]) %>% str_remove_all(".tsv.gz")

traits_to_run_for_gwas <- traits_for_gwas[!(traits_for_gwas %in% already_run_traits)]

lapply(traits_to_run_for_gwas, gwas_one_trait)

unlink(list.files("gwas_data/derived", pattern = "focal_lines", full.names = T))
unlink(c("gwas_data/derived/lines_to_keep.txt"))
```


```{r eval=FALSE, echo=FALSE, include=FALSE}
# experiments with bivariate greml in gcta64. Seems to fail very often, and gives untrustworthy estimates. Maybe not useful for the DGRP?
# Make the genomic relatedness matrix for all the lines
run_command(glue("{gcta} --bfile dgrp2_QC_all_lines_imputed_correct",
                 " --make-grm",
                 " --out gcta_GRM"), path = "/gwas_data/derived/")

trait_guilds <- read_csv("data/all.dgrp.phenos_unscaled.csv") %>% 
  select(Trait, `Trait guild`) %>% 
  distinct() %>% 
  mutate(`Trait guild` = replace(`Trait guild`, is.na(`Trait guild`), "Other"),
         `Trait guild` = replace(`Trait guild`, `Trait guild` == "?", "Other"),
         `Trait guild` = replace(`Trait guild`, `Trait guild` == "Population", "Other"),
         `Trait guild` = replace(`Trait guild`, `Trait guild` == "Reproduction", "Other"))
guilds <- trait_guilds$`Trait guild` %>% unique()

combos <- map_df(1:length(guilds), ~ {
  t(combn(traits_for_gwas[traits_for_gwas %in% c(
    "fitness.early.f", "fitness.early.m",
    trait_guilds$Trait[trait_guilds$`Trait guild` == guilds[.x]])], 2)) %>% 
    as.data.frame() %>% 
    rename(trait1 = V1, trait2 = V2) %>% as_tibble() %>% 
    mutate(guild = guilds[.x]) %>% select(guild, trait1, trait2)
  })

combos <- combos %>% filter(guild %in% c("Other", "Behavioural", "Morphological", "Physiological", "Life history"))


do_bivar_greml <- function(trait1, trait2){
  print(paste(trait1, trait2))
  # First make 'focal_data', a 2-column data frame with the line and the focal phenotype value
  focal_data <- line_mean_phenotypes %>%
    select(line, !! trait1, !! trait2)
  names(focal_data)[2:3] <- c("focal_pheno1", "focal_pheno2")
  
  # Write a file with the line and phenotype data called phenotype.txt, for gcta64
  pheno_data <- focal_data %>%
    mutate(line_copy = line) %>%
    select(line, line_copy, focal_pheno1, focal_pheno2) %>%
    as.matrix()
  
  pheno_data %>%
    write.table(row.names = FALSE, col.names = FALSE,
                file = "gwas_data/derived/phenotype_pair.txt", quote = FALSE)
  
  # Note that the warnings come when gcta could not fit the model properly. I save these errors especially, so no need to print them.
  console_output <- suppressWarnings(capture.output(
    run_command(glue("{gcta}  --reml-bivar --reml-bivar-lrt-rg 0 --reml-maxit 10000 --grm gcta_GRM  --pheno phenotype_pair.txt  --out delete_me"), 
                path = "/gwas_data/derived/")))
  
  errors <- console_output[str_detect(console_output, "[Ee]rror")]
  if(length(errors) == 0) errors <- NA
  if(length(errors) > 1) errors <- paste(errors, sep = "; ")
  traits <- sort(c(trait1, trait2))
  
  rbind(read.delim("gwas_data/derived/delete_me.hsq", sep = "\t") , 
        data.frame(Source = "Errors", Variance = errors, SE = NA)) %>% 
    mutate(trait1 = traits[1], trait2 = traits[2]) %>% select(trait1, trait2, everything())
}
# do_bivar_greml("lifespan.25C.2020.m", "fitness.early.f")
do_bivar_greml("fitness.early.m", "fitness.late.f")

if(!file.exists("gwas_data/derived/bivariate_greml_results.tsv")){
  all_gen_corrs <- map_df(1:nrow(combos), ~ do_bivar_greml(combos$trait1[.x], combos$trait2[.x]))
  write_tsv(all_gen_corrs, "gwas_data/derived/bivariate_greml_results.tsv")
}

unlink(c("gwas_data/derived/delete_me.log", "gwas_data/derived/phenotype_pair.txt", "gwas_data/derived/delete_me.hsq"))
unlink(list.files("gwas_data/derived/", pattern = "gcta_GRM", full.names = TRUE))
```

